#对对象进行创建对象----
#IUA-BYQ-----
IUA_BYQ_count=Read10X("/mnt/chentw/zigongneimo/outs/IUA-BYQ/outs/filtered_feature_bc_matrix/")
IUA_BYQ <- CreateSeuratObject(counts = IUA_BYQ_count, project = "IUA_BYQ", min.cells = 3)
IUA_BYQ[["percent.mt"]] <- PercentageFeatureSet(IUA_BYQ, pattern = "^MT-")
VlnPlot(IUA_BYQ, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
IUA_BYQ <- subset(IUA_BYQ, subset = nFeature_RNA < 5500 & percent.mt < 20 & 
                    nFeature_RNA > 200)
IUA_BYQ <- NormalizeData(IUA_BYQ, normalization.method = "LogNormalize", 
                         scale.factor = 10000)
IUA_BYQ <- FindVariableFeatures(IUA_BYQ, selection.method = "vst", nfeatures = 2000)
VlnPlot(IUA_BYQ, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)

#IUA_BYQ进行标准化然后PCA分析
DefaultAssay(IUA_BYQ) <- "RNA"
all.genes_IUA_BYQ <- rownames(IUA_BYQ)
IUA_BYQ <- ScaleData(object = IUA_BYQ, features = all.genes_IUA_BYQ)
IUA_BYQ <- RunPCA(IUA_BYQ, features = VariableFeatures(object = IUA_BYQ))

#确定IUA_BYQ的PC维数
IUA_BYQ <- JackStraw(IUA_BYQ, num.replicate = 100) 
IUA_BYQ <- ScoreJackStraw(IUA_BYQ, dims = 1:20) 
ElbowPlot(IUA_BYQ,ndims = 30, reduction = "pca") 

#IUA_BYQ进行细胞细胞聚类分析(UMAP)
IUA_BYQ=RunUMAP(IUA_BYQ, dims = 1:30)
DimPlot(IUA_BYQ,reduction="umap",label=TRUE)
IUA_BYQ <- FindNeighbors(IUA_BYQ, dims = 1:30)
IUA_BYQ <- FindClusters(IUA_BYQ, resolution = 0.3)
IUA_BYQ <- FindClusters(IUA_BYQ, resolution = 0.6)
DimPlot(IUA_BYQ,reduction="umap",label=TRUE)


#IUA_BYQ进行细胞细胞聚类分析(tsne)
IUA_BYQ=RunTSNE(IUA_BYQ, dims = 1:30)
DimPlot(IUA_BYQ,reduction="tsne",label=TRUE)
IUA_BYQ <- FindNeighbors(IUA_BYQ, dims = 1:30)
IUA_BYQ <- FindClusters(IUA_BYQ, resolution = 0.3)
DimPlot(IUA_BYQ,reduction="tsne",label=TRUE)

#保存对象
saveRDS(IUA_BYQ,"/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/IUA_BYQ.rds")




#IUA_WJM-----
IUA_WJM_count=Read10X("/mnt/chentw/zigongneimo/outs/IUA-WJM/IUA-WJM-outs/filtered_feature_bc_matrix/")
IUA_WJM <- CreateSeuratObject(counts = IUA_WJM_count, project = "IUA_WJM", min.cells = 3)
IUA_WJM[["percent.mt"]] <- PercentageFeatureSet(IUA_WJM, pattern = "^MT-")
VlnPlot(IUA_WJM, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
IUA_WJM <- subset(IUA_WJM, subset = nFeature_RNA < 5500 &nFeature_RNA >200 &
                    percent.mt < 20)
IUA_WJM <- NormalizeData(IUA_WJM, normalization.method = "LogNormalize", 
                         scale.factor = 10000)
IUA_WJM <- FindVariableFeatures(IUA_WJM, selection.method = "vst", nfeatures = 2000)
VlnPlot(IUA_WJM, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

#IUA_WJM进行标准化然后PCA分析
DefaultAssay(IUA_WJM) <- "RNA"
all.genes_IUA_WJM <- rownames(IUA_WJM)
IUA_WJM <- ScaleData(object = IUA_WJM, features = all.genes_IUA_WJM)
IUA_WJM <- RunPCA(IUA_WJM, features = VariableFeatures(object = IUA_WJM))

#确定IUA_WJM的PC维数
IUA_WJM <- JackStraw(IUA_WJM, num.replicate = 100) 
IUA_WJM <- ScoreJackStraw(IUA_WJM, dims = 1:20) 
ElbowPlot(IUA_WJM,ndims = 30, reduction = "pca") 

#IUA_WJM进行细胞细胞聚类分析(UMAP)
IUA_WJM=RunUMAP(IUA_WJM, dims = 1:30)
DimPlot(IUA_WJM,reduction="umap",label=TRUE)
IUA_WJM <- FindNeighbors(IUA_WJM, dims = 1:30)
IUA_WJM <- FindClusters(IUA_WJM, resolution = 0.3)
DimPlot(IUA_WJM,reduction="umap",label=TRUE)


#IUA_WJM进行细胞细胞聚类分析(tsne)
IUA_WJM=RunTSNE(IUA_WJM, dims = 1:30)
DimPlot(IUA_WJM,reduction="tsne",label=TRUE)
IUA_WJM <- FindNeighbors(IUA_WJM, dims = 1:30)
IUA_WJM <- FindClusters(IUA_WJM, resolution = 0.3)
DimPlot(IUA_WJM,reduction="tsne",label=TRUE)

#保存对象
saveRDS(IUA_WJM,"/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/IUA_WJM.rds")


#IUA_GWY_additional----
IUA_GWY_additional_count=Read10X("/mnt/chentw/zigongneimo/outs/IUA-GWY_additional/outs/filtered_feature_bc_matrix/")
IUA_GWY_additional <- CreateSeuratObject(counts = IUA_GWY_additional_count, 
                                         project = "IUA_GWY_additional", min.cells = 3)
IUA_GWY_additional[["percent.mt"]] <- PercentageFeatureSet(IUA_GWY_additional, pattern = "^MT")
VlnPlot(IUA_GWY_additional, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"),
        ncol = 3,
        pt.size = 0)
IUA_GWY_additional <- subset(IUA_GWY_additional, subset = nFeature_RNA < 5500 & 
                               nFeature_RNA > 200 & percent.mt < 20)
IUA_GWY_additional <- NormalizeData(IUA_GWY_additional, normalization.method = "LogNormalize", 
                                    scale.factor = 10000)
IUA_GWY_additional <- FindVariableFeatures(IUA_GWY_additional, selection.method = "vst", 
                                           nfeatures = 2000)
VlnPlot(IUA_GWY_additional, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), 
        ncol = 3,pt.size = 0)

#IUA_GWY_additional进行标准化然后PCA分析
DefaultAssay(IUA_GWY_additional) <- "RNA"
all.genes_IUA_GWY_additional <- rownames(IUA_GWY_additional)
IUA_GWY_additional <- ScaleData(object = IUA_GWY_additional, features = all.genes_IUA_GWY_additional)
IUA_GWY_additional <- RunPCA(IUA_GWY_additional, features = VariableFeatures(object = IUA_GWY_additional))

#确定IUA_GWY_additional的PC维数
IUA_GWY_additional <- JackStraw(IUA_GWY_additional, num.replicate = 100) 
IUA_GWY_additional <- ScoreJackStraw(IUA_GWY_additional, dims = 1:20) 
ElbowPlot(IUA_GWY_additional,ndims = 30, reduction = "pca") 

#IUA_GWY_additional进行细胞细胞聚类分析(UMAP)
IUA_GWY_additional=RunUMAP(IUA_GWY_additional, dims = 1:30)
DimPlot(IUA_GWY_additional,reduction="umap",label=TRUE)
IUA_GWY_additional <- FindNeighbors(IUA_GWY_additional, dims = 1:30)
IUA_GWY_additional <- FindClusters(IUA_GWY_additional, resolution = 0.3)
DimPlot(IUA_GWY_additional,reduction="umap",label=TRUE)


#IUA_GWY_additional进行细胞细胞聚类分析(tsne)
IUA_GWY_additional=RunTSNE(IUA_GWY_additional, dims = 1:30)
DimPlot(IUA_GWY_additional,reduction="tsne",label=TRUE)
IUA_GWY_additional <- FindNeighbors(IUA_GWY_additional, dims = 1:30)
IUA_GWY_additional <- FindClusters(IUA_GWY_additional, resolution = 0.3)
DimPlot(IUA_GWY_additional,reduction="tsne",label=TRUE)

#保存对象
saveRDS(IUA_GWY_additional,"/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/IUA_GWY_additional.rds")


#H-----
library(Seurat)
H_count=Read10X("/mnt/chentw/zigongneimo/outs/H/outs/filtered_feature_bc_matrix/")
H <- CreateSeuratObject(counts = H_count, project = "H", min.cells = 3)
H[["percent.mt"]] <- PercentageFeatureSet(H, pattern = "^MT-")
VlnPlot(H, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,
        pt.size = 0)
H <- subset(H, subset = nFeature_RNA < 5500 & nFeature_RNA>200 &
              percent.mt < 20)
H <- NormalizeData(H, normalization.method = "LogNormalize", 
                   scale.factor = 10000)
H <- FindVariableFeatures(H, selection.method = "vst", nfeatures = 2000)
VlnPlot(H, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,
        pt.size = 0)

#H进行标准化然后PCA分析
DefaultAssay(H) <- "RNA"
all.genes_H <- rownames(H)
H <- ScaleData(object = H, features = all.genes_H)
H <- RunPCA(H, features = VariableFeatures(object = H))

#确定H的PC维数
H <- JackStraw(H, num.replicate = 100) 
H <- ScoreJackStraw(H, dims = 1:20) 
ElbowPlot(H,ndims = 30, reduction = "pca") 

#H进行细胞细胞聚类分析(UMAP)
H=RunUMAP(H, dims = 1:30)
DimPlot(H,reduction="umap",label=TRUE)
H <- FindNeighbors(H, dims = 1:30)
H <- FindClusters(H, resolution = 0.3)
DimPlot(H,reduction="umap",label=TRUE)


#H进行细胞细胞聚类分析(tsne)
H=RunTSNE(H, dims = 1:30)
DimPlot(H,reduction="tsne",label=TRUE)
H <- FindNeighbors(H, dims = 1:30)
H <- FindClusters(H, resolution = 0.3)
DimPlot(H,reduction="tsne",label=TRUE)
saveRDS(H,"/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/H.rds")

#li-----
library(Seurat)
li_count=Read10X("/mnt/chentw/zigongneimo/outs/li/outs/filtered_feature_bc_matrix/")
li <- CreateSeuratObject(counts = li_count, project = "li", min.cells = 3)
li[["percent.mt"]] <- PercentageFeatureSet(li, pattern = "^MT-")
VlnPlot(li, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,
        pt.size = 0)
li <- subset(li, subset = nFeature_RNA < 5500 &nFeature_RNA >200 &
               percent.mt < 20)
li <- NormalizeData(li, normalization.method = "LogNormalize", scale.factor = 10000)
li <- FindVariableFeatures(li, selection.method = "vst", nfeatures = 2000)
VlnPlot(li, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,
        pt.size = 0)

#li进行标准化然后PCA分析
DefaultAssay(li) <- "RNA"
all.genes_li <- rownames(li)
li <- ScaleData(object = li, features = all.genes_li)
li <- RunPCA(li, features = VariableFeatures(object = li))

#确定li的PC维数
li <- JackStraw(li, num.replicate = 100) 
li <- ScoreJackStraw(li, dims = 1:20) 
ElbowPlot(li,ndims = 30, reduction = "pca") 

#li进行细胞细胞聚类分析(UMAP)
li=RunUMAP(li, dims = 1:30)
DimPlot(li,reduction="umap",label=TRUE)
li <- FindNeighbors(li, dims = 1:30)
li <- FindClusters(li, resolution = 0.3)
DimPlot(li,reduction="umap",label=TRUE)


#li进行细胞细胞聚类分析(tsne)
li=RunTSNE(li, dims = 1:30)
DimPlot(li,reduction="tsne",label=TRUE)
li <- FindNeighbors(li, dims = 1:30)
li <- FindClusters(li, resolution = 0.3)
DimPlot(li,reduction="tsne",label=TRUE)
saveRDS(li,"/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/li.rds")






#qian-----
qian_count=Read10X("/mnt/chentw/zigongneimo/outs/qian/outs/filtered_feature_bc_matrix/")
qian <- CreateSeuratObject(counts = qian_count, project = "qian", min.cells = 3)
qian[["percent.mt"]] <- PercentageFeatureSet(qian, pattern = "^MT-")
VlnPlot(qian, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,
        pt.size = 0)
qian <- subset(qian, subset = nFeature_RNA < 5500 & nFeature_RNA > 200 &
                 percent.mt < 20)
qian <- NormalizeData(qian, normalization.method = "LogNormalize", 
                      scale.factor = 10000)
qian <- FindVariableFeatures(qian, selection.method = "vst", nfeatures = 2000)
VlnPlot(qian, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,
        pt.size = 0)

#qian进行标准化然后PCA分析
DefaultAssay(qian) <- "RNA"
all.genes_qian <- rownames(qian)
qian <- ScaleData(object = qian, features = all.genes_qian)
qian <- RunPCA(qian, features = VariableFeatures(object = qian))

#确定qian的PC维数
qian <- JackStraw(qian, num.replicate = 100) 
qian <- ScoreJackStraw(qian, dims = 1:20) 
ElbowPlot(qian,ndims = 30, reduction = "pca") 

#qian进行细胞细胞聚类分析(UMAP)
qian=RunUMAP(qian, dims = 1:30)
DimPlot(qian,reduction="umap",label=TRUE)
qian <- FindNeighbors(qian, dims = 1:30)
qian <- FindClusters(qian, resolution = 0.3)
DimPlot(qian,reduction="umap",label=TRUE)


#qian进行细胞细胞聚类分析(tsne)
qian=RunTSNE(qian, dims = 1:30)
DimPlot(qian,reduction="tsne",label=TRUE)
qian <- FindNeighbors(qian, dims = 1:30)
qian <- FindClusters(qian, resolution = 0.3)
DimPlot(qian,reduction="tsne",label=TRUE)
saveRDS(qian,"/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/qian.rds")






#IUAfeng-----
IUAfeng_count=Read10X("/mnt/chentw/zigongneimo/outs/IUAfeng/outs/filtered_feature_bc_matrix/")
IUAfeng <- CreateSeuratObject(counts = IUAfeng_count, project = "IUAfeng", min.cells = 3)
IUAfeng[["percent.mt"]] <- PercentageFeatureSet(IUAfeng, pattern = "^MT-")
VlnPlot(IUAfeng, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,
        pt.size=0)
IUAfeng <- subset(IUAfeng, subset = nFeature_RNA < 5500 & nFeature_RNA>200 &
                    percent.mt < 20)
IUAfeng <- NormalizeData(IUAfeng, normalization.method = "LogNormalize", 
                         scale.factor = 10000)
IUAfeng <- FindVariableFeatures(IUAfeng, selection.method = "vst", nfeatures = 2000)
VlnPlot(IUAfeng, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,
        pt.size=0)

#IUAfeng进行标准化然后PCA分析
DefaultAssay(IUAfeng) <- "RNA"
all.genes_IUAfeng <- rownames(IUAfeng)
IUAfeng <- ScaleData(object = IUAfeng, features = all.genes_IUAfeng)
IUAfeng <- RunPCA(IUAfeng, features = VariableFeatures(object = IUAfeng))

#确定IUAfeng的PC维数
IUAfeng <- JackStraw(IUAfeng, num.replicate = 100) 
IUAfeng <- ScoreJackStraw(IUAfeng, dims = 1:20) 
ElbowPlot(IUAfeng,ndims = 30, reduction = "pca") 

#IUAfeng进行细胞细胞聚类分析(UMAP)
IUAfeng=RunUMAP(IUAfeng, dims = 1:30)
DimPlot(IUAfeng,reduction="umap",label=TRUE)
IUAfeng <- FindNeighbors(IUAfeng, dims = 1:30)
IUAfeng <- FindClusters(IUAfeng, resolution = 0.3)
DimPlot(IUAfeng,reduction="umap",label=TRUE)


#IUAfeng进行细胞细胞聚类分析(tsne)
IUAfeng=RunTSNE(IUAfeng, dims = 1:30)
DimPlot(IUAfeng,reduction="tsne",label=TRUE)
IUAfeng <- FindNeighbors(IUAfeng, dims = 1:30)
IUAfeng <- FindClusters(IUAfeng, resolution = 0.3)
DimPlot(IUAfeng,reduction="tsne",label=TRUE)
saveRDS(IUAfeng,"/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/IUAfeng.rds")



#IUAshou-----
IUAshou_count=Read10X("/mnt/chentw/zigongneimo/outs/IUAshou/outs/filtered_feature_bc_matrix/")
IUAshou <- CreateSeuratObject(counts = IUAshou_count, project = "IUAshou", min.cells = 3)
IUAshou[["percent.mt"]] <- PercentageFeatureSet(IUAshou, pattern = "^MT-")
VlnPlot(IUAshou, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,
        pt.size=0)
IUAshou <- subset(IUAshou, subset = nFeature_RNA < 5500 & nFeature_RNA>200 &
                    percent.mt < 20)
IUAshou <- NormalizeData(IUAshou, normalization.method = "LogNormalize", 
                         scale.factor = 10000)
IUAshou <- FindVariableFeatures(IUAshou, selection.method = "vst", nfeatures = 2000)
VlnPlot(IUAshou, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,
        pt.size=0)

#IUAshou进行标准化然后PCA分析
DefaultAssay(IUAshou) <- "RNA"
all.genes_IUAshou <- rownames(IUAshou)
IUAshou <- ScaleData(object = IUAshou, features = all.genes_IUAshou)
IUAshou <- RunPCA(IUAshou, features = VariableFeatures(object = IUAshou))

#确定IUAshou的PC维数
IUAshou <- JackStraw(IUAshou, num.replicate = 100) 
IUAshou <- ScoreJackStraw(IUAshou, dims = 1:20) 
ElbowPlot(IUAshou,ndims = 30, reduction = "pca") 

#IUAshou进行细胞细胞聚类分析(UMAP)
IUAshou=RunUMAP(IUAshou, dims = 1:30)
DimPlot(IUAshou,reduction="umap",label=TRUE)
IUAshou <- FindNeighbors(IUAshou, dims = 1:30)
IUAshou <- FindClusters(IUAshou, resolution = 0.3)
DimPlot(IUAshou,reduction="umap",label=TRUE)


#IUAshou进行细胞细胞聚类分析(tsne)
IUAshou=RunTSNE(IUAshou, dims = 1:30)
DimPlot(IUAshou,reduction="tsne",label=TRUE)
IUAshou <- FindNeighbors(IUAshou, dims = 1:30)
IUAshou <- FindClusters(IUAshou, resolution = 0.3)
DimPlot(IUAshou,reduction="tsne",label=TRUE)
saveRDS(IUAshou,"/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/IUAshou.rds")








#IUAxia-----
IUAxia_count=Read10X("/mnt/chentw/zigongneimo/outs/IUAxia/outs/filtered_feature_bc_matrix/")
IUAxia <- CreateSeuratObject(counts = IUAxia_count, project = "IUAxia", min.cells = 3)
IUAxia[["percent.mt"]] <- PercentageFeatureSet(IUAxia, pattern = "^MT-")
VlnPlot(IUAxia, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,
        pt.size=0)
IUAxia <- subset(IUAxia, subset = nFeature_RNA < 5500 & nFeature_RNA >200 &
                   percent.mt < 20)
IUAxia <- NormalizeData(IUAxia, normalization.method = "LogNormalize", 
                        scale.factor = 10000)
IUAxia <- FindVariableFeatures(IUAxia, selection.method = "vst", nfeatures = 2000)
VlnPlot(IUAxia, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,
        pt.size=0)

#IUAxia进行标准化然后PCA分析
DefaultAssay(IUAxia) <- "RNA"
all.genes_IUAxia <- rownames(IUAxia)
IUAxia <- ScaleData(object = IUAxia, features = all.genes_IUAxia)
IUAxia <- RunPCA(IUAxia, features = VariableFeatures(object = IUAxia))

#确定IUAxia的PC维数
IUAxia <- JackStraw(IUAxia, num.replicate = 100) 
IUAxia <- ScoreJackStraw(IUAxia, dims = 1:20) 
ElbowPlot(IUAxia,ndims = 30, reduction = "pca") 

#IUAxia进行细胞细胞聚类分析(UMAP)
IUAxia=RunUMAP(IUAxia, dims = 1:30)
DimPlot(IUAxia,reduction="umap",label=TRUE)
IUAxia <- FindNeighbors(IUAxia, dims = 1:30)
IUAxia <- FindClusters(IUAxia, resolution = 0.3)
DimPlot(IUAxia,reduction="umap",label=TRUE)


#IUAxia进行细胞细胞聚类分析(tsne)
IUAxia=RunTSNE(IUAxia, dims = 1:30)
DimPlot(IUAxia,reduction="tsne",label=TRUE)
IUAxia <- FindNeighbors(IUAxia, dims = 1:30)
IUAxia <- FindClusters(IUAxia, resolution = 0.3)
DimPlot(IUAxia,reduction="tsne",label=TRUE)
saveRDS(IUAxia,"/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/IUAxia.rds")









#进行去除双细胞以后进行整合----
CTRL1=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/H.rds")
CTRL2=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/qian.rds")

POLYP1=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/yang.rds")
POLYP2=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/Z.rds")
POLYP3=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/POLYP_GLR_additional.rds")
POLYP4=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/POLYP_ZJX.rds")

IUA1=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/IUAshou.rds")
IUA2=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/IUAxia.rds")
IUA3=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/li.rds")
IUA4=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/IUAfeng.rds")
IUA5=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/IUA_GWY_additional.rds")
IUA6=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/IUA_BYQ.rds")
IUA7=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/IUA_WJM.rds")


#Ctrl和IUA进行合并
##给样本加上注释信息
###type1
CTRL1$type1="CTRL"
CTRL2$type1="CTRL"
POLYP1$type1="POLYP"
POLYP2$type1="POLYP"
POLYP3$type1="POLYP"
POLYP4$type1="POLYP"
IUA1$type1="IUA"
IUA2$type1="IUA"
IUA3$type1="IUA"
IUA4$type1="IUA"
IUA5$type1="IUA"
IUA6$type1="IUA"
IUA7$type1="IUA"

###type2
CTRL1$type2="CTRL"
CTRL2$type2="CTRL"
POLYP1$type2="POLYP1"
POLYP2$type2="POLYP1"
POLYP3$type2="POLYP2"
POLYP4$type2="POLYP2"
IUA1$type2="IUA1"
IUA2$type2="IUA1"
IUA3$type2="IUA1"
IUA4$type2="IUA1"
IUA5$type2="IUA2"
IUA6$type2="IUA2"
IUA7$type2="IUA2"

###type3
CTRL1$type3="CTRL1"
CTRL2$type3="CTRL2"
POLYP1$type3="POLYP1"
POLYP2$type3="POLYP2"
POLYP3$type3="POLYP3"
POLYP4$type3="POLYP4"
IUA1$type3="IUA1"
IUA2$type3="IUA2"
IUA3$type3="IUA3"
IUA4$type3="IUA4"
IUA5$type3="IUA5"
IUA6$type3="IUA6"
IUA7$type3="IUA7"




#数据进行合并(intergrad)
umap_cltrl_IUA=FindIntegrationAnchors(object.list = list(CTRL1,CTRL2,IUA1,IUA2,IUA3,IUA4,IUA5,IUA6,IUA7),dims=1:30)
umap_cltrl_IUA=IntegrateData(anchorset=umap_cltrl_IUA,dims=1:30)

#进行标准化然后PCA分析
DefaultAssay(umap_cltrl_IUA) <- "RNA"
all.genes <- rownames(umap_cltrl_IUA)
umap_cltrl_IUA <- ScaleData(object = umap_cltrl_IUA, features = all.genes)

DefaultAssay(umap_cltrl_IUA) <- "integrated"
all.genes1 <- rownames(umap_cltrl_IUA)
umap_cltrl_IUA <- ScaleData(object = umap_cltrl_IUA, features = all.genes1)
umap_cltrl_IUA <- RunPCA(umap_cltrl_IUA, features = VariableFeatures(object = umap_cltrl_IUA))

#确定PC维数(整合型样本可以不做这一步)
umap_cltrl_IUA <- JackStraw(umap_cltrl_IUA, num.replicate = 100) 
umap_cltrl_IUA <- ScoreJackStraw(umap_cltrl_IUA, dims = 1:20) 
ElbowPlot(umap_cltrl_IUA,ndims = 30, reduction = "pca") 

#进行细胞细胞聚类分析(umap_cltrl_IUA)
umap_cltrl_IUA=RunUMAP(umap_cltrl_IUA, dims = 1:30)
DimPlot(umap_cltrl_IUA,reduction="umap",label=TRUE)
umap_cltrl_IUA <- FindNeighbors(umap_cltrl_IUA, dims = 1:30)
umap_cltrl_IUA <- FindClusters(umap_cltrl_IUA, resolution = 0.3)
umap_cltrl_IUA <- FindClusters(umap_cltrl_IUA, resolution = 0.6)
umap_cltrl_IUA <- FindClusters(umap_cltrl_IUA, resolution = 0.9)
DefaultAssay(umap_cltrl_IUA)<-"RNA"
DimPlot(umap_cltrl_IUA,label=TRUE)



#umap_cltrl_IUA进行细胞细胞聚类分析(tsne)
umap_cltrl_IUA=RunTSNE(umap_cltrl_IUA, dims = 1:30)
DimPlot(umap_cltrl_IUA,reduction="tsne",label=TRUE)
umap_cltrl_IUA<- FindNeighbors(umap_cltrl_IUA, dims = 1:30)
umap_cltrl_IUA <- FindClusters(umap_cltrl_IUA, resolution = 0.3)
umap_cltrl_IUA <- FindClusters(umap_cltrl_IUA, resolution = 0.6)
umap_cltrl_IUA <- FindClusters(umap_cltrl_IUA, resolution = 0.9)
DimPlot(umap_cltrl_IUA,reduction="tsne",label=TRUE)
saveRDS(umap_cltrl_IUA,"/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_cltrl_IUA.rds")


#进行后续的简单绘图
umap_cltrl_IUA=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_cltrl_IUA.rds")
table(umap_cltrl_IUA$Double)

table(umap_cltrl_IUA$type3,umap_cltrl_IUA$Double)

table(umap_cltrl_IUA$type3,umap_cltrl_IUA$Double.1)

table(umap_cltrl_IUA$integrated_snn_res.0.3)

umap_cltrl_IUA$integrated_snn_res.0.3=factor(umap_cltrl_IUA$integrated_snn_res.0.3,levels = 0:19)

DimPlot(umap_cltrl_IUA,group.by = "integrated_snn_res.0.3",label=T)

DimPlot(umap_cltrl_IUA,group.by = "Double",label=F)+ggtitle("")

ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/P20221130_2.jpeg",
       width = 6,height = 4,dpi = 900)

DimPlot(umap_cltrl_IUA,group.by = "Double",label=F)+ggtitle("")+
  theme(legend.position = "none",panel.grid.major = element_blank(),panel.grid.minor = element_blank(),
        panel.border = element_blank(),axis.title = element_blank(),axis.text = element_blank(),
        axis.ticks = element_blank(),axis.line = element_blank())

ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/P20221130_2.jpeg",
       width = 6,height = 4,dpi = 900)

#去除双细胞并进行maker基因查找
umap_cltrl_IUA_allsingle=subset(umap_cltrl_IUA,Double%in%"Singlet")

#进行标准化然后PCA分析
DefaultAssay(umap_cltrl_IUA_allsingle) <- "RNA"
all.genes <- rownames(umap_cltrl_IUA_allsingle)
umap_cltrl_IUA_allsingle <- ScaleData(object = umap_cltrl_IUA_allsingle, features = all.genes)

DefaultAssay(umap_cltrl_IUA_allsingle) <- "integrated"
all.genes1 <- rownames(umap_cltrl_IUA_allsingle)
umap_cltrl_IUA_allsingle <- ScaleData(object = umap_cltrl_IUA_allsingle, features = all.genes1)



DimPlot(umap_cltrl_IUA_allsingle,group.by = "integrated_snn_res.0.3",label=T)


DefaultAssay(umap_cltrl_IUA_allsingle)<-"RNA"
umap_cltrl_IUA_allsingle@active.ident<-umap_cltrl_IUA_allsingle$integrated_snn_res.0.3
umap_cltrl_IUA_allsingle_cluster_markers <- FindAllMarkers(object = umap_cltrl_IUA_allsingle, only.pos = TRUE,min.pct = 0.25, 
                                                           logfc.threshold = 0.25)


umap_cltrl_IUA_allsingle_cluster_markers=subset(umap_cltrl_IUA_allsingle_cluster_markers,umap_cltrl_IUA_allsingle_cluster_markers$p_val < 0.01)

write.csv(umap_cltrl_IUA_allsingle_cluster_markers,"/data/DATAprocess/chentw/zigongneimo/umap_cltrl_IUA_allsingle/umap_cltrl_IUA_allsingle_cluster_markers.csv")




#对9个样本进行python处理后的进一步处理尤其是双细胞去除步骤------
library(SeuratDisk)
library(patchwork)
library(Seurat)
library(dplyr)
library(ggplot2)

CTRL1=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/H.rds")
CTRL2=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/qian.rds")

IUA1=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/IUAshou.rds")
IUA2=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/IUAxia.rds")
IUA3=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/li.rds")
IUA4=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/IUAfeng.rds")
IUA5=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/IUA_GWY_additional.rds")
IUA6=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/IUA_BYQ.rds")
IUA7=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/IUA_WJM.rds")


#Ctrl和IUA进行合并
##给样本加上注释信息
###type1
CTRL1$type1="CTRL"
CTRL2$type1="CTRL"
IUA1$type1="IUA"
IUA2$type1="IUA"
IUA3$type1="IUA"
IUA4$type1="IUA"
IUA5$type1="IUA"
IUA6$type1="IUA"
IUA7$type1="IUA"

###type2
CTRL1$type2="CTRL"
CTRL2$type2="CTRL"
IUA1$type2="IUA1"
IUA2$type2="IUA1"
IUA3$type2="IUA1"
IUA4$type2="IUA1"
IUA5$type2="IUA2"
IUA6$type2="IUA2"
IUA7$type2="IUA2"

###type3
CTRL1$type3="CTRL1"
CTRL2$type3="CTRL2"
IUA1$type3="IUA1"
IUA2$type3="IUA2"
IUA3$type3="IUA3"
IUA4$type3="IUA4"
IUA5$type3="IUA5"
IUA6$type3="IUA6"
IUA7$type3="IUA7"

#读取双细胞信息
CTRL1_double=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/double/H_doublet.txt")
CTRL2_double=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/double/qian_doublet.txt")
IUA1_double=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/double/IUAshou_doublet.txt")
IUA2_double=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/double/IUAxia_doublet.txt")
IUA3_double=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/double/li_doublet.txt")
IUA4_double=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/double/IUAfeng_doublet.txt")
IUA5_double=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/double/IUAGWY_doublet.txt")
IUA6_double=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/double/IUABYQ_doublet.txt")
IUA7_double=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/double/IUAWJM_doublet.txt")

#根据推测的双细胞数目将一定值以上的score设置为双细胞
quantile(CTRL1_double$doublet_scores,0.856) #0.3125
CTRL1_double$predicted_doublets="False"
CTRL1_double$predicted_doublets[CTRL1_double$doublet_scores>=0.3125]="True"

quantile(CTRL2_double$doublet_scores,0.888)
CTRL2_double$predicted_doublets="False"
CTRL2_double$predicted_doublets[CTRL2_double$doublet_scores>=0.2537815]="True"

quantile(IUA1_double$doublet_scores,0.824)
IUA1_double$predicted_doublets="False"
IUA1_double$predicted_doublets[IUA1_double$doublet_scores>=0.2982346]="True"

quantile(IUA2_double$doublet_scores,0.808)
IUA2_double$predicted_doublets="False"
IUA2_double$predicted_doublets[IUA2_double$doublet_scores>=0.3602122]="True"

quantile(IUA3_double$doublet_scores,0.912)
IUA3_double$predicted_doublets="False"
IUA3_double$predicted_doublets[IUA3_double$doublet_scores>=0.1908303 ]="True"

quantile(IUA4_double$doublet_scores,0.94)
IUA4_double$predicted_doublets="False"
IUA4_double$predicted_doublets[IUA4_double$doublet_scores>=0.1504745 ]="True"

quantile(IUA5_double$doublet_scores,0.776)
IUA5_double$predicted_doublets="False"
IUA5_double$predicted_doublets[IUA5_double$doublet_scores>=0.337931 ]="True"

quantile(IUA6_double$doublet_scores,0.92)
IUA6_double$predicted_doublets="False"
IUA6_double$predicted_doublets[IUA6_double$doublet_scores>=0.2044025 ]="True"

quantile(IUA7_double$doublet_scores,0.976)
IUA7_double$predicted_doublets="False"
IUA7_double$predicted_doublets[IUA7_double$doublet_scores>=0.06270879 ]="True"




CTRL1_single=subset(CTRL1_double,CTRL1_double$predicted_doublets%in%"False")$barcode
CTRL2_single=subset(CTRL2_double,CTRL2_double$predicted_doublets%in%"False")$barcode
IUA1_single=subset(IUA1_double,IUA1_double$predicted_doublets%in%"False")$barcode
IUA2_single=subset(IUA2_double,IUA2_double$predicted_doublets%in%"False")$barcode
IUA3_single=subset(IUA3_double,IUA3_double$predicted_doublets%in%"False")$barcode
IUA4_single=subset(IUA4_double,IUA4_double$predicted_doublets%in%"False")$barcode
IUA5_single=subset(IUA5_double,IUA5_double$predicted_doublets%in%"False")$barcode
IUA6_single=subset(IUA6_double,IUA6_double$predicted_doublets%in%"False")$barcode
IUA7_single=subset(IUA7_double,IUA7_double$predicted_doublets%in%"False")$barcode

CTRL1_single=CTRL1[,CTRL1_single]
CTRL2_single=CTRL2[,CTRL2_single]
IUA1_single=IUA1[,IUA1_single]
IUA2_single=IUA2[,IUA2_single]
IUA3_single=IUA3[,IUA3_single]
IUA4_single=IUA4[,IUA4_single]
IUA5_single=IUA5[,IUA5_single]
IUA6_single=IUA6[,IUA6_single]
IUA7_single=IUA7[,IUA7_single]

table(DOUBLE_all$predicted_doublets)



#开始整合
harmony_all=merge(CTRL1_single,y = c(CTRL2_single,IUA1_single,IUA2_single,IUA3_single,
                                     IUA4_single,IUA5_single,IUA6_single,IUA7_single))

mtGenes=grep('^MT-',rownames(harmony_all),value=TRUE)
RPGenes=grep('^RP',rownames(harmony_all),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
harmony_alless_gene=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
                      "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene=c(mtGenes,RPGenes,HB.genes,g2m.genes,harmony_alless_gene,s.genes)





###尝试harmony
harmony_all <- NormalizeData(harmony_all, normalization.method = "LogNormalize", 
                             scale.factor = 10000)
harmony_all <- FindVariableFeatures(harmony_all, selection.method = "vst", 
                                    nfeatures = 2000)

harmony_all@assays$RNA@var.features = setdiff(harmony_all@assays$RNA@var.features,subgene)  #去除subgene影响



#harmony_all进行标准化然后PCA分析
DefaultAssay(harmony_all) <- "RNA"
all.genes_harmony_all <- rownames(harmony_all)
harmony_all <- ScaleData(object = harmony_all, features = all.genes_harmony_all)
harmony_all <- RunPCA(harmony_all, features = VariableFeatures(object = harmony_all))

#确定harmony_all的PC维数
harmony_all <- JackStraw(harmony_all, num.replicate = 50) 
harmony_all <- ScoreJackStraw(harmony_all, dims = 1:20) 
ElbowPlot(harmony_all,ndims = 30, reduction = "pca") 


#harmony_all进行细胞细胞聚类分析(UMAP)
harmony_all=RunUMAP(harmony_all, dims = 1:30)
DimPlot(harmony_all,reduction="umap",label=TRUE)
harmony_all <- FindNeighbors(harmony_all, dims = 1:30)
harmony_all <- FindClusters(harmony_all, resolution = 0.15)
harmony_all <- FindClusters(harmony_all, resolution = 0.3)
harmony_all <- FindClusters(harmony_all, resolution = 0.45)
harmony_all <- FindClusters(harmony_all, resolution = 0.6)

DimPlot(harmony_all,reduction="umap",label=TRUE)
DimPlot(harmony_all,reduction="umap",label=TRUE,group.by = "RNA_snn_res.0.15")
DimPlot(harmony_all,reduction="umap",label=TRUE,group.by = "RNA_snn_res.0.3")

harmony_all$type3=umap_WEN2_harmony_all_sub$type3

DimPlot(harmony_all,reduction="umap",label=TRUE,group.by = "type3")


saveRDS(harmony_all,"/data/DATAprocess/chentw/zigongneimo/原图绘制/rds/harmony_all_harmony.rds")

harmony_all=readRDS("/data/DATAprocess/chentw/zigongneimo/原图绘制/rds/harmony_all_harmony.rds")

DimPlot(harmony_all,reduction="umap",label=TRUE)


harmony_all=harmony_all%>%RunHarmony("type3",plot_covergence=T)

harmony_all=RunUMAP(harmony_all,reduction="harmony",dims=1:30)
harmony_all=FindNeighbors(harmony_all,dims=1:30,reduction = "harmony")
harmony_all=FindClusters(harmony_all, resolution = 0.15,reduction = "harmony")
harmony_all=FindClusters(harmony_all, resolution = 0.3,reduction = "harmony")
harmony_all=FindClusters(harmony_all, resolution = 0.6,reduction = "harmony")

harmony_all$RNA_snn_res.0.6=factor(harmony_all$RNA_snn_res.0.6,levels = 0:23)
harmony_all$RNA_snn_res.0.3=factor(harmony_all$RNA_snn_res.0.3,levels = 0:16)
DimPlot(harmony_all,reduction="umap",label=TRUE,group.by = "RNA_snn_res.0.3")


DimPlot(harmony_all,reduction="umap",label=TRUE,group.by = "type3")




###查看一些基因的表达分布

STR_maker=c("LUM","COL1A1","DCN","MMP11")
FeaturePlot(harmony_all,features = STR_maker)
ggsave("/data/DATAprocess/chentw/zigongneimo/python运行/harmony_all/细胞类型定义/STR.png",width = 8,height = 8)

EPI_maker=c("EPCAM","KRT18","MMP7","WFDC2")
FeaturePlot(harmony_all,features = EPI_maker)
ggsave("/data/DATAprocess/chentw/zigongneimo/python运行/harmony_all/细胞类型定义/EPI.png",width = 8,height = 8)


Endothelium_maker=c("VWF","CDH5")
FeaturePlot(harmony_all,features = Endothelium_maker)
ggsave("/data/DATAprocess/chentw/zigongneimo/python运行/harmony_all/细胞类型定义/Endothelium.png",width = 8,height = 4)

smooth_maker=c("MCAM","RGS5","ACTA2")
FeaturePlot(harmony_all,features = smooth_maker)
ggsave("/data/DATAprocess/chentw/zigongneimo/python运行/harmony_all/细胞类型定义/smooth_muscle.png",width = 8,height = 8)

T_maker=c("CD3D","CD3E","CXCR4","PTPRC","GZMK","CD4","CD8A")
FeaturePlot(harmony_all,features = T_maker,max.cutoff = 5)
ggsave("/data/DATAprocess/chentw/zigongneimo/python运行/harmony_all/细胞类型定义/T-cell.png",width = 12,height = 12)

NK_maker=c("NCAM1","GNLY","GZMA")
FeaturePlot(harmony_all,features = NK_maker)
ggsave("/data/DATAprocess/chentw/zigongneimo/python运行/harmony_all/细胞类型定义/NK.png",width = 8,height = 8)


MAST_maker=c("MS4A2","TPSB2","TPSAB1","CPA3")
FeaturePlot(harmony_all,features =MAST_maker)
ggsave("/data/DATAprocess/chentw/zigongneimo/python运行/harmony_all/细胞类型定义/Mast.png",width = 8,height = 8)

B_maker=c("CD79A","CD74")
FeaturePlot(harmony_all,features =B_maker)
ggsave("/data/DATAprocess/chentw/zigongneimo/python运行/harmony_all/细胞类型定义/B_cell.png",width = 8,height = 4)


macr_maker=c("HLA-DRB1","CSF1R","CD14")
FeaturePlot(harmony_all,features =macr_maker)
ggsave("/data/DATAprocess/chentw/zigongneimo/python运行/harmony_all/细胞类型定义/macrophage.png",width = 8,height = 8)


Epithelium=c("EPCAM","WFDC2","KRT18","MMP7","CLDN4")
Endothelium=c("VWF","PECAM1","CDH5","CD34","CLDN5")
Immune=c("PTPRC","CD14","CD79A","NKG7","CD3E","MS4A2")
STR=c("LUM","DCN","COL1A1","MMP11","PDGFRB")


FeaturePlot(harmony_all,features =STR)
FeaturePlot(harmony_all,features =Epithelium)
FeaturePlot(harmony_all,features =Endothelium)
FeaturePlot(harmony_all,features =Immune)



#对细胞类型进行赋值
harmony_all$cell_type1221="1221"
harmony_all$cell_type1221[harmony_all$RNA_snn_res.0.3%in%c("0","2","5","7","16","10")]<-"Stromal cells"

harmony_all$cell_type1221[harmony_all$RNA_snn_res.0.3%in%c("9")]<-"Endothelium cells"

harmony_all$cell_type1221[harmony_all$RNA_snn_res.0.3%in%c("1","6","11","3","12","13","14","15")]<-"Immune cells"

harmony_all$cell_type1221[harmony_all$RNA_snn_res.0.3%in%c("4","8")]<-"Epithelium cells"


DimPlot(harmony_all,group.by = "cell_type1221",label = T)


#根据空间位置取出基质细胞并进行重新整合
#根据umap空间位置取出细胞
library(dplyr)
harmony_all=readRDS("/data/DATAprocess/chentw/zigongneimo/原图绘制/rds/harmony_all_harmony.rds")

DimPlot(harmony_all,group.by = "cell_type1221")
harmony_all$cell_type1221="1221"
harmony_all$cell_type1221[harmony_all$RNA_snn_res.0.3%in%c("0","2","5","7","16","10")]<-"Stromal cells"

harmony_all$cell_type1221[harmony_all$RNA_snn_res.0.3%in%c("9")]<-"Endothelium cells"

harmony_all$cell_type1221[harmony_all$RNA_snn_res.0.3%in%c("1","6","11","3","12","13","14","15")]<-"Immune cells"

harmony_all$cell_type1221[harmony_all$RNA_snn_res.0.3%in%c("4","8")]<-"Epithelium cells"



harmony_all_embeddings=harmony_all$umap@cell.embeddings
harmony_all_embeddings=as.data.frame(harmony_all_embeddings)
harmony_all_embeddings$cell_type="cell_type1221"
harmony_all_embeddings$cell_type=harmony_all$cell_type1221
harmony_all_embeddings$UMAP_1=as.numeric(as.character(harmony_all_embeddings$UMAP_1))
harmony_all_embeddings$UMAP_2=as.numeric(as.character(harmony_all_embeddings$UMAP_2))
harmony_all$UMAP_1=harmony_all_embeddings$UMAP_1
harmony_all$UMAP_2=harmony_all_embeddings$UMAP_2

harmony_all_STR=subset(harmony_all,cell_type1221%in%"Stromal cells")

DimPlot(harmony_all_STR,group.by = "cell_type1221")

harmony_all_STR=subset(harmony_all_STR,UMAP_1<5&UMAP_2<8&UMAP_2>-7)

DimPlot(harmony_all_STR,group.by = "cell_type1221")



#开始重新整合基质细胞
STR.list <- SplitObject(harmony_all_STR, split.by = "type3")

STR=merge(STR.list[[1]],y = c(STR.list[[2]],STR.list[[3]],STR.list[[4]],
                              STR.list[[5]],
                              STR.list[[6]],STR.list[[7]],STR.list[[8]],
                              STR.list[[9]]))

mtGenes=grep('^MT-',rownames(STR),value=TRUE)
RPGenes=grep('^RP',rownames(STR),value=TRUE)
HSPGenes=grep('^HSP',rownames(STR),value=TRUE)
HBGenes=grep('^HB',rownames(STR),value=TRUE)

HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")

s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes

stress_gene=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
              "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene=c(mtGenes,RPGenes,HB.genes,g2m.genes,stress_gene,s.genes)


###尝试harmony

STR <- NormalizeData(STR, normalization.method = "LogNormalize", 
                     scale.factor = 10000)
STR <- FindVariableFeatures(STR, selection.method = "vst", 
                            nfeatures = 2000)
STR@assays$RNA@var.features = setdiff(STR@assays$RNA@var.features,subgene)  #去除subgene影响

L=STR@assays$RNA@var.features

#STR进行标准化然后PCA分析
DefaultAssay(STR) <- "RNA"
all.genes_STR <- rownames(STR)
STR <- ScaleData(object = STR, features = all.genes_STR)
STR <- ScaleData(STR, vars.to.regress = c("S.Score", "G2M.Score"))


STR <- RunPCA(STR, features = VariableFeatures(object = STR))


#STR进行细胞细胞聚类分析(UMAP)
STR=RunUMAP(STR, dims = 1:30)
DimPlot(STR,reduction="umap",label=TRUE)
STR <- FindNeighbors(STR, dims = 1:30)
STR <- FindClusters(STR, resolution = 0.15)
STR <- FindClusters(STR, resolution = 0.3)
STR <- FindClusters(STR, resolution = 0.45)
STR <- FindClusters(STR, resolution = 0.6)

DimPlot(STR,reduction="umap",label=TRUE)
DimPlot(STR,reduction="umap",label=TRUE,group.by = "RNA_snn_res.0.15")
DimPlot(STR,reduction="umap",label=TRUE,group.by = "RNA_snn_res.0.3")


DimPlot(STR,reduction="umap",label=TRUE,group.by = "type3")



STR=STR%>%RunHarmony("type3",plot_covergence=T)

DefaultAssay(STR) <- "RNA"
all.genes_STR <- rownames(STR)
STR <- ScaleData(object = STR, features = all.genes_STR)
STR <- ScaleData(STR, vars.to.regress = c("S.Score", "G2M.Score"))


STR <- RunPCA(STR, features = VariableFeatures(object = STR))


STR=RunUMAP(STR,reduction="harmony",dims=1:30)
STR=FindNeighbors(STR,dims=1:30,reduction = "harmony")
STR=FindClusters(STR, resolution = 0.15,reduction = "harmony")
STR=FindClusters(STR, resolution = 0.3,reduction = "harmony")
STR=FindClusters(STR, resolution = 0.6,reduction = "harmony")

DimPlot(STR,reduction="umap",label=TRUE,group.by = "RNA_snn_res.0.3")


DimPlot(STR,reduction="umap",label=TRUE,group.by = "type3")


DimPlot(STR,reduction="umap",label=TRUE,group.by = "RNA_snn_res.0.6")


saveRDS(STR,"/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/STR_harmony.rds")

STR=readRDS("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/STR_harmony.rds")



##周期性分析绘图
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/STR_harmony.rds")
DimPlot(STR,reduction="umap",label=TRUE,group.by = "type3")


s.genes <- cc.genes$s.genes
g2m.genes <- cc.genes$g2m.genes
STR <- CellCycleScoring(STR, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)

STR@active.ident=STR$integrated_snn_res.0.3
DimPlot(STR)
DimPlot(STR,group.by = "Phase")+ggtitle("")

#去除细胞周期的影响
STR <- ScaleData(STR, vars.to.regress = c("S.Score", "G2M.Score"))


STR <- RunPCA(STR, features = VariableFeatures(object = STR))

#STR进行细胞细胞聚类分析(UMAP)
STR=RunUMAP(STR, dims = 1:30)
DimPlot(STR,reduction="umap",label=TRUE)
STR <- FindNeighbors(STR, dims = 1:30)
STR <- FindClusters(STR, resolution = 0.15)
STR <- FindClusters(STR, resolution = 0.3)
STR <- FindClusters(STR, resolution = 0.45)
STR <- FindClusters(STR, resolution = 0.6)

DimPlot(STR,reduction="umap",label=TRUE)
DimPlot(STR,reduction="umap",label=TRUE,group.by = "RNA_snn_res.0.15")
DimPlot(STR,reduction="umap",label=TRUE,group.by = "RNA_snn_res.0.3")


DimPlot(STR,reduction="umap",label=TRUE,group.by = "type3")

saveRDS(STR,"/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/STR_harmony2.rds")


STR=STR%>%RunHarmony("type3",plot_covergence=T)



STR=RunUMAP(STR,reduction="harmony",dims=1:30)
STR=FindNeighbors(STR,dims=1:30,reduction = "harmony")
STR=FindClusters(STR, resolution = 0.15,reduction = "harmony")
STR=FindClusters(STR, resolution = 0.3,reduction = "harmony")
STR=FindClusters(STR, resolution = 0.6,reduction = "harmony")

DimPlot(STR,reduction="umap",label=TRUE,group.by = "RNA_snn_res.0.3")


DimPlot(STR,reduction="umap",label=TRUE,group.by = "type3")

saveRDS(STR,"/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/STR_harmony3.rds")





DimPlot(STR,reduction="umap",label=TRUE,group.by = "type3")


s.genes <- cc.genes$s.genes
g2m.genes <- cc.genes$g2m.genes
STR <- CellCycleScoring(STR, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)

STR@active.ident=STR$integrated_snn_res.0.3
DimPlot(STR)
DimPlot(STR,group.by = "Phase")+ggtitle("")


DimPlot(STR,reduction="umap",label=TRUE,group.by = "RNA_snn_res.0.6")

DimPlot(STR,reduction="pca",label=TRUE,group.by = "Phase")

STR$RNA_snn_res.0.6=factor(STR$RNA_snn_res.0.6,levels =0:16 )



#对各个cluster进行marker基因寻找
STR@active.ident=STR$RNA_snn_res.0.6
STR_0.6_markers <- FindAllMarkers(object = STR, only.pos = TRUE,min.pct = 0.25, 
                                  logfc.threshold = 0.25)

STR_0.6_markers=subset(STR_0.6_markers,STR_0.6_markers$p_val_adj<0.01)

STR_0.6_markers_top5 <- STR_0.6_markers %>% group_by(cluster) %>% top_n(n = 5, wt = avg_log2FC)

DefaultAssay(STR_0.6_markers)<-"RNA"
STR_0.6_markers_top5=unique(STR_0.6_markers_top5$gene)

DotPlot(STR,features=STR_0.6_markers_top5,group.by = "RNA_snn_res.0.6")+
  theme(axis.text.x= element_text(size=9,angle = 45,vjust = 1,hjust = 1,face= "bold"))+theme(axis.text.x=element_text(size=11),axis.text.y=element_text(size=22),axis.title = element_text(size=22))+xlab("")+ylab("Cluster")+
  theme(legend.key.height=unit(2,"line"),legend.key.width=unit(2,"line"))        

#对marker基因进行富集分析
GO_STR=function(X){
  N=as.character(X)
  N=subset(STR_0.6_markers,STR_0.6_markers$cluster%in%N)
  N=N$gene
  G <- bitr(N, fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
  G=G$ENTREZID
  ego <- enrichGO(OrgDb="org.Hs.eg.db", gene = G, ont = "ALL",readable= TRUE)
  file1=paste("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/STR_harmony2/res0.6/",as.character(X),"_ego.csv",sep = "")
  print(file1)
  write.csv(ego,file1)
}
for (i in 0:16) {GO_STR(i)}

#挑前3富集通路进行绘图-气泡图

ego0=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/STR_harmony2/res0.6/0_ego.csv")
ego1=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/STR_harmony2/res0.6/1_ego.csv")
ego2=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/STR_harmony2/res0.6/2_ego.csv")
ego3=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/STR_harmony2/res0.6/3_ego.csv")
ego4=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/STR_harmony2/res0.6/4_ego.csv")
ego5=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/STR_harmony2/res0.6/5_ego.csv")
ego6=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/STR_harmony2/res0.6/6_ego.csv")
ego7=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/STR_harmony2/res0.6/7_ego.csv")
ego8=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/STR_harmony2/res0.6/8_ego.csv")
ego9=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/STR_harmony2/res0.6/9_ego.csv")
ego10=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/STR_harmony2/res0.6/10_ego.csv")
ego11=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/STR_harmony2/res0.6/11_ego.csv")
ego12=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/STR_harmony2/res0.6/12_ego.csv")
ego13=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/STR_harmony2/res0.6/13_ego.csv")
ego14=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/STR_harmony2/res0.6/14_ego.csv")
ego15=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/STR_harmony2/res0.6/15_ego.csv")
ego16=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/STR_harmony2/res0.6/16_ego.csv")


ego0=ego0[1:3,]
ego1=ego1[1:3,]
ego2=ego2[1:3,]
ego3=ego3[1:3,]
ego4=ego4[1:3,]
ego5=ego5[1:3,]
ego6=ego6[1:3,]
ego7=ego7[1:3,]
ego8=ego8[1:3,]
ego9=ego9[1:3,]
ego10=ego10[1:3,]
ego11=ego11[1:3,]
ego12=ego12[1:3,]
ego13=ego13[1:3,]
ego14=ego14[1:3,]
ego15=ego15[1:3,]
ego16=ego16[1:3,]



ego0$cluster="0"
ego1$cluster="1"
ego2$cluster="2"
ego3$cluster="3"
ego4$cluster="4"
ego5$cluster="5"
ego6$cluster="6"
ego7$cluster="7"
ego8$cluster="8"
ego9$cluster="9"
ego10$cluster="10"
ego11$cluster="11"
ego12$cluster="12"
ego13$cluster="13"
ego14$cluster="14"
ego15$cluster="15"
ego16$cluster="16"




ego_all=rbind(ego0,ego1,ego2,ego3,ego4,ego5,
              ego6,ego7,ego8,ego9,ego10,ego11,
              ego12,ego13,ego14,ego15,ego16)

ego_all$cluster=factor(ego_all$cluster,levels = 0:16)


ggplot(data=ego_all)+aes(x=cluster,y=Description,color=-log10(pvalue))+geom_point(aes(size=Count,color= -log10(pvalue)))+
  scale_y_discrete(limits=unique(ego_all$Description))+theme_bw()+xlab("Cluster")+
  scale_color_gradient(low="blue",high = "red")



#进行下游绘图----
#进行后续的简单绘图
umap_cltrl_IUA=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_cltrl_IUA.rds")
table(umap_cltrl_IUA$Double)

table(umap_cltrl_IUA$type3,umap_cltrl_IUA$Double)

table(umap_cltrl_IUA$type3,umap_cltrl_IUA$Double.1)

table(umap_cltrl_IUA$integrated_snn_res.0.3)

umap_cltrl_IUA$integrated_snn_res.0.3=factor(umap_cltrl_IUA$integrated_snn_res.0.3,levels = 0:19)

DimPlot(umap_cltrl_IUA,group.by = "integrated_snn_res.0.3",label=T)

DimPlot(umap_cltrl_IUA,group.by = "Double",label=T)
ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/P20221130_2.jpeg",width = 6,height = 4,
       dpi = 300)



#去除双细胞并进行maker基因查找
umap_ctrl_IUA_allsingle=subset(umap_cltrl_IUA,Double%in%"Singlet")

#进行标准化然后PCA分析
DefaultAssay(umap_cltrl_IUA_allsingle) <- "RNA"



#根据之前的细胞类型定义进行赋值
umap_cltrl_IUA=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_cltrl_IUA.rds")
umap_cltrl_IUA2=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/umap_cltrl_IUA.rds")




umap_cltrl_IUA$cell_type=umap_cltrl_IUA2$cell_type


#去除双细胞并进行maker基因查找
umap_ctrl_IUA_allsingle=subset(umap_cltrl_IUA,Double%in%"Singlet")

#进行标准化然后PCA分析
DefaultAssay(umap_ctrl_IUA_allsingle) <- "RNA"

DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type",label = T)

table(umap_cltrl_IUA2$cell_type)

table(umap_ctrl_IUA_allsingle$cell_type)


#进行分辨率查找
umap_cltrl_IUA=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_cltrl_IUA.rds")

umap_ctrl_IUA_allsingle=subset(umap_cltrl_IUA,Double%in%"Singlet")
DimPlot(umap_ctrl_IUA_allsingle,label = T)

umap_ctrl_IUA_allsingle2=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/umap_ctrl_IUA_allsingle.rds")

umap_ctrl_IUA_allsingle3=umap_ctrl_IUA_allsingle[,colnames(umap_ctrl_IUA_allsingle)%in%colnames(umap_ctrl_IUA_allsingle2)]



umap_ctrl_IUA_allsingle3$cell_type=umap_ctrl_IUA_allsingle2$cell_type[colnames(umap_ctrl_IUA_allsingle2)%in%colnames(umap_ctrl_IUA_allsingle2)]

DimPlot(umap_ctrl_IUA_allsingle3,group.by = "cell_type",label = T)

DimPlot(umap_ctrl_IUA_allsingle2,group.by = "cell_type",label = T)

saveRDS(umap_ctrl_IUA_allsingle3,"/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")

#图1部分
umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")

umap_ctrl_IUA_allsingle$integrated_snn_res.0.3=factor(umap_ctrl_IUA_allsingle$integrated_snn_res.0.3,levels = 0:19)


DimPlot(umap_ctrl_IUA_allsingle,group.by = "integrated_snn_res.0.3",label = T,label.size = 4)+
  ggtitle("")


ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/P1.jpeg",width = 7,height = 5,dpi = 2800)

#将细胞类型定义为四类并继续进行绘图-图2
umap_ctrl_IUA_allsingle@active.ident<-umap_ctrl_IUA_allsingle$integrated_snn_res.0.3
umap_ctrl_IUA_allsingle@active.ident=factor(umap_ctrl_IUA_allsingle@active.ident,levels = 0:19)
new.cluster.ids <- c("Stromal cells","Stromal cells","Stromal cells","Immune cells","Immune cells","Epithelium cells",
                     "Stromal cells","Stromal cells","Stromal cells","Immune cells","Immune cells","Epithelium cells",
                     "Endothelium cells","Stromal cells","Stromal cells","Immune cells","Immune cells","Immune cells","Immune cells","Endothelium cells")


names(new.cluster.ids) <- levels(umap_ctrl_IUA_allsingle)
umap_ctrl_IUA_allsingle <- RenameIdents(umap_ctrl_IUA_allsingle, new.cluster.ids)

umap_ctrl_IUA_allsingle$cell_type1110=umap_ctrl_IUA_allsingle@active.ident

table(umap_ctrl_IUA_allsingle$cell_type1110)



DimPlot(umap_ctrl_IUA_allsingle,label = TRUE,repel = T,label.size = 7,group.by = "cell_type1110")+theme(legend.position ="top")+
  theme(axis.text.x= element_text(size=12,vjust = 1,hjust = 1,face= "bold"),legend.position ="top")+
  theme(axis.text.y= element_text(size=12,vjust = 1,hjust = 1,face= "bold"),axis.title= element_text(size=18))+
  theme(legend.text = element_text(size = 14),legend.title = element_text(size = 19))





ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/P2.jpeg",
       width = 9,height = 6,dpi = 2800)



DimPlot(umap_ctrl_IUA_allsingle,label =F,repel = T,label.size = 7,group.by = "cell_type1110")+theme(legend.position ="top")+
  theme(axis.text.x= element_text(size=0,vjust = 1,hjust = 1,face= "bold"),legend.position ="top")+
  theme(axis.text.y= element_text(size=0,vjust = 1,hjust = 1,face= "bold"),axis.title= element_text(size=18))+
  theme(legend.text = element_text(size = 14),legend.title = element_text(size = 19))+ggtitle("")+
  theme(title = element_text(size = 25),legend.key.size = unit(1,'cm'),legend.position = "none",panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),panel.border = element_blank(),axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),axis.line = element_blank())

ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/P2-2.jpeg",
       width = 9,height = 6,dpi = 2800)

#图3
gene_1227=c("EPCAM","WFDC2","KRT18","MMP7","CLDN4",
            "PECAM1","CDH5","CD34","CLDN5",
            "PTPRC","CD14","CD79A","NKG7","CD3E","MS4A2",
            "LUM","DCN","COL1A1","MMP11","PDGFRB")

umap_ctrl_IUA_allsingle@active.ident=factor(umap_ctrl_IUA_allsingle$integrated_snn_res.0.3,levels = as.character(0:19))


pdf(file="/data/DATAprocess/chentw/zigongneimo/原图绘制/P3.pdf",width = 9,height = 6)

DotPlot(umap_ctrl_IUA_allsingle,features = gene_1227)+
  theme(axis.text.x= element_text(size=12,angle = 45,vjust = 1,hjust = 1,face= "bold"),legend.position ="top")+ylab('Cluster')+
  theme(axis.text.y= element_text(size=12,vjust = 1,hjust = 1,face= "bold"))+ylab('Cluster')+
  theme(legend.text = element_text(size = 14),legend.title = element_text(size = 14))+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])))+
  xlab("")+ylab("")

dev.off()



#图3-2
umap_ctrl_IUA_allsingle$integrated_snn_res.0.3=factor(umap_ctrl_IUA_allsingle$integrated_snn_res.0.3,levels = 0:19)


gene_1227=c("EPCAM","WFDC2","KRT18","MMP7","CLDN4",
            "PECAM1","CDH5","CD34","CLDN5",
            "PTPRC","CD14","CD79A","NKG7","CD3E","MS4A2",
            "LUM","DCN","COL1A1","MMP11","PDGFRB")

bubble.df=as.matrix(umap_ctrl_IUA_allsingle[["RNA"]]@data[gene_1227,])
bubble.df=t(bubble.df)
bubble.df=as.data.frame(scale(bubble.df))
bubble.df$CB=rownames(bubble.df)
umap_ctrl_IUA_allsingle$CB=bubble.df$CB
bubble.df=merge(bubble.df,umap_ctrl_IUA_allsingle@meta.data[,c("CB","integrated_snn_res.0.3")],by = "CB")


bubble.df$CB=NULL

bubble.df$integrated_snn_res.0.3=as.factor(bubble.df$integrated_snn_res.0.3)

integrated_snn_res.0.3_v=c()
gene_v=c()
mean_v=c()
ratio_v=c()
for (i in unique(bubble.df$integrated_snn_res.0.3)) {
  bubble.df_small=bubble.df%>%filter(integrated_snn_res.0.3==i)
  for (j in gene_1227) {
    exp_mean=mean(bubble.df_small[,j])
    exp_ratio=sum(bubble.df_small[,j] > min(bubble.df_small[,j])) / length(bubble.df_small[,j])
    integrated_snn_res.0.3_v=append(integrated_snn_res.0.3_v,i)
    gene_v=append(gene_v,j)
    mean_v=append(mean_v,exp_mean)
    ratio_v=append(ratio_v,exp_ratio)
  }
}

plotdf=data.frame(
  integrated_snn_res.0.3=integrated_snn_res.0.3_v,
  gene=gene_v,
  exp=mean_v,
  ratio=ratio_v
)

plotdf$ratio=plotdf$ratio*100

plotdf$integrated_snn_res.0.3=factor(plotdf$integrated_snn_res.0.3,levels = sort(unique(plotdf$integrated_snn_res.0.3)))
plotdf$gene=factor(plotdf$gene,levels = rev(as.character(gene_1227$gene)))
plotdf$exp=ifelse(plotdf$exp>3,3,plotdf$exp)

plotdf$integrated_snn_res.0.3=factor(plotdf$integrated_snn_res.0.3,levels = 0:19)


plotdf%>%ggplot(aes(x=integrated_snn_res.0.3,y=gene,size=ratio,color=exp))+geom_point()+
  scale_x_discrete("")+scale_y_discrete("")+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])))+
  scale_size_continuous(limits = c(0,100),range = c(2,9))+theme_bw()+
  theme(axis.text.x.bottom = element_text(hjust = 1, vjust = 1, angle = 45)
  )+labs(size="Percent",color="Expression")+
  ylab("")+scale_y_discrete(limits=gene_1227)



#图4

umap_ctrl_IUA_allsingle$type4='type4'
umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type3%in%c("CTRL1","CTRL2")]<-"CTRL"
umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type3%in%c("IUA1","IUA2","IUA5","IUA6")]<-"IUA-mild"
umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type3%in%c("IUA3","IUA4",'IUA7')]<-"IUA-moderate"
saveRDS(umap_ctrl_IUA_allsingle,"/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/umap_ctrl_IUA_allsingle.rds")

umap_ctrl_IUA_allsingle@active.ident=factor(umap_ctrl_IUA_allsingle@active.ident,levels = 0:19)
#CTRL
DimPlot(subset(umap_ctrl_IUA_allsingle,type1%in%'CTRL'),label = T,label.size = 7,repel = T)+ggtitle("Ctrl")+theme(title = element_text(size = 15),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 20),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),axis.line = element_blank())+theme(legend.position = "none")


umap_ctrl_IUA_allsingle_IUA1mild=subset(umap_ctrl_IUA_allsingle,type2%in%'IUA1')
umap_ctrl_IUA_allsingle_IUA1mild=subset(umap_ctrl_IUA_allsingle_IUA1mild,type4%in%'IUA-mild')
DimPlot(umap_ctrl_IUA_allsingle_IUA1mild,label=T,label.size = 7,repel = T)+ggtitle("Mild (Tissue adjacent to adhesion)")+theme(title = element_text(size = 15),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 20),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),axis.line = element_blank())+theme(legend.position = "none")

umap_ctrl_IUA_allsingle_IUA2mild=subset(umap_ctrl_IUA_allsingle,type2%in%'IUA2')
umap_ctrl_IUA_allsingle_IUA2mild=subset(umap_ctrl_IUA_allsingle_IUA2mild,type4%in%'IUA-mild')
DimPlot(umap_ctrl_IUA_allsingle_IUA2mild,label=T,label.size = 7,repel = T)+ggtitle("Mild (adhesion tissue)")+theme(title = element_text(size = 15),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 20),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),axis.line = element_blank())+theme(legend.position = "none")

umap_ctrl_IUA_allsingle_IUA1moderate=subset(umap_ctrl_IUA_allsingle,type2%in%'IUA1')
umap_ctrl_IUA_allsingle_IUA1moderate=subset(umap_ctrl_IUA_allsingle_IUA1moderate,type4%in%'IUA-moderate')
DimPlot(umap_ctrl_IUA_allsingle_IUA1moderate,label=T,label.size = 7,repel = T)+ggtitle("Moderate (Tissue adjacent to adhesion)")+theme(title = element_text(size = 15),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 20),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),axis.line = element_blank())+theme(legend.position = "none")

umap_ctrl_IUA_allsingle_IUA2moderate=subset(umap_ctrl_IUA_allsingle,type2%in%'IUA2')
umap_ctrl_IUA_allsingle_IUA2moderate=subset(umap_ctrl_IUA_allsingle_IUA2moderate,type4%in%'IUA-moderate')
DimPlot(umap_ctrl_IUA_allsingle_IUA2moderate,label=T,label.size = 7,repel = T)+ggtitle("Moderate (adhesion tissue)")+theme(title = element_text(size = 15),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 20),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),axis.line = element_blank())+theme(legend.position = "none")


type4_celltype2_table=as.data.frame(cbind(table(subset(umap_ctrl_IUA_allsingle,type1%in%'CTRL')@active.ident),table(umap_ctrl_IUA_allsingle_IUA1mild@active.ident),table(umap_ctrl_IUA_allsingle_IUA1moderate@active.ident),table(umap_ctrl_IUA_allsingle_IUA2mild@active.ident),table(umap_ctrl_IUA_allsingle_IUA2moderate@active.ident)))



#图5
library(ggsci)
library(ggplot2)


umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")

umap_ctrl_IUA_allsingle$integrated_snn_res.0.3=factor(umap_ctrl_IUA_allsingle$integrated_snn_res.0.3,levels = 0:19)


DimPlot(umap_ctrl_IUA_allsingle,group.by = "integrated_snn_res.0.3",label = T,label.size = 4)+
  ggtitle("")

#图6
umap_ctrl_IUA_allsingle$type4='type4'
umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type3%in%c("CTRL1","CTRL2")]<-"CTRL"
umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type3%in%c("IUA1","IUA2","IUA5","IUA6")]<-"IUA-mild"
umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type3%in%c("IUA3","IUA4",'IUA7')]<-"IUA-moderate"


table(umap_ctrl_IUA_allsingle$type4,umap_ctrl_IUA_allsingle$type3)

table(umap_ctrl_IUA_allsingle$type3)

umap_ctrl_IUA_allsingle$stage="stage"

umap_ctrl_IUA_allsingle$stage[umap_ctrl_IUA_allsingle$type4%in%"CTRL"]<-"0"
umap_ctrl_IUA_allsingle$stage[umap_ctrl_IUA_allsingle$type4%in%"IUA-mild"]<-"1"
umap_ctrl_IUA_allsingle$stage[umap_ctrl_IUA_allsingle$type4%in%"IUA-moderate"]<-"2"


DimPlot(umap_ctrl_IUA_allsingle,group.by = "stage",split.by = "type1",label = F)+ggtitle("")

ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/P20221130_1.jpeg",
       width = 12,height = 6,dpi = 300)


DimPlot(umap_ctrl_IUA_allsingle,group.by = "type3",split.by = "stage")+ggtitle("")

ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/P20221130_2.jpeg",
       width = 18,height = 6,dpi = 900)





#图6-对九个样本进行细胞类型的四种的饼形绘制----
umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")

umap_ctrl_IUA_allsingle@active.ident<-umap_ctrl_IUA_allsingle$integrated_snn_res.0.3
umap_ctrl_IUA_allsingle@active.ident=factor(umap_ctrl_IUA_allsingle@active.ident,levels = 0:19)
new.cluster.ids <- c("Stromal cells","Stromal cells","Stromal cells","Immune cells","Immune cells","Epithelium cells",
                     "Stromal cells","Stromal cells","Stromal cells","Immune cells","Immune cells","Epithelium cells",
                     "Endothelium cells","Stromal cells","Stromal cells","Immune cells","Immune cells","Immune cells","Immune cells","Endothelium cells")


names(new.cluster.ids) <- levels(umap_ctrl_IUA_allsingle)
umap_ctrl_IUA_allsingle <- RenameIdents(umap_ctrl_IUA_allsingle, new.cluster.ids)

umap_ctrl_IUA_allsingle$cell_type1110=umap_ctrl_IUA_allsingle@active.ident

table(umap_ctrl_IUA_allsingle$cell_type1110)
umap_ctrl_IUA_allsingle@active.ident=factor(umap_ctrl_IUA_allsingle@active.ident,levels = 0:19)


cell_type1110_table=as.data.frame(table(umap_ctrl_IUA_allsingle$cell_type1110,umap_ctrl_IUA_allsingle$type3))
colnames(cell_type1110_table)[1]<-"cell type"

###CTRL1
cell_type1110_table_CTRL1=subset(cell_type1110_table,Var2%in%"CTRL1")


myLabel = as.vector(cell_type1110_table_CTRL1$`cell type`)

myLabel = paste(myLabel, "(", round(cell_type1110_table_CTRL1$Freq / sum(cell_type1110_table_CTRL1$Freq) * 100, 2), "%)        ", sep = "") 

P_CTRL1=ggplot(cell_type1110_table_CTRL1, aes(x = "", fill = `cell type`, y =Freq))+ 
  geom_bar(stat = "identity", width = 1)+  
  coord_polar(theta = "y")+theme_classic()+ 
  labs(x = "", y = "", title = "") + 
  theme(axis.ticks = element_blank())+ 
  theme(axis.ticks = element_blank()) + 
  theme(legend.title = element_blank()) + 
  scale_fill_discrete(breaks = cell_type1110_table_CTRL1$`cell type`, labels = myLabel)+
  theme(axis.line = element_blank())+ 
  theme(axis.text.x = element_blank())+ggtitle("CTRL1")

###CTRL2
cell_type1110_table_CTRL2=subset(cell_type1110_table,Var2%in%"CTRL2")


myLabel = as.vector(cell_type1110_table_CTRL2$`cell type`)

myLabel = paste(myLabel, "(", round(cell_type1110_table_CTRL2$Freq / sum(cell_type1110_table_CTRL2$Freq) * 100, 2), "%)        ", sep = "") 

P_CTRL2=ggplot(cell_type1110_table_CTRL2, aes(x = "", fill = `cell type`, y =Freq))+ 
  geom_bar(stat = "identity", width = 1)+  
  coord_polar(theta = "y")+theme_classic()+ 
  labs(x = "", y = "", title = "") + 
  theme(axis.ticks = element_blank())+ 
  theme(axis.ticks = element_blank()) + 
  theme(legend.title = element_blank()) + 
  scale_fill_discrete(breaks = cell_type1110_table_CTRL2$`cell type`, labels = myLabel)+
  theme(axis.line = element_blank())+ 
  theme(axis.text.x = element_blank())+ggtitle("CTRL2")


###IUA1
cell_type1110_table_IUA1=subset(cell_type1110_table,Var2%in%"IUA1")


myLabel = as.vector(cell_type1110_table_IUA1$`cell type`)

myLabel = paste(myLabel, "(", round(cell_type1110_table_IUA1$Freq / sum(cell_type1110_table_IUA1$Freq) * 100, 2), "%)        ", sep = "") 

P_IUA1=ggplot(cell_type1110_table_IUA1, aes(x = "", fill = `cell type`, y =Freq))+ 
  geom_bar(stat = "identity", width = 1)+  
  coord_polar(theta = "y")+theme_classic()+ 
  labs(x = "", y = "", title = "") + 
  theme(axis.ticks = element_blank())+ 
  theme(axis.ticks = element_blank()) + 
  theme(legend.title = element_blank()) + 
  scale_fill_discrete(breaks = cell_type1110_table_IUA1$`cell type`, labels = myLabel)+
  theme(axis.line = element_blank())+ 
  theme(axis.text.x = element_blank())+ggtitle("IUA1")



###IUA2
cell_type1110_table_IUA2=subset(cell_type1110_table,Var2%in%"IUA2")


myLabel = as.vector(cell_type1110_table_IUA2$`cell type`)

myLabel = paste(myLabel, "(", round(cell_type1110_table_IUA2$Freq / sum(cell_type1110_table_IUA2$Freq) * 100, 2), "%)        ", sep = "") 

P_IUA2=ggplot(cell_type1110_table_IUA2, aes(x = "", fill = `cell type`, y =Freq))+ 
  geom_bar(stat = "identity", width = 1)+  
  coord_polar(theta = "y")+theme_classic()+ 
  labs(x = "", y = "", title = "") + 
  theme(axis.ticks = element_blank())+ 
  theme(axis.ticks = element_blank()) + 
  theme(legend.title = element_blank()) + 
  scale_fill_discrete(breaks = cell_type1110_table_IUA2$`cell type`, labels = myLabel)+
  theme(axis.line = element_blank())+ 
  theme(axis.text.x = element_blank())+ggtitle("IUA2")


###IUA3
cell_type1110_table_IUA3=subset(cell_type1110_table,Var2%in%"IUA3")


myLabel = as.vector(cell_type1110_table_IUA3$`cell type`)

myLabel = paste(myLabel, "(", round(cell_type1110_table_IUA3$Freq / sum(cell_type1110_table_IUA3$Freq) * 100, 2), "%)        ", sep = "") 

P_IUA3=ggplot(cell_type1110_table_IUA3, aes(x = "", fill = `cell type`, y =Freq))+ 
  geom_bar(stat = "identity", width = 1)+  
  coord_polar(theta = "y")+theme_classic()+ 
  labs(x = "", y = "", title = "") + 
  theme(axis.ticks = element_blank())+ 
  theme(axis.ticks = element_blank()) + 
  theme(legend.title = element_blank()) + 
  scale_fill_discrete(breaks = cell_type1110_table_IUA3$`cell type`, labels = myLabel)+
  theme(axis.line = element_blank())+ 
  theme(axis.text.x = element_blank())+ggtitle("IUA3")


###IUA4
cell_type1110_table_IUA4=subset(cell_type1110_table,Var2%in%"IUA4")


myLabel = as.vector(cell_type1110_table_IUA4$`cell type`)

myLabel = paste(myLabel, "(", round(cell_type1110_table_IUA4$Freq / sum(cell_type1110_table_IUA4$Freq) * 100, 2), "%)        ", sep = "") 

P_IUA4=ggplot(cell_type1110_table_IUA4, aes(x = "", fill = `cell type`, y =Freq))+ 
  geom_bar(stat = "identity", width = 1)+  
  coord_polar(theta = "y")+theme_classic()+ 
  labs(x = "", y = "", title = "") + 
  theme(axis.ticks = element_blank())+ 
  theme(axis.ticks = element_blank()) + 
  theme(legend.title = element_blank()) + 
  scale_fill_discrete(breaks = cell_type1110_table_IUA4$`cell type`, labels = myLabel)+
  theme(axis.line = element_blank())+ 
  theme(axis.text.x = element_blank())+ggtitle("IUA4")

###IUA5
cell_type1110_table_IUA5=subset(cell_type1110_table,Var2%in%"IUA5")


myLabel = as.vector(cell_type1110_table_IUA5$`cell type`)

myLabel = paste(myLabel, "(", round(cell_type1110_table_IUA5$Freq / sum(cell_type1110_table_IUA5$Freq) * 100, 2), "%)        ", sep = "") 

P_IUA5=ggplot(cell_type1110_table_IUA5, aes(x = "", fill = `cell type`, y =Freq))+ 
  geom_bar(stat = "identity", width = 1)+  
  coord_polar(theta = "y")+theme_classic()+ 
  labs(x = "", y = "", title = "") + 
  theme(axis.ticks = element_blank())+ 
  theme(axis.ticks = element_blank()) + 
  theme(legend.title = element_blank()) + 
  scale_fill_discrete(breaks = cell_type1110_table_IUA5$`cell type`, labels = myLabel)+
  theme(axis.line = element_blank())+ 
  theme(axis.text.x = element_blank())+ggtitle("IUA5")


###IUA6
cell_type1110_table_IUA6=subset(cell_type1110_table,Var2%in%"IUA6")


myLabel = as.vector(cell_type1110_table_IUA6$`cell type`)

myLabel = paste(myLabel, "(", round(cell_type1110_table_IUA6$Freq / sum(cell_type1110_table_IUA6$Freq) * 100, 2), "%)        ", sep = "") 

P_IUA6=ggplot(cell_type1110_table_IUA6, aes(x = "", fill = `cell type`, y =Freq))+ 
  geom_bar(stat = "identity", width = 1)+  
  coord_polar(theta = "y")+theme_classic()+ 
  labs(x = "", y = "", title = "") + 
  theme(axis.ticks = element_blank())+ 
  theme(axis.ticks = element_blank()) + 
  theme(legend.title = element_blank()) + 
  scale_fill_discrete(breaks = cell_type1110_table_IUA6$`cell type`, labels = myLabel)+
  theme(axis.line = element_blank())+ 
  theme(axis.text.x = element_blank())+ggtitle("IUA6")

###IUA7
cell_type1110_table_IUA7=subset(cell_type1110_table,Var2%in%"IUA7")


myLabel = as.vector(cell_type1110_table_IUA7$`cell type`)

myLabel = paste(myLabel, "(", round(cell_type1110_table_IUA7$Freq / sum(cell_type1110_table_IUA7$Freq) * 100, 2), "%)        ", sep = "") 

P_IUA7=ggplot(cell_type1110_table_IUA7, aes(x = "", fill = `cell type`, y =Freq))+ 
  geom_bar(stat = "identity", width = 1)+  
  coord_polar(theta = "y")+theme_classic()+ 
  labs(x = "", y = "", title = "") + 
  theme(axis.ticks = element_blank())+ 
  theme(axis.ticks = element_blank()) + 
  theme(legend.title = element_blank()) + 
  scale_fill_discrete(breaks = cell_type1110_table_IUA7$`cell type`, labels = myLabel)+
  theme(axis.line = element_blank())+ 
  theme(axis.text.x = element_blank())+ggtitle("IUA7")

ggarrange(P_CTRL1,P_CTRL2,P_IUA1,P_IUA2,P_IUA3,P_IUA4,P_IUA5,P_IUA6,P_IUA7,
          ncol=3,nrow = 3)

#带文字版
P_CTRL1+scale_fill_manual(values = c("#63A0A7","#D38167","#304553","#C03335"))
ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/饼状图/P_CTRL1.jpeg",width = 6,height = 6,
       dpi = 300)


P_CTRL2+scale_fill_manual(values = c("#63A0A7","#D38167","#304553","#C03335"))
ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/饼状图/P_CTRL2.jpeg",width = 6,height = 6,
       dpi = 300)


P_IUA1+scale_fill_manual(values = c("#63A0A7","#D38167","#304553","#C03335"))
ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/饼状图/P_IUA1.jpeg",width = 6,height = 6,
       dpi = 300)

P_IUA2+scale_fill_manual(values = c("#63A0A7","#D38167","#304553","#C03335"))
ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/饼状图/P_IUA2.jpeg",width = 6,height = 6,
       dpi = 300)

P_IUA3+scale_fill_manual(values = c("#63A0A7","#D38167","#304553","#C03335"))
ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/饼状图/P_IUA3.jpeg",width = 6,height = 6,
       dpi = 300)

P_IUA4+scale_fill_manual(values = c("#63A0A7","#D38167","#304553","#C03335"))
ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/饼状图/P_IUA4.jpeg",width = 6,height = 6,
       dpi = 300)

P_IUA5+scale_fill_manual(values = c("#63A0A7","#D38167","#304553","#C03335"))
ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/饼状图/P_IUA5.jpeg",width = 6,height = 6,
       dpi = 300)

P_IUA6+scale_fill_manual(values = c("#63A0A7","#D38167","#304553","#C03335"))
ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/饼状图/P_IUA6.jpeg",width = 6,height = 6,
       dpi = 300)

P_IUA7+scale_fill_manual(values = c("#63A0A7","#D38167","#304553","#C03335"))
ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/饼状图/P_IUA7.jpeg",width = 6,height = 6,
       dpi = 300)

#不带文字版
P_CTRL1+scale_fill_manual(values = c("#63A0A7","#D38167","#304553","#C03335"))+ggtitle("")+
  theme(legend.position = "none")
ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/饼状图2/P_CTRL1.jpeg",width = 6,height = 6,
       dpi = 300)


P_CTRL2+scale_fill_manual(values = c("#63A0A7","#D38167","#304553","#C03335"))+ggtitle("")+
  theme(legend.position = "none")
ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/饼状图2/P_CTRL2.jpeg",width = 6,height = 6,
       dpi = 300)


P_IUA1+scale_fill_manual(values = c("#63A0A7","#D38167","#304553","#C03335"))+ggtitle("")+
  theme(legend.position = "none")
ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/饼状图2/P_IUA1.jpeg",width = 6,height = 6,
       dpi = 300)

P_IUA2+scale_fill_manual(values = c("#63A0A7","#D38167","#304553","#C03335"))+ggtitle("")+
  theme(legend.position = "none")
ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/饼状图2/P_IUA2.jpeg",width = 6,height = 6,
       dpi = 300)

P_IUA3+scale_fill_manual(values = c("#63A0A7","#D38167","#304553","#C03335"))+ggtitle("")+
  theme(legend.position = "none")
ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/饼状图2/P_IUA3.jpeg",width = 6,height = 6,
       dpi = 300)

P_IUA4+scale_fill_manual(values = c("#63A0A7","#D38167","#304553","#C03335"))+ggtitle("")+
  theme(legend.position = "none")
ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/饼状图2/P_IUA4.jpeg",width = 6,height = 6,
       dpi = 300)

P_IUA5+scale_fill_manual(values = c("#63A0A7","#D38167","#304553","#C03335"))+ggtitle("")+
  theme(legend.position = "none")
ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/饼状图2/P_IUA5.jpeg",width = 6,height = 6,
       dpi = 300)

P_IUA6+scale_fill_manual(values = c("#63A0A7","#D38167","#304553","#C03335"))+ggtitle("")+
  theme(legend.position = "none")
ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/饼状图2/P_IUA6.jpeg",width = 6,height = 6,
       dpi = 300)

P_IUA7+scale_fill_manual(values = c("#63A0A7","#D38167","#304553","#C03335"))+ggtitle("")+
  theme(legend.position = "none")
ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/饼状图2/P_IUA7.jpeg",width = 6,height = 6,
       dpi = 300)


#将比例的平均值进行柱形图绘制
#CTRL组-0
cell_type1110_table_CTRL1$ratio=cell_type1110_table_CTRL1$Freq/sum(cell_type1110_table_CTRL1$Freq)
cell_type1110_table_CTRL2$ratio=cell_type1110_table_CTRL2$Freq/sum(cell_type1110_table_CTRL2$Freq)
cell_type1110_table_CTRL1=cell_type1110_table_CTRL1[,c(1,4)]
cell_type1110_table_CTRL2=cell_type1110_table_CTRL2[,c(1,4)]

colnames(cell_type1110_table_CTRL1)[1]<-"cell_type"
colnames(cell_type1110_table_CTRL2)[1]<-"cell_type"
CTRL=dplyr::left_join(cell_type1110_table_CTRL1,cell_type1110_table_CTRL2,by="cell_type")

CTRL$ratio=(CTRL$ratio.x+CTRL$ratio.y)/2


myLabel = as.vector(CTRL$cell_type)

myLabel = paste(myLabel, "(", as.character(round(CTRL$ratio* 100,2)),"%", ")", sep = "") 





ggplot(CTRL, aes(x = "", fill = CTRL$cell_type, y =CTRL$ratio))+ 
  geom_bar(stat = "identity", width = 1)+  
  coord_polar(theta = "y")+theme_classic()+ 
  labs(x = "", y = "", title = "") + 
  theme(axis.ticks = element_blank())+ 
  theme(axis.ticks = element_blank()) + 
  theme(legend.title = element_blank()) + 
  scale_fill_discrete(breaks = CTRL$cell_type,label=myLabel)+
  theme(axis.line = element_blank())+ 
  theme(axis.text.x = element_blank())+ggtitle("0")+
  scale_fill_manual(values = c("#63A0A7","#D38167","#304553","#C03335"),limits=CTRL$cell_type,label=myLabel)

ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/饼状图3/0.jpeg",width = 6,height = 6,
       dpi = 300)

ggplot(CTRL, aes(x = "", fill = CTRL$cell_type, y =CTRL$ratio))+ 
  geom_bar(stat = "identity", width = 1)+  
  coord_polar(theta = "y")+theme_classic()+ 
  labs(x = "", y = "", title = "") + 
  theme(axis.ticks = element_blank())+ 
  theme(axis.ticks = element_blank()) + 
  theme(legend.title = element_blank()) + 
  scale_fill_discrete(breaks = CTRL$cell_type,label=)+
  theme(axis.line = element_blank())+ 
  theme(axis.text.x = element_blank())+ggtitle("0")+
  scale_fill_manual(values = c("#63A0A7","#D38167","#304553","#C03335"))+ggtitle("")+
  theme(legend.position = "none")+
  scale_fill_manual(values = c("#63A0A7","#D38167","#304553","#C03335"),limits=CTRL$cell_type,label=myLabel)


ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/饼状图3/0_2.jpeg",width = 6,height = 6,
       dpi = 300)


#IUA组-1
cell_type1110_table_IUA1$ratio=cell_type1110_table_IUA1$Freq/sum(cell_type1110_table_IUA1$Freq)
cell_type1110_table_IUA2$ratio=cell_type1110_table_IUA2$Freq/sum(cell_type1110_table_IUA2$Freq)
cell_type1110_table_IUA5$ratio=cell_type1110_table_IUA5$Freq/sum(cell_type1110_table_IUA5$Freq)
cell_type1110_table_IUA6$ratio=cell_type1110_table_IUA6$Freq/sum(cell_type1110_table_IUA6$Freq)



cell_type1110_table_IUA1=cell_type1110_table_IUA1[,c(1,4)]
cell_type1110_table_IUA2=cell_type1110_table_IUA2[,c(1,4)]
cell_type1110_table_IUA5=cell_type1110_table_IUA5[,c(1,4)]
cell_type1110_table_IUA6=cell_type1110_table_IUA6[,c(1,4)]

colnames(cell_type1110_table_IUA1)[1]<-"cell_type"
colnames(cell_type1110_table_IUA2)[1]<-"cell_type"
colnames(cell_type1110_table_IUA5)[1]<-"cell_type"
colnames(cell_type1110_table_IUA6)[1]<-"cell_type"


IUA=cbind(cell_type1110_table_IUA1,cell_type1110_table_IUA2$ratio,cell_type1110_table_IUA5$ratio,
          cell_type1110_table_IUA6$ratio)

IUA$ratio=(IUA$ratio+IUA$`cell_type1110_table_IUA2$ratio`+IUA$`cell_type1110_table_IUA5$ratio`+IUA$`cell_type1110_table_IUA6$ratio`)/4


myLabel = as.vector(IUA$cell_type)

myLabel = paste(myLabel, "(", as.character(round(IUA$ratio* 100,2)),"%", ")", sep = "") 



ggplot(IUA, aes(x = "", fill = IUA$cell_type, y =IUA$ratio))+ 
  geom_bar(stat = "identity", width = 1)+  
  coord_polar(theta = "y")+theme_classic()+ 
  labs(x = "", y = "", title = "") + 
  theme(axis.ticks = element_blank())+ 
  theme(axis.ticks = element_blank()) + 
  theme(legend.title = element_blank()) + 
  scale_fill_discrete(breaks = IUA$cell_type)+
  theme(axis.line = element_blank())+ 
  theme(axis.text.x = element_blank())+ggtitle("1")+
  scale_fill_manual(values = c("#63A0A7","#D38167","#304553","#C03335"))+
  scale_fill_manual(values = c("#63A0A7","#D38167","#304553","#C03335"),limits=CTRL$cell_type,label=myLabel)



ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/饼状图3/1_1.jpeg",width = 6,height = 6,
       dpi = 300)

ggplot(IUA, aes(x = "", fill = IUA$cell_type, y =IUA$ratio))+ 
  geom_bar(stat = "identity", width = 1)+  
  coord_polar(theta = "y")+theme_classic()+ 
  labs(x = "", y = "", title = "") + 
  theme(axis.ticks = element_blank())+ 
  theme(axis.ticks = element_blank()) + 
  theme(legend.title = element_blank()) + 
  scale_fill_discrete(breaks = IUA$cell_type)+
  theme(axis.line = element_blank())+ 
  theme(axis.text.x = element_blank())+ggtitle("1")+
  scale_fill_manual(values = c("#63A0A7","#D38167","#304553","#C03335"))+theme(legend.position = "none")+
  ggtitle("")

ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/饼状图3/1_2.jpeg",width = 6,height = 6,
       dpi = 300)






#IUA组-2
cell_type1110_table_IUA3$ratio=cell_type1110_table_IUA3$Freq/sum(cell_type1110_table_IUA3$Freq)
cell_type1110_table_IUA4$ratio=cell_type1110_table_IUA4$Freq/sum(cell_type1110_table_IUA4$Freq)
cell_type1110_table_IUA7$ratio=cell_type1110_table_IUA7$Freq/sum(cell_type1110_table_IUA7$Freq)



cell_type1110_table_IUA3=cell_type1110_table_IUA3[,c(1,4)]
cell_type1110_table_IUA4=cell_type1110_table_IUA4[,c(1,4)]
cell_type1110_table_IUA7=cell_type1110_table_IUA7[,c(1,4)]

colnames(cell_type1110_table_IUA3)[1]<-"cell_type"
colnames(cell_type1110_table_IUA4)[1]<-"cell_type"
colnames(cell_type1110_table_IUA7)[1]<-"cell_type"


IUA=cbind(cell_type1110_table_IUA3,cell_type1110_table_IUA4$ratio,cell_type1110_table_IUA7$ratio)

IUA$ratio=(IUA$ratio+IUA$`cell_type1110_table_IUA4$ratio`+IUA$`cell_type1110_table_IUA7$ratio`)/3


myLabel = as.vector(IUA$cell_type)

myLabel = paste(myLabel, "(", as.character(round(IUA$ratio* 100,2)),"%", ")", sep = "") 


ggplot(IUA, aes(x = "", fill = IUA$cell_type, y =IUA$ratio))+ 
  geom_bar(stat = "identity", width = 1)+  
  coord_polar(theta = "y")+theme_classic()+ 
  labs(x = "", y = "", title = "") + 
  theme(axis.ticks = element_blank())+ 
  theme(axis.ticks = element_blank()) + 
  theme(legend.title = element_blank()) + 
  scale_fill_discrete(breaks = IUA$cell_type)+
  theme(axis.line = element_blank())+ 
  theme(axis.text.x = element_blank())+ggtitle("2")+
  scale_fill_manual(values = c("#63A0A7","#D38167","#304553","#C03335"))+
  scale_fill_manual(values = c("#63A0A7","#D38167","#304553","#C03335"),limits=CTRL$cell_type,label=myLabel)




ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/饼状图3/2_1.jpeg",width = 6,height = 6,
       dpi = 300)

ggplot(IUA, aes(x = "", fill = IUA$cell_type, y =IUA$ratio))+ 
  geom_bar(stat = "identity", width = 1)+  
  coord_polar(theta = "y")+theme_classic()+ 
  labs(x = "", y = "", title = "") + 
  theme(axis.ticks = element_blank())+ 
  theme(axis.ticks = element_blank()) + 
  theme(legend.title = element_blank()) + 
  scale_fill_discrete(breaks = IUA$cell_type)+
  theme(axis.line = element_blank())+ 
  theme(axis.text.x = element_blank())+ggtitle("1")+
  scale_fill_manual(values = c("#63A0A7","#D38167","#304553","#C03335"))+theme(legend.position = "none")+
  ggtitle("")

ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/饼状图3/2_2.jpeg",width = 6,height = 6,
       dpi = 300)












#原图绘制1-----
#小提琴图1

umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")



gene1116=c("EPCAM","CDH1","COL1A1","PDGFRA","VWF","PECAM1","PTPRC",
           "CD3G")

vln1116=function(X){VlnPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type1110",features =X,
                            pt.size = 0)+xlab("")+ylab(X)+theme(axis.text.y = element_blank(),axis.text.x = element_blank(),
                                                                legend.position = "none")+ggtitle("")
}


P=lapply(gene1116,vln1116)

P[[8]]=VlnPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type1110",features ="CD3G",
               pt.size = 0)+xlab("")+ylab("CD3G")+theme(axis.text.y = element_blank(),legend.position = "none")+
  ggtitle("")

ggarrange(P[[1]],P[[2]],P[[3]],P[[4]],P[[5]],P[[6]],P[[7]],P[[8]],
          common.legend = F,ncol = 1,heights = c(2,2,2,2,2,2,2,3))

ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/P6-1.jpeg",width = 4,height = 17.5,dpi = 1800)




P[[1]]
ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/P6-1.jpeg",width = 4,height = 2,dpi = 1800)

P[[2]]
ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/P6-2.jpeg",width = 4,height = 2,dpi = 1800)

P[[3]]
ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/P6-3.jpeg",width = 4,height = 2,dpi = 1800)

P[[4]]
ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/P6-4.jpeg",width = 4,height = 2,dpi = 1800)

P[[5]]
ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/P6-5.jpeg",width = 4,height = 2,dpi = 1800)

P[[6]]
ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/P6-6.jpeg",width = 4,height = 2,dpi = 1800)

P[[7]]
ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/P6-7.jpeg",width = 4,height = 2,dpi = 1800)

P[[8]]
ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/P6-8.jpeg",width = 4,height = 2,dpi = 1800)








#小提琴图2

umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")

umap_ctrl_IUA_allsingle@active.ident<-umap_ctrl_IUA_allsingle$integrated_snn_res.0.3
umap_ctrl_IUA_allsingle@active.ident=factor(umap_ctrl_IUA_allsingle@active.ident,levels = 0:19)
new.cluster.ids <- c("Stromal cells","Stromal cells","Stromal cells","Immune cells","Immune cells","Epithelium cells",
                     "Stromal cells","Stromal cells","Stromal cells","Immune cells","Immune cells","Epithelium cells",
                     "Endothelium cells","Stromal cells","Stromal cells","Immune cells","Immune cells","Immune cells","Immune cells","Endothelium cells")


names(new.cluster.ids) <- levels(umap_ctrl_IUA_allsingle)
umap_ctrl_IUA_allsingle <- RenameIdents(umap_ctrl_IUA_allsingle, new.cluster.ids)

umap_ctrl_IUA_allsingle$cell_type1110=umap_ctrl_IUA_allsingle@active.ident

table(umap_ctrl_IUA_allsingle$cell_type1110)

gene1116=c("EPCAM","CDH1","COL1A1","PDGFRA","VWF","PECAM1","PTPRC",
           "CD3G")

umap_ctrl_IUA_allsingle_data=
  as.data.frame(umap_ctrl_IUA_allsingle@assays$RNA@data)

umap_ctrl_IUA_allsingle_data=subset(umap_ctrl_IUA_allsingle_data,
                                    rownames(umap_ctrl_IUA_allsingle_data)%in%gene1116)


umap_ctrl_IUA_allsingle_data=as.data.frame(t(umap_ctrl_IUA_allsingle_data))

umap_ctrl_IUA_allsingle_data$cell_type1110=umap_ctrl_IUA_allsingle$cell_type1110

umap_ctrl_IUA_allsingle_data=gather(umap_ctrl_IUA_allsingle_data,gene1116, value, colnames(umap_ctrl_IUA_allsingle_data),-cell_type1110)





ggplot(umap_ctrl_IUA_allsingle_data, aes(x=gene1116, y=value, fill=cell_type1110))+facet_grid(~gene1116, scales = "free_x")+
  geom_violin()+theme_bw()+xlab("")+
  ylab("")


umap_ctrl_IUA_allsingle_data1=subset(umap_ctrl_IUA_allsingle_data,gene1116%in%"NCAM1")
quantile(umap_ctrl_IUA_allsingle_data1$value,0.1)


umap_ctrl_IUA_allsingle_data1=subset(umap_ctrl_IUA_allsingle_data1,cell_type1110%in%"Immune cells")
quantile(umap_ctrl_IUA_allsingle_data1$value,0.90)



#查看COL1A1在各个细胞类型中的表达比例
umap_ctrl_IUA_allsingle_data2=subset(umap_ctrl_IUA_allsingle_data,
                                     gene1116%in%"COL1A1")

umap_ctrl_IUA_allsingle_data2$IF="T"
umap_ctrl_IUA_allsingle_data2$IF[umap_ctrl_IUA_allsingle_data2$value%in%0]="F"
table(umap_ctrl_IUA_allsingle_data2$IF,umap_ctrl_IUA_allsingle_data2$cell_type1110)

#进行umap图绘制(区分)
umap_ctrl_IUA_allsingle$type4<-"type4"
umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type3%in%c("IUA1","IUA2","IUA5","IUA6")]<-"IUA-mild"
umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type3%in%c("IUA3","IUA4","IUA7")]<-"IUA-moderate"
umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type1%in%c("CTRL")]<-"CTRL"

umap_ctrl_IUA_allsingle$type5<-"type5"
umap_ctrl_IUA_allsingle$type5[umap_ctrl_IUA_allsingle$type1%in%"CTRL"]<- "Ctrl"
umap_ctrl_IUA_allsingle$type5[umap_ctrl_IUA_allsingle$type2%in%"IUA1"&umap_ctrl_IUA_allsingle$type4%in%"IUA-mild"]<- "miIUA-a"
umap_ctrl_IUA_allsingle$type5[umap_ctrl_IUA_allsingle$type2%in%"IUA2"&umap_ctrl_IUA_allsingle$type4%in%"IUA-mild"]<- "miIUA-I"
umap_ctrl_IUA_allsingle$type5[umap_ctrl_IUA_allsingle$type2%in%"IUA1"&umap_ctrl_IUA_allsingle$type4%in%"IUA-moderate"]<- "moIUA-a"
umap_ctrl_IUA_allsingle$type5[umap_ctrl_IUA_allsingle$type2%in%"IUA2"&umap_ctrl_IUA_allsingle$type4%in%"IUA-moderate"]<- "moIUA-I"


DimPlot(umap_ctrl_IUA_allsingle,group.by = "type5",label = F,label.size = 2)+
  theme(title = element_text(size = 25),legend.key.size = unit(1,'cm'),legend.position = "none",
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),
        panel.border = element_blank(),axis.title = element_blank(),axis.text = element_blank(),
        axis.ticks = element_blank(),axis.line = element_blank())+ggtitle("")





ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/P4.jpeg",
       width = 9,height = 6,dpi = 2800)




DimPlot(umap_ctrl_IUA_allsingle,group.by = "type5",label = T,repel = T)+
  ggtitle("")





ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/P4-2.jpeg",
       width = 9,height = 6,dpi = 2800)



DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type1110",split.by = "type5",label = T)


umap_ctrl_IUA_allsingle$integrated_snn_res.0.3=factor(umap_ctrl_IUA_allsingle$integrated_snn_res.0.3,levels = 0:19)

DimPlot(umap_ctrl_IUA_allsingle,group.by = "integrated_snn_res.0.3",split.by = "type5",label = T)


#进行原图中的指控的图的绘制
View(umap_ctrl_IUA_allsingle@meta.data)

VlnPlot(umap_ctrl_IUA_allsingle, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)

umap_quality_plot=umap_ctrl_IUA_allsingle@meta.data[c("nFeature_RNA","nCount_RNA","type5")]


umap_quality_plot_P1=as.data.frame(table(umap_quality_plot$type5))

P1=ggplot(data=umap_quality_plot_P1)+aes(x=Var1,y=Freq,fill=Var1)+geom_bar(stat="identity",width=0.85)+
  scale_y_continuous(expand=c(0,0))+
  theme_bw()+theme(panel.grid.major=element_blank(),panel.grid.minor=element_blank())+
  theme(legend.position="none")+theme(panel.border = element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank(),
                                      axis.ticks.y = element_blank())+  
  theme(axis.line = element_line(size=0.4, colour = "black"))+coord_flip()+
  scale_x_discrete(limits=c("Ctrl","miIUA-a","miIUA-I","moIUA-a","moIUA-I"))+
  ylab("cells")+xlab("")+scale_fill_manual(values=c("#3B6241","#797592","#596C94","#363F70","#A3C5B8"))



umap_quality_plot_P2=umap_ctrl_IUA_allsingle@meta.data[c("nFeature_RNA","type5")]

P2=ggplot(data=umap_quality_plot_P2)+aes(x=nFeature_RNA,y=type5,fill=type5)+
  geom_violin()+theme_bw()+ylab("")+theme(legend.position = "none")+xlab("Genes")+
  geom_boxplot(width=.1,col="black",fill="black")+scale_fill_manual(values=c("#A3C5B8","#363F70","#596C94","#797592","#3B6241"))+
  scale_y_discrete(limits=c("Ctrl","miIUA-a","miIUA-I","moIUA-a","moIUA-I"))+
  theme(axis.text.y =element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank(),axis.ticks.y = element_blank())






umap_quality_plot_P3=umap_ctrl_IUA_allsingle@meta.data[c("nCount_RNA","type5")]

P3=ggplot(data=umap_quality_plot_P3)+aes(x=nCount_RNA,y=type5,fill=type5)+
  geom_violin()+theme_bw()+ylab("")+theme(legend.position = "none")+xlab("UMI")+
  geom_boxplot(width=.1,col="black",fill="black")+scale_fill_manual(values=c("#A3C5B8","#363F70","#596C94","#797592","#3B6241"))+
  scale_y_discrete(limits=c("Ctrl","miIUA-a","miIUA-I","moIUA-a","moIUA-I"))+
  theme(axis.text.y =element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank(),axis.ticks.y = element_blank())




ggarrange(P1,P2,P3,ncol = 3)



###桑葚图的绘制
table(umap_ctrl_IUA_allsingle$type3)

umap_ctrl_IUA_allsingle$type5=factor(umap_ctrl_IUA_allsingle$type5,
                                     levels = c("moIUA-I","moIUA-a","miIUA-I","miIUA-a","Ctrl"))

umap_ctrl_IUA_allsingle$type3=factor(umap_ctrl_IUA_allsingle$type3,
                                     levels = c("IUA7","IUA6","IUA5","IUA4",
                                                "IUA3","IUA2","IUA1","CTRL2","CTRL1"))


P4_table=as.data.frame(table(umap_ctrl_IUA_allsingle$type3,umap_ctrl_IUA_allsingle$type5))
P4_table=subset(P4_table,P4_table$Freq!=0)
P4_table=P4_table[1:2]

colnames(P4_table)[1]<-"sample"
colnames(P4_table)[2]<-"Group"



P4_table=rbind(P4_table,P4_table[P4_table$sample%in%c("IUA7"),])

P4_table$link <- 1
P4_table <- reshape::melt(P4_table, id = 'link')

variable <- summary(P4_table$variable)
P4_table$flow <- rep(1:variable[1], length(variable))


P4=ggplot(P4_table, aes(x = variable, y = link,
                        stratum = value, alluvium = flow, fill = value)) +
  geom_stratum() +  
  geom_flow(aes.flow = 'forward')+theme_bw()+
  labs(x = '', y = '') +
  scale_fill_manual(values = c("#F8766C","#E38900",
                               "#C49A00","#99A800","#52B400",
                               "#00BC56","#00C094","#00BFC4","#00B6EB",
                               "#A3C5B8","#363F70","#596C94","#797592","#3B6241"))+
  scale_y_continuous(expand=c(0,0))+geom_text(stat = 'stratum', infer.label = TRUE, size = 5.5) + 
  theme(title = element_text(size = 25),legend.key.size = unit(1,'cm'),
        legend.position = "none",panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.title = element_blank(),axis.text = element_blank(),
        axis.ticks = element_blank(),axis.line = element_blank())


ggarrange(P4,P1,P2,P3,ncol = 4,widths = c(3,2,1,1))

ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/P1.jpeg",width = 17,height = 6,
       dpi = 1800)




#桑葚图绘制


###桑葚图的绘制
View(umap_ctrl_IUA_allsingle@meta.data)

VlnPlot(umap_ctrl_IUA_allsingle, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)

umap_quality_plot=umap_ctrl_IUA_allsingle@meta.data[c("nFeature_RNA","nCount_RNA","type2")]


umap_quality_plot_P1=as.data.frame(table(umap_quality_plot$type2))
table(umap_ctrl_IUA_allsingle$type3)

table(umap_ctrl_IUA_allsingle$type2)

umap_ctrl_IUA_allsingle$type2[umap_ctrl_IUA_allsingle$type2%in%"CTRL"]="Ctrl"
umap_ctrl_IUA_allsingle$type2[umap_ctrl_IUA_allsingle$type2%in%"IUA1"]="Adjacent"
umap_ctrl_IUA_allsingle$type2[umap_ctrl_IUA_allsingle$type2%in%"IUA2"]="Adhesion"




umap_ctrl_IUA_allsingle$type2=factor(umap_ctrl_IUA_allsingle$type2,
                                     levels = c("Ctrl","Adjacent","Adhesion"))

umap_ctrl_IUA_allsingle$type3=factor(umap_ctrl_IUA_allsingle$type3,
                                     levels = c("IUA7","IUA6","IUA5","IUA4",
                                                "IUA3","IUA2","IUA1","CTRL2","CTRL1"))


P1=ggplot(data=umap_quality_plot_P1)+aes(x=Var1,y=Freq,fill=Var1)+geom_bar(stat="identity",width=0.85)+
  scale_y_continuous(expand=c(0,0))+
  theme_bw()+theme(panel.grid.major=element_blank(),panel.grid.minor=element_blank())+
  theme(legend.position="none")+theme(panel.border = element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank(),
                                      axis.ticks.y = element_blank())+  
  theme(axis.line = element_line(size=0.4, colour = "black"))+coord_flip()+
  ylab("cells")+xlab("")+scale_fill_manual(values=c("#3B6241","#797592","#596C94","#363F70","#A3C5B8"))+
  scale_x_discrete(limits=c("Ctrl","Adjacent","Adhesion"))



umap_quality_plot_P2=umap_ctrl_IUA_allsingle@meta.data[c("nFeature_RNA","type2")]

P2=ggplot(data=umap_quality_plot_P2)+aes(x=nFeature_RNA,y=type2,fill=type2)+
  geom_violin()+theme_bw()+ylab("")+theme(legend.position = "none")+xlab("Genes")+
  geom_boxplot(width=.1,col="black",fill="black")+scale_fill_manual(values=c("#A3C5B8","#363F70","#596C94","#797592","#3B6241"))+
  theme(axis.text.y =element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank(),axis.ticks.y = element_blank())






umap_quality_plot_P3=umap_ctrl_IUA_allsingle@meta.data[c("nCount_RNA","type2")]

P3=ggplot(data=umap_quality_plot_P3)+aes(x=nCount_RNA,y=type2,fill=type2)+
  geom_violin()+theme_bw()+ylab("")+theme(legend.position = "none")+xlab("UMI")+
  geom_boxplot(width=.1,col="black",fill="black")+scale_fill_manual(values=c("#A3C5B8","#363F70","#596C94","#797592","#3B6241"))+
  theme(axis.text.y =element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank(),axis.ticks.y = element_blank())


P4_table=as.data.frame(table(umap_ctrl_IUA_allsingle$type3,umap_ctrl_IUA_allsingle$type2))
P4_table=subset(P4_table,P4_table$Freq!=0)
P4_table=P4_table[1:2]

colnames(P4_table)[1]<-"sample"
colnames(P4_table)[2]<-"Group"



P4_table=rbind(P4_table,P4_table[P4_table$sample%in%c("IUA7"),])

P4_table$link <- 1
P4_table <- reshape::melt(P4_table, id = 'link')

variable <- summary(P4_table$variable)
P4_table$flow <- rep(1:variable[1], length(variable))


P4=ggplot(P4_table, aes(x = variable, y = link,
                        stratum = value, alluvium = flow, fill = value)) +
  geom_stratum() +  
  geom_flow(aes.flow = 'forward')+theme_bw()+
  labs(x = '', y = '') +
  scale_fill_manual(values = c("#F8766C","#E38900",
                               "#C49A00","#99A800","#52B400",
                               "#00BC56","#00C094","#00BFC4","#00B6EB",
                               "#A3C5B8","#363F70","#596C94","#797592","#3B6241"))+
  scale_y_continuous(expand=c(0,0))+geom_text(stat = 'stratum', infer.label = TRUE, size = 5.5) + 
  theme(title = element_text(size = 25),legend.key.size = unit(1,'cm'),
        legend.position = "none",panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.title = element_blank(),axis.text = element_blank(),
        axis.ticks = element_blank(),axis.line = element_blank())


P4

ggarrange(P4,P1,P2,P3,ncol = 4,widths = c(3,2,1,1))




#进行原图中的指控的图的绘制2
View(umap_ctrl_IUA_allsingle@meta.data)

VlnPlot(umap_ctrl_IUA_allsingle, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)

umap_quality_plot=umap_ctrl_IUA_allsingle@meta.data[c("nFeature_RNA","nCount_RNA","type5")]


umap_quality_plot_P1=as.data.frame(table(umap_quality_plot$type5))

P1=ggplot(data=umap_quality_plot_P1)+aes(x=Var1,y=Freq,fill=Var1)+geom_bar(stat="identity",width=0.85)+
  scale_y_continuous(expand=c(0,0))+
  theme_bw()+theme(panel.grid.major=element_blank(),panel.grid.minor=element_blank())+
  theme(legend.position="none")+theme(panel.border = element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank(),
                                      axis.ticks.y = element_blank())+  
  theme(axis.line = element_line(size=0.4, colour = "black"),text = element_blank())+coord_flip()+
  scale_x_discrete(limits=c("Ctrl","miIUA-a","miIUA-I","moIUA-a","moIUA-I"))+
  ylab("cells")+xlab("")+scale_fill_manual(values=c("#3B6241","#797592","#596C94","#363F70","#A3C5B8"))




umap_quality_plot_P2=umap_ctrl_IUA_allsingle@meta.data[c("nFeature_RNA","type5")]

P2=ggplot(data=umap_quality_plot_P2)+aes(x=nFeature_RNA,y=type5,fill=type5)+
  geom_violin()+theme_bw()+ylab("")+theme(legend.position = "none",text = element_blank())+xlab("Genes")+
  geom_boxplot(width=.1,col="black",fill="black")+scale_fill_manual(values=c("#A3C5B8","#363F70","#596C94","#797592","#3B6241"))+
  scale_y_discrete(limits=c("Ctrl","miIUA-a","miIUA-I","moIUA-a","moIUA-I"))+
  theme(axis.text.y =element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank(),axis.ticks.y = element_blank())






umap_quality_plot_P3=umap_ctrl_IUA_allsingle@meta.data[c("nCount_RNA","type5")]

P3=ggplot(data=umap_quality_plot_P3)+aes(x=nCount_RNA,y=type5,fill=type5)+
  geom_violin()+theme_bw()+ylab("")+theme(legend.position = "none",text = element_blank())+xlab("UMI")+
  geom_boxplot(width=.1,col="black",fill="black")+scale_fill_manual(values=c("#A3C5B8","#363F70","#596C94","#797592","#3B6241"))+
  scale_y_discrete(limits=c("Ctrl","miIUA-a","miIUA-I","moIUA-a","moIUA-I"))+
  theme(axis.text.y =element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank(),axis.ticks.y = element_blank())




ggarrange(P1,P2,P3,ncol = 3)



###桑葚图的绘制
table(umap_ctrl_IUA_allsingle$type3)

umap_ctrl_IUA_allsingle$type5=factor(umap_ctrl_IUA_allsingle$type5,
                                     levels = c("moIUA-I","moIUA-a","miIUA-I","miIUA-a","Ctrl"))

umap_ctrl_IUA_allsingle$type3=factor(umap_ctrl_IUA_allsingle$type3,
                                     levels = c("IUA7","IUA6","IUA5","IUA4",
                                                "IUA3","IUA2","IUA1","CTRL2","CTRL1"))


P4_table=as.data.frame(table(umap_ctrl_IUA_allsingle$type3,umap_ctrl_IUA_allsingle$type5))
P4_table=subset(P4_table,P4_table$Freq!=0)
P4_table=P4_table[1:2]

colnames(P4_table)[1]<-"sample"
colnames(P4_table)[2]<-"Group"



P4_table=rbind(P4_table,P4_table[P4_table$sample%in%c("IUA7"),])

P4_table$link <- 1
P4_table <- reshape::melt(P4_table, id = 'link')

variable <- summary(P4_table$variable)
P4_table$flow <- rep(1:variable[1], length(variable))





P4=ggplot(P4_table, aes(x = variable, y = link,
                        stratum = value, alluvium = flow, fill = value)) +
  geom_stratum() +  
  geom_flow(aes.flow = 'forward')+theme_bw()+
  labs(x = '', y = '') +
  scale_fill_manual(values = c("#F8766C","#E38900",
                               "#C49A00","#99A800","#52B400",
                               "#00BC56","#00C094","#00BFC4","#00B6EB",
                               "#A3C5B8","#363F70","#596C94","#797592","#3B6241"))+
  scale_y_continuous(expand=c(0,0))+ 
  theme(title = element_text(size = 25),legend.key.size = unit(1,'cm'),
        legend.position = "none",panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.title = element_blank(),axis.text = element_blank(),
        axis.ticks = element_blank(),axis.line = element_blank(),text = element_blank())


ggarrange(P4,P1,P2,P3,ncol = 4,widths = c(3,2,1,1))

ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/P1.jpeg",width = 17,height = 6,
       dpi = 1800)



#cell_type1110的柱形图绘制
umap_ctrl_IUA_allsingle_table2=table(umap_ctrl_IUA_allsingle$type5,umap_ctrl_IUA_allsingle$cell_type1110)
umap_ctrl_IUA_allsingle_table2=as.data.frame(umap_ctrl_IUA_allsingle_table2)

colnames(umap_ctrl_IUA_allsingle_table2)[1]<-"type5"
colnames(umap_ctrl_IUA_allsingle_table2)[2]<-"celltype"
colnames(umap_ctrl_IUA_allsingle_table2)[3]<-"value"

ggplot(data=umap_ctrl_IUA_allsingle_table2,fill=celltype)+aes(x=type5,y=stat(umap_ctrl_IUA_allsingle_table2$value),fill=celltype)+
  geom_bar(stat="identity",width=0.55, position="fill")+theme_bw()+
  theme(legend.title=element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank())+ylab("Ratio")+xlab("celltype")+
  theme(axis.text.x = element_text(angle=45,hjust = 1,vjust = 1))+scale_fill_manual(values=c("#FF8C00","#CAFF70","#8B008B","brown1",
                                                                                             "#7FFF00","#98F5FF","#B4EEB4","#00BFFF","firebrick1",
                                                                                             "gold1","maroon1","#191970","#FFA07A","#8470FF",
                                                                                             "red3"))+theme(text = element_text(size=27))+
  xlab("")+ylab("")+ scale_y_continuous(labels = scales :: percent,expand=c(0,0))+theme(plot.margin=unit(rep(1,4),"lines"))







#对文章2进行绘图复现-----
install.packages("remotes")
remotes::install_github("mojaveazure/seurat-disk",force = TRUE)
library(SeuratDisk)
library(patchwork)
library(Seurat)
library(dplyr)
library(ggplot2)


Convert("/data/DATAprocess/chentw/zigongneimo/WEN2/h5ad文件/endo-2022_lymphocyte.h5ad","h5seurat",
        overwrite=TRUE,assay="RNA") 


lymphocyte<-LoadH5Seurat("/data/DATAprocess/chentw/zigongneimo/WEN2/h5ad文件/endo-2022_lymphocyte.h5seurat",
                         assays="RNA")


DimPlot(lymphocyte,group.by = "subtypes",label = T)





Convert("/data/DATAprocess/chentw/zigongneimo/WEN2/h5ad文件/endo-2022_global.h5ad","h5seurat",
        overwrite=TRUE,assay="RNA") 


global<-LoadH5Seurat("/data/DATAprocess/chentw/zigongneimo/WEN2/h5ad文件/endo-2022_global.h5seurat",
                     assays="RNA")

DimPlot(global,group.by = "celltype_main",label = T)


VlnPlot(global,features = "COL6A1",group.by = "celltype_main",pt.size = 0)


VlnPlot(global,features = "COL1A1",group.by = "celltype_main",pt.size = 0)


VlnPlot(global,features = "COL1A1",group.by = "celltype_main",pt.size = 0.1)


DimPlot(global,group.by = "sample_type_rename",label = T)


DimPlot(global,group.by = "celltype_main",label = T,split.by = "sample_type_rename")



#将文章的基质细胞尝试抽出来进行harmony整合-----
global<-LoadH5Seurat("/data/DATAprocess/chentw/zigongneimo/WEN2/h5ad文件/endo-2022_global.h5seurat",
                     assays="RNA")

DimPlot(global,group.by = "celltype_main",label = T)

DimPlot(global,group.by = "celltype_main",label = T,split.by = "celltype_main")

WEN2_STR=subset(global,celltype_main%in%"stromal")


STR.list <- SplitObject(WEN2_STR, split.by="PID")


STR=merge(STR.list[[1]],y = c(STR.list[[2]],STR.list[[3]],STR.list[[4]],STR.list[[5]],
                              STR.list[[6]],STR.list[[7]],STR.list[[8]],STR.list[[9]],
                              STR.list[[10]],STR.list[[11]],STR.list[[12]],STR.list[[13]],
                              STR.list[[14]]))

mtGenes=grep('^MT-',rownames(STR),value=TRUE)
RPGenes=grep('^RP',rownames(STR),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes

stress_gene=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
              "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene=c(mtGenes,RPGenes,HB.genes,g2m.genes,stress_gene,s.genes)


###尝试harmony

STR <- NormalizeData(STR, normalization.method = "LogNormalize", 
                     scale.factor = 10000)
STR <- FindVariableFeatures(STR, selection.method = "vst", 
                            nfeatures = 2000)

STR@assays$RNA@var.features = setdiff(STR@assays$RNA@var.features,subgene)  #去除subgene影响



#STR进行标准化然后PCA分析
DefaultAssay(STR) <- "RNA"
all.genes_STR <- rownames(STR)
STR <- ScaleData(object = STR, features = all.genes_STR)
STR <- RunPCA(STR, features = VariableFeatures(object = STR))

#确定STR的PC维数
STR <- JackStraw(STR, num.replicate = 50) 
STR <- ScoreJackStraw(STR, dims = 1:20) 
ElbowPlot(STR,ndims = 30, reduction = "pca") 


#STR进行细胞细胞聚类分析(UMAP)
STR=RunUMAP(STR, dims = 1:30)
DimPlot(STR,reduction="umap",label=TRUE)
STR <- FindNeighbors(STR, dims = 1:30)
STR <- FindClusters(STR, resolution = 0.15)
STR <- FindClusters(STR, resolution = 0.3)
STR <- FindClusters(STR, resolution = 0.45)
STR <- FindClusters(STR, resolution = 0.6)

DimPlot(STR,reduction="umap",label=TRUE)
DimPlot(STR,reduction="umap",label=TRUE,group.by = "RNA_snn_res.0.15")
DimPlot(STR,reduction="umap",label=TRUE,group.by = "RNA_snn_res.0.3")

STR$type3=umap_WEN2_STR_sub$type3

DimPlot(STR,reduction="umap",label=TRUE,group.by = "PID")


#进行harmony
STR_harmony=STR%>%RunHarmony("PID",plot_covergence=T)

STR_harmony=RunUMAP(STR_harmony,reduction="harmony",dims=1:30)
STR_harmony=FindNeighbors(STR_harmony,dims=1:30,reduction = "harmony")
STR_harmony=FindClusters(STR_harmony, resolution = 0.15,reduction = "harmony")
STR_harmony=FindClusters(STR_harmony, resolution = 0.3,reduction = "harmony")
STR_harmony=FindClusters(STR_harmony, resolution = 0.6,reduction = "harmony")

DimPlot(STR_harmony,reduction = "umap",group.by = "RNA_snn_res.0.3",label = T)
DimPlot(STR_harmony,reduction = "harmony",group.by = "RNA_snn_res.0.3",label = T)



DimPlot(STR_harmony,reduction = "umap",group.by = "RNA_snn_res.0.3",label = T)
DimPlot(STR_harmony,reduction = "umap",group.by = "RNA_snn_res.0.6",label = T)

DimPlot(STR_harmony,reduction = "umap",group.by = "PID",label = T)


#对各个cluster进行marker基因寻找
STR_harmony@active.ident=STR_harmony$RNA_snn_res.0.6
STR_harmony_0.6_markers <- FindAllMarkers(object = STR_harmony, only.pos = TRUE,min.pct = 0.25, 
                                          logfc.threshold = 0.25)

STR_harmony_0.6_markers=subset(STR_harmony_0.6_markers,STR_harmony_0.6_markers$p_val_adj<0.01)


saveRDS(STR_harmony,"/data/DATAprocess/chentw/zigongneimo/rds文件存放点/STR_harmony_WEN2.rds")





STR_harmony_0.6_markers_top5 <- STR_harmony_0.6_markers %>% group_by(cluster) %>% top_n(n = 5, wt = avg_log2FC)

DotPlot(STR_harmony,group.by = "RNA_snn_res.0.6",features = unique(STR_harmony_0.6_markers_top5$gene))+
  theme(axis.text.x = element_text(angle=45,hjust = 1,vjust = 1))



#尝试对WEN2_STR进行integrated-----
global<-LoadH5Seurat("/data/DATAprocess/chentw/zigongneimo/WEN2/h5ad文件/endo-2022_global.h5seurat",
                     assays="RNA")

DimPlot(global,group.by = "celltype_main",label = T)

DimPlot(global,group.by = "celltype_main",label = T,split.by = "celltype_main")

WEN2_STR=subset(global,celltype_main%in%"stromal")



mtGenes=grep('^MT-',rownames(STR),value=TRUE)
RPGenes=grep('^RP',rownames(STR),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes

stress_gene=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
              "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene=c(mtGenes,RPGenes,HB.genes,g2m.genes,stress_gene,s.genes)



WEN2_STR.list <- SplitObject(WEN2_STR, split.by="PID")
WEN2_STR.list <- lapply(X = WEN2_STR.list, FUN = function(x) {
  x <- NormalizeData(x)
  x <- FindVariableFeatures(x, selection.method = "vst", nfeatures = 2000)
})

k.fliter=min(200,min(sapply(WEN2_STR.list,ncol)))
WEN2_STR <- FindIntegrationAnchors(object.list = WEN2_STR.list, dims = 1:30,k.filter=k.fliter)
WEN2_STR <- IntegrateData(anchorset = WEN2_STR, dims = 1:30,k.weight=k.fliter)

#进行标准化并进行PCA
DefaultAssay(WEN2_STR) <- "RNA"
all.genes.WEN2_STR <- rownames(WEN2_STR)
WEN2_STR <- ScaleData(object = WEN2_STR, features = all.genes.WEN2_STR)

DefaultAssay(WEN2_STR) <- "integrated"
all.genes.WEN2_STR1 <- rownames(WEN2_STR)
WEN2_STR <- ScaleData(object = WEN2_STR, features = all.genes.WEN2_STR1)
WEN2_STR <- RunPCA(WEN2_STR, features = VariableFeatures(object = WEN2_STR))
DimPlot(WEN2_STR,label = T)

#定义pca维数
WEN2_STR <- JackStraw(WEN2_STR, num.replicate = 100) 
WEN2_STR <- ScoreJackStraw(WEN2_STR, dims = 1:30) 
ElbowPlot(WEN2_STR,ndims = 30, reduction = "pca") 

#进行WEN2_STR细胞细胞聚类分析(先runumap降维确定形状再确定cluster)
#进行细胞细胞聚类分析(WEN2_STR)
DefaultAssay(WEN2_STR) <- "integrated"
WEN2_STR=RunUMAP(WEN2_STR, dims = 1:30)
DimPlot(WEN2_STR,reduction="umap",label=TRUE)
WEN2_STR <- FindNeighbors(WEN2_STR, dims = 1:30)
WEN2_STR <- FindClusters(WEN2_STR, resolution = 0.3)
WEN2_STR <- FindClusters(WEN2_STR, resolution = 0.15)

WEN2_STR <- FindClusters(WEN2_STR, resolution = 0.6)
WEN2_STR <- FindClusters(WEN2_STR, resolution = 0.45)
DefaultAssay(WEN2_STR)<-"RNA"




DimPlot(WEN2_STR,reduction="umap",label=TRUE,group.by = "integrated_snn_res.0.45")






#WEN2_STR进行细胞细胞聚类分析(tsne)
WEN2_STR=RunTSNE(WEN2_S,.TR, dims = 1:30)
DimPlot(WEN2_STR,reduction="tsne",label=TRUE)
WEN2_STR<- FindNeighbors(WEN2_STR, dims = 1:30)
WEN2_STR <- FindClusters(WEN2_STR, resolution = 0.3)
WEN2_STR <- FindClusters(WEN2_STR, resolution = 0.15)
DimPlot(WEN2_STR,reduction="tsne",label=TRUE)



WEN2_STR_0.45_markers <- FindAllMarkers(object = WEN2_STR, only.pos = TRUE,min.pct = 0.25, 
                                        logfc.threshold = 0.25)

WEN2_STR_0.45_markers=subset(WEN2_STR_0.45_markers,WEN2_STR_0.45_markers$p_val_adj<0.01)


###TOP5marker基因气泡图绘制

WEN2_STR_0.45_markers_top5 <- WEN2_STR_0.45_markers %>% group_by(cluster) %>% top_n(n = 5, wt = avg_log2FC)


DotPlot(WEN2_STR,group.by = "integrated_snn_res.0.45",features = WEN2_STR_0.45_markers_top5$gene)+
  theme(axis.text.x = element_text(angle=45,hjust = 1,vjust = 1))



WEN2_STR_0.45_markers_top3 <- WEN2_STR_0.45_markers %>% group_by(cluster) %>% top_n(n = 3, wt = avg_log2FC)


DotPlot(WEN2_STR,group.by = "integrated_snn_res.0.45",features = WEN2_STR_0.45_markers_top3$gene)+
  theme(axis.text.x = element_text(angle=45,hjust = 1,vjust = 1))




#对marker基因进行富集分析
GO_STR=function(X){
  N=as.character(X)
  N=subset(WEN2_STR_0.45_markers,WEN2_STR_0.45_markers$cluster%in%N)
  N=N$gene
  G <- bitr(N, fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
  G=G$ENTREZID
  ego <- enrichGO(OrgDb="org.Hs.eg.db", gene = G, ont = "ALL",readable= TRUE)
  file1=paste("/data/DATAprocess/chentw/zigongneimo/python运行/WEN2_STR1/res0.45marker基因富集结果/",as.character(X),"_ego.csv",sep = "")
  print(file1)
  write.csv(ego,file1)
}
for (i in 0:13) {GO_STR(i)}

#挑前3富集通路进行绘图-气泡图

ego0=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN2_STR1/res0.45marker基因富集结果/0_ego.csv")
ego1=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN2_STR1/res0.45marker基因富集结果/1_ego.csv")
ego2=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN2_STR1/res0.45marker基因富集结果/2_ego.csv")
ego3=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN2_STR1/res0.45marker基因富集结果/3_ego.csv")
ego4=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN2_STR1/res0.45marker基因富集结果/4_ego.csv")
ego5=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN2_STR1/res0.45marker基因富集结果/5_ego.csv")
ego6=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN2_STR1/res0.45marker基因富集结果/6_ego.csv")
ego7=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN2_STR1/res0.45marker基因富集结果/7_ego.csv")
ego8=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN2_STR1/res0.45marker基因富集结果/8_ego.csv")
ego9=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN2_STR1/res0.45marker基因富集结果/9_ego.csv")
ego10=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN2_STR1/res0.45marker基因富集结果/10_ego.csv")
ego11=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN2_STR1/res0.45marker基因富集结果/11_ego.csv")
ego12=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN2_STR1/res0.45marker基因富集结果/12_ego.csv")
ego13=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN2_STR1/res0.45marker基因富集结果/13_ego.csv")


ego0=ego0[1:3,]
ego1=ego1[1:3,]
ego2=ego2[1:3,]
ego3=ego3[1:3,]
ego4=ego4[1:3,]
ego5=ego5[1:3,]
ego6=ego6[1:3,]
ego7=ego7[1:3,]
ego8=ego8[1:3,]
ego9=ego9[1:3,]
ego10=ego10[1:3,]
ego11=ego11[1:3,]
ego12=ego12[1:3,]
ego13=ego13[1:3,]


ego0$cluster="0"
ego1$cluster="1"
ego2$cluster="2"
ego3$cluster="3"
ego4$cluster="4"
ego5$cluster="5"
ego6$cluster="6"
ego7$cluster="7"
ego8$cluster="8"
ego9$cluster="9"
ego10$cluster="10"
ego11$cluster="11"
ego12$cluster="12"
ego13$cluster="13"




ego_all=rbind(ego0,ego1,ego2,ego3,ego4,ego5,
              ego6,ego7,ego8,ego9,ego10,ego11,
              ego12,ego13)

ego_all$cluster=factor(ego_all$cluster,levels = 0:13)


ggplot(data=ego_all)+aes(x=cluster,y=Description,color=-log10(pvalue))+geom_point(aes(size=Count,color= -log10(pvalue)))+
  scale_y_discrete(limits=unique(ego_all$Description))+theme_bw()+xlab("Cluster")+
  scale_color_gradient(low="blue",high = "red")









#对文章分离出来的基质细胞进行操作-----
WEN_STR<-LoadH5Seurat("/data/DATAprocess/chentw/zigongneimo/WEN2/h5ad文件/endo-2022_stromal.h5seurat",
                      assays="RNA")

#进行数据cluster的查找并进行marker基因的查找
DimPlot(WEN_STR,reduction="umap",label=TRUE)
DimPlot(WEN_STR,group.by = "PID")




WEN_STR <- FindNeighbors(WEN_STR, dims = 1:30)
WEN_STR <- FindClusters(WEN_STR, resolution = 0.3)
WEN_STR <- FindClusters(WEN_STR, resolution = 0.15)

WEN_STR <- FindClusters(WEN_STR, resolution = 0.6)
WEN_STR <- FindClusters(WEN_STR, resolution = 0.45)
DefaultAssay(WEN_STR)<-"RNA"
DimPlot(WEN_STR,reduction="umap",label=TRUE)


#对12个cluster进行marker基因的寻找
table(WEN_STR$RNA_snn_res.0.45)
table(WEN_STR$RNA_snn_res.0.6)

WEN_STR$RNA_snn_res.0.45=factor(WEN_STR$RNA_snn_res.0.45,levels = 0:11)

WEN_STR$RNA_snn_res.0.6=factor(WEN_STR$RNA_snn_res.0.6,levels = 0:14)

DimPlot(WEN_STR,reduction="umap",label=TRUE,group.by = "RNA_snn_res.0.45")


WEN_STR@active.ident=WEN_STR$RNA_snn_res.0.45

WEN_STR_0.45_markers <- FindAllMarkers(object = WEN_STR, only.pos = TRUE,min.pct = 0.25, 
                                       logfc.threshold = 0.25)

WEN_STR_0.45_markers=subset(WEN_STR_0.45_markers,WEN_STR_0.45_markers$p_val_adj<0.01)



WEN_STR_0.45_markers_top5 <- WEN_STR_0.45_markers %>% group_by(cluster) %>% top_n(n = 5, wt = avg_log2FC)


DotPlot(WEN_STR,group.by = "RNA_snn_res.0.45",features = unique(WEN_STR_0.45_markers_top5$gene))+
  theme(axis.text.x = element_text(angle=45,hjust = 1,vjust = 1))



#对marker基因进行富集分析
GO_STR=function(X){
  N=as.character(X)
  N=subset(WEN_STR_0.45_markers,WEN_STR_0.45_markers$cluster%in%N)
  N=N$gene
  G <- bitr(N, fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
  G=G$ENTREZID
  ego <- enrichGO(OrgDb="org.Hs.eg.db", gene = G, ont = "ALL",readable= TRUE)
  file1=paste("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/res0.45的marker基因的富集分析结果/",as.character(X),"_ego.csv",sep = "")
  print(file1)
  write.csv(ego,file1)
}
for (i in 0:11) {GO_STR(i)}

#挑前3富集通路进行绘图-气泡图
ego0=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/res0.45的marker基因的富集分析结果/0_ego.csv")
ego1=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/res0.45的marker基因的富集分析结果/1_ego.csv")
ego2=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/res0.45的marker基因的富集分析结果/2_ego.csv")
ego3=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/res0.45的marker基因的富集分析结果/3_ego.csv")
ego4=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/res0.45的marker基因的富集分析结果/4_ego.csv")
ego5=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/res0.45的marker基因的富集分析结果/5_ego.csv")
ego6=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/res0.45的marker基因的富集分析结果/6_ego.csv")
ego7=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/res0.45的marker基因的富集分析结果/7_ego.csv")
ego8=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/res0.45的marker基因的富集分析结果/8_ego.csv")
ego9=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/res0.45的marker基因的富集分析结果/9_ego.csv")
ego10=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/res0.45的marker基因的富集分析结果/10_ego.csv")
ego11=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/res0.45的marker基因的富集分析结果/11_ego.csv")


ego0=ego0[1:3,]
ego1=ego1[1:3,]
ego2=ego2[1:3,]
ego3=ego3[1:3,]
ego4=ego4[1:3,]
ego5=ego5[1:3,]
ego6=ego6[1:3,]
ego7=ego7[1:3,]
ego8=ego8[1:3,]
ego9=ego9[1:3,]
ego10=ego10[1:3,]
ego11=ego11[1:3,]

ego0$cluster="0"
ego1$cluster="1"
ego2$cluster="2"
ego3$cluster="3"
ego4$cluster="4"
ego5$cluster="5"
ego6$cluster="6"
ego7$cluster="7"
ego8$cluster="8"
ego9$cluster="9"
ego10$cluster="10"
ego11$cluster="11"




ego_all=rbind(ego0,ego1,ego2,ego3,ego4,ego5,
              ego6,ego7,ego8,ego9,ego10,ego11)

ego_all$cluster=factor(ego_all$cluster,levels = 0:11)


ggplot(data=ego_all)+aes(x=cluster,y=Description,color=-log10(pvalue))+geom_point(aes(size=Count,color= -log10(pvalue)))+
  scale_y_discrete(limits=unique(ego_all$Description))+theme_bw()+xlab("Cluster")+
  scale_color_gradient(low="blue",high = "red")




#将文章的基质细胞和我们宫腔粘连项目的重新过滤的双细胞的基质细胞进行整合并进行marker基因查找和观测
WEN_STR<-LoadH5Seurat("/data/DATAprocess/chentw/zigongneimo/WEN2/h5ad文件/endo-2022_stromal.h5seurat",
                      assays="RNA")

#进行数据cluster的查找并进行marker基因的查找
DimPlot(WEN_STR,reduction="umap",label=TRUE)
WEN_STR <- FindNeighbors(WEN_STR, dims = 1:30)
WEN_STR <- FindClusters(WEN_STR, resolution = 0.3)
WEN_STR <- FindClusters(WEN_STR, resolution = 0.15)

WEN_STR <- FindClusters(WEN_STR, resolution = 0.6)
WEN_STR <- FindClusters(WEN_STR, resolution = 0.45)
DefaultAssay(WEN_STR)<-"RNA"
DimPlot(WEN_STR,reduction="umap",label=TRUE)


FeaturePlot(WEN_STR,features = stress_gene)





STR_harmony=readRDS("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/STR_harmony.rds")

DimPlot(STR_harmony)

WEN_STR$type3=WEN_STR$PID

table(STR_harmony$type3)

STR_WEN2_all=merge(WEN_STR,STR_harmony)

STR_WEN2_all.list <- SplitObject(STR_WEN2_all, split.by="type3")
STR_WEN2_all.list <- lapply(X = STR_WEN2_all.list, FUN = function(x) {
  x <- NormalizeData(x)
  x <- FindVariableFeatures(x, selection.method = "vst", nfeatures = 2000)
})

k.fliter=min(200,min(sapply(STR_WEN2_all.list,ncol)))
STR_WEN2_all <- FindIntegrationAnchors(object.list = STR_WEN2_all.list, dims = 1:30,k.filter=k.fliter)
STR_WEN2_all <- IntegrateData(anchorset = STR_WEN2_all, dims = 1:30,k.weight=k.fliter)

#进行标准化并进行PCA
DefaultAssay(STR_WEN2_all) <- "RNA"
all.genes.STR_WEN2_all <- rownames(STR_WEN2_all)
STR_WEN2_all <- ScaleData(object = STR_WEN2_all, features = all.genes.STR_WEN2_all)

DefaultAssay(STR_WEN2_all) <- "integrated"
all.genes.STR_WEN2_all1 <- rownames(STR_WEN2_all)
STR_WEN2_all <- ScaleData(object = STR_WEN2_all, features = all.genes.STR_WEN2_all1)
STR_WEN2_all <- RunPCA(STR_WEN2_all, features = VariableFeatures(object = STR_WEN2_all))
DimPlot(STR_WEN2_all,label = T)

#定义pca维数
STR_WEN2_all <- JackStraw(STR_WEN2_all, num.replicate = 100) 
STR_WEN2_all <- ScoreJackStraw(STR_WEN2_all, dims = 1:30) 
ElbowPlot(STR_WEN2_all,ndims = 30, reduction = "pca") 

#进行STR_WEN2_all细胞细胞聚类分析(先runumap降维确定形状再确定cluster)
#进行细胞细胞聚类分析(STR_WEN2_all)
DefaultAssay(STR_WEN2_all) <- "integrated"
STR_WEN2_all=RunUMAP(STR_WEN2_all, dims = 1:30)
DimPlot(STR_WEN2_all,reduction="umap",label=TRUE)
STR_WEN2_all <- FindNeighbors(STR_WEN2_all, dims = 1:30)
STR_WEN2_all <- FindClusters(STR_WEN2_all, resolution = 0.3)
STR_WEN2_all <- FindClusters(STR_WEN2_all, resolution = 0.15)

STR_WEN2_all <- FindClusters(STR_WEN2_all, resolution = 0.6)
STR_WEN2_all <- FindClusters(STR_WEN2_all, resolution = 0.45)
STR_WEN2_all <- FindClusters(STR_WEN2_all, resolution = 0.05)
DefaultAssay(STR_WEN2_all)<-"RNA"








#STR_WEN2_all进行细胞细胞聚类分析(tsne)
STR_WEN2_all=RunTSNE(WEN2_S,.TR, dims = 1:30)
DimPlot(STR_WEN2_all,reduction="tsne",label=TRUE)
STR_WEN2_all<- FindNeighbors(STR_WEN2_all, dims = 1:30)
STR_WEN2_all <- FindClusters(STR_WEN2_all, resolution = 0.3)
STR_WEN2_all <- FindClusters(STR_WEN2_all, resolution = 0.15)
STR_WEN2_all <- FindClusters(STR_WEN2_all, resolution = 0.05)
DimPlot(STR_WEN2_all,reduction="tsne",label=TRUE)

saveRDS(STR_WEN2_all,"/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/STR_WEN2_all.rds")


STR_WEN2_all$integrated_snn_res.0.15=factor(STR_WEN2_all$integrated_snn_res.0.15,levels = 0:17)

DimPlot(STR_WEN2_all,reduction="umap",label=TRUE,group.by = "integrated_snn_res.0.15")

table(STR_WEN2_all$type3)

STR_WEN2_all$type1225="WEN"

STR_WEN2_all$type1225[STR_WEN2_all$type3%in%c("CTRL1","CTRL2","IUA1","IUA2","IUA3","IUA4",
                                              "IUA5","IUA6","IUA7")]<-"NEW"

STR_WEN2_all$celltype=STR_WEN2_all$subtypes

STR_WEN2_all$celltype[STR_WEN2_all$type1225%in%"NEW"]<-STR_WEN2_all$integrated_snn_res.0.3[STR_WEN2_all$type1225%in%"NEW"]


DimPlot(STR_WEN2_all,reduction="umap",label=TRUE,group.by = "celltype",split.by = "type1225")


STR_WEN2=subset(STR_WEN2_all,type1225%in%"WEN")



DimPlot(STR_WEN2,reduction="umap",label=TRUE,group.by = "subtypes")
DimPlot(STR_WEN2,reduction="umap",label=TRUE,group.by = "sample_type_rename")




#####将STR_WEN2_all2进行harmony的整合看看
WEN_STR<-LoadH5Seurat("/data/DATAprocess/chentw/zigongneimo/WEN2/h5ad文件/endo-2022_stromal.h5seurat",
                      assays="RNA")

#进行数据cluster的查找并进行marker基因的查找
DimPlot(WEN_STR,reduction="umap",label=TRUE)
WEN_STR <- FindNeighbors(WEN_STR, dims = 1:30)
WEN_STR <- FindClusters(WEN_STR, resolution = 0.3)
WEN_STR <- FindClusters(WEN_STR, resolution = 0.15)

WEN_STR <- FindClusters(WEN_STR, resolution = 0.6)
WEN_STR <- FindClusters(WEN_STR, resolution = 0.45)
DefaultAssay(WEN_STR)<-"RNA"
DimPlot(WEN_STR,reduction="umap",label=TRUE)



STR_harmony=readRDS("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/STR_harmony.rds")

DimPlot(STR_harmony)

WEN_STR$type3=WEN_STR$PID

table(STR_harmony$type3)

STR_WEN2_all2=merge(WEN_STR,STR_harmony)





mtGenes=grep('^MT-',rownames(STR),value=TRUE)
RPGenes=grep('^RP',rownames(STR),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
stress_gene=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
              "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene=c(mtGenes,RPGenes,HB.genes,g2m.genes,stress_gene,s.genes)


###尝试harmony
STR_WEN2_all2 <- NormalizeData(STR_WEN2_all2, normalization.method = "LogNormalize", 
                               scale.factor = 10000)
STR_WEN2_all2 <- FindVariableFeatures(STR_WEN2_all2, selection.method = "vst", 
                                      nfeatures = 2000)

STR_WEN2_all2@assays$RNA@var.features = setdiff(STR_WEN2_all2@assays$RNA@var.features,subgene)  #去除subgene影响



#STR_WEN2_all2进行标准化然后PCA分析
DefaultAssay(STR_WEN2_all2) <- "RNA"
all.genes_STR_WEN2_all2 <- rownames(STR_WEN2_all2)
STR_WEN2_all2 <- ScaleData(object = STR_WEN2_all2, features = all.genes_STR_WEN2_all2)
STR_WEN2_all2 <- RunPCA(STR_WEN2_all2, features = VariableFeatures(object = STR_WEN2_all2))

#确定STR_WEN2_all2的PC维数
STR_WEN2_all2 <- JackStraw(STR_WEN2_all2, num.replicate = 50) 
STR_WEN2_all2 <- ScoreJackStraw(STR_WEN2_all2, dims = 1:20) 
ElbowPlot(STR_WEN2_all2,ndims = 30, reduction = "pca") 


#STR_WEN2_all2进行细胞细胞聚类分析(UMAP)
STR_WEN2_all2=RunUMAP(STR_WEN2_all2, dims = 1:30)
DimPlot(STR_WEN2_all2,reduction="umap",label=TRUE)
STR_WEN2_all2 <- FindNeighbors(STR_WEN2_all2, dims = 1:30)
STR_WEN2_all2 <- FindClusters(STR_WEN2_all2, resolution = 0.15)
STR_WEN2_all2 <- FindClusters(STR_WEN2_all2, resolution = 0.3)
STR_WEN2_all2 <- FindClusters(STR_WEN2_all2, resolution = 0.45)
STR_WEN2_all2 <- FindClusters(STR_WEN2_all2, resolution = 0.6)

DimPlot(STR_WEN2_all2,reduction="umap",label=TRUE)
DimPlot(STR_WEN2_all2,reduction="umap",label=TRUE,group.by = "RNA_snn_res.0.15")
DimPlot(STR_WEN2_all2,reduction="umap",label=TRUE,group.by = "RNA_snn_res.0.3")
DimPlot(STR_WEN2_all2,reduction="umap",label=TRUE,group.by = "RNA_snn_res.0.05")
STR_WEN2_all2$type3=umap_WEN2_STR_WEN2_all2$type3

DimPlot(STR_WEN2_all2,reduction="umap",label=TRUE,group.by = "type3")


saveRDS(STR_WEN2_all2,"/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/STR_WEN2_all2.rds")

STR_WEN2_all2=readRDS("/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/STR_WEN2_all2.rds")



STR_WEN2_all2_harmony=STR_WEN2_all2%>%RunHarmony("type3",plot_covergence=T)

STR_WEN2_all2_harmony=RunUMAP(STR_WEN2_all2_harmony,reduction="harmony",dims=1:30)
STR_WEN2_all2_harmony=FindNeighbors(STR_WEN2_all2_harmony,dims=1:30,reduction = "harmony")
STR_WEN2_all2_harmony=FindClusters(STR_WEN2_all2_harmony, resolution = 0.15,reduction = "harmony")
STR_WEN2_all2_harmony=FindClusters(STR_WEN2_all2_harmony, resolution = 0.3,reduction = "harmony")
STR_WEN2_all2_harmony=FindClusters(STR_WEN2_all2_harmony, resolution = 0.6,reduction = "harmony")
STR_WEN2_all2_harmony=FindClusters(STR_WEN2_all2_harmony, resolution = 0.05,reduction = "harmony")
DimPlot(STR_WEN2_all2_harmony,reduction = "umap",group.by = "RNA_snn_res.0.3",label = T)
DimPlot(STR_WEN2_all2_harmony,reduction = "harmony",group.by = "RNA_snn_res.0.3",label = T)



DimPlot(STR_WEN2_all2_harmony,reduction = "umap",group.by = "RNA_snn_res.0.05",label = T)

saveRDS(STR_WEN2_all2_harmony,"/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/STR_WEN2_all2_harmony.rds")




STR_WEN2_all2_harmony$type1225="WEN"

STR_WEN2_all2_harmony$type1225[STR_WEN2_all2_harmony$type3%in%c("CTRL1","CTRL2","IUA1","IUA2","IUA3","IUA4",
                                                                "IUA5","IUA6","IUA7")]<-"NEW"

STR_WEN2_all2_harmony$celltype=STR_WEN2_all2_harmony$subtypes

STR_WEN2_all2_harmony$celltype[STR_WEN2_all2_harmony$type1225%in%"NEW"]<-STR_WEN2_all2_harmony$RNA_snn_res.0.3[STR_WEN2_all2_harmony$type1225%in%"NEW"]


DimPlot(STR_WEN2_all2_harmony,reduction="umap",label=TRUE,group.by = "celltype",split.by = "type1225")


STR_WEN2=subset(STR_WEN2_all2_harmony,type1225%in%"WEN")



DimPlot(STR_WEN2,reduction="umap",label=TRUE,group.by = "subtypes")

table(STR_WEN2$PID,STR_WEN2$subtypes)
DimPlot(STR_WEN2,reduction="umap",label=TRUE,group.by = "sample_type_rename")







#将基质细胞抽出来进行重新整合2【去除部分离散细胞，整合以后靠近免疫细胞的部分】----
umap_WEN2=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_WEN2.rds")

DimPlot(umap_WEN2,split.by = "type1124")




umap_WEN2$type1124="WEN2"

umap_WEN2$type1124[cell_type1122%in%"DE_Stromal"]="UMAP"

umap_WEN2_STR=subset(umap_WEN2,cell_type1122%in%"DE_Stromal")
DimPlot(umap_WEN2_STR)

#根据umap空间位置取出细胞
DimPlot(umap_WEN2_STR,group.by = "cell_type1122")


umap_WEN2_STR_embeddings=umap_WEN2_STR$umap@cell.embeddings
umap_WEN2_STR_embeddings=as.data.frame(umap_WEN2_STR_embeddings)
umap_WEN2_STR_embeddings$cell_type="cell_type"
umap_WEN2_STR_embeddings$cell_type=umap_WEN2_STR$cell_type
umap_WEN2_STR_embeddings$UMAP_1=as.numeric(as.character(umap_WEN2_STR_embeddings$UMAP_1))
umap_WEN2_STR_embeddings$UMAP_2=as.numeric(as.character(umap_WEN2_STR_embeddings$UMAP_2))

umap_WEN2_STR_sub=subset(umap_WEN2_STR,UMAP_1>-1.5&UMAP_2<0)


umap_WEN2_STR$sub="res"
umap_WEN2_STR$sub[colnames(umap_WEN2_STR)%in%colnames(umap_WEN2_STR_sub)]="sub"


DimPlot(umap_WEN2_STR,group.by = "sub")


umap_WEN2_STR_sub=subset(umap_WEN2_STR,sub%in%"res")

STR.list <- SplitObject(umap_WEN2_STR_sub, split.by = "type3")

DimPlot(umap_WEN2_STR_sub,group.by = "sub")


STR=merge(STR.list[[1]],y = c(STR.list[[2]],STR.list[[3]],STR.list[[4]],
                              STR.list[[5]],
                              STR.list[[6]],STR.list[[7]],STR.list[[8]],
                              STR.list[[9]]))

mtGenes=grep('^MT-',rownames(STR),value=TRUE)
RPGenes=grep('^RP',rownames(STR),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
stress_gene=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
              "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene=c(mtGenes,RPGenes,HB.genes,g2m.genes,stress_gene,s.genes)



stress_genes=adata.var_names.str.startswith('FOSB','FOS','JUN','JUNB','JUND','ATF3','EGR1','HSPA1A','HSPA1B','HSP90AB1','HSPA8','HSPB1','IER3','IER2','BTG1','BTG2','DUSP1')





###尝试harmony
STR <- CreateSeuratObject(counts = STR@assays$RNA@counts, 
                          project = "STR", min.cells = 3)

STR =STR[!rownames(STR) %in% subgene,]


STR <- NormalizeData(STR, normalization.method = "LogNormalize", 
                     scale.factor = 10000)
STR <- FindVariableFeatures(STR, selection.method = "vst", 
                            nfeatures = 2000)

STR@assays$RNA@var.features = setdiff(STR@assays$RNA@var.features,subgene)  #去除subgene影响



#STR进行标准化然后PCA分析
DefaultAssay(STR) <- "RNA"
all.genes_STR <- rownames(STR)
STR <- ScaleData(object = STR, features = all.genes_STR)
STR <- RunPCA(STR, features = VariableFeatures(object = STR))

#确定STR的PC维数
STR <- JackStraw(STR, num.replicate = 50) 
STR <- ScoreJackStraw(STR, dims = 1:20) 
ElbowPlot(STR,ndims = 30, reduction = "pca") 


#STR进行细胞细胞聚类分析(UMAP)
STR=RunUMAP(STR, dims = 1:30)
DimPlot(STR,reduction="umap",label=TRUE)
STR <- FindNeighbors(STR, dims = 1:30)
STR <- FindClusters(STR, resolution = 0.15)
STR <- FindClusters(STR, resolution = 0.3)
STR <- FindClusters(STR, resolution = 0.45)
STR <- FindClusters(STR, resolution = 0.6)

DimPlot(STR,reduction="umap",label=TRUE)
DimPlot(STR,reduction="umap",label=TRUE,group.by = "RNA_snn_res.0.15")
DimPlot(STR,reduction="umap",label=TRUE,group.by = "RNA_snn_res.0.3")

STR$type3=umap_WEN2_STR_sub$type3

DimPlot(STR,reduction="umap",label=TRUE,group.by = "type3")


saveRDS(STR,"/data/DATAprocess/chentw/zigongneimo/原图绘制/rds/STR_harmony.rds")

STR=readRDS("/data/DATAprocess/chentw/zigongneimo/原图绘制/rds/STR_harmony.rds")




STR_harmony=STR%>%RunHarmony("type3",plot_covergence=T)

STR_harmony=RunUMAP(STR_harmony,reduction="harmony",dims=1:30)
STR_harmony=FindNeighbors(STR_harmony,dims=1:30,reduction = "harmony")
STR_harmony=FindClusters(STR_harmony, resolution = 0.15,reduction = "harmony")
STR_harmony=FindClusters(STR_harmony, resolution = 0.3,reduction = "harmony")
STR_harmony=FindClusters(STR_harmony, resolution = 0.6,reduction = "harmony")

DimPlot(STR_harmony,reduction = "umap",group.by = "RNA_snn_res.0.3",label = T)
DimPlot(STR_harmony,reduction = "harmony",group.by = "RNA_snn_res.0.3",label = T)



DimPlot(STR_harmony,reduction = "umap",group.by = "RNA_snn_res.0.3",label = T)

saveRDS(STR_harmony,"/data/DATAprocess/chentw/zigongneimo/原图绘制/rds/STR_harmony.rds")


STR_harmony=readRDS("/data/DATAprocess/chentw/zigongneimo/原图绘制/rds/STR_harmony.rds")

DimPlot(STR_harmony,reduction = "umap",group.by = "RNA_snn_res.0.3",label = T)
DimPlot(STR_harmony,reduction = "umap",group.by = "type3",label = T)

#查看细胞周期的影响
s.genes=Seurat::cc.genes.updated.2019$s.genes
g2m.genes=Seurat::cc.genes.updated.2019$g2m.genes
STR_harmony <- CellCycleScoring(STR_harmony, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
DefaultAssay(STR_harmony)<-"RNA"
DimPlot(STR_harmony,group.by = "Phase",reduction = "umap")



#进行marker基因的查找-方法1-findallmarkers-res0.15并进行富集
STR_harmony@active.ident=as.factor(STR_harmony$RNA_snn_res.0.15)

DimPlot(STR_harmony,reduction = "umap",group.by = "RNA_snn_res.0.15",label = T)

STR_harmony_0.15_markers <- FindAllMarkers(object = STR_harmony, only.pos = TRUE,min.pct = 0.25, 
                                           logfc.threshold = 0.25)

STR_harmony_0.15_markers=subset(STR_harmony_0.15_markers,STR_harmony_0.15_markers$p_val_adj<0.01)

write.csv(STR_harmony_0.15_markers,"/data/DATAprocess/chentw/zigongneimo/原图绘制/STR_harmnoy/res0.15_marker/STR_harmony_0.15_markers.csv")


###TOP5marker基因气泡图绘制

STR_harmony_0.15_markers_top5 <- STR_harmony_0.15_markers %>% group_by(cluster) %>% top_n(n = 5, wt = avg_log2FC)


DotPlot(STR_harmony,group.by = "RNA_snn_res.0.15",features = STR_harmony_0.15_markers_top5$gene)+
  theme(axis.text.x = element_text(angle=45,hjust = 1,vjust = 1))



##开始富集

STR_harmony_0.15_markers=read.csv("/data/DATAprocess/chentw/zigongneimo/原图绘制/STR_harmnoy/res0.15_marker/STR_harmony_0.15_markers.csv")

GO_STR=function(X){N=as.character(X)
N=subset(STR_harmony_0.15_markers,STR_harmony_0.15_markers$cluster%in%N)
N=N$gene
G <- bitr(N, fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
G=G$ENTREZID
ego <- enrichGO(OrgDb="org.Hs.eg.db", gene = G, ont = "ALL",readable= TRUE)
}
ego0=GO_STR(0)
write.csv(ego0,
          "/data/DATAprocess/chentw/zigongneimo/原图绘制/STR_harmnoy/res0.15_marker/S0.csv")

ego1=GO_STR(1)
write.csv(ego1,
          "/data/DATAprocess/chentw/zigongneimo/原图绘制/STR_harmnoy/res0.15_marker/S1.csv")

ego2=GO_STR(2)
write.csv(ego2,
          "/data/DATAprocess/chentw/zigongneimo/原图绘制/STR_harmnoy/res0.15_marker/S2.csv")

ego3=GO_STR(3)
write.csv(ego3,
          "/data/DATAprocess/chentw/zigongneimo/原图绘制/STR_harmnoy/res0.15_marker/S3.csv")

ego4=GO_STR(4)
write.csv(ego4,
          "/data/DATAprocess/chentw/zigongneimo/原图绘制/STR_harmnoy/res0.15_marker/S4.csv")

ego5=GO_STR(5)
write.csv(ego5,
          "/data/DATAprocess/chentw/zigongneimo/原图绘制/STR_harmnoy/res0.15_marker/S5.csv")

ego6=GO_STR(6)
write.csv(ego6,
          "/data/DATAprocess/chentw/zigongneimo/原图绘制/STR_harmnoy/res0.15_marker/S6.csv")

ego7=GO_STR(7)
write.csv(ego7,
          "/data/DATAprocess/chentw/zigongneimo/原图绘制/STR_harmnoy/res0.15_marker/S7.csv")


###寻找unique的基因并进行保存











#将单细胞数据和文章全部细胞进行整合(integrated)----
umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")
global<-LoadH5Seurat("/data/DATAprocess/chentw/zigongneimo/WEN2/h5ad文件/endo-2022_global.h5seurat",
                     assays="RNA")

DimPlot(global,split.by = "PID",label = T)

global$type3=global$PID


global$type1117="WEN2"

umap_ctrl_IUA_allsingle$type1117="umap"






umap_WEN2=merge(umap_ctrl_IUA_allsingle,global)

#重新整合
umap_WEN2.list <- SplitObject(umap_WEN2, split.by = "type3")
umap_WEN2.list <- lapply(X = umap_WEN2.list, FUN = function(x) {
  x <- NormalizeData(x)
  x <- FindVariableFeatures(x, selection.method = "vst", nfeatures = 2000)
})



#数据进行合并(intergrad)
umap_WEN2=FindIntegrationAnchors(object.list = umap_WEN2.list,dims=1:30)
umap_WEN2=IntegrateData(anchorset=umap_WEN2,dims=1:30)

#进行标准化然后PCA分析
DefaultAssay(umap_WEN2) <- "RNA"
all.genes <- rownames(umap_WEN2)
umap_WEN2 <- ScaleData(object = umap_WEN2, features = all.genes)

DefaultAssay(umap_WEN2) <- "integrated"
all.genes1 <- rownames(umap_WEN2)
umap_WEN2 <- ScaleData(object = umap_WEN2, features = all.genes1)
umap_WEN2 <- RunPCA(umap_WEN2, features = VariableFeatures(object = umap_WEN2))



#进行细胞细胞聚类分析(umap_WEN2)
umap_WEN2=RunUMAP(umap_WEN2, dims = 1:30)
DimPlot(umap_WEN2,reduction="umap",label=TRUE)
umap_WEN2 <- FindNeighbors(umap_WEN2, dims = 1:30)
umap_WEN2 <- FindClusters(umap_WEN2, resolution = 0.3)
DefaultAssay(umap_WEN2)<-"RNA"

#umap_WEN2进行细胞细胞聚类分析(tsne)
umap_WEN2=RunTSNE(umap_WEN2, dims = 1:30)
DimPlot(umap_WEN2,reduction="tsne",label=TRUE)
umap_WEN2<- FindNeighbors(umap_WEN2, dims = 1:30)
umap_WEN2 <- FindClusters(umap_WEN2, resolution = 0.3)
umap_WEN2 <- FindClusters(umap_WEN2, resolution = 0.6)
DimPlot(umap_WEN2,reduction="tsne",label=TRUE)
saveRDS(umap_WEN2,"/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_WEN2.rds")

umap_WEN2=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_WEN2.rds")

DimPlot(umap_WEN2,group.by = "type1117")




umap_WEN2$cell_type1122=umap_WEN2$celltype_main

umap_WEN2$cell_type1122[umap_WEN2$type1117%in%"umap"]=umap_ctrl_IUA_allsingle$cell_type1110

umap_WEN2$cell_type1122[umap_WEN2$cell_type1122%in%"1"]<-"DE_Stromal"
umap_WEN2$cell_type1122[umap_WEN2$cell_type1122%in%"2"]<-"DE_Immune"
umap_WEN2$cell_type1122[umap_WEN2$cell_type1122%in%"3"]<-"DE_Epithelium"
umap_WEN2$cell_type1122[umap_WEN2$cell_type1122%in%"4"]<-"DE_Endothelium"

DimPlot(umap_WEN2,group.by = "cell_type1122",label = T,repel = T,reduction = "umap")

DimPlot(umap_WEN2,group.by = "cell_type1122",label = T,repel = T,split.by = "type1117",reduction = "umap")+
  theme(legend.position = "none",panel.grid.major = element_blank(),panel.grid.minor = element_blank(),
        panel.border = element_blank(),axis.title = element_blank(),axis.text = element_blank(),
        axis.ticks = element_blank(),axis.line = element_blank())


#查看细胞周期的影响
s.genes=Seurat::cc.genes.updated.2019$s.genes
g2m.genes=Seurat::cc.genes.updated.2019$g2m.genes
umap_WEN2 <- CellCycleScoring(umap_WEN2, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
umap_WEN2 <- RunPCA(umap_WEN2, features = c(s.genes, g2m.genes))
DefaultAssay(umap_WEN2)<-"RNA"
DimPlot(umap_WEN2,group.by = "Phase",reduction = "umap")





#对宫腔粘连数据的基质细胞进行操作----
STR_ctrl_IUA_allsingle=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/STR_ctrl_IUA_allsingle.rds")

DefaultAssay(STR_ctrl_IUA_allsingle)<-"RNA"

DimPlot(STR_ctrl_IUA_allsingle)

FeaturePlot(STR_ctrl_IUA_allsingle,features = c("TGM2",
                                                "PAGE4","TAGLN","RAMP1","ACTA2"),min.cutoff = 0)

#对宫腔粘连数据的基质细胞进行操作（integrated并去除细胞周期）-----
library(dplyr)
library(Seurat)
library(ggplot2)

harmony_all=readRDS("/data/DATAprocess/chentw/zigongneimo/原图绘制/rds/harmony_all_harmony.rds")

DimPlot(harmony_all,group.by = "cell_type1221")
harmony_all$cell_type1221="1221"
harmony_all$cell_type1221[harmony_all$RNA_snn_res.0.3%in%c("0","2","5","7","16","10")]<-"Stromal cells"

harmony_all$cell_type1221[harmony_all$RNA_snn_res.0.3%in%c("9")]<-"Endothelium cells"

harmony_all$cell_type1221[harmony_all$RNA_snn_res.0.3%in%c("1","6","11","3","12","13","14","15")]<-"Immune cells"

harmony_all$cell_type1221[harmony_all$RNA_snn_res.0.3%in%c("4","8")]<-"Epithelium cells"



harmony_all_embeddings=harmony_all$umap@cell.embeddings
harmony_all_embeddings=as.data.frame(harmony_all_embeddings)
harmony_all_embeddings$cell_type="cell_type1221"
harmony_all_embeddings$cell_type=harmony_all$cell_type1221
harmony_all_embeddings$UMAP_1=as.numeric(as.character(harmony_all_embeddings$UMAP_1))
harmony_all_embeddings$UMAP_2=as.numeric(as.character(harmony_all_embeddings$UMAP_2))
harmony_all$UMAP_1=harmony_all_embeddings$UMAP_1
harmony_all$UMAP_2=harmony_all_embeddings$UMAP_2

STR_integrated=subset(harmony_all,cell_type1221%in%"Stromal cells")

DimPlot(STR_integrated,group.by = "cell_type1221")

STR_integrated=subset(STR_integrated,UMAP_1<5&UMAP_2<8&UMAP_2>-7)

DimPlot(STR_integrated,group.by = "cell_type1221")




#重新整合
STR_integrated.list <- SplitObject(STR_integrated, split.by = "type3")
STR_integrated.list <- lapply(X = STR_integrated.list, FUN = function(x) {
  x <- NormalizeData(x)
  x <- FindVariableFeatures(x, selection.method = "vst", nfeatures = 2000)
})



#数据进行合并(intergrad)
STR_integrated=FindIntegrationAnchors(object.list = STR_integrated.list,dims=1:30)

STR_integrated=IntegrateData(anchorset=STR_integrated,dims=1:30,k.weight = 67)

#进行标准化然后PCA分析
DefaultAssay(STR_integrated) <- "RNA"
all.genes <- rownames(STR_integrated)


STR_integrated <- ScaleData(object = STR_integrated)

s.genes <- cc.genes$s.genes
g2m.genes <- cc.genes$g2m.genes
STR_integrated <- CellCycleScoring(STR_integrated, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)

STR_integrated@active.ident=STR_integrated$integrated_snn_res.0.3
DimPlot(STR_integrated)
DimPlot(STR_integrated,group.by = "Phase")+ggtitle("")

#去除细胞周期的影响
STR_integrated<- ScaleData(STR_integrated, vars.to.regress = c("S.Score", "G2M.Score"))


DefaultAssay(STR_integrated) <- "integrated"
all.genes1 <- rownames(STR_integrated)
STR_integrated <- ScaleData(object = STR_integrated)
STR_integrated <- RunPCA(STR_integrated, features = VariableFeatures(object = STR_integrated))



#进行细胞细胞聚类分析(STR_integrated)
STR_integrated=RunUMAP(STR_integrated, dims = 1:30)
DimPlot(STR_integrated,reduction="umap",label=TRUE)
STR_integrated <- FindNeighbors(STR_integrated, dims = 1:30)
STR_integrated <- FindClusters(STR_integrated, resolution = 0.025)
STR_integrated <- FindClusters(STR_integrated, resolution = 0.05)
STR_integrated <- FindClusters(STR_integrated, resolution = 0.1)
STR_integrated <- FindClusters(STR_integrated, resolution = 0.2)
STR_integrated <- FindClusters(STR_integrated, resolution = 0.3)
DefaultAssay(STR_integrated)<-"RNA"

#STR_integrated进行细胞细胞聚类分析(tsne)
STR_integrated=RunTSNE(STR_integrated, dims = 1:30)
DimPlot(STR_integrated,reduction="tsne",label=TRUE)
STR_integrated<- FindNeighbors(STR_integrated, dims = 1:30)
STR_integrated <- FindClusters(STR_integrated, resolution = 0.3)
STR_integrated <- FindClusters(STR_integrated, resolution = 0.6)
DimPlot(STR_integrated,reduction="tsne",label=TRUE)


saveRDS(STR_integrated,"/data/DATAprocess/chentw/zigongneimo/python运行/WEN_STR20221224/STR_integrated.rds")








#对文章的基质细胞进行下游操作----
WEN2_STR<-LoadH5Seurat("/data/DATAprocess/chentw/zigongneimo/WEN2/h5ad文件/endo-2022_stromal.h5seurat",
                       assays="RNA")


color=c(rep("grey",3),rep("red",1),rep("grey",8))

DimPlot(WEN2_STR,group.by = "subtypes",label = T)+
  scale_color_manual(values=color)

FeaturePlot(WEN2_STR,features = c("TGM2",
                                  "PAGE4","TAGLN","RAMP1","ACTA2"),min.cutoff = 0)


WEN2_STR@active.ident=as.factor(WEN2_STR$subtypes)

DefaultAssay(WEN2_STR)<-"RNA"
WEN2_STR_markers <- FindAllMarkers(object=WEN2_STR, only.pos = TRUE,min.pct = 0.25, 
                                   logfc.threshold = 0.25)
WEN2_STR_markers <-subset(WEN2_STR_markers,WEN2_STR_markers$p_val_adj<0.01)


write.csv(WEN2_STR_markers,"/data/DATAprocess/chentw/zigongneimo/WEN2/marker基因/WEN2_STR_subtypes_markers.csv")



WEN2_STR_markers_top5 <- WEN2_STR_markers %>% group_by(cluster) %>% top_n(n = 5, wt = avg_log2FC)

write.csv(WEN2_STR_markers_top5,"/data/DATAprocess/chentw/zigongneimo/WEN2/marker基因/WEN2_STR_markers_top5.csv")


WEN2_STR_markers=read.csv("/data/DATAprocess/chentw/zigongneimo/WEN2/marker基因/WEN2_STR_subtypes_markers.csv")


WEN2_STR_markers_top10 <- WEN2_STR_markers %>% group_by(cluster) %>% top_n(n = 10, wt = avg_log2FC)


WEN2_STR_markers3=subset(WEN2_STR_markers,substr(WEN2_STR_markers$X,1,2)%in%"MT")


#和文章的基质细胞进行重新整合-----
#把文章的基质细胞和之前的基质细胞进行处理并且进行赋值
Convert("/data/DATAprocess/chentw/zigongneimo/WEN2/h5ad文件/endo-2022_stromal.h5ad","h5seurat",
        overwrite=TRUE,assay="RNA") 


WEN2_STR<-LoadH5Seurat("/data/DATAprocess/chentw/zigongneimo/WEN2/h5ad文件/endo-2022_stromal.h5seurat",
                       assays="RNA")

umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")

umap_ctrl_IUA_allsingle@active.ident<-umap_ctrl_IUA_allsingle$integrated_snn_res.0.3
umap_ctrl_IUA_allsingle@active.ident=factor(umap_ctrl_IUA_allsingle@active.ident,levels = 0:19)
new.cluster.ids <- c("Stromal cells","Stromal cells","Stromal cells","Immune cells","Immune cells","Epithelium cells",
                     "Stromal cells","Stromal cells","Stromal cells","Immune cells","Immune cells","Epithelium cells",
                     "Endothelium cells","Stromal cells","Stromal cells","Immune cells","Immune cells","Immune cells","Immune cells","Endothelium cells")


names(new.cluster.ids) <- levels(umap_ctrl_IUA_allsingle)
umap_ctrl_IUA_allsingle <- RenameIdents(umap_ctrl_IUA_allsingle, new.cluster.ids)

umap_ctrl_IUA_allsingle$cell_type1110=umap_ctrl_IUA_allsingle@active.ident

STR_ctrl_IUA_allsingle=subset(umap_ctrl_IUA_allsingle,cell_type1110%in%"Stromal cells")

table(WEN2_STR$PID)

WEN2_STR$type3=WEN2_STR$PID


WEN2_STR$type1124="WEN2"


STR_ctrl_IUA_allsingle$type1124="UMAP"


umap_WEN2_STR=merge(WEN2_STR,STR_ctrl_IUA_allsingle)



#重新整合
umap_WEN2_STR.list <- SplitObject(umap_WEN2_STR, split.by = "type3")
umap_WEN2_STR.list <- lapply(X = umap_WEN2_STR.list, FUN = function(x) {
  x <- NormalizeData(x)
  x <- FindVariableFeatures(x, selection.method = "vst", nfeatures = 2000)
})



#数据进行合并(intergrad)
umap_WEN2_STR=FindIntegrationAnchors(object.list = umap_WEN2_STR.list,dims=1:30)

umap_WEN2_STR=IntegrateData(anchorset=umap_WEN2_STR,dims=1:30,k.weight = 67)

#进行标准化然后PCA分析
DefaultAssay(umap_WEN2_STR) <- "RNA"
all.genes <- rownames(umap_WEN2_STR)
umap_WEN2_STR <- ScaleData(object = umap_WEN2_STR, features = all.genes)

DefaultAssay(umap_WEN2_STR) <- "integrated"
all.genes1 <- rownames(umap_WEN2_STR)
umap_WEN2_STR <- ScaleData(object = umap_WEN2_STR, features = all.genes1)
umap_WEN2_STR <- RunPCA(umap_WEN2_STR, features = VariableFeatures(object = umap_WEN2_STR))



#进行细胞细胞聚类分析(umap_WEN2_STR)
umap_WEN2_STR=RunUMAP(umap_WEN2_STR, dims = 1:30)
DimPlot(umap_WEN2_STR,reduction="umap",label=TRUE)
umap_WEN2_STR <- FindNeighbors(umap_WEN2_STR, dims = 1:30)
umap_WEN2_STR <- FindClusters(umap_WEN2_STR, resolution = 0.025)
umap_WEN2_STR <- FindClusters(umap_WEN2_STR, resolution = 0.05)
umap_WEN2_STR <- FindClusters(umap_WEN2_STR, resolution = 0.1)
umap_WEN2_STR <- FindClusters(umap_WEN2_STR, resolution = 0.2)
umap_WEN2_STR <- FindClusters(umap_WEN2_STR, resolution = 0.3)
DefaultAssay(umap_WEN2_STR)<-"RNA"

#umap_WEN2_STR进行细胞细胞聚类分析(tsne)
umap_WEN2_STR=RunTSNE(umap_WEN2_STR, dims = 1:30)
DimPlot(umap_WEN2_STR,reduction="tsne",label=TRUE)
umap_WEN2_STR<- FindNeighbors(umap_WEN2_STR, dims = 1:30)
umap_WEN2_STR <- FindClusters(umap_WEN2_STR, resolution = 0.3)
umap_WEN2_STR <- FindClusters(umap_WEN2_STR, resolution = 0.6)
DimPlot(umap_WEN2_STR,reduction="tsne",label=TRUE)
saveRDS(umap_WEN2_STR,"/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_WEN2_STR.rds")


umap_WEN2_STR=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_WEN2_STR.rds")

umap_WEN2_STR$type1124="WEN2"

umap_WEN2_STR$type1124[umap_WEN2_STR$cell_type1110%in%"Stromal cells"]="UMAP"

type3
DimPlot(umap_WEN2_STR,split.by = "type1124",label = T,repel = T)


table(WEN2_STR$celltype)


DimPlot(WEN2_STR,group.by = "subtypes",label = T)


umap_WEN2_STR$celltype1126=WEN2_STR$subtypes

table(umap_WEN2_STR$type1124)
umap_WEN2_STR$type1124=as.factor(umap_WEN2_STR$type1124)

table(umap_WEN2_STR$celltype1126)

umap_WEN2_STR$celltype1126="UMAP"

umap_WEN2_STR$subtypes[umap_WEN2_STR$type1124%in%"UMAP"]="UMAP"


DimPlot(umap_WEN2_STR,split.by = "type1124",label = T,repel = T,group.by = "subtypes")


DimPlot(umap_WEN2_STR,split.by = "type1124",label = T,repel = T,group.by = "subtypes")+
  scale_color_manual(values = colors)

umap_WEN2_STR$subtypes[umap_WEN2_STR$type1124%in%"UMAP"]=umap_WEN2_STR$integrated_snn_res.0.1[umap_WEN2_STR$type1124%in%"UMAP"]


DimPlot(umap_WEN2_STR,split.by = "type1124",label = T,repel = T,group.by = "subtypes")

umap_WEN2_STR$celltype1126="UMAP"

umap_WEN2_STR$celltype[umap_WEN2_STR$type1124%in%"UMAP"]="UMAP"

DimPlot(umap_WEN2_STR,split.by = "type1124",label = T,repel = T,group.by = "celltype")



DimPlot(umap_WEN2_STR,split.by = "type1124",label = T,repel = T,group.by = "celltype")+
  scale_fill_manual(values = c("grey"))  



table(umap_WEN2_STR$celltype)



#对res0.1进行marker基因的寻找
DimPlot(umap_WEN2_STR,split.by = "type1124",label = T,repel = T,group.by = "subtypes")

umap_WEN2_STR_UMAP=subset(umap_WEN2_STR,type1124%in%"UMAP")

DimPlot(umap_WEN2_STR_UMAP,label = T,repel = T,group.by = "subtypes")

DefaultAssay(umap_WEN2_STR_UMAP)<-"RNA"
umap_WEN2_STR_UMAP@active.ident=as.factor(umap_WEN2_STR_UMAP$subtypes)
umap_WEN2_STR_UMAP_cluster_markers <- FindAllMarkers(object = umap_WEN2_STR_UMAP, only.pos = TRUE,min.pct = 0.25, 
                                                     logfc.threshold = 0.25)



write.csv(umap_WEN2_STR_UMAP_cluster_markers,
          "/data/DATAprocess/chentw/zigongneimo/WEN2/marker基因/umap_WEN2_STR_UMAP_cluster_markers.csv")



umap_WEN2_STR_UMAP_cluster_markers_top5 <- umap_WEN2_STR_UMAP_cluster_markers %>% group_by(cluster) %>% top_n(n = 5, wt = avg_log2FC)

write.csv(umap_WEN2_STR_UMAP_cluster_markers_top5,"/data/DATAprocess/chentw/zigongneimo/WEN2/marker基因/umap_WEN2_STR_UMAP_cluster_markers_top5.csv")




#对umap_WEN2_STR进行线粒体，细胞周期，核糖体，应激相关基因的影响



#将umap_ctrl_IUA_allsingle转成h5ad
umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")

SaveH5Seurat(umap_ctrl_IUA_allsingle, filename = "/data/DATAprocess/chentw/zigongneimo/python运行/loom文件存放点/umap_ctrl_IUA_allsingle.h5Seurat",overwrite = TRUE)

setwd("/data/DATAprocess/chentw/zigongneimo/python运行/loom文件存放点/")
Convert("/data/DATAprocess/chentw/zigongneimo/python运行/loom文件存放点/umap_ctrl_IUA_allsingle.h5Seurat", dest = "h5ad",overwrite = TRUE)


#将STR_ctrl_IUA_allsingle转成h5ad
STR_ctrl_IUA_allsingle=subset(umap_ctrl_IUA_allsingle,cell_type1110%in%"Stromal cells")
SaveH5Seurat(STR_ctrl_IUA_allsingle, filename = "/data/DATAprocess/chentw/zigongneimo/python运行/loom文件存放点/STR_ctrl_IUA_allsingle.h5Seurat",overwrite = TRUE)

setwd("/data/DATAprocess/chentw/zigongneimo/python运行/loom文件存放点/")
Convert("/data/DATAprocess/chentw/zigongneimo/python运行/loom文件存放点/STR_ctrl_IUA_allsingle.h5Seurat", dest = "h5ad",overwrite = TRUE)







#将基质细胞和文章的基质细胞进行merge和时期相关基因的时期变化热图----
umap_cltrl_IUA=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_cltrl_IUA.rds")

table(umap_cltrl_IUA$Double,umap_cltrl_IUA$type3)

DimPlot(umap_cltrl_IUA,group.by = "Double")+ggtitle("")



umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")

umap_ctrl_IUA_allsingle@active.ident<-umap_ctrl_IUA_allsingle$integrated_snn_res.0.3
umap_ctrl_IUA_allsingle@active.ident=factor(umap_ctrl_IUA_allsingle@active.ident,levels = 0:19)
new.cluster.ids <- c("Stromal cells","Stromal cells","Stromal cells","Immune cells","Immune cells","Epithelium cells",
                     "Stromal cells","Stromal cells","Stromal cells","Immune cells","Immune cells","Epithelium cells",
                     "Endothelium cells","Stromal cells","Stromal cells","Immune cells","Immune cells","Immune cells","Immune cells","Endothelium cells")


names(new.cluster.ids) <- levels(umap_ctrl_IUA_allsingle)
umap_ctrl_IUA_allsingle <- RenameIdents(umap_ctrl_IUA_allsingle, new.cluster.ids)

umap_ctrl_IUA_allsingle$cell_type1110=umap_ctrl_IUA_allsingle@active.ident

STR_ctrl_IUA_allsingle=subset(umap_ctrl_IUA_allsingle,cell_type1110%in%"Stromal cells")



STRWEN=readRDS("/mnt/chentw/zigongneimo/photo/文章10x和我们数据整合分析/STRWEN.rds")
STRWEN$type3 <- "proliferative"
STRWEN$type3[STRWEN$donor==c("proliferative")] <- "proliferative"
STRWEN$type3[STRWEN$donor==c("secretory_early")] <- "secretory_early"
STRWEN$type3[STRWEN$donor==c("secretory_early-mid")] <- "secretory_early-mid"
STRWEN$type3[STRWEN$donor==c("secretory_mid")] <- "secretory_mid"
STRWEN$type3[STRWEN$donor==c("secretory_late")] <- "secretory_late"


STR_allsingle_STR_wen=merge(STR_ctrl_IUA_allsingle,STRWEN)

STR_allsingle_STR_wen=NormalizeData(STR_allsingle_STR_wen)

#进行标准化
DefaultAssay(STR_allsingle_STR_wen) <- "RNA"
all.genes.STR_allsingle_STR_wen <- rownames(STR_allsingle_STR_wen)
STR_allsingle_STR_wen <- ScaleData(object = STR_allsingle_STR_wen, features = all.genes.STR_allsingle_STR_wen)



#绘图
DimPlot(STR_allsingle_STR_wen,group.by = "type3")
prolifer_maker=c("STC1","PMAIP1","MMP11","SFRP1","WNT5A",
                 "S100A4","DKK1","LMCD1")


DefaultAssay(STR_allsingle_STR_wen)<-"RNA"


pdf(file = "/data/DATAprocess/chentw/zigongneimo/原图绘制/和NC文章的整合/STR2.pdf",width = 40,height = 16)
DoHeatmap(STR_allsingle_STR_wen,features = prolifer_maker,group.by = "type3",label = F,slot = "data")+
  theme(axis.title= element_text(size=60),axis.text=element_text(size=30),legend.text = element_text(size = 30),legend.title = element_text(size =30))+
  theme(legend.key.height=unit(4,"line"),legend.key.width =unit(4,"line"),legend.box = "horizontal",legend.key.size=unit(30,"line"))+scale_size(range=c(20,20))+
  scale_fill_gradientn(colors = c("blue", "white", "yellow"))+ylab("STR")
dev.off()






pdf(file = "/data/DATAprocess/chentw/zigongneimo/原图绘制/和NC文章的整合/prolifer.pdf",width = 40,height = 16)
DoHeatmap(STR_allsingle_STR_wen,features = prolifer_maker,group.by = "type3",label = F)+
  theme(axis.title= element_text(size=60),axis.text=element_text(size=30),legend.text = element_text(size = 30),legend.title = element_text(size =30))+
  theme(legend.key.height=unit(4,"line"),legend.key.width =unit(4,"line"),legend.box = "horizontal",legend.key.size=unit(30,"line"))+scale_size(range=c(20,20))+
  scale_fill_gradientn(colors = c("blue", "white", "yellow"))+ylab("STR")
dev.off()



jpeg(file = "/mnt/chentw/zigongneimo/photo/11-samples/绘图存放点/STR_ctrl_IUA_allsingle/和NC文章的整合绘图/prolifer.jpeg",width = 200,height = 1200)
DoHeatmap(STR_allsingle_STR_wen,features = prolifer_maker,group.by = "type3",label = F)
dev.off()

pdf(file = "/mnt/chentw/zigongneimo/photo/11-samples/绘图存放点/STR_ctrl_IUA_allsingle/和NC文章的整合绘图/prolifer2.pdf")
DoHeatmap(STR_allsingle_STR_wen,features = prolifer_maker,group.by = "type3",label = F)
dev.off()


STR_maker=c("LUM","COL1A1","DCN","MMP11")
FeaturePlot(IUA_BYQ,features = STR_maker)
ggsave("/mnt/chentw/zigongneimo/photo/11-samples/绘图存放点/IUA_BYQ/STR.png")

EPI_maker=c("EPCAM","KRT18","MMP7","WFDC2")
FeaturePlot(IUA_BYQ,features = EPI_maker)
ggsave("/mnt/chentw/zigongneimo/photo/11-samples/绘图存放点/IUA_BYQ/EPI.png")


Endothelium_maker=c("VWF","CDH5")
FeaturePlot(IUA_BYQ,features = Endothelium_maker)
ggsave("/mnt/chentw/zigongneimo/photo/11-samples/绘图存放点/IUA_BYQ/Endothelium.png")

smooth_maker=c("MCAM","RGS5","ACTA2")
FeaturePlot(IUA_BYQ,features = smooth_maker)
ggsave("/mnt/chentw/zigongneimo/photo/11-samples/绘图存放点/IUA_BYQ/smooth_muscle.png")









#将基质细胞和文章的基质细胞进行merge和时期相关基因的时期变化热图分开绘制----
STRWEN=readRDS("/mnt/chentw/zigongneimo/photo/文章10x和我们数据整合分析/STRWEN.rds")
STRWEN$type3 <- "proliferative"
STRWEN$type3[STRWEN$donor==c("proliferative")] <- "proliferative"
STRWEN$type3[STRWEN$donor==c("secretory_early")] <- "secretory_early"
STRWEN$type3[STRWEN$donor==c("secretory_early-mid")] <- "secretory_early-mid"
STRWEN$type3[STRWEN$donor==c("secretory_mid")] <- "secretory_mid"
STRWEN$type3[STRWEN$donor==c("secretory_late")] <- "secretory_late"


prolifer_maker=c("STC1","PMAIP1","MMP11","SFRP1","WNT5A",
                 "S100A4","DKK1","LMCD1")



STRWEN=as.data.frame(AverageExpression(STRWEN,
                                       features =prolifer_maker,group.by = "donor")$RNA)

library(pheatmap)


pheatmap(STRWEN,scale="row",cluster_rows = F,cluster_cols = F,
         color = colorRampPalette(c("#164180","white","#6B0823"))(100), border=FALSE)


bk=c(seq(-1.5,-0.1,by=0.01),seq(0,1.5,by=0.01))


pheatmap(STRWEN,scale="row",cluster_rows = F,cluster_cols = F, border=FALSE,
         color = c(colorRampPalette(colors = c("#3C2154","white"))(length(bk)/2),colorRampPalette(colors = c("white","#DCAB5E"))(length(bk)/2)),
         breaks=bk)





STR_pheatmap=as.data.frame(AverageExpression(STR_ctrl_IUA_allsingle,
                                             features =prolifer_maker,group.by = "type3")$RNA)

library(pheatmap)


P1=pheatmap(STR_pheatmap,scale="row",cluster_rows = F,cluster_cols = F,
            color = colorRampPalette(c("#164180","white","#6B0823"))(100), border=FALSE)


bk=c(seq(-1.5,-0.1,by=0.01),seq(0,1.5,by=0.01))


P2=pheatmap(STR_pheatmap,scale="row",cluster_rows = F,cluster_cols = F, border=FALSE,
            color = c(colorRampPalette(colors = c("#3C2154","white"))(length(bk)/2),colorRampPalette(colors = c("white","#DCAB5E"))(length(bk)/2)),
            breaks=bk)


#进行组合绘图
P1=pheatmap(STRWEN,scale="row",cluster_rows = F,cluster_cols = F, border=FALSE,
            color = c(colorRampPalette(colors = c("#3C2154","white"))(length(bk)/2),colorRampPalette(colors = c("white","#DCAB5E"))(length(bk)/2)),
            breaks=bk,show_colnames=F)

P2=pheatmap(STR_pheatmap,scale="row",cluster_rows = F,cluster_cols = F, border=FALSE,
            color = c(colorRampPalette(colors = c("#3C2154","white"))(length(bk)/2),colorRampPalette(colors = c("white","#DCAB5E"))(length(bk)/2)),
            breaks=bk,show_colnames=F)

P1=as.ggplot(P1)

P1=P1+theme(legend.position = "none")

P2=as.ggplot(P2)

P1+P2

ggarrange(P1,P2,common.legend = T,widths = c(3,5))

graph2ppt(file = "/data/DATAprocess/chentw/zigongneimo/原图绘制/20221203/P1", width=6, height=5, append=TRUE)



pheatmap(STR_pheatmap,scale="row",cluster_rows = F,cluster_cols = F, border=FALSE,
         color = c(colorRampPalette(colors = c("#3C2154","white"))(length(bk)/2),colorRampPalette(colors = c("white","#DCAB5E"))(length(bk)/2)),
         breaks=bk)


pheatmap(STRWEN,scale="row",cluster_rows = F,cluster_cols = F, border=FALSE,
         color = c(colorRampPalette(colors = c("#3C2154","white"))(length(bk)/2),colorRampPalette(colors = c("white","#DCAB5E"))(length(bk)/2)),
         breaks=bk)

#将基质细胞和文章的基质细胞进行时期相关基因的气泡图的分开绘制----
prolifer_maker=c("STC1","PMAIP1","MMP11","SFRP1","WNT5A",
                 "S100A4","DKK1","LMCD1")




umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")

umap_ctrl_IUA_allsingle@active.ident<-umap_ctrl_IUA_allsingle$integrated_snn_res.0.3
umap_ctrl_IUA_allsingle@active.ident=factor(umap_ctrl_IUA_allsingle@active.ident,levels = 0:19)
new.cluster.ids <- c("Stromal cells","Stromal cells","Stromal cells","Immune cells","Immune cells","Epithelium cells",
                     "Stromal cells","Stromal cells","Stromal cells","Immune cells","Immune cells","Epithelium cells",
                     "Endothelium cells","Stromal cells","Stromal cells","Immune cells","Immune cells","Immune cells","Immune cells","Endothelium cells")


names(new.cluster.ids) <- levels(umap_ctrl_IUA_allsingle)
umap_ctrl_IUA_allsingle <- RenameIdents(umap_ctrl_IUA_allsingle, new.cluster.ids)

umap_ctrl_IUA_allsingle$cell_type1110=umap_ctrl_IUA_allsingle@active.ident

STR_ctrl_IUA_allsingle=subset(umap_ctrl_IUA_allsingle,cell_type1110%in%"Stromal cells")



DotPlot(STR_ctrl_IUA_allsingle,group.by = "type3",features = prolifer_maker)+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),limits=c(-5,3))+
  scale_size_continuous(range = c(2,11))+xlab("")+ylab("")


STRWEN=readRDS("/mnt/chentw/zigongneimo/photo/文章10x和我们数据整合分析/STRWEN.rds")

DotPlot(STRWEN,group.by = "donor",features = prolifer_maker)+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])))



#将STR_harmony转入python进行操作【按照子宫内膜异位症的流程】-----
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/原图绘制/rds/STR_harmony.rds")


SaveH5Seurat(STR, filename = "/data/DATAprocess/chentw/zigongneimo/python运行/loom文件存放点/STR.h5Seurat",overwrite = TRUE)

setwd("/data/DATAprocess/chentw/zigongneimo/python运行/loom文件存放点/")
Convert("/data/DATAprocess/chentw/zigongneimo/python运行/loom文件存放点/STR.h5Seurat", dest = "h5ad",overwrite = TRUE)






#将基质细胞和文章的基质细胞进行相关性分析【样本对应文章时期】-----
umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")

umap_ctrl_IUA_allsingle@active.ident<-umap_ctrl_IUA_allsingle$integrated_snn_res.0.3
umap_ctrl_IUA_allsingle@active.ident=factor(umap_ctrl_IUA_allsingle@active.ident,levels = 0:19)
new.cluster.ids <- c("Stromal cells","Stromal cells","Stromal cells","Immune cells","Immune cells","Epithelium cells",
                     "Stromal cells","Stromal cells","Stromal cells","Immune cells","Immune cells","Epithelium cells",
                     "Endothelium cells","Stromal cells","Stromal cells","Immune cells","Immune cells","Immune cells","Immune cells","Endothelium cells")


names(new.cluster.ids) <- levels(umap_ctrl_IUA_allsingle)
umap_ctrl_IUA_allsingle <- RenameIdents(umap_ctrl_IUA_allsingle, new.cluster.ids)

umap_ctrl_IUA_allsingle$cell_type1110=umap_ctrl_IUA_allsingle@active.ident

STR_ctrl_IUA_allsingle=subset(umap_ctrl_IUA_allsingle,cell_type1110%in%"Stromal cells")



STRWEN=readRDS("/mnt/chentw/zigongneimo/photo/文章10x和我们数据整合分析/STRWEN.rds")
STRWEN$type3 <- "proliferative"
STRWEN$type3[STRWEN$donor==c("proliferative")] <- "proliferative"
STRWEN$type3[STRWEN$donor==c("secretory_early")] <- "secretory_early"
STRWEN$type3[STRWEN$donor==c("secretory_early-mid")] <- "secretory_early-mid"
STRWEN$type3[STRWEN$donor==c("secretory_mid")] <- "secretory_mid"
STRWEN$type3[STRWEN$donor==c("secretory_late")] <- "secretory_late"


STR_allsingle_STR_wen=merge(STR_ctrl_IUA_allsingle,STRWEN)

STR_allsingle_STR_wen=NormalizeData(STR_allsingle_STR_wen)

#进行标准化
DefaultAssay(STR_allsingle_STR_wen) <- "RNA"
all.genes.STR_allsingle_STR_wen <- rownames(STR_allsingle_STR_wen)
STR_allsingle_STR_wen <- ScaleData(object = STR_allsingle_STR_wen, features = all.genes.STR_allsingle_STR_wen)


#开始相关性热图绘制-1000
av=AverageExpression(STR_allsingle_STR_wen,
                     group.by = "type3",
                     assays = "RNA")

av=av[[1]]

cg=names(tail(sort(apply(av,1,sd)),1000))


View(av[cg,])


pheatmap(cor(av[cg,],method = 'spearman'))



#开始相关性热图绘制-2000
av=AverageExpression(STR_allsingle_STR_wen,
                     group.by = "type3",
                     assays = "RNA")

av=av[[1]]

cg=names(tail(sort(apply(av,1,sd)),2000))


View(av[cg,])


pheatmap(cor(av[cg,],method = 'spearman'),cluster_rows = F,cluster_cols = F)





#将上皮细胞和文章的上皮细胞进行时期相关基因的气泡图的分开绘制----
umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")

umap_ctrl_IUA_allsingle@active.ident<-umap_ctrl_IUA_allsingle$integrated_snn_res.0.3
umap_ctrl_IUA_allsingle@active.ident=factor(umap_ctrl_IUA_allsingle@active.ident,levels = 0:19)
new.cluster.ids <- c("Stromal cells","Stromal cells","Stromal cells","Immune cells","Immune cells","Epithelium cells",
                     "Stromal cells","Stromal cells","Stromal cells","Immune cells","Immune cells","Epithelium cells",
                     "Endothelium cells","Stromal cells","Stromal cells","Immune cells","Immune cells","Immune cells","Immune cells","Endothelium cells")


names(new.cluster.ids) <- levels(umap_ctrl_IUA_allsingle)
umap_ctrl_IUA_allsingle <- RenameIdents(umap_ctrl_IUA_allsingle, new.cluster.ids)

umap_ctrl_IUA_allsingle$cell_type1110=umap_ctrl_IUA_allsingle@active.ident

EPI_ctrl_IUA_allsingle=subset(umap_ctrl_IUA_allsingle,cell_type1110%in%"Epithelium cells")

EPIWEN=readRDS("/mnt/chentw/zigongneimo/photo/文章10x和我们数据整合分析/EPIWEN.rds")


EPI_donor_marker=c("PLAU","MMP7","THBS1","CADM1","NPAS3","ATP1A1",
                   "ANK3","ALPL","TRAK1","SCGB1D2","MT1F","MT1X",
                   "MT1E","MT1G","CXCL14","MAOA","DPP4","NUPR1",
                   "GPX3","PAEP")


DotPlot(EPI_ctrl_IUA_allsingle,group.by = "type3",features = EPI_donor_marker)+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])))


DotPlot(EPIWEN,group.by = "donor",features = EPI_donor_marker)+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])))








#进行上皮细胞气泡图合起来绘制-----
umap_ctrl_IUA_allsingle@active.ident<-umap_ctrl_IUA_allsingle$integrated_snn_res.0.3
umap_ctrl_IUA_allsingle@active.ident=factor(umap_ctrl_IUA_allsingle@active.ident,levels = 0:19)
new.cluster.ids <- c("Stromal cells","Stromal cells","Stromal cells","Immune cells","Immune cells","Epithelium cells",
                     "Stromal cells","Stromal cells","Stromal cells","Immune cells","Immune cells","Epithelium cells",
                     "Endothelium cells","Stromal cells","Stromal cells","Immune cells","Immune cells","Immune cells","Immune cells","Endothelium cells")


names(new.cluster.ids) <- levels(umap_ctrl_IUA_allsingle)
umap_ctrl_IUA_allsingle <- RenameIdents(umap_ctrl_IUA_allsingle, new.cluster.ids)

umap_ctrl_IUA_allsingle$cell_type1110=umap_ctrl_IUA_allsingle@active.ident

EPI_ctrl_IUA_allsingle=subset(umap_ctrl_IUA_allsingle,cell_type1110%in%"Epithelium cells")

EPIWEN=readRDS("/mnt/chentw/zigongneimo/photo/文章10x和我们数据整合分析/EPIWEN.rds")

EPIWEN$type3 <- "proliferative"
EPIWEN$type3[EPIWEN$donor==c("proliferative")] <- "proliferative"
EPIWEN$type3[EPIWEN$donor==c("secretory_early")] <- "secretory_early"
EPIWEN$type3[EPIWEN$donor==c("secretory_early-mid")] <- "secretory_early-mid"
EPIWEN$type3[EPIWEN$donor==c("secretory_mid")] <- "secretory_mid"
EPIWEN$type3[EPIWEN$donor==c("secretory_late")] <- "secretory_late"


EPI_allsingle_EPI_wen=merge(EPI_ctrl_IUA_allsingle,EPIWEN)

EPI_allsingle_EPI_wen=NormalizeData(EPI_allsingle_EPI_wen)

#进行标准化
DefaultAssay(EPI_allsingle_EPI_wen) <- "RNA"
all.genes.EPI_allsingle_EPI_wen <- rownames(EPI_allsingle_EPI_wen)
EPI_allsingle_EPI_wen <- ScaleData(object = EPI_allsingle_EPI_wen, features = all.genes.EPI_allsingle_EPI_wen)

EPI_donor_marker=c("PLAU","MMP7","THBS1","CADM1","NPAS3","ATP1A1",
                   "ANK3","ALPL","TRAK1","SCGB1D2","MT1F","MT1X",
                   "MT1E","MT1G","CXCL14","MAOA","DPP4","NUPR1",
                   "GPX3","PAEP")

EPI_allsingle_EPI_wen$type3[EPI_allsingle_EPI_wen$type3%in%"secretory_early-mid"]<-"secretory_mid"

EPI_allsingle_EPI_wen$type3=factor(EPI_allsingle_EPI_wen$type3,levels = c("CTRL1","CTRL2","IUA1","IUA2","IUA3","IUA4","IUA5","IUA6","IUA7",
                                                                          "proliferative","secretory_early","secretory_mid","secretory_late"))
EPI_allsingle_EPI_wen$type3=as.factor(EPI_allsingle_EPI_wen$type3)
table(EPI_allsingle_EPI_wen$type3)


jpeg("/data/DATAprocess/chentw/zigongneimo/原图绘制/20221203/EPI_1.jpeg",height = 3600,width = 5500,res=800)
DotPlot(EPI_allsingle_EPI_wen,group.by = "type3",features = EPI_donor_marker)+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+coord_flip()+
  theme(axis.text.x = element_blank(),axis.text.y = element_blank(),panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),axis.title = element_blank(),
        axis.text = element_blank(),axis.ticks = element_blank(),axis.line = element_blank(),
        legend.position = "none")+xlab("")+ylab("")

dev.off()

jpeg("/data/DATAprocess/chentw/zigongneimo/原图绘制/20221203/EPI_1.jpeg",height = 4500,width = 5500,res=800)
DotPlot(EPI_allsingle_EPI_wen,group.by = "type3",features = EPI_donor_marker)+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+coord_flip()+
  theme(legend.position = "none")+xlab("")+ylab("")

dev.off()




#将上皮细胞和文章的上皮细胞进行相关性分析并进行热图绘制-----
umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")

umap_ctrl_IUA_allsingle@active.ident<-umap_ctrl_IUA_allsingle$integrated_snn_res.0.3
umap_ctrl_IUA_allsingle@active.ident=factor(umap_ctrl_IUA_allsingle@active.ident,levels = 0:19)
new.cluster.ids <- c("Stromal cells","Stromal cells","Stromal cells","Immune cells","Immune cells","Epithelium cells",
                     "Stromal cells","Stromal cells","Stromal cells","Immune cells","Immune cells","Epithelium cells",
                     "Endothelium cells","Stromal cells","Stromal cells","Immune cells","Immune cells","Immune cells","Immune cells","Endothelium cells")


names(new.cluster.ids) <- levels(umap_ctrl_IUA_allsingle)
umap_ctrl_IUA_allsingle <- RenameIdents(umap_ctrl_IUA_allsingle, new.cluster.ids)

umap_ctrl_IUA_allsingle$cell_type1110=umap_ctrl_IUA_allsingle@active.ident

EPI_ctrl_IUA_allsingle=subset(umap_ctrl_IUA_allsingle,cell_type1110%in%"Epithelium cells")

EPIWEN=readRDS("/mnt/chentw/zigongneimo/photo/文章10x和我们数据整合分析/EPIWEN.rds")

EPIWEN$type3 <- "proliferative"
EPIWEN$type3[EPIWEN$donor==c("proliferative")] <- "proliferative"
EPIWEN$type3[EPIWEN$donor==c("secretory_early")] <- "secretory_early"
EPIWEN$type3[EPIWEN$donor==c("secretory_early-mid")] <- "secretory_early-mid"
EPIWEN$type3[EPIWEN$donor==c("secretory_mid")] <- "secretory_mid"
EPIWEN$type3[EPIWEN$donor==c("secretory_late")] <- "secretory_late"


EPI_allsingle_EPI_wen=merge(EPI_ctrl_IUA_allsingle,EPIWEN)

EPI_allsingle_EPI_wen=NormalizeData(EPI_allsingle_EPI_wen)

#进行标准化
DefaultAssay(EPI_allsingle_EPI_wen) <- "RNA"
all.genes.EPI_allsingle_EPI_wen <- rownames(EPI_allsingle_EPI_wen)
EPI_allsingle_EPI_wen <- ScaleData(object = EPI_allsingle_EPI_wen, features = all.genes.EPI_allsingle_EPI_wen)


#开始相关性热图绘制-1000
av=AverageExpression(EPI_allsingle_EPI_wen,
                     group.by = "type3",
                     assays = "RNA")

av=av[[1]]

cg=names(tail(sort(apply(av,1,sd)),1000))


View(av[cg,])


pheatmap(cor(av[cg,],method = 'spearman'))



#开始相关性热图绘制-2000
av=AverageExpression(EPI_allsingle_EPI_wen,
                     group.by = "type3",
                     assays = "RNA")

av=av[[1]]

cg=names(tail(sort(apply(av,1,sd)),2000))


View(av[cg,])


pheatmap(cor(av[cg,],method = 'spearman'),cluster_rows = F,cluster_cols = F)

#----
CTRL1=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/H.rds")
CTRL2=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/qian.rds")

IUA1=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/IUAshou.rds")
IUA2=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/IUAxia.rds")
IUA3=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/li.rds")
IUA4=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/IUAfeng.rds")
IUA5=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/IUA_GWY_additional.rds")
IUA6=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/IUA_BYQ.rds")
IUA7=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/IUA_WJM.rds")


#Ctrl和IUA进行合并
##给样本加上注释信息
###type1
CTRL1$type1="CTRL"
CTRL2$type1="CTRL"
IUA1$type1="IUA"
IUA2$type1="IUA"
IUA3$type1="IUA"
IUA4$type1="IUA"
IUA5$type1="IUA"
IUA6$type1="IUA"
IUA7$type1="IUA"

###type2
CTRL1$type2="CTRL"
CTRL2$type2="CTRL"
IUA1$type2="IUA1"
IUA2$type2="IUA1"
IUA3$type2="IUA1"
IUA4$type2="IUA1"
IUA5$type2="IUA2"
IUA6$type2="IUA2"
IUA7$type2="IUA2"

###type3
CTRL1$type3="CTRL1"
CTRL2$type3="CTRL2"
IUA1$type3="IUA1"
IUA2$type3="IUA2"
IUA3$type3="IUA3"
IUA4$type3="IUA4"
IUA5$type3="IUA5"
IUA6$type3="IUA6"
IUA7$type3="IUA7"

harmony_all=merge(CTRL1,y=c(CTRL2,IUA1,IUA2,IUA3,IUA4,IUA5,IUA6,IUA7))


harmony_all=NormalizeData(harmony_all)
harmony_all <- FindVariableFeatures(harmony_all, selection.method = "vst", nfeatures = 2000)
VlnPlot(harmony_all, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)

#harmony_all进行标准化然后PCA分析
DefaultAssay(harmony_all) <- "RNA"
all.genes_harmony_all <- rownames(harmony_all)
harmony_all <- ScaleData(object = harmony_all, features = all.genes_harmony_all)
harmony_all <- RunPCA(harmony_all, features = VariableFeatures(object = harmony_all))


#harmony_all进行细胞细胞聚类分析(UMAP)
harmony_all=RunUMAP(harmony_all, dims = 1:30)
DimPlot(harmony_all,reduction="umap",label=TRUE)
harmony_all <- FindNeighbors(harmony_all, dims = 1:30)
harmony_all <- FindClusters(harmony_all, resolution = 0.3)
harmony_all <- FindClusters(harmony_all, resolution = 0.6)
DimPlot(harmony_all,reduction="umap",label=TRUE)


#harmony_all进行细胞细胞聚类分析(tsne)
harmony_all=RunTSNE(harmony_all, dims = 1:30)
DimPlot(harmony_all,reduction="tsne",label=TRUE)
harmony_all <- FindNeighbors(harmony_all, dims = 1:30)
harmony_all <- FindClusters(harmony_all, resolution = 0.3)
DimPlot(harmony_all,reduction="tsne",label=TRUE)

harmony_all=harmony_all%>%RunHarmony("type3",plot_covergence=T)

harmony_all=RunUMAP(harmony_all,reduction="harmony",dims=1:30)
harmony_all=FindNeighbors(harmony_all,reduction="harmony",dims=1:30)
harmony_all=FindClusters(harmony_all, resolution = 0.3,reduction="harmony")
harmony_all=FindClusters(harmony_all, resolution = 0.6,reduction="harmony")
DimPlot(harmony_all,reduction="umap",label=TRUE)

DimPlot(harmony_all,reduction="umap",label=TRUE)


#去除检测的双细胞部分并重新找高可变基因并且绘图
double1=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/double/H_doublet.txt")
double2=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/double/IUABYQ_doublet.txt")
double3=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/double/IUAfeng_doublet.txt")
double4=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/double/IUAGWY_doublet.txt")
double5=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/double/IUAshou_doublet.txt")
double6=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/double/IUAWJM_doublet.txt")
double7=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/double/IUAxia_doublet.txt")
double8=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/double/li_doublet.txt")
double9=read.csv("/data/DATAprocess/chentw/zigongneimo/python运行/double/qian_doublet.txt")

CTRL1_single=CTRL1[,colnames(CTRL1)%in%subset(double1,double1$predicted_doublets%in%"False")$"barcode"]
CTRL2_single=CTRL2[,colnames(CTRL2)%in%subset(double9,double9$predicted_doublets%in%"False")$"barcode"]
IUA1_single=IUA1[,colnames(IUA1)%in%subset(double5,double5$predicted_doublets%in%"False")$"barcode"]

IUA2_single=IUA2[,colnames(IUA2)%in%subset(double7,double7$predicted_doublets%in%"False")$"barcode"]

IUA3_single=IUA3[,colnames(IUA3)%in%subset(double8,double8$predicted_doublets%in%"False")$"barcode"]

IUA4_single=IUA4[,colnames(IUA4)%in%subset(double3,double3$predicted_doublets%in%"False")$"barcode"]

IUA5_single=IUA5[,colnames(IUA5)%in%subset(double4,double4$predicted_doublets%in%"False")$"barcode"]

IUA6_single=IUA6[,colnames(IUA6)%in%subset(double2,double2$predicted_doublets%in%"False")$"barcode"]

IUA7_single=IUA7[,colnames(IUA7)%in%subset(double6,double6$predicted_doublets%in%"False")$"barcode"]




harmony_allsingle=merge(CTRL1_single,y=c(CTRL2_single,IUA1_single,IUA2,IUA3_single,
                                         IUA4_single,IUA5_single,IUA6_single,IUA7_single))


harmony_allsingle=NormalizeData(harmony_allsingle)
harmony_allsingle <- FindVariableFeatures(harmony_allsingle, selection.method = "vst", nfeatures = 2000)
VlnPlot(harmony_allsingle, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)

mtGenes=grep('^MT-',rownames(harmony_allsingle),value=TRUE)
RPGenes=grep('^RP',rownames(harmony_allsingle),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
stress_gene=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
              "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")
stress_genes=c('FOSB','FOS','JUN','JUNB','JUND','ATF3','EGR1','HSPA1A','HSPA1B','HSP90AB1','HSPA8','HSPB1','IER3','IER2','BTG1','BTG2','DUSP1')


subgene=c(mtGenes,RPGenes,HB.genes,g2m.genes,stress_gene,s.genes)

harmony_allsingle@assays$RNA@var.features = setdiff(harmony_allsingle@assays$RNA@var.features,subgene)  #去除subgene影响







#harmony_allsingle进行标准化然后PCA分析
DefaultAssay(harmony_allsingle) <- "RNA"
all.genes_harmony_allsingle <- rownames(harmony_allsingle)
harmony_allsingle <- ScaleData(object = harmony_allsingle, features = all.genes_harmony_allsingle)
harmony_allsingle <- RunPCA(harmony_allsingle, features = VariableFeatures(object = harmony_allsingle))


#harmony_allsingle进行细胞细胞聚类分析(UMAP)
harmony_allsingle=RunUMAP(harmony_allsingle, dims = 1:30)
DimPlot(harmony_allsingle,reduction="umap",label=TRUE)
harmony_allsingle <- FindNeighbors(harmony_allsingle, dims = 1:30)
harmony_allsingle <- FindClusters(harmony_allsingle, resolution = 0.3)
harmony_allsingle <- FindClusters(harmony_allsingle, resolution = 0.6)
DimPlot(harmony_allsingle,reduction="umap",label=TRUE)


#harmony_allsingle进行细胞细胞聚类分析(tsne)
harmony_allsingle=RunTSNE(harmony_allsingle, dims = 1:30)
DimPlot(harmony_allsingle,reduction="tsne",label=TRUE)
harmony_allsingle <- FindNeighbors(harmony_allsingle, dims = 1:30)
harmony_allsingle <- FindClusters(harmony_allsingle, resolution = 0.3)
DimPlot(harmony_allsingle,reduction="tsne",label=TRUE)

harmony_allsingle=harmony_allsingle%>%RunHarmony("type3",plot_covergence=T)

harmony_allsingle=RunUMAP(harmony_allsingle,reduction="harmony",dims=1:30)
harmony_allsingle=FindNeighbors(harmony_allsingle,reduction="harmony",dims=1:30)
harmony_allsingle=FindClusters(harmony_allsingle, resolution = 0.3,reduction="harmony")
harmony_allsingle=FindClusters(harmony_allsingle, resolution = 0.6,reduction="harmony")
DimPlot(harmony_allsingle,reduction="umap",label=TRUE)

DimPlot(harmony_allsingle,reduction="umap",label=TRUE)


















#对基质细胞进行harmony整合分析----
library(devtools) 
install_github("immunogenomics/harmony")
library(harmony)

umap_WEN2_STR=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_WEN2_STR.rds")

umap_WEN2_STR2=umap_WEN2_STR%>%RunHarmony("type3",plot_covergence=T)

umap_WEN2_STR2=RunUMAP(umap_WEN2_STR2,reduction="harmony",dims=1:30)
umap_WEN2_STR2=FindNeighbors(umap_WEN2_STR2,reduction="harmony",dims=1:30)
umap_WEN2_STR2=FindClusters(umap_WEN2_STR2, resolution = 0.3)
umap_WEN2_STR2=FindClusters(umap_WEN2_STR2, resolution = 0.6)

identity(umap_WEN2_STR2)

DimPlot(umap_WEN2_STR2,reduction = "umap",group.by = "type1")

umap_WEN2_STR2$celltype1126="UMAP"

umap_WEN2_STR2$subtypes[umap_WEN2_STR2$type1124%in%"UMAP"]="UMAP"

umap_WEN2_STR2$type1124="WEN2"

umap_WEN2_STR2$type1124[umap_WEN2_STR2$cell_type1110%in%"Stromal cells"]="UMAP"

color=c(rep("red",1),rep("grey",12))

DimPlot(umap_WEN2_STR2,split.by = "type1124",label = F,repel = T,group.by = "subtypes")+
  scale_color_manual(values = color)


#对基质细胞进行直接merge----
library(Seurat)
umap_WEN2_STR=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_WEN2_STR.rds")

umap_WEN2_STR.list <- SplitObject(umap_WEN2_STR, split.by = "type3")

umap_WEN2_STR_merge=merge(umap_WEN2_STR.list[[1]],y = c(umap_WEN2_STR.list[[2]],umap_WEN2_STR.list[[3]],umap_WEN2_STR.list[[4]],
                                                        umap_WEN2_STR.list[[5]],
                                                        umap_WEN2_STR.list[[6]],umap_WEN2_STR.list[[7]],umap_WEN2_STR.list[[8]],
                                                        umap_WEN2_STR.list[[9]],umap_WEN2_STR.list[[10]],umap_WEN2_STR.list[[11]],
                                                        umap_WEN2_STR.list[[12]],umap_WEN2_STR.list[[13]],umap_WEN2_STR.list[[14]],
                                                        umap_WEN2_STR.list[[15]],umap_WEN2_STR.list[[16]],umap_WEN2_STR.list[[17]],
                                                        umap_WEN2_STR.list[[18]],umap_WEN2_STR.list[[19]],umap_WEN2_STR.list[[20]],
                                                        umap_WEN2_STR.list[[21]],umap_WEN2_STR.list[[22]],umap_WEN2_STR.list[[23]]))


umap_WEN2_STR_merge=NormalizeData(umap_WEN2_STR_merge)
umap_WEN2_STR_merge <- FindVariableFeatures(umap_WEN2_STR_merge, selection.method = "vst", nfeatures = 2000)
VlnPlot(umap_WEN2_STR_merge, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)

#umap_WEN2_STR_merge进行标准化然后PCA分析
DefaultAssay(umap_WEN2_STR_merge) <- "RNA"
all.genes_umap_WEN2_STR_merge <- rownames(umap_WEN2_STR_merge)
umap_WEN2_STR_merge <- ScaleData(object = umap_WEN2_STR_merge, features = all.genes_umap_WEN2_STR_merge)
umap_WEN2_STR_merge <- RunPCA(umap_WEN2_STR_merge, features = VariableFeatures(object = umap_WEN2_STR_merge))


#umap_WEN2_STR_merge进行细胞细胞聚类分析(UMAP)
umap_WEN2_STR_merge=RunUMAP(umap_WEN2_STR_merge, dims = 1:30)
DimPlot(umap_WEN2_STR_merge,reduction="umap",label=TRUE)
umap_WEN2_STR_merge <- FindNeighbors(umap_WEN2_STR_merge, dims = 1:30)
umap_WEN2_STR_merge <- FindClusters(umap_WEN2_STR_merge, resolution = 0.3)
umap_WEN2_STR_merge <- FindClusters(umap_WEN2_STR_merge, resolution = 0.6)
DimPlot(umap_WEN2_STR_merge,reduction="umap",label=TRUE)


#umap_WEN2_STR_merge进行细胞细胞聚类分析(tsne)
umap_WEN2_STR_merge=RunTSNE(umap_WEN2_STR_merge, dims = 1:30)
DimPlot(umap_WEN2_STR_merge,reduction="tsne",label=TRUE)
umap_WEN2_STR_merge <- FindNeighbors(umap_WEN2_STR_merge, dims = 1:30)
umap_WEN2_STR_merge <- FindClusters(umap_WEN2_STR_merge, resolution = 0.3)
DimPlot(umap_WEN2_STR_merge,reduction="tsne",label=TRUE)

#进行绘图
umap_WEN2_STR_merge$type1124="WEN2"

umap_WEN2_STR_merge$subtypes[umap_WEN2_STR_merge$type1124%in%"UMAP"]<-"UMAP"

DimPlot(umap_WEN2_STR_merge,reduction="umap",label=T,group.by = "type3")




#复现文章的数据【所有细胞】------
library(Seurat)
global<-LoadH5Seurat("/data/DATAprocess/chentw/zigongneimo/WEN2/h5ad文件/endo-2022_global.h5seurat",
                     assays="RNA")

DimPlot(global,group.by = "celltype_main")

STR=subset(global,celltype_main%in%"stromal")


STR.list <- SplitObject(STR, split.by = "PID")

STR=merge(STR.list[[1]],y = c(STR.list[[2]],STR.list[[3]],STR.list[[4]],
                              STR.list[[5]],
                              STR.list[[6]],STR.list[[7]],STR.list[[8]],
                              STR.list[[9]],STR.list[[10]],STR.list[[11]],
                              STR.list[[12]],STR.list[[13]],STR.list[[14]]))

STR=NormalizeData(STR)
STR <- FindVariableFeatures(STR, selection.method = "vst", nfeatures = 2000)
VlnPlot(STR, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)



s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
STR <- CellCycleScoring(STR,s.features = s.genes, g2m.features = g2m.genes,set.ident = T)

STR[["percent.mt"]]<-PercentageFeatureSet(STR,pattern = "^MT-")
STR[["percent.rp"]]<-PercentageFeatureSet(STR,pattern = "^RP")

HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")

STR[["percent.HB"]]<-PercentageFeatureSet(STR,features = HB.genes)

head(STR@meta.data)






#STR进行标准化然后PCA分析
DefaultAssay(STR) <- "RNA"
all.genes_STR <- rownames(STR)
STR <- ScaleData(object = STR, features = all.genes_STR,
                 vars.to.regress = c("percent.mt","S.Score","G2M.Score","percent.rp","percent.HB"))

STR <- RunPCA(STR, features = VariableFeatures(object = STR))






#STR进行细胞细胞聚类分析(UMAP)
STR=RunUMAP(STR, dims = 1:30)
DimPlot(STR,reduction="umap",label=TRUE)
STR <- FindNeighbors(STR, dims = 1:30)
STR <- FindClusters(STR, resolution = 0.3)
STR <- FindClusters(STR, resolution = 0.6)
DimPlot(STR,reduction="umap",label=TRUE)


library(harmony)

STR=STR%>%RunHarmony("PID",plot_covergence=T)

STR=RunUMAP(STR,reduction="harmony",dims=1:30)
STR=FindNeighbors(STR,reduction="harmony",dims=1:30)
STR=FindClusters(STR, resolution = 0.3)
STR=FindClusters(STR, resolution = 0.6)

identity(STR)

DimPlot(STR,reduction="umap",label=TRUE,group.by = "celltype")

#去除细胞周期影响并观察








STR2<-LoadH5Seurat("/data/DATAprocess/chentw/zigongneimo/WEN2/h5ad文件/endo-2022_stromal.h5seurat",
                   assays="RNA")


STR2=STR2%>%RunHarmony("PID",plot_covergence=T)

STR2=RunUMAP(STR2,reduction="umap",dims=1:10)
STR2=FindNeighbors(STR2,reduction="umap",dims=1:10)
STR2=FindClusters(STR2, resolution = 0.3)
STR2=FindClusters(STR2, resolution = 0.6)

identity(STR2)

DimPlot(STR2,reduction="umap",label=TRUE,group.by = "subtypes")

#开始STR的harmony整合-----
library(Seurat)
library(SeuratDisk)
library(harmony)
library(dplyr)
library(tidyr)
library(ggplot2)
library(clusterProfiler)
library(org.Hs.eg.db)
library(RColorBrewer)
library(export)




umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")

DimPlot(umap_ctrl_IUA_allsingle,label = T,group.by = "cell_type1110",label.size = 8)+
  ggtitle("")

gene=c("ACTA2","PDGFRB","RGS5")


FeaturePlot(umap_ctrl_IUA_allsingle,features = gene)

DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type",label = T)+
  ggtitle("")


#重新定义并且开始绘制原图
umap_ctrl_IUA_allsingle$cell_type0216=
  umap_ctrl_IUA_allsingle$cell_type1110

umap_ctrl_IUA_allsingle$cell_type0216=as.character(umap_ctrl_IUA_allsingle$cell_type0216)

DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type")

DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type0216",label = T)+ggtitle("")+theme(legend.position = "none")


umap_ctrl_IUA_allsingle=subset(umap_ctrl_IUA_allsingle,cell_type!="Erythrocyte")
DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type1110")
table(umap_ctrl_IUA_allsingle$cell_type0216)

STR_allsingle=subset(umap_ctrl_IUA_allsingle,cell_type1110%in%"Stromal cells")




STR_allsingle_embeddings=STR_allsingle$umap@cell.embeddings
STR_allsingle_embeddings=as.data.frame(STR_allsingle_embeddings)

STR_allsingle_embeddings$UMAP_1=as.numeric(as.character(STR_allsingle_embeddings$UMAP_1))
STR_allsingle_embeddings$UMAP_2=as.numeric(as.character(STR_allsingle_embeddings$UMAP_2))
STR_allsingle$UMAP_1=STR_allsingle_embeddings$UMAP_1
STR_allsingle$UMAP_2=STR_allsingle_embeddings$UMAP_2

STR_allsingle=subset(STR_allsingle,UMAP_1<6&UMAP_2>-7)

DimPlot(STR_allsingle)

RPGenes=c(grep('^RPL',rownames(STR_allsingle),value=TRUE),grep('^RPS',rownames(STR_allsingle),value=TRUE))

STR_allsingle[["percent.rp"]] <- PercentageFeatureSet(STR_allsingle, features = RPGenes)
VlnPlot(STR_allsingle, features = c("nFeature_RNA", "nCount_RNA", "percent.rp","percent.mt"), ncol = 4)

DimPlot(STR_allsingle,reduction="umap")

quantile(STR_allsingle$percent.rp)

FeaturePlot(STR_allsingle,features = "percent.rp",
            min.cutoff = 20,order = T,cols = c("grey","red"))

STR_allsingle=subset(STR_allsingle,percent.rp<25)

STR_maker=c("LUM","COL1A1","DCN","MMP11")
EPI_maker=c("EPCAM","KRT18","MMP7","WFDC2","CD9")
Endothelium_maker=c("VWF","CDH5")
smooth_maker=c("MCAM","RGS5","ACTA2")
T_maker=c("CD3D","CD3E","CXCR4","PTPRC","GZMK","CD4","CD8A")
NK_maker=c("NCAM1","GNLY","GZMA")
MAST_maker=c("MS4A2","TPSB2","TPSAB1","CPA3")
B_maker=c("CD79A","CD74")
macr_maker=c("HLA-DRB1","CSF1R","CD14")



#开始进行细胞的过滤【免疫细胞为主】
STR_allsingle_data=
  as.data.frame(STR_allsingle@assays$RNA@data)

STR_allsingle_data=subset(STR_allsingle_data,
                          rownames(STR_allsingle_data)%in%T_maker)

STR_allsingle_data=as.data.frame(t(STR_allsingle_data))


STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$PTPRC<=0)


STR_allsingle=STR_allsingle[,rownames(STR_allsingle_data)]

T_maker=c("CD3D","CD3E","CXCR4","PTPRC","GZMK","CD4","CD8A")
FeaturePlot(STR_allsingle,features = T_maker,min.cutoff = 0)

STR_allsingle_data=
  as.data.frame(STR_allsingle@assays$RNA@data)



STR_allsingle_data=
  as.data.frame(STR_allsingle@assays$RNA@data)

STR_allsingle_data=subset(STR_allsingle_data,
                          rownames(STR_allsingle_data)%in%c(EPI_maker,NK_maker,
                                                            Endothelium_maker,
                                                            smooth_maker,STR_maker,
                                                            T_maker,B_maker,macr_maker,
                                                            c("CD74","LYZ","WFDC2","PAEP","HLA-DRA","CD14")))

STR_allsingle_data=as.data.frame(t(STR_allsingle_data))


quantile(STR_allsingle_data$GNLY,0.9)###
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$GNLY<=0.95)


quantile(STR_allsingle_data$EPCAM,0.92)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$EPCAM<=0.5)


quantile(STR_allsingle_data$VWF,0.99)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$VWF<=0)

quantile(STR_allsingle_data$CXCR4,0.90)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$CXCR4<=0)

quantile(STR_allsingle_data$WFDC2,0.80)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$WFDC2<=0)

quantile(STR_allsingle_data$EPCAM,0.95)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$EPCAM<=0)

quantile(STR_allsingle_data$KRT18,0.95)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$KRT18<=0)


quantile(STR_allsingle_data$CD14,0.95)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$CD14<=0)

quantile(STR_allsingle_data$CD74,0.95)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$CD74<=0)

quantile(STR_allsingle_data$LYZ,0.95)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$LYZ<=0)

quantile(STR_allsingle_data$PAEP,0.95)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$PAEP<=0)


quantile(STR_allsingle_data$`HLA-DRA`,0.95)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$`HLA-DRA`<=0)


STR_allsingle=STR_allsingle[,rownames(STR_allsingle_data)]


table(STR_allsingle$type3)

STR_allsingle=subset(STR_allsingle,type3!="IUA7")


####开始进行重新整合
N=STR_allsingle@assays$RNA@counts

STR_allsingle=CreateSeuratObject(counts = N, meta.data = STR_allsingle@meta.data)

STR_allsingle=NormalizeData(STR_allsingle, normalization.method = "LogNormalize", 
                            scale.factor = 10000)
STR_allsingle <- FindVariableFeatures(STR_allsingle, selection.method = "vst", nfeatures = 2000)
VlnPlot(STR_allsingle, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)

mtGenes=grep('^MT-',rownames(STR_allsingle),value=TRUE)
RPGenes=c(grep('^RPL',rownames(STR_allsingle),value=TRUE),grep('^RPS',rownames(STR_allsingle),value=TRUE))
stress_gene2=grep('^HSP',rownames(STR_allsingle),value=TRUE)
HB_gene2=grep('^HB',rownames(STR_allsingle),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
stress_gene1=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
               "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/gene1.csv")
subgene3=subgene3$cellcyle_gene
subgene4=c("FOS","CXCL2","ZFP36","FOSB","DUSP1","ATF3","CXCL8","NR4A1","CXCL3","PPP1R15A",
           "JUNB","EGR1","HSPA1A","HSPA1B","SOCS3","KLF6","JUN","IER2","CXCL1","NFKBIA",
           "HSPA6","DNAJB1","IER3","CCNL1","MTRNR2L2","IER4","ID1","CEBPD","KRT6A","CYR61",
           "DEPP1","CLDN4","IRF1","DUSP2","BTG2","PLAUR","MAFF","KLF4","PHLDA2","TNFAIP3")
subgene5=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/512个应激反应相关基因.csv")
subgene5=subgene5$gene_symbol

subgene=c(mtGenes,RPGenes,stress_gene2,HB_gene2,HB.genes,s.genes,g2m.genes,stress_gene1,"MKI67",subgene3,subgene4,subgene5)

STR_allsingle@assays$RNA@var.features = setdiff(STR_allsingle@assays$RNA@var.features,subgene)  #去除subgene影响

STR_allsingle@assays$RNA@var.features

#STR_allsingle进行标准化然后PCA分析
DefaultAssay(STR_allsingle) <- "RNA"
all.genes_STR_allsingle <- rownames(STR_allsingle)
STR_allsingle <- ScaleData(object = STR_allsingle, features = all.genes_STR_allsingle)
STR_allsingle <- RunPCA(STR_allsingle, features = VariableFeatures(object = STR_allsingle))


#STR_allsingle进行细胞细胞聚类分析(UMAP)
STR_allsingle=RunUMAP(STR_allsingle, dims = 1:30)
DimPlot(STR_allsingle,reduction="umap",label=TRUE)
STR_allsingle <- FindNeighbors(STR_allsingle, dims = 1:30)
STR_allsingle <- FindClusters(STR_allsingle, resolution = 0.3)
STR_allsingle <- FindClusters(STR_allsingle, resolution = 0.6)
DimPlot(STR_allsingle,reduction="umap",label=TRUE)


#STR_allsingle进行细胞细胞聚类分析(tsne)
STR_allsingle=RunTSNE(STR_allsingle, dims = 1:30)
DimPlot(STR_allsingle,reduction="tsne",label=TRUE)
STR_allsingle <- FindNeighbors(STR_allsingle, dims = 1:30)
STR_allsingle <- FindClusters(STR_allsingle, resolution = 0.3)
DimPlot(STR_allsingle,reduction="tsne",label=TRUE)

DimPlot(STR_allsingle,reduction="umap",label=TRUE)



#开始尝试去除细胞样本批次并且进行新的一个保存

STR_allsingle=STR_allsingle%>%RunHarmony("type3",plot_covergence=T)

DefaultAssay(STR_allsingle) <- "RNA"


STR_allsingle <- RunPCA(STR_allsingle, features = VariableFeatures(object = STR_allsingle))


STR_allsingle=RunUMAP(STR_allsingle,reduction="harmony",dims=1:30)
STR_allsingle=FindNeighbors(STR_allsingle,dims=1:30,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.15,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.17,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.18,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.2,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.3,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.4,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.45,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.5,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.55,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.6,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.7,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.9,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 1,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 1.1,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 1.2,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 1.3,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 1.4,reduction = "harmony")



#查看细胞周期情况
s.genes=Seurat::cc.genes.updated.2019$s.genes
g2m.genes=Seurat::cc.genes.updated.2019$g2m.genes
STR_allsingle <- CellCycleScoring(STR_allsingle, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
DimPlot(STR_allsingle,reduction="Phase",label=TRUE)



DimPlot(STR_allsingle,reduction="umap",label=TRUE,group.by = "Phase")+
  DimPlot(STR_allsingle,reduction="umap",label=TRUE,group.by = "RNA_snn_res.0.3")



DimPlot(STR_allsingle,group.by = "cell_type",label=TRUE)



STR_allsingle=subset(STR_allsingle,cell_type%in%c("Stromal cells",
                                                  "Perivascular cells",
                                                  "Transient state cells","Cycling stromal cells"))


DimPlot(STR_allsingle,group.by = "cell_type",label=TRUE)

FeaturePlot(STR_allsingle,features = c("ACTA2","TAGLN"),
            min.cutoff = 1,order = T,cols = c("grey","red"))

saveRDS(STR_allsingle,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")


#开始定义细胞-----
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

DimPlot(STR,group.by = "cell_type",label = T,label.size = 8)+ggtitle("")

DimPlot(STR,label = T,label.size = 8,group.by = "RNA_snn_res.0.3")+ggtitle("")

STR$cell_type=factor(STR$cell_type)


STR$type4=factor(STR$type4,levels =c("Ctrl","Adjacent","Adhesion") )

#开始对最早的定义结果进行柱形图统计比例并绘制1
table_plot=as.data.frame(table(STR$type4,STR$cell_type))
ggplot(data=table_plot,fill=Var2)+aes(x=Var1,y=stat(table_plot$Freq),fill=Var2)+geom_bar(stat="identity",width=0.75, position="fill")+
  theme_bw()+theme(legend.title=element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank())+ylab("Integrative Celltype(%)")+scale_fill_brewer(palette="Paired")+xlab("")+
  scale_y_continuous(expand=c(0,0),limits=c(0,1),breaks = seq(0,1,0.25),labels = c("0","25","50","75","100"))+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+scale_x_discrete(limits=c("Ctrl","Adjacent","Adhesion"))




#开始对最早的定义结果进行柱形图统计比例并绘制2
table_plot=as.data.frame(table(STR$cell_type,STR$type4))
ggplot(data=table_plot,fill=Var2)+aes(x=Var1,y=stat(table_plot$Freq),fill=Var2)+geom_bar(stat="identity",width=0.75, position="fill")+
  theme_bw()+theme(legend.title=element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank())+ylab("Integrative Celltype(%)")+scale_fill_brewer(palette="Paired")+xlab("")+
  scale_y_continuous(expand=c(0,0),limits=c(0,1),breaks = seq(0,1,0.25),labels = c("0","25","50","75","100"))+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))





#通过之前的marker来定义细胞类型
gene=c("TGFBI","SERPINF1","BMP7","REV3L","RGS5","ISG15","APOD","PDGFRA","PDGFRB","CD34","THY1",
       "SFRP2","SFRP4","FAP","PDPN","COL1A1","MCAM","XIST")

VlnPlot(STR,features = gene,group.by = "RNA_snn_res.0.3",pt.size = 0,ncol = 9)

STR$cell_type0305<-"K"
STR$cell_type0305[STR$RNA_snn_res.0.3%in%"0"]="S0"
STR$cell_type0305[STR$RNA_snn_res.0.3%in%"1"]="REV3L"
STR$cell_type0305[STR$RNA_snn_res.0.3%in%"2"]="S1"
STR$cell_type0305[STR$RNA_snn_res.0.3%in%"3"]="Cycling"
STR$cell_type0305[STR$RNA_snn_res.0.3%in%"4"]="BMP7"
STR$cell_type0305[STR$RNA_snn_res.0.3%in%"5"]="P"
STR$cell_type0305[STR$RNA_snn_res.0.3%in%"6"]="ISG15"
STR$cell_type0305[STR$RNA_snn_res.0.3%in%"7"]="APOD"

STR$cell_type0305=factor(STR$cell_type0305,levels = c("S0","S1","BMP7","Cycling","REV3L","ISG15","APOD","P"))

DimPlot(STR,label = T,label.size = 12,group.by = "cell_type0305",pt.size = 1)+ggtitle("")+
  scale_color_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"))


VlnPlot(STR,group.by = "cell_type0305",features = "IGF1R",pt.size = 0)

STR_gene=c("PDGFRB","LUM","DCN","VIM","TGFBI","PGRMC1","SERPINF1",
           "OGN","BMP7","MKI67","REV3L","APOD","ISG15","RGS5","NOTCH3")
VlnPlot(STR,group.by = "cell_type0305",features =STR_gene,pt.size = 0,
        stack = T)


###marker基因查找并进行富marker基因气泡图绘制并尝试定义
DefaultAssay(STR)<-"RNA"
STR@active.ident=STR$cell_type0305
STR_markers <- FindAllMarkers(object=STR, only.pos = TRUE,min.pct = 0.25, 
                              logfc.threshold = 0.25)
STR_markers <-subset(STR_markers,STR_markers$p_val<0.05)


write.csv(STR_markers,
          "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/STR_markers.csv")

mtGenes=grep('^MT-',rownames(STR),value=TRUE)
RPGenes=c(grep('^RPL',rownames(STR),value=TRUE),grep('^RPS',rownames(STR),value=TRUE))
stress_gene2=grep('^HSP',rownames(STR),value=TRUE)
HB_gene2=grep('^HB',rownames(STR),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
stress_gene1=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
               "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/gene1.csv")
subgene3=subgene3$cellcyle_gene
subgene4=c("FOS","CXCL2","ZFP36","FOSB","DUSP1","ATF3","CXCL8","NR4A1","CXCL3","PPP1R15A",
           "JUNB","EGR1","HSPA1A","HSPA1B","SOCS3","KLF6","JUN","IER2","CXCL1","NFKBIA",
           "HSPA6","DNAJB1","IER3","CCNL1","MTRNR2L2","IER4","ID1","CEBPD","KRT6A","CYR61",
           "DEPP1","CLDN4","IRF1","DUSP2","BTG2","PLAUR","MAFF","KLF4","PHLDA2","TNFAIP3")
subgene5=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/512个应激反应相关基因.csv")
subgene5=subgene5$gene_symbol

subgene=c(mtGenes,RPGenes,stress_gene2,HB_gene2,HB.genes,s.genes,g2m.genes,stress_gene1,"MKI67",subgene3,subgene4,subgene5)


#包含非特异的marker
STR_markers=subset(STR_markers,!gene%in%subgene)

STR_markers_top10 <- STR_markers %>% group_by(cluster) %>% top_n(n = 10, wt = avg_log2FC)

DefaultAssay(STR)<-"RNA"

STR_markers_top10=unique(STR_markers_top10$gene)

DotPlot(STR,features=STR_markers_top10,group.by = "cell_type0305")+
  theme(axis.text.x= element_text(size=9,angle = 45,vjust = 1,hjust = 1,face= "bold"))+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])))+xlab("")+ylab("")



#特异的marker
STR_markers=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/STR_markers.csv")

Z=STR_markers[duplicated(STR_markers$gene),]

N=subset(STR_markers,!gene%in%unique(Z$gene))

write.csv(N,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/STR_markers_unique.csv")

table(duplicated(N$gene))

STR_markers_top10 <- N %>% group_by(cluster) %>% top_n(n = 10, wt = avg_log2FC)

DefaultAssay(STR)<-"RNA"

STR_markers_top10=unique(STR_markers_top10$gene)

DotPlot(STR,features=STR_markers_top10,group.by = "cell_type0305")+
  theme(axis.text.x= element_text(size=9,angle = 45,vjust = 1,hjust = 1,face= "bold"))+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])))+xlab("")+ylab("")

#开始对最早的定义结果进行柱形图统计比例并绘制1
table_plot=as.data.frame(table(STR$type4,STR$cell_type0305))
ggplot(data=table_plot,fill=Var2)+aes(x=Var1,y=stat(table_plot$Freq),fill=Var2)+geom_bar(stat="identity",width=0.75, position="fill")+
  theme_bw()+theme(legend.title=element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank())+ylab("Integrative Celltype(%)")+scale_fill_brewer(palette="Paired")+xlab("")+
  scale_y_continuous(expand=c(0,0),limits=c(0,1),breaks = seq(0,1,0.25),labels = c("0","25","50","75","100"))+scale_x_discrete(limits=c("Ctrl","Adjacent","Adhesion"))+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"))+
  theme(axis.text.x =element_text(size = 20) )+ylab("")

#开始对最早的定义结果进行柱形图统计比例并绘制2
table_plot=as.data.frame(table(STR$cell_type0305,STR$type4))
ggplot(data=table_plot,fill=Var2)+aes(x=Var1,y=stat(table_plot$Freq),fill=Var2)+geom_bar(stat="identity",width=0.75, position="fill")+
  theme_bw()+theme(legend.title=element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank())+ylab("Integrative Celltype(%)")+scale_fill_brewer(palette="Paired")+xlab("")+
  scale_y_continuous(expand=c(0,0),limits=c(0,1),breaks = seq(0,1,0.25),labels = c("0","25","50","75","100"))+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+ylab("")


#开始富集分析
STR_markers=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/STR_markers.csv")

#对marker基因进行富集分析
GO_STR=function(X){
  N=as.character(X)
  N=subset(STR_markers,STR_markers$cluster%in%N)
  N=N$gene
  G <- bitr(N, fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
  G=G$ENTREZID
  ego <- enrichGO(OrgDb="org.Hs.eg.db", gene = G, ont = "ALL",readable= TRUE)
  file1=paste("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/",as.character(X),"_ego.csv",sep = "")
  print(file1)
  write.csv(ego,file1)
}

for (i in names(table(STR_markers$cluster))) {GO_STR(i)
}

#挑前8富集通路进行绘图-气泡图---unique的通路-----
S0_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/S0_ego.csv")
S1_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/S1_ego.csv")
BMP7_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/BMP7_ego.csv")
Cycling_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/Cycling_ego.csv")
REV3L_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/REV3L_ego.csv")
ISG15_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/ISG15_ego.csv")
APOD_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/APOD_ego.csv")
P_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/P_ego.csv")

S0_ego$cluster="S0"
S1_ego$cluster="S1"
BMP7_ego$cluster="BMP7"
Cycling_ego$cluster="Cycling"
REV3L_ego$cluster="REV3L"
ISG15_ego$cluster="ISG15"
APOD_ego$cluster="APOD"
P_ego$cluster="P"

S0_ego=S0_ego[order(S0_ego$pvalue),]
S1_ego=S1_ego[order(S1_ego$pvalue),]
BMP7_ego=BMP7_ego[order(BMP7_ego$pvalue),]
Cycling_ego=Cycling_ego[order(Cycling_ego$pvalue),]
REV3L_ego=REV3L_ego[order(REV3L_ego$pvalue),]
ISG15_ego=ISG15_ego[order(ISG15_ego$pvalue),]
APOD_ego=APOD_ego[order(APOD_ego$pvalue),]
P_ego=P_ego[order(P_ego$pvalue),]

ego_all=rbind(S0_ego,S1_ego,BMP7_ego,Cycling_ego,REV3L_ego,
              ISG15_ego,APOD_ego,P_ego)

Z=ego_all[duplicated(ego_all$Description),]

N=subset(ego_all,!Description%in%unique(Z$Description))

table(duplicated(N$Description))

S0_ego=subset(N,cluster%in%"S0")
S1_ego=subset(N,cluster%in%"S1")
BMP7_ego=subset(N,cluster%in%"BMP7")
Cycling_ego=subset(N,cluster%in%"Cycling")
REV3L_ego=subset(N,cluster%in%"REV3L")
ISG15_ego=subset(N,cluster%in%"ISG15")
APOD_ego=subset(N,cluster%in%"APOD")
P_ego=subset(N,cluster%in%"P")

S0_des=S0_ego[1:8,]$Description
S1_des=S1_ego[1:8,]$Description
BMP7_des=BMP7_ego[1:8,]$Description
Cycling_des=Cycling_ego[1:8,]$Description
REV3L_des=REV3L_ego[1:8,]$Description
ISG15_des=ISG15_ego[1:8,]$Description
APOD_des=APOD_ego[1:8,]$Description
P_des=P_ego[1:8,]$Description



S0_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/S0_ego.csv")
S1_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/S1_ego.csv")
BMP7_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/BMP7_ego.csv")
Cycling_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/Cycling_ego.csv")
REV3L_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/REV3L_ego.csv")
ISG15_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/ISG15_ego.csv")
APOD_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/APOD_ego.csv")
P_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/P_ego.csv")

S0_ego$cluster="S0"
S1_ego$cluster="S1"
BMP7_ego$cluster="BMP7"
Cycling_ego$cluster="Cycling"
REV3L_ego$cluster="REV3L"
ISG15_ego$cluster="ISG15"
APOD_ego$cluster="APOD"
P_ego$cluster="P"


S0_ego=S0_ego[order(S0_ego$pvalue),]
S1_ego=S1_ego[order(S1_ego$pvalue),]
BMP7_ego=BMP7_ego[order(BMP7_ego$pvalue),]
Cycling_ego=Cycling_ego[order(Cycling_ego$pvalue),]
REV3L_ego=REV3L_ego[order(REV3L_ego$pvalue),]
ISG15_ego=ISG15_ego[order(ISG15_ego$pvalue),]
APOD_ego=APOD_ego[order(APOD_ego$pvalue),]
P_ego=P_ego[order(P_ego$pvalue),]


ego_all=rbind(S0_ego,S1_ego,BMP7_ego,Cycling_ego,REV3L_ego,
              ISG15_ego,APOD_ego,P_ego)

ego_all=subset(ego_all,Description%in%c(S0_des,S1_des,BMP7_des,Cycling_des,REV3L_des,
                                        ISG15_des,APOD_des,P_des))



ego_all$cluster=factor(ego_all$cluster,levels = c("S0","S1","BMP7","Cycling","REV3L","ISG15","APOD","P"))


ggplot(data=ego_all)+aes(x=cluster,y=Description,color=-log10(pvalue))+geom_point(aes(size=Count,color= -log10(pvalue)))+
  scale_y_discrete(limits=unique(ego_all$Description))+theme_bw()+xlab("Cluster")+
  scale_color_gradient(low="blue",high = "red")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+xlab("")+
  ylab("")



#挑前8富集通路进行绘图-气泡图---unique的通路-----
S0_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/S0_ego.csv")
S1_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/S1_ego.csv")
BMP7_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/BMP7_ego.csv")
Cycling_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/Cycling_ego.csv")
REV3L_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/REV3L_ego.csv")
ISG15_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/ISG15_ego.csv")
APOD_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/APOD_ego.csv")
P_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/P_ego.csv")

S0_ego$cluster="S0"
S1_ego$cluster="S1"
BMP7_ego$cluster="BMP7"
Cycling_ego$cluster="Cycling"
REV3L_ego$cluster="REV3L"
ISG15_ego$cluster="ISG15"
APOD_ego$cluster="APOD"
P_ego$cluster="P"

S0_ego=S0_ego[order(S0_ego$pvalue),]
S1_ego=S1_ego[order(S1_ego$pvalue),]
BMP7_ego=BMP7_ego[order(BMP7_ego$pvalue),]
Cycling_ego=Cycling_ego[order(Cycling_ego$pvalue),]
REV3L_ego=REV3L_ego[order(REV3L_ego$pvalue),]
ISG15_ego=ISG15_ego[order(ISG15_ego$pvalue),]
APOD_ego=APOD_ego[order(APOD_ego$pvalue),]
P_ego=P_ego[order(P_ego$pvalue),]

ego_all=rbind(S0_ego,S1_ego,BMP7_ego,Cycling_ego,REV3L_ego,
              ISG15_ego,APOD_ego,P_ego)

Z=ego_all[duplicated(ego_all$Description),]

N=subset(ego_all,!Description%in%unique(Z$Description))

table(duplicated(N$Description))

S0_ego=subset(N,cluster%in%"S0")
S1_ego=subset(N,cluster%in%"S1")
BMP7_ego=subset(N,cluster%in%"BMP7")
Cycling_ego=subset(N,cluster%in%"Cycling")
REV3L_ego=subset(N,cluster%in%"REV3L")
ISG15_ego=subset(N,cluster%in%"ISG15")
APOD_ego=subset(N,cluster%in%"APOD")
P_ego=subset(N,cluster%in%"P")

S0_des=S0_ego[1:8,]$Description
S1_des=S1_ego[1:8,]$Description
BMP7_des=BMP7_ego[1:8,]$Description
Cycling_des=Cycling_ego[1:8,]$Description
REV3L_des=REV3L_ego[1:8,]$Description
ISG15_des=ISG15_ego[1:8,]$Description
APOD_des=APOD_ego[1:8,]$Description
P_des=P_ego[1:8,]$Description



S0_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/S0_ego.csv")
S1_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/S1_ego.csv")
BMP7_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/BMP7_ego.csv")
Cycling_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/Cycling_ego.csv")
REV3L_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/REV3L_ego.csv")
ISG15_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/ISG15_ego.csv")
APOD_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/APOD_ego.csv")
P_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/P_ego.csv")

S0_ego$cluster="S0"
S1_ego$cluster="S1"
BMP7_ego$cluster="BMP7"
Cycling_ego$cluster="Cycling"
REV3L_ego$cluster="REV3L"
ISG15_ego$cluster="ISG15"
APOD_ego$cluster="APOD"
P_ego$cluster="P"


S0_ego=S0_ego[order(S0_ego$pvalue),]
S1_ego=S1_ego[order(S1_ego$pvalue),]
BMP7_ego=BMP7_ego[order(BMP7_ego$pvalue),]
Cycling_ego=Cycling_ego[order(Cycling_ego$pvalue),]
REV3L_ego=REV3L_ego[order(REV3L_ego$pvalue),]
ISG15_ego=ISG15_ego[order(ISG15_ego$pvalue),]
APOD_ego=APOD_ego[order(APOD_ego$pvalue),]
P_ego=P_ego[order(P_ego$pvalue),]


ego_all=rbind(S0_ego,REV3L_ego,S1_ego,Cycling_ego,BMP7_ego,
              P_ego,ISG15_ego,APOD_ego)

ego_all=subset(ego_all,Description%in%c(S0_des,REV3L_des,S1_des,Cycling_des,BMP7_des,
                                        P_des,ISG15_des,APOD_des))



ego_all$cluster=factor(ego_all$cluster,levels = c("S0","REV3L","S1","Cycling","BMP7","P","ISG15","APOD"))


ggplot(data=ego_all)+aes(x=cluster,y=Description,color=-log10(pvalue))+geom_point(aes(size=Count,color= -log10(pvalue)))+
  scale_y_discrete(limits=unique(ego_all$Description))+theme_bw()+xlab("Cluster")+
  scale_color_gradient(low="blue",high = "red")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+xlab("")+
  ylab("")+scale_x_discrete(labels=c("FIB1","FIB2","FIB3","FIB4","FIB5","FIB6","FIB7","FIB8"))

write.csv(ego_all,
          "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/原图绘制/2024/GO_PLOT/ego_all.csv")


#挑前8富集通路进行绘图-气泡图---非unique的通路
S0_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/S0_ego.csv")
S1_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/S1_ego.csv")
BMP7_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/BMP7_ego.csv")
Cycling_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/Cycling_ego.csv")
REV3L_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/REV3L_ego.csv")
ISG15_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/ISG15_ego.csv")
APOD_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/APOD_ego.csv")
P_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker基因表_celltype0305/P_ego.csv")

S0_ego$cluster="S0"
S1_ego$cluster="S1"
BMP7_ego$cluster="BMP7"
Cycling_ego$cluster="Cycling"
REV3L_ego$cluster="REV3L"
ISG15_ego$cluster="ISG15"
APOD_ego$cluster="APOD"
P_ego$cluster="P"

S0_ego=S0_ego[order(S0_ego$pvalue),]
S1_ego=S1_ego[order(S1_ego$pvalue),]
BMP7_ego=BMP7_ego[order(BMP7_ego$pvalue),]
Cycling_ego=Cycling_ego[order(Cycling_ego$pvalue),]
REV3L_ego=REV3L_ego[order(REV3L_ego$pvalue),]
ISG15_ego=ISG15_ego[order(ISG15_ego$pvalue),]
APOD_ego=APOD_ego[order(APOD_ego$pvalue),]
P_ego=P_ego[order(P_ego$pvalue),]





S0_des=S0_ego[1:8,]$Description
S1_des=S1_ego[1:8,]$Description
BMP7_des=BMP7_ego[1:8,]$Description
Cycling_des=Cycling_ego[1:8,]$Description
REV3L_des=REV3L_ego[1:8,]$Description
ISG15_des=ISG15_ego[1:8,]$Description
APOD_des=APOD_ego[1:8,]$Description
P_des=P_ego[1:8,]$Description

ego_all=rbind(S0_ego,S1_ego,BMP7_ego,Cycling_ego,REV3L_ego,
              ISG15_ego,APOD_ego,P_ego)

ego_all=subset(ego_all,Description%in%c(S0_des,S1_des,BMP7_des,Cycling_des,REV3L_des,
                                        ISG15_des,APOD_des,P_des))



ego_all$cluster=factor(ego_all$cluster,levels = c("S0","S1","BMP7","Cycling","REV3L","ISG15","APOD","P"))


ggplot(data=ego_all)+aes(x=cluster,y=Description,color=-log10(pvalue))+geom_point(aes(size=Count,color= -log10(pvalue)))+
  scale_y_discrete(limits=unique(ego_all$Description))+theme_bw()+xlab("Cluster")+
  scale_color_gradient(low="blue",high = "red")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+xlab("")+
  ylab("")













#增大tsne的cluster分群查看异质性-----
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")
STR=RunTSNE(STR,reduction="harmony",dims=1:30)
STR=FindNeighbors(STR,dims=1:30,reduction = "harmony")
STR=FindClusters(STR, resolution = 4,reduction = "harmony")

STR=FindClusters(STR, resolution = 3,reduction = "harmony")

DimPlot(STR,reduction = "tsne",label = T,split.by = "type1")


#不同细胞亚型之间相关性的绘图-----
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")



#3组间
STR$type4=factor(STR$type4,levels = c("Ctrl","Adjacent","Adhesion"))

STR$cell_type0305_1=paste(STR$type2,"_",STR$cell_type0305,sep = "")

STR$cell_type0305_1=factor(STR$cell_type0305_1,levels = c(paste("CTRL_",names(table(STR$cell_type0305)),sep = ""),
                                                          paste("IUA1_",names(table(STR$cell_type0305)),sep = ""),
                                                          paste("IUA2_",names(table(STR$cell_type0305)),sep = "")))

av=AverageExpression(STR,
                     group.by = "cell_type0305_1",
                     assays = "RNA")

av=av[[1]]


cg=names(tail(sort(apply(av,1,sd)),1000))


View(av[cg,])

N=cor(av[cg,],method = 'spearman')

N=as.data.frame(N)

N=N[1:8,]
N=N[,-c(1:8)]

N$CTRL=1
N$IUA1=1
N$IUA2=1

N$IUA1[1]=N[1,1]
N$IUA1[2]=N[2,2]
N$IUA1[3]=N[3,3]
N$IUA1[4]=N[4,4]
N$IUA1[5]=N[5,5]
N$IUA1[6]=N[6,6]
N$IUA1[7]=N[7,7]
N$IUA1[8]=N[8,8]

N$IUA2[1]=N[1,9]
N$IUA2[2]=N[2,10]
N$IUA2[3]=N[3,11]
N$IUA2[4]=N[4,12]
N$IUA2[5]=N[5,13]
N$IUA2[6]=N[6,14]
N$IUA2[7]=N[7,15]
N$IUA2[8]=N[8,16]

N=N[17:19]
pheatmap(N,cluster_cols = F, display_numbers = TRUE,cluster_rows = F)

colnames(N)[2]="Adjacent"
colnames(N)[3]="Adhesion"

rownames(N)=gsub("CTRL_","",rownames(N))
par(mar=c(3,9,2,3))

jpeg(file ="/data/DATAprocess/chentw/zigongneimo/temp/P1.jpeg",
     width =1000*6,height = 1000*6,units = "px",res =72*15)
pheatmap(N,cluster_cols = F, display_numbers = TRUE,cluster_rows = F,
         fontsize = 20)
dev.off()




#2组间
STR$type4=factor(STR$type4,levels = c("Ctrl","Adjacent","Adhesion"))

STR$cell_type0305_2=paste(STR$type1,"_",STR$cell_type0305,sep = "")


STR$cell_type0305_2=factor(STR$cell_type0305_2,levels = c(paste("CTRL_",names(table(STR$cell_type0305)),sep = ""),
                                                          paste("IUA_",names(table(STR$cell_type0305)),sep = "")))

av=AverageExpression(STR,
                     group.by = "cell_type0305_2",
                     assays = "RNA")

av=av[[1]]


cg=names(tail(sort(apply(av,1,sd)),1000))


View(av[cg,])

N=cor(av[cg,],method = 'spearman')

N=as.data.frame(N)

N=N[1:8,]
N=N[,-c(1:8)]

N$CTRL=1
N$IUA=1


N$IUA[1]=N[1,1]
N$IUA[2]=N[2,2]
N$IUA[3]=N[3,3]
N$IUA[4]=N[4,4]
N$IUA[5]=N[5,5]
N$IUA[6]=N[6,6]
N$IUA[7]=N[7,7]
N$IUA[8]=N[8,8]



N=N[9:10]

pheatmap(N,cluster_cols = F, display_numbers = TRUE,cluster_rows = F)

rownames(N)=gsub("CTRL_","",rownames(N))


pheatmap(N,cluster_cols = F, display_numbers = TRUE,cluster_rows = F,
         fontsize = 25)


###进行ctrl和ADJACENT和Adhesion之间部分基因的比较
gene=c("ACTA2","COL")

VlnPlot(STR,group.by = "type4",features = gene,pt.size = 0)






#不同亚型之间的比例统计以及绘图-----
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")


table_plot=as.data.frame(table(STR$type4,STR$cell_type0305))


table_plot_Ctrl=subset(table_plot,table_plot$Var1%in%"Ctrl")
table_plot_Ctrl$ratio=table_plot_Ctrl$Freq/sum(table_plot_Ctrl$Freq)*100

table_plot_Adjacent=subset(table_plot,table_plot$Var1%in%"Adjacent")
table_plot_Adjacent$ratio=table_plot_Adjacent$Freq/sum(table_plot_Adjacent$Freq)*100

table_plot_Adhesion=subset(table_plot,table_plot$Var1%in%"Adhesion")
table_plot_Adhesion$ratio=table_plot_Adhesion$Freq/sum(table_plot_Adhesion$Freq)*100

table_plot=rbind(table_plot_Ctrl,table_plot_Adjacent,table_plot_Adhesion)

ggplot(data=table_plot,fill=Var2)+aes(x=Var2,y=stat(table_plot$ratio),fill=Var1)+geom_bar(position="dodge", stat="identity")+
  theme_bw()+theme(legend.title=element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank())+ylab("Integrative Celltype(%)")+scale_fill_brewer(palette="Paired")+
  xlab("")+scale_y_continuous(expand=c(0,0),limits=c(0,45))

ggplot(data, aes(fill=condition, y=value, x=specie)) + 
  geom_bar(position="dodge", stat="identity")






#对细胞亚型内的adhesion和CTRL进行差异基因查找富集以及绘图-----
library(Seurat)
library(ggplot2)
library(clusterProfiler)
library(dplyr)
library(org.Hs.eg.db)


STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

DimPlot(STR,group.by = "cell_type0305",label = T,label.size = 12)


mtGenes=grep('^MT-',rownames(STR),value=TRUE)
RPGenes=c(grep('^RPL',rownames(STR),value=TRUE),grep('^RPS',rownames(STR),value=TRUE))
stress_gene2=grep('^HSP',rownames(STR),value=TRUE)
HB_gene2=grep('^HB',rownames(STR),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
stress_gene1=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
               "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/gene1.csv")
subgene3=subgene3$cellcyle_gene
subgene4=c("FOS","CXCL2","ZFP36","FOSB","DUSP1","ATF3","CXCL8","NR4A1","CXCL3","PPP1R15A",
           "JUNB","EGR1","HSPA1A","HSPA1B","SOCS3","KLF6","JUN","IER2","CXCL1","NFKBIA",
           "HSPA6","DNAJB1","IER3","CCNL1","MTRNR2L2","IER4","ID1","CEBPD","KRT6A","CYR61",
           "DEPP1","CLDN4","IRF1","DUSP2","BTG2","PLAUR","MAFF","KLF4","PHLDA2","TNFAIP3")
subgene5=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/512个应激反应相关基因.csv")
subgene5=subgene5$gene_symbol

subgene=c(mtGenes,RPGenes,stress_gene2,HB_gene2,HB.genes,s.genes,g2m.genes,stress_gene1,"MKI67",subgene3,subgene4,subgene5)


####S0-----
STR$cell_type0305_2="Other"
STR$cell_type0305_2[STR$cell_type0305%in%"S0"]<-"S0"

STR$type4="Adhesion"
STR$type4[STR$type2%in%"CTRL"]="Ctrl"
STR$type4[STR$type2%in%"IUA1"]="Adjacent"



STR$cell_type0305_2=factor(STR$cell_type0305_2,levels = c("S0","Other"))

DimPlot(STR,group.by = "cell_type0305_2",label = T,label.size = 12)+
  scale_color_manual(values = c("#DB0C15","grey"))+
  ggtitle("")

STR$cell_type0305_2=paste(STR$type4,"_",STR$cell_type0305_2,sep = "")



S0_deseq=FindMarkers(STR,ident.1 = "Adhesion_S0",ident.2 = "Ctrl_S0",
                     min.pct = 0.2,group.by = "cell_type0305_2")

S0_deseq=
  subset(S0_deseq,S0_deseq$p_val<0.05)

write.csv(S0_deseq,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/S0/S0_deseq.csv")

S0_up=subset(S0_deseq,S0_deseq$avg_log2FC>0)

S0_up=rownames(S0_up)
G <- bitr(S0_up, fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
G=G$ENTREZID
ego <- enrichGO(OrgDb="org.Hs.eg.db", gene = G, ont = "ALL",readable= TRUE)
write.csv(ego,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/S0/S0_UP_ego.csv")



S0_down=subset(S0_deseq,S0_deseq$avg_log2FC<0)
S0_down=rownames(S0_down)
G <- bitr(S0_down, fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
G=G$ENTREZID
ego <- enrichGO(OrgDb="org.Hs.eg.db", gene = G, ont = "ALL",readable= TRUE)
write.csv(ego,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/S0/S0_DOWN_ego.csv")


#热图绘制
S0=subset(STR,type2%in%c("CTRL","IUA2"))
S0=subset(S0,cell_type0305%in%"S0")

DefaultAssay(S0) <- "RNA"
all.genes.S0 <- rownames(S0)
S0 <- ScaleData(object = S0, features = all.genes.S0)

S0_deseq=S0_deseq[order(S0_deseq$avg_log2FC),]

S0_deseq$gene=rownames(S0_deseq)

S0_deseq=subset(S0_deseq,!gene%in%subgene)


big50=c(rownames(S0_deseq[1:25,]),rownames(S0_deseq[as.numeric(length(rownames(S0_deseq))-24):length(rownames(S0_deseq)),]))


S0$type4=factor(S0$type4,levels = c("Ctrl","Adhesion"))

DoHeatmap(S0,features = big50,group.by = "type4",label = F)


#将UP和DOWN的ego画在一起前十五[特有]
ego_up=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/S0/S0_UP_ego.csv")

ego_down=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/S0/S0_DOWN_ego.csv")

ego_down=ego_down[order(ego_down$pvalue),]
ego_up=ego_up[order(ego_up$pvalue),]

ego_up$cluster="UP"
ego_down$cluster="DOWN"

ego_all=rbind(ego_up,ego_down)

Z=ego_all[duplicated(ego_all$Description),]

N=subset(ego_all,!Description%in%unique(Z$Description))

table(duplicated(N$Description))

ego_up=subset(N,cluster%in%"UP")
ego_down=subset(N,cluster%in%"DOWN")






UP_des=ego_up[1:15,]$Description
DOWN_des=ego_down[1:15,]$Description


ego_des=c(UP_des,DOWN_des)

ego_up=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/S0/S0_UP_ego.csv")
ego_down=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/S0/S0_DOWN_ego.csv")
ego_up$cluster="UP"
ego_down$cluster="DOWN"

ego_all=rbind(ego_down,ego_up)



ego_all=subset(ego_all,ego_all$Description%in%ego_des)


ggplot(data=ego_all)+aes(x=cluster,y=Description,color=-log10(pvalue))+geom_point(aes(size=Count,color= -log10(pvalue)))+
  scale_y_discrete(limits=unique(ego_all$Description))+theme_bw()+xlab("")+ylab("")+
  scale_color_gradient(low="blue",high = "red")+theme(axis.text = element_text(size=15))+
  guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))


####S1-----
STR$cell_type0305_2="Other"
STR$cell_type0305_2[STR$cell_type0305%in%"S1"]<-"S1"

STR$type4="Adhesion"
STR$type4[STR$type2%in%"CTRL"]="Ctrl"
STR$type4[STR$type2%in%"IUA1"]="Adjacent"



STR$cell_type0305_2=factor(STR$cell_type0305_2,levels = c("S1","Other"))

DimPlot(STR,group.by = "cell_type0305_2",label = T,label.size = 12,pt.size = 0.8)+
  scale_color_manual(values = c("#4CA636","grey"))+
  ggtitle("")

STR$cell_type0305_2=paste(STR$type4,"_",STR$cell_type0305_2,sep = "")



S1_deseq=FindMarkers(STR,ident.1 = "Adhesion_S1",ident.2 = "Ctrl_S1",
                     min.pct = 0.2,group.by = "cell_type0305_2")

S1_deseq=
  subset(S1_deseq,S1_deseq$p_val<0.05)

write.csv(S1_deseq,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/S1/S1_deseq.csv")

S1_up=subset(S1_deseq,S1_deseq$avg_log2FC>0)

S1_up=rownames(S1_up)
G <- bitr(S1_up, fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
G=G$ENTREZID
ego <- enrichGO(OrgDb="org.Hs.eg.db", gene = G, ont = "ALL",readable= TRUE)
write.csv(ego,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/S1/S1_UP_ego.csv")



S1_down=subset(S1_deseq,S1_deseq$avg_log2FC<0)
S1_down=rownames(S1_down)
G <- bitr(S1_down, fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
G=G$ENTREZID
ego <- enrichGO(OrgDb="org.Hs.eg.db", gene = G, ont = "ALL",readable= TRUE)
write.csv(ego,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/S1/S1_DOWN_ego.csv")


#热图绘制
S1=subset(STR,type2%in%c("CTRL","IUA2"))
S1=subset(S1,cell_type0305%in%"S1")

DefaultAssay(S1) <- "RNA"
all.genes.S1 <- rownames(S1)
S1 <- ScaleData(object = S1, features = all.genes.S1)

S1_deseq=S1_deseq[order(S1_deseq$avg_log2FC),]

S1_deseq$gene=rownames(S1_deseq)

S1_deseq=subset(S1_deseq,!gene%in%subgene)


big50=c(rownames(S1_deseq[1:25,]),rownames(S1_deseq[as.numeric(length(rownames(S1_deseq))-24):length(rownames(S1_deseq)),]))


S1$type4=factor(S1$type4,levels = c("Ctrl","Adhesion"))

DoHeatmap(S1,features = big50,group.by = "type4",label = F)


#将UP和DOWN的ego画在一起前十五[特有]
ego_up=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/S1/S1_UP_ego.csv")

ego_down=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/S1/S1_DOWN_ego.csv")

ego_down=ego_down[order(ego_down$pvalue),]
ego_up=ego_up[order(ego_up$pvalue),]

ego_up$cluster="UP"
ego_down$cluster="DOWN"

ego_all=rbind(ego_up,ego_down)

Z=ego_all[duplicated(ego_all$Description),]

N=subset(ego_all,!Description%in%unique(Z$Description))

table(duplicated(N$Description))

ego_up=subset(N,cluster%in%"UP")
ego_down=subset(N,cluster%in%"DOWN")






UP_des=ego_up[1:15,]$Description
DOWN_des=ego_down[1:15,]$Description


ego_des=c(UP_des,DOWN_des)

ego_up=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/S1/S1_UP_ego.csv")
ego_down=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/S1/S1_DOWN_ego.csv")
ego_up$cluster="UP"
ego_down$cluster="DOWN"

ego_all=rbind(ego_down,ego_up)



ego_all=subset(ego_all,ego_all$Description%in%ego_des)


ggplot(data=ego_all)+aes(x=cluster,y=Description,color=-log10(pvalue))+geom_point(aes(size=Count,color= -log10(pvalue)))+
  scale_y_discrete(limits=unique(ego_all$Description))+theme_bw()+xlab("")+ylab("")+
  scale_color_gradient(low="blue",high = "red")+theme(axis.text = element_text(size=15))+
  guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))


####BMP7-----
STR$cell_type0305_2="Other"
STR$cell_type0305_2[STR$cell_type0305%in%"BMP7"]<-"BMP7"

STR$type4="Adhesion"
STR$type4[STR$type2%in%"CTRL"]="Ctrl"
STR$type4[STR$type2%in%"IUA1"]="Adjacent"



STR$cell_type0305_2=factor(STR$cell_type0305_2,levels = c("BMP7","Other"))

DimPlot(STR,group.by = "cell_type0305_2",label = T,label.size = 12,pt.size = 0.8)+
  scale_color_manual(values = c("#9E4894","grey"))+
  ggtitle("")

STR$cell_type0305_2=paste(STR$type4,"_",STR$cell_type0305_2,sep = "")



BMP7_deseq=FindMarkers(STR,ident.1 = "Adhesion_BMP7",ident.2 = "Ctrl_BMP7",
                       min.pct = 0.2,group.by = "cell_type0305_2")

BMP7_deseq=
  subset(BMP7_deseq,BMP7_deseq$p_val<0.05)

write.csv(BMP7_deseq,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/BMP7/BMP7_deseq.csv")

BMP7_up=subset(BMP7_deseq,BMP7_deseq$avg_log2FC>0)

BMP7_up=rownames(BMP7_up)
G <- bitr(BMP7_up, fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
G=G$ENTREZID
ego <- enrichGO(OrgDb="org.Hs.eg.db", gene = G, ont = "ALL",readable= TRUE)
write.csv(ego,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/BMP7/BMP7_UP_ego.csv")



BMP7_down=subset(BMP7_deseq,BMP7_deseq$avg_log2FC<0)
BMP7_down=rownames(BMP7_down)
G <- bitr(BMP7_down, fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
G=G$ENTREZID
ego <- enrichGO(OrgDb="org.Hs.eg.db", gene = G, ont = "ALL",readable= TRUE)
write.csv(ego,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/BMP7/BMP7_DOWN_ego.csv")


#热图绘制
BMP7=subset(STR,type2%in%c("CTRL","IUA2"))
BMP7=subset(BMP7,cell_type0305%in%"BMP7")

DefaultAssay(BMP7) <- "RNA"
all.genes.BMP7 <- rownames(BMP7)
BMP7 <- ScaleData(object = BMP7, features = all.genes.BMP7)

BMP7_deseq=BMP7_deseq[order(BMP7_deseq$avg_log2FC),]

BMP7_deseq$gene=rownames(BMP7_deseq)

BMP7_deseq=subset(BMP7_deseq,!gene%in%subgene)


big50=c(rownames(BMP7_deseq[1:25,]),rownames(BMP7_deseq[as.numeric(length(rownames(BMP7_deseq))-24):length(rownames(BMP7_deseq)),]))


BMP7$type4=factor(BMP7$type4,levels = c("Ctrl","Adhesion"))

DoHeatmap(BMP7,features = big50,group.by = "type4",label = F)


#将UP和DOWN的ego画在一起前十五[特有]
ego_up=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/BMP7/BMP7_UP_ego.csv")

ego_down=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/BMP7/BMP7_DOWN_ego.csv")

ego_down=ego_down[order(ego_down$pvalue),]
ego_up=ego_up[order(ego_up$pvalue),]

ego_up$cluster="UP"
ego_down$cluster="DOWN"

ego_all=rbind(ego_up,ego_down)

Z=ego_all[duplicated(ego_all$Description),]

N=subset(ego_all,!Description%in%unique(Z$Description))

table(duplicated(N$Description))

ego_up=subset(N,cluster%in%"UP")
ego_down=subset(N,cluster%in%"DOWN")






UP_des=ego_up[1:15,]$Description
DOWN_des=ego_down[1:15,]$Description


ego_des=c(UP_des,DOWN_des)

ego_up=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/BMP7/BMP7_UP_ego.csv")
ego_down=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/BMP7/BMP7_DOWN_ego.csv")
ego_up$cluster="UP"
ego_down$cluster="DOWN"

ego_all=rbind(ego_down,ego_up)



ego_all=subset(ego_all,ego_all$Description%in%ego_des)


ggplot(data=ego_all)+aes(x=cluster,y=Description,color=-log10(pvalue))+geom_point(aes(size=Count,color= -log10(pvalue)))+
  scale_y_discrete(limits=unique(ego_all$Description))+theme_bw()+xlab("")+ylab("")+
  scale_color_gradient(low="blue",high = "red")+theme(axis.text = element_text(size=15))+
  guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))


####Cycling-----
STR$cell_type0305_2="Other"
STR$cell_type0305_2[STR$cell_type0305%in%"Cycling"]<-"Cycling"

STR$type4="Adhesion"
STR$type4[STR$type2%in%"CTRL"]="Ctrl"
STR$type4[STR$type2%in%"IUA1"]="Adjacent"



STR$cell_type0305_2=factor(STR$cell_type0305_2,levels = c("Cycling","Other"))

DimPlot(STR,group.by = "cell_type0305_2",label = T,label.size = 12,pt.size = 0.8)+
  scale_color_manual(values = c("#EE7A00","grey"))+
  ggtitle("")

STR$cell_type0305_2=paste(STR$type4,"_",STR$cell_type0305_2,sep = "")



Cycling_deseq=FindMarkers(STR,ident.1 = "Adhesion_Cycling",ident.2 = "Ctrl_Cycling",
                          min.pct = 0.2,group.by = "cell_type0305_2")

Cycling_deseq=
  subset(Cycling_deseq,Cycling_deseq$p_val<0.05)

write.csv(Cycling_deseq,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/Cycling/Cycling_deseq.csv")

Cycling_up=subset(Cycling_deseq,Cycling_deseq$avg_log2FC>0)

Cycling_up=rownames(Cycling_up)
G <- bitr(Cycling_up, fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
G=G$ENTREZID
ego <- enrichGO(OrgDb="org.Hs.eg.db", gene = G, ont = "ALL",readable= TRUE)
write.csv(ego,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/Cycling/Cycling_UP_ego.csv")



Cycling_down=subset(Cycling_deseq,Cycling_deseq$avg_log2FC<0)
Cycling_down=rownames(Cycling_down)
G <- bitr(Cycling_down, fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
G=G$ENTREZID
ego <- enrichGO(OrgDb="org.Hs.eg.db", gene = G, ont = "ALL",readable= TRUE)
write.csv(ego,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/Cycling/Cycling_DOWN_ego.csv")


#热图绘制
Cycling=subset(STR,type2%in%c("CTRL","IUA2"))
Cycling=subset(Cycling,cell_type0305%in%"Cycling")

DefaultAssay(Cycling) <- "RNA"
all.genes.Cycling <- rownames(Cycling)
Cycling <- ScaleData(object = Cycling, features = all.genes.Cycling)

Cycling_deseq=Cycling_deseq[order(Cycling_deseq$avg_log2FC),]

Cycling_deseq$gene=rownames(Cycling_deseq)

Cycling_deseq=subset(Cycling_deseq,!gene%in%subgene)


big50=c(rownames(Cycling_deseq[1:25,]),rownames(Cycling_deseq[as.numeric(length(rownames(Cycling_deseq))-24):length(rownames(Cycling_deseq)),]))


Cycling$type4=factor(Cycling$type4,levels = c("Ctrl","Adhesion"))

DoHeatmap(Cycling,features = big50,group.by = "type4",label = F)


#将UP和DOWN的ego画在一起前十五[特有]
ego_up=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/Cycling/Cycling_UP_ego.csv")

ego_down=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/Cycling/Cycling_DOWN_ego.csv")

ego_down=ego_down[order(ego_down$pvalue),]
ego_up=ego_up[order(ego_up$pvalue),]

ego_up$cluster="UP"
ego_down$cluster="DOWN"

ego_all=rbind(ego_up,ego_down)

Z=ego_all[duplicated(ego_all$Description),]

N=subset(ego_all,!Description%in%unique(Z$Description))

table(duplicated(N$Description))

ego_up=subset(N,cluster%in%"UP")
ego_down=subset(N,cluster%in%"DOWN")






UP_des=ego_up[1:15,]$Description
DOWN_des=ego_down[1:15,]$Description


ego_des=c(UP_des,DOWN_des)

ego_up=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/Cycling/Cycling_UP_ego.csv")
ego_down=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/Cycling/Cycling_DOWN_ego.csv")
ego_up$cluster="UP"
ego_down$cluster="DOWN"

ego_all=rbind(ego_down,ego_up)



ego_all=subset(ego_all,ego_all$Description%in%ego_des)


ggplot(data=ego_all)+aes(x=cluster,y=Description,color=-log10(pvalue))+geom_point(aes(size=Count,color= -log10(pvalue)))+
  scale_y_discrete(limits=unique(ego_all$Description))+theme_bw()+xlab("")+ylab("")+
  scale_color_gradient(low="blue",high = "red")+theme(axis.text = element_text(size=15))+
  guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))


####REV3L-----
STR$cell_type0305_2="Other"
STR$cell_type0305_2[STR$cell_type0305%in%"REV3L"]<-"REV3L"

STR$type4="Adhesion"
STR$type4[STR$type2%in%"CTRL"]="Ctrl"
STR$type4[STR$type2%in%"IUA1"]="Adjacent"



STR$cell_type0305_2=factor(STR$cell_type0305_2,levels = c("REV3L","Other"))

DimPlot(STR,group.by = "cell_type0305_2",label = T,label.size = 12,pt.size = 0.8)+
  scale_color_manual(values = c("#387DB9","grey"))+
  ggtitle("")

STR$cell_type0305_2=paste(STR$type4,"_",STR$cell_type0305_2,sep = "")



REV3L_deseq=FindMarkers(STR,ident.1 = "Adhesion_REV3L",ident.2 = "Ctrl_REV3L",
                        min.pct = 0.2,group.by = "cell_type0305_2")

REV3L_deseq=
  subset(REV3L_deseq,REV3L_deseq$p_val<0.05)

write.csv(REV3L_deseq,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/REV3L/REV3L_deseq.csv")

REV3L_up=subset(REV3L_deseq,REV3L_deseq$avg_log2FC>0)

REV3L_up=rownames(REV3L_up)
G <- bitr(REV3L_up, fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
G=G$ENTREZID
ego <- enrichGO(OrgDb="org.Hs.eg.db", gene = G, ont = "ALL",readable= TRUE)
write.csv(ego,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/REV3L/REV3L_UP_ego.csv")



REV3L_down=subset(REV3L_deseq,REV3L_deseq$avg_log2FC<0)
REV3L_down=rownames(REV3L_down)
G <- bitr(REV3L_down, fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
G=G$ENTREZID
ego <- enrichGO(OrgDb="org.Hs.eg.db", gene = G, ont = "ALL",readable= TRUE)
write.csv(ego,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/REV3L/REV3L_DOWN_ego.csv")


#热图绘制
REV3L=subset(STR,type2%in%c("CTRL","IUA2"))
REV3L=subset(REV3L,cell_type0305%in%"REV3L")

DefaultAssay(REV3L) <- "RNA"
all.genes.REV3L <- rownames(REV3L)
REV3L <- ScaleData(object = REV3L, features = all.genes.REV3L)

REV3L_deseq=REV3L_deseq[order(REV3L_deseq$avg_log2FC),]

REV3L_deseq$gene=rownames(REV3L_deseq)

REV3L_deseq=subset(REV3L_deseq,!gene%in%subgene)


big50=c(rownames(REV3L_deseq[1:25,]),rownames(REV3L_deseq[as.numeric(length(rownames(REV3L_deseq))-24):length(rownames(REV3L_deseq)),]))


REV3L$type4=factor(REV3L$type4,levels = c("Ctrl","Adhesion"))

DoHeatmap(REV3L,features = big50,group.by = "type4",label = F)


#将UP和DOWN的ego画在一起前十五[特有]
ego_up=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/REV3L/REV3L_UP_ego.csv")

ego_down=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/REV3L/REV3L_DOWN_ego.csv")

ego_down=ego_down[order(ego_down$pvalue),]
ego_up=ego_up[order(ego_up$pvalue),]

ego_up$cluster="UP"
ego_down$cluster="DOWN"

ego_all=rbind(ego_up,ego_down)

Z=ego_all[duplicated(ego_all$Description),]

N=subset(ego_all,!Description%in%unique(Z$Description))

table(duplicated(N$Description))

ego_up=subset(N,cluster%in%"UP")
ego_down=subset(N,cluster%in%"DOWN")






UP_des=ego_up[1:15,]$Description
DOWN_des=ego_down[1:15,]$Description


ego_des=c(UP_des,DOWN_des)

ego_up=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/REV3L/REV3L_UP_ego.csv")
ego_down=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/REV3L/REV3L_DOWN_ego.csv")
ego_up$cluster="UP"
ego_down$cluster="DOWN"

ego_all=rbind(ego_down,ego_up)



ego_all=subset(ego_all,ego_all$Description%in%ego_des)


ggplot(data=ego_all)+aes(x=cluster,y=Description,color=-log10(pvalue))+geom_point(aes(size=Count,color= -log10(pvalue)))+
  scale_y_discrete(limits=unique(ego_all$Description))+theme_bw()+xlab("")+ylab("")+
  scale_color_gradient(low="blue",high = "red")+theme(axis.text = element_text(size=15))+
  guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))


####ISG15-----
STR$cell_type0305_2="Other"
STR$cell_type0305_2[STR$cell_type0305%in%"ISG15"]<-"ISG15"

STR$type4="Adhesion"
STR$type4[STR$type2%in%"CTRL"]="Ctrl"
STR$type4[STR$type2%in%"IUA1"]="Adjacent"



STR$cell_type0305_2=factor(STR$cell_type0305_2,levels = c("ISG15","Other"))

DimPlot(STR,group.by = "cell_type0305_2",label = T,label.size = 12,pt.size = 0.8)+
  scale_color_manual(values = c("#482013","grey"))+
  ggtitle("")

STR$cell_type0305_2=paste(STR$type4,"_",STR$cell_type0305_2,sep = "")



ISG15_deseq=FindMarkers(STR,ident.1 = "Adhesion_ISG15",ident.2 = "Ctrl_ISG15",
                        min.pct = 0.2,group.by = "cell_type0305_2")

ISG15_deseq=
  subset(ISG15_deseq,ISG15_deseq$p_val<0.05)

write.csv(ISG15_deseq,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/ISG15/ISG15_deseq.csv")

ISG15_up=subset(ISG15_deseq,ISG15_deseq$avg_log2FC>0)

ISG15_up=rownames(ISG15_up)
G <- bitr(ISG15_up, fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
G=G$ENTREZID
ego <- enrichGO(OrgDb="org.Hs.eg.db", gene = G, ont = "ALL",readable= TRUE)
write.csv(ego,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/ISG15/ISG15_UP_ego.csv")



ISG15_down=subset(ISG15_deseq,ISG15_deseq$avg_log2FC<0)
ISG15_down=rownames(ISG15_down)
G <- bitr(ISG15_down, fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
G=G$ENTREZID
ego <- enrichGO(OrgDb="org.Hs.eg.db", gene = G, ont = "ALL",readable= TRUE)
write.csv(ego,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/ISG15/ISG15_DOWN_ego.csv")


#热图绘制
ISG15=subset(STR,type2%in%c("CTRL","IUA2"))
ISG15=subset(ISG15,cell_type0305%in%"ISG15")

DefaultAssay(ISG15) <- "RNA"
all.genes.ISG15 <- rownames(ISG15)
ISG15 <- ScaleData(object = ISG15, features = all.genes.ISG15)

ISG15_deseq=ISG15_deseq[order(ISG15_deseq$avg_log2FC),]

ISG15_deseq$gene=rownames(ISG15_deseq)

ISG15_deseq=subset(ISG15_deseq,!gene%in%subgene)


big50=c(rownames(ISG15_deseq[1:25,]),rownames(ISG15_deseq[as.numeric(length(rownames(ISG15_deseq))-24):length(rownames(ISG15_deseq)),]))


ISG15$type4=factor(ISG15$type4,levels = c("Ctrl","Adhesion"))

DoHeatmap(ISG15,features = big50,group.by = "type4",label = F)


#将UP和DOWN的ego画在一起前十五[特有]
ego_up=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/ISG15/ISG15_UP_ego.csv")

ego_down=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/ISG15/ISG15_DOWN_ego.csv")

ego_down=ego_down[order(ego_down$pvalue),]
ego_up=ego_up[order(ego_up$pvalue),]

ego_up$cluster="UP"
ego_down$cluster="DOWN"

ego_all=rbind(ego_up,ego_down)

Z=ego_all[duplicated(ego_all$Description),]

N=subset(ego_all,!Description%in%unique(Z$Description))

table(duplicated(N$Description))

ego_up=subset(N,cluster%in%"UP")
ego_down=subset(N,cluster%in%"DOWN")






UP_des=ego_up[1:15,]$Description
DOWN_des=ego_down[1:15,]$Description


ego_des=c(UP_des,DOWN_des)

ego_up=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/ISG15/ISG15_UP_ego.csv")
ego_down=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/ISG15/ISG15_DOWN_ego.csv")
ego_up$cluster="UP"
ego_down$cluster="DOWN"

ego_all=rbind(ego_down,ego_up)



ego_all=subset(ego_all,ego_all$Description%in%ego_des)


ggplot(data=ego_all)+aes(x=cluster,y=Description,color=-log10(pvalue))+geom_point(aes(size=Count,color= -log10(pvalue)))+
  scale_y_discrete(limits=unique(ego_all$Description))+theme_bw()+xlab("")+ylab("")+
  scale_color_gradient(low="blue",high = "red")+theme(axis.text = element_text(size=15))+
  guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))


####APOD-----
STR$cell_type0305_2="Other"
STR$cell_type0305_2[STR$cell_type0305%in%"APOD"]<-"APOD"

STR$type4="Adhesion"
STR$type4[STR$type2%in%"CTRL"]="Ctrl"
STR$type4[STR$type2%in%"IUA1"]="Adjacent"



STR$cell_type0305_2=factor(STR$cell_type0305_2,levels = c("APOD","Other"))

DimPlot(STR,group.by = "cell_type0305_2",label = T,label.size = 12,pt.size = 0.8)+
  scale_color_manual(values = c("#F9E800","grey"))+
  ggtitle("")

STR$cell_type0305_2=paste(STR$type4,"_",STR$cell_type0305_2,sep = "")



APOD_deseq=FindMarkers(STR,ident.1 = "Adhesion_APOD",ident.2 = "Ctrl_APOD",
                       min.pct = 0.2,group.by = "cell_type0305_2")

APOD_deseq=
  subset(APOD_deseq,APOD_deseq$p_val<0.05)

write.csv(APOD_deseq,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/APOD/APOD_deseq.csv")

APOD_up=subset(APOD_deseq,APOD_deseq$avg_log2FC>0)

APOD_up=rownames(APOD_up)
G <- bitr(APOD_up, fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
G=G$ENTREZID
ego <- enrichGO(OrgDb="org.Hs.eg.db", gene = G, ont = "ALL",readable= TRUE)
write.csv(ego,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/APOD/APOD_UP_ego.csv")



APOD_down=subset(APOD_deseq,APOD_deseq$avg_log2FC<0)
APOD_down=rownames(APOD_down)
G <- bitr(APOD_down, fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
G=G$ENTREZID
ego <- enrichGO(OrgDb="org.Hs.eg.db", gene = G, ont = "ALL",readable= TRUE)
write.csv(ego,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/APOD/APOD_DOWN_ego.csv")


#热图绘制
APOD=subset(STR,type2%in%c("CTRL","IUA2"))
APOD=subset(APOD,cell_type0305%in%"APOD")

DefaultAssay(APOD) <- "RNA"
all.genes.APOD <- rownames(APOD)
APOD <- ScaleData(object = APOD, features = all.genes.APOD)

APOD_deseq=APOD_deseq[order(APOD_deseq$avg_log2FC),]

APOD_deseq$gene=rownames(APOD_deseq)

APOD_deseq=subset(APOD_deseq,!gene%in%subgene)


big50=c(rownames(APOD_deseq[1:25,]),rownames(APOD_deseq[as.numeric(length(rownames(APOD_deseq))-24):length(rownames(APOD_deseq)),]))


APOD$type4=factor(APOD$type4,levels = c("Ctrl","Adhesion"))

DoHeatmap(APOD,features = big50,group.by = "type4",label = F)


#将UP和DOWN的ego画在一起前十五[特有]
ego_up=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/APOD/APOD_UP_ego.csv")

ego_down=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/APOD/APOD_DOWN_ego.csv")

ego_down=ego_down[order(ego_down$pvalue),]
ego_up=ego_up[order(ego_up$pvalue),]

ego_up$cluster="UP"
ego_down$cluster="DOWN"

ego_all=rbind(ego_up,ego_down)

Z=ego_all[duplicated(ego_all$Description),]

N=subset(ego_all,!Description%in%unique(Z$Description))

table(duplicated(N$Description))

ego_up=subset(N,cluster%in%"UP")
ego_down=subset(N,cluster%in%"DOWN")






UP_des=ego_up[1:15,]$Description
DOWN_des=ego_down[1:15,]$Description


ego_des=c(UP_des,DOWN_des)

ego_up=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/APOD/APOD_UP_ego.csv")
ego_down=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/APOD/APOD_DOWN_ego.csv")
ego_up$cluster="UP"
ego_down$cluster="DOWN"

ego_all=rbind(ego_down,ego_up)



ego_all=subset(ego_all,ego_all$Description%in%ego_des)


ggplot(data=ego_all)+aes(x=cluster,y=Description,color=-log10(pvalue))+geom_point(aes(size=Count,color= -log10(pvalue)))+
  scale_y_discrete(limits=unique(ego_all$Description))+theme_bw()+xlab("")+ylab("")+
  scale_color_gradient(low="blue",high = "red")+theme(axis.text = element_text(size=15))+
  guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))


####P-----
STR$cell_type0305_2="Other"
STR$cell_type0305_2[STR$cell_type0305%in%"P"]<-"P"

STR$type4="Adhesion"
STR$type4[STR$type2%in%"CTRL"]="Ctrl"
STR$type4[STR$type2%in%"IUA1"]="Adjacent"



STR$cell_type0305_2=factor(STR$cell_type0305_2,levels = c("P","Other"))

DimPlot(STR,group.by = "cell_type0305_2",label = T,label.size = 12,pt.size = 0.8)+
  scale_color_manual(values = c("#5555A4","grey"))+
  ggtitle("")

STR$cell_type0305_2=paste(STR$type4,"_",STR$cell_type0305_2,sep = "")



P_deseq=FindMarkers(STR,ident.1 = "Adhesion_P",ident.2 = "Ctrl_P",
                    min.pct = 0.2,group.by = "cell_type0305_2")

P_deseq=
  subset(P_deseq,P_deseq$p_val<0.05)

write.csv(P_deseq,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/P/P_deseq.csv")

P_up=subset(P_deseq,P_deseq$avg_log2FC>0)

P_up=rownames(P_up)
G <- bitr(P_up, fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
G=G$ENTREZID
ego <- enrichGO(OrgDb="org.Hs.eg.db", gene = G, ont = "ALL",readable= TRUE)
write.csv(ego,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/P/P_UP_ego.csv")



P_down=subset(P_deseq,P_deseq$avg_log2FC<0)
P_down=rownames(P_down)
G <- bitr(P_down, fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
G=G$ENTREZID
ego <- enrichGO(OrgDb="org.Hs.eg.db", gene = G, ont = "ALL",readable= TRUE)
write.csv(ego,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/P/P_DOWN_ego.csv")


#热图绘制
P=subset(STR,type2%in%c("CTRL","IUA2"))
P=subset(P,cell_type0305%in%"P")

DefaultAssay(P) <- "RNA"
all.genes.P <- rownames(P)
P <- ScaleData(object = P, features = all.genes.P)

P_deseq=P_deseq[order(P_deseq$avg_log2FC),]

P_deseq$gene=rownames(P_deseq)

P_deseq=subset(P_deseq,!gene%in%subgene)


big50=c(rownames(P_deseq[1:25,]),rownames(P_deseq[as.numeric(length(rownames(P_deseq))-24):length(rownames(P_deseq)),]))


P$type4=factor(P$type4,levels = c("Ctrl","Adhesion"))

DoHeatmap(P,features = big50,group.by = "type4",label = F)


#将UP和DOWN的ego画在一起前十五[特有]
ego_up=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/P/P_UP_ego.csv")

ego_down=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/P/P_DOWN_ego.csv")

ego_down=ego_down[order(ego_down$pvalue),]
ego_up=ego_up[order(ego_up$pvalue),]

ego_up$cluster="UP"
ego_down$cluster="DOWN"

ego_all=rbind(ego_up,ego_down)

Z=ego_all[duplicated(ego_all$Description),]

N=subset(ego_all,!Description%in%unique(Z$Description))

table(duplicated(N$Description))

ego_up=subset(N,cluster%in%"UP")
ego_down=subset(N,cluster%in%"DOWN")






UP_des=ego_up[1:15,]$Description
DOWN_des=ego_down[1:15,]$Description


ego_des=c(UP_des,DOWN_des)

ego_up=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/P/P_UP_ego.csv")
ego_down=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/P/P_DOWN_ego.csv")
ego_up$cluster="UP"
ego_down$cluster="DOWN"

ego_all=rbind(ego_down,ego_up)



ego_all=subset(ego_all,ego_all$Description%in%ego_des)


ggplot(data=ego_all)+aes(x=cluster,y=Description,color=-log10(pvalue))+geom_point(aes(size=Count,color= -log10(pvalue)))+
  scale_y_discrete(limits=unique(ego_all$Description))+theme_bw()+xlab("")+ylab("")+
  scale_color_gradient(low="blue",high = "red")+theme(axis.text = element_text(size=15))+
  guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))


#对细胞亚型内的adhesion和CTRL进行差异基因查找富集以及绘图KOBAS结果-------
###S0-------
S0_down_ego=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/S0_down_ego.csv')

S0_down_ego=S0_down_ego[order(S0_down_ego$P.Value),]

S0_down_ego=S0_down_ego[1:30,]

S0_down_ego$Input.number=as.numeric(S0_down_ego$Input.number)
S0_down_ego$Background.number=as.numeric(S0_down_ego$Background.number)

S0_down_ego$richFactor<-S0_down_ego$Input.number/S0_down_ego$Background.number

S0_down_ego=S0_down_ego[order(S0_down_ego$P.Value),]

S0_down_ego$Cluster="DOWN"


S0_up_ego=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/S0_up_ego.csv')

S0_up_ego=S0_up_ego[order(S0_up_ego$P.Value),]

S0_up_ego=S0_up_ego[1:30,]

S0_up_ego$Input.number=as.numeric(S0_up_ego$Input.number)
S0_up_ego$Background.number=as.numeric(S0_up_ego$Background.number)

S0_up_ego$richFactor<-S0_up_ego$Input.number/S0_up_ego$Background.number

S0_up_ego=S0_up_ego[order(S0_up_ego$P.Value),]

S0_up_ego$Cluster="UP"


S0_ego=rbind(S0_down_ego,S0_up_ego)


ggplot(data=S0_ego)+aes(x=Cluster,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(S0_ego$X.Term))+theme_bw()+xlab("")+ylab("")+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.1,1))+
  guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))


S0_down_kegg=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/S0_down_kegg.csv')

S0_down_kegg=S0_down_kegg[order(S0_down_kegg$P.Value),]

S0_down_kegg=S0_down_kegg[1:30,]

S0_down_kegg$Input.number=as.numeric(S0_down_kegg$Input.number)
S0_down_kegg$Background.number=as.numeric(S0_down_kegg$Background.number)

S0_down_kegg$richFactor<-S0_down_kegg$Input.number/S0_down_kegg$Background.number

S0_down_kegg=S0_down_kegg[order(S0_down_kegg$P.Value),]

S0_down_kegg$Cluster="DOWN"


S0_up_kegg=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/S0_up_kegg.csv')

S0_up_kegg=S0_up_kegg[order(S0_up_kegg$P.Value),]

S0_up_kegg=S0_up_kegg[1:30,]

S0_up_kegg$Input.number=as.numeric(S0_up_kegg$Input.number)
S0_up_kegg$Background.number=as.numeric(S0_up_kegg$Background.number)

S0_up_kegg$richFactor<-S0_up_kegg$Input.number/S0_up_kegg$Background.number

S0_up_kegg=S0_up_kegg[order(S0_up_kegg$P.Value),]

S0_up_kegg$Cluster="UP"


S0_kegg=rbind(S0_down_kegg,S0_up_kegg)


ggplot(data=S0_kegg)+aes(x=Cluster,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(S0_kegg$X.Term))+theme_bw()+xlab("")+ylab("")+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.1,1))+
  guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))


###S1-------
S1_down_ego=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/S1_down_ego.csv')

S1_down_ego=S1_down_ego[order(S1_down_ego$P.Value),]

S1_down_ego=S1_down_ego[1:30,]

S1_down_ego$Input.number=as.numeric(S1_down_ego$Input.number)
S1_down_ego$Background.number=as.numeric(S1_down_ego$Background.number)

S1_down_ego$richFactor<-S1_down_ego$Input.number/S1_down_ego$Background.number

S1_down_ego=S1_down_ego[order(S1_down_ego$P.Value),]

S1_down_ego$Cluster="DOWN"


S1_up_ego=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/S1_up_ego.csv')

S1_up_ego=S1_up_ego[order(S1_up_ego$P.Value),]

S1_up_ego=S1_up_ego[1:30,]

S1_up_ego$Input.number=as.numeric(S1_up_ego$Input.number)
S1_up_ego$Background.number=as.numeric(S1_up_ego$Background.number)

S1_up_ego$richFactor<-S1_up_ego$Input.number/S1_up_ego$Background.number

S1_up_ego=S1_up_ego[order(S1_up_ego$P.Value),]

S1_up_ego$Cluster="UP"


S1_ego=rbind(S1_down_ego,S1_up_ego)


ggplot(data=S1_ego)+aes(x=Cluster,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(S1_ego$X.Term))+theme_bw()+xlab("")+ylab("")+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.1,1))+
  guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))


S1_down_kegg=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/S1_down_kegg.csv')

S1_down_kegg=S1_down_kegg[order(S1_down_kegg$P.Value),]

S1_down_kegg=S1_down_kegg[1:30,]

S1_down_kegg$Input.number=as.numeric(S1_down_kegg$Input.number)
S1_down_kegg$Background.number=as.numeric(S1_down_kegg$Background.number)

S1_down_kegg$richFactor<-S1_down_kegg$Input.number/S1_down_kegg$Background.number

S1_down_kegg=S1_down_kegg[order(S1_down_kegg$P.Value),]

S1_down_kegg$Cluster="DOWN"


S1_up_kegg=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/S1_up_kegg.csv')

S1_up_kegg=S1_up_kegg[order(S1_up_kegg$P.Value),]

S1_up_kegg=S1_up_kegg[1:30,]

S1_up_kegg$Input.number=as.numeric(S1_up_kegg$Input.number)
S1_up_kegg$Background.number=as.numeric(S1_up_kegg$Background.number)

S1_up_kegg$richFactor<-S1_up_kegg$Input.number/S1_up_kegg$Background.number

S1_up_kegg=S1_up_kegg[order(S1_up_kegg$P.Value),]

S1_up_kegg$Cluster="UP"


S1_kegg=rbind(S1_down_kegg,S1_up_kegg)


ggplot(data=S1_kegg)+aes(x=Cluster,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(S1_kegg$X.Term))+theme_bw()+xlab("")+ylab("")+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.1,1))+
  guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))


###BMP7-------
BMP7_down_ego=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/BMP7_down_ego.csv')

BMP7_down_ego=BMP7_down_ego[order(BMP7_down_ego$P.Value),]

BMP7_down_ego=BMP7_down_ego[1:30,]

BMP7_down_ego$Input.number=as.numeric(BMP7_down_ego$Input.number)
BMP7_down_ego$Background.number=as.numeric(BMP7_down_ego$Background.number)

BMP7_down_ego$richFactor<-BMP7_down_ego$Input.number/BMP7_down_ego$Background.number

BMP7_down_ego=BMP7_down_ego[order(BMP7_down_ego$P.Value),]

BMP7_down_ego$Cluster="DOWN"


BMP7_up_ego=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/BMP7_up_ego.csv')

BMP7_up_ego=BMP7_up_ego[order(BMP7_up_ego$P.Value),]

BMP7_up_ego=BMP7_up_ego[1:30,]

BMP7_up_ego$Input.number=as.numeric(BMP7_up_ego$Input.number)
BMP7_up_ego$Background.number=as.numeric(BMP7_up_ego$Background.number)

BMP7_up_ego$richFactor<-BMP7_up_ego$Input.number/BMP7_up_ego$Background.number

BMP7_up_ego=BMP7_up_ego[order(BMP7_up_ego$P.Value),]

BMP7_up_ego$Cluster="UP"


BMP7_ego=rbind(BMP7_down_ego,BMP7_up_ego)


ggplot(data=BMP7_ego)+aes(x=Cluster,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(BMP7_ego$X.Term))+theme_bw()+xlab("")+ylab("")+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.1,1))+
  guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))


BMP7_down_kegg=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/BMP7_down_kegg.csv')

BMP7_down_kegg=BMP7_down_kegg[order(BMP7_down_kegg$P.Value),]

BMP7_down_kegg=BMP7_down_kegg[1:30,]

BMP7_down_kegg$Input.number=as.numeric(BMP7_down_kegg$Input.number)
BMP7_down_kegg$Background.number=as.numeric(BMP7_down_kegg$Background.number)

BMP7_down_kegg$richFactor<-BMP7_down_kegg$Input.number/BMP7_down_kegg$Background.number

BMP7_down_kegg=BMP7_down_kegg[order(BMP7_down_kegg$P.Value),]

BMP7_down_kegg$Cluster="DOWN"


BMP7_up_kegg=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/BMP7_up_kegg.csv')

BMP7_up_kegg=BMP7_up_kegg[order(BMP7_up_kegg$P.Value),]

BMP7_up_kegg=BMP7_up_kegg[1:30,]

BMP7_up_kegg$Input.number=as.numeric(BMP7_up_kegg$Input.number)
BMP7_up_kegg$Background.number=as.numeric(BMP7_up_kegg$Background.number)

BMP7_up_kegg$richFactor<-BMP7_up_kegg$Input.number/BMP7_up_kegg$Background.number

BMP7_up_kegg=BMP7_up_kegg[order(BMP7_up_kegg$P.Value),]

BMP7_up_kegg$Cluster="UP"


BMP7_kegg=rbind(BMP7_down_kegg,BMP7_up_kegg)


ggplot(data=BMP7_kegg)+aes(x=Cluster,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(BMP7_kegg$X.Term))+theme_bw()+xlab("")+ylab("")+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.1,1))+
  guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))


###Cycling-------
Cycling_down_ego=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/Cycling_down_ego.csv')

Cycling_down_ego=Cycling_down_ego[order(Cycling_down_ego$P.Value),]

Cycling_down_ego=Cycling_down_ego[1:30,]

Cycling_down_ego$Input.number=as.numeric(Cycling_down_ego$Input.number)
Cycling_down_ego$Background.number=as.numeric(Cycling_down_ego$Background.number)

Cycling_down_ego$richFactor<-Cycling_down_ego$Input.number/Cycling_down_ego$Background.number

Cycling_down_ego=Cycling_down_ego[order(Cycling_down_ego$P.Value),]

Cycling_down_ego$Cluster="DOWN"


Cycling_up_ego=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/Cycling_up_ego.csv')

Cycling_up_ego=Cycling_up_ego[order(Cycling_up_ego$P.Value),]

Cycling_up_ego=Cycling_up_ego[1:30,]

Cycling_up_ego$Input.number=as.numeric(Cycling_up_ego$Input.number)
Cycling_up_ego$Background.number=as.numeric(Cycling_up_ego$Background.number)

Cycling_up_ego$richFactor<-Cycling_up_ego$Input.number/Cycling_up_ego$Background.number

Cycling_up_ego=Cycling_up_ego[order(Cycling_up_ego$P.Value),]

Cycling_up_ego$Cluster="UP"
colnames(Cycling_up_ego)=colnames(Cycling_down_ego)

Cycling_ego=rbind(Cycling_down_ego,Cycling_up_ego)


ggplot(data=Cycling_ego)+aes(x=Cluster,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(Cycling_ego$X.Term))+theme_bw()+xlab("")+ylab("")+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.1,1))+
  guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))


Cycling_down_kegg=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/Cycling_down_kegg.csv')

Cycling_down_kegg=Cycling_down_kegg[order(Cycling_down_kegg$P.Value),]

Cycling_down_kegg=Cycling_down_kegg[1:30,]

Cycling_down_kegg$Input.number=as.numeric(Cycling_down_kegg$Input.number)
Cycling_down_kegg$Background.number=as.numeric(Cycling_down_kegg$Background.number)

Cycling_down_kegg$richFactor<-Cycling_down_kegg$Input.number/Cycling_down_kegg$Background.number

Cycling_down_kegg=Cycling_down_kegg[order(Cycling_down_kegg$P.Value),]

Cycling_down_kegg$Cluster="DOWN"


Cycling_up_kegg=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/Cycling_up_kegg.csv')

Cycling_up_kegg=Cycling_up_kegg[order(Cycling_up_kegg$P.Value),]

Cycling_up_kegg=Cycling_up_kegg[1:30,]

Cycling_up_kegg$Input.number=as.numeric(Cycling_up_kegg$Input.number)
Cycling_up_kegg$Background.number=as.numeric(Cycling_up_kegg$Background.number)

Cycling_up_kegg$richFactor<-Cycling_up_kegg$Input.number/Cycling_up_kegg$Background.number

Cycling_up_kegg=Cycling_up_kegg[order(Cycling_up_kegg$P.Value),]

Cycling_up_kegg$Cluster="UP"


Cycling_kegg=rbind(Cycling_down_kegg,Cycling_up_kegg)


ggplot(data=Cycling_kegg)+aes(x=Cluster,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(Cycling_kegg$X.Term))+theme_bw()+xlab("")+ylab("")+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.1,1))+
  guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))


###REV3L-------
REV3L_down_ego=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/REV3L_down_ego.csv')

REV3L_down_ego=REV3L_down_ego[order(REV3L_down_ego$P.Value),]

REV3L_down_ego=REV3L_down_ego[1:30,]

REV3L_down_ego$Input.number=as.numeric(REV3L_down_ego$Input.number)
REV3L_down_ego$Background.number=as.numeric(REV3L_down_ego$Background.number)

REV3L_down_ego$richFactor<-REV3L_down_ego$Input.number/REV3L_down_ego$Background.number

REV3L_down_ego=REV3L_down_ego[order(REV3L_down_ego$P.Value),]

REV3L_down_ego$Cluster="DOWN"


REV3L_up_ego=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/REV3L_up_ego.csv')

REV3L_up_ego=REV3L_up_ego[order(REV3L_up_ego$P.Value),]

REV3L_up_ego=REV3L_up_ego[1:30,]

REV3L_up_ego$Input.number=as.numeric(REV3L_up_ego$Input.number)
REV3L_up_ego$Background.number=as.numeric(REV3L_up_ego$Background.number)

REV3L_up_ego$richFactor<-REV3L_up_ego$Input.number/REV3L_up_ego$Background.number

REV3L_up_ego=REV3L_up_ego[order(REV3L_up_ego$P.Value),]

REV3L_up_ego$Cluster="UP"


REV3L_ego=rbind(REV3L_down_ego,REV3L_up_ego)


ggplot(data=REV3L_ego)+aes(x=Cluster,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(REV3L_ego$X.Term))+theme_bw()+xlab("")+ylab("")+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.1,1))+
  guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))


REV3L_down_kegg=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/REV3L_down_kegg.csv')

REV3L_down_kegg=REV3L_down_kegg[order(REV3L_down_kegg$P.Value),]

REV3L_down_kegg=REV3L_down_kegg[1:30,]

REV3L_down_kegg$Input.number=as.numeric(REV3L_down_kegg$Input.number)
REV3L_down_kegg$Background.number=as.numeric(REV3L_down_kegg$Background.number)

REV3L_down_kegg$richFactor<-REV3L_down_kegg$Input.number/REV3L_down_kegg$Background.number

REV3L_down_kegg=REV3L_down_kegg[order(REV3L_down_kegg$P.Value),]

REV3L_down_kegg$Cluster="DOWN"


REV3L_up_kegg=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/REV3L_up_kegg.csv')

REV3L_up_kegg=REV3L_up_kegg[order(REV3L_up_kegg$P.Value),]

REV3L_up_kegg=REV3L_up_kegg[1:30,]

REV3L_up_kegg$Input.number=as.numeric(REV3L_up_kegg$Input.number)
REV3L_up_kegg$Background.number=as.numeric(REV3L_up_kegg$Background.number)

REV3L_up_kegg$richFactor<-REV3L_up_kegg$Input.number/REV3L_up_kegg$Background.number

REV3L_up_kegg=REV3L_up_kegg[order(REV3L_up_kegg$P.Value),]

REV3L_up_kegg$Cluster="UP"


REV3L_kegg=rbind(REV3L_down_kegg,REV3L_up_kegg)


ggplot(data=REV3L_kegg)+aes(x=Cluster,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(REV3L_kegg$X.Term))+theme_bw()+xlab("")+ylab("")+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.1,1))+
  guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))


###ISG15-------
ISG15_down_ego=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/ISG15_down_ego.csv')

ISG15_down_ego=ISG15_down_ego[order(ISG15_down_ego$P.Value),]

ISG15_down_ego=ISG15_down_ego[1:30,]

ISG15_down_ego$Input.number=as.numeric(ISG15_down_ego$Input.number)
ISG15_down_ego$Background.number=as.numeric(ISG15_down_ego$Background.number)

ISG15_down_ego$richFactor<-ISG15_down_ego$Input.number/ISG15_down_ego$Background.number

ISG15_down_ego=ISG15_down_ego[order(ISG15_down_ego$P.Value),]

ISG15_down_ego$Cluster="DOWN"


ISG15_up_ego=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/ISG15_up_ego.csv')

ISG15_up_ego=ISG15_up_ego[order(ISG15_up_ego$P.Value),]

ISG15_up_ego=ISG15_up_ego[1:30,]

ISG15_up_ego$Input.number=as.numeric(ISG15_up_ego$Input.number)
ISG15_up_ego$Background.number=as.numeric(ISG15_up_ego$Background.number)

ISG15_up_ego$richFactor<-ISG15_up_ego$Input.number/ISG15_up_ego$Background.number

ISG15_up_ego=ISG15_up_ego[order(ISG15_up_ego$P.Value),]

ISG15_up_ego$Cluster="UP"


ISG15_ego=rbind(ISG15_down_ego,ISG15_up_ego)


ggplot(data=ISG15_ego)+aes(x=Cluster,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(ISG15_ego$X.Term))+theme_bw()+xlab("")+ylab("")+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.1,1))+
  guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))


ISG15_down_kegg=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/ISG15_down_kegg.csv')

ISG15_down_kegg=ISG15_down_kegg[order(ISG15_down_kegg$P.Value),]

ISG15_down_kegg=ISG15_down_kegg[1:30,]

ISG15_down_kegg$Input.number=as.numeric(ISG15_down_kegg$Input.number)
ISG15_down_kegg$Background.number=as.numeric(ISG15_down_kegg$Background.number)

ISG15_down_kegg$richFactor<-ISG15_down_kegg$Input.number/ISG15_down_kegg$Background.number

ISG15_down_kegg=ISG15_down_kegg[order(ISG15_down_kegg$P.Value),]

ISG15_down_kegg$Cluster="DOWN"


ISG15_up_kegg=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/ISG15_up_kegg.csv')

ISG15_up_kegg=ISG15_up_kegg[order(ISG15_up_kegg$P.Value),]

ISG15_up_kegg=ISG15_up_kegg[1:30,]

ISG15_up_kegg$Input.number=as.numeric(ISG15_up_kegg$Input.number)
ISG15_up_kegg$Background.number=as.numeric(ISG15_up_kegg$Background.number)

ISG15_up_kegg$richFactor<-ISG15_up_kegg$Input.number/ISG15_up_kegg$Background.number

ISG15_up_kegg=ISG15_up_kegg[order(ISG15_up_kegg$P.Value),]

ISG15_up_kegg$Cluster="UP"


ISG15_kegg=rbind(ISG15_down_kegg,ISG15_up_kegg)


ggplot(data=ISG15_kegg)+aes(x=Cluster,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(ISG15_kegg$X.Term))+theme_bw()+xlab("")+ylab("")+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.1,1))+
  guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))


###APOD-------
APOD_down_ego=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/APOD_down_ego.csv')

APOD_down_ego=APOD_down_ego[order(APOD_down_ego$P.Value),]

APOD_down_ego=APOD_down_ego[1:30,]

APOD_down_ego$Input.number=as.numeric(APOD_down_ego$Input.number)
APOD_down_ego$Background.number=as.numeric(APOD_down_ego$Background.number)

APOD_down_ego$richFactor<-APOD_down_ego$Input.number/APOD_down_ego$Background.number

APOD_down_ego=APOD_down_ego[order(APOD_down_ego$P.Value),]

APOD_down_ego$Cluster="DOWN"


APOD_up_ego=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/APOD_up_ego.csv')

APOD_up_ego=APOD_up_ego[order(APOD_up_ego$P.Value),]

APOD_up_ego=APOD_up_ego[1:30,]

APOD_up_ego$Input.number=as.numeric(APOD_up_ego$Input.number)
APOD_up_ego$Background.number=as.numeric(APOD_up_ego$Background.number)

APOD_up_ego$richFactor<-APOD_up_ego$Input.number/APOD_up_ego$Background.number

APOD_up_ego=APOD_up_ego[order(APOD_up_ego$P.Value),]

APOD_up_ego$Cluster="UP"


APOD_ego=rbind(APOD_down_ego,APOD_up_ego)


ggplot(data=APOD_ego)+aes(x=Cluster,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(APOD_ego$X.Term))+theme_bw()+xlab("")+ylab("")+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.1,1))+
  guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))


APOD_down_kegg=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/APOD_down_kegg.csv')

APOD_down_kegg=APOD_down_kegg[order(APOD_down_kegg$P.Value),]

APOD_down_kegg=APOD_down_kegg[1:30,]

APOD_down_kegg$Input.number=as.numeric(APOD_down_kegg$Input.number)
APOD_down_kegg$Background.number=as.numeric(APOD_down_kegg$Background.number)

APOD_down_kegg$richFactor<-APOD_down_kegg$Input.number/APOD_down_kegg$Background.number

APOD_down_kegg=APOD_down_kegg[order(APOD_down_kegg$P.Value),]

APOD_down_kegg$Cluster="DOWN"


APOD_up_kegg=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/APOD_up_kegg.csv')

APOD_up_kegg=APOD_up_kegg[order(APOD_up_kegg$P.Value),]

APOD_up_kegg=APOD_up_kegg[1:30,]

APOD_up_kegg$Input.number=as.numeric(APOD_up_kegg$Input.number)
APOD_up_kegg$Background.number=as.numeric(APOD_up_kegg$Background.number)

APOD_up_kegg$richFactor<-APOD_up_kegg$Input.number/APOD_up_kegg$Background.number

APOD_up_kegg=APOD_up_kegg[order(APOD_up_kegg$P.Value),]

APOD_up_kegg$Cluster="UP"


APOD_kegg=rbind(APOD_down_kegg,APOD_up_kegg)


ggplot(data=APOD_kegg)+aes(x=Cluster,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(APOD_kegg$X.Term))+theme_bw()+xlab("")+ylab("")+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.1,1))+
  guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))


###P-------
P_down_ego=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/P_down_ego.csv')

P_down_ego=P_down_ego[order(P_down_ego$P.Value),]

P_down_ego=P_down_ego[1:30,]

P_down_ego$Input.number=as.numeric(P_down_ego$Input.number)
P_down_ego$Background.number=as.numeric(P_down_ego$Background.number)

P_down_ego$richFactor<-P_down_ego$Input.number/P_down_ego$Background.number

P_down_ego=P_down_ego[order(P_down_ego$P.Value),]

P_down_ego$Cluster="DOWN"


P_up_ego=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/P_up_ego.csv')

P_up_ego=P_up_ego[order(P_up_ego$P.Value),]

P_up_ego=P_up_ego[1:30,]

P_up_ego$Input.number=as.numeric(P_up_ego$Input.number)
P_up_ego$Background.number=as.numeric(P_up_ego$Background.number)

P_up_ego$richFactor<-P_up_ego$Input.number/P_up_ego$Background.number

P_up_ego=P_up_ego[order(P_up_ego$P.Value),]

P_up_ego$Cluster="UP"


P_ego=rbind(P_down_ego,P_up_ego)


ggplot(data=P_ego)+aes(x=Cluster,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(P_ego$X.Term))+theme_bw()+xlab("")+ylab("")+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.1,1))+
  guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))


P_down_kegg=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/P_down_kegg.csv')

P_down_kegg=P_down_kegg[order(P_down_kegg$P.Value),]

P_down_kegg=P_down_kegg[1:30,]

P_down_kegg$Input.number=as.numeric(P_down_kegg$Input.number)
P_down_kegg$Background.number=as.numeric(P_down_kegg$Background.number)

P_down_kegg$richFactor<-P_down_kegg$Input.number/P_down_kegg$Background.number

P_down_kegg=P_down_kegg[order(P_down_kegg$P.Value),]

P_down_kegg$Cluster="DOWN"


P_up_kegg=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/P_up_kegg.csv')

P_up_kegg=P_up_kegg[order(P_up_kegg$P.Value),]

P_up_kegg=P_up_kegg[1:30,]

P_up_kegg$Input.number=as.numeric(P_up_kegg$Input.number)
P_up_kegg$Background.number=as.numeric(P_up_kegg$Background.number)

P_up_kegg$richFactor<-P_up_kegg$Input.number/P_up_kegg$Background.number

P_up_kegg=P_up_kegg[order(P_up_kegg$P.Value),]

P_up_kegg$Cluster="UP"


P_kegg=rbind(P_down_kegg,P_up_kegg)


ggplot(data=P_kegg)+aes(x=Cluster,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(P_kegg$X.Term))+theme_bw()+xlab("")+ylab("")+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.1,1))+
  guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))


#对STR开始开始进行monocle[所有细胞，所有分组]-1-----
library(monocle)
library(Seurat)
library(SeuratDisk)
library(harmony)
library(dplyr)
library(tidyr)
library(ggplot2)
library(clusterProfiler)
library(org.Hs.eg.db)
library(RColorBrewer)
library(export)


STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")


STR$type0227=STR$type2

STR$type0227[STR$type0227%in%"IUA1"]<-"Adjacent"
STR$type0227[STR$type0227%in%"IUA2"]<-"Adhesion"

DimPlot(STR,group.by = "type2",label = T)

DimPlot(STR,group.by = "type0227",label = T)

DefaultAssay(STR)<-"RNA"

STR_matrix<-as(as.matrix(STR@assays$RNA@counts),'sparseMatrix')

STR_ann<-data.frame(gene_id=rownames(STR_matrix),gene_short_name=rownames(STR_matrix))

rownames(STR_ann)<-rownames(STR_matrix)

STR_fd<-new("AnnotatedDataFrame", data = STR_ann)


sample_ann<-STR@meta.data
rownames(sample_ann)<-colnames(STR_matrix)

STR_pd<-new("AnnotatedDataFrame", data =sample_ann)

STR.cds<-newCellDataSet(STR_matrix,phenoData =STR_pd,
                        featureData =STR_fd,expressionFamily=negbinomial.size())


#预处理
STR.cds <- estimateSizeFactors(STR.cds)
STR.cds <- estimateDispersions(STR.cds)

STR.cds <- detectGenes(STR.cds, min_expr = 0.1)

disp_table <- dispersionTable(STR.cds)

unsup_clustering_genes <- subset(disp_table, mean_expression >= 0.1)

STR.cds <- setOrderingFilter(STR.cds, unsup_clustering_genes$gene_id)


STR.cds<- reduceDimension(STR.cds, max_components = 2, num_dim = 8, 
                          reduction_method = 'tSNE', verbose = T)

STR.cds<- clusterCells(STR.cds, num_clusters = 8)


expressed_genes <- row.names(subset(fData(STR.cds),
                                    num_cells_expressed >= 10))

clustering_DEG_genes <- differentialGeneTest(STR.cds[expressed_genes,],
                                             fullModelFormulaStr = "~ type0227",cores = 8)

ordering_genes <- row.names(clustering_DEG_genes)[order(clustering_DEG_genes$qval)][1:800] #排序后取前800个基因

STR.cds<- setOrderingFilter(STR.cds, ordering_genes)

#开始绘图用DDrtree 方法进行降维、画图
STR.cds <- reduceDimension(
  STR.cds,
  max_components = 2,
  method = 'DDRTree')

STR.cds <- orderCells(STR.cds)
head(pData(STR.cds))


STR_P1=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "type0227",label=T)+ggtitle("")

STR_P2=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "Pseudotime")

STR_P3=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "type0227",label=T)+ggtitle("")+
  facet_wrap(~type0227, nrow = 1)

STR_P3
STR_P1+STR_P2

STR.cds$cell_type0305

STR_P1=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "cell_type0305",label=T)+ggtitle("")

STR_P3=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "cell_type0305",label=T)+ggtitle("")+
  facet_wrap(~cell_type0305, nrow = 2)
STR_P3
STR_P1+STR_P2

saveRDS(STR.cds,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/文件存放点/STR_0303_1.rds")

#开始寻找分支热图的绘制
#节点1
BEAM_res <- BEAM(STR.cds, branch_point = 1,cores = 8) #第一个结点

BEAM_res <- BEAM_res[order(BEAM_res$qval),]

BEAM_res <- BEAM_res[,c("gene_short_name", "pval", "qval")]

mtGenes=grep('^MT-',rownames(STR),value=TRUE)
RPGenes=c(grep('^RPL',rownames(STR),value=TRUE),grep('^RPS',rownames(STR),value=TRUE))
stress_gene2=grep('^HSP',rownames(STR),value=TRUE)
HB_gene2=grep('^HB',rownames(STR),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
stress_gene1=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
               "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/gene1.csv")
subgene3=subgene3$cellcyle_gene
subgene4=c("FOS","CXCL2","ZFP36","FOSB","DUSP1","ATF3","CXCL8","NR4A1","CXCL3","PPP1R15A",
           "JUNB","EGR1","HSPA1A","HSPA1B","SOCS3","KLF6","JUN","IER2","CXCL1","NFKBIA",
           "HSPA6","DNAJB1","IER3","CCNL1","MTRNR2L2","IER4","ID1","CEBPD","KRT6A","CYR61",
           "DEPP1","CLDN4","IRF1","DUSP2","BTG2","PLAUR","MAFF","KLF4","PHLDA2","TNFAIP3")
subgene5=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/512个应激反应相关基因.csv")
subgene5=subgene5$gene_symbol

subgene=c(mtGenes,RPGenes,stress_gene2,HB_gene2,HB.genes,s.genes,g2m.genes,stress_gene1,"MKI67",subgene3,subgene4,subgene5)

BEAM_res=subset(BEAM_res,!gene_short_name%in%subgene)


write.csv(BEAM_res,
          "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_all_0305/BEAM_res_1.csv")
saveRDS(STR.cds,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/文件存放点/STR_0303_1.rds")

STR.cds=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/文件存放点/STR_0303_1.rds")

BEAM_res=
  read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_all_0305/BEAM_res_1.csv",row.names = 1)




BEAM_res2=BEAM_res[1:800,]

p_BEAM = plot_genes_branched_heatmap(STR.cds[row.names(subset(BEAM_res2,qval<1e-4)),],branch_point = 1,
                                     cores = 8,branch_colors = c("#979797", "#F05662", "#7990C8"),
                                     use_gene_short_name = T,hmcols = colorRampPalette(rev(brewer.pal(11, "Spectral")))(70),
                                     show_rownames = F,return_heatmap = T)






#对上面这个monocle的结果进行分支确定cellfate并进行特定cluster基因的富集分析----
STR.cds=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/文件存放点/STR_0303_1.rds")

STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

STR$State=STR.cds$State

BEAM_res=
  read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_all_0305/BEAM_res_1.csv",row.names = 1)




BEAM_res2=BEAM_res[1:800,]

p_BEAM = plot_genes_branched_heatmap(STR.cds[row.names(subset(BEAM_res2,qval<1e-4)),],branch_point = 1,
                                     cores = 8,branch_colors = c("#979797", "#F05662", "#7990C8"),
                                     use_gene_short_name = T,hmcols = colorRampPalette(rev(brewer.pal(11, "Spectral")))(70),
                                     show_rownames = F,return_heatmap = T)



matrix1=as.data.frame(p_BEAM$BranchA_exprs)
matrix2=as.data.frame(p_BEAM$BranchB_exprs)

matrix=cbind(matrix1,matrix2)

matrix3=as.data.frame(p_BEAM$annotation_col)

matrix4=as.data.frame(p_BEAM$annotation_row)

gene_cluster3=subset(matrix4,matrix4$Cluster%in%"3")

K=matrix[rownames(matrix)%in%rownames(gene_cluster3),]


AverageExpression(STR,group.by = "State",features = "ITGA6")

###cluster3进行富集分析GO和kegg------
gene_cluster3=rownames(subset(matrix4,matrix4$Cluster%in%"3"))

###GO富集
cluster3_gene <- bitr(gene_cluster3, fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
cluster3_gene=cluster3_gene$ENTREZID
cluster3_ego <- enrichGO(OrgDb="org.Hs.eg.db", gene = cluster3_gene, ont = "ALL",readable= TRUE)

write.csv(cluster3_ego,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_all_0305/富集分析/分支1/cluster3_ego.csv")



cluster3_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_all_0305/富集分析/分支1/cluster3_ego.csv")


tonumeric <- function(x){
  x <- sub(" ", "+", x, fixed = TRUE)
  return(unlist(lapply(x, function(x) eval(parse(text=x)))))
}

cluster3_ego$GeneRatio=tonumeric(cluster3_ego$GeneRatio)

cluster3_ego=cluster3_ego[order(cluster3_ego$GeneRatio),]


ggplot(data=cluster3_ego)+aes(x=GeneRatio,y=Description,color=-log10(pvalue))+geom_point(aes(size=Count,color= -log10(pvalue)))+
  scale_y_discrete(limits=cluster3_ego$Description)+theme_bw()+
  theme(axis.text.y = element_text(size=11))+guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))+
  xlab("Cluster")+ylab("")+scale_size_continuous(range = c(3,10))+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values =c(0,0.01,1))+
  xlab("")+ylab("")+
  theme(panel.border = element_blank(),axis.title = element_blank(),axis.text = element_blank(),
        axis.ticks = element_blank(),axis.line = element_blank())



###kegg
gene_cluster3=rownames(subset(matrix4,matrix4$Cluster%in%"3"))
cluster3_gene <- bitr(gene_cluster3, fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
cluster3_gene=cluster3_gene$ENTREZID

cluster3_ekk <- enrichKEGG(gene= cluster3_gene,organism= "hsa", 
                           pvalueCutoff=0.05,keyType = "kegg")




###cluster2进行富集分析GO和kegg------
gene_cluster2=rownames(subset(matrix4,matrix4$Cluster%in%"2"))



###GO富集
cluster2_gene <- bitr(gene_cluster2, fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
cluster2_gene=cluster2_gene$ENTREZID
cluster2_ego <- enrichGO(OrgDb="org.Hs.eg.db", gene = cluster2_gene, ont = "ALL",readable= TRUE)

write.csv(cluster2_ego,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_all_0305/富集分析/分支1/cluster2_ego.csv")



cluster2_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_all_0305/富集分析/分支1/cluster2_ego.csv")

cluster2_ego=cluster2_ego[order(cluster2_ego$BgRatio),]
cluster2_ego=cluster2_ego[1:30,]

tonumeric <- function(x){
  x <- sub(" ", "+", x, fixed = TRUE)
  return(unlist(lapply(x, function(x) eval(parse(text=x)))))
}

cluster2_ego$GeneRatio=tonumeric(cluster2_ego$GeneRatio)

cluster2_ego=cluster2_ego[order(cluster2_ego$GeneRatio),]


ggplot(data=cluster2_ego)+aes(x=GeneRatio,y=Description,color=-log10(pvalue))+geom_point(aes(size=Count,color= -log10(pvalue)))+
  scale_y_discrete(limits=cluster2_ego$Description)+theme_bw()+
  theme(axis.text.y = element_text(size=11))+guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))+
  xlab("Cluster")+ylab("")+scale_size_continuous(range = c(3,10))+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values =c(0,0.01,1))+
  xlab("")+ylab("")+
  theme(panel.border = element_blank(),axis.title = element_blank(),axis.text = element_blank(),
        axis.ticks = element_blank(),axis.line = element_blank())



###kegg
gene_cluster2=rownames(subset(matrix4,matrix4$Cluster%in%"3"))
cluster2_ekk <- enrichKEGG(gene= cluster2_gene,organism= "human", 
                           pvalueCutoff=0.05,keyType = "kegg")

###cluster1进行富集分析GO和kegg------
gene_cluster1=rownames(subset(matrix4,matrix4$Cluster%in%"1"))



###GO富集
cluster1_gene <- bitr(gene_cluster1, fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
cluster1_gene=cluster1_gene$ENTREZID
cluster1_ego <- enrichGO(OrgDb="org.Hs.eg.db", gene = cluster1_gene, ont = "ALL",readable= TRUE)

write.csv(cluster1_ego,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_all_0305/富集分析/分支1/cluster1_ego.csv")



cluster1_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_all_0305/富集分析/分支1/cluster1_ego.csv")

cluster1_ego=cluster1_ego[order(cluster1_ego$BgRatio),]
cluster1_ego=cluster1_ego[1:30,]

tonumeric <- function(x){
  x <- sub(" ", "+", x, fixed = TRUE)
  return(unlist(lapply(x, function(x) eval(parse(text=x)))))
}

cluster1_ego$GeneRatio=tonumeric(cluster1_ego$GeneRatio)

cluster1_ego=cluster1_ego[order(cluster1_ego$GeneRatio),]


ggplot(data=cluster1_ego)+aes(x=GeneRatio,y=Description,color=-log10(pvalue))+geom_point(aes(size=Count,color= -log10(pvalue)))+
  scale_y_discrete(limits=cluster1_ego$Description)+theme_bw()+
  theme(axis.text.y = element_text(size=11))+guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))+
  xlab("Cluster")+ylab("")+scale_size_continuous(range = c(3,10))+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values =c(0,0.01,1))+
  xlab("")+ylab("")+
  theme(panel.border = element_blank(),axis.title = element_blank(),axis.text = element_blank(),
        axis.ticks = element_blank(),axis.line = element_blank())



###kegg
gene_cluster1=rownames(subset(matrix4,matrix4$Cluster%in%"3"))

cluster1_ekk <- enrichKEGG(gene= cluster1_gene,organism= "human", 
                           pvalueCutoff=0.05,keyType = "kegg")



###cluster5进行富集分析GO和kegg------
gene_cluster5=rownames(subset(matrix4,matrix4$Cluster%in%"5"))



###GO富集
cluster5_gene <- bitr(gene_cluster5, fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
cluster5_gene=cluster5_gene$ENTREZID
cluster5_ego <- enrichGO(OrgDb="org.Hs.eg.db", gene = cluster5_gene, ont = "ALL",readable= TRUE)

write.csv(cluster5_ego,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_all_0305/富集分析/分支1/cluster5_ego.csv")



cluster5_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_all_0305/富集分析/分支1/cluster5_ego.csv")

cluster5_ego=cluster5_ego[order(cluster5_ego$BgRatio),]

tonumeric <- function(x){
  x <- sub(" ", "+", x, fixed = TRUE)
  return(unlist(lapply(x, function(x) eval(parse(text=x)))))
}

cluster5_ego$GeneRatio=tonumeric(cluster5_ego$GeneRatio)

cluster5_ego=cluster5_ego[order(cluster5_ego$GeneRatio),]


ggplot(data=cluster5_ego)+aes(x=GeneRatio,y=Description,color=-log10(pvalue))+geom_point(aes(size=Count,color= -log10(pvalue)))+
  scale_y_discrete(limits=cluster5_ego$Description)+theme_bw()+
  theme(axis.text.y = element_text(size=11))+guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))+
  xlab("Cluster")+ylab("")+scale_size_continuous(range = c(3,10))+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values =c(0,0.01,1))+
  xlab("")+ylab("")+
  theme(panel.border = element_blank(),axis.title = element_blank(),axis.text = element_blank(),
        axis.ticks = element_blank(),axis.line = element_blank())



###kegg
gene_cluster5=rownames(subset(matrix4,matrix4$Cluster%in%"3"))

cluster5_ekk <- enrichKEGG(gene= cluster5_gene,organism= "human", 
                           pvalueCutoff=0.05,keyType = "kegg")



#将所有cluster的一个基因出完整的表并用kobas做富集-----
###文件准备
STR.cds=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/文件存放点/STR_0303_1.rds")

STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

STR$State=STR.cds$State

BEAM_res=
  read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_all_0305/BEAM_res_1.csv",row.names = 1)




BEAM_res2=BEAM_res[1:800,]

p_BEAM = plot_genes_branched_heatmap(STR.cds[row.names(subset(BEAM_res2,qval<1e-4)),],branch_point = 1,
                                     cores = 8,branch_colors = c("#979797", "#F05662", "#7990C8"),
                                     use_gene_short_name = T,hmcols = colorRampPalette(rev(brewer.pal(11, "Spectral")))(70),
                                     show_rownames = F,return_heatmap = T)
matrix4=as.data.frame(p_BEAM$annotation_row)
matrix4$gene=rownames(matrix4)

gene_cluster1=rownames(subset(matrix4,matrix4$Cluster%in%"1"))
gene_cluster2=rownames(subset(matrix4,matrix4$Cluster%in%"2"))
gene_cluster3=rownames(subset(matrix4,matrix4$Cluster%in%"3"))
gene_cluster4=rownames(subset(matrix4,matrix4$Cluster%in%"4"))
gene_cluster5=rownames(subset(matrix4,matrix4$Cluster%in%"5"))
gene_cluster6=rownames(subset(matrix4,matrix4$Cluster%in%"6"))


Z=rbind.fill(list(as.data.frame(t(gene_cluster1)),as.data.frame(t(gene_cluster2)),
                  as.data.frame(t(gene_cluster3)),as.data.frame(t(gene_cluster4)),
                  as.data.frame(t(gene_cluster5)),as.data.frame(t(gene_cluster6))
))
Z=as.data.frame(t(Z))

write.csv(Z,
          "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_all_0305/富集分析/分支1/cluster_all.csv")

#对kobas网站的结果做图-GO富集-----
###cluster1
C1_GO=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_all_0305/富集分析/分支1/kobas/20230309/cluster1_GO.csv')

C1_GO=C1_GO[order(C1_GO$P.Value),]

C1_GO=C1_GO[1:30,]

C1_GO$Input.number=as.numeric(C1_GO$Input.number)
C1_GO$Background.number=as.numeric(C1_GO$Background.number)

C1_GO$richFactor<-C1_GO$Input.number/C1_GO$Background.number

C1_GO=C1_GO[order(C1_GO$richFactor),]


ggplot(data=C1_GO)+aes(x=richFactor,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(C1_GO$X.Term))+theme_bw()+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.1,1))+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+xlab("")+
  ylab("")



###cluster2
C2_GO=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_all_0305/富集分析/分支1/kobas/20230309/cluster2_GO.csv')

C2_GO=C2_GO[order(C2_GO$P.Value),]

C2_GO=C2_GO[1:30,]

C2_GO$Input.number=as.numeric(C2_GO$Input.number)
C2_GO$Background.number=as.numeric(C2_GO$Background.number)

C2_GO$richFactor<-C2_GO$Input.number/C2_GO$Background.number

C2_GO=C2_GO[order(C2_GO$richFactor),]


ggplot(data=C2_GO)+aes(x=richFactor,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(C2_GO$X.Term))+theme_bw()+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.1,1))+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+xlab("")+
  ylab("")


###cluster3
C3_GO=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_all_0305/富集分析/分支1/kobas/20230309/cluster3_GO.csv')

C3_GO=C3_GO[order(C3_GO$P.Value),]

C3_GO=C3_GO[1:30,]

C3_GO$Input.number=as.numeric(C3_GO$Input.number)
C3_GO$Background.number=as.numeric(C3_GO$Background.number)

C3_GO$richFactor<-C3_GO$Input.number/C3_GO$Background.number

C3_GO=C3_GO[order(C3_GO$richFactor),]


ggplot(data=C3_GO)+aes(x=richFactor,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(C3_GO$X.Term))+theme_bw()+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.1,1))+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+xlab("")+
  ylab("")


###cluster4
C4_GO=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_all_0305/富集分析/分支1/kobas/20230309/cluster4_GO.csv')

C4_GO=C4_GO[order(C4_GO$P.Value),]

C4_GO=C4_GO[1:30,]

C4_GO$Input.number=as.numeric(C4_GO$Input.number)
C4_GO$Background.number=as.numeric(C4_GO$Background.number)

C4_GO$richFactor<-C4_GO$Input.number/C4_GO$Background.number

C4_GO=C4_GO[order(C4_GO$richFactor),]


ggplot(data=C4_GO)+aes(x=richFactor,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(C4_GO$X.Term))+theme_bw()+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.1,1))+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+xlab("")+
  ylab("")+theme(axis.text.y = element_text(size = 10))


###cluster5
C5_GO=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_all_0305/富集分析/分支1/kobas/20230309/cluster5_GO.csv')

C5_GO=C5_GO[order(C5_GO$P.Value),]

C5_GO=C5_GO[1:30,]

C5_GO$Input.number=as.numeric(C5_GO$Input.number)
C5_GO$Background.number=as.numeric(C5_GO$Background.number)

C5_GO$richFactor<-C5_GO$Input.number/C5_GO$Background.number

C5_GO=C5_GO[order(C5_GO$richFactor),]


ggplot(data=C5_GO)+aes(x=richFactor,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(C5_GO$X.Term))+theme_bw()+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.1,1))+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+xlab("")+
  ylab("")+theme(axis.text.y = element_text(size = 10))


###cluster5
C6_GO=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_all_0305/富集分析/分支1/kobas/20230309/cluster6_GO.csv')

C6_GO=C6_GO[order(C6_GO$P.Value),]

C6_GO=C6_GO[1:30,]

C6_GO$Input.number=as.numeric(C6_GO$Input.number)
C6_GO$Background.number=as.numeric(C6_GO$Background.number)

C6_GO$richFactor<-C6_GO$Input.number/C6_GO$Background.number

C6_GO=C6_GO[order(C6_GO$richFactor),]


ggplot(data=C6_GO)+aes(x=richFactor,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(C6_GO$X.Term))+theme_bw()+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.1,1))+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+xlab("")+
  ylab("")+theme(axis.text.y = element_text(size = 10))




#对kobas网站的结果做图-kegg富集-----
###cluster1
C1_kegg=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_all_0305/富集分析/分支1/kobas/20230309/cluster1_kegg.csv')

C1_kegg=na.omit(C1_kegg)

C1_kegg=C1_kegg[order(C1_kegg$P.Value),]

C1_kegg=C1_kegg[1:30,]

C1_kegg$Input.number=as.numeric(C1_kegg$Input.number)
C1_kegg$Background.number=as.numeric(C1_kegg$Background.number)

C1_kegg$richFactor<-C1_kegg$Input.number/C1_kegg$Background.number

C1_kegg=C1_kegg[order(C1_kegg$richFactor),]


ggplot(data=C1_kegg)+aes(x=richFactor,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(C1_kegg$X.Term))+theme_bw()+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.1,1))+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+xlab("")+
  ylab("")



###cluster2
C2_kegg=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_all_0305/富集分析/分支1/kobas/20230309/cluster2_kegg.csv')

C2_kegg=C2_kegg[order(C2_kegg$P.Value),]

C2_kegg=C2_kegg[1:30,]

C2_kegg$Input.number=as.numeric(C2_kegg$Input.number)
C2_kegg$Background.number=as.numeric(C2_kegg$Background.number)

C2_kegg$richFactor<-C2_kegg$Input.number/C2_kegg$Background.number

C2_kegg=C2_kegg[order(C2_kegg$richFactor),]


ggplot(data=C2_kegg)+aes(x=richFactor,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(C2_kegg$X.Term))+theme_bw()+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.1,1))+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+xlab("")+
  ylab("")


###cluster3
C3_kegg=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_all_0305/富集分析/分支1/kobas/20230309/cluster3_kegg.csv')

C3_kegg=C3_kegg[order(C3_kegg$P.Value),]

C3_kegg=C3_kegg[1:30,]

C3_kegg$Input.number=as.numeric(C3_kegg$Input.number)
C3_kegg$Background.number=as.numeric(C3_kegg$Background.number)

C3_kegg$richFactor<-C3_kegg$Input.number/C3_kegg$Background.number

C3_kegg=C3_kegg[order(C3_kegg$richFactor),]


ggplot(data=C3_kegg)+aes(x=richFactor,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(C3_kegg$X.Term))+theme_bw()+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.1,1))+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+xlab("")+
  ylab("")


###cluster4
C4_kegg=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_all_0305/富集分析/分支1/kobas/20230309/cluster4_kegg.csv')

C4_kegg=C4_kegg[order(C4_kegg$P.Value),]

C4_kegg=C4_kegg[1:30,]

C4_kegg$Input.number=as.numeric(C4_kegg$Input.number)
C4_kegg$Background.number=as.numeric(C4_kegg$Background.number)

C4_kegg$richFactor<-C4_kegg$Input.number/C4_kegg$Background.number

C4_kegg=C4_kegg[order(C4_kegg$richFactor),]


ggplot(data=C4_kegg)+aes(x=richFactor,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(C4_kegg$X.Term))+theme_bw()+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.1,1))+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+xlab("")+
  ylab("")+theme(axis.text.y = element_text(size = 10))


###cluster5
C5_kegg=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_all_0305/富集分析/分支1/kobas/20230309/cluster5_kegg.csv')

C5_kegg=C5_kegg[order(C5_kegg$P.Value),]

C5_kegg=C5_kegg[1:30,]

C5_kegg$Input.number=as.numeric(C5_kegg$Input.number)
C5_kegg$Background.number=as.numeric(C5_kegg$Background.number)

C5_kegg$richFactor<-C5_kegg$Input.number/C5_kegg$Background.number

C5_kegg=C5_kegg[order(C5_kegg$richFactor),]


ggplot(data=C5_kegg)+aes(x=richFactor,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(C5_kegg$X.Term))+theme_bw()+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.1,1))+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+xlab("")+
  ylab("")+theme(axis.text.y = element_text(size = 10))


###cluster6
C6_kegg=read.csv('/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_all_0305/富集分析/分支1/kobas/20230309/cluster6_kegg.csv')

C6_kegg=C6_kegg[order(C6_kegg$P.Value),]

C6_kegg=C6_kegg[1:30,]

C6_kegg$Input.number=as.numeric(C6_kegg$Input.number)
C6_kegg$Background.number=as.numeric(C6_kegg$Background.number)

C6_kegg$richFactor<-C6_kegg$Input.number/C6_kegg$Background.number

C6_kegg=C6_kegg[order(C6_kegg$richFactor),]


ggplot(data=C6_kegg)+aes(x=richFactor,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(C6_kegg$X.Term))+theme_bw()+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.1,1))+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+xlab("")+
  ylab("")+theme(axis.text.y = element_text(size = 10))




#对monocle上述结果进行伪时间分布的umap绘图-----
STR.cds=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/文件存放点/STR_0303_1.rds")

STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

STR$State=STR.cds$State

STR$time=STR.cds$Pseudotime


FeaturePlot(STR,features = "time",cols = c("#23126F","#F9E048"))






#对STR开始开始进行monocle[所有细胞，CTRL+Adhesion]-1-----
library(monocle)
library(Seurat)
library(SeuratDisk)
library(harmony)
library(dplyr)
library(tidyr)
library(ggplot2)
library(clusterProfiler)
library(org.Hs.eg.db)
library(RColorBrewer)
library(export)


STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")


STR=subset(STR,type4%in%c("Ctrl","Adhesion"))

STR$type0227=STR$type2

STR$type0227[STR$type0227%in%"IUA1"]<-"Adjacent"
STR$type0227[STR$type0227%in%"IUA2"]<-"Adhesion"

DimPlot(STR,group.by = "type2",label = T)

DimPlot(STR,group.by = "type0227",label = T)

DefaultAssay(STR)<-"RNA"

STR_matrix<-as(as.matrix(STR@assays$RNA@counts),'sparseMatrix')

STR_ann<-data.frame(gene_id=rownames(STR_matrix),gene_short_name=rownames(STR_matrix))

rownames(STR_ann)<-rownames(STR_matrix)

STR_fd<-new("AnnotatedDataFrame", data = STR_ann)


sample_ann<-STR@meta.data
rownames(sample_ann)<-colnames(STR_matrix)

STR_pd<-new("AnnotatedDataFrame", data =sample_ann)

STR.cds<-newCellDataSet(STR_matrix,phenoData =STR_pd,
                        featureData =STR_fd,expressionFamily=negbinomial.size())


#预处理
STR.cds <- estimateSizeFactors(STR.cds)
STR.cds <- estimateDispersions(STR.cds)

STR.cds <- detectGenes(STR.cds, min_expr = 0.1)

disp_table <- dispersionTable(STR.cds)

unsup_clustering_genes <- subset(disp_table, mean_expression >= 0.1)

STR.cds <- setOrderingFilter(STR.cds, unsup_clustering_genes$gene_id)


STR.cds<- reduceDimension(STR.cds, max_components = 2, num_dim = 8, 
                          reduction_method = 'tSNE', verbose = T)

STR.cds<- clusterCells(STR.cds, num_clusters = 8)


expressed_genes <- row.names(subset(fData(STR.cds),
                                    num_cells_expressed >= 10))

clustering_DEG_genes <- differentialGeneTest(STR.cds[expressed_genes,],
                                             fullModelFormulaStr = "~ type0227",cores = 8)

ordering_genes <- row.names(clustering_DEG_genes)[order(clustering_DEG_genes$qval)][1:800] #排序后取前800个基因

STR.cds<- setOrderingFilter(STR.cds, ordering_genes)

#开始绘图用DDrtree 方法进行降维、画图
STR.cds <- reduceDimension(
  STR.cds,
  max_components = 2,
  method = 'DDRTree')

STR.cds <- orderCells(STR.cds)
head(pData(STR.cds))


STR_P1=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "type0227",label=T)+ggtitle("")

STR_P2=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "Pseudotime")

STR_P3=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "type0227",label=T)+ggtitle("")+
  facet_wrap(~type0227, nrow = 1)

STR_P3
STR_P1+STR_P2

STR.cds$cell_type0305

STR_P1=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "cell_type0305",label=T)+ggtitle("")

STR_P3=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "cell_type0305",label=T)+ggtitle("")+
  facet_wrap(~cell_type0305, nrow = 2)

saveRDS(STR.cds,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/文件存放点/STR_0307_1.rds")

#开始寻找分支热图的绘制
#节点1
BEAM_res <- BEAM(STR.cds, branch_point = 1,cores = 8) #第一个结点

BEAM_res <- BEAM_res[order(BEAM_res$qval),]

BEAM_res <- BEAM_res[,c("gene_short_name", "pval", "qval")]

mtGenes=grep('^MT-',rownames(STR),value=TRUE)
RPGenes=c(grep('^RPL',rownames(STR),value=TRUE),grep('^RPS',rownames(STR),value=TRUE))
stress_gene2=grep('^HSP',rownames(STR),value=TRUE)
HB_gene2=grep('^HB',rownames(STR),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
stress_gene1=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
               "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/gene1.csv")
subgene3=subgene3$cellcyle_gene
subgene4=c("FOS","CXCL2","ZFP36","FOSB","DUSP1","ATF3","CXCL8","NR4A1","CXCL3","PPP1R15A",
           "JUNB","EGR1","HSPA1A","HSPA1B","SOCS3","KLF6","JUN","IER2","CXCL1","NFKBIA",
           "HSPA6","DNAJB1","IER3","CCNL1","MTRNR2L2","IER4","ID1","CEBPD","KRT6A","CYR61",
           "DEPP1","CLDN4","IRF1","DUSP2","BTG2","PLAUR","MAFF","KLF4","PHLDA2","TNFAIP3")
subgene5=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/512个应激反应相关基因.csv")
subgene5=subgene5$gene_symbol

subgene=c(mtGenes,RPGenes,stress_gene2,HB_gene2,HB.genes,s.genes,g2m.genes,stress_gene1,"MKI67",subgene3,subgene4,subgene5)

BEAM_res=subset(BEAM_res,!gene_short_name%in%subgene)


write.csv(BEAM_res,
          "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_CTRL_adhesion_0307_1/BEAM_res_1.csv")

STR.cds=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/文件存放点/STR_0307_1.rds")

BEAM_res=
  read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_CTRL_adhesion_0307_1/BEAM_res_1.csv",row.names = 1)




BEAM_res2=BEAM_res[1:800,]

p_BEAM = plot_genes_branched_heatmap(STR.cds[row.names(subset(BEAM_res2,qval<1e-4)),],branch_point = 1,
                                     cores = 8,branch_colors = c("#979797", "#F05662", "#7990C8"),
                                     use_gene_short_name = T,hmcols = colorRampPalette(rev(brewer.pal(11, "Spectral")))(70),
                                     show_rownames = F,return_heatmap = T)






#对STR开始开始进行monocle[所有细胞，CTRL+Adhesion]-2-----
library(monocle)
library(Seurat)
library(SeuratDisk)
library(harmony)
library(dplyr)
library(tidyr)
library(ggplot2)
library(clusterProfiler)
library(org.Hs.eg.db)
library(RColorBrewer)
library(export)


STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")


STR=subset(STR,type4%in%c("Ctrl","Adhesion"))

STR$type0227=STR$type2

STR$type0227[STR$type0227%in%"IUA1"]<-"Adjacent"
STR$type0227[STR$type0227%in%"IUA2"]<-"Adhesion"

DimPlot(STR,group.by = "type2",label = T)

DimPlot(STR,group.by = "type0227",label = T)

DefaultAssay(STR)<-"RNA"

STR_matrix<-as(as.matrix(STR@assays$RNA@counts),'sparseMatrix')

STR_ann<-data.frame(gene_id=rownames(STR_matrix),gene_short_name=rownames(STR_matrix))

rownames(STR_ann)<-rownames(STR_matrix)

STR_fd<-new("AnnotatedDataFrame", data = STR_ann)


sample_ann<-STR@meta.data
rownames(sample_ann)<-colnames(STR_matrix)

STR_pd<-new("AnnotatedDataFrame", data =sample_ann)

STR.cds<-newCellDataSet(STR_matrix,phenoData =STR_pd,
                        featureData =STR_fd,expressionFamily=negbinomial.size())


#预处理
STR.cds <- estimateSizeFactors(STR.cds)
STR.cds <- estimateDispersions(STR.cds)

STR.cds <- detectGenes(STR.cds, min_expr = 0.1)

disp_table <- dispersionTable(STR.cds)

unsup_clustering_genes <- subset(disp_table, mean_expression >= 0.1)

STR.cds <- setOrderingFilter(STR.cds, unsup_clustering_genes$gene_id)


STR.cds<- reduceDimension(STR.cds, max_components = 2, num_dim = 8, 
                          reduction_method = 'tSNE', verbose = T)

STR.cds<- clusterCells(STR.cds, num_clusters = 8)


expressed_genes <- row.names(subset(fData(STR.cds),
                                    num_cells_expressed >= 10))

clustering_DEG_genes <- differentialGeneTest(STR.cds[expressed_genes,],
                                             fullModelFormulaStr = "~ cell_type0305",cores = 8)

ordering_genes <- row.names(clustering_DEG_genes)[order(clustering_DEG_genes$qval)][1:800] #排序后取前800个基因

STR.cds<- setOrderingFilter(STR.cds, ordering_genes)

#开始绘图用DDrtree 方法进行降维、画图
STR.cds <- reduceDimension(
  STR.cds,
  max_components = 2,
  method = 'DDRTree')

STR.cds <- orderCells(STR.cds)
head(pData(STR.cds))


STR_P1=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "type0227",label=T)+ggtitle("")

STR_P2=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "Pseudotime")

STR_P3=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "type0227",label=T)+ggtitle("")+
  facet_wrap(~type0227, nrow = 1)

STR_P3
STR_P1+STR_P2

STR.cds$cell_type0305

STR_P1=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "cell_type0305",label=T)+ggtitle("")

STR_P3=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "cell_type0305",label=T)+ggtitle("")+
  facet_wrap(~cell_type0305, nrow = 2)

STR_P3

STR_P1+STR_P2

saveRDS(STR.cds,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/文件存放点/STR_0307_2.rds")

#开始寻找分支热图的绘制
#节点1
BEAM_res <- BEAM(STR.cds, branch_point = 1,cores = 8) #第一个结点

BEAM_res <- BEAM_res[order(BEAM_res$qval),]

BEAM_res <- BEAM_res[,c("gene_short_name", "pval", "qval")]

mtGenes=grep('^MT-',rownames(STR),value=TRUE)
RPGenes=c(grep('^RPL',rownames(STR),value=TRUE),grep('^RPS',rownames(STR),value=TRUE))
stress_gene2=grep('^HSP',rownames(STR),value=TRUE)
HB_gene2=grep('^HB',rownames(STR),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
stress_gene1=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
               "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/gene1.csv")
subgene3=subgene3$cellcyle_gene
subgene4=c("FOS","CXCL2","ZFP36","FOSB","DUSP1","ATF3","CXCL8","NR4A1","CXCL3","PPP1R15A",
           "JUNB","EGR1","HSPA1A","HSPA1B","SOCS3","KLF6","JUN","IER2","CXCL1","NFKBIA",
           "HSPA6","DNAJB1","IER3","CCNL1","MTRNR2L2","IER4","ID1","CEBPD","KRT6A","CYR61",
           "DEPP1","CLDN4","IRF1","DUSP2","BTG2","PLAUR","MAFF","KLF4","PHLDA2","TNFAIP3")
subgene5=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/512个应激反应相关基因.csv")
subgene5=subgene5$gene_symbol

subgene=c(mtGenes,RPGenes,stress_gene2,HB_gene2,HB.genes,s.genes,g2m.genes,stress_gene1,"MKI67",subgene3,subgene4,subgene5)

BEAM_res=subset(BEAM_res,!gene_short_name%in%subgene)


write.csv(BEAM_res,
          "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_CTRL_adhesion_0307_2/BEAM_res_1.csv")

STR.cds=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/文件存放点/STR_0307_2.rds")

BEAM_res=
  read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_CTRL_adhesion_0307_2/BEAM_res_1.csv",row.names = 1)




BEAM_res2=BEAM_res[1:800,]

p_BEAM = plot_genes_branched_heatmap(STR.cds[row.names(subset(BEAM_res2,qval<1e-4)),],branch_point = 1,
                                     cores = 8,branch_colors = c("#979797", "#F05662", "#7990C8"),
                                     use_gene_short_name = T,hmcols = colorRampPalette(rev(brewer.pal(11, "Spectral")))(70),
                                     show_rownames = F,return_heatmap = T)






#对STR开始开始进行monocle[所有细胞，所有分组]-2-----
library(monocle)
library(Seurat)
library(SeuratDisk)
library(harmony)
library(dplyr)
library(tidyr)
library(ggplot2)
library(clusterProfiler)
library(org.Hs.eg.db)
library(RColorBrewer)
library(export)


STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")


STR$type0227=STR$type2

STR$type0227[STR$type0227%in%"IUA1"]<-"Adjacent"
STR$type0227[STR$type0227%in%"IUA2"]<-"Adhesion"

DimPlot(STR,group.by = "type2",label = T)

DimPlot(STR,group.by = "type0227",label = T)

DefaultAssay(STR)<-"RNA"

STR_matrix<-as(as.matrix(STR@assays$RNA@counts),'sparseMatrix')

STR_ann<-data.frame(gene_id=rownames(STR_matrix),gene_short_name=rownames(STR_matrix))

rownames(STR_ann)<-rownames(STR_matrix)

STR_fd<-new("AnnotatedDataFrame", data = STR_ann)


sample_ann<-STR@meta.data
rownames(sample_ann)<-colnames(STR_matrix)

STR_pd<-new("AnnotatedDataFrame", data =sample_ann)

STR.cds<-newCellDataSet(STR_matrix,phenoData =STR_pd,
                        featureData =STR_fd,expressionFamily=negbinomial.size())


#预处理
STR.cds <- estimateSizeFactors(STR.cds)
STR.cds <- estimateDispersions(STR.cds)

STR.cds <- detectGenes(STR.cds, min_expr = 0.1)

disp_table <- dispersionTable(STR.cds)

unsup_clustering_genes <- subset(disp_table, mean_expression >= 0.1)

STR.cds <- setOrderingFilter(STR.cds, unsup_clustering_genes$gene_id)


STR.cds<- reduceDimension(STR.cds, max_components = 2, num_dim = 8, 
                          reduction_method = 'tSNE', verbose = T)

STR.cds<- clusterCells(STR.cds, num_clusters = 8)


expressed_genes <- row.names(subset(fData(STR.cds),
                                    num_cells_expressed >= 10))

clustering_DEG_genes <- differentialGeneTest(STR.cds[expressed_genes,],
                                             fullModelFormulaStr = "~ cell_type0305",cores = 8)

ordering_genes <- row.names(clustering_DEG_genes)[order(clustering_DEG_genes$qval)][1:800] #排序后取前800个基因

STR.cds<- setOrderingFilter(STR.cds, ordering_genes)

#开始绘图用DDrtree 方法进行降维、画图
STR.cds <- reduceDimension(
  STR.cds,
  max_components = 2,
  method = 'DDRTree')

STR.cds <- orderCells(STR.cds)
head(pData(STR.cds))


STR_P1=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "type0227",label=T)+ggtitle("")

STR_P2=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "Pseudotime")

STR_P3=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "type0227",label=T)+ggtitle("")+
  facet_wrap(~type0227, nrow = 1)

STR_P3
STR_P1+STR_P2

STR.cds$cell_type0305

STR_P1=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "cell_type0305",label=T)+ggtitle("")

STR_P3=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "cell_type0305",label=T)+ggtitle("")+
  facet_wrap(~cell_type0305, nrow = 2)

saveRDS(STR.cds,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/文件存放点/STR_0303_2.rds")

#开始寻找分支热图的绘制
#节点1
BEAM_res <- BEAM(STR.cds, branch_point = 1,cores = 8) #第一个结点

BEAM_res <- BEAM_res[order(BEAM_res$qval),]

BEAM_res <- BEAM_res[,c("gene_short_name", "pval", "qval")]

mtGenes=grep('^MT-',rownames(STR),value=TRUE)
RPGenes=c(grep('^RPL',rownames(STR),value=TRUE),grep('^RPS',rownames(STR),value=TRUE))
stress_gene2=grep('^HSP',rownames(STR),value=TRUE)
HB_gene2=grep('^HB',rownames(STR),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
stress_gene1=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
               "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/gene1.csv")
subgene3=subgene3$cellcyle_gene
subgene4=c("FOS","CXCL2","ZFP36","FOSB","DUSP1","ATF3","CXCL8","NR4A1","CXCL3","PPP1R15A",
           "JUNB","EGR1","HSPA1A","HSPA1B","SOCS3","KLF6","JUN","IER2","CXCL1","NFKBIA",
           "HSPA6","DNAJB1","IER3","CCNL1","MTRNR2L2","IER4","ID1","CEBPD","KRT6A","CYR61",
           "DEPP1","CLDN4","IRF1","DUSP2","BTG2","PLAUR","MAFF","KLF4","PHLDA2","TNFAIP3")
subgene5=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/512个应激反应相关基因.csv")
subgene5=subgene5$gene_symbol

subgene=c(mtGenes,RPGenes,stress_gene2,HB_gene2,HB.genes,s.genes,g2m.genes,stress_gene1,"MKI67",subgene3,subgene4,subgene5)

BEAM_res=subset(BEAM_res,!gene_short_name%in%subgene)


write.csv(BEAM_res,
          "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_all_0305_2/BEAM_res_1.csv")

BEAM_res=
  read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR_all_0305_2/BEAM_res_1.csv",row.names = 1)




BEAM_res2=BEAM_res[1:800,]

p_BEAM = plot_genes_branched_heatmap(STR.cds[row.names(subset(BEAM_res2,qval<1e-4)),],branch_point = 1,
                                     cores = 8,branch_colors = c("#979797", "#F05662", "#7990C8"),
                                     use_gene_short_name = T,hmcols = colorRampPalette(rev(brewer.pal(11, "Spectral")))(70),
                                     show_rownames = F,return_heatmap = T)






#对STR进行velovyto.py-ALL----
###前期文件准备
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/velocyto/ALL-ALL/")

STR$typenew=STR$cell_type0305


DimPlot(STR,group.by = "typenew",label = T,label.size = 7)+ggtitle("")


cellID_humanpost10x=Cells(STR)
cellID_humanpost10x=as.data.frame(cellID_humanpost10x)
colnames(cellID_humanpost10x)[1]<-"x"

cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"]=paste("H:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"]=paste("qian:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"]=paste("IUAshou:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"]=paste("IUAxia:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"]=paste("li:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"]=paste("IUAfeng:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"]=paste("IUA-GWY_additional:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"]=paste("IUA-BYQ:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"]=paste("IUA-WJM:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"],sep = "")
cellID_humanpost10x$x=gsub("-1_1","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_2","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_3","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_4","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_5","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_6","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_7","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_8","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_9","x",cellID_humanpost10x$x)

write.csv(cellID_humanpost10x, file = "cellID_humanpost10x.csv", row.names = FALSE)
#And finally we can extract our clusters with:
celltype=STR$typenew
celltype=as.data.frame(celltype)
rownames(celltype)=cellID_humanpost10x$x
write.csv(celltype, file = "celltype.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(STR, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")



rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"]=paste(rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"],"-0",sep = "")
rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"]=paste(rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"],"-1",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"],"-2",sep = "")
rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"]=paste(rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"],"-3",sep = "")
rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"]=paste(rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"],"-4",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"],"-5",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"],"-6",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"],"-7",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"],"-8",sep = "")
write.csv(celltype, file = "celltype2.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(STR, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")

DimPlot(STR,label = T,group.by = "typenew")+ggtitle("")










#对STR进行velovyto.py-CTRL----
###前期文件准备
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

STR_CTRL=subset(STR,type2%in%"CTRL")

setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/velocyto/CTRL-ALL/")

STR$typenew=STR$cell_type0305


cellID_humanpost10x=Cells(STR_CTRL)
cellID_humanpost10x=as.data.frame(cellID_humanpost10x)
colnames(cellID_humanpost10x)[1]<-"x"
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"]=paste("H:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"]=paste("qian:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"],sep = "")



cellID_humanpost10x$x=gsub("-1_1","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_2","x",cellID_humanpost10x$x)


write.csv(cellID_humanpost10x, file = "cellID_humanpost10x.csv", row.names = FALSE)

#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(STR_CTRL, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")

#And finally we can extract our clusters with:
celltype=STR_CTRL$cell_type0305
celltype=as.data.frame(celltype)
rownames(celltype)=cellID_humanpost10x$x
write.csv(celltype, file = "celltype.csv")

rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"]=paste(rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"],"-0",sep = "")
rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"]=paste(rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"],"-1",sep = "")

write.csv(celltype, file = "celltype2.csv")


Embeddings=Embeddings(STR_CTRL, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")



#对STR进行velovyto.py-Adjacent-----
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")


STR_Adjacent=subset(STR,type2%in%"IUA1")

DimPlot(STR_Adjacent,group.by = "cell_type0305",label = T)


setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/velocyto/Adjacent-ALL/")

cellID_humanpost10x=Cells(STR_Adjacent)
cellID_humanpost10x=as.data.frame(cellID_humanpost10x)
colnames(cellID_humanpost10x)[1]<-"x"


cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"]=paste("IUAshou:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"]=paste("IUAxia:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"]=paste("li:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"]=paste("IUAfeng:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"],sep = "")

cellID_humanpost10x$x=gsub("-1_1","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_2","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_3","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_4","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_5","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_6","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_7","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_8","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_9","x",cellID_humanpost10x$x)

write.csv(cellID_humanpost10x, file = "cellID_humanpost10x.csv", row.names = FALSE)

#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(STR_Adjacent, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")

#And finally we can extract our clusters with:
celltype=STR_Adjacent$cell_type0305
celltype=as.data.frame(celltype)
rownames(celltype)=cellID_humanpost10x$x
write.csv(celltype, file = "celltype.csv")


rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"],"-0",sep = "")
rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"]=paste(rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"],"-1",sep = "")
rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"]=paste(rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"],"-2",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"],"-3",sep = "")


write.csv(celltype, file = "celltype2.csv")




Embeddings=Embeddings(STR_Adjacent, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")




###Adhesion----
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")


STR_Adhesion=subset(STR,type2%in%"IUA2")

DimPlot(STR_Adhesion,group.by = "cell_type0305",label = T)


setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/velocyto/Adhesion-ALL/")


cellID_humanpost10x=Cells(STR_Adhesion)
cellID_humanpost10x=as.data.frame(cellID_humanpost10x)
colnames(cellID_humanpost10x)[1]<-"x"



cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"]=paste("IUA-GWY_additional:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"]=paste("IUA-BYQ:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"],sep = "")

cellID_humanpost10x$x=gsub("-1_7","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_8","x",cellID_humanpost10x$x)


write.csv(cellID_humanpost10x, file = "cellID_humanpost10x.csv", row.names = FALSE)

#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(STR_Adhesion, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")

#And finally we can extract our clusters with:
celltype=STR_Adhesion$cell_type0305
celltype=as.data.frame(celltype)
rownames(celltype)=cellID_humanpost10x$x
write.csv(celltype, file = "celltype.csv")



rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"],"-0",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"],"-1",sep = "")

write.csv(celltype, file = "celltype2.csv")

#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(STR_Adhesion, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")



#对STR去除ALLgroups里面的不同cell_type来进行一个velocyto.py的准备-----
#S0----
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")



setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/velocyto/ALL_subtypes/S0/")

S0=subset(STR,cell_type0305%in%"S0")

S0$typenew=S0$type4


DimPlot(S0,group.by = "typenew",label = T,label.size = 7)+ggtitle("")


cellID_humanpost10x=Cells(S0)
cellID_humanpost10x=as.data.frame(cellID_humanpost10x)
colnames(cellID_humanpost10x)[1]<-"x"

cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"]=paste("H:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"]=paste("qian:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"]=paste("IUAshou:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"]=paste("IUAxia:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"]=paste("li:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"]=paste("IUAfeng:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"]=paste("IUA-GWY_additional:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"]=paste("IUA-BYQ:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"]=paste("IUA-WJM:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"],sep = "")
cellID_humanpost10x$x=gsub("-1_1","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_2","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_3","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_4","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_5","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_6","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_7","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_8","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_9","x",cellID_humanpost10x$x)

write.csv(cellID_humanpost10x, file = "cellID_humanpost10x.csv", row.names = FALSE)
#And finally we can extract our clusters with:
celltype=S0$typenew
celltype=as.data.frame(celltype)
rownames(celltype)=cellID_humanpost10x$x
write.csv(celltype, file = "celltype.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(S0, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")



rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"]=paste(rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"],"-0",sep = "")
rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"]=paste(rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"],"-1",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"],"-2",sep = "")
rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"]=paste(rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"],"-3",sep = "")
rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"]=paste(rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"],"-4",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"],"-5",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"],"-6",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"],"-7",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"],"-8",sep = "")
write.csv(celltype, file = "celltype2.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(S0, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")

DimPlot(S0,label = T,group.by = "typenew")+ggtitle("")


#S1----
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")



setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/velocyto/ALL_subtypes/S1/")

S1=subset(STR,cell_type0305%in%"S1")

S1$typenew=S1$type4


DimPlot(S1,group.by = "typenew",label = T,label.size = 7)+ggtitle("")


cellID_humanpost10x=Cells(S1)
cellID_humanpost10x=as.data.frame(cellID_humanpost10x)
colnames(cellID_humanpost10x)[1]<-"x"

cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"]=paste("H:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"]=paste("qian:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"]=paste("IUAshou:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"]=paste("IUAxia:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"]=paste("li:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"]=paste("IUAfeng:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"]=paste("IUA-GWY_additional:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"]=paste("IUA-BYQ:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"]=paste("IUA-WJM:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"],sep = "")
cellID_humanpost10x$x=gsub("-1_1","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_2","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_3","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_4","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_5","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_6","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_7","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_8","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_9","x",cellID_humanpost10x$x)

write.csv(cellID_humanpost10x, file = "cellID_humanpost10x.csv", row.names = FALSE)
#And finally we can extract our clusters with:
celltype=S1$typenew
celltype=as.data.frame(celltype)
rownames(celltype)=cellID_humanpost10x$x
write.csv(celltype, file = "celltype.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(S1, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")



rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"]=paste(rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"],"-0",sep = "")
rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"]=paste(rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"],"-1",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"],"-2",sep = "")
rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"]=paste(rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"],"-3",sep = "")
rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"]=paste(rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"],"-4",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"],"-5",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"],"-6",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"],"-7",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"],"-8",sep = "")
write.csv(celltype, file = "celltype2.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(S1, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")

DimPlot(S1,label = T,group.by = "typenew")+ggtitle("")















#BMP7----
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")



setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/velocyto/ALL_subtypes/BMP7/")

BMP7=subset(STR,cell_type0305%in%"BMP7")

BMP7$typenew=BMP7$type4


DimPlot(BMP7,group.by = "typenew",label = T,label.size = 7)+ggtitle("")


cellID_humanpost10x=Cells(BMP7)
cellID_humanpost10x=as.data.frame(cellID_humanpost10x)
colnames(cellID_humanpost10x)[1]<-"x"

cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"]=paste("H:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"]=paste("qian:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"]=paste("IUAshou:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"]=paste("IUAxia:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"]=paste("li:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"]=paste("IUAfeng:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"]=paste("IUA-GWY_additional:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"]=paste("IUA-BYQ:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"]=paste("IUA-WJM:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"],sep = "")
cellID_humanpost10x$x=gsub("-1_1","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_2","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_3","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_4","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_5","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_6","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_7","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_8","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_9","x",cellID_humanpost10x$x)

write.csv(cellID_humanpost10x, file = "cellID_humanpost10x.csv", row.names = FALSE)
#And finally we can extract our clusters with:
celltype=BMP7$typenew
celltype=as.data.frame(celltype)
rownames(celltype)=cellID_humanpost10x$x
write.csv(celltype, file = "celltype.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(BMP7, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")



rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"]=paste(rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"],"-0",sep = "")
rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"]=paste(rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"],"-1",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"],"-2",sep = "")
rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"]=paste(rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"],"-3",sep = "")
rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"]=paste(rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"],"-4",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"],"-5",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"],"-6",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"],"-7",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"],"-8",sep = "")
write.csv(celltype, file = "celltype2.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(BMP7, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")

DimPlot(BMP7,label = T,group.by = "typenew")+ggtitle("")















#REV3L----
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")



setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/velocyto/ALL_subtypes/REV3L/")

REV3L=subset(STR,cell_type0305%in%"REV3L")

REV3L$typenew=REV3L$type4


DimPlot(REV3L,group.by = "typenew",label = T,label.size = 7)+ggtitle("")


cellID_humanpost10x=Cells(REV3L)
cellID_humanpost10x=as.data.frame(cellID_humanpost10x)
colnames(cellID_humanpost10x)[1]<-"x"

cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"]=paste("H:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"]=paste("qian:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"]=paste("IUAshou:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"]=paste("IUAxia:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"]=paste("li:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"]=paste("IUAfeng:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"]=paste("IUA-GWY_additional:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"]=paste("IUA-BYQ:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"]=paste("IUA-WJM:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"],sep = "")
cellID_humanpost10x$x=gsub("-1_1","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_2","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_3","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_4","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_5","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_6","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_7","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_8","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_9","x",cellID_humanpost10x$x)

write.csv(cellID_humanpost10x, file = "cellID_humanpost10x.csv", row.names = FALSE)
#And finally we can extract our clusters with:
celltype=REV3L$typenew
celltype=as.data.frame(celltype)
rownames(celltype)=cellID_humanpost10x$x
write.csv(celltype, file = "celltype.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(REV3L, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")



rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"]=paste(rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"],"-0",sep = "")
rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"]=paste(rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"],"-1",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"],"-2",sep = "")
rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"]=paste(rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"],"-3",sep = "")
rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"]=paste(rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"],"-4",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"],"-5",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"],"-6",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"],"-7",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"],"-8",sep = "")
write.csv(celltype, file = "celltype2.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(REV3L, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")

DimPlot(REV3L,label = T,group.by = "typenew")+ggtitle("")















#ISG15----
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")



setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/velocyto/ALL_subtypes/ISG15/")

ISG15=subset(STR,cell_type0305%in%"ISG15")

ISG15$typenew=ISG15$type4


DimPlot(ISG15,group.by = "typenew",label = T,label.size = 7)+ggtitle("")


cellID_humanpost10x=Cells(ISG15)
cellID_humanpost10x=as.data.frame(cellID_humanpost10x)
colnames(cellID_humanpost10x)[1]<-"x"

cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"]=paste("H:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"]=paste("qian:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"]=paste("IUAshou:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"]=paste("IUAxia:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"]=paste("li:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"]=paste("IUAfeng:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"]=paste("IUA-GWY_additional:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"]=paste("IUA-BYQ:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"]=paste("IUA-WJM:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"],sep = "")
cellID_humanpost10x$x=gsub("-1_1","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_2","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_3","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_4","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_5","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_6","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_7","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_8","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_9","x",cellID_humanpost10x$x)

write.csv(cellID_humanpost10x, file = "cellID_humanpost10x.csv", row.names = FALSE)
#And finally we can extract our clusters with:
celltype=ISG15$typenew
celltype=as.data.frame(celltype)
rownames(celltype)=cellID_humanpost10x$x
write.csv(celltype, file = "celltype.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(ISG15, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")



rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"]=paste(rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"],"-0",sep = "")
rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"]=paste(rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"],"-1",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"],"-2",sep = "")
rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"]=paste(rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"],"-3",sep = "")
rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"]=paste(rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"],"-4",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"],"-5",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"],"-6",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"],"-7",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"],"-8",sep = "")
write.csv(celltype, file = "celltype2.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(ISG15, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")

DimPlot(ISG15,label = T,group.by = "typenew")+ggtitle("")





#APOD----
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")



setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/velocyto/ALL_subtypes/APOD/")

APOD=subset(STR,cell_type0305%in%"APOD")

APOD$typenew=APOD$type4


DimPlot(APOD,group.by = "typenew",label = T,label.size = 7)+ggtitle("")


cellID_humanpost10x=Cells(APOD)
cellID_humanpost10x=as.data.frame(cellID_humanpost10x)
colnames(cellID_humanpost10x)[1]<-"x"

cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"]=paste("H:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"]=paste("qian:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"]=paste("IUAshou:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"]=paste("IUAxia:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"]=paste("li:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"]=paste("IUAfeng:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"]=paste("IUA-GWY_additional:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"]=paste("IUA-BYQ:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"]=paste("IUA-WJM:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"],sep = "")
cellID_humanpost10x$x=gsub("-1_1","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_2","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_3","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_4","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_5","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_6","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_7","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_8","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_9","x",cellID_humanpost10x$x)

write.csv(cellID_humanpost10x, file = "cellID_humanpost10x.csv", row.names = FALSE)
#And finally we can extract our clusters with:
celltype=APOD$typenew
celltype=as.data.frame(celltype)
rownames(celltype)=cellID_humanpost10x$x
write.csv(celltype, file = "celltype.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(APOD, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")



rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"]=paste(rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"],"-0",sep = "")
rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"]=paste(rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"],"-1",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"],"-2",sep = "")
rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"]=paste(rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"],"-3",sep = "")
rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"]=paste(rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"],"-4",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"],"-5",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"],"-6",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"],"-7",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"],"-8",sep = "")
write.csv(celltype, file = "celltype2.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(APOD, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")

DimPlot(APOD,label = T,group.by = "typenew")+ggtitle("")





#Cycling----
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")



setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/velocyto/ALL_subtypes/Cycling/")

Cycling=subset(STR,cell_type0305%in%"Cycling")

Cycling$typenew=Cycling$type4


DimPlot(Cycling,group.by = "typenew",label = T,label.size = 7)+ggtitle("")


cellID_humanpost10x=Cells(Cycling)
cellID_humanpost10x=as.data.frame(cellID_humanpost10x)
colnames(cellID_humanpost10x)[1]<-"x"

cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"]=paste("H:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"]=paste("qian:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"]=paste("IUAshou:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"]=paste("IUAxia:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"]=paste("li:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"]=paste("IUAfeng:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"]=paste("IUA-GWY_additional:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"]=paste("IUA-BYQ:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"]=paste("IUA-WJM:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"],sep = "")
cellID_humanpost10x$x=gsub("-1_1","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_2","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_3","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_4","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_5","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_6","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_7","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_8","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_9","x",cellID_humanpost10x$x)

write.csv(cellID_humanpost10x, file = "cellID_humanpost10x.csv", row.names = FALSE)
#And finally we can extract our clusters with:
celltype=Cycling$typenew
celltype=as.data.frame(celltype)
rownames(celltype)=cellID_humanpost10x$x
write.csv(celltype, file = "celltype.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(Cycling, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")



rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"]=paste(rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"],"-0",sep = "")
rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"]=paste(rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"],"-1",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"],"-2",sep = "")
rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"]=paste(rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"],"-3",sep = "")
rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"]=paste(rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"],"-4",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"],"-5",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"],"-6",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"],"-7",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"],"-8",sep = "")
write.csv(celltype, file = "celltype2.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(Cycling, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")

DimPlot(Cycling,label = T,group.by = "typenew")+ggtitle("")





#P----
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")



setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/velocyto/ALL_subtypes/P/")

P=subset(STR,cell_type0305%in%"P")

P$typenew=P$type4


DimPlot(P,group.by = "typenew",label = T,label.size = 7)+ggtitle("")


cellID_humanpost10x=Cells(P)
cellID_humanpost10x=as.data.frame(cellID_humanpost10x)
colnames(cellID_humanpost10x)[1]<-"x"

cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"]=paste("H:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"]=paste("qian:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"]=paste("IUAshou:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"]=paste("IUAxia:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"]=paste("li:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"]=paste("IUAfeng:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"]=paste("IUA-GWY_additional:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"]=paste("IUA-BYQ:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"]=paste("IUA-WJM:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"],sep = "")
cellID_humanpost10x$x=gsub("-1_1","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_2","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_3","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_4","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_5","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_6","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_7","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_8","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_9","x",cellID_humanpost10x$x)

write.csv(cellID_humanpost10x, file = "cellID_humanpost10x.csv", row.names = FALSE)
#And finally we can extract our clusters with:
celltype=P$typenew
celltype=as.data.frame(celltype)
rownames(celltype)=cellID_humanpost10x$x
write.csv(celltype, file = "celltype.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(P, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")



rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"]=paste(rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"],"-0",sep = "")
rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"]=paste(rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"],"-1",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"],"-2",sep = "")
rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"]=paste(rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"],"-3",sep = "")
rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"]=paste(rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"],"-4",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"],"-5",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"],"-6",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"],"-7",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"],"-8",sep = "")
write.csv(celltype, file = "celltype2.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(P, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")

DimPlot(P,label = T,group.by = "typenew")+ggtitle("")





#STR----
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")



setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/velocyto/ALL_subtypes/ALL/")

STR$typenew=STR$type4


DimPlot(STR,group.by = "typenew",label = T,label.size = 7)+ggtitle("")


cellID_humanpost10x=Cells(STR)
cellID_humanpost10x=as.data.frame(cellID_humanpost10x)
colnames(cellID_humanpost10x)[1]<-"x"

cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"]=paste("H:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"]=paste("qian:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"]=paste("IUAshou:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"]=paste("IUAxia:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"]=paste("li:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"]=paste("IUAfeng:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"]=paste("IUA-GWY_additional:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"]=paste("IUA-BYQ:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"]=paste("IUA-WJM:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"],sep = "")
cellID_humanpost10x$x=gsub("-1_1","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_2","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_3","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_4","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_5","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_6","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_7","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_8","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_9","x",cellID_humanpost10x$x)

write.csv(cellID_humanpost10x, file = "cellID_humanpost10x.csv", row.names = FALSE)
#And finally we can extract our clusters with:
celltype=STR$typenew
celltype=as.data.frame(celltype)
rownames(celltype)=cellID_humanpost10x$x
write.csv(celltype, file = "celltype.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(STR, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")



rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"]=paste(rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"],"-0",sep = "")
rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"]=paste(rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"],"-1",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"],"-2",sep = "")
rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"]=paste(rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"],"-3",sep = "")
rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"]=paste(rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"],"-4",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"],"-5",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"],"-6",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"],"-7",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"],"-8",sep = "")
write.csv(celltype, file = "celltype2.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(STR, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")

DimPlot(STR,label = T,group.by = "typenew")+ggtitle("")





###REV3L和S1----
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")



setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/velocyto/ALL_subtypes/REV3L_S1/")

STR0313=subset(STR,cell_type0305%in%c("REV3L","S1"))

STR0313$typenew=STR0313$cell_type0305


DimPlot(STR0313,group.by = "typenew",label = T,label.size = 7)+ggtitle("")


cellID_humanpost10x=Cells(STR0313)
cellID_humanpost10x=as.data.frame(cellID_humanpost10x)
colnames(cellID_humanpost10x)[1]<-"x"

cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"]=paste("H:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"]=paste("qian:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"]=paste("IUAshou:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"]=paste("IUAxia:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"]=paste("li:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"]=paste("IUAfeng:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"]=paste("IUA-GWY_additional:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"]=paste("IUA-BYQ:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"]=paste("IUA-WJM:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"],sep = "")
cellID_humanpost10x$x=gsub("-1_1","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_2","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_3","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_4","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_5","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_6","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_7","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_8","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_9","x",cellID_humanpost10x$x)

write.csv(cellID_humanpost10x, file = "cellID_humanpost10x.csv", row.names = FALSE)
#And finally we can extract our clusters with:
celltype=STR0313$typenew
celltype=as.data.frame(celltype)
rownames(celltype)=cellID_humanpost10x$x
write.csv(celltype, file = "celltype.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(STR0313, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")



rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"]=paste(rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"],"-0",sep = "")
rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"]=paste(rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"],"-1",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"],"-2",sep = "")
rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"]=paste(rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"],"-3",sep = "")
rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"]=paste(rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"],"-4",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"],"-5",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"],"-6",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"],"-7",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"],"-8",sep = "")
write.csv(celltype, file = "celltype2.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(STR0313, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")

DimPlot(STR0313,label = T,group.by = "typenew")+ggtitle("")



###Cycling和S0----
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")



setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/velocyto/ALL_subtypes/Cycling_S0/")

STR0313=subset(STR,cell_type0305%in%c("Cycling","S0"))

STR0313$typenew=STR0313$cell_type0305


DimPlot(STR0313,group.by = "typenew",label = T,label.size = 7)+ggtitle("")


cellID_humanpost10x=Cells(STR0313)
cellID_humanpost10x=as.data.frame(cellID_humanpost10x)
colnames(cellID_humanpost10x)[1]<-"x"

cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"]=paste("H:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"]=paste("qian:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"]=paste("IUAshou:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"]=paste("IUAxia:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"]=paste("li:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"]=paste("IUAfeng:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"]=paste("IUA-GWY_additional:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"]=paste("IUA-BYQ:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"]=paste("IUA-WJM:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"],sep = "")
cellID_humanpost10x$x=gsub("-1_1","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_2","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_3","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_4","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_5","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_6","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_7","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_8","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_9","x",cellID_humanpost10x$x)

write.csv(cellID_humanpost10x, file = "cellID_humanpost10x.csv", row.names = FALSE)
#And finally we can extract our clusters with:
celltype=STR0313$typenew
celltype=as.data.frame(celltype)
rownames(celltype)=cellID_humanpost10x$x
write.csv(celltype, file = "celltype.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(STR0313, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")



rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"]=paste(rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"],"-0",sep = "")
rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"]=paste(rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"],"-1",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"],"-2",sep = "")
rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"]=paste(rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"],"-3",sep = "")
rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"]=paste(rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"],"-4",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"],"-5",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"],"-6",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"],"-7",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"],"-8",sep = "")
write.csv(celltype, file = "celltype2.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(STR0313, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")

DimPlot(STR0313,label = T,group.by = "typenew")+ggtitle("")



#查看不同细胞类型间的纤维化相关marker的表达情况-----
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")
gene_0314_1=c("TAGLN","IGFBP7","IGFBP5","IGFBP3","MYL9","SPP1","MMP9")
VlnPlot(STR,features = gene_0314_1,group.by = "cell_type0305",pt.size = 0,ncol = 7)

gene_0314_2=c("LUM","CD24","WNT5A","FN1","IGF1","PDGFRB","TNFRSF12A")
VlnPlot(STR,features = gene_0314_2,group.by = "cell_type0305",pt.size = 0,ncol = 7)

gene_0314_3=c("VCAN","CD44","NAMPT","LOX","HMGB1","PTGS2","TGFB1")
VlnPlot(STR,features = gene_0314_3,group.by = "cell_type0305",pt.size = 0,ncol = 7)

gene_0314_4=c("CD163","MARCO","SMAD1","SMAD2","SMAD3","SMAD5","SMAD8","SMAD9")
VlnPlot(STR,features = gene_0314_4,group.by = "cell_type0305",pt.size = 0,ncol = 8)

gene_0314_3=c("TGFB1","TGFB2","TGFB3","TGFB4")
VlnPlot(STR,features = gene_0314_3,group.by = "cell_type0305",pt.size = 0,ncol = 4)

#将STR中各个亚型的kobas的marker基因富集结果进行绘图并统一观察GO富集-----
#挑前8富集通路进行绘图-气泡图---unique的通路
S0_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/S0_ego.csv")
S1_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/S1_ego.csv")
BMP7_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/BMP7_ego.csv")
Cycling_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/Cycling_ego.csv")
REV3L_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/REV3L_ego.csv")
ISG15_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/ISG15_ego.csv")
APOD_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/APOD_ego.csv")
P_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/P_ego.csv")

S0_ego$cluster="S0"
S1_ego$cluster="S1"
BMP7_ego$cluster="BMP7"
Cycling_ego$cluster="Cycling"
REV3L_ego$cluster="REV3L"
ISG15_ego$cluster="ISG15"
APOD_ego$cluster="APOD"
P_ego$cluster="P"

S0_ego=S0_ego[order(S0_ego$P.Value),]
S1_ego=S1_ego[order(S1_ego$P.Value),]
BMP7_ego=BMP7_ego[order(BMP7_ego$P.Value),]
Cycling_ego=Cycling_ego[order(Cycling_ego$P.Value),]
REV3L_ego=REV3L_ego[order(REV3L_ego$P.Value),]
ISG15_ego=ISG15_ego[order(ISG15_ego$P.Value),]
APOD_ego=APOD_ego[order(APOD_ego$P.Value),]
P_ego=P_ego[order(P_ego$P.Value),]

ego_all=rbind(S0_ego,S1_ego,BMP7_ego,Cycling_ego,REV3L_ego,
              ISG15_ego,APOD_ego,P_ego)

Z=ego_all[duplicated(ego_all$X.Term),]

N=subset(ego_all,!X.Term%in%unique(Z$X.Term))

table(duplicated(N$X.Term))

S0_ego=subset(N,cluster%in%"S0")
S1_ego=subset(N,cluster%in%"S1")
BMP7_ego=subset(N,cluster%in%"BMP7")
Cycling_ego=subset(N,cluster%in%"Cycling")
REV3L_ego=subset(N,cluster%in%"REV3L")
ISG15_ego=subset(N,cluster%in%"ISG15")
APOD_ego=subset(N,cluster%in%"APOD")
P_ego=subset(N,cluster%in%"P")

S0_des=S0_ego[1:10,]$X.Term
S1_des=S1_ego[1:10,]$X.Term
BMP7_des=BMP7_ego[1:10,]$X.Term
Cycling_des=Cycling_ego[1:10,]$X.Term
REV3L_des=REV3L_ego[1:10,]$X.Term
ISG15_des=ISG15_ego[1:10,]$X.Term
APOD_des=APOD_ego[1:10,]$X.Term
P_des=P_ego[1:10,]$X.Term



S0_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/S0_ego.csv")
S1_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/S1_ego.csv")
BMP7_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/BMP7_ego.csv")
Cycling_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/Cycling_ego.csv")
REV3L_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/REV3L_ego.csv")
ISG15_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/ISG15_ego.csv")
APOD_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/APOD_ego.csv")
P_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/P_ego.csv")

S0_ego$cluster="S0"
S1_ego$cluster="S1"
BMP7_ego$cluster="BMP7"
Cycling_ego$cluster="Cycling"
REV3L_ego$cluster="REV3L"
ISG15_ego$cluster="ISG15"
APOD_ego$cluster="APOD"
P_ego$cluster="P"


S0_ego=S0_ego[order(S0_ego$P.Value),]
S1_ego=S1_ego[order(S1_ego$P.Value),]
BMP7_ego=BMP7_ego[order(BMP7_ego$P.Value),]
Cycling_ego=Cycling_ego[order(Cycling_ego$P.Value),]
REV3L_ego=REV3L_ego[order(REV3L_ego$P.Value),]
ISG15_ego=ISG15_ego[order(ISG15_ego$P.Value),]
APOD_ego=APOD_ego[order(APOD_ego$P.Value),]
P_ego=P_ego[order(P_ego$P.Value),]


ego_all=rbind(S0_ego,S1_ego,BMP7_ego,Cycling_ego,REV3L_ego,
              ISG15_ego,APOD_ego,P_ego)

ego_all=subset(ego_all,X.Term%in%c(S0_des,S1_des,BMP7_des,Cycling_des,REV3L_des,
                                   ISG15_des,APOD_des,P_des))



ego_all$cluster=factor(ego_all$cluster,levels = c("S0","S1","BMP7","Cycling","REV3L","ISG15","APOD","P"))


ggplot(data=ego_all)+aes(x=cluster,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(ego_all$X.Term))+theme_bw()+xlab("Cluster")+
  scale_color_gradient(low="blue",high = "red")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+xlab("")+
  ylab("")




#挑前8富集通路进行绘图-气泡图---非unique的通路
S0_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/S0_ego.csv")
S1_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/S1_ego.csv")
BMP7_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/BMP7_ego.csv")
Cycling_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/Cycling_ego.csv")
REV3L_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/REV3L_ego.csv")
ISG15_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/ISG15_ego.csv")
APOD_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/APOD_ego.csv")
P_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/P_ego.csv")

S0_ego$cluster="S0"
S1_ego$cluster="S1"
BMP7_ego$cluster="BMP7"
Cycling_ego$cluster="Cycling"
REV3L_ego$cluster="REV3L"
ISG15_ego$cluster="ISG15"
APOD_ego$cluster="APOD"
P_ego$cluster="P"

S0_ego=S0_ego[order(S0_ego$P.Value),]
S1_ego=S1_ego[order(S1_ego$P.Value),]
BMP7_ego=BMP7_ego[order(BMP7_ego$P.Value),]
Cycling_ego=Cycling_ego[order(Cycling_ego$P.Value),]
REV3L_ego=REV3L_ego[order(REV3L_ego$P.Value),]
ISG15_ego=ISG15_ego[order(ISG15_ego$P.Value),]
APOD_ego=APOD_ego[order(APOD_ego$P.Value),]
P_ego=P_ego[order(P_ego$P.Value),]





S0_des=S0_ego[1:10,]$X.Term
S1_des=S1_ego[1:10,]$X.Term
BMP7_des=BMP7_ego[1:10,]$X.Term
Cycling_des=Cycling_ego[1:10,]$X.Term
REV3L_des=REV3L_ego[1:10,]$X.Term
ISG15_des=ISG15_ego[1:10,]$X.Term
APOD_des=APOD_ego[1:10,]$X.Term
P_des=P_ego[1:10,]$X.Term

ego_all=rbind(S0_ego,S1_ego,BMP7_ego,Cycling_ego,REV3L_ego,
              ISG15_ego,APOD_ego,P_ego)

ego_all=subset(ego_all,X.Term%in%c(S0_des,S1_des,BMP7_des,Cycling_des,REV3L_des,
                                   ISG15_des,APOD_des,P_des))



ego_all$cluster=factor(ego_all$cluster,levels = c("S0","S1","BMP7","Cycling","REV3L","ISG15","APOD","P"))
ego_all=subset(ego_all,ego_all$P.Value<0.05)

ggplot(data=ego_all)+aes(x=cluster,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(ego_all$X.Term))+theme_bw()+xlab("Cluster")+
  scale_color_gradient(low="blue",high = "red")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+xlab("")+
  ylab("")






#将STR中各个亚型的kobas的marker基因富集结果进行绘图并统一观察KEGG富集-----
#挑前8富集通路进行绘图-气泡图---unique的通路
S0_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/S0_kegg.csv")
S1_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/S1_kegg.csv")
BMP7_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/BMP7_kegg.csv")
Cycling_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/Cycling_kegg.csv")
REV3L_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/REV3L_kegg.csv")
ISG15_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/ISG15_kegg.csv")
APOD_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/APOD_kegg.csv")
P_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/P_kegg.csv")

S0_kegg=na.omit(S0_kegg)
S1_kegg=na.omit(S1_kegg)
BMP7_kegg=na.omit(BMP7_kegg)
Cycling_kegg=na.omit(Cycling_kegg)
REV3L_kegg=na.omit(REV3L_kegg)
ISG15_kegg=na.omit(ISG15_kegg)
APOD_kegg=na.omit(APOD_kegg)
P_kegg=na.omit(P_kegg)




S0_kegg$cluster="S0"
S1_kegg$cluster="S1"
BMP7_kegg$cluster="BMP7"
Cycling_kegg$cluster="Cycling"
REV3L_kegg$cluster="REV3L"
ISG15_kegg$cluster="ISG15"
APOD_kegg$cluster="APOD"
P_kegg$cluster="P"

S0_kegg=S0_kegg[order(S0_kegg$P.Value),]
S1_kegg=S1_kegg[order(S1_kegg$P.Value),]
BMP7_kegg=BMP7_kegg[order(BMP7_kegg$P.Value),]
Cycling_kegg=Cycling_kegg[order(Cycling_kegg$P.Value),]
REV3L_kegg=REV3L_kegg[order(REV3L_kegg$P.Value),]
ISG15_kegg=ISG15_kegg[order(ISG15_kegg$P.Value),]
APOD_kegg=APOD_kegg[order(APOD_kegg$P.Value),]
P_kegg=P_kegg[order(P_kegg$P.Value),]

kegg_all=rbind(S0_kegg,S1_kegg,BMP7_kegg,Cycling_kegg,REV3L_kegg,
               ISG15_kegg,APOD_kegg,P_kegg)

Z=kegg_all[duplicated(kegg_all$X.Term),]

N=subset(kegg_all,!X.Term%in%unique(Z$X.Term))

table(duplicated(N$X.Term))

S0_kegg=subset(N,cluster%in%"S0")
S1_kegg=subset(N,cluster%in%"S1")
BMP7_kegg=subset(N,cluster%in%"BMP7")
Cycling_kegg=subset(N,cluster%in%"Cycling")
REV3L_kegg=subset(N,cluster%in%"REV3L")
ISG15_kegg=subset(N,cluster%in%"ISG15")
APOD_kegg=subset(N,cluster%in%"APOD")
P_kegg=subset(N,cluster%in%"P")

S0_des=S0_kegg[1:10,]$X.Term
S1_des=S1_kegg[1:10,]$X.Term
BMP7_des=BMP7_kegg[1:10,]$X.Term
Cycling_des=Cycling_kegg[1:10,]$X.Term
REV3L_des=REV3L_kegg[1:10,]$X.Term
ISG15_des=ISG15_kegg[1:10,]$X.Term
APOD_des=APOD_kegg[1:10,]$X.Term
P_des=P_kegg[1:10,]$X.Term



S0_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/S0_kegg.csv")
S1_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/S1_kegg.csv")
BMP7_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/BMP7_kegg.csv")
Cycling_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/Cycling_kegg.csv")
REV3L_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/REV3L_kegg.csv")
ISG15_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/ISG15_kegg.csv")
APOD_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/APOD_kegg.csv")
P_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/P_kegg.csv")

S0_kegg=na.omit(S0_kegg)
S1_kegg=na.omit(S1_kegg)
BMP7_kegg=na.omit(BMP7_kegg)
Cycling_kegg=na.omit(Cycling_kegg)
REV3L_kegg=na.omit(REV3L_kegg)
ISG15_kegg=na.omit(ISG15_kegg)
APOD_kegg=na.omit(APOD_kegg)
P_kegg=na.omit(P_kegg)



S0_kegg$cluster="S0"
S1_kegg$cluster="S1"
BMP7_kegg$cluster="BMP7"
Cycling_kegg$cluster="Cycling"
REV3L_kegg$cluster="REV3L"
ISG15_kegg$cluster="ISG15"
APOD_kegg$cluster="APOD"
P_kegg$cluster="P"


S0_kegg=S0_kegg[order(S0_kegg$P.Value),]
S1_kegg=S1_kegg[order(S1_kegg$P.Value),]
BMP7_kegg=BMP7_kegg[order(BMP7_kegg$P.Value),]
Cycling_kegg=Cycling_kegg[order(Cycling_kegg$P.Value),]
REV3L_kegg=REV3L_kegg[order(REV3L_kegg$P.Value),]
ISG15_kegg=ISG15_kegg[order(ISG15_kegg$P.Value),]
APOD_kegg=APOD_kegg[order(APOD_kegg$P.Value),]
P_kegg=P_kegg[order(P_kegg$P.Value),]


kegg_all=rbind(S0_kegg,S1_kegg,BMP7_kegg,Cycling_kegg,REV3L_kegg,
               ISG15_kegg,APOD_kegg,P_kegg)

kegg_all=subset(kegg_all,X.Term%in%c(S0_des,S1_des,BMP7_des,Cycling_des,REV3L_des,
                                     ISG15_des,APOD_des,P_des))



kegg_all$cluster=factor(kegg_all$cluster,levels = c("S0","S1","BMP7","Cycling","REV3L","ISG15","APOD","P"))


ggplot(data=kegg_all)+aes(x=cluster,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(kegg_all$X.Term))+theme_bw()+xlab("Cluster")+
  scale_color_gradient(low="blue",high = "red")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+xlab("")+
  ylab("")




#挑前8富集通路进行绘图-气泡图---非unique的通路
S0_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/S0_kegg.csv")
S1_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/S1_kegg.csv")
BMP7_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/BMP7_kegg.csv")
Cycling_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/Cycling_kegg.csv")
REV3L_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/REV3L_kegg.csv")
ISG15_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/ISG15_kegg.csv")
APOD_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/APOD_kegg.csv")
P_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/P_kegg.csv")

S0_kegg$cluster="S0"
S1_kegg$cluster="S1"
BMP7_kegg$cluster="BMP7"
Cycling_kegg$cluster="Cycling"
REV3L_kegg$cluster="REV3L"
ISG15_kegg$cluster="ISG15"
APOD_kegg$cluster="APOD"
P_kegg$cluster="P"

S0_kegg=S0_kegg[order(S0_kegg$P.Value),]
S1_kegg=S1_kegg[order(S1_kegg$P.Value),]
BMP7_kegg=BMP7_kegg[order(BMP7_kegg$P.Value),]
Cycling_kegg=Cycling_kegg[order(Cycling_kegg$P.Value),]
REV3L_kegg=REV3L_kegg[order(REV3L_kegg$P.Value),]
ISG15_kegg=ISG15_kegg[order(ISG15_kegg$P.Value),]
APOD_kegg=APOD_kegg[order(APOD_kegg$P.Value),]
P_kegg=P_kegg[order(P_kegg$P.Value),]





S0_des=S0_kegg[1:10,]$X.Term
S1_des=S1_kegg[1:10,]$X.Term
BMP7_des=BMP7_kegg[1:10,]$X.Term
Cycling_des=Cycling_kegg[1:10,]$X.Term
REV3L_des=REV3L_kegg[1:10,]$X.Term
ISG15_des=ISG15_kegg[1:10,]$X.Term
APOD_des=APOD_kegg[1:10,]$X.Term
P_des=P_kegg[1:10,]$X.Term

kegg_all=rbind(S0_kegg,S1_kegg,BMP7_kegg,Cycling_kegg,REV3L_kegg,
               ISG15_kegg,APOD_kegg,P_kegg)

kegg_all=subset(kegg_all,X.Term%in%c(S0_des,S1_des,BMP7_des,Cycling_des,REV3L_des,
                                     ISG15_des,APOD_des,P_des))



kegg_all$cluster=factor(kegg_all$cluster,levels = c("S0","S1","BMP7","Cycling","REV3L","ISG15","APOD","P"))


ggplot(data=kegg_all)+aes(x=cluster,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(kegg_all$X.Term))+theme_bw()+xlab("Cluster")+
  scale_color_gradient(low="blue",high = "red")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+xlab("")+
  ylab("")






#对Adhesion组中的不同细胞亚型进行marker基因查找以及富集工作-----
library(Seurat)
library(ggplot2)

STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")
STR_adhesion=subset(STR,type4%in%"Adhesion")
DimPlot(STR_adhesion,label = T,label.size = 12,group.by = "cell_type0305",pt.size = 1)+ggtitle("")+
  scale_color_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"))

DefaultAssay(STR_adhesion)<-"RNA"
STR_adhesion@active.ident=STR_adhesion$cell_type0305
STR_adhesion_markers <- FindAllMarkers(object=STR_adhesion, only.pos = TRUE,min.pct = 0.25, 
                                       logfc.threshold = 0.25)
STR_adhesion_markers <-subset(STR_adhesion_markers,STR_adhesion_markers$p_val<0.05)


write.csv(STR_adhesion_markers,
          "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/Adhesion/cell_type0305marker基因查找/STR_markers.csv")



mtGenes=grep('^MT-',rownames(STR_adhesion),value=TRUE)
RPGenes=c(grep('^RPL',rownames(STR_adhesion),value=TRUE),grep('^RPS',rownames(STR_adhesion),value=TRUE))
STR_adhesioness_gene2=grep('^HSP',rownames(STR_adhesion),value=TRUE)
HB_gene2=grep('^HB',rownames(STR_adhesion),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
STR_adhesioness_gene1=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
                        "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/gene1.csv")
subgene3=subgene3$cellcyle_gene
subgene4=c("FOS","CXCL2","ZFP36","FOSB","DUSP1","ATF3","CXCL8","NR4A1","CXCL3","PPP1R15A",
           "JUNB","EGR1","HSPA1A","HSPA1B","SOCS3","KLF6","JUN","IER2","CXCL1","NFKBIA",
           "HSPA6","DNAJB1","IER3","CCNL1","MTRNR2L2","IER4","ID1","CEBPD","KRT6A","CYR61",
           "DEPP1","CLDN4","IRF1","DUSP2","BTG2","PLAUR","MAFF","KLF4","PHLDA2","TNFAIP3")
subgene5=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/512个应激反应相关基因.csv")
subgene5=subgene5$gene_symbol

subgene=c(mtGenes,RPGenes,STR_adhesioness_gene2,HB_gene2,HB.genes,s.genes,g2m.genes,STR_adhesioness_gene1,"MKI67",subgene3,subgene4,subgene5)


#包含非特异的marker
STR_adhesion_markers=subset(STR_adhesion_markers,!gene%in%subgene)

STR_adhesion_markers_top10 <- STR_adhesion_markers %>% group_by(cluster) %>% top_n(n = 10, wt = avg_log2FC)

DefaultAssay(STR_adhesion)<-"RNA"

STR_adhesion_markers_top10=unique(STR_adhesion_markers_top10$gene)

DotPlot(STR_adhesion,features=STR_adhesion_markers_top10,group.by = "cell_type0305")+
  theme(axis.text.x= element_text(size=9,angle = 45,vjust = 1,hjust = 1,face= "bold"))+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])))+xlab("")+ylab("")



#特异的marker
Z=STR_adhesion_markers[duplicated(STR_adhesion_markers$gene),]

N=subset(STR_adhesion_markers,!gene%in%unique(Z$gene))

table(duplicated(N$gene))

STR_adhesion_markers_top10 <- N %>% group_by(cluster) %>% top_n(n = 10, wt = avg_log2FC)

DefaultAssay(STR_adhesion)<-"RNA"

STR_adhesion_markers_top10=unique(STR_adhesion_markers_top10$gene)

DotPlot(STR_adhesion,features=STR_adhesion_markers_top10,group.by = "cell_type0305")+
  theme(axis.text.x= element_text(size=9,angle = 45,vjust = 1,hjust = 1,face= "bold"))+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])))+xlab("")+ylab("")




#将Adhesion中各个亚型的kobas的marker基因富集结果进行绘图并统一观察GO富集-----
#挑前8富集通路进行绘图-气泡图---unique的通路
S0_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/Adhesion/cell_type0305marker基因查找/S0_ego.csv")
S1_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/Adhesion/cell_type0305marker基因查找/S1_ego.csv")
BMP7_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/Adhesion/cell_type0305marker基因查找/BMP7_ego.csv")
Cycling_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/Adhesion/cell_type0305marker基因查找/Cycling_ego.csv")
REV3L_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/Adhesion/cell_type0305marker基因查找/REV3L_ego.csv")
ISG15_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/Adhesion/cell_type0305marker基因查找/ISG15_ego.csv")
APOD_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/Adhesion/cell_type0305marker基因查找/APOD_ego.csv")
P_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/Adhesion/cell_type0305marker基因查找/P_ego.csv")

S0_ego$cluster="S0"
S1_ego$cluster="S1"
BMP7_ego$cluster="BMP7"
Cycling_ego$cluster="Cycling"
REV3L_ego$cluster="REV3L"
ISG15_ego$cluster="ISG15"
APOD_ego$cluster="APOD"
P_ego$cluster="P"

S0_ego=S0_ego[order(S0_ego$P.Value),]
S1_ego=S1_ego[order(S1_ego$P.Value),]
BMP7_ego=BMP7_ego[order(BMP7_ego$P.Value),]
Cycling_ego=Cycling_ego[order(Cycling_ego$P.Value),]
REV3L_ego=REV3L_ego[order(REV3L_ego$P.Value),]
ISG15_ego=ISG15_ego[order(ISG15_ego$P.Value),]
APOD_ego=APOD_ego[order(APOD_ego$P.Value),]
P_ego=P_ego[order(P_ego$P.Value),]

ego_all=rbind(S0_ego,S1_ego,BMP7_ego,Cycling_ego,REV3L_ego,
              ISG15_ego,APOD_ego,P_ego)

Z=ego_all[duplicated(ego_all$X.Term),]

N=subset(ego_all,!X.Term%in%unique(Z$X.Term))

table(duplicated(N$X.Term))

S0_ego=subset(N,cluster%in%"S0")
S1_ego=subset(N,cluster%in%"S1")
BMP7_ego=subset(N,cluster%in%"BMP7")
Cycling_ego=subset(N,cluster%in%"Cycling")
REV3L_ego=subset(N,cluster%in%"REV3L")
ISG15_ego=subset(N,cluster%in%"ISG15")
APOD_ego=subset(N,cluster%in%"APOD")
P_ego=subset(N,cluster%in%"P")

S0_des=S0_ego[1:10,]$X.Term
S1_des=S1_ego[1:10,]$X.Term
BMP7_des=BMP7_ego[1:10,]$X.Term
Cycling_des=Cycling_ego[1:10,]$X.Term
REV3L_des=REV3L_ego[1:10,]$X.Term
ISG15_des=ISG15_ego[1:10,]$X.Term
APOD_des=APOD_ego[1:10,]$X.Term
P_des=P_ego[1:10,]$X.Term



S0_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/Adhesion/cell_type0305marker基因查找/S0_ego.csv")
S1_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/Adhesion/cell_type0305marker基因查找/S1_ego.csv")
BMP7_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/Adhesion/cell_type0305marker基因查找/BMP7_ego.csv")
Cycling_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/Adhesion/cell_type0305marker基因查找/Cycling_ego.csv")
REV3L_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/Adhesion/cell_type0305marker基因查找/REV3L_ego.csv")
ISG15_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/Adhesion/cell_type0305marker基因查找/ISG15_ego.csv")
APOD_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/Adhesion/cell_type0305marker基因查找/APOD_ego.csv")
P_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/Adhesion/cell_type0305marker基因查找/P_ego.csv")

S0_ego$cluster="S0"
S1_ego$cluster="S1"
BMP7_ego$cluster="BMP7"
Cycling_ego$cluster="Cycling"
REV3L_ego$cluster="REV3L"
ISG15_ego$cluster="ISG15"
APOD_ego$cluster="APOD"
P_ego$cluster="P"


S0_ego=S0_ego[order(S0_ego$P.Value),]
S1_ego=S1_ego[order(S1_ego$P.Value),]
BMP7_ego=BMP7_ego[order(BMP7_ego$P.Value),]
Cycling_ego=Cycling_ego[order(Cycling_ego$P.Value),]
REV3L_ego=REV3L_ego[order(REV3L_ego$P.Value),]
ISG15_ego=ISG15_ego[order(ISG15_ego$P.Value),]
APOD_ego=APOD_ego[order(APOD_ego$P.Value),]
P_ego=P_ego[order(P_ego$P.Value),]


ego_all=rbind(S0_ego,S1_ego,BMP7_ego,Cycling_ego,REV3L_ego,
              ISG15_ego,APOD_ego,P_ego)

ego_all=subset(ego_all,X.Term%in%c(S0_des,S1_des,BMP7_des,Cycling_des,REV3L_des,
                                   ISG15_des,APOD_des,P_des))



ego_all$cluster=factor(ego_all$cluster,levels = c("S0","S1","BMP7","Cycling","REV3L","ISG15","APOD","P"))


ggplot(data=ego_all)+aes(x=cluster,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(ego_all$X.Term))+theme_bw()+xlab("Cluster")+
  scale_color_gradient(low="blue",high = "red")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+xlab("")+
  ylab("")




#挑前8富集通路进行绘图-气泡图---非unique的通路
S0_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/Adhesion/cell_type0305marker基因查找/S0_ego.csv")
S1_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/Adhesion/cell_type0305marker基因查找/S1_ego.csv")
BMP7_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/Adhesion/cell_type0305marker基因查找/BMP7_ego.csv")
Cycling_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/Adhesion/cell_type0305marker基因查找/Cycling_ego.csv")
REV3L_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/Adhesion/cell_type0305marker基因查找/REV3L_ego.csv")
ISG15_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/Adhesion/cell_type0305marker基因查找/ISG15_ego.csv")
APOD_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/Adhesion/cell_type0305marker基因查找/APOD_ego.csv")
P_ego=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/Adhesion/cell_type0305marker基因查找/P_ego.csv")

S0_ego$cluster="S0"
S1_ego$cluster="S1"
BMP7_ego$cluster="BMP7"
Cycling_ego$cluster="Cycling"
REV3L_ego$cluster="REV3L"
ISG15_ego$cluster="ISG15"
APOD_ego$cluster="APOD"
P_ego$cluster="P"

S0_ego=S0_ego[order(S0_ego$P.Value),]
S1_ego=S1_ego[order(S1_ego$P.Value),]
BMP7_ego=BMP7_ego[order(BMP7_ego$P.Value),]
Cycling_ego=Cycling_ego[order(Cycling_ego$P.Value),]
REV3L_ego=REV3L_ego[order(REV3L_ego$P.Value),]
ISG15_ego=ISG15_ego[order(ISG15_ego$P.Value),]
APOD_ego=APOD_ego[order(APOD_ego$P.Value),]
P_ego=P_ego[order(P_ego$P.Value),]





S0_des=S0_ego[1:10,]$X.Term
S1_des=S1_ego[1:10,]$X.Term
BMP7_des=BMP7_ego[1:10,]$X.Term
Cycling_des=Cycling_ego[1:10,]$X.Term
REV3L_des=REV3L_ego[1:10,]$X.Term
ISG15_des=ISG15_ego[1:10,]$X.Term
APOD_des=APOD_ego[1:10,]$X.Term
P_des=P_ego[1:10,]$X.Term

ego_all=rbind(S0_ego,S1_ego,BMP7_ego,Cycling_ego,REV3L_ego,
              ISG15_ego,APOD_ego,P_ego)

ego_all=subset(ego_all,X.Term%in%c(S0_des,S1_des,BMP7_des,Cycling_des,REV3L_des,
                                   ISG15_des,APOD_des,P_des))



ego_all$cluster=factor(ego_all$cluster,levels = c("S0","S1","BMP7","Cycling","REV3L","ISG15","APOD","P"))


ggplot(data=ego_all)+aes(x=cluster,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(ego_all$X.Term))+theme_bw()+xlab("Cluster")+
  scale_color_gradient(low="blue",high = "red")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+xlab("")+
  ylab("")


des=as.data.frame(unique(ego_all$X.Term))



#将Adhesion中各个亚型的kobas的marker基因富集结果进行绘图并统一观察KEGG富集-----
#挑前8富集通路进行绘图-气泡图---unique的通路
S0_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/S0_kegg.csv")
S1_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/S1_kegg.csv")
BMP7_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/BMP7_kegg.csv")
Cycling_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/Cycling_kegg.csv")
REV3L_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/REV3L_kegg.csv")
ISG15_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/ISG15_kegg.csv")
APOD_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/APOD_kegg.csv")
P_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/P_kegg.csv")

S0_kegg=na.omit(S0_kegg)
S1_kegg=na.omit(S1_kegg)
BMP7_kegg=na.omit(BMP7_kegg)
Cycling_kegg=na.omit(Cycling_kegg)
REV3L_kegg=na.omit(REV3L_kegg)
ISG15_kegg=na.omit(ISG15_kegg)
APOD_kegg=na.omit(APOD_kegg)
P_kegg=na.omit(P_kegg)




S0_kegg$cluster="S0"
S1_kegg$cluster="S1"
BMP7_kegg$cluster="BMP7"
Cycling_kegg$cluster="Cycling"
REV3L_kegg$cluster="REV3L"
ISG15_kegg$cluster="ISG15"
APOD_kegg$cluster="APOD"
P_kegg$cluster="P"

S0_kegg=S0_kegg[order(S0_kegg$P.Value),]
S1_kegg=S1_kegg[order(S1_kegg$P.Value),]
BMP7_kegg=BMP7_kegg[order(BMP7_kegg$P.Value),]
Cycling_kegg=Cycling_kegg[order(Cycling_kegg$P.Value),]
REV3L_kegg=REV3L_kegg[order(REV3L_kegg$P.Value),]
ISG15_kegg=ISG15_kegg[order(ISG15_kegg$P.Value),]
APOD_kegg=APOD_kegg[order(APOD_kegg$P.Value),]
P_kegg=P_kegg[order(P_kegg$P.Value),]

kegg_all=rbind(S0_kegg,S1_kegg,BMP7_kegg,Cycling_kegg,REV3L_kegg,
               ISG15_kegg,APOD_kegg,P_kegg)

Z=kegg_all[duplicated(kegg_all$X.Term),]

N=subset(kegg_all,!X.Term%in%unique(Z$X.Term))

table(duplicated(N$X.Term))

S0_kegg=subset(N,cluster%in%"S0")
S1_kegg=subset(N,cluster%in%"S1")
BMP7_kegg=subset(N,cluster%in%"BMP7")
Cycling_kegg=subset(N,cluster%in%"Cycling")
REV3L_kegg=subset(N,cluster%in%"REV3L")
ISG15_kegg=subset(N,cluster%in%"ISG15")
APOD_kegg=subset(N,cluster%in%"APOD")
P_kegg=subset(N,cluster%in%"P")

S0_des=S0_kegg[1:10,]$X.Term
S1_des=S1_kegg[1:10,]$X.Term
BMP7_des=BMP7_kegg[1:10,]$X.Term
Cycling_des=Cycling_kegg[1:10,]$X.Term
REV3L_des=REV3L_kegg[1:10,]$X.Term
ISG15_des=ISG15_kegg[1:10,]$X.Term
APOD_des=APOD_kegg[1:10,]$X.Term
P_des=P_kegg[1:10,]$X.Term



S0_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/S0_kegg.csv")
S1_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/S1_kegg.csv")
BMP7_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/BMP7_kegg.csv")
Cycling_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/Cycling_kegg.csv")
REV3L_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/REV3L_kegg.csv")
ISG15_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/ISG15_kegg.csv")
APOD_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/APOD_kegg.csv")
P_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/P_kegg.csv")

S0_kegg=na.omit(S0_kegg)
S1_kegg=na.omit(S1_kegg)
BMP7_kegg=na.omit(BMP7_kegg)
Cycling_kegg=na.omit(Cycling_kegg)
REV3L_kegg=na.omit(REV3L_kegg)
ISG15_kegg=na.omit(ISG15_kegg)
APOD_kegg=na.omit(APOD_kegg)
P_kegg=na.omit(P_kegg)



S0_kegg$cluster="S0"
S1_kegg$cluster="S1"
BMP7_kegg$cluster="BMP7"
Cycling_kegg$cluster="Cycling"
REV3L_kegg$cluster="REV3L"
ISG15_kegg$cluster="ISG15"
APOD_kegg$cluster="APOD"
P_kegg$cluster="P"


S0_kegg=S0_kegg[order(S0_kegg$P.Value),]
S1_kegg=S1_kegg[order(S1_kegg$P.Value),]
BMP7_kegg=BMP7_kegg[order(BMP7_kegg$P.Value),]
Cycling_kegg=Cycling_kegg[order(Cycling_kegg$P.Value),]
REV3L_kegg=REV3L_kegg[order(REV3L_kegg$P.Value),]
ISG15_kegg=ISG15_kegg[order(ISG15_kegg$P.Value),]
APOD_kegg=APOD_kegg[order(APOD_kegg$P.Value),]
P_kegg=P_kegg[order(P_kegg$P.Value),]


kegg_all=rbind(S0_kegg,S1_kegg,BMP7_kegg,Cycling_kegg,REV3L_kegg,
               ISG15_kegg,APOD_kegg,P_kegg)

kegg_all=subset(kegg_all,X.Term%in%c(S0_des,S1_des,BMP7_des,Cycling_des,REV3L_des,
                                     ISG15_des,APOD_des,P_des))



kegg_all$cluster=factor(kegg_all$cluster,levels = c("S0","S1","BMP7","Cycling","REV3L","ISG15","APOD","P"))


ggplot(data=kegg_all)+aes(x=cluster,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(kegg_all$X.Term))+theme_bw()+xlab("Cluster")+
  scale_color_gradient(low="blue",high = "red")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+xlab("")+
  ylab("")




#挑前8富集通路进行绘图-气泡图---非unique的通路
S0_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/S0_kegg.csv")
S1_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/S1_kegg.csv")
BMP7_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/BMP7_kegg.csv")
Cycling_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/Cycling_kegg.csv")
REV3L_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/REV3L_kegg.csv")
ISG15_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/ISG15_kegg.csv")
APOD_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/APOD_kegg.csv")
P_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/P_kegg.csv")

S0_kegg$cluster="S0"
S1_kegg$cluster="S1"
BMP7_kegg$cluster="BMP7"
Cycling_kegg$cluster="Cycling"
REV3L_kegg$cluster="REV3L"
ISG15_kegg$cluster="ISG15"
APOD_kegg$cluster="APOD"
P_kegg$cluster="P"

S0_kegg=S0_kegg[order(S0_kegg$P.Value),]
S1_kegg=S1_kegg[order(S1_kegg$P.Value),]
BMP7_kegg=BMP7_kegg[order(BMP7_kegg$P.Value),]
Cycling_kegg=Cycling_kegg[order(Cycling_kegg$P.Value),]
REV3L_kegg=REV3L_kegg[order(REV3L_kegg$P.Value),]
ISG15_kegg=ISG15_kegg[order(ISG15_kegg$P.Value),]
APOD_kegg=APOD_kegg[order(APOD_kegg$P.Value),]
P_kegg=P_kegg[order(P_kegg$P.Value),]





S0_des=S0_kegg[1:10,]$X.Term
S1_des=S1_kegg[1:10,]$X.Term
BMP7_des=BMP7_kegg[1:10,]$X.Term
Cycling_des=Cycling_kegg[1:10,]$X.Term
REV3L_des=REV3L_kegg[1:10,]$X.Term
ISG15_des=ISG15_kegg[1:10,]$X.Term
APOD_des=APOD_kegg[1:10,]$X.Term
P_des=P_kegg[1:10,]$X.Term

kegg_all=rbind(S0_kegg,S1_kegg,BMP7_kegg,Cycling_kegg,REV3L_kegg,
               ISG15_kegg,APOD_kegg,P_kegg)

kegg_all=subset(kegg_all,X.Term%in%c(S0_des,S1_des,BMP7_des,Cycling_des,REV3L_des,
                                     ISG15_des,APOD_des,P_des))



kegg_all$cluster=factor(kegg_all$cluster,levels = c("S0","S1","BMP7","Cycling","REV3L","ISG15","APOD","P"))


ggplot(data=kegg_all)+aes(x=cluster,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(kegg_all$X.Term))+theme_bw()+xlab("Cluster")+
  scale_color_gradient(low="blue",high = "red")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+xlab("")+
  ylab("")






#对Adhesion的不同基因进行vlnplot-----
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")
STR_adhesion=subset(STR,type4%in%"Adhesion")

gene_0314_1=c("IL18","IGFBP7","IGFBP5","IGFBP3","IGF1","SPP1","MMP9")
VlnPlot(STR_adhesion,features = gene_0314_1,group.by = "cell_type0305",pt.size = 0,ncol = 7)

gene_0314_2=c("LUM","CD24","WNT5A","FN1","IGF1","PDGFRB","TNFRSF12A")
VlnPlot(STR_adhesion,features = gene_0314_2,group.by = "cell_type0305",pt.size = 0,ncol = 7)

gene_0314_3=c("VCAN","CD44","NAMPT","LOX","HMGB1","PTGS2","TGFB1")
VlnPlot(STR_adhesion,features = gene_0314_3,group.by = "cell_type0305",pt.size = 0,ncol = 7)

gene_0314_4=c("CD163","MARCO","SMAD1","SMAD2","SMAD3","SMAD5","SMAD9")
VlnPlot(STR_adhesion,features = gene_0314_4,group.by = "cell_type0305",pt.size = 0,ncol = 7)

gene_0314_5=c("TGFB1","TGFB2","TGFB3","TGFB4")
VlnPlot(STR_adhesion,features = gene_0314_5,group.by = "cell_type0305",pt.size = 0,ncol = 4)

FeaturePlot(STR_adhesion,features = gene_0314_5,min.cutoff = 0,
            order = T)


FeaturePlot(STR,features = gene_0314_4,min.cutoff = 0,
            order = T)

gene_0314_6=c('COL1A1','COL1A2','COL3A1','TIMP1')

VlnPlot(STR,features = gene_0314_6,group.by = "cell_type0305",split.by = "type4",ncol = 4,pt.size = 0)

gene_0314_7=c('PDGFRA','PDGFRB',"EGFR")

VlnPlot(STR,features = gene_0314_7,group.by = "cell_type0305",split.by = "type4",ncol = 3,pt.size = 0)



#只做STR中的CTRL和ADheison
STR_adhesion=subset(STR,type4%in%c("Adhesion","Ctrl"))
gene_0314_6=c('COL1A1','COL1A2','COL3A1','TIMP1','PDGFRA','PDGFRB',"EGFR")

VlnPlot(STR_adhesion,features = gene_0314_6,group.by = "cell_type0305",split.by = "type4",ncol = 4,pt.size = 0,
        combine = T)


STR_adhesiondata=
  as.data.frame(STR_adhesion@assays$RNA@data)

STR_adhesiondata=subset(STR_adhesiondata,rownames(STR_adhesiondata)%in%gene_0314_6)


STR_adhesiondata=as.data.frame(t(STR_adhesiondata))

STR_adhesiondata$type2=STR_adhesion$type4

STR_adhesiondata$cell_type0305=STR_adhesion$cell_type0305



###开始对每种细胞类型进行一个P值统计
###S0
STR_adhesiondata_S0=subset(STR_adhesiondata,cell_type0305%in%"S0")

STR_adhesiondata_S0=STR_adhesiondata_S0[,1:8]

STR_adhesiondata_S0=gather(STR_adhesiondata_S0,gene_0314_6, value, colnames(STR_adhesiondata_S0),-type2)

my_comparisons=list(c("Ctrl","Adhesion")) 

STR_adhesiondata_S0_1=subset(STR_adhesiondata_S0,gene_0314_6%in%"COL3A1")

ggplot(STR_adhesiondata_S0, aes(x=gene_0314_6, y=value, fill=type2))+
  stat_compare_means(comparisons = my_comparisons,label = "p.signif")+facet_grid(~gene_0314_6, scales = "free_x")+
  geom_violin()+theme_bw()+xlab("")+
  ylab("")+ggtitle("S0")



###S1
STR_adhesiondata_S1=subset(STR_adhesiondata,cell_type0305%in%"S1")

STR_adhesiondata_S1=STR_adhesiondata_S1[,1:8]

STR_adhesiondata_S1=gather(STR_adhesiondata_S1,gene_0314_6, value, colnames(STR_adhesiondata_S1),-type2)

my_comparisons=list(c("Ctrl","Adhesion")) 

STR_adhesiondata_S1_1=subset(STR_adhesiondata_S1,gene_0314_6%in%"COL3A1")

ggplot(STR_adhesiondata_S1, aes(x=gene_0314_6, y=value, fill=type2))+
  stat_compare_means()+facet_grid(~gene_0314_6, scales = "free_x")+
  geom_violin()+theme_bw()+xlab("")+
  ylab("")+ggtitle("S1")

###BMP7
STR_adhesiondata_BMP7=subset(STR_adhesiondata,cell_type0305%in%"BMP7")

STR_adhesiondata_BMP7=STR_adhesiondata_BMP7[,1:8]

STR_adhesiondata_BMP7=gather(STR_adhesiondata_BMP7,gene_0314_6, value, colnames(STR_adhesiondata_BMP7),-type2)

my_comparisons=list(c("Ctrl","Adhesion")) 

STR_adhesiondata_BMP7_1=subset(STR_adhesiondata_BMP7,gene_0314_6%in%"COL3A1")

ggplot(STR_adhesiondata_BMP7, aes(x=gene_0314_6, y=value, fill=type2))+
  stat_compare_means()+facet_grid(~gene_0314_6, scales = "free_x")+
  geom_violin()+theme_bw()+xlab("")+
  ylab("")+ggtitle("BMP7")


###REV3L
STR_adhesiondata_REV3L=subset(STR_adhesiondata,cell_type0305%in%"REV3L")

STR_adhesiondata_REV3L=STR_adhesiondata_REV3L[,1:8]

STR_adhesiondata_REV3L=gather(STR_adhesiondata_REV3L,gene_0314_6, value, colnames(STR_adhesiondata_REV3L),-type2)

my_comparisons=list(c("Ctrl","Adhesion")) 

STR_adhesiondata_REV3L_1=subset(STR_adhesiondata_REV3L,gene_0314_6%in%"COL3A1")

ggplot(STR_adhesiondata_REV3L, aes(x=gene_0314_6, y=value, fill=type2))+
  stat_compare_means()+facet_grid(~gene_0314_6, scales = "free_x")+
  geom_violin()+theme_bw()+xlab("")+
  ylab("")+ggtitle("REV3L")

###ISG15
STR_adhesiondata_ISG15=subset(STR_adhesiondata,cell_type0305%in%"ISG15")

STR_adhesiondata_ISG15=STR_adhesiondata_ISG15[,1:8]

STR_adhesiondata_ISG15=gather(STR_adhesiondata_ISG15,gene_0314_6, value, colnames(STR_adhesiondata_ISG15),-type2)

my_comparisons=list(c("Ctrl","Adhesion")) 

STR_adhesiondata_ISG15_1=subset(STR_adhesiondata_ISG15,gene_0314_6%in%"COL3A1")

ggplot(STR_adhesiondata_ISG15, aes(x=gene_0314_6, y=value, fill=type2))+
  stat_compare_means()+facet_grid(~gene_0314_6, scales = "free_x")+
  geom_violin()+theme_bw()+xlab("")+
  ylab("")+ggtitle("ISG15")



###APOD
STR_adhesiondata_APOD=subset(STR_adhesiondata,cell_type0305%in%"APOD")

STR_adhesiondata_APOD=STR_adhesiondata_APOD[,1:8]

STR_adhesiondata_APOD=gather(STR_adhesiondata_APOD,gene_0314_6, value, colnames(STR_adhesiondata_APOD),-type2)

my_comparisons=list(c("Ctrl","Adhesion")) 

STR_adhesiondata_APOD_1=subset(STR_adhesiondata_APOD,gene_0314_6%in%"COL3A1")

ggplot(STR_adhesiondata_APOD, aes(x=gene_0314_6, y=value, fill=type2))+
  stat_compare_means()+facet_grid(~gene_0314_6, scales = "free_x")+
  geom_violin()+theme_bw()+xlab("")+
  ylab("")+ggtitle("APOD")


###Cycling
STR_adhesiondata_Cycling=subset(STR_adhesiondata,cell_type0305%in%"Cycling")

STR_adhesiondata_Cycling=STR_adhesiondata_Cycling[,1:8]

STR_adhesiondata_Cycling=gather(STR_adhesiondata_Cycling,gene_0314_6, value, colnames(STR_adhesiondata_Cycling),-type2)

my_comparisons=list(c("Ctrl","Adhesion")) 

STR_adhesiondata_Cycling_1=subset(STR_adhesiondata_Cycling,gene_0314_6%in%"COL3A1")

ggplot(STR_adhesiondata_Cycling, aes(x=gene_0314_6, y=value, fill=type2))+
  stat_compare_means()+facet_grid(~gene_0314_6, scales = "free_x")+
  geom_violin()+theme_bw()+xlab("")+
  ylab("")+ggtitle("Cycling")




###P
STR_adhesiondata_P=subset(STR_adhesiondata,cell_type0305%in%"P")

STR_adhesiondata_P=STR_adhesiondata_P[,1:8]

STR_adhesiondata_P=gather(STR_adhesiondata_P,gene_0314_6, value, colnames(STR_adhesiondata_P),-type2)

my_comparisons=list(c("Ctrl","Adhesion")) 

STR_adhesiondata_P_1=subset(STR_adhesiondata_P,gene_0314_6%in%"COL3A1")

ggplot(STR_adhesiondata_P, aes(x=gene_0314_6, y=value, fill=type2))+
  stat_compare_means()+facet_grid(~gene_0314_6, scales = "free_x")+
  geom_violin()+theme_bw()+xlab("")+
  ylab("")+ggtitle("P")








#所有velocyto的ALL-STR的结果并进行kobas富集以后对不同模块的富集结果进行绘图以及解读-----
module1=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/20230314_2/module1_ego.csv")
module2=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/20230314_2/module2_ego.csv")
module3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/20230314_2/module3_ego.csv")
module4=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/20230314_2/module4_ego.csv")
module5=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/20230314_2/module5_ego.csv")
module6=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/20230314_2/module6_ego.csv")

module1$cluster="1"
module2$cluster="2"
module3$cluster="3"
module4$cluster="4"
module5$cluster="5"
module6$cluster="6"


module1=module1[order(module1$P.Value),]
module2=module2[order(module2$P.Value),]
module3=module3[order(module3$P.Value),]
module4=module4[order(module4$P.Value),]
module5=module5[order(module5$P.Value),]
module6=module6[order(module6$P.Value),]




ego_all=rbind(module1,module2,module3,module4,module5,module6)

Z=ego_all[duplicated(ego_all$X.Term),]

N=subset(ego_all,!X.Term%in%unique(Z$X.Term))

table(duplicated(N$X.Term))

module1=subset(N,cluster%in%"1")
module2=subset(N,cluster%in%"2")
module3=subset(N,cluster%in%"3")
module4=subset(N,cluster%in%"4")
module5=subset(N,cluster%in%"5")
module6=subset(N,cluster%in%"6")

module1_des=module1[1:15,]$X.Term
module2_des=module2[1:15,]$X.Term
module3_des=module3[1:15,]$X.Term
module4_des=module4[1:15,]$X.Term
module5_des=module5[1:15,]$X.Term
module6_des=module6[1:15,]$X.Term


module1=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/20230314_2/module1_ego.csv")
module2=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/20230314_2/module2_ego.csv")
module3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/20230314_2/module3_ego.csv")
module4=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/20230314_2/module4_ego.csv")
module5=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/20230314_2/module5_ego.csv")
module6=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/20230314_2/module6_ego.csv")

module1$cluster="1"
module2$cluster="2"
module3$cluster="3"
module4$cluster="4"
module5$cluster="5"
module6$cluster="6"


module1=module1[order(module1$P.Value),]
module2=module2[order(module2$P.Value),]
module3=module3[order(module3$P.Value),]
module4=module4[order(module4$P.Value),]
module5=module5[order(module5$P.Value),]
module6=module6[order(module6$P.Value),]




ego_all=rbind(module1,module2,module3,module4,module5,module6)


ego_all=subset(ego_all,X.Term%in%c(module1_des,module2_des,module3_des,
                                   module4_des,module5_des,module6_des))


ego_all=subset(ego_all,ego_all$P.Value<0.05)
ego_all$cluster=factor(ego_all$cluster,levels = c("1","2","3","4","5","6"))


ggplot(data=ego_all)+aes(x=cluster,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(ego_all$X.Term))+theme_bw()+xlab("Cluster")+
  scale_color_gradient(low="blue",high = "red")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+xlab("")+
  ylab("")




#挑前8富集通路进行绘图-气泡图---非unique的通路
module1=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/20230314_2/module1_ego.csv")
module2=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/20230314_2/module2_ego.csv")
module3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/20230314_2/module3_ego.csv")
module4=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/20230314_2/module4_ego.csv")
module5=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/20230314_2/module5_ego.csv")
module6=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/20230314_2/module6_ego.csv")

module1$cluster="1"
module2$cluster="2"
module3$cluster="3"
module4$cluster="4"
module5$cluster="5"
module6$cluster="6"


module1=module1[order(module1$P.Value),]
module2=module2[order(module2$P.Value),]
module3=module3[order(module3$P.Value),]
module4=module4[order(module4$P.Value),]
module5=module5[order(module5$P.Value),]
module6=module6[order(module6$P.Value),]


module1_des=module1[1:15,]$X.Term
module2_des=module2[1:15,]$X.Term
module3_des=module3[1:15,]$X.Term
module4_des=module4[1:15,]$X.Term
module5_des=module5[1:15,]$X.Term
module6_des=module6[1:15,]$X.Term



ego_all=rbind(module1,module2,module3,module4,module5,module6)



ego_all=subset(ego_all,X.Term%in%c(module1_des,module2_des,module3_des,
                                   module4_des,module5_des,module6_des))


ego_all$cluster=factor(ego_all$cluster,levels = c("1","2","3","4","5","6"))

ego_all=subset(ego_all,ego_all$P.Value<0.05)

ggplot(data=ego_all)+aes(x=cluster,y=X.Term,color=-log10(P.Value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(ego_all$X.Term))+theme_bw()+xlab("Cluster")+
  scale_color_gradient(low="blue",high = "red")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+xlab("")+
  ylab("")


des=as.data.frame(unique(ego_all$X.Term))



#开始cellphonedb的文件准备【上皮细胞和基质细胞】[CTRL和IUA所有]1-----
umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")
DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type1110")

EPI=subset(umap_ctrl_IUA_allsingle,cell_type1110%in%"Epithelium cells")

STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

EPI$cell_type0305="EPI"
table(STR$cell_type0305)

STR_EPI=merge(STR,EPI)



table(STR_EPI$cell_type0305)

##CTRL1的受体配体文件准备
STR_EPI_CTRL1=subset(STR_EPI,type3%in%c("CTRL1"))
write.table(as.matrix(subset(STR_EPI,type3%in%c("CTRL1"))@assays$RNA@data), '/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/CTRL1/cellphonedb_count.txt', sep='\t', quote=F)
meta_STR_EPI_CTRL1 <- cbind(rownames(STR_EPI_CTRL1@meta.data), STR_EPI_CTRL1@meta.data[,'cell_type0305', drop=F])  
meta_STR_EPI_CTRL1 <- as.matrix(meta_STR_EPI_CTRL1)
meta_STR_EPI_CTRL1[is.na(meta_STR_EPI_CTRL1)] = "Unkown" #  细胞类型中不能有NA
write.table(meta_STR_EPI_CTRL1, '/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/CTRL1/meta_STR_EPI_CTRL1.txt', sep='\t', quote=F, row.names=F)

##CTRL2的受体配体文件准备
STR_EPI_CTRL2=subset(STR_EPI,type3%in%c("CTRL2"))
write.table(as.matrix(subset(STR_EPI,type3%in%c("CTRL2"))@assays$RNA@data), '/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/CTRL2/cellphonedb_count.txt', sep='\t', quote=F)
meta_STR_EPI_CTRL2 <- cbind(rownames(STR_EPI_CTRL2@meta.data), STR_EPI_CTRL2@meta.data[,'cell_type0305', drop=F])  
meta_STR_EPI_CTRL2 <- as.matrix(meta_STR_EPI_CTRL2)
meta_STR_EPI_CTRL2[is.na(meta_STR_EPI_CTRL2)] = "Unkown" #  细胞类型中不能有NA
write.table(meta_STR_EPI_CTRL2, '/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/CTRL2/meta_STR_EPI_CTRL2.txt', sep='\t', quote=F, row.names=F)

##IUA1的受体配体文件准备
STR_EPI_IUA1=subset(STR_EPI,type3%in%c("IUA1"))
write.table(as.matrix(subset(STR_EPI,type3%in%c("IUA1"))@assays$RNA@data), '/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/IUA1/cellphonedb_count.txt', sep='\t', quote=F)
meta_STR_EPI_IUA1 <- cbind(rownames(STR_EPI_IUA1@meta.data), STR_EPI_IUA1@meta.data[,'cell_type0305', drop=F])  
meta_STR_EPI_IUA1 <- as.matrix(meta_STR_EPI_IUA1)
meta_STR_EPI_IUA1[is.na(meta_STR_EPI_IUA1)] = "Unkown" #  细胞类型中不能有NA
write.table(meta_STR_EPI_IUA1, '/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/IUA1/meta_STR_EPI_IUA1.txt', sep='\t', quote=F, row.names=F)

##IUA2的受体配体文件准备
STR_EPI_IUA2=subset(STR_EPI,type3%in%c("IUA2"))
write.table(as.matrix(subset(STR_EPI,type3%in%c("IUA2"))@assays$RNA@data), '/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/IUA2/cellphonedb_count.txt', sep='\t', quote=F)
meta_STR_EPI_IUA2 <- cbind(rownames(STR_EPI_IUA2@meta.data), STR_EPI_IUA2@meta.data[,'cell_type0305', drop=F])  
meta_STR_EPI_IUA2 <- as.matrix(meta_STR_EPI_IUA2)
meta_STR_EPI_IUA2[is.na(meta_STR_EPI_IUA2)] = "Unkown" #  细胞类型中不能有NA
write.table(meta_STR_EPI_IUA2, '/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/IUA2/meta_STR_EPI_IUA2.txt', sep='\t', quote=F, row.names=F)

##IUA3的受体配体文件准备
STR_EPI_IUA3=subset(STR_EPI,type3%in%c("IUA3"))
write.table(as.matrix(subset(STR_EPI,type3%in%c("IUA3"))@assays$RNA@data), '/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/IUA3/cellphonedb_count.txt', sep='\t', quote=F)
meta_STR_EPI_IUA3 <- cbind(rownames(STR_EPI_IUA3@meta.data), STR_EPI_IUA3@meta.data[,'cell_type0305', drop=F])  
meta_STR_EPI_IUA3 <- as.matrix(meta_STR_EPI_IUA3)
meta_STR_EPI_IUA3[is.na(meta_STR_EPI_IUA3)] = "Unkown" #  细胞类型中不能有NA
write.table(meta_STR_EPI_IUA3, '/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/IUA3/meta_STR_EPI_IUA3.txt', sep='\t', quote=F, row.names=F)

##IUA4的受体配体文件准备
STR_EPI_IUA4=subset(STR_EPI,type3%in%c("IUA4"))
write.table(as.matrix(subset(STR_EPI,type3%in%c("IUA4"))@assays$RNA@data), '/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/IUA4/cellphonedb_count.txt', sep='\t', quote=F)
meta_STR_EPI_IUA4 <- cbind(rownames(STR_EPI_IUA4@meta.data), STR_EPI_IUA4@meta.data[,'cell_type0305', drop=F])  
meta_STR_EPI_IUA4 <- as.matrix(meta_STR_EPI_IUA4)
meta_STR_EPI_IUA4[is.na(meta_STR_EPI_IUA4)] = "Unkown" #  细胞类型中不能有NA
write.table(meta_STR_EPI_IUA4, '/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/IUA4/meta_STR_EPI_IUA4.txt', sep='\t', quote=F, row.names=F)


##IUA5的受体配体文件准备
STR_EPI_IUA5=subset(STR_EPI,type3%in%c("IUA5"))
write.table(as.matrix(subset(STR_EPI,type3%in%c("IUA5"))@assays$RNA@data), '/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/IUA5/cellphonedb_count.txt', sep='\t', quote=F)
meta_STR_EPI_IUA5 <- cbind(rownames(STR_EPI_IUA5@meta.data), STR_EPI_IUA5@meta.data[,'cell_type0305', drop=F])  
meta_STR_EPI_IUA5 <- as.matrix(meta_STR_EPI_IUA5)
meta_STR_EPI_IUA5[is.na(meta_STR_EPI_IUA5)] = "Unkown" #  细胞类型中不能有NA
write.table(meta_STR_EPI_IUA5, '/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/IUA5/meta_STR_EPI_IUA5.txt', sep='\t', quote=F, row.names=F)

##IUA6的受体配体文件准备
STR_EPI_IUA6=subset(STR_EPI,type3%in%c("IUA6"))
write.table(as.matrix(subset(STR_EPI,type3%in%c("IUA6"))@assays$RNA@data), '/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/IUA6/cellphonedb_count.txt', sep='\t', quote=F)
meta_STR_EPI_IUA6 <- cbind(rownames(STR_EPI_IUA6@meta.data), STR_EPI_IUA6@meta.data[,'cell_type0305', drop=F])  
meta_STR_EPI_IUA6 <- as.matrix(meta_STR_EPI_IUA6)
meta_STR_EPI_IUA6[is.na(meta_STR_EPI_IUA6)] = "Unkown" #  细胞类型中不能有NA
write.table(meta_STR_EPI_IUA6, '/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/IUA6/meta_STR_EPI_IUA6.txt', sep='\t', quote=F, row.names=F)


#开始进行分析和绘图
###开始处理数据
CTRL1_sign_means=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/绘图文件/CTRL1_significant_means.csv",row.names = 1)
CTRL2_sign_means=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/绘图文件/CTRL2_significant_means.csv",row.names = 1)
IUA1_sign_means=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/绘图文件/IUA1_significant_means.csv",row.names = 1)
IUA2_sign_means=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/绘图文件/IUA2_significant_means.csv",row.names = 1)
IUA3_sign_means=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/绘图文件/IUA3_significant_means.csv",row.names = 1)
IUA4_sign_means=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/绘图文件/IUA4_significant_means.csv",row.names = 1)
IUA5_sign_means=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/绘图文件/IUA5_significant_means.csv",row.names = 1)
IUA6_sign_means=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/绘图文件/IUA6_significant_means.csv",row.names = 1)

rownames(CTRL1_sign_means)=CTRL1_sign_means$interacting_pair
rownames(CTRL2_sign_means)=CTRL2_sign_means$interacting_pair
rownames(IUA1_sign_means)=IUA1_sign_means$interacting_pair
rownames(IUA2_sign_means)=IUA2_sign_means$interacting_pair
rownames(IUA3_sign_means)=IUA3_sign_means$interacting_pair
rownames(IUA4_sign_means)=IUA4_sign_means$interacting_pair
rownames(IUA5_sign_means)=IUA5_sign_means$interacting_pair
rownames(IUA6_sign_means)=IUA6_sign_means$interacting_pair

CTRL1_sign_means=CTRL1_sign_means[12:92]
CTRL2_sign_means=CTRL2_sign_means[12:92]
IUA1_sign_means=IUA1_sign_means[12:92]
IUA2_sign_means=IUA2_sign_means[12:92]
IUA3_sign_means=IUA3_sign_means[12:92]
IUA4_sign_means=IUA4_sign_means[12:92]
IUA5_sign_means=IUA5_sign_means[12:92]
IUA6_sign_means=IUA6_sign_means[12:92]

CTRL1_sign_means=CTRL1_sign_means[,grep("EPI",colnames(CTRL1_sign_means))]
CTRL2_sign_means=CTRL2_sign_means[,grep("EPI",colnames(CTRL2_sign_means))]
IUA1_sign_means=IUA1_sign_means[,grep("EPI",colnames(IUA1_sign_means))]
IUA2_sign_means=IUA2_sign_means[,grep("EPI",colnames(IUA2_sign_means))]
IUA3_sign_means=IUA3_sign_means[,grep("EPI",colnames(IUA3_sign_means))]
IUA4_sign_means=IUA4_sign_means[,grep("EPI",colnames(IUA4_sign_means))]
IUA5_sign_means=IUA5_sign_means[,grep("EPI",colnames(IUA5_sign_means))]
IUA6_sign_means=IUA6_sign_means[,grep("EPI",colnames(IUA6_sign_means))]


CTRL1_sign_means=CTRL1_sign_means[apply(CTRL1_sign_means, 1, function(y) any(!is.na(y))),]
CTRL2_sign_means=CTRL2_sign_means[apply(CTRL2_sign_means, 1, function(y) any(!is.na(y))),]
IUA1_sign_means=IUA1_sign_means[apply(IUA1_sign_means, 1, function(y) any(!is.na(y))),]
IUA2_sign_means=IUA2_sign_means[apply(IUA2_sign_means, 1, function(y) any(!is.na(y))),]
IUA3_sign_means=IUA3_sign_means[apply(IUA3_sign_means, 1, function(y) any(!is.na(y))),]
IUA4_sign_means=IUA4_sign_means[apply(IUA4_sign_means, 1, function(y) any(!is.na(y))),]
IUA5_sign_means=IUA5_sign_means[apply(IUA5_sign_means, 1, function(y) any(!is.na(y))),]
IUA6_sign_means=IUA6_sign_means[apply(IUA6_sign_means, 1, function(y) any(!is.na(y))),]


name=Reduce(intersect,list(rownames(CTRL1_sign_means),rownames(CTRL2_sign_means),
                           rownames(IUA1_sign_means),rownames(IUA2_sign_means),
                           rownames(IUA3_sign_means),rownames(IUA4_sign_means),rownames(IUA5_sign_means),rownames(IUA6_sign_means)))


CTRL1_means=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/绘图文件/CTRL1_means.csv",row.names = 1)
CTRL2_means=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/绘图文件/CTRL2_means.csv",row.names = 1)
IUA1_means=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/绘图文件/IUA1_means.csv",row.names = 1)
IUA2_means=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/绘图文件/IUA2_means.csv",row.names = 1)
IUA3_means=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/绘图文件/IUA3_means.csv",row.names = 1)
IUA4_means=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/绘图文件/IUA4_means.csv",row.names = 1)
IUA5_means=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/绘图文件/IUA5_means.csv",row.names = 1)
IUA6_means=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/绘图文件/IUA6_means.csv",row.names = 1)

rownames(CTRL1_means)=CTRL1_means$interacting_pair
rownames(CTRL2_means)=CTRL2_means$interacting_pair
rownames(IUA1_means)=IUA1_means$interacting_pair
rownames(IUA2_means)=IUA2_means$interacting_pair
rownames(IUA3_means)=IUA3_means$interacting_pair
rownames(IUA4_means)=IUA4_means$interacting_pair
rownames(IUA5_means)=IUA5_means$interacting_pair
rownames(IUA6_means)=IUA6_means$interacting_pair

CTRL1_means=CTRL1_means[11:91]
CTRL2_means=CTRL2_means[11:91]
IUA1_means=IUA1_means[11:91]
IUA2_means=IUA2_means[11:91]
IUA3_means=IUA3_means[11:91]
IUA4_means=IUA4_means[11:91]
IUA5_means=IUA5_means[11:91]
IUA6_means=IUA6_means[11:91]

CTRL1_means=CTRL1_means[,grep("EPI",colnames(CTRL1_means))]
CTRL2_means=CTRL2_means[,grep("EPI",colnames(CTRL2_means))]
IUA1_means=IUA1_means[,grep("EPI",colnames(IUA1_means))]
IUA2_means=IUA2_means[,grep("EPI",colnames(IUA2_means))]
IUA3_means=IUA3_means[,grep("EPI",colnames(IUA3_means))]
IUA4_means=IUA4_means[,grep("EPI",colnames(IUA4_means))]
IUA5_means=IUA5_means[,grep("EPI",colnames(IUA5_means))]
IUA6_means=IUA6_means[,grep("EPI",colnames(IUA6_means))]


CTRL1_means=CTRL1_means[apply(CTRL1_means, 1, function(y) any(!is.na(y))),]
CTRL2_means=CTRL2_means[apply(CTRL2_means, 1, function(y) any(!is.na(y))),]
IUA1_means=IUA1_means[apply(IUA1_means, 1, function(y) any(!is.na(y))),]
IUA2_means=IUA2_means[apply(IUA2_means, 1, function(y) any(!is.na(y))),]
IUA3_means=IUA3_means[apply(IUA3_means, 1, function(y) any(!is.na(y))),]
IUA4_means=IUA4_means[apply(IUA4_means, 1, function(y) any(!is.na(y))),]
IUA5_means=IUA5_means[apply(IUA5_means, 1, function(y) any(!is.na(y))),]
IUA6_means=IUA6_means[apply(IUA6_means, 1, function(y) any(!is.na(y))),]



#分离出相应name
CTRL1_means=subset(CTRL1_means,rownames(CTRL1_means)%in%name)
CTRL1_means=CTRL1_means+0.001

CTRL2_means=subset(CTRL2_means,rownames(CTRL2_means)%in%name)
CTRL2_means=CTRL2_means+0.001

IUA1_means=subset(IUA1_means,rownames(IUA1_means)%in%name)
IUA1_means=IUA1_means+0.001

IUA2_means=subset(IUA2_means,rownames(IUA2_means)%in%name)
IUA2_means=IUA2_means+0.001

IUA3_means=subset(IUA3_means,rownames(IUA3_means)%in%name)
IUA3_means=IUA3_means+0.001

IUA4_means=subset(IUA4_means,rownames(IUA4_means)%in%name)
IUA4_means=IUA4_means+0.001

IUA5_means=subset(IUA5_means,rownames(IUA5_means)%in%name)
IUA5_means=IUA5_means+0.001

IUA6_means=subset(IUA6_means,rownames(IUA6_means)%in%name)
IUA6_means=IUA6_means+0.001



NORMAL_means=(CTRL1_means+CTRL2_means)/2
IUAs_means=(IUA1_means+IUA2_means+IUA3_means+IUA4_means+IUA5_means+IUA6_means)/6

cellphone_heatmap=log2(IUAs_means/NORMAL_means)
cellphone_heatmap=cellphone_heatmap[apply(cellphone_heatmap,1,max)>2|apply(cellphone_heatmap,1,min)< -2,]

name2=rownames(cellphone_heatmap)
cellphone_heatmap=cellphone_heatmap %>% {.$pair<-rownames(.);.} %>% gather(ColA,ColB,-pair)







#对筛选出来的可能差异的71个受体配体对进行p值的计算
CTRL1_means_p=subset(CTRL1_means,rownames(CTRL1_means)%in%name2)
CTRL2_means_p=subset(CTRL2_means,rownames(CTRL2_means)%in%name2)
IUA1_means_p=subset(IUA1_means,rownames(IUA1_means)%in%name2)
IUA2_means_p=subset(IUA2_means,rownames(IUA2_means)%in%name2)
IUA3_means_p=subset(IUA3_means,rownames(IUA3_means)%in%name2)
IUA4_means_p=subset(IUA4_means,rownames(IUA4_means)%in%name2)
IUA5_means_p=subset(IUA5_means,rownames(IUA5_means)%in%name2)
IUA6_means_p=subset(IUA6_means,rownames(IUA6_means)%in%name2)


CTRL1_means_p=CTRL1_means_p %>% {.$pair<-rownames(.);.} %>% gather(ColA,valueCTRL1,-pair)
CTRL2_means_p=CTRL2_means_p %>% {.$pair<-rownames(.);.} %>% gather(ColA,valueCTRL2,-pair)
IUA1_means_p=IUA1_means_p %>% {.$pair<-rownames(.);.} %>% gather(ColA,valueIUA1,-pair)
IUA2_means_p=IUA2_means_p %>% {.$pair<-rownames(.);.} %>% gather(ColA,valueIUA2,-pair)
IUA3_means_p=IUA3_means_p %>% {.$pair<-rownames(.);.} %>% gather(ColA,valueIUA3,-pair)
IUA4_means_p=IUA4_means_p %>% {.$pair<-rownames(.);.} %>% gather(ColA,valueIUA4,-pair)
IUA5_means_p=IUA5_means_p %>% {.$pair<-rownames(.);.} %>% gather(ColA,valueIUA5,-pair)
IUA6_means_p=IUA6_means_p %>% {.$pair<-rownames(.);.} %>% gather(ColA,valueIUA6,-pair)


all_p=data_frame(CTRL1_means_p$pair,CTRL1_means_p$valueCTRL1,CTRL2_means_p$valueCTRL2,IUA1_means_p$valueIUA1,
                 IUA2_means_p$valueIUA2,IUA3_means_p$valueIUA3,IUA4_means_p$valueIUA4,IUA5_means_p$valueIUA5,
                 IUA6_means_p$valueIUA6)
write.csv(all_p,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/绘图文件/all_p.csv")

#在这中间要对all_p文件进行t检验计算p值然后上传以后重新读回去
all_p=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/绘图文件/all_p.csv")
cellphone_heatmap$pvalue=all_p$pvalue
cellphone_heatmap$pvalue=as.numeric(cellphone_heatmap$pvalue)
colnames(cellphone_heatmap)[3]<-"log2FC"
write.csv(cellphone_heatmap,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/绘图文件/cellphone_heatmap.csv")
cellphone_heatmap=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/绘图文件/cellphone_heatmap.csv")

ggplot(data=cellphone_heatmap)+aes(x=ColA,y=pair)+geom_point(aes(color=log2FC,size=-log2(pvalue)))+theme_bw()+
  theme(panel.grid.major=element_blank(),panel.grid.minor=element_blank())+
  scale_colour_gradient2(low="blue",mid="white",high="red")+theme(axis.text.x= element_text(angle = 45,vjust = 1,hjust = 1,face= "bold"))
ggsave('/mnt/chentw/zigongneimo/photo/11-samples/绘图存放点/umap_ctrl_IUA_allsingle/cellphonedb/umap_allsingle_allcelltype_cellphonedb.jpeg',height = 13,width = 12)










#开始cellphonedb的文件准备【上皮细胞和基质细胞】[CTRL和IUA所有]2-----
umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")
DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type1110")

EPI=subset(umap_ctrl_IUA_allsingle,cell_type1110%in%"Epithelium cells")

STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

EPI$cell_type0305="EPI"
table(STR$cell_type0305)

STR_EPI=merge(STR,EPI)



table(STR_EPI$cell_type0305)

##ALL受体配体文件准备
write.table(as.matrix(STR_EPI@assays$RNA@data), '/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/ALL/cellphonedb_count.txt', sep='\t', quote=F)
meta_STR_EPI <- cbind(rownames(STR_EPI@meta.data), STR_EPI@meta.data[,'cell_type0305', drop=F])  
meta_STR_EPI <- as.matrix(meta_STR_EPI)
meta_STR_EPI[is.na(meta_STR_EPI)] = "Unkown" #  细胞类型中不能有NA
write.table(meta_STR_EPI, '/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/ALL/meta_STR_EPI.txt', sep='\t', quote=F, row.names=F)







#开始cellchat【基质细胞】[CTRL]------
library(CellChat)
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")
STR=subset(STR,type1%in%"CTRL")


cellchat <- createCellChat(STR@assays$RNA@data,meta = STR@meta.data,group.by = "cell_type0305")
cellchat=setIdent(cellchat,ident.use = "cell_type0305")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
write.csv(df.net,"/mnt/chentw/zigongneimo/photo/11-samples/绘图存放点/STR/cellchat/EPI3_net.csv")
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
write.csv(df.netp,"/mnt/chentw/zigongneimo/photo/11-samples/绘图存放点/STR/cellchat/EPI3_netp.csv")
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = F,title.name = "Interaction weights/strength")
cellchat@netP$pathways
par(mfrow=c(1,1))
netVisual_heatmap(cellchat,signaling = "IGF",color.heatmap = "Reds")
netAnalysis_contribution(cellchat,signaling = "IGF")

pair.IGF=extractEnrichedLR(cellchat,signaling = "IGF",geneLR.return = F)
LR.show=pair.IGF[1,]

vertex.receiver=c(1,2,4,6)


netVisual_individual(cellchat,signaling = "IGF",pairLR.use = pair.IGF,layout = "circle")

netVisual_bubble(cellchat,remove.isolate = F)+
  theme_bw()+theme(panel.grid.major=element_blank(),panel.grid.minor=element_blank())


cellchat@netP$pathways


pairLR.use <- extractEnrichedLR(cellchat, signaling = c("TGFb","IGF"))
netVisual_bubble(cellchat, targets.use = 4,
                 pairLR.use = pairLR.use, remove.isolate = TRUE)

saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL/STR_CTRL.rds")

cellchat=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL/STR_CTRL.rds")




pairLR.use <- extractEnrichedLR(cellchat, signaling = cellchat@netP$pathways)

netVisual_bubble(cellchat, 
                 pairLR.use = pairLR.use, remove.isolate = TRUE)+
  ggtitle("CTRL")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/temp/P3.jpeg",height = 20,dpi = 1000)



#开始cellchat【基质细胞】[Adhesion]------
library(CellChat)
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")
STR=subset(STR,type1%in%"Adhesion")

cellchat <- createCellChat(STR@assays$RNA@data,meta = STR@meta.data,group.by = "cell_type0305")
cellchat=setIdent(cellchat,ident.use = "cell_type0305")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
write.csv(df.net,"/mnt/chentw/zigongneimo/photo/11-samples/绘图存放点/STR/cellchat/EPI3_net.csv")
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
write.csv(df.netp,"/mnt/chentw/zigongneimo/photo/11-samples/绘图存放点/STR/cellchat/EPI3_netp.csv")
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = F,title.name = "Interaction weights/strength")
cellchat@netP$pathways
par(mfrow=c(1,1))
netVisual_heatmap(cellchat,signaling = "IGF",color.heatmap = "Reds")
netAnalysis_contribution(cellchat,signaling = "IGF")
pair.IGF=extractEnrichedLR(cellchat,signaling = "IGF",geneLR.return = F)
LR.show=pair.IGF[1,]
vertex.receiver=c(1,2,4,6)


netVisual_individual(cellchat,signaling = "IGF",pairLR.use = pair.IGF,layout = "circle")

netVisual_bubble(cellchat,remove.isolate = F)+
  theme_bw()+theme(panel.grid.major=element_blank(),panel.grid.minor=element_blank())


cellchat@netP$pathways


pairLR.use <- extractEnrichedLR(cellchat, signaling = c("TGFb","IGF"))
netVisual_bubble(cellchat, targets.use = 4,
                 pairLR.use = pairLR.use, remove.isolate = TRUE)

saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL/STR_Adhesion.rds")

cellchat=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL/STR_Adhesion.rds")




pairLR.use <- extractEnrichedLR(cellchat, signaling = cellchat@netP$pathways)

netVisual_bubble(cellchat, 
                 pairLR.use = pairLR.use, remove.isolate = TRUE)+
  ggtitle("Adhesion")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/temp/P4.jpeg",height = 20,dpi = 1000)


#开始cellchat【上皮细胞和基质细胞】[CTRL和IUA所有]------
umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")
DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type1110")

EPI=subset(umap_ctrl_IUA_allsingle,cell_type1110%in%"Epithelium cells")

STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

EPI$cell_type0305="EPI"
table(STR$cell_type0305)

STR_EPI=merge(STR,EPI)

cellchat <- createCellChat(STR_EPI@assays$RNA@data,meta = STR_EPI@meta.data,group.by = "cell_type0305")
cellchat=setIdent(cellchat,ident.use = "cell_type0305")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
write.csv(df.net,"/mnt/chentw/zigongneimo/photo/11-samples/绘图存放点/STR_EPI/cellchat/EPI3_net.csv")
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
write.csv(df.netp,"/mnt/chentw/zigongneimo/photo/11-samples/绘图存放点/STR_EPI/cellchat/EPI3_netp.csv")
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = F,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = F,title.name = "Interaction weights/strength")
cellchat@netP$pathways
par(mfrow=c(1,1))
netVisual_heatmap(cellchat,signaling = "IGF",color.heatmap = "Reds")
netAnalysis_contribution(cellchat,signaling = "IGF")
pair.IGF=extractEnrichedLR(cellchat,signaling = "IGF",geneLR.return = F)
LR.show=pair.IGF[1,]
vertex.receiver=c(1,2,4,6)


netVisual_individual(cellchat,signaling = "IGF",pairLR.use = pair.IGF,layout = "circle")

netVisual_bubble(cellchat,remove.isolate = F)+
  theme_bw()+theme(panel.grid.major=element_blank(),panel.grid.minor=element_blank())


cellchat@netP$pathways


pairLR.use <- extractEnrichedLR(cellchat, signaling = c("TGFb","IGF"))
netVisual_bubble(cellchat, targets.use = 4,
                 pairLR.use = pairLR.use, remove.isolate = TRUE)

saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL/STR_EPI.rds")



#开始cellchat【内皮细胞和基质细胞】[CTRL和IUA所有]------
umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")
DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type1110")

Endo=subset(umap_ctrl_IUA_allsingle,cell_type1110%in%"Endothelium cells")

STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

Endo$cell_type0305="Endo"
table(STR$cell_type0305)

STR_Endo=merge(STR,Endo)

cellchat <- createCellChat(STR_Endo@assays$RNA@data,meta = STR_Endo@meta.data,group.by = "cell_type0305")
cellchat=setIdent(cellchat,ident.use = "cell_type0305")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
write.csv(df.net,"/mnt/chentw/zigongneimo/photo/11-samples/绘图存放点/STR_Endo/cellchat/EPI3_net.csv")
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
write.csv(df.netp,"/mnt/chentw/zigongneimo/photo/11-samples/绘图存放点/STR_Endo/cellchat/EPI3_netp.csv")
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = F,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = F,title.name = "Interaction weights/strength")
cellchat@netP$pathways
par(mfrow=c(1,1))
netVisual_heatmap(cellchat,signaling = "IGF",color.heatmap = "Reds")
netAnalysis_contribution(cellchat,signaling = "IGF")
pair.IGF=extractEnrichedLR(cellchat,signaling = "IGF",geneLR.return = F)
LR.show=pair.IGF[1,]
vertex.receiver=c(1,2,4,6)


netVisual_individual(cellchat,signaling = "IGF",pairLR.use = pair.IGF,layout = "circle")

netVisual_bubble(cellchat,remove.isolate = F)+
  theme_bw()+theme(panel.grid.major=element_blank(),panel.grid.minor=element_blank())


cellchat@netP$pathways


pairLR.use <- extractEnrichedLR(cellchat, signaling = c("TGFb"))
netVisual_bubble(cellchat, sources.use = 4,
                 pairLR.use = pairLR.use, remove.isolate = TRUE)


saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL/STR_Endo.rds")





#开始cellchat【内皮细胞和基质细胞和上皮细胞】[CTRL和IUA所有，celltype0305]------
library(CellChat)
library(ggplot2)
library(Seurat)


umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")
DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type1110")

Endo=subset(umap_ctrl_IUA_allsingle,cell_type1110%in%"Endothelium cells")

EPI=subset(umap_ctrl_IUA_allsingle,cell_type1110%in%"Epithelium cells")

STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

Endo$cell_type0305="Endo"
EPI$cell_type0305="EPI"
table(STR$cell_type0305)

STR_Endo_EPI=merge(STR,y=c(EPI,Endo))

cellchat <- createCellChat(STR_Endo_EPI@assays$RNA@data,meta = STR_Endo_EPI@meta.data,group.by = "cell_type0305")
cellchat=setIdent(cellchat,ident.use = "cell_type0305")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
write.csv(df.net,"/mnt/chentw/zigongneimo/photo/11-samples/绘图存放点/STR_Endo_EPI/cellchat/EPI3_net.csv")
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
write.csv(df.netp,"/mnt/chentw/zigongneimo/photo/11-samples/绘图存放点/STR_Endo_EPI/cellchat/EPI3_netp.csv")
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = F,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = F,title.name = "Interaction weights/strength")
cellchat@netP$pathways
par(mfrow=c(1,1))
netVisual_heatmap(cellchat,signaling = "IGF",color.heatmap = "Reds")
netAnalysis_contribution(cellchat,signaling = "IGF")
pair.IGF=extractEnrichedLR(cellchat,signaling = "IGF",geneLR.return = F)
LR.show=pair.IGF[1,]
vertex.receiver=c(1,2,4,6)


netVisual_individual(cellchat,signaling = "IGF",pairLR.use = pair.IGF,layout = "circle")

netVisual_bubble(cellchat,remove.isolate = F)+
  theme_bw()+theme(panel.grid.major=element_blank(),panel.grid.minor=element_blank())


cellchat@netP$pathways


pairLR.use <- extractEnrichedLR(cellchat, signaling = c("TGFb"))

netVisual_bubble(cellchat, targets.use = 5,
                 pairLR.use = pairLR.use, remove.isolate = TRUE)


saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL/STR_Endo_EPI.rds")


#挑选通路进行绘图
cellchat=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL/STR_Endo_EPI.rds")

cellchat@netP$pathways

pairLR.use <- extractEnrichedLR(cellchat, signaling = c("TGFb","BMP","IGF","WNT","PRL"))

netVisual_bubble(cellchat, targets.use = 5,
                 pairLR.use = pairLR.use, remove.isolate = TRUE)























#开始cellchat【内皮细胞和基质细胞和上皮细胞】[CTRL和IUA所有，celltype1110]-----
library(CellChat)
library(ggplot2)
library(Seurat)


umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")
DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type1110")

Endo=subset(umap_ctrl_IUA_allsingle,cell_type1110%in%"Endothelium cells")

EPI=subset(umap_ctrl_IUA_allsingle,cell_type1110%in%"Epithelium cells")

STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

STR$cell_type0305="STR"
Endo$cell_type0305="Endo"
EPI$cell_type0305="EPI"
table(STR$cell_type0305)

STR_Endo_EPI=merge(STR,y=c(EPI,Endo))

cellchat <- createCellChat(STR_Endo_EPI@assays$RNA@data,meta = STR_Endo_EPI@meta.data,group.by = "cell_type0305")
cellchat=setIdent(cellchat,ident.use = "cell_type0305")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
write.csv(df.net,"/mnt/chentw/zigongneimo/photo/11-samples/绘图存放点/STR_Endo_EPI/cellchat/EPI3_net.csv")
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
write.csv(df.netp,"/mnt/chentw/zigongneimo/photo/11-samples/绘图存放点/STR_Endo_EPI/cellchat/EPI3_netp.csv")
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = F,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = F,title.name = "Interaction weights/strength")
cellchat@netP$pathways
par(mfrow=c(1,1))
netVisual_heatmap(cellchat,signaling = "IGF",color.heatmap = "Reds")
netAnalysis_contribution(cellchat,signaling = "IGF")
pair.IGF=extractEnrichedLR(cellchat,signaling = "IGF",geneLR.return = F)
LR.show=pair.IGF[1,]
vertex.receiver=c(1,2,4,6)


netVisual_individual(cellchat,signaling = "IGF",pairLR.use = pair.IGF,layout = "circle")

netVisual_bubble(cellchat,remove.isolate = F)+
  theme_bw()+theme(panel.grid.major=element_blank(),panel.grid.minor=element_blank())


cellchat@netP$pathways


pairLR.use <- extractEnrichedLR(cellchat, signaling = c("TGFb"))

netVisual_bubble(cellchat, targets.use = 5,
                 pairLR.use = pairLR.use, remove.isolate = TRUE)


saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL/STR_Endo_EPI2.rds")


#挑选通路进行绘图
cellchat=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL/STR_Endo_EPI2.rds")

cellchat@netP$pathways

pairLR.use <- extractEnrichedLR(cellchat, signaling = c("TGFb","BMP","IGF","WNT","PRL"))

netVisual_bubble(cellchat, 
                 pairLR.use = pairLR.use, remove.isolate = TRUE)



#开始cellchat【内皮细胞和基质细胞和上皮细胞】[CTRL]------
library(CellChat)
library(ggplot2)
library(Seurat)


umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")



DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type1110")

Endo=subset(umap_ctrl_IUA_allsingle,cell_type1110%in%"Endothelium cells")

EPI=subset(umap_ctrl_IUA_allsingle,cell_type1110%in%"Epithelium cells")

STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

Endo$cell_type0305="Endo"
EPI$cell_type0305="EPI"
table(STR$cell_type0305)

STR_Endo_EPI=merge(STR,y=c(EPI,Endo))

STR_Endo_EPI=subset(STR_Endo_EPI,type1%in%"CTRL")

cellchat <- createCellChat(STR_Endo_EPI@assays$RNA@data,meta = STR_Endo_EPI@meta.data,group.by = "cell_type0305")
cellchat=setIdent(cellchat,ident.use = "cell_type0305")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
write.csv(df.net,"/mnt/chentw/zigongneimo/photo/11-samples/绘图存放点/STR_Endo_EPI/cellchat/EPI3_net.csv")
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
write.csv(df.netp,"/mnt/chentw/zigongneimo/photo/11-samples/绘图存放点/STR_Endo_EPI/cellchat/EPI3_netp.csv")
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = F,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = F,title.name = "Interaction weights/strength")
cellchat@netP$pathways
par(mfrow=c(1,1))
netVisual_heatmap(cellchat,signaling = "IGF",color.heatmap = "Reds")
netAnalysis_contribution(cellchat,signaling = "IGF")
pair.IGF=extractEnrichedLR(cellchat,signaling = "IGF",geneLR.return = F)
LR.show=pair.IGF[1,]
vertex.receiver=c(1,2,4,6)


netVisual_individual(cellchat,signaling = "IGF",pairLR.use = pair.IGF,layout = "circle")

netVisual_bubble(cellchat,remove.isolate = F)+
  theme_bw()+theme(panel.grid.major=element_blank(),panel.grid.minor=element_blank())


cellchat@netP$pathways


pairLR.use <- extractEnrichedLR(cellchat, signaling = c("TGFb"))

netVisual_bubble(cellchat, targets.use = 5,
                 pairLR.use = pairLR.use, remove.isolate = TRUE)


saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL/STR_Endo_EPI_ctrl.rds")


#挑选通路进行绘图
cellchat=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL/STR_Endo_EPI_ctrl.rds")

cellchat@netP$pathways

pairLR.use <- extractEnrichedLR(cellchat, signaling = c("TGFb","BMP","IGF","WNT","PRL","SPP1"))

netVisual_bubble(cellchat, targets.use = 5,
                 pairLR.use = pairLR.use, remove.isolate = TRUE)

netVisual_bubble(cellchat,sources.use = 5,
                 pairLR.use = pairLR.use, remove.isolate = TRUE)




#开始cellchat【内皮细胞和基质细胞和上皮细胞】[CTRL,cell_type1110]-------
library(CellChat)
library(ggplot2)
library(Seurat)


umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")



DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type1110")

Endo=subset(umap_ctrl_IUA_allsingle,cell_type1110%in%"Endothelium cells")

EPI=subset(umap_ctrl_IUA_allsingle,cell_type1110%in%"Epithelium cells")

STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

STR$cell_type0305="STR"
Endo$cell_type0305="Endo"
EPI$cell_type0305="EPI"
table(STR$cell_type0305)

STR_Endo_EPI=merge(STR,y=c(EPI,Endo))

STR_Endo_EPI=subset(STR_Endo_EPI,type1%in%"CTRL")

cellchat <- createCellChat(STR_Endo_EPI@assays$RNA@data,meta = STR_Endo_EPI@meta.data,group.by = "cell_type0305")
cellchat=setIdent(cellchat,ident.use = "cell_type0305")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
write.csv(df.net,"/mnt/chentw/zigongneimo/photo/11-samples/绘图存放点/STR_Endo_EPI/cellchat/EPI3_net.csv")
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
write.csv(df.netp,"/mnt/chentw/zigongneimo/photo/11-samples/绘图存放点/STR_Endo_EPI/cellchat/EPI3_netp.csv")
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Interaction weights/strength")
cellchat@netP$pathways
par(mfrow=c(1,1))
netVisual_heatmap(cellchat,signaling = "IGF",color.heatmap = "Reds")
netAnalysis_contribution(cellchat,signaling = "IGF")
pair.IGF=extractEnrichedLR(cellchat,signaling = "IGF",geneLR.return = F)
LR.show=pair.IGF[1,]
vertex.receiver=c(1,2,4,6)


netVisual_individual(cellchat,signaling = "IGF",pairLR.use = pair.IGF,layout = "circle")

netVisual_bubble(cellchat,remove.isolate = F)+
  theme_bw()+theme(panel.grid.major=element_blank(),panel.grid.minor=element_blank())


cellchat@netP$pathways


pairLR.use <- extractEnrichedLR(cellchat, signaling = c("TGFb"))

netVisual_bubble(cellchat, targets.use = 5,
                 pairLR.use = pairLR.use, remove.isolate = TRUE)


saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL/STR_Endo_EPI_ctrl2.rds")


#挑选通路进行绘图
cellchat=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL/STR_Endo_EPI_ctrl2.rds")

pairLR.use=cellchat@netP$pathways

pairLR.use <- extractEnrichedLR(cellchat, signaling = cellchat@netP$pathways)

netVisual_bubble(cellchat, 
                 pairLR.use = pairLR.use, remove.isolate = TRUE)+
  ggtitle("CTRL")
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/temp/P1.jpeg",height = 20,
       dpi = 1000)



#开始cellchat【内皮细胞和基质细胞和上皮细胞】[Adhesion,cell_type1110]-------
library(CellChat)
library(ggplot2)
library(Seurat)


umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")



DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type1110")

Endo=subset(umap_ctrl_IUA_allsingle,cell_type1110%in%"Endothelium cells")

EPI=subset(umap_ctrl_IUA_allsingle,cell_type1110%in%"Epithelium cells")

STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

STR$cell_type0305="STR"
Endo$cell_type0305="Endo"
EPI$cell_type0305="EPI"
table(STR$cell_type0305)

STR_Endo_EPI=merge(STR,y=c(EPI,Endo))

STR_Endo_EPI=subset(STR_Endo_EPI,type2%in%"IUA2")

cellchat <- createCellChat(STR_Endo_EPI@assays$RNA@data,meta = STR_Endo_EPI@meta.data,group.by = "cell_type0305")
cellchat=setIdent(cellchat,ident.use = "cell_type0305")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
write.csv(df.net,"/mnt/chentw/zigongneimo/photo/11-samples/绘图存放点/STR_Endo_EPI/cellchat/EPI3_net.csv")
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
write.csv(df.netp,"/mnt/chentw/zigongneimo/photo/11-samples/绘图存放点/STR_Endo_EPI/cellchat/EPI3_netp.csv")
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = F,title.name = "Interaction weights/strength")
cellchat@netP$pathways
par(mfrow=c(1,1))
netVisual_heatmap(cellchat,signaling = "IGF",color.heatmap = "Reds")
netAnalysis_contribution(cellchat,signaling = "IGF")
pair.IGF=extractEnrichedLR(cellchat,signaling = "IGF",geneLR.return = F)
LR.show=pair.IGF[1,]
vertex.receiver=c(1,2,4,6)


netVisual_individual(cellchat,signaling = "IGF",pairLR.use = pair.IGF,layout = "circle")

netVisual_bubble(cellchat,remove.isolate = F)+
  theme_bw()+theme(panel.grid.major=element_blank(),panel.grid.minor=element_blank())


cellchat@netP$pathways


pairLR.use <- extractEnrichedLR(cellchat, signaling = c("TGFb"))

netVisual_bubble(cellchat, targets.use = 5,
                 pairLR.use = pairLR.use, remove.isolate = TRUE)


saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL/STR_Endo_EPI_adhesion1.rds")


#挑选通路进行绘图
cellchat=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL/STR_Endo_EPI_adhesion1.rds")

cellchat@netP$pathways

pairLR.use <- extractEnrichedLR(cellchat, signaling = cellchat@netP$pathways)

netVisual_bubble(cellchat, 
                 pairLR.use = pairLR.use, remove.isolate = TRUE)+ggtitle("Adhesion
                                                                         ")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/temp/Cellchat_Adhesion.jpeg",height = 20,
       dpi = 1000)




#cellchat，三种主要细胞类型Adhesion和CTRL比较-----
library(CellChat)
library(ggplot2)
library(Seurat)


umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")

umap_ctrl_IUA_allsingle$type4="type4"
umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type2%in%"CTRL"]="Ctrl"
umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type2%in%"IUA1"]="Adjacent"
umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type2%in%"IUA2"]="Adhesion"

DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type1110")

Endo=subset(umap_ctrl_IUA_allsingle,cell_type1110%in%"Endothelium cells")

EPI=subset(umap_ctrl_IUA_allsingle,cell_type1110%in%"Epithelium cells")

STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

STR$cell_type0305="STR"
Endo$cell_type0305="Endo"
EPI$cell_type0305="EPI"
table(STR$cell_type0305)

STR_Endo_EPI=merge(STR,y=c(EPI,Endo))

STR_Endo_EPI=subset(STR_Endo_EPI,type4%in%c("Ctrl","Adhesion"))
CellChatDB <- CellChatDB.human

sc.3=subset(STR_Endo_EPI,type4%in%"Ctrl")
sc.11=subset(STR_Endo_EPI,type4%in%"Adhesion")

cellchat.sc11 <- createCellChat(object =sc.11@assays$RNA@data, meta =sc.11@meta.data,  group.by ="cell_type0305")
cellchat.sc3 <- createCellChat(object =sc.3@assays$RNA@data, meta =sc.3@meta.data,  group.by ="cell_type0305")

setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/20230403/")

##Adhesion
cellchat=cellchat.sc11 
cellchat@DB  <- subsetDB(CellChatDB, search = "Secreted Signaling") # use Secreted Signaling
cellchat <- subsetData(cellchat)
future::plan("multiprocess", workers = 4)
cellchat <- identifyOverExpressedGenes(cellchat)
cellchat <- identifyOverExpressedInteractions(cellchat)
cellchat <- projectData(cellchat, PPI.human)
cellchat <- computeCommunProb(cellchat, raw.use = TRUE,population.size =T)
cellchat <- filterCommunication(cellchat, min.cells = 3)
cellchat <- computeCommunProbPathway(cellchat)
cellchat <- aggregateNet(cellchat)
cellchat <- netAnalysis_computeCentrality(cellchat, slot.name = "netP")
cc.sc11 = cellchat

##CTRL
cellchat=cellchat.sc3
cellchat@DB  <- subsetDB(CellChatDB, search = "Secreted Signaling") # use Secreted Signaling
cellchat <- subsetData(cellchat)
future::plan("multiprocess", workers = 4)
cellchat <- identifyOverExpressedGenes(cellchat)
cellchat <- identifyOverExpressedInteractions(cellchat)
cellchat <- projectData(cellchat, PPI.human)
cellchat <- computeCommunProb(cellchat, raw.use = TRUE,population.size =T)
cellchat <- filterCommunication(cellchat, min.cells = 3)
cellchat <- computeCommunProbPathway(cellchat)
cellchat <- aggregateNet(cellchat)
cellchat <- netAnalysis_computeCentrality(cellchat, slot.name = "netP")
cc.sc3 = cellchat


###
Ctrl=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL/STR_Endo_EPI_ctrl2.rds")
Ctrl$type4="Ctrl"

Adhesion=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL/STR_Endo_EPI_adhesion1.rds")
Adhesion$type4="Adhesion"

cc.list=list(Ctrl=Ctrl,Adhesion=Adhesion)
cellchat=mergeCellChat(cc.list,cell.prefix = T,add.names = names(cc.list))
##可视化
##所有细胞群总体观：通讯数量与强度对比
P1=compareInteractions(cellchat,show.legend = F,group = c(1,3),measure = "count")
P2=compareInteractions(cellchat,show.legend = F,group = c(1,3),measure = "weight")

P1+P2

##第一个图展示通讯数量之间的差异，第二个图展示通讯强度之间的差异。 

##数量与强度差异网络图
netVisual_diffInteraction(cellchat,weight.scale = T)
netVisual_diffInteraction(cellchat,weight.scale = T,measure = "weight")
##红色是case相对于control上调的，蓝色是下调的

#数量与强度差异热图
netVisual_heatmap(cellchat)
netVisual_heatmap(cellchat,measure = "weight")
#case和control对比，红色是上调，蓝色是下调

#保守和特异性信号通路的识别与可视化
rankNet(cellchat,mode = "comparison",stacked = T,do.stat = T)
rankNet(cellchat,mode = "comparison",stacked =F,do.stat = T)
##左图最下面多个信号通路是case组独有的


##细胞互作数量对比网络图
cc.list=list(Ctrl=Ctrl,Adhesion=Adhesion)
weight.max=getMaxWeight(cc.list,attribute = c("idents","count"))
netVisual_circle(cc.list[[1]]@net$count,weight.scale = T,label.edge = F,
                 edge.weight.max =weight.max[2],edge.width.max = 12,title.name = "sc11" )

netVisual_circle(cc.list[[2]]@net$count,weight.scale = T,label.edge = F,
                 edge.weight.max =weight.max[2],edge.width.max = 12,title.name = "sc3" )


plotGeneExpression(cellchat, signaling = 'TGFb',group.by = "type1")


cellchat@meta$cell_type0403="K"

cellchat@meta$cell_type0403=paste(cellchat@meta$type1,"_",cellchat@meta$cell_type0305,sep = "")
plotGeneExpression(cellchat, signaling = 'TGFb',group.by = "cell_type0403")
















#新的整合20230321----
library(Seurat)
library(SeuratDisk)
library(harmony)
library(dplyr)
library(tidyr)
library(ggplot2)
library(clusterProfiler)
library(org.Hs.eg.db)
library(RColorBrewer)
library(export)




umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")
DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type1110")
DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type",label = T)


#重新定义并且开始绘制原图
umap_ctrl_IUA_allsingle$cell_type0216=
  umap_ctrl_IUA_allsingle$cell_type1110

umap_ctrl_IUA_allsingle$cell_type0216=as.character(umap_ctrl_IUA_allsingle$cell_type0216)

DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type")

DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type0216",label = T)+ggtitle("")+theme(legend.position = "none")


umap_ctrl_IUA_allsingle=subset(umap_ctrl_IUA_allsingle,cell_type!="Erythrocyte")
DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type1110")
table(umap_ctrl_IUA_allsingle$cell_type0216)

STR_allsingle=subset(umap_ctrl_IUA_allsingle,cell_type1110%in%"Stromal cells")

STR_allsingle=subset(STR_allsingle,type2%in%c("CTRL","IUA2"))

STR_allsingle=subset(STR_allsingle,type3!="IUA7")


STR_allsingle_embeddings=STR_allsingle$umap@cell.embeddings
STR_allsingle_embeddings=as.data.frame(STR_allsingle_embeddings)

STR_allsingle_embeddings$UMAP_1=as.numeric(as.character(STR_allsingle_embeddings$UMAP_1))
STR_allsingle_embeddings$UMAP_2=as.numeric(as.character(STR_allsingle_embeddings$UMAP_2))
STR_allsingle$UMAP_1=STR_allsingle_embeddings$UMAP_1
STR_allsingle$UMAP_2=STR_allsingle_embeddings$UMAP_2

STR_allsingle=subset(STR_allsingle,UMAP_1<6&UMAP_2>-7)

DimPlot(STR_allsingle)

RPGenes=c(grep('^RPL',rownames(STR_allsingle),value=TRUE),grep('^RPS',rownames(STR_allsingle),value=TRUE))

STR_allsingle[["percent.rp"]] <- PercentageFeatureSet(STR_allsingle, features = RPGenes)
VlnPlot(STR_allsingle, features = c("nFeature_RNA", "nCount_RNA", "percent.rp","percent.mt"), ncol = 4)

DimPlot(STR_allsingle,reduction="umap")

quantile(STR_allsingle$percent.rp)

FeaturePlot(STR_allsingle,features = "percent.rp",
            min.cutoff = 20,order = T,cols = c("grey","red"))

STR_allsingle=subset(STR_allsingle,percent.rp<25)

STR_maker=c("LUM","COL1A1","DCN","MMP11")
EPI_maker=c("EPCAM","KRT18","MMP7","WFDC2","CD9")
Endothelium_maker=c("VWF","CDH5")
smooth_maker=c("MCAM","RGS5","ACTA2")
T_maker=c("CD3D","CD3E","CXCR4","PTPRC","GZMK","CD4","CD8A")
NK_maker=c("NCAM1","GNLY","GZMA")
MAST_maker=c("MS4A2","TPSB2","TPSAB1","CPA3")
B_maker=c("CD79A","CD74")
macr_maker=c("HLA-DRB1","CSF1R","CD14")



#开始进行细胞的过滤【免疫细胞为主】
STR_allsingle_data=
  as.data.frame(STR_allsingle@assays$RNA@data)

STR_allsingle_data=subset(STR_allsingle_data,
                          rownames(STR_allsingle_data)%in%T_maker)

STR_allsingle_data=as.data.frame(t(STR_allsingle_data))


STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$PTPRC<=0)


STR_allsingle=STR_allsingle[,rownames(STR_allsingle_data)]

T_maker=c("CD3D","CD3E","CXCR4","PTPRC","GZMK","CD4","CD8A")
FeaturePlot(STR_allsingle,features = T_maker,min.cutoff = 0)

STR_allsingle_data=
  as.data.frame(STR_allsingle@assays$RNA@data)



STR_allsingle_data=
  as.data.frame(STR_allsingle@assays$RNA@data)

STR_allsingle_data=subset(STR_allsingle_data,
                          rownames(STR_allsingle_data)%in%c(EPI_maker,NK_maker,
                                                            Endothelium_maker,
                                                            smooth_maker,STR_maker,
                                                            T_maker,B_maker,macr_maker,
                                                            c("CD74","LYZ","WFDC2","PAEP","HLA-DRA","CD14")))

STR_allsingle_data=as.data.frame(t(STR_allsingle_data))


quantile(STR_allsingle_data$GNLY,0.9)###
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$GNLY<=0.95)


quantile(STR_allsingle_data$EPCAM,0.92)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$EPCAM<=0.5)


quantile(STR_allsingle_data$VWF,0.99)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$VWF<=0)

quantile(STR_allsingle_data$CXCR4,0.90)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$CXCR4<=0)

quantile(STR_allsingle_data$WFDC2,0.80)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$WFDC2<=0)

quantile(STR_allsingle_data$EPCAM,0.95)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$EPCAM<=0)

quantile(STR_allsingle_data$KRT18,0.95)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$KRT18<=0)


quantile(STR_allsingle_data$CD14,0.95)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$CD14<=0)

quantile(STR_allsingle_data$CD74,0.95)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$CD74<=0)

quantile(STR_allsingle_data$LYZ,0.95)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$LYZ<=0)

quantile(STR_allsingle_data$PAEP,0.95)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$PAEP<=0)


quantile(STR_allsingle_data$`HLA-DRA`,0.95)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$`HLA-DRA`<=0)


STR_allsingle=STR_allsingle[,rownames(STR_allsingle_data)]


table(STR_allsingle$type2)


####开始进行重新整合
N=STR_allsingle@assays$RNA@counts

STR_allsingle=CreateSeuratObject(counts = N, meta.data = STR_allsingle@meta.data)

STR_allsingle=NormalizeData(STR_allsingle, normalization.method = "LogNormalize", 
                            scale.factor = 10000)
STR_allsingle <- FindVariableFeatures(STR_allsingle, selection.method = "vst", nfeatures = 2000)
VlnPlot(STR_allsingle, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)

mtGenes=grep('^MT-',rownames(STR_allsingle),value=TRUE)
RPGenes=c(grep('^RPL',rownames(STR_allsingle),value=TRUE),grep('^RPS',rownames(STR_allsingle),value=TRUE))
stress_gene2=grep('^HSP',rownames(STR_allsingle),value=TRUE)
HB_gene2=grep('^HB',rownames(STR_allsingle),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
stress_gene1=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
               "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/gene1.csv")
subgene3=subgene3$cellcyle_gene
subgene4=c("FOS","CXCL2","ZFP36","FOSB","DUSP1","ATF3","CXCL8","NR4A1","CXCL3","PPP1R15A",
           "JUNB","EGR1","HSPA1A","HSPA1B","SOCS3","KLF6","JUN","IER2","CXCL1","NFKBIA",
           "HSPA6","DNAJB1","IER3","CCNL1","MTRNR2L2","IER4","ID1","CEBPD","KRT6A","CYR61",
           "DEPP1","CLDN4","IRF1","DUSP2","BTG2","PLAUR","MAFF","KLF4","PHLDA2","TNFAIP3")
subgene5=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/512个应激反应相关基因.csv")
subgene5=subgene5$gene_symbol

subgene=c(mtGenes,RPGenes,stress_gene2,HB_gene2,HB.genes,s.genes,g2m.genes,stress_gene1,"MKI67",subgene3,subgene4,subgene5)

STR_allsingle@assays$RNA@var.features = setdiff(STR_allsingle@assays$RNA@var.features,subgene)  #去除subgene影响

STR_allsingle@assays$RNA@var.features

#STR_allsingle进行标准化然后PCA分析
DefaultAssay(STR_allsingle) <- "RNA"
all.genes_STR_allsingle <- rownames(STR_allsingle)
STR_allsingle <- ScaleData(object = STR_allsingle, features = all.genes_STR_allsingle)
STR_allsingle <- RunPCA(STR_allsingle, features = VariableFeatures(object = STR_allsingle))


#STR_allsingle进行细胞细胞聚类分析(UMAP)
STR_allsingle=RunUMAP(STR_allsingle, dims = 1:30)
DimPlot(STR_allsingle,reduction="umap",label=TRUE)
STR_allsingle <- FindNeighbors(STR_allsingle, dims = 1:30)
STR_allsingle <- FindClusters(STR_allsingle, resolution = 0.3)
STR_allsingle <- FindClusters(STR_allsingle, resolution = 0.6)
DimPlot(STR_allsingle,reduction="umap",label=TRUE)


#STR_allsingle进行细胞细胞聚类分析(tsne)
STR_allsingle=RunTSNE(STR_allsingle, dims = 1:30)
DimPlot(STR_allsingle,reduction="tsne",label=TRUE)
STR_allsingle <- FindNeighbors(STR_allsingle, dims = 1:30)
STR_allsingle <- FindClusters(STR_allsingle, resolution = 0.3)
DimPlot(STR_allsingle,reduction="tsne",label=TRUE)

DimPlot(STR_allsingle,reduction="umap",label=TRUE)



#开始尝试去除细胞样本批次并且进行新的一个保存

STR_allsingle=STR_allsingle%>%RunHarmony("type2",plot_covergence=T)

DefaultAssay(STR_allsingle) <- "RNA"


STR_allsingle <- RunPCA(STR_allsingle, features = VariableFeatures(object = STR_allsingle))


STR_allsingle=RunUMAP(STR_allsingle,reduction="harmony",dims=1:30)
STR_allsingle=FindNeighbors(STR_allsingle,dims=1:30,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.15,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.17,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.18,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.2,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.3,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.4,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.45,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.5,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.55,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 1,reduction = "harmony")




STR_allsingle=subset(STR_allsingle,cell_type%in%c("Stromal cells",
                                                  "Perivascular cells",
                                                  "Transient state cells","Cycling stromal cells"))


DimPlot(STR_allsingle,split.by = "type2",label=TRUE,group.by = "RNA_snn_res.0.6")


FeaturePlot(STR_allsingle,features = c("ACTA2","TAGLN"),
            min.cutoff = 1,order = T,cols = c("grey","red"))


#附上细胞类型查看特有cluster
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

STR_sub=subset(STR,type2%in%c("CTRL","IUA2"))
STR_allsingle$cell_type0305=STR_sub$cell_type0305

DimPlot(STR_allsingle,split.by = "type2",label=TRUE)

DimPlot(STR_allsingle,split.by = "type2",label=TRUE,group.by = "cell_type0305")

#cellphonedb绘图20230322------
#开始进行分析和绘图
###开始处理数据
sign_means=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/绘图文件2/significant_means.csv",row.names = 1)



sign_means=sign_means[,grep("EPI",colnames(sign_means))]

sign_means=sign_means[apply(sign_means, 1, function(y) any(!is.na(y))),]

name=rownames(sign_means)

means=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/绘图文件2/means.csv",row.names = 1)


means=means[,grep("EPI",colnames(means))]


means=means[apply(means, 1, function(y) any(!is.na(y))),]




#分离出相应name
means=subset(means,rownames(means)%in%name)


means=means %>% {.$pair<-rownames(.);.} %>% gather(ColA,value,-pair)

all_p=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellphonedb/20230316/绘图文件2/pvalues.csv",row.names = 1)
all_p=subset(all_p,rownames(all_p)%in%name)
all_p=all_p[,grep("EPI",colnames(all_p))]

all_p=all_p %>% {.$pair<-rownames(.);.} %>% gather(ColA,value_P,-pair)



means$pvalue=all_p$value_P

means$pvalue[means$pvalue>0.05]<-1

ggplot(data=means)+aes(x=ColA,y=pair)+geom_point(aes(color=value,size=-pvalue))+theme_bw()+
  theme(panel.grid.major=element_blank(),panel.grid.minor=element_blank())+
  scale_colour_gradient(low="white",high="red")+theme(axis.text.x= element_text(angle = 45,vjust = 1,hjust = 1,face= "bold"))+
  scale_size_continuous(range = c(-1,5),limits = c(-1,0))+xlab("")





#对上皮细胞进行重新的分群聚类-----
library(Seurat)
library(SeuratDisk)
library(harmony)
library(dplyr)
library(tidyr)
library(ggplot2)
library(clusterProfiler)
library(org.Hs.eg.db)
library(RColorBrewer)
library(export)




umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")
DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type1110")
DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type",label = T)


#重新定义并且开始绘制原图
umap_ctrl_IUA_allsingle$cell_type0216=
  umap_ctrl_IUA_allsingle$cell_type1110

umap_ctrl_IUA_allsingle$cell_type0216=as.character(umap_ctrl_IUA_allsingle$cell_type0216)

DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type")

DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type0216",label = T)+ggtitle("")+theme(legend.position = "none")



EPI_allsingle=subset(umap_ctrl_IUA_allsingle,cell_type1110%in%"Epithelium cells")

EPI_allsingle=subset(EPI_allsingle,cell_type%in%c("Epithelials cells","Cliated epithelial cells","Cycling epithelial cells"))


EPI_allsingle_embeddings=EPI_allsingle$umap@cell.embeddings
EPI_allsingle_embeddings=as.data.frame(EPI_allsingle_embeddings)

EPI_allsingle_embeddings$UMAP_1=as.numeric(as.character(EPI_allsingle_embeddings$UMAP_1))
EPI_allsingle_embeddings$UMAP_2=as.numeric(as.character(EPI_allsingle_embeddings$UMAP_2))
EPI_allsingle$UMAP_1=EPI_allsingle_embeddings$UMAP_1
EPI_allsingle$UMAP_2=EPI_allsingle_embeddings$UMAP_2

EPI_allsingle=subset(EPI_allsingle,UMAP_1<5 & UMAP_2< -5)

DimPlot(EPI_allsingle)

RPGenes=c(grep('^RPL',rownames(EPI_allsingle),value=TRUE),grep('^RPS',rownames(EPI_allsingle),value=TRUE))

EPI_allsingle[["percent.rp"]] <- PercentageFeatureSet(EPI_allsingle, features = RPGenes)
VlnPlot(EPI_allsingle, features = c("nFeature_RNA", "nCount_RNA", "percent.rp","percent.mt"), ncol = 4)

DimPlot(EPI_allsingle,reduction="umap")

quantile(EPI_allsingle$percent.rp)

EPI_maker=c("LUM","COL1A1","DCN","MMP11")
EPI_maker=c("EPCAM","KRT18","MMP7","WFDC2","CD9")
Endothelium_maker=c("VWF","CDH5")
smooth_maker=c("MCAM","RGS5","ACTA2")
T_maker=c("CD3D","CD3E","CXCR4","PTPRC","GZMK","CD4","CD8A")
NK_maker=c("NCAM1","GNLY","GZMA")
MAST_maker=c("MS4A2","TPSB2","TPSAB1","CPA3")
B_maker=c("CD79A","CD74")
macr_maker=c("HLA-DRB1","CSF1R","CD14")



#开始进行细胞的过滤【免疫细胞为主】
EPI_allsingle_data=
  as.data.frame(EPI_allsingle@assays$RNA@data)

EPI_allsingle_data=subset(EPI_allsingle_data,
                          rownames(EPI_allsingle_data)%in%T_maker)

EPI_allsingle_data=as.data.frame(t(EPI_allsingle_data))


EPI_allsingle_data=subset(EPI_allsingle_data,EPI_allsingle_data$PTPRC<=0)


EPI_allsingle=EPI_allsingle[,rownames(EPI_allsingle_data)]

T_maker=c("CD3D","CD3E","CXCR4","PTPRC","GZMK","CD4","CD8A")
FeaturePlot(EPI_allsingle,features = T_maker,min.cutoff = 0)

EPI_allsingle_data=
  as.data.frame(EPI_allsingle@assays$RNA@data)



EPI_allsingle_data=
  as.data.frame(EPI_allsingle@assays$RNA@data)

EPI_allsingle_data=subset(EPI_allsingle_data,
                          rownames(EPI_allsingle_data)%in%c(EPI_maker,NK_maker,
                                                            Endothelium_maker,
                                                            smooth_maker,EPI_maker,
                                                            T_maker,B_maker,macr_maker,
                                                            c("CD74","LYZ","WFDC2","PAEP","HLA-DRA","CD14")))

EPI_allsingle_data=as.data.frame(t(EPI_allsingle_data))


quantile(EPI_allsingle_data$GNLY,0.9)###
EPI_allsingle_data=subset(EPI_allsingle_data,EPI_allsingle_data$GNLY<=0)

EPI_allsingle=EPI_allsingle[,rownames(EPI_allsingle_data)]


table(EPI_allsingle$type3)



####开始进行重新整合
N=EPI_allsingle@assays$RNA@counts

EPI_allsingle=CreateSeuratObject(counts = N, meta.data = EPI_allsingle@meta.data)

EPI_allsingle=NormalizeData(EPI_allsingle, normalization.method = "LogNormalize", 
                            scale.factor = 10000)
EPI_allsingle <- FindVariableFeatures(EPI_allsingle, selection.method = "vst", nfeatures = 2000)
VlnPlot(EPI_allsingle, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)

mtGenes=grep('^MT-',rownames(EPI_allsingle),value=TRUE)
RPGenes=c(grep('^RPL',rownames(EPI_allsingle),value=TRUE),grep('^RPS',rownames(EPI_allsingle),value=TRUE))
stress_gene2=grep('^HSP',rownames(EPI_allsingle),value=TRUE)
HB_gene2=grep('^HB',rownames(EPI_allsingle),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
stress_gene1=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
               "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/gene1.csv")
subgene3=subgene3$cellcyle_gene
subgene4=c("FOS","CXCL2","ZFP36","FOSB","DUSP1","ATF3","CXCL8","NR4A1","CXCL3","PPP1R15A",
           "JUNB","EGR1","HSPA1A","HSPA1B","SOCS3","KLF6","JUN","IER2","CXCL1","NFKBIA",
           "HSPA6","DNAJB1","IER3","CCNL1","MTRNR2L2","IER4","ID1","CEBPD","KRT6A","CYR61",
           "DEPP1","CLDN4","IRF1","DUSP2","BTG2","PLAUR","MAFF","KLF4","PHLDA2","TNFAIP3")
subgene5=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/512个应激反应相关基因.csv")
subgene5=subgene5$gene_symbol

subgene=c(mtGenes,RPGenes,stress_gene2,HB_gene2,HB.genes,s.genes,g2m.genes,stress_gene1,"MKI67",subgene3,subgene4,subgene5)

EPI_allsingle@assays$RNA@var.features = setdiff(EPI_allsingle@assays$RNA@var.features,subgene)  #去除subgene影响

EPI_allsingle@assays$RNA@var.features

#EPI_allsingle进行标准化然后PCA分析
DefaultAssay(EPI_allsingle) <- "RNA"
all.genes_EPI_allsingle <- rownames(EPI_allsingle)
EPI_allsingle <- ScaleData(object = EPI_allsingle, features = all.genes_EPI_allsingle)
EPI_allsingle <- RunPCA(EPI_allsingle, features = VariableFeatures(object = EPI_allsingle))


#EPI_allsingle进行细胞细胞聚类分析(UMAP)
EPI_allsingle=RunUMAP(EPI_allsingle, dims = 1:30)
DimPlot(EPI_allsingle,reduction="umap",label=TRUE)
EPI_allsingle <- FindNeighbors(EPI_allsingle, dims = 1:30)
EPI_allsingle <- FindClusters(EPI_allsingle, resolution = 0.3)
EPI_allsingle <- FindClusters(EPI_allsingle, resolution = 0.6)
DimPlot(EPI_allsingle,reduction="umap",label=TRUE)


#EPI_allsingle进行细胞细胞聚类分析(tsne)
EPI_allsingle=RunTSNE(EPI_allsingle, dims = 1:30)
DimPlot(EPI_allsingle,reduction="tsne",label=TRUE)
EPI_allsingle <- FindNeighbors(EPI_allsingle, dims = 1:30)
EPI_allsingle <- FindClusters(EPI_allsingle, resolution = 0.3)
DimPlot(EPI_allsingle,reduction="tsne",label=TRUE)

DimPlot(EPI_allsingle,reduction="umap",label=TRUE)



#开始尝试去除细胞样本批次并且进行新的一个保存
EPI_allsingle=subset(EPI_allsingle,type2%in%c("CTRL","IUA2"))

EPI_allsingle=EPI_allsingle%>%RunHarmony("type3",plot_covergence=T)

DefaultAssay(EPI_allsingle) <- "RNA"


EPI_allsingle <- RunPCA(EPI_allsingle, features = VariableFeatures(object = EPI_allsingle))


EPI_allsingle=RunUMAP(EPI_allsingle,reduction="harmony",dims=1:30)
EPI_allsingle=FindNeighbors(EPI_allsingle,dims=1:30,reduction = "harmony")
EPI_allsingle=FindClusters(EPI_allsingle, resolution = 0.15,reduction = "harmony")
EPI_allsingle=FindClusters(EPI_allsingle, resolution = 0.17,reduction = "harmony")
EPI_allsingle=FindClusters(EPI_allsingle, resolution = 0.18,reduction = "harmony")
EPI_allsingle=FindClusters(EPI_allsingle, resolution = 0.2,reduction = "harmony")
EPI_allsingle=FindClusters(EPI_allsingle, resolution = 0.3,reduction = "harmony")
EPI_allsingle=FindClusters(EPI_allsingle, resolution = 0.4,reduction = "harmony")
EPI_allsingle=FindClusters(EPI_allsingle, resolution = 0.45,reduction = "harmony")
EPI_allsingle=FindClusters(EPI_allsingle, resolution = 0.5,reduction = "harmony")
EPI_allsingle=FindClusters(EPI_allsingle, resolution = 0.55,reduction = "harmony")
EPI_allsingle=FindClusters(EPI_allsingle, resolution = 0.6,reduction = "harmony")



DimPlot(EPI_allsingle,reduction="umap",label=TRUE,group.by = "Phase")+
  DimPlot(EPI_allsingle,reduction="umap",label=TRUE,group.by = "RNA_snn_res.0.3")



DimPlot(EPI_allsingle,group.by = "cell_type",label=TRUE)


DimPlot(EPI_allsingle,group.by = "cell_type",label=TRUE)

DimPlot(EPI_allsingle,split.by = "type2",
        label=TRUE,group.by = "RNA_snn_res.0.3")

saveRDS(EPI_allsingle,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/EPI_allsingle.rds")










#20230323原图绘制-----
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")


#图1-----
DimPlot(STR,reduction="umap",label=F,label.size =7,pt.size = 0.6,group.by = "cell_type0305")+ggtitle("")+
  theme(title = element_text(size =0),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),axis.line = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 0))+theme(text = element_text(size=0))+
  scale_color_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"))



ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/原图绘制/20230323/P1-1.jpeg",
       width = 9,height = 6,dpi = 1000)


#图1-2
DimPlot(STR,reduction="umap",label=T,label.size =7,pt.size = 0.6,group.by = "cell_type0305")+ggtitle("")+
  theme(title = element_text(size =0),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),axis.line = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 0))+
  scale_color_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4"))



ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/原图绘制/20230323/P1-2.jpeg",
       width = 9,height = 6,dpi = 1000)





#图2-------
gene=c("BMP7","RGS5","APOD","ISG15","REV3L","THY1","PDGFRA","COL1A1")

table(STR$cell_type0305)

#2-1
P1=VlnPlot(STR,features = gene[1],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P2=VlnPlot(STR,features = gene[2],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P3=VlnPlot(STR,features = gene[3],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P4=VlnPlot(STR,features = gene[4],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P5=VlnPlot(STR,features = gene[5],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P6=VlnPlot(STR,features = gene[6],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P7=VlnPlot(STR,features = gene[7],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P8=VlnPlot(STR,features = gene[8],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

ggarrange(P1,P2,P3,P4,P5,P6,P7,P8,common.legend = T,ncol = 4,legend = "none",nrow = 2)


jpeg("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/原图绘制/20230323/P2-1.jpeg",width =16000,height = 10000,res = 1200,
     quality = 100)
ggarrange(P1,P2,P3,P4,P5,P6,P7,P8,common.legend = F,ncol = 4,nrow = 2)+theme(legend.position = "none")
dev.off()

#2-2
P1=VlnPlot(STR,features = gene[1],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4"))+coord_flip()+
  theme(legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P2=VlnPlot(STR,features = gene[2],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4"))+coord_flip()+
  theme(legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P3=VlnPlot(STR,features = gene[3],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4"))+coord_flip()+
  theme(legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P4=VlnPlot(STR,features = gene[4],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4"))+coord_flip()+
  theme(legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P5=VlnPlot(STR,features = gene[5],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4"))+coord_flip()+
  theme(legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P6=VlnPlot(STR,features = gene[6],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4"))+coord_flip()+
  theme(legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P7=VlnPlot(STR,features = gene[7],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4"))+coord_flip()+
  theme(legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P8=VlnPlot(STR,features = gene[8],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4"))+coord_flip()+
  theme(legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

ggarrange(P1,P2,P3,P4,P5,P6,P7,P8,common.legend = T,ncol = 4,legend = "none",nrow = 2)


jpeg("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/原图绘制/20230323/P2-2.jpeg",width =20000,height = 10000,res = 1200,
     quality = 100)
ggarrange(P1,P2,P3,P4,P5,P6,P7,P8,common.legend = F,ncol = 4,nrow = 2)+theme(legend.position = "none")
dev.off()



#图3------
#图3-1
STR$type4=factor(STR$type4,levels = c("Ctrl","Adjacent","Adhesion"))

STR$cell_type0305_1=paste(STR$type2,"_",STR$cell_type0305,sep = "")

STR$cell_type0305_1=factor(STR$cell_type0305_1,levels = c(paste("CTRL_",names(table(STR$cell_type0305)),sep = ""),
                                                          paste("IUA1_",names(table(STR$cell_type0305)),sep = ""),
                                                          paste("IUA2_",names(table(STR$cell_type0305)),sep = "")))

av=AverageExpression(STR,
                     group.by = "cell_type0305_1",
                     assays = "RNA")

av=av[[1]]


cg=names(tail(sort(apply(av,1,sd)),2000))


View(av[cg,])

N=cor(av[cg,],method = 'spearman')

N=as.data.frame(N)

N=N[1:8,]
N=N[,-c(1:8)]

N$CTRL=1
N$IUA1=1
N$IUA2=1

N$IUA1[1]=N[1,1]
N$IUA1[2]=N[2,2]
N$IUA1[3]=N[3,3]
N$IUA1[4]=N[4,4]
N$IUA1[5]=N[5,5]
N$IUA1[6]=N[6,6]
N$IUA1[7]=N[7,7]
N$IUA1[8]=N[8,8]

N$IUA2[1]=N[1,9]
N$IUA2[2]=N[2,10]
N$IUA2[3]=N[3,11]
N$IUA2[4]=N[4,12]
N$IUA2[5]=N[5,13]
N$IUA2[6]=N[6,14]
N$IUA2[7]=N[7,15]
N$IUA2[8]=N[8,16]

N=N[17:19]
pheatmap(N,cluster_cols = F, display_numbers = TRUE,cluster_rows = F)

colnames(N)[2]="Adjacent"
colnames(N)[3]="Adhesion"

rownames(N)=gsub("CTRL_","",rownames(N))


XX1=pheatmap(N,cluster_cols = F, display_numbers = F,cluster_rows = F,
             fontsize = 18,show_rownames = F,show_colnames = F,legend = F)

save_pheatmap_pdf <- function(x, filename, width=7000, height=7000) {
  stopifnot(!missing(x))
  stopifnot(!missing(filename))
  jpeg(filename, width=width, height=height)
  grid::grid.newpage()
  grid::grid.draw(x$gtable)
  dev.off()
}

save_pheatmap_pdf(XX1, "3-2.jpeg")


#2组间
STR$type4=factor(STR$type4,levels = c("Ctrl","Adjacent","Adhesion"))

STR$cell_type0305_2=paste(STR$type1,"_",STR$cell_type0305,sep = "")


STR$cell_type0305_2=factor(STR$cell_type0305_2,levels = c(paste("CTRL_",names(table(STR$cell_type0305)),sep = ""),
                                                          paste("IUA_",names(table(STR$cell_type0305)),sep = "")))

av=AverageExpression(STR,
                     group.by = "cell_type0305_2",
                     assays = "RNA")

av=av[[1]]


cg=names(tail(sort(apply(av,1,sd)),2000))


View(av[cg,])

N=cor(av[cg,],method = 'spearman')

N=as.data.frame(N)

N=N[1:8,]
N=N[,-c(1:8)]

N$CTRL=1
N$IUA=1


N$IUA[1]=N[1,1]
N$IUA[2]=N[2,2]
N$IUA[3]=N[3,3]
N$IUA[4]=N[4,4]
N$IUA[5]=N[5,5]
N$IUA[6]=N[6,6]
N$IUA[7]=N[7,7]
N$IUA[8]=N[8,8]



N=N[9:10]

pheatmap(N,cluster_cols = F, display_numbers = TRUE,cluster_rows = F)

rownames(N)=gsub("CTRL_","",rownames(N))


pheatmap(N,cluster_cols = F, display_numbers = TRUE,cluster_rows = F,
         fontsize = 25)

XX1=pheatmap(N,cluster_cols = F, display_numbers = F,cluster_rows = F,
             fontsize = 18,show_rownames = F,show_colnames = F,legend = F)

save_pheatmap_pdf <- function(x, filename, width=4000, height=7000) {
  stopifnot(!missing(x))
  stopifnot(!missing(filename))
  jpeg(filename, width=width, height=height)
  grid::grid.newpage()
  grid::grid.draw(x$gtable)
  dev.off()
}

save_pheatmap_pdf(XX1, "4-2.jpeg")



#图5
av=AverageExpression(STR,
                     group.by = "cell_type0305",
                     assays = "RNA")

av=av[[1]]


cg=names(tail(sort(apply(av,1,sd)),2000))

N=cor(av[cg,],method = 'spearman')

N=as.data.frame(N)

pheatmap(N,cluster_cols = F, display_numbers = F,cluster_rows = F,
         fontsize = 18,)

XX1=pheatmap(N,display_numbers = F,
             fontsize = 18)

save_pheatmap_pdf <- function(x, filename, width=7000, height=7000) {
  stopifnot(!missing(x))
  stopifnot(!missing(filename))
  jpeg(filename, width=width, height=height)
  grid::grid.newpage()
  grid::grid.draw(x$gtable)
  dev.off()
}

save_pheatmap_pdf(XX1, "P5-2.jpeg")


#图6-----
STR_markers=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/STR_markers.csv")

mtGenes=grep('^MT-',rownames(STR),value=TRUE)
RPGenes=c(grep('^RPL',rownames(STR),value=TRUE),grep('^RPS',rownames(STR),value=TRUE))
stress_gene2=grep('^HSP',rownames(STR),value=TRUE)
HB_gene2=grep('^HB',rownames(STR),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
stress_gene1=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
               "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/gene1.csv")
subgene3=subgene3$cellcyle_gene
subgene4=c("FOS","CXCL2","ZFP36","FOSB","DUSP1","ATF3","CXCL8","NR4A1","CXCL3","PPP1R15A",
           "JUNB","EGR1","HSPA1A","HSPA1B","SOCS3","KLF6","JUN","IER2","CXCL1","NFKBIA",
           "HSPA6","DNAJB1","IER3","CCNL1","MTRNR2L2","IER4","ID1","CEBPD","KRT6A","CYR61",
           "DEPP1","CLDN4","IRF1","DUSP2","BTG2","PLAUR","MAFF","KLF4","PHLDA2","TNFAIP3")
subgene5=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/512个应激反应相关基因.csv")
subgene5=subgene5$gene_symbol

subgene=c(mtGenes,RPGenes,stress_gene2,HB_gene2,HB.genes,s.genes,g2m.genes,stress_gene1,"MKI67",subgene3,subgene4,subgene5)


#包含非特异的marker
STR_markers=subset(STR_markers,!gene%in%subgene)

STR_markers_top10 <- STR_markers %>% group_by(cluster) %>% top_n(n = 10, wt = avg_log2FC)

DefaultAssay(STR)<-"RNA"

STR_markers_top10=unique(STR_markers_top10$gene)


jpeg("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/原图绘制/20230323/P6_1.jpeg",
     width = 9000,height = 2600,quality = 100,res = 600)

DotPlot(STR,features=STR_markers_top10,group.by = "cell_type0305")+
  theme(axis.text.x= element_text(size=9,angle = 45,vjust = 1,hjust = 1,face= "bold"))+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])))+xlab("")+ylab("")


dev.off()



#开始重新整合基质细胞【不要cycling】得到STR3------
library(Seurat)
library(SeuratDisk)
library(harmony)
library(dplyr)
library(tidyr)
library(ggplot2)
library(clusterProfiler)
library(org.Hs.eg.db)
library(RColorBrewer)
library(export)




umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")
DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type1110")
DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type",label = T)


#重新定义并且开始绘制原图
umap_ctrl_IUA_allsingle$cell_type0216=
  umap_ctrl_IUA_allsingle$cell_type1110

umap_ctrl_IUA_allsingle$cell_type0216=as.character(umap_ctrl_IUA_allsingle$cell_type0216)

DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type")

DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type0216",label = T)+ggtitle("")+theme(legend.position = "none")


umap_ctrl_IUA_allsingle=subset(umap_ctrl_IUA_allsingle,cell_type!="Erythrocyte")
DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type1110")
table(umap_ctrl_IUA_allsingle$cell_type0216)

STR_allsingle=subset(umap_ctrl_IUA_allsingle,cell_type1110%in%"Stromal cells")




STR_allsingle_embeddings=STR_allsingle$umap@cell.embeddings
STR_allsingle_embeddings=as.data.frame(STR_allsingle_embeddings)

STR_allsingle_embeddings$UMAP_1=as.numeric(as.character(STR_allsingle_embeddings$UMAP_1))
STR_allsingle_embeddings$UMAP_2=as.numeric(as.character(STR_allsingle_embeddings$UMAP_2))
STR_allsingle$UMAP_1=STR_allsingle_embeddings$UMAP_1
STR_allsingle$UMAP_2=STR_allsingle_embeddings$UMAP_2

STR_allsingle=subset(STR_allsingle,UMAP_1<6&UMAP_2>-7)

DimPlot(STR_allsingle)

RPGenes=c(grep('^RPL',rownames(STR_allsingle),value=TRUE),grep('^RPS',rownames(STR_allsingle),value=TRUE))

STR_allsingle[["percent.rp"]] <- PercentageFeatureSet(STR_allsingle, features = RPGenes)
VlnPlot(STR_allsingle, features = c("nFeature_RNA", "nCount_RNA", "percent.rp","percent.mt"), ncol = 4)

DimPlot(STR_allsingle,reduction="umap")

quantile(STR_allsingle$percent.rp)

FeaturePlot(STR_allsingle,features = "percent.rp",
            min.cutoff = 20,order = T,cols = c("grey","red"))

STR_allsingle=subset(STR_allsingle,percent.rp<25)

STR_maker=c("LUM","COL1A1","DCN","MMP11")
EPI_maker=c("EPCAM","KRT18","MMP7","WFDC2","CD9")
Endothelium_maker=c("VWF","CDH5")
smooth_maker=c("MCAM","RGS5","ACTA2")
T_maker=c("CD3D","CD3E","CXCR4","PTPRC","GZMK","CD4","CD8A")
NK_maker=c("NCAM1","GNLY","GZMA")
MAST_maker=c("MS4A2","TPSB2","TPSAB1","CPA3")
B_maker=c("CD79A","CD74")
macr_maker=c("HLA-DRB1","CSF1R","CD14")



#开始进行细胞的过滤【免疫细胞为主】
STR_allsingle_data=
  as.data.frame(STR_allsingle@assays$RNA@data)

STR_allsingle_data=subset(STR_allsingle_data,
                          rownames(STR_allsingle_data)%in%T_maker)

STR_allsingle_data=as.data.frame(t(STR_allsingle_data))


STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$PTPRC<=0)


STR_allsingle=STR_allsingle[,rownames(STR_allsingle_data)]

T_maker=c("CD3D","CD3E","CXCR4","PTPRC","GZMK","CD4","CD8A")
FeaturePlot(STR_allsingle,features = T_maker,min.cutoff = 0)

STR_allsingle_data=
  as.data.frame(STR_allsingle@assays$RNA@data)



STR_allsingle_data=
  as.data.frame(STR_allsingle@assays$RNA@data)

STR_allsingle_data=subset(STR_allsingle_data,
                          rownames(STR_allsingle_data)%in%c(EPI_maker,NK_maker,
                                                            Endothelium_maker,
                                                            smooth_maker,STR_maker,
                                                            T_maker,B_maker,macr_maker,
                                                            c("CD74","LYZ","WFDC2","PAEP","HLA-DRA","CD14")))

STR_allsingle_data=as.data.frame(t(STR_allsingle_data))


quantile(STR_allsingle_data$GNLY,0.9)###
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$GNLY<=0.95)


quantile(STR_allsingle_data$EPCAM,0.92)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$EPCAM<=0.5)


quantile(STR_allsingle_data$VWF,0.99)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$VWF<=0)

quantile(STR_allsingle_data$CXCR4,0.90)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$CXCR4<=0)

quantile(STR_allsingle_data$WFDC2,0.80)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$WFDC2<=0)

quantile(STR_allsingle_data$EPCAM,0.95)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$EPCAM<=0)

quantile(STR_allsingle_data$KRT18,0.95)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$KRT18<=0)


quantile(STR_allsingle_data$CD14,0.95)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$CD14<=0)

quantile(STR_allsingle_data$CD74,0.95)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$CD74<=0)

quantile(STR_allsingle_data$LYZ,0.95)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$LYZ<=0)

quantile(STR_allsingle_data$PAEP,0.95)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$PAEP<=0)


quantile(STR_allsingle_data$`HLA-DRA`,0.95)
STR_allsingle_data=subset(STR_allsingle_data,STR_allsingle_data$`HLA-DRA`<=0)


STR_allsingle=STR_allsingle[,rownames(STR_allsingle_data)]


table(STR_allsingle$type3)

STR_allsingle=subset(STR_allsingle,type3!="IUA7")


STR_allsingle=subset(STR_allsingle,cell_type%in%c("Stromal cells",
                                                  "Perivascular cells",
                                                  "Transient state cells"))

####开始进行重新整合
N=STR_allsingle@assays$RNA@counts

STR_allsingle=CreateSeuratObject(counts = N, meta.data = STR_allsingle@meta.data)

STR_allsingle=NormalizeData(STR_allsingle, normalization.method = "LogNormalize", 
                            scale.factor = 10000)
STR_allsingle <- FindVariableFeatures(STR_allsingle, selection.method = "vst", nfeatures = 2000)
VlnPlot(STR_allsingle, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)

mtGenes=grep('^MT-',rownames(STR_allsingle),value=TRUE)
RPGenes=c(grep('^RPL',rownames(STR_allsingle),value=TRUE),grep('^RPS',rownames(STR_allsingle),value=TRUE))
stress_gene2=grep('^HSP',rownames(STR_allsingle),value=TRUE)
HB_gene2=grep('^HB',rownames(STR_allsingle),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
stress_gene1=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
               "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/gene1.csv")
subgene3=subgene3$cellcyle_gene
subgene4=c("FOS","CXCL2","ZFP36","FOSB","DUSP1","ATF3","CXCL8","NR4A1","CXCL3","PPP1R15A",
           "JUNB","EGR1","HSPA1A","HSPA1B","SOCS3","KLF6","JUN","IER2","CXCL1","NFKBIA",
           "HSPA6","DNAJB1","IER3","CCNL1","MTRNR2L2","IER4","ID1","CEBPD","KRT6A","CYR61",
           "DEPP1","CLDN4","IRF1","DUSP2","BTG2","PLAUR","MAFF","KLF4","PHLDA2","TNFAIP3")
subgene5=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/512个应激反应相关基因.csv")
subgene5=subgene5$gene_symbol

subgene=c(mtGenes,RPGenes,stress_gene2,HB_gene2,HB.genes,s.genes,g2m.genes,stress_gene1,"MKI67",subgene3,subgene4,subgene5)

STR_allsingle@assays$RNA@var.features = setdiff(STR_allsingle@assays$RNA@var.features,subgene)  #去除subgene影响

STR_allsingle@assays$RNA@var.features

#STR_allsingle进行标准化然后PCA分析
DefaultAssay(STR_allsingle) <- "RNA"
all.genes_STR_allsingle <- rownames(STR_allsingle)
STR_allsingle <- ScaleData(object = STR_allsingle, features = all.genes_STR_allsingle)
STR_allsingle <- RunPCA(STR_allsingle, features = VariableFeatures(object = STR_allsingle))


#STR_allsingle进行细胞细胞聚类分析(UMAP)
STR_allsingle=RunUMAP(STR_allsingle, dims = 1:30)
DimPlot(STR_allsingle,reduction="umap",label=TRUE)
STR_allsingle <- FindNeighbors(STR_allsingle, dims = 1:30)
STR_allsingle <- FindClusters(STR_allsingle, resolution = 0.3)
STR_allsingle <- FindClusters(STR_allsingle, resolution = 0.6)
DimPlot(STR_allsingle,reduction="umap",label=TRUE)


#STR_allsingle进行细胞细胞聚类分析(tsne)
STR_allsingle=RunTSNE(STR_allsingle, dims = 1:30)
DimPlot(STR_allsingle,reduction="tsne",label=TRUE)
STR_allsingle <- FindNeighbors(STR_allsingle, dims = 1:30)
STR_allsingle <- FindClusters(STR_allsingle, resolution = 0.3)
DimPlot(STR_allsingle,reduction="tsne",label=TRUE)

DimPlot(STR_allsingle,reduction="umap",label=TRUE)



#开始尝试去除细胞样本批次并且进行新的一个保存

STR_allsingle=STR_allsingle%>%RunHarmony("type3",plot_covergence=T)

DefaultAssay(STR_allsingle) <- "RNA"


STR_allsingle <- RunPCA(STR_allsingle, features = VariableFeatures(object = STR_allsingle))


STR_allsingle=RunUMAP(STR_allsingle,reduction="harmony",dims=1:30)
STR_allsingle=FindNeighbors(STR_allsingle,dims=1:30,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.15,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.17,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.18,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.2,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.3,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.4,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.45,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.5,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.55,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.6,reduction = "harmony")




DimPlot(STR_allsingle,group.by = "cell_type",label=TRUE)


DimPlot(STR_allsingle,label=TRUE)



gene=c("BMP7","RGS5","APOD","ISG15","REV3L","THY1","PDGFRA","COL1A1","REV3L")

VlnPlot(STR_allsingle,features = gene,pt.size = 0)


#进行细胞类型的定义
STR_allsingle$cell_type0323="0323"

STR_allsingle$cell_type0323[STR_allsingle@active.ident%in%c("0","1")]<-"S0"
STR_allsingle$cell_type0323[STR_allsingle@active.ident%in%c("2","5")]<-"REV3L"
STR_allsingle$cell_type0323[STR_allsingle@active.ident%in%c("3")]<-"S1"
STR_allsingle$cell_type0323[STR_allsingle@active.ident%in%c("4")]<-"BMP7"
STR_allsingle$cell_type0323[STR_allsingle@active.ident%in%c("6")]<-"P"
STR_allsingle$cell_type0323[STR_allsingle@active.ident%in%c("7")]<-"ISG15"
STR_allsingle$cell_type0323[STR_allsingle@active.ident%in%c("8")]<-"APOD"

DimPlot(STR_allsingle,group.by = "cell_type0323",label = T)

STR_allsingle_embeddings=STR_allsingle$umap@cell.embeddings
STR_allsingle_embeddings=as.data.frame(STR_allsingle_embeddings)

STR_allsingle_embeddings$UMAP_1=as.numeric(as.character(STR_allsingle_embeddings$UMAP_1))
STR_allsingle_embeddings$UMAP_2=as.numeric(as.character(STR_allsingle_embeddings$UMAP_2))
STR_allsingle$UMAP_1=STR_allsingle_embeddings$UMAP_1
STR_allsingle$UMAP_2=STR_allsingle_embeddings$UMAP_2

STR_allsingle$cell_type0323[STR_allsingle$cell_type0323%in%"P"&STR_allsingle$UMAP_2<0]<-"S0"


STR_allsingle2=STR_allsingle
STR_allsingle2$cell_type0323[STR_allsingle2$cell_type0323%in%"REV3L"&STR_allsingle2$UMAP_2<1.8&STR_allsingle2$UMAP_1<2.3]<-"S1"

DimPlot(STR_allsingle2,group.by = "cell_type0323",label = T)


STR_allsingle2$cell_type0323=factor(STR_allsingle2$cell_type0323,
                                    levels = c('S0',"S1","BMP7","REV3L","ISG15","APOD","P"))

DimPlot(STR_allsingle2,group.by = "cell_type0323",label = T)+
  scale_color_manual(values =c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4"))+
  ggtitle("")


gene=c("BMP7","RGS5","APOD","ISG15","REV3L","THY1","PDGFRA","COL1A1","REV3L")

VlnPlot(STR_allsingle2,features = gene,pt.size = 0,group.by = "cell_type0323")
saveRDS(STR_allsingle2,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0323.rds")



#开始对STR3进行各类原图绘制-----
#对STR3原图绘制-----
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0323.rds")


#图1-----
DimPlot(STR,reduction="umap",label=F,label.size =7,pt.size = 0.6,group.by = "cell_type0323")+ggtitle("")+
  theme(title = element_text(size =0),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),axis.line = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 0))+theme(text = element_text(size=0))+
  scale_color_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"))



ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/原图绘制/20230323_2/P1-1.jpeg",
       width = 9,height = 6,dpi = 1000)


#图1-2
DimPlot(STR,reduction="umap",label=T,label.size =7,pt.size = 0.6,group.by = "cell_type0323")+ggtitle("")+
  theme(title = element_text(size =0),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),axis.line = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 0))+
  scale_color_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"))



ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/原图绘制/20230323_2/P1-2.jpeg",
       width = 9,height = 6,dpi = 1000)





#图2-------
gene=c("BMP7","RGS5","APOD","ISG15","REV3L","THY1","PDGFRA","COL1A1")

table(STR$cell_type0323)

#2-1
P1=VlnPlot(STR,features = gene[1],pt.size = 0,ncol = 1,group.by = "cell_type0323")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P2=VlnPlot(STR,features = gene[2],pt.size = 0,ncol = 1,group.by = "cell_type0323")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P3=VlnPlot(STR,features = gene[3],pt.size = 0,ncol = 1,group.by = "cell_type0323")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P4=VlnPlot(STR,features = gene[4],pt.size = 0,ncol = 1,group.by = "cell_type0323")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P5=VlnPlot(STR,features = gene[5],pt.size = 0,ncol = 1,group.by = "cell_type0323")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P6=VlnPlot(STR,features = gene[6],pt.size = 0,ncol = 1,group.by = "cell_type0323")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P7=VlnPlot(STR,features = gene[7],pt.size = 0,ncol = 1,group.by = "cell_type0323")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P8=VlnPlot(STR,features = gene[8],pt.size = 0,ncol = 1,group.by = "cell_type0323")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

ggarrange(P1,P2,P3,P4,P5,P6,P7,P8,common.legend = T,ncol = 4,legend = "none",nrow = 2)


jpeg("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/原图绘制/20230323_2/P2-1.jpeg",width =16000,height = 10000,res = 1200,
     quality = 100)
ggarrange(P1,P2,P3,P4,P5,P6,P7,P8,common.legend = F,ncol = 4,nrow = 2)+theme(legend.position = "none")
dev.off()

#2-2
P1=VlnPlot(STR,features = gene[1],pt.size = 0,ncol = 1,group.by = "cell_type0323")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P2=VlnPlot(STR,features = gene[2],pt.size = 0,ncol = 1,group.by = "cell_type0323")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P3=VlnPlot(STR,features = gene[3],pt.size = 0,ncol = 1,group.by = "cell_type0323")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P4=VlnPlot(STR,features = gene[4],pt.size = 0,ncol = 1,group.by = "cell_type0323")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P5=VlnPlot(STR,features = gene[5],pt.size = 0,ncol = 1,group.by = "cell_type0323")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P6=VlnPlot(STR,features = gene[6],pt.size = 0,ncol = 1,group.by = "cell_type0323")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P7=VlnPlot(STR,features = gene[7],pt.size = 0,ncol = 1,group.by = "cell_type0323")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P8=VlnPlot(STR,features = gene[8],pt.size = 0,ncol = 1,group.by = "cell_type0323")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

ggarrange(P1,P2,P3,P4,P5,P6,P7,P8,common.legend = T,ncol = 4,legend = "none",nrow = 2)


jpeg("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/原图绘制/20230323_2/P2-2.jpeg",width =20000,height = 10000,res = 1200,
     quality = 100)
ggarrange(P1,P2,P3,P4,P5,P6,P7,P8,common.legend = F,ncol = 4,nrow = 2)+theme(legend.position = "none")
dev.off()



#图3------
#图3-1
setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/原图绘制/20230323_2/")

STR$type4="type4"
STR$type4[STR$type2%in%"CTRL"]<-'Ctrl'
STR$type4[STR$type2%in%"IUA1"]<-'Adjacent'
STR$type4[STR$type2%in%"IUA2"]<-'Adhesion'


STR$type4=factor(STR$type4,levels = c("Ctrl","Adjacent","Adhesion"))

STR$cell_type0323_1=paste(STR$type2,"_",STR$cell_type0323,sep = "")

STR$cell_type0323_1=factor(STR$cell_type0323_1,levels = c(paste("CTRL_",names(table(STR$cell_type0323)),sep = ""),
                                                          paste("IUA1_",names(table(STR$cell_type0323)),sep = ""),
                                                          paste("IUA2_",names(table(STR$cell_type0323)),sep = "")))

av=AverageExpression(STR,
                     group.by = "cell_type0323_1",
                     assays = "RNA")

av=av[[1]]


cg=names(tail(sort(apply(av,1,sd)),2000))


View(av[cg,])

N=cor(av[cg,],method = 'spearman')

N=as.data.frame(N)

N=N[1:7,]
N=N[,-c(1:7)]

N$CTRL=1
N$IUA1=1
N$IUA2=1

N$IUA1[1]=N[1,1]
N$IUA1[2]=N[2,2]
N$IUA1[3]=N[3,3]
N$IUA1[4]=N[4,4]
N$IUA1[5]=N[5,5]
N$IUA1[6]=N[6,6]
N$IUA1[7]=N[7,7]


N$IUA2[1]=N[1,8]
N$IUA2[2]=N[2,9]
N$IUA2[3]=N[3,10]
N$IUA2[4]=N[4,11]
N$IUA2[5]=N[5,12]
N$IUA2[6]=N[6,13]
N$IUA2[7]=N[7,14]

N=N[15:17]
pheatmap(N,cluster_cols = F, display_numbers = TRUE,cluster_rows = F)

colnames(N)[2]="Adjacent"
colnames(N)[3]="Adhesion"

rownames(N)=gsub("CTRL_","",rownames(N))

pheatmap(N,cluster_cols = F, display_numbers = T,cluster_rows = F,
         fontsize = 23)

XX1=pheatmap(N,cluster_cols = F, display_numbers = F,cluster_rows = F,
             fontsize = 18,show_rownames = F,show_colnames = F,legend = F)

save_pheatmap_pdf <- function(x, filename, width=7000, height=7000) {
  stopifnot(!missing(x))
  stopifnot(!missing(filename))
  jpeg(filename, width=width, height=height)
  grid::grid.newpage()
  grid::grid.draw(x$gtable)
  dev.off()
}

save_pheatmap_pdf(XX1, "3-2.jpeg")


#2组间
STR$type4=factor(STR$type4,levels = c("Ctrl","Adjacent","Adhesion"))

STR$cell_type0323_2=paste(STR$type1,"_",STR$cell_type0323,sep = "")


STR$cell_type0323_2=factor(STR$cell_type0323_2,levels = c(paste("CTRL_",names(table(STR$cell_type0323)),sep = ""),
                                                          paste("IUA_",names(table(STR$cell_type0323)),sep = "")))

av=AverageExpression(STR,
                     group.by = "cell_type0323_2",
                     assays = "RNA")

av=av[[1]]


cg=names(tail(sort(apply(av,1,sd)),2000))


View(av[cg,])

N=cor(av[cg,],method = 'spearman')

N=as.data.frame(N)

N=N[1:7,]
N=N[,-c(1:7)]

N$CTRL=1
N$IUA=1


N$IUA[1]=N[1,1]
N$IUA[2]=N[2,2]
N$IUA[3]=N[3,3]
N$IUA[4]=N[4,4]
N$IUA[5]=N[5,5]
N$IUA[6]=N[6,6]
N$IUA[7]=N[7,7]



N=N[8:9]

pheatmap(N,cluster_cols = F, display_numbers = TRUE,cluster_rows = F)

rownames(N)=gsub("CTRL_","",rownames(N))


pheatmap(N,cluster_cols = F, display_numbers = TRUE,cluster_rows = F,
         fontsize = 25)

XX1=pheatmap(N,cluster_cols = F, display_numbers = F,cluster_rows = F,
             fontsize = 18,show_rownames = F,show_colnames = F,legend = F)

save_pheatmap_pdf <- function(x, filename, width=4000, height=7000) {
  stopifnot(!missing(x))
  stopifnot(!missing(filename))
  jpeg(filename, width=width, height=height)
  grid::grid.newpage()
  grid::grid.draw(x$gtable)
  dev.off()
}

save_pheatmap_pdf(XX1, "P4-2.jpeg")



#图5
av=AverageExpression(STR,
                     group.by = "cell_type0323",
                     assays = "RNA")

av=av[[1]]


cg=names(tail(sort(apply(av,1,sd)),2000))

N=cor(av[cg,],method = 'spearman')

N=as.data.frame(N)

pheatmap(N,cluster_cols = F, display_numbers = T,cluster_rows = F,
         fontsize = 20)

XX1=pheatmap(N,cluster_cols = F, display_numbers = F,cluster_rows = F,
             fontsize = 18,show_rownames = F,show_colnames = F,legend = F)

save_pheatmap_pdf <- function(x, filename, width=7000, height=7000) {
  stopifnot(!missing(x))
  stopifnot(!missing(filename))
  jpeg(filename, width=width, height=height)
  grid::grid.newpage()
  grid::grid.draw(x$gtable)
  dev.off()
}

save_pheatmap_pdf(XX1, "P5-2.jpeg")


#图6-----
STR@active.ident=STR$cell_type0323
STR_allsingle_markers <- FindAllMarkers(object=STR, only.pos = TRUE,min.pct = 0.25, 
                                        logfc.threshold = 0.25)
STR_allsingle_markers <-subset(STR_allsingle_markers,STR_allsingle_markers$p_val<0.05)


write.csv(STR_allsingle_markers,
          "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/原图绘制/20230323_2/STR_markers_cell_type0323.csv")



STR_markers=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/原图绘制/20230323_2/STR_markers_cell_type0323.csv")


mtGenes=grep('^MT-',rownames(STR),value=TRUE)
RPGenes=c(grep('^RPL',rownames(STR),value=TRUE),grep('^RPS',rownames(STR),value=TRUE))
stress_gene2=grep('^HSP',rownames(STR),value=TRUE)
HB_gene2=grep('^HB',rownames(STR),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
stress_gene1=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
               "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/gene1.csv")
subgene3=subgene3$cellcyle_gene
subgene4=c("FOS","CXCL2","ZFP36","FOSB","DUSP1","ATF3","CXCL8","NR4A1","CXCL3","PPP1R15A",
           "JUNB","EGR1","HSPA1A","HSPA1B","SOCS3","KLF6","JUN","IER2","CXCL1","NFKBIA",
           "HSPA6","DNAJB1","IER3","CCNL1","MTRNR2L2","IER4","ID1","CEBPD","KRT6A","CYR61",
           "DEPP1","CLDN4","IRF1","DUSP2","BTG2","PLAUR","MAFF","KLF4","PHLDA2","TNFAIP3")
subgene5=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/512个应激反应相关基因.csv")
subgene5=subgene5$gene_symbol

subgene=c(mtGenes,RPGenes,stress_gene2,HB_gene2,HB.genes,s.genes,g2m.genes,stress_gene1,"MKI67",subgene3,subgene4,subgene5)


#包含非特异的marker
STR_markers=subset(STR_markers,!gene%in%subgene)

STR_markers_top10 <- STR_markers %>% group_by(cluster) %>% top_n(n = 10, wt = avg_log2FC)

DefaultAssay(STR)<-"RNA"

STR_markers_top10=unique(STR_markers_top10$gene)


jpeg("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/原图绘制/20230323_2/P6_1.jpeg",
     width = 9000,height = 2600,quality = 100,res = 600)

DotPlot(STR,features=STR_markers_top10,group.by = "cell_type0323")+
  theme(axis.text.x= element_text(size=9,angle = 45,vjust = 1,hjust = 1,face= "bold"))+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])))+xlab("")+ylab("")


dev.off()


#特异基因气泡图
Z=STR_markers[duplicated(STR_markers$gene),]

N=subset(STR_markers,!gene%in%unique(Z$gene))

table(duplicated(N$gene))

STR_markers_top10 <- N %>% group_by(cluster) %>% top_n(n = 10, wt = avg_log2FC)

DefaultAssay(STR)<-"RNA"

STR_markers_top10=unique(STR_markers_top10$gene)

jpeg("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/原图绘制/20230323_2/P6_2.jpeg",
     width = 9000,height = 2600,quality = 100,res = 600)

DotPlot(STR,features=STR_markers_top10,group.by = "cell_type0323")+
  theme(axis.text.x= element_text(size=9,angle = 45,vjust = 1,hjust = 1,face= "bold"))+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])))+xlab("")+ylab("")


dev.off()



#20230325绘图-----
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

STR_allsingle=subset(STR,cell_type0305!="Cycling")


#STR_allsingle进行标准化然后PCA分析
DefaultAssay(STR_allsingle) <- "RNA"
all.genes_STR_allsingle <- rownames(STR_allsingle)
STR_allsingle <- ScaleData(object = STR_allsingle, features = all.genes_STR_allsingle)
STR_allsingle <- RunPCA(STR_allsingle, features = VariableFeatures(object = STR_allsingle))


#STR_allsingle进行细胞细胞聚类分析(UMAP)
STR_allsingle=RunUMAP(STR_allsingle, dims = 1:30)
DimPlot(STR_allsingle,reduction="umap",label=TRUE)
STR_allsingle <- FindNeighbors(STR_allsingle, dims = 1:30)
STR_allsingle <- FindClusters(STR_allsingle, resolution = 0.3)
STR_allsingle <- FindClusters(STR_allsingle, resolution = 0.6)
DimPlot(STR_allsingle,reduction="umap",label=TRUE)


#STR_allsingle进行细胞细胞聚类分析(tsne)
STR_allsingle=RunTSNE(STR_allsingle, dims = 1:30)
DimPlot(STR_allsingle,reduction="tsne",label=TRUE)
STR_allsingle <- FindNeighbors(STR_allsingle, dims = 1:30)
STR_allsingle <- FindClusters(STR_allsingle, resolution = 0.3)
DimPlot(STR_allsingle,reduction="tsne",label=TRUE)

DimPlot(STR_allsingle,reduction="umap",label=TRUE)



#开始尝试去除细胞样本批次并且进行新的一个保存

STR_allsingle=STR_allsingle%>%RunHarmony("type3",plot_covergence=T)

DefaultAssay(STR_allsingle) <- "RNA"


STR_allsingle <- RunPCA(STR_allsingle, features = VariableFeatures(object = STR_allsingle))


STR_allsingle=RunUMAP(STR_allsingle,reduction="harmony",dims=1:30)
STR_allsingle=FindNeighbors(STR_allsingle,dims=1:30,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.15,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.17,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.18,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.2,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.3,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.4,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.45,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.5,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.55,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.6,reduction = "harmony")

DimPlot(STR_allsingle,reduction="umap",label=TRUE,group.by = "cell_type0305")
DimPlot(STR_allsingle,reduction="umap",label=TRUE,group.by = "type3")
STR=STR_allsingle

DimPlot(STR,reduction="umap",label=TRUE,group.by = "cell_type0305")
saveRDS(STR,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0324.rds")

STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0324.rds")

#图1-----
STR$cell_type0323=factor(STR$cell_type0305,
                         levels = c('S0',"S1","BMP7","REV3L","ISG15","APOD","P"))

DimPlot(STR,label=F,label.size =7,pt.size = 0.6,group.by = "cell_type0305")+ggtitle("")+
  theme(title = element_text(size =0),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),axis.line = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 0))+theme(text = element_text(size=0))+
  scale_color_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"))



ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/原图绘制/20230325/P1-1.jpeg",
       width = 9,height = 6,dpi = 1000)


#图1-2
DimPlot(STR,reduction="umap",label=T,label.size =7,pt.size = 0.6,group.by = "cell_type0305")+ggtitle("")+
  theme(title = element_text(size =0),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),axis.line = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 0))+
  scale_color_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"))



ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/原图绘制/20230325/P1-2.jpeg",
       width = 9,height = 6,dpi = 1000)





#图2-------
gene=c("BMP7","RGS5","APOD","ISG15","REV3L","THY1","PDGFRA","COL1A1")

table(STR$cell_type0305)

#2-1
P1=VlnPlot(STR,features = gene[1],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P2=VlnPlot(STR,features = gene[2],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P3=VlnPlot(STR,features = gene[3],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P4=VlnPlot(STR,features = gene[4],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P5=VlnPlot(STR,features = gene[5],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P6=VlnPlot(STR,features = gene[6],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P7=VlnPlot(STR,features = gene[7],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P8=VlnPlot(STR,features = gene[8],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

ggarrange(P1,P2,P3,P4,P5,P6,P7,P8,common.legend = T,ncol = 4,legend = "none",nrow = 2)


jpeg("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/原图绘制/20230325/P2-1.jpeg",width =16000,height = 10000,res = 1200,
     quality = 100)
ggarrange(P1,P2,P3,P4,P5,P6,P7,P8,common.legend = F,ncol = 4,nrow = 2)+theme(legend.position = "none")
dev.off()

#2-2
P1=VlnPlot(STR,features = gene[1],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P2=VlnPlot(STR,features = gene[2],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P3=VlnPlot(STR,features = gene[3],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P4=VlnPlot(STR,features = gene[4],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P5=VlnPlot(STR,features = gene[5],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P6=VlnPlot(STR,features = gene[6],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P7=VlnPlot(STR,features = gene[7],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P8=VlnPlot(STR,features = gene[8],pt.size = 0,ncol = 1,group.by = "cell_type0305")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"),limits=c("S0","S1","BMP7","REV3L","ISG15","APOD","P"))+coord_flip()+
  theme(legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

ggarrange(P1,P2,P3,P4,P5,P6,P7,P8,common.legend = T,ncol = 4,legend = "none",nrow = 2)


jpeg("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/原图绘制/20230325/P2-2.jpeg",width =20000,height = 10000,res = 1200,
     quality = 100)
ggarrange(P1,P2,P3,P4,P5,P6,P7,P8,common.legend = F,ncol = 4,nrow = 2)+theme(legend.position = "none")
dev.off()



#图3------
#图3-1
setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/原图绘制/20230325/")
STR=STR_allsingle3

STR$type4="type4"
STR$type4[STR$type2%in%"CTRL"]<-'Ctrl'
STR$type4[STR$type2%in%"IUA1"]<-'Adjacent'
STR$type4[STR$type2%in%"IUA2"]<-'Adhesion'


STR$type4=factor(STR$type4,levels = c("Ctrl","Adjacent","Adhesion"))

STR$cell_type0305_1=paste(STR$type2,"_",STR$cell_type0305,sep = "")

STR$cell_type0305_1=factor(STR$cell_type0305_1,levels = c(paste("CTRL_",names(table(STR$cell_type0305)),sep = ""),
                                                          paste("IUA1_",names(table(STR$cell_type0305)),sep = ""),
                                                          paste("IUA2_",names(table(STR$cell_type0305)),sep = "")))

av=AverageExpression(STR,
                     group.by = "cell_type0305_1",
                     assays = "RNA")

av=av[[1]]


cg=names(tail(sort(apply(av,1,sd)),2000))


View(av[cg,])

N=cor(av[cg,],method = 'spearman')

N=as.data.frame(N)

N=N[1:7,]
N=N[,-c(1:7)]

N$CTRL=1
N$IUA1=1
N$IUA2=1

N$IUA1[1]=N[1,1]
N$IUA1[2]=N[2,2]
N$IUA1[3]=N[3,3]
N$IUA1[4]=N[4,4]
N$IUA1[5]=N[5,5]
N$IUA1[6]=N[6,6]
N$IUA1[7]=N[7,7]


N$IUA2[1]=N[1,8]
N$IUA2[2]=N[2,9]
N$IUA2[3]=N[3,10]
N$IUA2[4]=N[4,11]
N$IUA2[5]=N[5,12]
N$IUA2[6]=N[6,13]
N$IUA2[7]=N[7,14]

N=N[15:17]
pheatmap(N,cluster_cols = F, display_numbers = TRUE,cluster_rows = F)

colnames(N)[2]="Adjacent"
colnames(N)[3]="Adhesion"

rownames(N)=gsub("CTRL_","",rownames(N))

pheatmap(N,cluster_cols = F, display_numbers = T,cluster_rows = F,
         fontsize = 23)

XX1=pheatmap(N,cluster_cols = F, display_numbers = F,cluster_rows = F,
             fontsize = 18,show_rownames = F,show_colnames = F,legend = F)

save_pheatmap_pdf <- function(x, filename, width=7000, height=7000) {
  stopifnot(!missing(x))
  stopifnot(!missing(filename))
  jpeg(filename, width=width, height=height)
  grid::grid.newpage()
  grid::grid.draw(x$gtable)
  dev.off()
}

save_pheatmap_pdf(XX1, "P3-2.jpeg")


#2组间
STR$type4=factor(STR$type4,levels = c("Ctrl","Adjacent","Adhesion"))

STR$cell_type0305_2=paste(STR$type1,"_",STR$cell_type0305,sep = "")


STR$cell_type0305_2=factor(STR$cell_type0305_2,levels = c(paste("CTRL_",names(table(STR$cell_type0305)),sep = ""),
                                                          paste("IUA_",names(table(STR$cell_type0305)),sep = "")))

av=AverageExpression(STR,
                     group.by = "cell_type0305_2",
                     assays = "RNA")

av=av[[1]]


cg=names(tail(sort(apply(av,1,sd)),2000))


View(av[cg,])

N=cor(av[cg,],method = 'spearman')

N=as.data.frame(N)

N=N[1:7,]
N=N[,-c(1:7)]

N$CTRL=1
N$IUA=1


N$IUA[1]=N[1,1]
N$IUA[2]=N[2,2]
N$IUA[3]=N[3,3]
N$IUA[4]=N[4,4]
N$IUA[5]=N[5,5]
N$IUA[6]=N[6,6]
N$IUA[7]=N[7,7]



N=N[8:9]

pheatmap(N,cluster_cols = F, display_numbers = TRUE,cluster_rows = F)

rownames(N)=gsub("CTRL_","",rownames(N))


pheatmap(N,cluster_cols = F, display_numbers = TRUE,cluster_rows = F,
         fontsize = 25)

XX1=pheatmap(N,cluster_cols = F, display_numbers = F,cluster_rows = F,
             fontsize = 18,show_rownames = F,show_colnames = F,legend = F)

save_pheatmap_pdf <- function(x, filename, width=4000, height=7000) {
  stopifnot(!missing(x))
  stopifnot(!missing(filename))
  jpeg(filename, width=width, height=height)
  grid::grid.newpage()
  grid::grid.draw(x$gtable)
  dev.off()
}

save_pheatmap_pdf(XX1, "P4-2.jpeg")



#图5
av=AverageExpression(STR,
                     group.by = "cell_type2",
                     assays = "RNA")

av=av[[1]]


cg=names(tail(sort(apply(av,1,sd)),2000))

N=cor(av[cg,],method = 'spearman')

N=as.data.frame(N)

pheatmap(N, display_numbers = T,
         fontsize = 20)

pheatmap(N,display_numbers = T,cellwidth = 30, 
         cellheight = 30)

save_pheatmap_pdf <- function(x, filename, width=7000, height=7000) {
  jpeg(filename, width=width, height=height)
  grid::grid.newpage()
  grid::grid.draw(x$gtable)
  dev.off()
}

save_pheatmap_pdf(XX1, "P5-2.jpeg")

save_pheatmap_pdf <- function(x, filename, width=7, height=7) {
  stopifnot(!missing(x))
  stopifnot(!missing(filename))
  pdf(filename, width=width, height=height)
  dev.off()
}

save_pheatmap_pdf(XX1, "test.pdf")






#图6-----
DefaultAssay(STR)<-'RNA'

STR@active.ident=STR$cell_type0305
STR_allsingle_markers <- FindAllMarkers(object=STR, only.pos = TRUE,min.pct = 0.25, 
                                        logfc.threshold = 0.25)
STR_allsingle_markers <-subset(STR_allsingle_markers,STR_allsingle_markers$p_val<0.05)


write.csv(STR_allsingle_markers,
          "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/原图绘制/20230325/STR_markers_cell_type0305.csv")



STR_markers=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/原图绘制/20230325/STR_markers_cell_type0305.csv")


mtGenes=grep('^MT-',rownames(STR),value=TRUE)
RPGenes=c(grep('^RPL',rownames(STR),value=TRUE),grep('^RPS',rownames(STR),value=TRUE))
stress_gene2=grep('^HSP',rownames(STR),value=TRUE)
HB_gene2=grep('^HB',rownames(STR),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
stress_gene1=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
               "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/gene1.csv")
subgene3=subgene3$cellcyle_gene
subgene4=c("FOS","CXCL2","ZFP36","FOSB","DUSP1","ATF3","CXCL8","NR4A1","CXCL3","PPP1R15A",
           "JUNB","EGR1","HSPA1A","HSPA1B","SOCS3","KLF6","JUN","IER2","CXCL1","NFKBIA",
           "HSPA6","DNAJB1","IER3","CCNL1","MTRNR2L2","IER4","ID1","CEBPD","KRT6A","CYR61",
           "DEPP1","CLDN4","IRF1","DUSP2","BTG2","PLAUR","MAFF","KLF4","PHLDA2","TNFAIP3")
subgene5=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/512个应激反应相关基因.csv")
subgene5=subgene5$gene_symbol

subgene=c(mtGenes,RPGenes,stress_gene2,HB_gene2,HB.genes,s.genes,g2m.genes,stress_gene1,"MKI67",subgene3,subgene4,subgene5)


#包含非特异的marker
STR_markers=subset(STR_markers,!gene%in%subgene)

STR_markers_top5 <- STR_markers %>% group_by(cluster) %>% top_n(n = 5, wt = avg_log2FC)

DefaultAssay(STR)<-"RNA"

STR_markers_top5=unique(STR_markers_top5$gene)


jpeg("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/原图绘制/20230325/P6_1.jpeg",
     width = 6000,height = 2600,quality = 100,res = 600)

DotPlot(STR,features=STR_markers_top5,group.by = "cell_type0305")+
  theme(axis.text.x= element_text(size=9,angle = 45,vjust = 1,hjust = 1,face= "bold"))+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])))+xlab("")+ylab("")


dev.off()


#特异基因气泡图
Z=STR_markers[duplicated(STR_markers$gene),]

N=subset(STR_markers,!gene%in%unique(Z$gene))

table(duplicated(N$gene))

STR_markers_top5 <- N %>% group_by(cluster) %>% top_n(n = 5, wt = avg_log2FC)

DefaultAssay(STR)<-"RNA"

STR_markers_top5=unique(STR_markers_top5$gene)

jpeg("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/原图绘制/20230325/P6_2.jpeg",
     width = 6000,height = 2600,quality = 100,res = 600)

DotPlot(STR,features=STR_markers_top5,group.by = "cell_type0305")+
  theme(axis.text.x= element_text(size=9,angle = 45,vjust = 1,hjust = 1,face= "bold"))+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])))+xlab("")+ylab("")


dev.off()






#对STR_0324进行scvelo----
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony_0324/scvelo/ALL_ALL/")

STR$typenew=STR$cell_type0305


DimPlot(STR,group.by = "typenew",label = T,label.size = 7)+ggtitle("")


cellID_humanpost10x=Cells(STR)
cellID_humanpost10x=as.data.frame(cellID_humanpost10x)
colnames(cellID_humanpost10x)[1]<-"x"

cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"]=paste("H:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"]=paste("qian:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"]=paste("IUAshou:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"]=paste("IUAxia:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"]=paste("li:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"]=paste("IUAfeng:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"]=paste("IUA-GWY_additional:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"]=paste("IUA-BYQ:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"]=paste("IUA-WJM:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"],sep = "")
cellID_humanpost10x$x=gsub("-1_1","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_2","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_3","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_4","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_5","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_6","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_7","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_8","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_9","x",cellID_humanpost10x$x)

write.csv(cellID_humanpost10x, file = "cellID_humanpost10x.csv", row.names = FALSE)
#And finally we can extract our clusters with:
celltype=STR$typenew
celltype=as.data.frame(celltype)
rownames(celltype)=cellID_humanpost10x$x
write.csv(celltype, file = "celltype.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(STR, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")



rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"]=paste(rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"],"-0",sep = "")
rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"]=paste(rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"],"-1",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"],"-2",sep = "")
rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"]=paste(rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"],"-3",sep = "")
rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"]=paste(rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"],"-4",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"],"-5",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"],"-6",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"],"-7",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"],"-8",sep = "")
write.csv(celltype, file = "celltype2.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(STR, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")

DimPlot(STR,label = T,group.by = "typenew")+ggtitle("")










#对STR_harmony2进行一个scenic 
mydbDIR <- "/mnt/chentw/zigongneimo/photo/R-data(6sample)/all-single-cell(6samples)/SCENIC/"
mydbs <- c("hg38__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr.feather",
           "hg38__refseq-r80__10kb_up_and_down_tss.mc9nr.feather")
names(mydbs) <- c("500bp", "10kb")












#对STR进行一个scenic分析----
library(SCENIC)
library(SCopeLoomR)
library(pheatmap)
library(RcisTarget)
setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/SCENIC/ALL/")


STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")
STR@active.ident=STR$cell_type0305

STR1 <- sample(colnames(STR),1500)
STR <- STR[,STR1]

mydbDIR <- "/mnt/chentw/zigongneimo/photo/R-data(6sample)/all-single-cell(6samples)/SCENIC/"



mydbs <- c("hg38__refseq-r80__500bp_up_and_100bp_down_tss.mc9nr.feather",
           "hg38__refseq-r80__10kb_up_and_down_tss.mc9nr.feather")
names(mydbs) <- c("500bp", "10kb")
scenicOptions <- initializeScenic(org="hgnc", 
                                  nCores=8,
                                  dbDir=mydbDIR, 
                                  dbs = mydbs)

exprMat <- as.matrix(STR@assays$RNA@counts)

genesKept <- geneFiltering(exprMat, scenicOptions, 
                           minCountsPerGene = 3 * 0.01 * ncol(exprMat), 
                           minSamples = ncol(exprMat) * 0.01)
exprMat_filtered <- exprMat[genesKept, ]

###获取转录因子表
K=as.data.frame(motifAnnotations_hgnc$TF)

colnames(K)[1]<-"TF"

New<-K[!duplicated(K$TF),]

write.csv(New,"/data/DATAprocess/chentw/temp/2022/scenic.csv")


##计算相关性矩阵
runCorrelation(exprMat_filtered, scenicOptions)
##TF-Targets相关性回归分析
exprMat_filtered_log <- log2(exprMat_filtered+1)

runGenie3(exprMat_filtered_log, scenicOptions, nParts = 2)
head("int/1.4_GENIE3_linkList.Rds")
head(readRDS("int/1.4_GENIE3_linkList.Rds") )

scenicOptions@settings$verbose <- TRUE
scenicOptions@settings$nCores <- 1
scenicOptions@settings$seed <- 123

scenicOptions <- runSCENIC_1_coexNetwork2modules(scenicOptions) #1. 获取共表达模块
scenicOptions <- runSCENIC_2_createRegulons(scenicOptions)  #2. 获取regulons




###进行基质细胞内部的亚型间三组来源的cellchat差异性分析-------
library(CellChat)
###CTRL
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")
setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/20230704/")

STR=subset(STR,type1%in%"CTRL")

cellchat <- createCellChat(STR@assays$RNA@data,meta = STR@meta.data,group.by = "cell_type0305")
cellchat=setIdent(cellchat,ident.use = "cell_type0305")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
write.csv(df.net,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/20230704/STR_ctrl.csv")
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
write.csv(df.netp,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/20230704/STR_ctrl_netp.csv")
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Interaction weights/strength")
cellchat@netP$pathways
par(mfrow=c(1,1))
netVisual_heatmap(cellchat,signaling = "IGF",color.heatmap = "Reds")
netAnalysis_contribution(cellchat,signaling = "IGF")
pair.IGF=extractEnrichedLR(cellchat,signaling = "IGF",geneLR.return = F)
LR.show=pair.IGF[1,]
vertex.receiver=c(1,2,4,6)


netVisual_individual(cellchat,signaling = "IGF",pairLR.use = pair.IGF,layout = "circle")

netVisual_bubble(cellchat,remove.isolate = F)+
  theme_bw()+theme(panel.grid.major=element_blank(),panel.grid.minor=element_blank())


cellchat@netP$pathways


pairLR.use <- extractEnrichedLR(cellchat, signaling = c("TGFb"))

netVisual_bubble(cellchat, targets.use = 5,
                 pairLR.use = pairLR.use, remove.isolate = TRUE)


saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/20230704/STR_ctrl.rds")

##Adjacent
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")
setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/20230704/")
table(STR$type4)
STR=subset(STR,type4%in%"Adjacent")

cellchat <- createCellChat(STR@assays$RNA@data,meta = STR@meta.data,group.by = "cell_type0305")
cellchat=setIdent(cellchat,ident.use = "cell_type0305")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
write.csv(df.net,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/20230704/STR_Adjacent_net.csv")
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
write.csv(df.netp,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/20230704/STR_Adjacent_netp.csv")
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Interaction weights/strength")
cellchat@netP$pathways
par(mfrow=c(1,1))
netVisual_heatmap(cellchat,signaling = "IGF",color.heatmap = "Reds")
netAnalysis_contribution(cellchat,signaling = "IGF")
pair.IGF=extractEnrichedLR(cellchat,signaling = "IGF",geneLR.return = F)
LR.show=pair.IGF[1,]
vertex.receiver=c(1,2,4,6)


netVisual_individual(cellchat,signaling = "IGF",pairLR.use = pair.IGF,layout = "circle")

netVisual_bubble(cellchat,remove.isolate = F)+
  theme_bw()+theme(panel.grid.major=element_blank(),panel.grid.minor=element_blank())


cellchat@netP$pathways


pairLR.use <- extractEnrichedLR(cellchat, signaling = c("TGFb"))

netVisual_bubble(cellchat, targets.use = 5,
                 pairLR.use = pairLR.use, remove.isolate = TRUE)


saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/20230704/STR_Adjacent.rds")




##Adhesion
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")
setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/20230704/")
table(STR$type4)
STR=subset(STR,type4%in%"Adhesion")

cellchat <- createCellChat(STR@assays$RNA@data,meta = STR@meta.data,group.by = "cell_type0305")
cellchat=setIdent(cellchat,ident.use = "cell_type0305")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
write.csv(df.net,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/20230704/STR_adhesion_net.csv")
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
write.csv(df.netp,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/20230704/STR_adhesion_netp.csv")
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Interaction weights/strength")
cellchat@netP$pathways
par(mfrow=c(1,1))
netVisual_heatmap(cellchat,signaling = "IGF",color.heatmap = "Reds")
netAnalysis_contribution(cellchat,signaling = "IGF")
pair.IGF=extractEnrichedLR(cellchat,signaling = "IGF",geneLR.return = F)
LR.show=pair.IGF[1,]
vertex.receiver=c(1,2,4,6)


netVisual_individual(cellchat,signaling = "IGF",pairLR.use = pair.IGF,layout = "circle")

netVisual_bubble(cellchat,remove.isolate = F)+
  theme_bw()+theme(panel.grid.major=element_blank(),panel.grid.minor=element_blank())


cellchat@netP$pathways


pairLR.use <- extractEnrichedLR(cellchat, signaling = c("TGFb"))

netVisual_bubble(cellchat, targets.use = 5,
                 pairLR.use = pairLR.use, remove.isolate = TRUE)


saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/20230704/STR_adhesion.rds")




Ctrl=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/20230704/STR_ctrl.rds")
Ctrl$type4="Ctrl"

Adhesion=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/20230704/STR_adhesion.rds")
Adhesion$type4="Adhesion"

cc.list=list(Ctrl=Ctrl,Adhesion=Adhesion)
cellchat=mergeCellChat(cc.list,cell.prefix = T,add.names = names(cc.list))
##可视化
##所有细胞群总体观：通讯数量与强度对比
P1=compareInteractions(cellchat,show.legend = F,group = c(1,3),measure = "count")
P2=compareInteractions(cellchat,show.legend = F,group = c(1,3),measure = "weight")

P1+P2

##第一个图展示通讯数量之间的差异，第二个图展示通讯强度之间的差异。 

##数量与强度差异网络图
netVisual_diffInteraction(cellchat,weight.scale = T)
netVisual_diffInteraction(cellchat,weight.scale = T,measure = "weight")
##红色是case相对于control上调的，蓝色是下调的

#数量与强度差异热图
netVisual_heatmap(cellchat)
netVisual_heatmap(cellchat,measure = "weight")
#case和control对比，红色是上调，蓝色是下调

#保守和特异性信号通路的识别与可视化
rankNet(cellchat,mode = "comparison",stacked = T,do.stat = T)
rankNet(cellchat,mode = "comparison",stacked =F,do.stat = T)
##左图最下面多个信号通路是case组独有的


##细胞互作数量对比网络图
weight.max=getMaxWeight(cc.list,attribute = c("idents","count"))
netVisual_circle(cc.list[[1]]@net$count,weight.scale = T,label.edge = F,
                 edge.weight.max =weight.max[2],edge.width.max = 12,title.name = "sc11" )

netVisual_circle(cc.list[[2]]@net$count,weight.scale = T,label.edge = F,
                 edge.weight.max =weight.max[2],edge.width.max = 12,title.name = "sc3" )


plotGeneExpression(cellchat, signaling = 'TGFb',group.by = "type1")




cellchat@meta$cell_type0403=paste(cellchat@meta$type1,"_",cellchat@meta$cell_type0305,sep = "")
plotGeneExpression(cellchat, signaling = 'TGFb',group.by = "cell_type0305")

#开始一些图的绘制
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

DimPlot(STR,reduction="umap",label=T,label.size =7,pt.size = 0.6,group.by = "cell_type0305")+ggtitle("")+
  theme(title = element_text(size =0),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),axis.line = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 0))+
  scale_color_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"))



###差异性绘图                                                                                                          
Ctrl=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/20230704/STR_ctrl.rds")
Ctrl$type4="Ctrl"

Adjacent=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/20230704/STR_Adjacent.rds")
Adjacent$type4="Adjacent"


Adhesion=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/20230704/STR_adhesion.rds")
Adhesion$type4="Adhesion"

cc.list=list(Ctrl=Ctrl,Adjacent=Adjacent,Adhesion=Adhesion)
cellchat=mergeCellChat(cc.list,cell.prefix = T,add.names = names(cc.list))

weight.max=getMaxWeight(cc.list,attribute = c("idents","count"))
par(mfrow=c(1,3),xpd=T)
netVisual_circle(cc.list[[1]]@net$count,weight.scale = T,label.edge = T,
                 edge.weight.max =weight.max[2],edge.width.max = 12,title.name = "Ctrl" )

netVisual_circle(cc.list[[2]]@net$count,weight.scale = T,label.edge = T,
                 edge.weight.max =weight.max[2],edge.width.max = 12,title.name = "Adjacent" )

netVisual_circle(cc.list[[3]]@net$count,weight.scale = T,label.edge = T,
                 edge.weight.max =weight.max[2],edge.width.max = 12,title.name = "Adhesion" )


par(mfrow=c(1,3),xpd=T)
weight.max=getMaxWeight(cc.list,attribute = c("idents","weight"))
netVisual_circle(cc.list[[1]]@net$weight,weight.scale = T,label.edge = F,
                 edge.weight.max =weight.max[2],edge.width.max = 12,title.name = "Ctrl" )

netVisual_circle(cc.list[[2]]@net$weight,weight.scale = T,label.edge = F,
                 edge.weight.max =weight.max[2],edge.width.max = 12,title.name = "Adjacent" )

netVisual_circle(cc.list[[3]]@net$weight,weight.scale = T,label.edge = F,
                 edge.weight.max =weight.max[2],edge.width.max = 12,title.name = "Adhesion" )


cc.list=list(Ctrl=Ctrl,Adjcent=Adjacent,Adhesion=Adhesion)
cellchat=mergeCellChat(cc.list,cell.prefix = T,add.names = names(cc.list))

rankNet(cellchat,mode = "comparison",stacked =T,do.stat = T,comparison = c(1,2,3))+
  rankNet(cellchat,mode = "comparison",stacked =T,do.stat = T,comparison = c(1,2))

rankNet(cellchat,mode = "comparison",stacked =T,do.stat = T,comparison = c(1,3))
pathways.show <- c("TGFb") 

weight.max <- getMaxWeight(object.list, slot.name = c("netP"), attribute = pathways.show) # control the edge weights across different datasets
par(mfrow = c(1,3), xpd=TRUE)
for (i in 1:length(object.list)) {
  netVisual_aggregate(object.list[[i]], signaling = pathways.show, layout = "circle", edge.weight.max = weight.max[1], edge.width.max = 10, signaling.name = paste(pathways.show, names(object.list)[i]))
}


pathways.show <- c("PDGF") 
weight.max <- getMaxWeight(object.list, slot.name = c("netP"), attribute = pathways.show) # control the edge weights across different datasets
par(mfrow = c(1,3), xpd=TRUE)
for (i in 1:length(object.list)) {
  netVisual_aggregate(object.list[[i]], signaling = pathways.show, layout = "circle", edge.weight.max = weight.max[1], edge.width.max = 10, signaling.name = paste(pathways.show, names(object.list)[i]))
}


###对挑选的信号通路进行气泡图和和弦图的展示
####PDGF
object.list=list(Ctrl=Ctrl,Adjacent=Adjacent,Adhesion=Adhesion)
pathways.show <- c("PDGF") 
weight.max <- getMaxWeight(object.list, slot.name = c("netP"), attribute = pathways.show) # control the edge weights across different datasets
par(mfrow = c(1,3), xpd=TRUE)
for (i in 1:length(object.list)) {
  netVisual_aggregate(object.list[[i]],vertex.label.cex = 1.8, signaling = pathways.show, layout = "circle", edge.weight.max = weight.max[1], edge.width.max = 10, signaling.name = paste(pathways.show, names(object.list)[i]))
}

pairLR.use <- extractEnrichedLR(cellchat, signaling = c("PDGF"))
netVisual_bubble(cellchat,
                 pairLR.use = pairLR.use, remove.isolate = TRUE,comparison = c(1,2,3))


####TGFb
object.list=list(Ctrl=Ctrl,Adjacent=Adjacent,Adhesion=Adhesion)
pathways.show <- c("TGFb") 
weight.max <- getMaxWeight(object.list, slot.name = c("netP"), attribute = pathways.show) # control the edge weights across different datasets
par(mfrow = c(1,3), xpd=TRUE)
for (i in 1:length(object.list)) {
  netVisual_aggregate(object.list[[i]],vertex.label.cex = 1.8, signaling = pathways.show, layout = "circle", edge.weight.max = weight.max[1], edge.width.max = 10, signaling.name = paste(pathways.show, names(object.list)[i]))
}

pairLR.use <- extractEnrichedLR(cellchat, signaling = c("TGFb"))
netVisual_bubble(cellchat,
                 pairLR.use = pairLR.use, remove.isolate = TRUE,comparison = c(1,2,3))



####IL6
object.list=list(Ctrl=Ctrl,Adjacent=Adjacent,Adhesion=Adhesion)
pathways.show <- c("IL6") 
weight.max <- getMaxWeight(object.list, slot.name = c("netP"), attribute = pathways.show) # control the edge weights across different datasets
par(mfrow = c(1,3), xpd=TRUE)
for (i in 1:length(object.list)) {
  netVisual_aggregate(object.list[[i]],vertex.label.cex = 1.8, signaling = pathways.show, layout = "circle", edge.weight.max = weight.max[1], edge.width.max = 10, signaling.name = paste(pathways.show, names(object.list)[i]))
}

pairLR.use <- extractEnrichedLR(cellchat, signaling = c("IL6"))
netVisual_bubble(cellchat,
                 pairLR.use = pairLR.use, remove.isolate = TRUE,comparison = c(1,2,3))




####BMP
object.list=list(Ctrl=Ctrl,Adjacent=Adjacent,Adhesion=Adhesion)
pathways.show <- c("BMP") 
weight.max <- getMaxWeight(object.list, slot.name = c("netP"), attribute = pathways.show) # control the edge weights across different datasets
par(mfrow = c(1,3), xpd=TRUE)
for (i in 1:length(object.list)) {
  netVisual_aggregate(object.list[[i]],vertex.label.cex = 1.8,title.space = 22,signaling = pathways.show, layout = "circle", edge.weight.max = weight.max[1], edge.width.max = 10, signaling.name = paste(pathways.show, names(object.list)[i]))
}

pairLR.use <- extractEnrichedLR(cellchat, signaling = c("BMP"))
netVisual_bubble(cellchat,
                 pairLR.use = pairLR.use, remove.isolate = TRUE,comparison = c(1,2,3))



####ANGPT
object.list=list(Ctrl=Ctrl,Adhesion=Adhesion)
pathways.show <- c("ANGPT") 
weight.max <- getMaxWeight(object.list, slot.name = c("netP"), attribute = pathways.show) # control the edge weights across different datasets
par(mfrow = c(1,2), xpd=TRUE)
for (i in 1:length(object.list)) {
  netVisual_aggregate(object.list[[i]],vertex.label.cex = 1.8,title.space = 22,signaling = pathways.show, layout = "circle", edge.weight.max = weight.max[1], edge.width.max = 10, signaling.name = paste(pathways.show, names(object.list)[i]))
}


pairLR.use <- extractEnrichedLR(cellchat, signaling = c("ANGPT"))
netVisual_bubble(cellchat,
                 pairLR.use = pairLR.use, remove.isolate = TRUE,comparison = c(1,3))


####BMP
object.list=list(Ctrl=Ctrl,Adjacent=Adjacent,Adhesion=Adhesion)
pathways.show <- c("BMP") 
weight.max <- getMaxWeight(object.list, slot.name = c("netP"), attribute = pathways.show) # control the edge weights across different datasets
par(mfrow = c(1,3), xpd=TRUE)
for (i in 1:length(object.list)) {
  netVisual_aggregate(object.list[[i]],vertex.label.cex = 1.8,title.space = 22,signaling = pathways.show, layout = "circle", edge.weight.max = weight.max[1], edge.width.max = 10, signaling.name = paste(pathways.show, names(object.list)[i]))
}

pairLR.use <- extractEnrichedLR(cellchat, signaling = c("BMP"))
netVisual_bubble(cellchat,
                 pairLR.use = pairLR.use, remove.isolate = TRUE,comparison = c(1,2,3))


####CCL
object.list=list(Ctrl=Ctrl,Adjacent=Adjacent)
pathways.show <- c("CCL") 
weight.max <- getMaxWeight(object.list, slot.name = c("netP"), attribute = pathways.show) # control the edge weights across different datasets
par(mfrow = c(1,2), xpd=TRUE)
for (i in 1:length(object.list)) {
  netVisual_aggregate(object.list[[i]],vertex.label.cex = 1.8, signaling = pathways.show, layout = "circle", edge.weight.max = weight.max[1], edge.width.max = 10, signaling.name = paste(pathways.show, names(object.list)[i]))
}

pairLR.use <- extractEnrichedLR(cellchat, signaling = c("CCL"))
netVisual_bubble(cellchat,
                 pairLR.use = pairLR.use, remove.isolate = TRUE,comparison = c(1,2))



####LIFR
object.list=list(Adjacent=Adjacent,Adhesion=Adhesion)
pathways.show <- c("LIFR") 
weight.max <- getMaxWeight(object.list, slot.name = c("netP"), attribute = pathways.show) # control the edge weights across different datasets
par(mfrow = c(1,2), xpd=TRUE)
for (i in 1:length(object.list)) {
  netVisual_aggregate(object.list[[i]],vertex.label.cex = 1.8,title.space = 22,signaling = pathways.show, layout = "circle", edge.weight.max = weight.max[1], edge.width.max = 10, signaling.name = paste(pathways.show, names(object.list)[i]))
}

pairLR.use <- extractEnrichedLR(cellchat, signaling = c("LIFR"))
netVisual_bubble(cellchat,
                 pairLR.use = pairLR.use, remove.isolate = TRUE,comparison = c(1,2,3))





#对STR、cycling、巨噬细胞进行cellchat差异性分析-----
umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")

umap_ctrl_IUA_allsingle$type4="n"
umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type2%in%"CTRL"]<-"Ctrl"
umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type2%in%"IUA1"]<-"Adjacent"
umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type2%in%"IUA2"]<-"Adhesion"

#先分离出来巨噬细胞
MACR=subset(umap_ctrl_IUA_allsingle,cell_type%in%"Macrophage")
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")
Cycling=subset(STR,cell_type0305%in%"Cycling")
STR=subset(STR,cell_type0305!="Cycling")

MACR$type5="Macrophage"
STR$type5="STR"
Cycling$type5="Cycling"

ALL=merge(MACR,y=c(STR,Cycling))


###CTRL
setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/S_C_Macr/")

STR=subset(ALL,type1%in%"CTRL")

cellchat <- createCellChat(STR@assays$RNA@data,meta = STR@meta.data,group.by = "type5")
cellchat=setIdent(cellchat,ident.use = "type5")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
write.csv(df.net,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/S_C_Macr/STR_ctrl.csv")
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
write.csv(df.netp,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/S_C_Macr/STR_ctrl_netp.csv")
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Interaction weights/strength")

saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/S_C_Macr/STR_ctrl.rds")

##Adjacent
setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/S_C_Macr/")
table(STR$type4)
STR=subset(ALL,type4%in%"Adjacent")

cellchat <- createCellChat(STR@assays$RNA@data,meta = STR@meta.data,group.by = "type5")
cellchat=setIdent(cellchat,ident.use = "type5")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
write.csv(df.net,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/S_C_Macr/STR_Adjacent_net.csv")
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
write.csv(df.netp,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/S_C_Macr/STR_Adjacent_netp.csv")
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Interaction weights/strength")

saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/S_C_Macr/STR_Adjacent.rds")




##Adhesion
setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/S_C_Macr/")
table(STR$type4)
STR=subset(ALL,type4%in%"Adhesion")

cellchat <- createCellChat(STR@assays$RNA@data,meta = STR@meta.data,group.by = "type5")
cellchat=setIdent(cellchat,ident.use = "type5")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
write.csv(df.net,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/S_C_Macr/STR_adhesion_net.csv")
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
write.csv(df.netp,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/S_C_Macr/STR_adhesion_netp.csv")
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Interaction weights/strength")

saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/S_C_Macr/STR_adhesion.rds")



###对3组进行差异性绘图
Ctrl=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/S_C_Macr/STR_ctrl.rds")
Ctrl$type4="Ctrl"

Adjacent=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/S_C_Macr/STR_Adjacent.rds")
Adjacent$type4="Adjacent"


Adhesion=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/S_C_Macr/STR_adhesion.rds")
Adhesion$type4="Adhesion"

object.list=list(Ctrl=Ctrl,Adjacent=Adjacent,Adhesion=Adhesion)
cellchat=mergeCellChat(cc.list,cell.prefix = T,add.names = names(cc.list))

weight.max=getMaxWeight(cc.list,attribute = c("idents","count"))
par(mfrow=c(1,3),xpd=T)



for (i in 1:length(object.list)) {
  netVisual_circle(object.list[[i]]@net$count,weight.scale = T,label.edge = T,vertex.label.cex = 1.8,edge.label.cex = 1.8,
                   edge.weight.max =weight.max[2],edge.width.max = 12,title.name = names(object.list)[i])
}


par(mfrow=c(1,3),xpd=T)
weight.max=getMaxWeight(object.list,attribute = c("idents","weight"))
for (i in 1:length(object.list)) {
  netVisual_circle(object.list[[i]]@net$weight,weight.scale = T,label.edge = F,vertex.label.cex = 1.8,edge.label.cex = 1.8,
                   edge.weight.max =weight.max[2],edge.width.max = 12,title.name = names(object.list)[i])
}


rankNet(cellchat,mode = "comparison",stacked =T,do.stat = T,comparison = c(1,2,3))+
  rankNet(cellchat,mode = "comparison",stacked =T,do.stat = T,comparison = c(1,3))




###对挑选的信号通路进行气泡图和和弦图的展示
####CCL
object.list=list(Ctrl=Ctrl,Adjacent=Adjacent,Adhesion=Adhesion)
pathways.show <- c("CCL") 
weight.max <- getMaxWeight(object.list, slot.name = c("netP"), attribute = pathways.show) # control the edge weights across different datasets
par(mfrow = c(1,3), xpd=TRUE)
for (i in 1:length(object.list)) {
  netVisual_aggregate(object.list[[i]],vertex.label.cex = 1.8, signaling = pathways.show, layout = "circle", edge.weight.max = weight.max[1], edge.width.max = 10, signaling.name = paste(pathways.show, names(object.list)[i]))
}

pairLR.use <- extractEnrichedLR(cellchat, signaling = c("CCL"))
netVisual_bubble(cellchat,
                 pairLR.use = pairLR.use, remove.isolate = TRUE,comparison = c(1,2,3))




#对STR、cycling、内皮细胞进行cellchat差异性分析-----
umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")

umap_ctrl_IUA_allsingle$type4="n"
umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type2%in%"CTRL"]<-"Ctrl"
umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type2%in%"IUA1"]<-"Adjacent"
umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type2%in%"IUA2"]<-"Adhesion"

#先分离出来巨噬细胞
ENDO=subset(umap_ctrl_IUA_allsingle,cell_type%in%"Endothelial cells")
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")
Cycling=subset(STR,cell_type0305%in%"Cycling")
STR=subset(STR,cell_type0305!="Cycling")

ENDO$type5="Endo"
STR$type5="STR"
Cycling$type5="Cycling"

ALL=merge(ENDO,y=c(STR,Cycling))


###CTRL
setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/S_C_Endo/")

STR=subset(ALL,type1%in%"CTRL")

cellchat <- createCellChat(STR@assays$RNA@data,meta = STR@meta.data,group.by = "type5")
cellchat=setIdent(cellchat,ident.use = "type5")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
write.csv(df.net,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/S_C_Endo/STR_ctrl.csv")
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
write.csv(df.netp,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/S_C_Endo/STR_ctrl_netp.csv")
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Interaction weights/strength")

saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/S_C_Endo/STR_ctrl.rds")

##Adjacent
setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/S_C_Endo/")
table(STR$type4)
STR=subset(ALL,type4%in%"Adjacent")

cellchat <- createCellChat(STR@assays$RNA@data,meta = STR@meta.data,group.by = "type5")
cellchat=setIdent(cellchat,ident.use = "type5")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
write.csv(df.net,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/S_C_Endo/STR_Adjacent_net.csv")
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
write.csv(df.netp,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/S_C_Endo/STR_Adjacent_netp.csv")
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Interaction weights/strength")

saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/S_C_Endo/STR_Adjacent.rds")




##Adhesion
setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/S_C_Endo/")
table(STR$type4)
STR=subset(ALL,type4%in%"Adhesion")

cellchat <- createCellChat(STR@assays$RNA@data,meta = STR@meta.data,group.by = "type5")
cellchat=setIdent(cellchat,ident.use = "type5")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
write.csv(df.net,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/S_C_Endo/STR_adhesion_net.csv")
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
write.csv(df.netp,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/S_C_Endo/STR_adhesion_netp.csv")
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Interaction weights/strength")

saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/S_C_Endo/STR_adhesion.rds")



###对3组进行差异性绘图
Ctrl=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/S_C_Endo/STR_ctrl.rds")
Ctrl$type4="Ctrl"

Adjacent=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/S_C_Endo/STR_Adjacent.rds")
Adjacent$type4="Adjacent"


Adhesion=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/S_C_Endo/STR_adhesion.rds")
Adhesion$type4="Adhesion"


object.list=list(Ctrl=Ctrl,Adjacent=Adjacent,Adhesion=Adhesion)
cellchat=mergeCellChat(cc.list,cell.prefix = T,add.names = names(cc.list))

weight.max=getMaxWeight(cc.list,attribute = c("idents","count"))
par(mfrow=c(1,3),xpd=T)



for (i in 1:length(object.list)) {
  netVisual_circle(object.list[[i]]@net$count,weight.scale = T,label.edge = T,vertex.label.cex = 1.8,edge.label.cex = 1.8,
                   edge.weight.max =weight.max[2],edge.width.max = 12,title.name = names(object.list)[i])
}


par(mfrow=c(1,3),xpd=T)
weight.max=getMaxWeight(object.list,attribute = c("idents","weight"))
for (i in 1:length(object.list)) {
  netVisual_circle(object.list[[i]]@net$weight,weight.scale = T,label.edge = F,vertex.label.cex = 1.8,edge.label.cex = 1.8,
                   edge.weight.max =weight.max[2],edge.width.max = 12,title.name = names(object.list)[i])
}


rankNet(cellchat,mode = "comparison",stacked =T,do.stat = T,comparison = c(1,2,3))+
  rankNet(cellchat,mode = "comparison",stacked =T,do.stat = T,comparison = c(1,3))


###对挑选的信号通路进行气泡图和和弦图的展示
####VEGF
object.list=list(Ctrl=Ctrl,Adjacent=Adjacent,Adhesion=Adhesion)
pathways.show <- c("VEGF") 
weight.max <- getMaxWeight(object.list, slot.name = c("netP"), attribute = pathways.show) # control the edge weights across different datasets
par(mfrow = c(1,3), xpd=TRUE)
for (i in 1:length(object.list)) {
  netVisual_aggregate(object.list[[i]],vertex.label.cex = 1.8, signaling = pathways.show, layout = "circle", edge.weight.max = weight.max[1], edge.width.max = 10, signaling.name = paste(pathways.show, names(object.list)[i]))
}

pairLR.use <- extractEnrichedLR(cellchat, signaling = c("VEGF"))
netVisual_bubble(cellchat,
                 pairLR.use = pairLR.use, remove.isolate = TRUE,comparison = c(1,2,3))


####CCL
object.list=list(Adhesion=Adhesion)
pathways.show <- c("CCL") 
weight.max <- getMaxWeight(object.list, slot.name = c("netP"), attribute = pathways.show) # control the edge weights across different datasets
par(mfrow = c(1,1), xpd=TRUE)
for (i in 1:length(object.list)) {
  netVisual_aggregate(object.list[[i]],vertex.label.cex = 1.8, signaling = pathways.show, layout = "circle", edge.weight.max = weight.max[1], edge.width.max = 10, signaling.name = paste(pathways.show, names(object.list)[i]))
}



####PDGF
object.list=list(Ctrl=Ctrl,Adjacent=Adjacent,Adhesion=Adhesion)
pathways.show <- c("PDGF") 
weight.max <- getMaxWeight(object.list, slot.name = c("netP"), attribute = pathways.show) # control the edge weights across different datasets
par(mfrow = c(1,3), xpd=TRUE)
for (i in 1:length(object.list)) {
  netVisual_aggregate(object.list[[i]],vertex.label.cex = 1.8, signaling = pathways.show, layout = "circle", edge.weight.max = weight.max[1], edge.width.max = 10, signaling.name = paste(pathways.show, names(object.list)[i]))
}













#对所有细胞类型进行cellchat差异性分析-----
umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")

umap_ctrl_IUA_allsingle$type4="n"
umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type2%in%"CTRL"]<-"Ctrl"
umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type2%in%"IUA1"]<-"Adjacent"
umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type2%in%"IUA2"]<-"Adhesion"

table(umap_ctrl_IUA_allsingle$cell_type,umap_ctrl_IUA_allsingle$cell_type1110)

umap_ctrl_IUA_allsingle=subset(umap_ctrl_IUA_allsingle,cell_type1110%in%c("Immune cells","Epithelium cells","Endothelium cells"))

umap_ctrl_IUA_allsingle=subset(umap_ctrl_IUA_allsingle,cell_type!="Stromal cells")
umap_ctrl_IUA_allsingle=subset(umap_ctrl_IUA_allsingle,cell_type!="Cycling stromal cells")
umap_ctrl_IUA_allsingle=subset(umap_ctrl_IUA_allsingle,cell_type!="Perivascular cells")
umap_ctrl_IUA_allsingle=subset(umap_ctrl_IUA_allsingle,cell_type!="Transient state cells")



table(umap_ctrl_IUA_allsingle$cell_type,umap_ctrl_IUA_allsingle$cell_type1110)

STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")
table(STR$cell_type0305)
Cycling=subset(STR,cell_type0305%in%"Cycling")
STR=subset(STR,cell_type0305!="Cycling")

STR$cell_type="STR"
Cycling$cell_type="Cycling"

table(umap_ctrl_IUA_allsingle$cell_type)


ALL=merge(STR,y = c(Cycling,umap_ctrl_IUA_allsingle))

###CTRL
setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL3/")

STR=subset(ALL,type1%in%"CTRL")

cellchat <- createCellChat(STR@assays$RNA@data,meta = STR@meta.data,group.by = "cell_type")
cellchat=setIdent(cellchat,ident.use = "cell_type")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
write.csv(df.net,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL3/STR_ctrl.csv")
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
write.csv(df.netp,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL3/STR_ctrl_netp.csv")
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Interaction weights/strength")

saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL3/STR_ctrl.rds")

##Adjacent
setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL3/")
table(STR$type4)
STR=subset(ALL,type4%in%"Adjacent")

cellchat <- createCellChat(STR@assays$RNA@data,meta = STR@meta.data,group.by = "cell_type")
cellchat=setIdent(cellchat,ident.use = "cell_type")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
write.csv(df.net,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL3/STR_Adjacent_net.csv")
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
write.csv(df.netp,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL3/STR_Adjacent_netp.csv")
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Interaction weights/strength")

saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL3/STR_Adjacent.rds")




##Adhesion
setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL3/")
table(STR$type4)
STR=subset(ALL,type4%in%"Adhesion")

cellchat <- createCellChat(STR@assays$RNA@data,meta = STR@meta.data,group.by = "cell_type")
cellchat=setIdent(cellchat,ident.use = "cell_type")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
write.csv(df.net,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL3/STR_adhesion_net.csv")
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
write.csv(df.netp,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL3/STR_adhesion_netp.csv")
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Interaction weights/strength")

saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL3/STR_adhesion.rds")



###对3组进行差异性绘图
Ctrl=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL3/STR_ctrl.rds")
Ctrl$type4="Ctrl"

Adjacent=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL3/STR_Adjacent.rds")
Adjacent$type4="Adjacent"


Adhesion=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/cellchat/ALL3/STR_adhesion.rds")
Adhesion$type4="Adhesion"


object.list=list(Ctrl=Ctrl,Adjacent=Adjacent,Adhesion=Adhesion)
cellchat=mergeCellChat(object.list,cell.prefix = T,add.names = names(object.list))

weight.max=getMaxWeight(object.list,attribute = c("idents","count"))
par(mfrow=c(1,3),xpd=T)



for (i in 1:length(object.list)) {
  netVisual_circle(object.list[[i]]@net$count,weight.scale = T,label.edge = T,vertex.label.cex = 1.8,edge.label.cex = 1.8,
                   edge.weight.max =weight.max[2],edge.width.max = 12,title.name = names(object.list)[i])
}


par(mfrow=c(1,3),xpd=T)
weight.max=getMaxWeight(object.list,attribute = c("idents","weight"))
for (i in 1:length(object.list)) {
  netVisual_circle(object.list[[i]]@net$weight,weight.scale = T,label.edge = F,vertex.label.cex = 1.8,edge.label.cex = 1.8,
                   edge.weight.max =weight.max[2],edge.width.max = 12,title.name = names(object.list)[i])
}


rankNet(cellchat,mode = "comparison",stacked =T,do.stat = T,comparison = c(1,2,3))+
  rankNet(cellchat,mode = "comparison",stacked =T,do.stat = T,comparison = c(1,3))


DimPlot(ALL,group.by = "cell_type",
        label = T)+ggtitle("")

par(mfrow = c(1,2), xpd=TRUE)
netVisual_diffInteraction(cellchat, weight.scale = T,comparison = c(1,3))
netVisual_diffInteraction(cellchat, weight.scale = T, measure = "weight",comparison = c(1,3))


####CCL
object.list=list(Ctrl=Ctrl,Adjacent=Adjacent,Adhesion=Adhesion)
pathways.show <- c("CCL") 
weight.max <- getMaxWeight(object.list, slot.name = c("netP"), attribute = pathways.show) # control the edge weights across different datasets
par(mfrow = c(1,3), xpd=TRUE)
for (i in 1:length(object.list)) {
  netVisual_aggregate(object.list[[i]],vertex.label.cex = 1.8,title.space = 22,signaling = pathways.show, layout = "circle", edge.weight.max = weight.max[1], edge.width.max = 10, signaling.name = paste(pathways.show, names(object.list)[i]))
}



####PDGF
object.list=list(Ctrl=Ctrl,Adjacent=Adjacent,Adhesion=Adhesion)
pathways.show <- c("PDGF") 
weight.max <- getMaxWeight(object.list, slot.name = c("netP"), attribute = pathways.show) # control the edge weights across different datasets
par(mfrow = c(1,3), xpd=TRUE)
for (i in 1:length(object.list)) {
  netVisual_aggregate(object.list[[i]],vertex.label.cex = 1.8,title.space = 22,signaling = pathways.show, layout = "circle", edge.weight.max = weight.max[1], edge.width.max = 10, signaling.name = paste(pathways.show, names(object.list)[i]))
}



####VEGF
object.list=list(Ctrl=Ctrl,Adjacent=Adjacent,Adhesion=Adhesion)
pathways.show <- c("VEGF") 
weight.max <- getMaxWeight(object.list, slot.name = c("netP"), attribute = pathways.show) # control the edge weights across different datasets
par(mfrow = c(1,3), xpd=TRUE)
for (i in 1:length(object.list)) {
  netVisual_aggregate(object.list[[i]],vertex.label.cex = 1.8,title.space = 22,signaling = pathways.show, layout = "circle", edge.weight.max = weight.max[1], edge.width.max = 10, signaling.name = paste(pathways.show, names(object.list)[i]))
}





####分CTRL,ADJACENT,ADHESION的3组的桑葚图的绘制----------
umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")

umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type2%in%"CTRL"]<-"Ctrl"
umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type2%in%"IUA1"]<-"Adjacent"
umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type2%in%"IUA2"]<-"Adhesion"




View(umap_ctrl_IUA_allsingle@meta.data)

VlnPlot(umap_ctrl_IUA_allsingle, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)

umap_quality_plot=umap_ctrl_IUA_allsingle@meta.data[c("nFeature_RNA","nCount_RNA","type4")]


umap_quality_plot_P1=as.data.frame(table(umap_quality_plot$type4))

P1=ggplot(data=umap_quality_plot_P1)+aes(x=Var1,y=Freq,fill=Var1)+geom_bar(stat="identity",width=0.85)+
  scale_y_continuous(expand=c(0,0))+
  theme_bw()+theme(panel.grid.major=element_blank(),panel.grid.minor=element_blank())+
  theme(legend.position="none")+theme(panel.border = element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank(),
                                      axis.ticks.y = element_blank())+  
  theme(axis.line = element_line(size=0.4, colour = "black"))+coord_flip()+
  scale_x_discrete(limits=c("Ctrl","Adjacent","Adhesion"))+
  ylab("cells")+xlab("")+scale_fill_manual(values=c("#3B6241","#797592","#596C94","#363F70","#A3C5B8"))



umap_quality_plot_P2=umap_ctrl_IUA_allsingle@meta.data[c("nFeature_RNA","type4")]

P2=ggplot(data=umap_quality_plot_P2)+aes(x=nFeature_RNA,y=type4,fill=type4)+
  geom_violin()+theme_bw()+ylab("")+theme(legend.position = "none")+xlab("Genes")+
  geom_boxplot(width=.1,col="black",fill="black")+scale_fill_manual(values=c("#A3C5B8","#363F70","#596C94","#797592","#3B6241"))+
  scale_y_discrete(limits=c("Ctrl","Adjacent","Adhesion"))+
  theme(axis.text.y =element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank(),axis.ticks.y = element_blank())






umap_quality_plot_P3=umap_ctrl_IUA_allsingle@meta.data[c("nCount_RNA","type4")]

P3=ggplot(data=umap_quality_plot_P3)+aes(x=nCount_RNA,y=type4,fill=type4)+
  geom_violin()+theme_bw()+ylab("")+theme(legend.position = "none")+xlab("UMI")+
  geom_boxplot(width=.1,col="black",fill="black")+scale_fill_manual(values=c("#A3C5B8","#363F70","#596C94","#797592","#3B6241"))+
  scale_y_discrete(limits=c("Ctrl","Adjacent","Adhesion"))+
  theme(axis.text.y =element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank(),axis.ticks.y = element_blank())




ggarrange(P1,P2,P3,ncol = 3)



###桑葚图的绘制
table(umap_ctrl_IUA_allsingle$type3)

umap_ctrl_IUA_allsingle$type4=factor(umap_ctrl_IUA_allsingle$type4,
                                     levels = c("Adhesion","Adjacent","Ctrl"))

umap_ctrl_IUA_allsingle$type3=factor(umap_ctrl_IUA_allsingle$type3,
                                     levels = c("IUA7","IUA6","IUA5","IUA4",
                                                "IUA3","IUA2","IUA1","CTRL2","CTRL1"))


P4_table=as.data.frame(table(umap_ctrl_IUA_allsingle$type3,umap_ctrl_IUA_allsingle$type4))
P4_table=subset(P4_table,P4_table$Freq!=0)
P4_table=P4_table[1:2]

colnames(P4_table)[1]<-"sample"
colnames(P4_table)[2]<-"Group"



P4_table=rbind(P4_table,P4_table[P4_table$sample%in%c("IUA7"),])

P4_table$link <- 1
P4_table <- reshape::melt(P4_table, id = 'link')

variable <- summary(P4_table$variable)
P4_table$flow <- rep(1:variable[1], length(variable))


P4=ggplot(P4_table, aes(x = variable, y = link,
                        stratum = value, alluvium = flow, fill = value)) +
  geom_stratum() +  
  geom_flow(aes.flow = 'forward')+theme_bw()+
  labs(x = '', y = '') +
  scale_fill_manual(values = c("#F8766C","#E38900",
                               "#C49A00","#99A800","#52B400",
                               "#00BC56","#00C094","#00BFC4","#00B6EB",
                               "#A3C5B8","#363F70","#596C94","#797592","#3B6241"))+
  scale_y_continuous(expand=c(0,0))+geom_text(stat = 'stratum', infer.label = TRUE, size = 5.5) + 
  theme(title = element_text(size = 25),legend.key.size = unit(1,'cm'),
        legend.position = "none",panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.title = element_blank(),axis.text = element_blank(),
        axis.ticks = element_blank(),axis.line = element_blank())


P1=ggarrange(P4,P1,P2,P3,ncol = 4,widths = c(3,2,1,1))


saveRDS(P1,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/PPT文件/P1.rds")


#进行原图中的指控的图的绘制2
VlnPlot(umap_ctrl_IUA_allsingle, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)

umap_quality_plot=umap_ctrl_IUA_allsingle@meta.data[c("nFeature_RNA","nCount_RNA","type4")]


umap_quality_plot_P1=as.data.frame(table(umap_quality_plot$type4))

P1=ggplot(data=umap_quality_plot_P1)+aes(x=Var1,y=Freq,fill=Var1)+geom_bar(stat="identity",width=0.85)+
  scale_y_continuous(expand=c(0,0))+
  theme_bw()+theme(panel.grid.major=element_blank(),panel.grid.minor=element_blank())+
  theme(legend.position="none")+theme(panel.border = element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank(),
                                      axis.ticks.y = element_blank())+  
  theme(axis.line = element_line(size=0.4, colour = "black"),text = element_blank())+coord_flip()+
  scale_x_discrete(limits=c("Adhesion","Adjacent","Ctrl"))+
  ylab("cells")+xlab("")+scale_fill_manual(values=c("#2A6343","#7A7592","#526D94"))




umap_quality_plot_P2=umap_ctrl_IUA_allsingle@meta.data[c("nFeature_RNA","type4")]

P2=ggplot(data=umap_quality_plot_P2)+aes(x=nFeature_RNA,y=type4,fill=type4)+
  geom_violin()+theme_bw()+ylab("")+theme(legend.position = "none",text = element_blank())+xlab("Genes")+
  geom_boxplot(width=.1,col="black",fill="black")+scale_fill_manual(values=c("#2A6343","#7A7592","#526D94"))+
  scale_y_discrete(limits=c("Adhesion","Adjacent","Ctrl"))+
  theme(axis.text.y =element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank(),axis.ticks.y = element_blank())






umap_quality_plot_P3=umap_ctrl_IUA_allsingle@meta.data[c("nCount_RNA","type4")]

P3=ggplot(data=umap_quality_plot_P3)+aes(x=nCount_RNA,y=type4,fill=type4)+
  geom_violin()+theme_bw()+ylab("")+theme(legend.position = "none",text = element_blank())+xlab("UMI")+
  geom_boxplot(width=.1,col="black",fill="black")+scale_fill_manual(values=c("#2A6343","#7A7592","#526D94"))+
  scale_y_discrete(limits=c("Adhesion","Adjacent","Ctrl"))+
  theme(axis.text.y =element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank(),axis.ticks.y = element_blank())




ggarrange(P1,P2,P3,ncol = 3)



###桑葚图的绘制
table(umap_ctrl_IUA_allsingle$type3)

umap_ctrl_IUA_allsingle$type4=factor(umap_ctrl_IUA_allsingle$type4,
                                     levels = c("Adhesion","Adjacent","Ctrl"))

umap_ctrl_IUA_allsingle$type3=factor(umap_ctrl_IUA_allsingle$type3,
                                     levels = c("IUA7","IUA6","IUA5","IUA4",
                                                "IUA3","IUA2","IUA1","CTRL2","CTRL1"))


P4_table=as.data.frame(table(umap_ctrl_IUA_allsingle$type3,umap_ctrl_IUA_allsingle$type4))
P4_table=subset(P4_table,P4_table$Freq!=0)
P4_table=P4_table[1:2]

colnames(P4_table)[1]<-"sample"
colnames(P4_table)[2]<-"Group"



P4_table=rbind(P4_table,P4_table[P4_table$sample%in%c("IUA7"),])

P4_table$link <- 1
P4_table <- reshape::melt(P4_table, id = 'link')

variable <- summary(P4_table$variable)
P4_table$flow <- rep(1:variable[1], length(variable))


P4=ggplot(P4_table, aes(x = variable, y = link,
                        stratum = value, alluvium = flow, fill = value)) +
  geom_stratum() +  
  geom_flow(aes.flow = 'forward')+theme_bw()+
  labs(x = '', y = '') +
  scale_fill_manual(values = c("#F8766C","#E38900",
                               "#C49A00","#99A800","#52B400",
                               "#00BC56","#00C094","#00BFC4","#00B6EB",
                               "#526D94","#7A7592","#2A6343"))+
  scale_y_continuous(expand=c(0,0))+ 
  theme(title = element_text(size = 25),legend.key.size = unit(1,'cm'),
        legend.position = "none",panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.title = element_blank(),axis.text = element_blank(),
        axis.ticks = element_blank(),axis.line = element_blank(),text = element_blank())


ggarrange(P4,P1,P2,P3,ncol = 4,widths = c(3,2,1,1))

ggsave("/data/DATAprocess/chentw/zigongneimo/原图绘制/P1.jpeg",width = 17,height = 6,
       dpi = 800)


####对基质细胞中的S0,S1,REV3L进行monocle2的伪时间分析-----
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

table(STR$cell_type0305)

STR=subset(STR,cell_type0305%in%c("S0","S1","REV3L"))

DefaultAssay(STR) <- "RNA"
all.genes.STR <- rownames(STR)
STR <- ScaleData(object = STR, features = all.genes.STR)

DefaultAssay(STR) <- "RNA"




DefaultAssay(STR)<-"RNA"
STR_matrix<-as(as.matrix(GetAssayData(STR,slot = "counts")),'sparseMatrix')
STR_ann<-data.frame(gene_id=rownames(STR_matrix),gene_short_name=rownames(STR_matrix))
rownames(STR_ann)<-rownames(STR_matrix)

STR_fd<-new("AnnotatedDataFrame", data = STR_ann)

sample_ann<-STR@meta.data
rownames(sample_ann)<-colnames(STR_matrix)

STR_pd<-new("AnnotatedDataFrame", data =sample_ann)

STR.cds<-newCellDataSet(STR_matrix,phenoData =STR_pd,featureData =STR_fd,expressionFamily=negbinomial.size())


#预处理
STR.cds <- estimateSizeFactors(STR.cds)
STR.cds <- estimateDispersions(STR.cds)

disp_table <- dispersionTable(STR.cds)
unsup_clustering_genes <- subset(disp_table, mean_expression >= 0.1)
STR.cds <- setOrderingFilter(STR.cds, unsup_clustering_genes$gene_id)

STR.cds <- reduceDimension(
  STR.cds,
  max_components = 2,
  method = 'DDRTree')

STR.cds <- orderCells(STR.cds)
head(pData(STR.cds))


STR_P1=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "cell_type0305",label=T)+ggtitle("STR")

STR_P2=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "Pseudotime")

STR_P3=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "type2",label=T)+ggtitle("STR")



STR_P1+STR_P2


STR_P2=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "Pseudotime")


plot_cell_trajectory(STR.cds,cell_size = 3,color_by = "cell_type0305",label=T)+ggtitle("")+
  facet_wrap(~type2, nrow = 1)




saveRDS(STR.cds,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/STR0705.cds")

####对基质细胞中的S0,S1,REV3L进行velocyto的伪时间分析-----
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

table(STR$cell_type0305)

STR0313=subset(STR,cell_type0305%in%c("S0","S1","REV3L"))

DefaultAssay(STR0313) <- "RNA"
all.genes.STR0313 <- rownames(STR0313)
STR0313 <- ScaleData(object = STR0313, features = all.genes.STR0313)

DefaultAssay(STR0313) <- "RNA"

setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/velocyto/S0_S1_REV3L/ALL/")


cellID_humanpost10x=Cells(STR0313)
cellID_humanpost10x=as.data.frame(cellID_humanpost10x)
colnames(cellID_humanpost10x)[1]<-"x"

cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"]=paste("H:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"]=paste("qian:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"]=paste("IUAshou:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"]=paste("IUAxia:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"]=paste("li:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"]=paste("IUAfeng:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"]=paste("IUA-GWY_additional:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"]=paste("IUA-BYQ:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"]=paste("IUA-WJM:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"],sep = "")
cellID_humanpost10x$x=gsub("-1_1","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_2","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_3","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_4","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_5","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_6","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_7","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_8","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_9","x",cellID_humanpost10x$x)

write.csv(cellID_humanpost10x, file = "cellID_humanpost10x.csv", row.names = FALSE)
#And finally we can extract our clusters with:
celltype=STR0313$cell_type0305
celltype=as.data.frame(celltype)
rownames(celltype)=cellID_humanpost10x$x
write.csv(celltype, file = "celltype.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(STR0313, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")



rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"]=paste(rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"],"-0",sep = "")
rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"]=paste(rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"],"-1",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"],"-2",sep = "")
rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"]=paste(rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"],"-3",sep = "")
rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"]=paste(rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"],"-4",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"],"-5",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"],"-6",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"],"-7",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"],"-8",sep = "")
write.csv(celltype, file = "celltype2.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(STR0313, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")

DimPlot(STR0313,label = T,group.by = "cell_type0305")+ggtitle("")


###统计7组细胞类型间差异基因的数目以及汇总[分上下调]-UPsetR包绘图-----
library(UpSetR)
library(RColorBrewer)

###上调
S0_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/S0/S0_deseq.csv")
S0=subset(S0_deseq,S0_deseq$avg_log2FC>0)$X

S1_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/S1/S1_deseq.csv")
S1=subset(S1_deseq,S1_deseq$avg_log2FC>0)$X

REV3L_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/REV3L/REV3L_deseq.csv")
REV3L=subset(REV3L_deseq,REV3L_deseq$avg_log2FC>0)$X


APOD_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/APOD/APOD_deseq.csv")
APOD=subset(APOD_deseq,APOD_deseq$avg_log2FC>0)$X

BMP7_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/BMP7/BMP7_deseq.csv")
BMP7=subset(BMP7_deseq,BMP7_deseq$avg_log2FC>0)$X


ISG15_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/ISG15/ISG15_deseq.csv")
ISG15=subset(ISG15_deseq,ISG15_deseq$avg_log2FC>0)$X


P_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/P/P_deseq.csv")
P=subset(P_deseq,P_deseq$avg_log2FC>0)$X


Cycling_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/Cycling/Cycling_deseq.csv")
Cycling=subset(Cycling_deseq,Cycling_deseq$avg_log2FC>0)$X


set_list <- list(S0=S0,S1=S1,BMP7=BMP7,Cycling=Cycling,REV3L=REV3L,ISG15=ISG15,P=P)

pdf(file = "/data/DATAprocess/chentw/zigongneimo/temp/STR3.pdf",
    width = 10,height = 6)

upset(
  fromList(set_list),
  order.by = "freq",nsets = 7,shade.color="red",
  sets.bar.color=c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800")
)

dev.off()

gene_interset=Reduce(intersect,list(S0,S1,BMP7,Cycling,REV3L,ISG15,APOD,P))
gene_interset


###根据这些共有的25个基因去除MT基因以后进行六组间的评分
gene_interset=gene_interset[-which(gene_interset%in%c("MT-ND5","MT-ND2"))]

STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

STR_data=as.data.frame(STR@assays$RNA@data)

STR_data=subset(STR_data,rownames(STR_data)%in%c(gene_interset))

STR_data=as.data.frame(t(STR_data))

STR_data$mean=apply(STR_data,1,mean)

STR_data$typ2=STR$type4
STR$gene=STR_data$mean


VlnPlot(STR,features = "gene",pt.size = 0,split.by = "type4",group.by = "cell_type0305")+ggtitle("")+
  xlab("")+ylab("Fibrosis Score")





###下调
S0_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/S0/S0_deseq.csv")
S0=subset(S0_deseq,S0_deseq$avg_log2FC<0)$X

S1_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/S1/S1_deseq.csv")
S1=subset(S1_deseq,S1_deseq$avg_log2FC<0)$X

REV3L_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/REV3L/REV3L_deseq.csv")
REV3L=subset(REV3L_deseq,REV3L_deseq$avg_log2FC<0)$X


APOD_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/APOD/APOD_deseq.csv")
APOD=subset(APOD_deseq,APOD_deseq$avg_log2FC<0)$X

BMP7_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/BMP7/BMP7_deseq.csv")
BMP7=subset(BMP7_deseq,BMP7_deseq$avg_log2FC<0)$X


ISG15_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/ISG15/ISG15_deseq.csv")
ISG15=subset(ISG15_deseq,ISG15_deseq$avg_log2FC<0)$X


P_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/P/P_deseq.csv")
P=subset(P_deseq,P_deseq$avg_log2FC<0)$X


Cycling_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/Cycling/Cycling_deseq.csv")
Cycling=subset(Cycling_deseq,Cycling_deseq$avg_log2FC<0)$X


set_list <- list(S0=S0,S1=S1,BMP7=BMP7,Cycling=Cycling,REV3L=REV3L,ISG15=ISG15,APOD=APOD,P=P)

upset(
  fromList(set_list),
  order.by = "freq",nsets = 9
)



gene_interset=Reduce(intersect,list(S0,S1,BMP7,Cycling,REV3L,ISG15,APOD,P))
gene_interset



###对STR中8种细胞亚型进行上调基因富集结果的气泡图展示------
###首先做的是GO富集
###选择的是p值最小的前30个通路并进行选择和展示
S0=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/S0_up_ego.csv")
S1=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/S1_up_ego.csv")
REV3L=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/REV3L_up_ego.csv")
ISG15=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/ISG15_up_ego.csv")
APOD=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/APOD_up_ego.csv")
BMP7=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/BMP7_up_ego.csv")
P=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/P_up_ego.csv")
Cycling=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/Cycling_up_ego.csv")


S0=S0[order(S0$P.Value),]
S1=S1[order(S1$P.Value),]
REV3L=REV3L[order(REV3L$P.Value),]
ISG15=ISG15[order(ISG15$P.Value),]
APOD=APOD[order(APOD$P.Value),]
BMP7=BMP7[order(BMP7$P.Value),]
P=P[order(P$P.Value),]
Cycling=Cycling[order(Cycling$P.Value),]

#取前30个富集通路
S0=S0[1:30,]
S1=S1[1:30,]
REV3L=REV3L[1:30,]
ISG15=ISG15[1:30,]
APOD=APOD[1:30,]
BMP7=BMP7[1:30,]
P=P[1:30,]
Cycling=Cycling[1:30,]


S0$Cluster="S0"
S1$Cluster="S1"
REV3L$Cluster="REV3L"
ISG15$Cluster="ISG15"
APOD$Cluster="APOD"
BMP7$Cluster="BMP7"
P$Cluster="P"
Cycling$Cluster="Cycling"




#去除cycling
description=Reduce(intersect,list(S0$X.Term,S1$X.Term,BMP7$X.Term,REV3L$X.Term,ISG15$X.Term,APOD$X.Term,P$X.Term))

ego_all=rbind(S0,S1,REV3L,ISG15,APOD,BMP7,P)
ego=subset(ego_all,X.Term%in%description)


ggplot(data=ego)+aes(x=Cluster,y=X.Term,color=-log10(pvalue))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(ego$X.Term))+theme_bw()+
  scale_color_gradientn(limits=c(-4,319),colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.1,1))+
  theme(axis.text.y = element_text(size=11))+guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))+
  xlab("Cluster")+ylab("")+xlab("")+theme(panel.border = element_blank(),axis.title = element_blank(),axis.ticks = element_blank(),axis.line = element_blank())+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,size = 17))+scale_size_continuous(range = c(5,9.5))





###先做交集部分的气泡图-所有细胞类型
description=Reduce(intersect,list(S0$X.Term,S1$X.Term,BMP7$X.Term,Cycling$X.Term,REV3L$X.Term,ISG15$X.Term,APOD$X.Term,P$X.Term))
ego_all=rbind(S0,S1,REV3L,ISG15,APOD,BMP7,P,Cycling)
ego=subset(ego_all,X.Term%in%description)


ggplot(data=ego)+aes(x=Cluster,y=X.Term,color=-log10(pvalue))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(ego$X.Term))+theme_bw()+
  scale_color_gradientn(limits=c(-4,319),colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.1,1))+
  theme(axis.text.y = element_text(size=11))+guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))+
  xlab("Cluster")+ylab("")+xlab("")+theme(panel.border = element_blank(),axis.title = element_blank(),axis.ticks = element_blank(),axis.line = element_blank())+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,size = 17))+scale_size_continuous(range = c(5,9.5))



des=c(S0$X.Term,S1$X.Term,BMP7$X.Term,Cycling$X.Term,REV3L$X.Term,ISG15$X.Term,APOD$X.Term,P$X.Term)
des=unique(des)
des=as.data.frame(des)



#取特定信号通路进行气泡图绘制
des2=c("focal adhesion","protein domain specific binding","cadherin binding","extracellular matrix organization",
       "collagen-containing extracellular matrix","integrin binding","","extracellular matrix structural constituent",
       "integrin-mediated signaling pathway","stress fiber","cell-matrix adhesion","wound healing",
       "collagen fibril organization","collagen binding","positive regulation of fibroblast proliferation",
       "angiogenesis","muscle contraction")

ego_all=rbind(S0,S1,REV3L,ISG15,APOD,BMP7,P,Cycling)
ego=subset(ego_all,X.Term%in%des2)


ggplot(data=ego)+aes(x=Cluster,y=X.Term,color=-log10(P.value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(ego$X.Term))+theme_bw()+
  scale_color_gradientn(limits=c(-4,50),colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.3,1))+
  theme(axis.text.y = element_text(size=11))+guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))+
  xlab("Cluster")+ylab("")+xlab("")+theme(panel.border = element_blank(),axis.title = element_blank(),axis.ticks = element_blank(),axis.line = element_blank())+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,size = 17))+scale_size_continuous(range = c(5,9.5))









###对STR中8种细胞亚型进行下调基因富集结果的气泡图展示------
###首先做的是GO富集
###选择的是p值最小的前30个通路并进行选择和展示
S0=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/S0_down_ego.csv")
S1=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/S1_down_ego.csv")
REV3L=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/REV3L_down_ego.csv")
ISG15=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/ISG15_down_ego.csv")
APOD=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/APOD_down_ego.csv")
BMP7=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/BMP7_down_ego.csv")
P=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/P_down_ego.csv")
Cycling=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/Cycling_down_ego.csv")


S0=S0[order(S0$P.Value),]
S1=S1[order(S1$P.Value),]
REV3L=REV3L[order(REV3L$P.Value),]
ISG15=ISG15[order(ISG15$P.Value),]
APOD=APOD[order(APOD$P.Value),]
BMP7=BMP7[order(BMP7$P.Value),]
P=P[order(P$P.Value),]
Cycling=Cycling[order(Cycling$P.Value),]

#取前30个富集通路
S0=S0[1:30,]
S1=S1[1:30,]
REV3L=REV3L[1:30,]
ISG15=ISG15[1:30,]
APOD=APOD[1:30,]
BMP7=BMP7[1:30,]
P=P[1:30,]
Cycling=Cycling[1:30,]


S0$Cluster="S0"
S1$Cluster="S1"
REV3L$Cluster="REV3L"
ISG15$Cluster="ISG15"
APOD$Cluster="APOD"
BMP7$Cluster="BMP7"
P$Cluster="P"
Cycling$Cluster="Cycling"



S0=S0[c("X.Term","Input.number","P.Value","Cluster")]
S1=S1[c("X.Term","Input.number","P.Value","Cluster")]
REV3L=REV3L[c("X.Term","Input.number","P.Value","Cluster")]
ISG15=ISG15[c("X.Term","Input.number","P.Value","Cluster")]
APOD=APOD[c("X.Term","Input.number","P.Value","Cluster")]
BMP7=BMP7[c("X.Term","Input.number","P.Value","Cluster")]
P=P[c("X.Term","Input.number","P.Value","Cluster")]
Cycling=Cycling[c("X.Term","Input.number","P.Value","Cluster")]




###先做交集部分的气泡图-所有细胞类型
description=Reduce(intersect,list(S0$X.Term,S1$X.Term,BMP7$X.Term,Cycling$X.Term,REV3L$X.Term,ISG15$X.Term,APOD$X.Term,P$X.Term))
ego_all=rbind(S0,S1,REV3L,ISG15,APOD,BMP7,P,Cycling)
ego=subset(ego_all,X.Term%in%description)



ggplot(data=ego)+aes(x=Cluster,y=X.Term,color=-log10(P.value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(ego$X.Term))+theme_bw()+
  scale_color_gradientn(limits=c(-4,150),colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.3,1))+
  theme(axis.text.y = element_text(size=11))+guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))+
  xlab("Cluster")+ylab("")+xlab("")+theme(panel.border = element_blank(),axis.title = element_blank(),axis.ticks = element_blank(),axis.line = element_blank())+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,size = 17))+scale_size_continuous(range = c(5,9.5))



###查看8组细胞的所有前30通路的交集并进行查看与挑选
des=c(S0$X.Term,S1$X.Term,BMP7$X.Term,Cycling$X.Term,REV3L$X.Term,ISG15$X.Term,APOD$X.Term,P$X.Term)
des=unique(des)
des=as.data.frame(des)



#取特定信号通路进行气泡图绘制
des2=c("NADH dehydrogenase (ubiquinone) activity","ribosome","mitochondrial inner membrane","mitochondrion",
       "small ribosomal subunit","mitochondrial translational elongation" ,"endoplasmic reticulum membrane",
       "mitochondrial translational termination","mitochondrial respiratory chain complex I","mitochondrial respiratory chain complex I assembly",
       "endoplasmic reticulum","cytosolic large ribosomal subunit","cytosolic small ribosomal subunit","structural constituent of ribosome","Golgi apparatus",
       "mitochondrial electron transport, NADH to ubiquinone")



ego_all=rbind(S0,S1,REV3L,ISG15,APOD,BMP7,P,Cycling)
ego=subset(ego_all,X.Term%in%des2)


ggplot(data=ego)+aes(x=Cluster,y=X.Term,color=-log10(P.value))+geom_point(aes(size=Input.number,color= -log10(P.Value)))+
  scale_y_discrete(limits=unique(ego$X.Term))+theme_bw()+
  scale_color_gradientn(limits=c(-4,80),colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])),values = c(0,0.3,1))+
  theme(axis.text.y = element_text(size=11))+guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))+
  xlab("Cluster")+ylab("")+xlab("")+theme(panel.border = element_blank(),axis.title = element_blank(),axis.ticks = element_blank(),axis.line = element_blank())+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,size = 17))+scale_size_continuous(range = c(5,9.5))

###开始定义细胞类型【肌成纤维细胞】------
umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")

STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

DimPlot(STR,label = T)

gene=c("RGS5","PDGFRB","ACTA2")

DimPlot(STR_allsingle,group.by = "RNA_snn_res.1.8",label = T,label.size = 7)

FeaturePlot(STR,features = gene,min.cutoff = 0,order = T)


STR_allsingle_embeddings=STR_allsingle$umap@cell.embeddings
STR_allsingle_embeddings=as.data.frame(STR_allsingle_embeddings)

STR_allsingle_embeddings$UMAP_1=as.numeric(as.character(STR_allsingle_embeddings$UMAP_1))
STR_allsingle_embeddings$UMAP_2=as.numeric(as.character(STR_allsingle_embeddings$UMAP_2))
STR_allsingle$UMAP_1=STR_allsingle_embeddings$UMAP_1
STR_allsingle$UMAP_2=STR_allsingle_embeddings$UMAP_2

STR_allsingle$cell_type2="other"

STR_allsingle$cell_type2[STR_allsingle$UMAP_1< -4.5]<-"myofibroblast"
STR_allsingle$type4="Adhesion"
STR_allsingle$type4[STR_allsingle$type2%in%"CTRL"]="Ctrl"
STR_allsingle$type4[STR_allsingle$type2%in%"IUA1"]="Adjacent"


table(STR_allsingle$cell_type2,STR_allsingle$type4)





DimPlot(STR_allsingle,group.by = "cell_type2",label = T,label.size = 7)



table_plot=as.data.frame(table(STR_allsingle$type4,STR_allsingle$cell_type2))


ggplot(data=table_plot,fill=Var2)+aes(x=Var1,y=stat(table_plot$Freq),fill=Var2)+geom_bar(stat="identity",width=0.75, position="fill")+
  theme_bw()+theme(legend.title=element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank())+ylab("Integrative Celltype(%)")+scale_fill_brewer(palette="Paired")+xlab("")+
  scale_y_continuous(expand=c(0,0),limits=c(0,1),breaks = seq(0,1,0.25),labels = c("0","25","50","75","100"))+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+scale_x_discrete(limits=c("Ctrl","Adjacent","Adhesion"))




#将文章中的数据和单细胞数据进行整合查看-----
umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")
WEN_umap=readRDS("/mnt/chentw/zigongneimo/photo/文献数据操作数据/WEN.umap.rds")

table(WEN_umap$donor)





WEN_umap$donor=factor(WEN_umap$donor,levels = c("proliferative","secretory_early",
                                                "secretory_early-mid","secretory_mid","secretory_late"))



VlnPlot(WEN_umap,features = "FOSL2",group.by = "cell_type",split.by = "donor",pt.size = 0)





table(WEN_umap$donor,WEN_umap$orig.ident)

WEN_umap=subset(WEN_umap,donor%in%c("secretory_early","secretory_early-mid",
                                    "secretory_late","secretory_mid"))



all=merge(umap_ctrl_IUA_allsingle,WEN_umap)


DefaultAssay(all)<-"RNA"

N=all@assays$RNA@counts

all=CreateSeuratObject(counts = N, meta.data = all@meta.data)

all=NormalizeData(all, normalization.method = "LogNormalize", 
                  scale.factor = 10000)
all <- FindVariableFeatures(all, selection.method = "vst", nfeatures = 2000)
VlnPlot(all, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)

mtGenes=grep('^MT-',rownames(all),value=TRUE)
RPGenes=c(grep('^RPL',rownames(all),value=TRUE),grep('^RPS',rownames(all),value=TRUE))
stress_gene2=grep('^HSP',rownames(all),value=TRUE)
HB_gene2=grep('^HB',rownames(all),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
stress_gene1=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
               "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/gene1.csv")
subgene3=subgene3$cellcyle_gene
subgene4=c("FOS","CXCL2","ZFP36","FOSB","DUSP1","ATF3","CXCL8","NR4A1","CXCL3","PPP1R15A",
           "JUNB","EGR1","HSPA1A","HSPA1B","SOCS3","KLF6","JUN","IER2","CXCL1","NFKBIA",
           "HSPA6","DNAJB1","IER3","CCNL1","MTRNR2L2","IER4","ID1","CEBPD","KRT6A","CYR61",
           "DEPP1","CLDN4","IRF1","DUSP2","BTG2","PLAUR","MAFF","KLF4","PHLDA2","TNFAIP3")
subgene5=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/512个应激反应相关基因.csv")
subgene5=subgene5$gene_symbol

subgene=c(mtGenes,RPGenes,stress_gene2,HB_gene2,HB.genes,s.genes,g2m.genes,stress_gene1,"MKI67",subgene3,subgene4,subgene5)

all@assays$RNA@var.features = setdiff(all@assays$RNA@var.features,subgene)  #去除subgene影响

all@assays$RNA@var.features

#all进行标准化然后PCA分析
DefaultAssay(all) <- "RNA"
all.genes_all <- rownames(all)
all <- ScaleData(object = all, features = all.genes_all)
all <- RunPCA(all, features = VariableFeatures(object = all))


#all进行细胞细胞聚类分析(UMAP)
all=RunUMAP(all, dims = 1:30)
DimPlot(all,reduction="umap",label=TRUE)
all <- FindNeighbors(all, dims = 1:30)
all <- FindClusters(all, resolution = 0.3)
all <- FindClusters(all, resolution = 0.6)
DimPlot(all,reduction="umap",label=TRUE)


#all进行细胞细胞聚类分析(tsne)
all=RunTSNE(all, dims = 1:30)
DimPlot(all,reduction="tsne",label=TRUE)
all <- FindNeighbors(all, dims = 1:30)
all <- FindClusters(all, resolution = 0.3)
DimPlot(all,reduction="tsne",label=TRUE)

DimPlot(all,reduction="umap",label=TRUE)



#开始尝试去除细胞样本批次并且进行新的一个保存

all=all%>%RunHarmony("orig.ident",plot_covergence=T)

DefaultAssay(all) <- "RNA"


all <- RunPCA(all, features = VariableFeatures(object = all))


all=RunUMAP(all,reduction="harmony",dims=1:30)

all=FindNeighbors(all,dims=1:30,reduction = "harmony")
all=FindClusters(all, resolution = 0.15,reduction = "harmony")
all=FindClusters(all, resolution = 0.3,reduction = "harmony")
all=FindClusters(all, resolution = 0.6,reduction = "harmony")
all=FindClusters(all, resolution = 0.9,reduction = "harmony")
all=FindClusters(all, resolution = 1.2,reduction = "harmony")


all$type4="Secretory"
all$type4[all$type2%in%"CTRL"]<-"Ctrl"
all$type4[all$type2%in%"IUA1"]<-"Adjacent"
all$type4[all$type2%in%"IUA2"]<-"Adhesion"

DimPlot(all,raster = F,split.by = "type4",
        label = T,group.by = "cell_type")




saveRDS(all,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/all.rds")



#对基质细胞的体内数据进行一个整合【和文章所有时期的基质细胞】
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

table(STR$type2)

WEN_umap=readRDS("/mnt/chentw/zigongneimo/photo/文献数据操作数据/WEN.umap.rds")

WEN_umap$type2=WEN_umap$donor

table(WEN_umap$cell_type)

WEN_umap=subset(WEN_umap,cell_type%in%"Stromal fibroblasts")

WEN_umap$cell_type0305="stromal"





STR_allsingle=merge(WEN_umap,STR)




####开始进行重新整合
N=STR_allsingle@assays$RNA@counts

STR_allsingle=CreateSeuratObject(counts = N, meta.data = STR_allsingle@meta.data)

STR_allsingle=NormalizeData(STR_allsingle, normalization.method = "LogNormalize", 
                            scale.factor = 10000)
STR_allsingle <- FindVariableFeatures(STR_allsingle, selection.method = "vst", nfeatures = 2000)
VlnPlot(STR_allsingle, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)

mtGenes=grep('^MT-',rownames(STR_allsingle),value=TRUE)
RPGenes=c(grep('^RPL',rownames(STR_allsingle),value=TRUE),grep('^RPS',rownames(STR_allsingle),value=TRUE))
stress_gene2=grep('^HSP',rownames(STR_allsingle),value=TRUE)
HB_gene2=grep('^HB',rownames(STR_allsingle),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
stress_gene1=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
               "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/gene1.csv")
subgene3=subgene3$cellcyle_gene
subgene4=c("FOS","CXCL2","ZFP36","FOSB","DUSP1","ATF3","CXCL8","NR4A1","CXCL3","PPP1R15A",
           "JUNB","EGR1","HSPA1A","HSPA1B","SOCS3","KLF6","JUN","IER2","CXCL1","NFKBIA",
           "HSPA6","DNAJB1","IER3","CCNL1","MTRNR2L2","IER4","ID1","CEBPD","KRT6A","CYR61",
           "DEPP1","CLDN4","IRF1","DUSP2","BTG2","PLAUR","MAFF","KLF4","PHLDA2","TNFAIP3")
subgene5=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/512个应激反应相关基因.csv")
subgene5=subgene5$gene_symbol

subgene=c(mtGenes,RPGenes,stress_gene2,HB_gene2,HB.genes,s.genes,g2m.genes,stress_gene1,"MKI67",subgene3,subgene4,subgene5)

STR_allsingle@assays$RNA@var.features = setdiff(STR_allsingle@assays$RNA@var.features,subgene)  #去除subgene影响

STR_allsingle@assays$RNA@var.features

#STR_allsingle进行标准化然后PCA分析
DefaultAssay(STR_allsingle) <- "RNA"
all.genes_STR_allsingle <- rownames(STR_allsingle)
STR_allsingle <- ScaleData(object = STR_allsingle, features = all.genes_STR_allsingle)
STR_allsingle <- RunPCA(STR_allsingle, features = VariableFeatures(object = STR_allsingle))


#STR_allsingle进行细胞细胞聚类分析(UMAP)
STR_allsingle=RunUMAP(STR_allsingle, dims = 1:30)
DimPlot(STR_allsingle,reduction="umap",label=TRUE)
STR_allsingle <- FindNeighbors(STR_allsingle, dims = 1:30)
STR_allsingle <- FindClusters(STR_allsingle, resolution = 0.3)
STR_allsingle <- FindClusters(STR_allsingle, resolution = 0.6)
DimPlot(STR_allsingle,reduction="umap",label=TRUE)


#STR_allsingle进行细胞细胞聚类分析(tsne)
STR_allsingle=RunTSNE(STR_allsingle, dims = 1:30)
DimPlot(STR_allsingle,reduction="tsne",label=TRUE)
STR_allsingle <- FindNeighbors(STR_allsingle, dims = 1:30)
STR_allsingle <- FindClusters(STR_allsingle, resolution = 0.3)
DimPlot(STR_allsingle,reduction="tsne",label=TRUE)

DimPlot(STR_allsingle,reduction="umap",label=TRUE)



#开始尝试去除细胞样本批次并且进行新的一个保存

STR_allsingle=STR_allsingle%>%RunHarmony("orig.ident",plot_covergence=T)

DefaultAssay(STR_allsingle) <- "RNA"


STR_allsingle <- RunPCA(STR_allsingle, features = VariableFeatures(object = STR_allsingle))


STR_allsingle=RunUMAP(STR_allsingle,reduction="harmony",dims=1:30)
STR_allsingle=FindNeighbors(STR_allsingle,dims=1:30,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.15,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.17,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.18,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.2,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.3,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.4,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.45,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.5,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.55,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.6,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.7,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 0.9,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 1,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 1.1,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 1.2,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 1.3,reduction = "harmony")
STR_allsingle=FindClusters(STR_allsingle, resolution = 1.4,reduction = "harmony")





DimPlot(STR_allsingle,group.by = "cell_type0305",label=TRUE)



saveRDS(STR_allsingle,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_all.rds")



###上面harmony的效果不太好，暴力整合，integrated[晚上跑]
####开始进行重新整合
STR_allsingle.list <- SplitObject(STR_allsingle, split.by = "type3")

STR_allsingle.list <- lapply(X = STR_allsingle.list, FUN = function(x) {
  x <- NormalizeData(x)
  x <- FindVariableFeatures(x, selection.method = "vst", nfeatures = 2000)
})


features <- SelectIntegrationFeatures(object.list = STR_allsingle.list)


mtGenes=grep('^MT-',rownames(STR_allsingle),value=TRUE)
RPGenes=grep('^RP',rownames(STR_allsingle),value=TRUE)
stress_gene2=grep('^HSP',rownames(STR_allsingle),value=TRUE)
HB_gene2=grep('^HB',rownames(STR_allsingle),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
stress_gene1=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
               "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/gene1.csv")
subgene3=subgene3$cellcyle_gene
subgene4=c("FOS","CXCL2","ZFP36","FOSB","DUSP1","ATF3","CXCL8","NR4A1","CXCL3","PPP1R15A",
           "JUNB","EGR1","HSPA1A","HSPA1B","SOCS3","KLF6","JUN","IER2","CXCL1","NFKBIA",
           "HSPA6","DNAJB1","IER3","CCNL1","MTRNR2L2","IER4","ID1","CEBPD","KRT6A","CYR61",
           "DEPP1","CLDN4","IRF1","DUSP2","BTG2","PLAUR","MAFF","KLF4","PHLDA2","TNFAIP3")



subgene=c(mtGenes,RPGenes,stress_gene2,HB_gene2,HB.genes,s.genes,g2m.genes,stress_gene1,"MKI67",subgene3,subgene4)


features=features[-which(features%in%subgene)]

STR_allsingle <- FindIntegrationAnchors(object.list = STR_allsingle.list, anchor.features = features)

STR_allsingle <- IntegrateData(anchorset = STR_allsingle)

#进行标准化并进行PCA
DefaultAssay(STR_allsingle) <- "RNA"
all.genes.STR_allsingle <- rownames(STR_allsingle)
STR_allsingle <- ScaleData(object = STR_allsingle, features = all.genes.STR_allsingle)

DefaultAssay(STR_allsingle) <- "integrated"
all.genes.STR_allsingle1 <- rownames(STR_allsingle)
STR_allsingle <- ScaleData(object = STR_allsingle, features = all.genes.STR_allsingle1)
STR_allsingle <- RunPCA(STR_allsingle, features = features)
DimPlot(STR_allsingle,label = T)

#进行STR_allsingle细胞细胞聚类分析(先runumap降维确定形状再确定cluster)
STR_allsingle=RunUMAP(STR_allsingle, dims = 1:30)
STR_allsingle <- FindNeighbors(STR_allsingle, dims = 1:30)
STR_allsingle <- FindClusters(STR_allsingle, resolution = 0.15)
STR_allsingle <- FindClusters(STR_allsingle, resolution = 0.16)
STR_allsingle <- FindClusters(STR_allsingle, resolution = 0.17)
STR_allsingle <- FindClusters(STR_allsingle, resolution = 0.18)
STR_allsingle <- FindClusters(STR_allsingle, resolution = 0.2)
STR_allsingle <- FindClusters(STR_allsingle, resolution = 0.25)
STR_allsingle <- FindClusters(STR_allsingle, resolution = 0.3)
STR_allsingle <- FindClusters(STR_allsingle, resolution = 0.6)
STR_allsingle <- FindClusters(STR_allsingle, resolution = 0.9)
STR_allsingle <- FindClusters(STR_allsingle, resolution = 1.2)
STR_allsingle <- FindClusters(STR_allsingle, resolution = 1.5)


STR_allsingle=RunTSNE(STR_allsingle, dims = 1:30)
DimPlot(STR_allsingle,reduction="tsne",label=TRUE)
STR_allsingle <- FindNeighbors(STR_allsingle, dims = 1:30)
STR_allsingle <- FindClusters(STR_allsingle, resolution = 0.3)





###对过滤以后的基质细胞进行映射----
WEN_STR<-LoadH5Seurat("/data/DATAprocess/chentw/zigongneimo/WEN2/h5ad文件/endo-2022_stromal.h5seurat",
                      assays="RNA")
STR_allsingle3=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")


#走一下标准化流程
WEN_STR <- SCTransform(WEN_STR)
WEN_STR <- RunPCA(WEN_STR)
WEN_STR <-RunUMAP(WEN_STR, return.model = TRUE,1:30)

DimPlot(WEN_STR,group.by = "subtypes",label = T)


STR_allsingle3 <- SCTransform(STR_allsingle3)

STR_allsingle3=STR_allsingle3%>%RunHarmony("type3",plot_covergence=T)

STR_allsingle3 <- RunPCA(STR_allsingle3)


STR_allsingle3=RunUMAP(STR_allsingle3,reduction="harmony",dims=1:30)
STR_allsingle3=FindNeighbors(STR_allsingle3,dims=1:30,reduction = "harmony")



DimPlot(STR_allsingle3,group.by = "cell_type",label = T)

DimPlot(STR_allsingle3,group.by = "orig.ident",label = T)

transfer.anchors <- FindTransferAnchors(reference = WEN_STR, query = STR_allsingle3, 
                                        dims = 1:30)

predictions <- TransferData(anchorset = transfer.anchors, refdata = WEN_STR$subtypes, 
                            dims = 1:30)

STR_allsingle3$predictions=predictions$predicted.id

DimPlot(STR_allsingle3,group.by = "predictions",label = T)

#对STR_allsingle3内部进行adhesion评分小提琴图绘制----
STR_allsingle3_data=as.data.frame(STR_allsingle3@assays$RNA@data)

STR_allsingle3_data=subset(STR_allsingle3_data,rownames(STR_allsingle3_data)%in%c(gene_interset))

STR_allsingle3_data=as.data.frame(t(STR_allsingle3_data))

STR_allsingle3_data$mean=apply(STR_allsingle3_data,1,mean)

STR_allsingle3_data$typ2=STR_allsingle3$type2
STR_allsingle3$gene=STR_allsingle3_data$mean


VlnPlot(STR_allsingle3,features = "gene",pt.size = 0,group.by = "cell_type0305")+ggtitle("")+
  xlab("")+ylab("Fibrosis Score")


##将基质细胞和异位症文章基质细胞进行整合----
library(SeuratDisk)
library(dplyr)
library(Seurat)


WEN_STR<-LoadH5Seurat("/data/DATAprocess/chentw/zigongneimo/WEN2/h5ad文件/endo-2022_stromal.h5seurat",
                      assays="RNA")
STR_allsingle3=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")


#走一下标准化流程
WEN_STR <- SCTransform(WEN_STR)
WEN_STR <- RunPCA(WEN_STR)
WEN_STR <-RunUMAP(WEN_STR, return.model = TRUE,1:30)

DimPlot(WEN_STR,group.by = "subtypes",label = T)




STR_allsingle3 <- SCTransform(STR_allsingle3)


WEN_STR$celltype=WEN_STR$subtypes
STR_allsingle3$celltype=STR_allsingle3$cell_type0305

WEN_STR$orig.ident=WEN_STR$PID

WEN_STR$type2="WEN"
WEN_STR$type4="WEN"
STR_allsingle3$type4="STR"


STR_integrated=merge(STR_allsingle3,WEN_STR)

features <- SelectIntegrationFeatures(object.list = STR_integrated.list)


mtGenes=grep('^MT-',rownames(STR_integrated),value=TRUE)
RPGenes=grep('^RP',rownames(STR_integrated),value=TRUE)
stress_gene2=grep('^HSP',rownames(STR_integrated),value=TRUE)
HB_gene2=grep('^HB',rownames(STR_integrated),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
stress_gene1=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
               "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/gene1.csv")
subgene3=subgene3$cellcyle_gene
subgene4=c("FOS","CXCL2","ZFP36","FOSB","DUSP1","ATF3","CXCL8","NR4A1","CXCL3","PPP1R15A",
           "JUNB","EGR1","HSPA1A","HSPA1B","SOCS3","KLF6","JUN","IER2","CXCL1","NFKBIA",
           "HSPA6","DNAJB1","IER3","CCNL1","MTRNR2L2","IER4","ID1","CEBPD","KRT6A","CYR61",
           "DEPP1","CLDN4","IRF1","DUSP2","BTG2","PLAUR","MAFF","KLF4","PHLDA2","TNFAIP3")



subgene=c(mtGenes,RPGenes,stress_gene2,HB_gene2,HB.genes,s.genes,g2m.genes,stress_gene1,"MKI67",subgene3,subgene4)



####开始进行重新整合
DefaultAssay(STR_integrated)<-'RNA'

####开始进行重新整合
STR_integrated.list <- SplitObject(STR_integrated, split.by = "orig.ident")

for (i in 1:length(STR_integrated.list)) {
  STR_integrated.list[[i]] <- SCTransform(STR_integrated.list[[i]], verbose = FALSE)
}


features <- SelectIntegrationFeatures(object.list = STR_integrated.list, nfeatures = 3000)


###由于代码冲突这一步不能删掉应激蛋白等基因暂时【后面思考调整】

STR_integrated.list <- PrepSCTIntegration(object.list = STR_integrated.list, anchor.features = features) 

STR_integrated.anchors <- FindIntegrationAnchors(object.list = STR_integrated.list, normalization.method = "SCT", 
                                                 anchor.features = features)

STR_integrated.integrated <- IntegrateData(anchorset = STR_integrated.anchors, normalization.method = "SCT", 
                                           verbose = FALSE)


#进行标准化并进行PCA
DefaultAssay(STR_integrated) <- "RNA"
all.genes.STR_integrated <- rownames(STR_integrated)
STR_integrated <- ScaleData(object = STR_integrated, features = all.genes.STR_integrated)

DefaultAssay(STR_integrated) <- "integrated"
all.genes.STR_integrated1 <- rownames(STR_integrated)
STR_integrated <- ScaleData(object = STR_integrated, features = all.genes.STR_integrated1)
STR_integrated <- RunPCA(STR_integrated, features = features)
DimPlot(STR_integrated,label = T)

#进行STR_integrated细胞细胞聚类分析(先runumap降维确定形状再确定cluster)
STR_integrated=RunUMAP(STR_integrated, dims = 1:30)
STR_integrated <- FindNeighbors(STR_integrated, dims = 1:30)
STR_integrated <- FindClusters(STR_integrated, resolution = 0.15)
STR_integrated <- FindClusters(STR_integrated, resolution = 0.16)
STR_integrated <- FindClusters(STR_integrated, resolution = 0.17)
STR_integrated <- FindClusters(STR_integrated, resolution = 0.18)
STR_integrated <- FindClusters(STR_integrated, resolution = 0.2)
STR_integrated <- FindClusters(STR_integrated, resolution = 0.25)
STR_integrated <- FindClusters(STR_integrated, resolution = 0.3)
STR_integrated <- FindClusters(STR_integrated, resolution = 0.6)
STR_integrated <- FindClusters(STR_integrated, resolution = 0.9)
STR_integrated <- FindClusters(STR_integrated, resolution = 1.2)
STR_integrated <- FindClusters(STR_integrated, resolution = 1.5)


STR_integrated=RunTSNE(STR_integrated, dims = 1:30)
DimPlot(STR_integrated,reduction="tsne",label=TRUE)
STR_integrated <- FindNeighbors(STR_integrated, dims = 1:30)
STR_integrated <- FindClusters(STR_integrated, resolution = 0.3)

saveRDS(STR_integrated,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_integrated.rds")

DimPlot(STR_integrated,group.by = "orig.ident",label = T)


###尝试定义肌成纤维细胞亚型-----
STR_allsingle3=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

STR_allsingle3_embeddings=STR_allsingle3$umap@cell.embeddings
STR_allsingle3_embeddings=as.data.frame(STR_allsingle3_embeddings)

STR_allsingle3_embeddings$UMAP_1=as.numeric(as.character(STR_allsingle3_embeddings$UMAP_1))
STR_allsingle3_embeddings$UMAP_2=as.numeric(as.character(STR_allsingle3_embeddings$UMAP_2))
STR_allsingle3$UMAP_1=STR_allsingle3_embeddings$UMAP_1
STR_allsingle3$UMAP_2=STR_allsingle3_embeddings$UMAP_2

STR_allsingle3_data=
  as.data.frame(STR_allsingle3@assays$RNA@data)

STR_allsingle3_data=subset(STR_allsingle3_data,
                           rownames(STR_allsingle3_data)%in%c("ACTA2","TAGLN"))

STR_allsingle3_data=as.data.frame(t(STR_allsingle3_data))

STR_allsingle3$ACTA2=STR_allsingle3_data$ACTA2
STR_allsingle3$TAGLN=STR_allsingle3_data$TAGLN

FeaturePlot(STR_allsingle3,features = c("ACTA2","TAGLN"),
            reduction = "UMAP",min.cutoff = 0,order = T)




FeaturePlot(STR_allsingle3,features = c("ACTA2","TAGLN"),
            reduction = "umap",min.cutoff = 0,order = T)

S0=subset(STR_allsingle3,cell_type0305%in%"S0")

FeaturePlot(S0,features = c("rna_ACTA2","rna_TAGLN"),
            reduction = "umap",min.cutoff = 1.2,order = T)

quantile(S0$ACTA2)
quantile(S0$TAGLN)


STR_allsingle3$cell_type1225="k"
STR_allsingle3$cell_type1225[STR_allsingle3$cell_type0305%in%"S0"&STR_allsingle3$UMAP_2>2&STR_allsingle3$ACTA2>1.2]<-"Myofibroblast"

STR_allsingle3$cell_type1225[STR_allsingle3$cell_type0305%in%"S1"]<-"S1"
STR_allsingle3$cell_type1225[STR_allsingle3$cell_type0305%in%"BMP7"]<-"BMP7"
STR_allsingle3$cell_type1225[STR_allsingle3$cell_type0305%in%"Cycling"]<-"Cycling"
STR_allsingle3$cell_type1225[STR_allsingle3$cell_type0305%in%"REV3L"]<-"REV3L"
STR_allsingle3$cell_type1225[STR_allsingle3$cell_type0305%in%"ISG15"]<-"ISG15"
STR_allsingle3$cell_type1225[STR_allsingle3$cell_type0305%in%"APOD"]<-"APOD"
STR_allsingle3$cell_type1225[STR_allsingle3$cell_type0305%in%"P"]<-"P"
STR_allsingle3$cell_type1225[STR_allsingle3$cell_type1225%in%"k"]<-"S0"

table(STR_allsingle3$cell_type1225)
STR_allsingle3$cell_type1225=factor(STR_allsingle3$cell_type1225,
                                    levels = c("S0","S1","BMP7","Cycling","REV3L","ISG15","APOD","P","Myofibroblast"))

DimPlot(STR_allsingle3,group.by = "cell_type1225",label = T,order = T)+
  scale_color_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4","#5555A4","#5555A4"))





VlnPlot()




STR_allsingle3_data=as.data.frame(STR_allsingle3@assays$RNA@data)

STR_allsingle3_data=subset(STR_allsingle3_data,rownames(STR_allsingle3_data)%in%c(gene_interset))

STR_allsingle3_data=as.data.frame(t(STR_allsingle3_data))

STR_allsingle3_data$mean=apply(STR_allsingle3_data,1,mean)

STR_allsingle3_data$typ2=STR_allsingle3$type2
STR_allsingle3$gene=STR_allsingle3_data$mean


VlnPlot(STR_allsingle3,features = "gene",pt.size = 0,group.by = "cell_type1225",split.by = "type2")+ggtitle("")+
  xlab("")+ylab("Fibrosis Score")


table_plot=as.data.frame(table(STR_allsingle3$type1,STR_allsingle3$cell_type1225))

ggplot(data=table_plot,fill=Var2)+aes(x=Var1,y=stat(table_plot$Freq),fill=Var2)+geom_bar(stat="identity",width=0.75, position="fill")+
  theme_bw()+theme(legend.title=element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank())+ylab("Integrative Celltype(%)")+scale_fill_brewer(palette="Paired")+xlab("")+
  scale_y_continuous(expand=c(0,0),limits=c(0,1),breaks = seq(0,1,0.25),labels = c("0","25","50","75","100"))+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))


#进行基质细胞内亚型比例统计8群细胞在ctrl和IUA之间的组内变化------
K=as.data.frame(table(STR_allsingle3$type3,STR_allsingle3$cell_type1225))

K$ratio="k"

K$ratio[K$Var1%in%"CTRL1"]<-K$Freq[K$Var1%in%"CTRL1"]/5914
K$ratio[K$Var1%in%"CTRL2"]<-K$Freq[K$Var1%in%"CTRL2"]/2869
K$ratio[K$Var1%in%"IUA1"]<-K$Freq[K$Var1%in%"IUA1"]/3174
K$ratio[K$Var1%in%"IUA2"]<-K$Freq[K$Var1%in%"IUA2"]/6721
K$ratio[K$Var1%in%"IUA3"]<-K$Freq[K$Var1%in%"IUA3"]/2602
K$ratio[K$Var1%in%"IUA4"]<-K$Freq[K$Var1%in%"IUA4"]/649
K$ratio[K$Var1%in%"IUA5"]<-K$Freq[K$Var1%in%"IUA5"]/1619
K$ratio[K$Var1%in%"IUA6"]<-K$Freq[K$Var1%in%"IUA6"]/982


write.csv(K,"/data/DATAprocess/chentw/zigongneimo/temp/ratio.csv")












#进行基质细胞内亚型比例统计8群细胞在ctrl和adjacent,adhesion之间的组内变化------
STR_allsingle3=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

STR_allsingle3$cell_type2="K"
STR_allsingle3$cell_type2[STR_allsingle3$RNA_snn_res.0.3%in%"0"]<-"FIB1"
STR_allsingle3$cell_type2[STR_allsingle3$RNA_snn_res.0.3%in%"1"]<-"FIB2"
STR_allsingle3$cell_type2[STR_allsingle3$RNA_snn_res.0.3%in%"2"]<-"FIB3"
STR_allsingle3$cell_type2[STR_allsingle3$RNA_snn_res.0.3%in%"3"]<-"FIB4"
STR_allsingle3$cell_type2[STR_allsingle3$RNA_snn_res.0.3%in%"4"]<-"FIB5"
STR_allsingle3$cell_type2[STR_allsingle3$RNA_snn_res.0.3%in%"5"]<-"FIB6"
STR_allsingle3$cell_type2[STR_allsingle3$RNA_snn_res.0.3%in%"6"]<-"FIB7"
STR_allsingle3$cell_type2[STR_allsingle3$RNA_snn_res.0.3%in%"7"]<-"FIB8"

DimPlot(STR_allsingle3,label = T,label.size = 12,group.by = "cell_type2",pt.size = 1)+
  ggtitle("")+
  scale_color_manual(values = c("#DB0C15","#387DB9","#4CA636","#EE7A00","#9E4894",
                                "#5555A4","#482013","#F9E800"))

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/temp/P1.jpeg",
       width = 9,height = 6,dpi = 1000)



DimPlot(STR_allsingle3,label = T,label.size = 6,group.by = "cell_type2",pt.size = 1,
        split.by = "type3")+
  ggtitle("")+
  scale_color_manual(values = c("#DB0C15","#387DB9","#4CA636","#EE7A00","#9E4894",
                                "#5555A4","#482013","#F9E800"))

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/temp/P1.jpeg",
       width = 35,height = 6,dpi = 500)




pdf(file = "/data/DATAprocess/chentw/zigongneimo/temp/STR2.pdf",
    width = 40,height = 6)

DimPlot(STR_allsingle3,label = T,label.size = 6,group.by = "cell_type2",pt.size = 1,
        split.by = "type3")+
  ggtitle("")+
  scale_color_manual(values = c("#DB0C15","#387DB9","#4CA636","#EE7A00","#9E4894",
                                "#5555A4","#482013","#F9E800"))
dev.off()



K=as.data.frame(table(STR_allsingle3$type3,STR_allsingle3$cell_type0305))

K$ratio="k"

K$ratio[K$Var1%in%"CTRL1"]<-K$Freq[K$Var1%in%"CTRL1"]/5914
K$ratio[K$Var1%in%"CTRL2"]<-K$Freq[K$Var1%in%"CTRL2"]/2869
K$ratio[K$Var1%in%"IUA1"]<-K$Freq[K$Var1%in%"IUA1"]/3174
K$ratio[K$Var1%in%"IUA2"]<-K$Freq[K$Var1%in%"IUA2"]/6721
K$ratio[K$Var1%in%"IUA3"]<-K$Freq[K$Var1%in%"IUA3"]/2602
K$ratio[K$Var1%in%"IUA4"]<-K$Freq[K$Var1%in%"IUA4"]/649
K$ratio[K$Var1%in%"IUA5"]<-K$Freq[K$Var1%in%"IUA5"]/1619
K$ratio[K$Var1%in%"IUA6"]<-K$Freq[K$Var1%in%"IUA6"]/982


write.csv(K,"/data/DATAprocess/chentw/zigongneimo/temp/ratio.csv")


ratio=read.csv("/data/DATAprocess/chentw/zigongneimo/temp/ratio.csv")

ratio=gather(ratio,Var2, Freq, colnames(ratio),-Var1)


##对比例进行叠加柱形图绘制
table_plot=as.data.frame(table(STR$type3,STR$cell_type2))
ggplot(data=table_plot,fill=Var2)+aes(x=Var1,y=stat(table_plot$Freq),fill=Var2)+geom_bar(stat="identity",width=0.75, position="fill")+
  theme_bw()+theme(legend.title=element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank())+ylab("Integrative Celltype(%)")+scale_fill_brewer(palette="Paired")+xlab("")+
  scale_y_continuous(expand=c(0,0),limits=c(0,1),breaks = seq(0,1,0.25),labels = c("0","25","50","75","100"))+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+
  scale_fill_manual(values = c("#DB0C15","#387DB9","#4CA636","#EE7A00","#9E4894",
                               "#5555A4","#482013","#F9E800"))








#开始定义细胞亚型-----
STR_allsingle3=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

STR_allsingle3@active.ident=STR_allsingle3$RNA_snn_res.0.3

DimPlot(STR_allsingle3,label = T,group.by = "cell_type2",label.size = 6)

C7_marker=c("C7","ACTA2","OGN")

FeaturePlot(STR_allsingle3,features = C7_marker,min.cutoff = 0,order = T,
            ncol = 3)

DotPlot(STR_allsingle3,group.by = "cell_type2",features = C7_marker)

STR_marker=c("IGF1","PCOLCE")


FeaturePlot(STR_allsingle3,features = STR_marker,
            min.cutoff = 0,order = T,
            ncol = 2)

DotPlot(STR_allsingle3,group.by = "cell_type2",features =STR_marker)


es_marker=c("MMP11","CRABP2","ECM1")

FeaturePlot(STR_allsingle3,features = es_marker,min.cutoff = 0,order = T,
            ncol = 3)

DotPlot(STR_allsingle3,group.by = "cell_type2",features = es_marker)


gene=c("ACTA2","PDGFRB","RGS5")

FeaturePlot(STR_allsingle3,features = gene,min.cutoff = 0,order = T,
            ncol = 3)

DotPlot(STR_allsingle3,group.by = "cell_type2",features = gene)


gene=c("BMP3","BMP4","BMP7")

FeaturePlot(STR_allsingle3,features = gene,min.cutoff = 0,order = T,
            ncol = 3)

DotPlot(STR_allsingle3,group.by = "cell_type2",features = gene)

gene=c("TOP2A","MKI67")

FeaturePlot(STR_allsingle3,features = gene,min.cutoff = 0,order = T,
            ncol = 2)

DotPlot(STR_allsingle3,group.by = "cell_type2",features = gene)







#对8群细胞进行部分marker基因的气泡图绘制------
STR_allsingle3=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

gene=c("DCN","LUM","DKK1","PDGFRA","SFRP4","HAND2","WT1","PCOLCE","TOP2A","MKI67","CD20","WNT2","WNT4",
       "BMP3","BMP4","BMP7","C7","OGN","ACTA2","RGS5","TAGLN","PDGFRB")


pdf("/data/DATAprocess/chentw/zigongneimo/temp/t1.pdf",width = 9,height = 5)
DotPlot(STR_allsingle3,group.by = "cell_type2",features = gene)+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1))+
  ggtitle('')+xlab("")+ylab("")+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])))+
  guides(color = guide_colorbar(order = 1),size=guide_legend(order=0))+
  labs(color = "Scaled \nexpression",size="Percentage\nsingle cells")
dev.off()


###对marker基因进行查找以及绘制
###marker基因查找并进行富marker基因气泡图绘制并尝试定义
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")


STR$cell_type2="K"
STR$cell_type2[STR$RNA_snn_res.0.3%in%"0"]<-"FIB1"
STR$cell_type2[STR$RNA_snn_res.0.3%in%"1"]<-"FIB2"
STR$cell_type2[STR$RNA_snn_res.0.3%in%"2"]<-"FIB3"
STR$cell_type2[STR$RNA_snn_res.0.3%in%"3"]<-"FIB4"
STR$cell_type2[STR$RNA_snn_res.0.3%in%"4"]<-"FIB5"
STR$cell_type2[STR$RNA_snn_res.0.3%in%"5"]<-"FIB6"
STR$cell_type2[STR$RNA_snn_res.0.3%in%"6"]<-"FIB7"
STR$cell_type2[STR$RNA_snn_res.0.3%in%"7"]<-"FIB8"



DefaultAssay(STR)<-"RNA"
STR@active.ident=as.factor(STR$cell_type2)

STR_markers <- FindAllMarkers(object=STR, only.pos = TRUE,min.pct = 0.25, 
                              logfc.threshold = 0.25)
STR_markers <-subset(STR_markers,STR_markers$p_val<0.05)


write.csv(STR_markers,
          "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/STR_markers.csv")

mtGenes=grep('^MT-',rownames(STR),value=TRUE)
RPGenes=c(grep('^RPL',rownames(STR),value=TRUE),grep('^RPS',rownames(STR),value=TRUE))
stress_gene2=grep('^HSP',rownames(STR),value=TRUE)
HB_gene2=grep('^HB',rownames(STR),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
stress_gene1=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
               "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/gene1.csv")
subgene3=subgene3$cellcyle_gene
subgene4=c("FOS","CXCL2","ZFP36","FOSB","DUSP1","ATF3","CXCL8","NR4A1","CXCL3","PPP1R15A",
           "JUNB","EGR1","HSPA1A","HSPA1B","SOCS3","KLF6","JUN","IER2","CXCL1","NFKBIA",
           "HSPA6","DNAJB1","IER3","CCNL1","MTRNR2L2","IER4","ID1","CEBPD","KRT6A","CYR61",
           "DEPP1","CLDN4","IRF1","DUSP2","BTG2","PLAUR","MAFF","KLF4","PHLDA2","TNFAIP3")
subgene5=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/512个应激反应相关基因.csv")
subgene5=subgene5$gene_symbol

subgene=c(mtGenes,RPGenes,stress_gene2,HB_gene2,HB.genes,s.genes,g2m.genes,stress_gene1,"MKI67",subgene3,subgene4,subgene5)


#包含非特异的marker
STR_markers=subset(STR_markers,!gene%in%subgene)

STR_markers_top10 <- STR_markers %>% group_by(cluster) %>% top_n(n = 10, wt = avg_log2FC)

DefaultAssay(STR)<-"RNA"

STR_markers_top10=unique(STR_markers_top10$gene)

DotPlot(STR,features=STR_markers_top10,group.by = "cell_type2")+
  theme(axis.text.x= element_text(size=9,angle = 45,vjust = 1,hjust = 1,face= "bold"))+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])))+xlab("")+ylab("")



#特异的marker
STR_markers=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/STR_markers.csv")

Z=STR_markers[duplicated(STR_markers$gene),]

N=subset(STR_markers,!gene%in%unique(Z$gene))

write.csv(N,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/marker以及富集表/STR_markers_unique.csv")

table(duplicated(N$gene))

STR_markers_top10 <- N %>% group_by(cluster) %>% top_n(n = 10, wt = avg_log2FC)

DefaultAssay(STR)<-"RNA"

STR_markers_top10=unique(STR_markers_top10$gene)


pdf("/data/DATAprocess/chentw/zigongneimo/temp/P1.pdf",width = 16,height = 5)
DotPlot(STR,features=STR_markers_top10,group.by = "cell_type2")+
  theme(axis.text.x= element_text(size=9,angle = 45,vjust = 1,hjust = 1,face= "bold"))+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])))+xlab("")+ylab("")


dev.off()







#对STR进行细胞类型定义----
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")


STR$cell_type2="K"
STR$cell_type2[STR$RNA_snn_res.0.3%in%"0"]<-"FIB1"
STR$cell_type2[STR$RNA_snn_res.0.3%in%"1"]<-"FIB2"
STR$cell_type2[STR$RNA_snn_res.0.3%in%"2"]<-"FIB3"
STR$cell_type2[STR$RNA_snn_res.0.3%in%"3"]<-"FIB4"
STR$cell_type2[STR$RNA_snn_res.0.3%in%"4"]<-"FIB5"
STR$cell_type2[STR$RNA_snn_res.0.3%in%"5"]<-"FIB6"
STR$cell_type2[STR$RNA_snn_res.0.3%in%"6"]<-"FIB7"
STR$cell_type2[STR$RNA_snn_res.0.3%in%"7"]<-"FIB8"


STR$cell_type0103="K"
STR$cell_type0103[STR$RNA_snn_res.0.3%in%"0"]<-"Normal"
STR$cell_type0103[STR$RNA_snn_res.0.3%in%"1"]<-"PGR+"
STR$cell_type0103[STR$RNA_snn_res.0.3%in%"2"]<-"OGN+"
STR$cell_type0103[STR$RNA_snn_res.0.3%in%"3"]<-"Cycling"
STR$cell_type0103[STR$RNA_snn_res.0.3%in%"4"]<-"BMP7+/CD34+"
STR$cell_type0103[STR$RNA_snn_res.0.3%in%"5"]<-"Myofibroblast/Pericyte"
STR$cell_type0103[STR$RNA_snn_res.0.3%in%"6"]<-"ISG15+"
STR$cell_type0103[STR$RNA_snn_res.0.3%in%"7"]<-"APOD+"

STR$cell_type0103=factor(STR$cell_type0103,
                         levels = c("Normal","PGR+","OGN+","Cycling",
                                    "BMP7+/CD34+","Myofibroblast/Pericyte",
                                    "ISG15+","APOD+"))


pdf("/data/DATAprocess/chentw/zigongneimo/temp/P1.pdf",width = 12,height = 8)
DimPlot(STR,reduction="umap",label=T,label.size =7,pt.size = 0.9,group.by = "cell_type0103")+ggtitle("")+
  theme(title = element_text(size =0),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),axis.line = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 0))+theme(text = element_text(size=0))+
  scale_color_manual(values = c("#DB0C15","#387DB9","#4CA636","#EE7A00","#9E4894",
                                "#5555A4","#482013","#F9E800"))

dev.off()



#对上述分群进行小提琴图绘制-----
gene=c("PGR","OGN","MKI67","BMP7","CD34","ACTA2","RGS5","ISG15","APOD",
       "THY1","PDGFRA","COL1A1")



#2-1
P1=VlnPlot(STR,features = gene[1],pt.size = 0,ncol = 1,group.by = "cell_type0103")+
  scale_fill_manual(values = c("#DB0C15","#387DB9","#4CA636","#EE7A00","#9E4894",
                               "#5555A4","#482013","#F9E800"),limits=c("Normal","PGR+","OGN+","Cycling",
                                                                       "BMP7+/CD34+","Myfibrolast/Pericyte","ISG15+","APOD+"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(axis.text = element_blank(),title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P2=VlnPlot(STR,features = gene[2],pt.size = 0,ncol = 1,group.by = "cell_type0103")+
  scale_fill_manual(values = c("#DB0C15","#387DB9","#4CA636","#EE7A00","#9E4894",
                               "#5555A4","#482013","#F9E800"),limits=c("Normal","PGR+","OGN+","Cycling","BMP7+/CD34+","Myfibrolast/Pericyte","ISG15+","APOD+"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P3=VlnPlot(STR,features = gene[3],pt.size = 0,ncol = 1,group.by = "cell_type0103")+
  scale_fill_manual(values = c("#DB0C15","#387DB9","#4CA636","#EE7A00","#9E4894",
                               "#5555A4","#482013","#F9E800"),limits=c("Normal","PGR+","OGN+","Cycling","BMP7+/CD34+","Myfibrolast/Pericyte","ISG15+","APOD+"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P4=VlnPlot(STR,features = gene[4],pt.size = 0,ncol = 1,group.by = "cell_type0103")+
  scale_fill_manual(values = c("#DB0C15","#387DB9","#4CA636","#EE7A00","#9E4894",
                               "#5555A4","#482013","#F9E800"),limits=c("Normal","PGR+","OGN+","Cycling","BMP7+/CD34+","Myfibrolast/Pericyte","ISG15+","APOD+"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P5=VlnPlot(STR,features = gene[5],pt.size = 0,ncol = 1,group.by = "cell_type0103")+
  scale_fill_manual(values = c("#DB0C15","#387DB9","#4CA636","#EE7A00","#9E4894",
                               "#5555A4","#482013","#F9E800"),limits=c("Normal","PGR+","OGN+","Cycling","BMP7+/CD34+","Myfibrolast/Pericyte","ISG15+","APOD+"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P6=VlnPlot(STR,features = gene[6],pt.size = 0,ncol = 1,group.by = "cell_type0103")+
  scale_fill_manual(values = c("#DB0C15","#387DB9","#4CA636","#EE7A00","#9E4894",
                               "#5555A4","#482013","#F9E800"),limits=c("Normal","PGR+","OGN+","Cycling","BMP7+/CD34+","Myfibrolast/Pericyte","ISG15+","APOD+"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P7=VlnPlot(STR,features = gene[7],pt.size = 0,ncol = 1,group.by = "cell_type0103")+
  scale_fill_manual(values = c("#DB0C15","#387DB9","#4CA636","#EE7A00","#9E4894",
                               "#5555A4","#482013","#F9E800"),limits=c("Normal","PGR+","OGN+","Cycling","BMP7+/CD34+","Myfibrolast/Pericyte","ISG15+","APOD+"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P8=VlnPlot(STR,features = gene[8],pt.size = 0,ncol = 1,group.by = "cell_type0103")+
  scale_fill_manual(values = c("#DB0C15","#387DB9","#4CA636","#EE7A00","#9E4894",
                               "#5555A4","#482013","#F9E800"),limits=c("Normal","PGR+","OGN+","Cycling","BMP7+/CD34+","Myfibrolast/Pericyte","ISG15+","APOD+"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))

P9=VlnPlot(STR,features = gene[9],pt.size = 0,ncol = 1,group.by = "cell_type0103")+
  scale_fill_manual(values = c("#DB0C15","#387DB9","#4CA636","#EE7A00","#9E4894",
                               "#5555A4","#482013","#F9E800"),limits=c("Normal","PGR+","OGN+","Cycling","BMP7+/CD34+","Myfibrolast/Pericyte","ISG15+","APOD+"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))


P10=VlnPlot(STR,features = gene[10],pt.size = 0,ncol = 1,group.by = "cell_type0103")+
  scale_fill_manual(values = c("#DB0C15","#387DB9","#4CA636","#EE7A00","#9E4894",
                               "#5555A4","#482013","#F9E800"),limits=c("Normal","PGR+","OGN+","Cycling","BMP7+/CD34+","Myfibrolast/Pericyte","ISG15+","APOD+"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))


P11=VlnPlot(STR,features = gene[11],pt.size = 0,ncol = 1,group.by = "cell_type0103")+
  scale_fill_manual(values = c("#DB0C15","#387DB9","#4CA636","#EE7A00","#9E4894",
                               "#5555A4","#482013","#F9E800"),limits=c("Normal","PGR+","OGN+","Cycling","BMP7+/CD34+","Myfibrolast/Pericyte","ISG15+","APOD+"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))


P12=VlnPlot(STR,features = gene[12],pt.size = 0,ncol = 1,group.by = "cell_type0103")+
  scale_fill_manual(values = c("#DB0C15","#387DB9","#4CA636","#EE7A00","#9E4894",
                               "#5555A4","#482013","#F9E800"),limits=c("Normal","PGR+","OGN+","Cycling","BMP7+/CD34+","Myfibrolast/Pericyte","ISG15+","APOD+"))+coord_flip()+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))



pdf("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/P2.pdf",
    width = 8,height = 9)
ggarrange(P1,P2,P3,P4,P5,P6,P7,P8,P9,P10,P11,P12,common.legend = T,ncol = 4,legend = "none",nrow = 3)
dev.off()


#对cell_type0103的细胞类型进行marker基因查找以及气泡图绘制-----
###marker基因查找并进行富marker基因气泡图绘制并尝试定义
DefaultAssay(STR)<-"RNA"
STR@active.ident=STR$cell_type0103
STR_markers <- FindAllMarkers(object=STR, only.pos = TRUE,min.pct = 0.25, 
                              logfc.threshold = 0.25)
STR_markers <-subset(STR_markers,STR_markers$p_val<0.05)


write.csv(STR_markers,
          "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/marker基因/STR_markers_0103.csv")

mtGenes=grep('^MT-',rownames(STR),value=TRUE)
RPGenes=c(grep('^RPL',rownames(STR),value=TRUE),grep('^RPS',rownames(STR),value=TRUE))
stress_gene2=grep('^HSP',rownames(STR),value=TRUE)
HB_gene2=grep('^HB',rownames(STR),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
stress_gene1=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
               "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/gene1.csv")
subgene3=subgene3$cellcyle_gene
subgene4=c("FOS","CXCL2","ZFP36","FOSB","DUSP1","ATF3","CXCL8","NR4A1","CXCL3","PPP1R15A",
           "JUNB","EGR1","HSPA1A","HSPA1B","SOCS3","KLF6","JUN","IER2","CXCL1","NFKBIA",
           "HSPA6","DNAJB1","IER3","CCNL1","MTRNR2L2","IER4","ID1","CEBPD","KRT6A","CYR61",
           "DEPP1","CLDN4","IRF1","DUSP2","BTG2","PLAUR","MAFF","KLF4","PHLDA2","TNFAIP3")
subgene5=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/512个应激反应相关基因.csv")
subgene5=subgene5$gene_symbol

subgene=c(mtGenes,RPGenes,stress_gene2,HB_gene2,HB.genes,s.genes,g2m.genes,stress_gene1,"MKI67",subgene3,subgene4,subgene5)


#包含非特异的marker
STR_markers=subset(STR_markers,!gene%in%subgene)

STR_markers_top10 <- STR_markers %>% group_by(cluster) %>% top_n(n = 10, wt = avg_log2FC)

DefaultAssay(STR)<-"RNA"

STR_markers_top10=unique(STR_markers_top10$gene)

DotPlot(STR,features=STR_markers_top10,group.by = "cell_type0103")+
  theme(axis.text.x= element_text(size=9,angle = 45,vjust = 1,hjust = 1,face= "bold"))+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])))+xlab("")+ylab("")



#特异的marker
STR_markers=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/marker基因/STR_markers_0103.csv")

Z=STR_markers[duplicated(STR_markers$gene),]

N=subset(STR_markers,!gene%in%unique(Z$gene))

write.csv(N,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/marker基因/STR_markers_0103_unique.csv")

table(duplicated(N$gene))

STR_markers_top5 <- N %>% group_by(cluster) %>% top_n(n = 5, wt = avg_log2FC)

DefaultAssay(STR)<-"RNA"

STR_markers_top5=unique(STR_markers_top5$gene)

DotPlot(STR,features=STR_markers_top5,group.by = "cell_type0103")+
  theme(axis.text.x= element_text(size=9,angle = 45,vjust = 1,hjust = 1,face= "bold"))+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])))+xlab("")+ylab("")


##保存STR_markers_top5并且更改，然后进行一些更改添加并重新绘图
STR_markers_top5=as.data.frame(STR_markers_top5)

STR_markers_top5=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/marker基因/STR_markers_top5.csv")

STR_markers_top5=STR_markers_top5$STR_markers_top5

pdf("/data/DATAprocess/chentw/zigongneimo/temp/t1.pdf",width = 12,height = 4)
DotPlot(STR,features=STR_markers_top5,group.by = "cell_type0103")+
  theme(axis.text.x= element_text(size=9,angle = 45,vjust = 1,hjust = 1,face= "bold"))+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])))+xlab("")+ylab("")
dev.off()




#upsetR图绘制-----
library(UpSetR)
library(RColorBrewer)

###上调
S0_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/S0/S0_deseq.csv")
S0=subset(S0_deseq,S0_deseq$avg_log2FC>0)$X

S1_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/S1/S1_deseq.csv")
S1=subset(S1_deseq,S1_deseq$avg_log2FC>0)$X

REV3L_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/REV3L/REV3L_deseq.csv")
REV3L=subset(REV3L_deseq,REV3L_deseq$avg_log2FC>0)$X


APOD_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/APOD/APOD_deseq.csv")
APOD=subset(APOD_deseq,APOD_deseq$avg_log2FC>0)$X

BMP7_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/BMP7/BMP7_deseq.csv")
BMP7=subset(BMP7_deseq,BMP7_deseq$avg_log2FC>0)$X


ISG15_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/ISG15/ISG15_deseq.csv")
ISG15=subset(ISG15_deseq,ISG15_deseq$avg_log2FC>0)$X


P_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/P/P_deseq.csv")
P=subset(P_deseq,P_deseq$avg_log2FC>0)$X


Cycling_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/Cycling/Cycling_deseq.csv")
Cycling=subset(Cycling_deseq,Cycling_deseq$avg_log2FC>0)$X


set_list <- list("Normal"=S0,"PGR+"=REV3L,"OGN+"=S1,"Cycling"=Cycling,
                 "BMP7+/CD34+"=BMP7,"Myofibroblast/Pericyte"=P,"ISG15"=ISG15,
                 "APOD"=APOD)

pdf(file = "/data/DATAprocess/chentw/zigongneimo/temp/STR3.pdf",
    width = 10,height = 6)

upset(
  fromList(set_list),
  order.by = "freq",nsets = 8,shade.color="red",
  sets.bar.color=c("#EE7A00","#DB0C15","#4CA636",
                   "#387DB9","#9E4894","#5555A4",
                   "#482013","#F9E800"),
  sets.x.label="UP genes",mainbar.y.label="No. of up genes per cell type"
)

dev.off()

gene_interset=Reduce(intersect,list(S0,S1,BMP7,Cycling,REV3L,ISG15,APOD,P))
gene_interset


gene_interset=Reduce(intersect,list(S0,S1,BMP7,Cycling,REV3L,ISG15,APOD,P))
gene_interset


###根据这些共有的25个基因去除MT基因以后进行六组间的评分
gene_interset=gene_interset[-which(gene_interset%in%c("MT-ND5","MT-ND2"))]

STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

STR_data=as.data.frame(STR@assays$RNA@data)

STR_data=subset(STR_data,rownames(STR_data)%in%c(gene_interset))

STR_data=as.data.frame(t(STR_data))

STR_data$mean=apply(STR_data,1,mean)

STR_data$typ2=STR$type2
STR$gene=STR_data$mean

STR$type4="Adhesion"
STR$type4[STR$type2%in%"CTRL"]="Ctrl"
STR$type4[STR$type2%in%"IUA1"]="Adjacent"

STR$type4=factor(STR$type4,levels = c("Ctrl","Adjacent","Adhesion"))

pdf("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/vln.pdf",
    width =7,height = 5 )
VlnPlot(STR,features = "gene",pt.size = 0,split.by = "type4",group.by = "cell_type0103")+ggtitle("")+
  xlab("")+ylab("Fibrosis Score")
dev.off()


P1=DoHeatmap(STR,group.by = "type4",features = gene_interset,label = F)


saveRDS(P1,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/temp/p1.rds")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/heatmap.pdf",
       width = 6,height =7,dpi = 400)

###下调
S0_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/S0/S0_deseq.csv")
S0=subset(S0_deseq,S0_deseq$avg_log2FC<0)$X

S1_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/S1/S1_deseq.csv")
S1=subset(S1_deseq,S1_deseq$avg_log2FC<0)$X

REV3L_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/REV3L/REV3L_deseq.csv")
REV3L=subset(REV3L_deseq,REV3L_deseq$avg_log2FC<0)$X


APOD_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/APOD/APOD_deseq.csv")
APOD=subset(APOD_deseq,APOD_deseq$avg_log2FC<0)$X

BMP7_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/BMP7/BMP7_deseq.csv")
BMP7=subset(BMP7_deseq,BMP7_deseq$avg_log2FC<0)$X


ISG15_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/ISG15/ISG15_deseq.csv")
ISG15=subset(ISG15_deseq,ISG15_deseq$avg_log2FC<0)$X


P_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/P/P_deseq.csv")
P=subset(P_deseq,P_deseq$avg_log2FC<0)$X


Cycling_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/Cycling/Cycling_deseq.csv")
Cycling=subset(Cycling_deseq,Cycling_deseq$avg_log2FC<0)$X


set_list <- list("Normal"=S0,"PGR+"=REV3L,"OGN+"=S1,"Cycling"=Cycling,
                 "BMP7+/CD34+"=BMP7,"Myofibroblast/Pericyte"=P,"ISG15"=ISG15,
                 "APOD"=APOD)

pdf(file = "/data/DATAprocess/chentw/zigongneimo/temp/STR4.pdf",
    width = 10,height = 6)

upset(
  fromList(set_list),
  order.by = "freq",nsets = 8,shade.color="red",
  sets.bar.color=c("#5555A4","#EE7A00","#9E4894","#482013",
                   "#4CA636","#DB0C15","#387DB9","#F9E800"),
  sets.x.label="DOWN genes",mainbar.y.label="No. of down genes per cell type"
)

dev.off()

#富集通路GOPLOT------
chord=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/GO_PLOT/chord.csv",
               row.names = 1,header = T)

chord$logFC=-log10(chord$logFC)

GOChord(chord, space = 0.02, gene.order = 'logFC', 
        gene.space = 0.2, gene.size = 2,
        ribbon.col=c("#DB0C15","#387DB9","#4CA636","#EE7A00","#9E4894",
                     "#5555A4","#482013","#F9E800"))

saveRDS(P1,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/GO_PLOT/P1.rds")

chord=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/GO_PLOT/kegg_chord.csv",
               row.names = 1,header = T)

chord$logFC=-log10(chord$logFC)

GOChord(chord, space = 0.02, gene.order = 'logFC', 
        gene.space = 0.2, gene.size = 2,
        ribbon.col=c("#DB0C15","#387DB9","#4CA636","#EE7A00","#9E4894",
                     "#5555A4","#482013","#F9E800"))








#利用DecontX预测和去除RNA污染[STR],尝试一下----



#开始对STR细胞进行映射定义-----
table(STR$CE)







#去除ctrl组的APOD群的5个细胞并重新保存,之后开启新的脚本，在新的数据中都用新的数据操作-----
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

STR$cell_type4=paste(STR$type1,STR$cell_type0305,sep = "")


STR2=subset(STR,cell_type4!="CTRLAPOD")


saveRDS(STR2,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/rds存放点/STR_20240104.rds")









#对PGR+基质细胞和所有巨噬细胞【定义为一种亚型】进行细胞通讯以及比较-----
umape=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")
PGR=subset(STR,cell_type0103%in%"PGR+")
Macr=subset(umape,cell_type%in%"Macrophage")


PGR$cell_type0305="PGR+"
Macr$cell_type0305="Macr"


table(STR$cell_type0305)

PGR_Macr=merge(PGR,Macr)

CellChatDB <- CellChatDB.human

sc.3=subset(PGR_Macr,type1%in%"CTRL")
sc.11=subset(PGR_Macr,type1%in%"IUA")

cellchat.sc11 <- createCellChat(object =sc.11@assays$RNA@data, meta =sc.11@meta.data,  group.by ="cell_type0305")
cellchat.sc3 <- createCellChat(object =sc.3@assays$RNA@data, meta =sc.3@meta.data,  group.by ="cell_type0305")

setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/cellchat/0104/")

##Adhesion
cellchat=cellchat.sc11 
cellchat@DB  <- subsetDB(CellChatDB, search = "Secreted Signaling") # use Secreted Signaling
cellchat <- subsetData(cellchat)
future::plan("multiprocess", workers = 4)
cellchat <- identifyOverExpressedGenes(cellchat)
cellchat <- identifyOverExpressedInteractions(cellchat)
cellchat <- projectData(cellchat, PPI.human)
cellchat <- computeCommunProb(cellchat, raw.use = TRUE,population.size =T)
cellchat <- filterCommunication(cellchat, min.cells = 3)
cellchat <- computeCommunProbPathway(cellchat)
cellchat <- aggregateNet(cellchat)
cellchat <- netAnalysis_computeCentrality(cellchat, slot.name = "netP")
cc.sc11 = cellchat

##CTRL
cellchat=cellchat.sc3
cellchat@DB  <- subsetDB(CellChatDB, search = "Secreted Signaling") # use Secreted Signaling
cellchat <- subsetData(cellchat)
future::plan("multiprocess", workers = 4)
cellchat <- identifyOverExpressedGenes(cellchat)
cellchat <- identifyOverExpressedInteractions(cellchat)
cellchat <- projectData(cellchat, PPI.human)
cellchat <- computeCommunProb(cellchat, raw.use = TRUE,population.size =T)
cellchat <- filterCommunication(cellchat, min.cells = 3)
cellchat <- computeCommunProbPathway(cellchat)
cellchat <- aggregateNet(cellchat)
cellchat <- netAnalysis_computeCentrality(cellchat, slot.name = "netP")
cc.sc3 = cellchat



cc.list=list(CTRL=cc.sc3,IUA=cc.sc11)
cellchat=mergeCellChat(cc.list,cell.prefix = T,add.names = names(cc.list))
##可视化
##所有细胞群总体观：通讯数量与强度对比
P1=compareInteractions(cellchat,show.legend = F,group = c(1,3),measure = "count")
P2=compareInteractions(cellchat,show.legend = F,group = c(1,3),measure = "weight")

P1+P2

##第一个图展示通讯数量之间的差异，第二个图展示通讯强度之间的差异。 

##数量与强度差异网络图
netVisual_diffInteraction(cellchat,weight.scale = T)
netVisual_diffInteraction(cellchat,weight.scale = T,measure = "weight")
##红色是case相对于control上调的，蓝色是下调的

#数量与强度差异热图
netVisual_heatmap(cellchat)
netVisual_heatmap(cellchat,measure = "weight")
#case和control对比，红色是上调，蓝色是下调

#保守和特异性信号通路的识别与可视化
rankNet(cellchat,mode = "comparison",stacked = T,do.stat = T)
rankNet(cellchat,mode = "comparison",stacked =F,do.stat = T)




par(mfrow=c(1,2),xpd=T)
weight.max=getMaxWeight(cc.list,attribute = c("idents","weight"))
for (i in 1:length(cc.list)) {
  netVisual_circle(cc.list[[i]]@net$weight,weight.scale = T,label.edge = F,vertex.label.cex = 1.8,edge.label.cex = 1.8,
                   edge.weight.max =weight.max[2],edge.width.max = 12,title.name = names(cc.list)[i])
}



par(mfrow=c(1,2),xpd=T)
weight.max=getMaxWeight(cc.list,attribute = c("idents","weight"))
for (i in 1:length(cc.list)) {
  netVisual_circle(cc.list[[i]]@net$weight,weight.scale = T,label.edge = F,vertex.label.cex = 1.8,edge.label.cex = 1.8,
                   edge.weight.max =weight.max[2],edge.width.max = 12,title.name = names(cc.list)[i])
}



object.list=cc.list


weight.max=getMaxWeight(object.list,attribute = c("idents","count"))
par(mfrow=c(1,2),xpd=T)



for (i in 1:length(object.list)) {
  netVisual_circle(object.list[[i]]@net$count,weight.scale = T,label.edge = T,vertex.label.cex = 1.8,edge.label.cex = 1.8,
                   edge.weight.max =weight.max[2],edge.width.max = 12,title.name = names(object.list)[i])
}



saveRDS(cc.sc3,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/cellchat/0104/CTRL.rds")

saveRDS(cc.sc11,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/cellchat/0104/IUA.rds")


#提取雌激素基因----
library(msigdbr)
all_gene_sets = msigdbr(species = "Homo sapiens")

K=as.data.frame(names(table(all_gene_sets$gs_description)))


write.csv(K,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/富集分析/msigdb_term.csv")


term1=c("Genes defining early response to estrogen.")
term2=c("Genes defining late response to estrogen.")
term3=c("Estrogen signaling pathway")

Z1=subset(all_gene_sets,all_gene_sets$gs_description%in%term1)
Z2=subset(all_gene_sets,all_gene_sets$gs_description%in%term2)
Z3=subset(all_gene_sets,all_gene_sets$gs_description%in%term3)


gene1=Z1$human_gene_symbol
gene2=Z2$human_gene_symbol
gene3=Z3$human_gene_symbol

gene1<-gene1[!duplicated(gene1)]
gene2<-gene2[!duplicated(gene2)]
gene3<-gene3[!duplicated(gene3)]

term=cbind(gene1,gene2)

colnames(term)=c("Early","late","Estrogen")


write.csv(term,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/富集分析/estrogen.csv")



###对基质细胞进行雌激素反应的各个cluster的小提琴图绘制
EnSc=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0718.rds")
DimPlot(EnSc,group.by = "cell_type0305",label = T)


EnSc=subset(EnSc,RNA_snn_res.0.3!="7")

#EnSc进行标准化然后PCA分析
DefaultAssay(EnSc) <- "RNA"
all.genes_EnSc <- rownames(EnSc)
EnSc <- ScaleData(object = EnSc, features = all.genes_EnSc)

EnSc$cell_type0718=paste("Cluster",EnSc$RNA_snn_res.0.3,sep = "")

DimPlot(EnSc,group.by = "cell_type0718",label = T)

EnSc =  AddModuleScore(object = EnSc,list(gene1),name = "Early")
EnSc =  AddModuleScore(object = EnSc,list(gene2),name = "Late")





VlnPlot(EnSc,group.by = "type4",features = "Cluster1",
        pt.size = 0)+ylab("Early estrogen response")+
  xlab("")+ggtitle("")+
  stat_compare_means(comparisons = my_comparisons)+
  ylim(-0.1,0.2)



VlnPlot(EnSc,group.by = "type4",features = "Late1",
        pt.size = 0)+ylab("Late estrogen response")+
  xlab("")+ggtitle("")+
  stat_compare_means(comparisons = my_comparisons)+
  ylim(-0.1,0.2)

K1=as.data.frame(AverageExpression(EnSc,features = "Early",group.by = "type4")$RNA)
K2=as.data.frame(AverageExpression(EnSc,features = "Late",group.by = "type4")$RNA)

K=rbind(K1,K2)

rownames(K)<-c("Early estrogen response","Late estrogen response")

pheatmap(K,cluster_rows = F,cluster_cols = F,scale = "row")

#基质细胞-----
##先把Ctrl和Adjacent进行比较
#对early和late进行半小提琴图绘制
meta=as.data.frame(EnSc@meta.data)


meta=meta[c("type3","type4","cell_type0718","Cluster1","Late1")]

meta$cell_type="EnSc"

colnames(meta)[2]<-"group"

meta=subset(meta,meta$group%in%c("Ctrl","Adjacent"))

meta$group=factor(meta$group)

ggplot()+
  geom_half_violin(
    data = meta %>% filter(group == "Ctrl"),
    aes(x = cell_type0718,y = Cluster1),colour="white",fill="#1ba7b3",side = "l")+
  geom_half_violin(
    data = meta %>% filter(group == "Adjacent"),
    aes(x = cell_type0718,y = Cluster1),colour="white",fill="#dfb424",side = "r")+
  theme_bw()+xlab("")+ylab("Early estrogen response")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,face = "bold",colour = "black",size =10 ),
        axis.text.y = element_text(face = "bold",colour = "black",size =10 ),
        axis.title.y =element_text(face = "bold",colour = "black",size =10 ))+
  stat_summary(data = meta, aes(x = cell_type0718,y = Cluster1, fill = group),
               fun.min = function(x){quantile(x)[2]},
               fun.max = function(x){quantile(x)[4]},
               geom = 'errorbar', color='black',
               width=0.01,size=0.5,
               position = position_dodge(width = 0.2))+
  stat_compare_means(data = meta, aes(x = cell_type0718,y = Cluster1, fill = group),
                     symnum.args=list(cutpoints = c(0, 0.001, 0.01, 0.05, 1),
                                      symbols = c("***", "**", "*", "-")),
                     label = "p.signif",
                     label.y = 0.11,
                     hide.ns = F)

ggsave("/data/DATAprocess/chentw/zigongneimo/temp/20240725/EnSc_adjacey_Early_1.pdf",
       width = 7,height = 3)

ggplot()+
  geom_half_violin(
    data = meta %>% filter(group == "Ctrl"),
    aes(x = cell_type,y = Cluster1),colour="white",fill="#1ba7b3",side = "l")+
  geom_half_violin(
    data = meta %>% filter(group == "Adjacent"),
    aes(x = cell_type,y = Cluster1),colour="white",fill="#dfb424",side = "r")+
  theme_bw()+xlab("")+ylab("Early estrogen response")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,face = "bold",colour = "black",size =10 ),
        axis.text.y = element_text(face = "bold",colour = "black",size =10 ),
        axis.title.y =element_text(face = "bold",colour = "black",size =10 ))+
  stat_summary(data = meta, aes(x = cell_type,y = Cluster1, fill = group),
               fun.min = function(x){quantile(x)[2]},
               fun.max = function(x){quantile(x)[4]},
               geom = 'errorbar', color='black',
               width=0.01,size=0.5,
               position = position_dodge(width = 0.2))+
  stat_compare_means(data = meta, aes(x = cell_type,y = Cluster1, fill = group),
                     symnum.args=list(cutpoints = c(0, 0.001, 0.01, 0.05, 1),
                                      symbols = c("***", "**", "*", "-")),
                     label = "p.signif",
                     label.y = 0.11,
                     hide.ns = F)

ggsave("/data/DATAprocess/chentw/zigongneimo/temp/20240725/EnSc_adjacey_Early_2.pdf",
       width = 3,height = 3)



ggplot()+
  geom_half_violin(
    data = meta %>% filter(group == "Ctrl"),
    aes(x = cell_type0718,y = Late1),colour="white",fill="#1ba7b3",side = "l")+
  geom_half_violin(
    data = meta %>% filter(group == "Adjacent"),
    aes(x = cell_type0718,y = Late1),colour="white",fill="#dfb424",side = "r")+
  theme_bw()+xlab("")+ylab("Late estrogen response")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,face = "bold",colour = "black",size =10 ),
        axis.text.y = element_text(face = "bold",colour = "black",size =10 ),
        axis.title.y =element_text(face = "bold",colour = "black",size =10 ))+
  stat_summary(data = meta, aes(x = cell_type0718,y = Late1, fill = group),
               fun.min = function(x){quantile(x)[2]},
               fun.max = function(x){quantile(x)[4]},
               geom = 'errorbar', color='black',
               width=0.01,size=0.5,
               position = position_dodge(width = 0.2))+
  stat_compare_means(data = meta, aes(x = cell_type0718,y = Late1, fill = group),
                     symnum.args=list(cutpoints = c(0, 0.001, 0.01, 0.05, 1),
                                      symbols = c("***", "**", "*", "-")),
                     label = "p.signif",
                     label.y = 0.11,
                     hide.ns = F)

ggsave("/data/DATAprocess/chentw/zigongneimo/temp/20240725/EnSc_adjacey_Late_1.pdf",
       width = 7,height = 3)

ggplot()+
  geom_half_violin(
    data = meta %>% filter(group == "Ctrl"),
    aes(x = cell_type,y = Late1),colour="white",fill="#1ba7b3",side = "l")+
  geom_half_violin(
    data = meta %>% filter(group == "Adjacent"),
    aes(x = cell_type,y = Late1),colour="white",fill="#dfb424",side = "r")+
  theme_bw()+xlab("")+ylab("Late estrogen response")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,face = "bold",colour = "black",size =10 ),
        axis.text.y = element_text(face = "bold",colour = "black",size =10 ),
        axis.title.y =element_text(face = "bold",colour = "black",size =10 ))+
  stat_summary(data = meta, aes(x = cell_type,y = Late1, fill = group),
               fun.min = function(x){quantile(x)[2]},
               fun.max = function(x){quantile(x)[4]},
               geom = 'errorbar', color='black',
               width=0.01,size=0.5,
               position = position_dodge(width = 0.2))+
  stat_compare_means(data = meta, aes(x = cell_type,y = Late1, fill = group),
                     symnum.args=list(cutpoints = c(0, 0.001, 0.01, 0.05, 1),
                                      symbols = c("***", "**", "*", "-")),
                     label = "p.signif",
                     label.y = 0.13,
                     hide.ns = F)

ggsave("/data/DATAprocess/chentw/zigongneimo/temp/20240725/EnSc_adjacey_Late_2.pdf",
       width = 3,height = 3)



meta=as.data.frame(EnSc@meta.data)


meta=meta[c("type3","type4","cell_type0718","Cluster1","Late1")]

meta$cell_type="EnSc"

colnames(meta)[2]<-"group"

meta=subset(meta,meta$group%in%c("Ctrl","Adhesion"))

meta$group=factor(meta$group)

ggplot()+
  geom_half_violin(
    data = meta %>% filter(group == "Ctrl"),
    aes(x = cell_type0718,y = Cluster1),colour="white",fill="#1ba7b3",side = "l")+
  geom_half_violin(
    data = meta %>% filter(group == "Adhesion"),
    aes(x = cell_type0718,y = Cluster1),colour="white",fill="#dfb424",side = "r")+
  theme_bw()+xlab("")+ylab("Early estrogen response")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,face = "bold",colour = "black",size =10 ),
        axis.text.y = element_text(face = "bold",colour = "black",size =10 ),
        axis.title.y =element_text(face = "bold",colour = "black",size =10 ))+
  stat_summary(data = meta, aes(x = cell_type0718,y = Cluster1, fill = group),
               fun.min = function(x){quantile(x)[2]},
               fun.max = function(x){quantile(x)[4]},
               geom = 'errorbar', color='black',
               width=0.01,size=0.5,
               position = position_dodge(width = 0.2))+
  stat_compare_means(data = meta, aes(x = cell_type0718,y = Cluster1, fill = group),
                     symnum.args=list(cutpoints = c(0, 0.001, 0.01, 0.05, 1),
                                      symbols = c("***", "**", "*", "-")),
                     label = "p.signif",
                     label.y = 0.09,
                     hide.ns = F)

ggsave("/data/DATAprocess/chentw/zigongneimo/temp/20240725/EnSc_adhesion_Early_1.pdf",
       width = 7,height = 3)





ggplot()+
  geom_half_violin(
    data = meta %>% filter(group == "Ctrl"),
    aes(x = cell_type,y = Cluster1),colour="white",fill="#1ba7b3",side = "l")+
  geom_half_violin(
    data = meta %>% filter(group == "Adhesion"),
    aes(x = cell_type,y = Cluster1),colour="white",fill="#dfb424",side = "r")+
  theme_bw()+xlab("")+ylab("Early estrogen response")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,face = "bold",colour = "black",size =10 ),
        axis.text.y = element_text(face = "bold",colour = "black",size =10 ),
        axis.title.y =element_text(face = "bold",colour = "black",size =10 ))+
  stat_summary(data = meta, aes(x = cell_type,y = Cluster1, fill = group),
               fun.min = function(x){quantile(x)[2]},
               fun.max = function(x){quantile(x)[4]},
               geom = 'errorbar', color='black',
               width=0.01,size=0.5,
               position = position_dodge(width = 0.2))+
  stat_compare_means(data = meta, aes(x = cell_type,y = Cluster1, fill = group),
                     symnum.args=list(cutpoints = c(0, 0.001, 0.01, 0.05, 1),
                                      symbols = c("***", "**", "*", "-")),
                     label = "p.signif",
                     label.y = 0.105,
                     hide.ns = F)

ggsave("/data/DATAprocess/chentw/zigongneimo/temp/20240725/EnSc_adhesion_Early_2.pdf",
       width = 3,height = 3)




ggplot()+
  geom_half_violin(
    data = meta %>% filter(group == "Ctrl"),
    aes(x = cell_type0718,y = Late1),colour="white",fill="#1ba7b3",side = "l")+
  geom_half_violin(
    data = meta %>% filter(group == "Adhesion"),
    aes(x = cell_type0718,y = Late1),colour="white",fill="#dfb424",side = "r")+
  theme_bw()+xlab("")+ylab("Late estrogen response")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,face = "bold",colour = "black",size =10 ),
        axis.text.y = element_text(face = "bold",colour = "black",size =10 ),
        axis.title.y =element_text(face = "bold",colour = "black",size =10 ))+
  stat_summary(data = meta, aes(x = cell_type0718,y = Late1, fill = group),
               fun.min = function(x){quantile(x)[2]},
               fun.max = function(x){quantile(x)[4]},
               geom = 'errorbar', color='black',
               width=0.01,size=0.5,
               position = position_dodge(width = 0.2))+
  stat_compare_means(data = meta, aes(x = cell_type0718,y = Late1, fill = group),
                     symnum.args=list(cutpoints = c(0, 0.001, 0.01, 0.05, 1),
                                      symbols = c("***", "**", "*", "-")),
                     label = "p.signif",
                     label.y = 0.11,
                     hide.ns = F)

ggsave("/data/DATAprocess/chentw/zigongneimo/temp/20240725/EnSc_adhesion_Late_1.pdf",
       width = 7,height = 3)





ggplot()+
  geom_half_violin(
    data = meta %>% filter(group == "Ctrl"),
    aes(x = cell_type,y = Late1),colour="white",fill="#1ba7b3",side = "l")+
  geom_half_violin(
    data = meta %>% filter(group == "Adhesion"),
    aes(x = cell_type,y = Late1),colour="white",fill="#dfb424",side = "r")+
  theme_bw()+xlab("")+ylab("Late estrogen response")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,face = "bold",colour = "black",size =10 ),
        axis.text.y = element_text(face = "bold",colour = "black",size =10 ),
        axis.title.y =element_text(face = "bold",colour = "black",size =10 ))+
  stat_summary(data = meta, aes(x = cell_type,y = Late1, fill = group),
               fun.min = function(x){quantile(x)[2]},
               fun.max = function(x){quantile(x)[4]},
               geom = 'errorbar', color='black',
               width=0.01,size=0.5,
               position = position_dodge(width = 0.2))+
  stat_compare_means(data = meta, aes(x = cell_type,y = Late1, fill = group),
                     symnum.args=list(cutpoints = c(0, 0.001, 0.01, 0.05, 1),
                                      symbols = c("***", "**", "*", "-")),
                     label = "p.signif",
                     label.y = 0.13,
                     hide.ns = F)

ggsave("/data/DATAprocess/chentw/zigongneimo/temp/20240725/EnSc_adhesion_Late_2.pdf",
       width = 3,height = 3)



##上皮细胞----
EPI=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/rds文件存放点/EPI_harmony_1218_2.rds")



EPI =  AddModuleScore(object = EPI,list(gene1),name = "Early")
EPI =  AddModuleScore(object = EPI,list(gene2),name = "Late")




meta=as.data.frame(EPI@meta.data)


meta=meta[c("type3","type4","cell_type0507","Cluster1","Late1")]

meta$cell_type="EPI"

colnames(meta)[2]<-"group"

meta=subset(meta,meta$group%in%c("Ctrl","Adjacent"))

meta$group=factor(meta$group)

meta$cell_type0507=gsub("Epithelium","",meta$cell_type0507)

ggplot()+
  geom_half_violin(
    data = meta %>% filter(group == "Ctrl"),
    aes(x = cell_type0507,y = Cluster1),colour="white",fill="#1ba7b3",side = "l")+
  geom_half_violin(
    data = meta %>% filter(group == "Adjacent"),
    aes(x = cell_type0507,y = Cluster1),colour="white",fill="#dfb424",side = "r")+
  theme_bw()+xlab("")+ylab("Early estrogen response")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,face = "bold",colour = "black",size =10 ),
        axis.text.y = element_text(face = "bold",colour = "black",size =10 ),
        axis.title.y =element_text(face = "bold",colour = "black",size =10 ))+
  stat_summary(data = meta, aes(x = cell_type0507,y = Cluster1, fill = group),
               fun.min = function(x){quantile(x)[2]},
               fun.max = function(x){quantile(x)[4]},
               geom = 'errorbar', color='black',
               width=0.01,size=0.5,
               position = position_dodge(width = 0.2))+
  stat_compare_means(data = meta, aes(x = cell_type0507,y = Cluster1, fill = group),
                     symnum.args=list(cutpoints = c(0, 0.001, 0.01, 0.05, 1),
                                      symbols = c("***", "**", "*", "-")),
                     label = "p.signif",
                     label.y = 0.16,
                     hide.ns = F)+ylim(-0.13,0.2)

ggsave("/data/DATAprocess/chentw/zigongneimo/temp/20240725/EPI_adjacey_Early_1.pdf",
       width = 7,height = 3)

ggplot()+
  geom_half_violin(
    data = meta %>% filter(group == "Ctrl"),
    aes(x = cell_type,y = Cluster1),colour="white",fill="#1ba7b3",side = "l")+
  geom_half_violin(
    data = meta %>% filter(group == "Adjacent"),
    aes(x = cell_type,y = Cluster1),colour="white",fill="#dfb424",side = "r")+
  theme_bw()+xlab("")+ylab("Early estrogen response")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,face = "bold",colour = "black",size =10 ),
        axis.text.y = element_text(face = "bold",colour = "black",size =10 ),
        axis.title.y =element_text(face = "bold",colour = "black",size =10 ))+
  stat_summary(data = meta, aes(x = cell_type,y = Cluster1, fill = group),
               fun.min = function(x){quantile(x)[2]},
               fun.max = function(x){quantile(x)[4]},
               geom = 'errorbar', color='black',
               width=0.01,size=0.5,
               position = position_dodge(width = 0.2))+
  stat_compare_means(data = meta, aes(x = cell_type,y = Cluster1, fill = group),
                     symnum.args=list(cutpoints = c(0, 0.001, 0.01, 0.05, 1),
                                      symbols = c("***", "**", "*", "-")),
                     label = "p.signif",
                     label.y = 0.145,
                     hide.ns = F)

ggsave("/data/DATAprocess/chentw/zigongneimo/temp/20240725/EPI_adjacey_Early_2.pdf",
       width = 3,height = 3)



ggplot()+
  geom_half_violin(
    data = meta %>% filter(group == "Ctrl"),
    aes(x = cell_type0507,y = Late1),colour="white",fill="#1ba7b3",side = "l")+
  geom_half_violin(
    data = meta %>% filter(group == "Adjacent"),
    aes(x = cell_type0507,y = Late1),colour="white",fill="#dfb424",side = "r")+
  theme_bw()+xlab("")+ylab("Late estrogen response")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,face = "bold",colour = "black",size =10 ),
        axis.text.y = element_text(face = "bold",colour = "black",size =10 ),
        axis.title.y =element_text(face = "bold",colour = "black",size =10 ))+
  stat_summary(data = meta, aes(x = cell_type0507,y = Late1, fill = group),
               fun.min = function(x){quantile(x)[2]},
               fun.max = function(x){quantile(x)[4]},
               geom = 'errorbar', color='black',
               width=0.01,size=0.5,
               position = position_dodge(width = 0.2))+
  stat_compare_means(data = meta, aes(x = cell_type0507,y = Late1, fill = group),
                     symnum.args=list(cutpoints = c(0, 0.001, 0.01, 0.05, 1),
                                      symbols = c("***", "**", "*", "-")),
                     label = "p.signif",
                     label.y = 0.16,
                     hide.ns = F)

ggsave("/data/DATAprocess/chentw/zigongneimo/temp/20240725/EPI_adjacey_Late_1.pdf",
       width = 7,height = 3)

ggplot()+
  geom_half_violin(
    data = meta %>% filter(group == "Ctrl"),
    aes(x = cell_type,y = Late1),colour="white",fill="#1ba7b3",side = "l")+
  geom_half_violin(
    data = meta %>% filter(group == "Adjacent"),
    aes(x = cell_type,y = Late1),colour="white",fill="#dfb424",side = "r")+
  theme_bw()+xlab("")+ylab("Late estrogen response")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,face = "bold",colour = "black",size =10 ),
        axis.text.y = element_text(face = "bold",colour = "black",size =10 ),
        axis.title.y =element_text(face = "bold",colour = "black",size =10 ))+
  stat_summary(data = meta, aes(x = cell_type,y = Late1, fill = group),
               fun.min = function(x){quantile(x)[2]},
               fun.max = function(x){quantile(x)[4]},
               geom = 'errorbar', color='black',
               width=0.01,size=0.5,
               position = position_dodge(width = 0.2))+
  stat_compare_means(data = meta, aes(x = cell_type,y = Late1, fill = group),
                     symnum.args=list(cutpoints = c(0, 0.001, 0.01, 0.05, 1),
                                      symbols = c("***", "**", "*", "-")),
                     label = "p.signif",
                     label.y = 0.18,
                     hide.ns = F)

ggsave("/data/DATAprocess/chentw/zigongneimo/temp/20240725/EPI_adjacey_Late_2.pdf",
       width = 3,height = 3)



meta=as.data.frame(EPI@meta.data)


meta=meta[c("type3","type4","cell_type0507","Cluster1","Late1")]

meta$cell_type="EPI"

colnames(meta)[2]<-"group"

meta=subset(meta,meta$group%in%c("Ctrl","Adhesion"))

meta$group=factor(meta$group)

meta$cell_type0507=gsub("Epithelium","",meta$cell_type0507)


ggplot()+
  geom_half_violin(
    data = meta %>% filter(group == "Ctrl"),
    aes(x = cell_type0507,y = Cluster1),colour="white",fill="#1ba7b3",side = "l")+
  geom_half_violin(
    data = meta %>% filter(group == "Adhesion"),
    aes(x = cell_type0507,y = Cluster1),colour="white",fill="#dfb424",side = "r")+
  theme_bw()+xlab("")+ylab("Early estrogen response")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,face = "bold",colour = "black",size =10 ),
        axis.text.y = element_text(face = "bold",colour = "black",size =10 ),
        axis.title.y =element_text(face = "bold",colour = "black",size =10 ))+
  stat_summary(data = meta, aes(x = cell_type0507,y = Cluster1, fill = group),
               fun.min = function(x){quantile(x)[2]},
               fun.max = function(x){quantile(x)[4]},
               geom = 'errorbar', color='black',
               width=0.01,size=0.5,
               position = position_dodge(width = 0.2))+
  stat_compare_means(data = meta, aes(x = cell_type0507,y = Cluster1, fill = group),
                     symnum.args=list(cutpoints = c(0, 0.001, 0.01, 0.05, 1),
                                      symbols = c("***", "**", "*", "-")),
                     label = "p.signif",
                     label.y = 0.17,
                     hide.ns = F)

ggsave("/data/DATAprocess/chentw/zigongneimo/temp/20240725/EPI_adhesion_Early_1.pdf",
       width = 7,height = 3)





ggplot()+
  geom_half_violin(
    data = meta %>% filter(group == "Ctrl"),
    aes(x = cell_type,y = Cluster1),colour="white",fill="#1ba7b3",side = "l")+
  geom_half_violin(
    data = meta %>% filter(group == "Adhesion"),
    aes(x = cell_type,y = Cluster1),colour="white",fill="#dfb424",side = "r")+
  theme_bw()+xlab("")+ylab("Early estrogen response")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,face = "bold",colour = "black",size =10 ),
        axis.text.y = element_text(face = "bold",colour = "black",size =10 ),
        axis.title.y =element_text(face = "bold",colour = "black",size =10 ))+
  stat_summary(data = meta, aes(x = cell_type,y = Cluster1, fill = group),
               fun.min = function(x){quantile(x)[2]},
               fun.max = function(x){quantile(x)[4]},
               geom = 'errorbar', color='black',
               width=0.01,size=0.5,
               position = position_dodge(width = 0.2))+
  stat_compare_means(data = meta, aes(x = cell_type,y = Cluster1, fill = group),
                     symnum.args=list(cutpoints = c(0, 0.001, 0.01, 0.05, 1),
                                      symbols = c("***", "**", "*", "-")),
                     label = "p.signif",
                     label.y = 0.16,
                     hide.ns = F)

ggsave("/data/DATAprocess/chentw/zigongneimo/temp/20240725/EPI_adhesion_Early_2.pdf",
       width = 3,height = 3)




ggplot()+
  geom_half_violin(
    data = meta %>% filter(group == "Ctrl"),
    aes(x = cell_type0507,y = Late1),colour="white",fill="#1ba7b3",side = "l")+
  geom_half_violin(
    data = meta %>% filter(group == "Adhesion"),
    aes(x = cell_type0507,y = Late1),colour="white",fill="#dfb424",side = "r")+
  theme_bw()+xlab("")+ylab("Late estrogen response")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,face = "bold",colour = "black",size =10 ),
        axis.text.y = element_text(face = "bold",colour = "black",size =10 ),
        axis.title.y =element_text(face = "bold",colour = "black",size =10 ))+
  stat_summary(data = meta, aes(x = cell_type0507,y = Late1, fill = group),
               fun.min = function(x){quantile(x)[2]},
               fun.max = function(x){quantile(x)[4]},
               geom = 'errorbar', color='black',
               width=0.01,size=0.5,
               position = position_dodge(width = 0.2))+
  stat_compare_means(data = meta, aes(x = cell_type0507,y = Late1, fill = group),
                     symnum.args=list(cutpoints = c(0, 0.001, 0.01, 0.05, 1),
                                      symbols = c("***", "**", "*", "-")),
                     label = "p.signif",
                     label.y = 0.18,
                     hide.ns = F)

ggsave("/data/DATAprocess/chentw/zigongneimo/temp/20240725/EPI_adhesion_Late_1.pdf",
       width = 7,height = 3)





ggplot()+
  geom_half_violin(
    data = meta %>% filter(group == "Ctrl"),
    aes(x = cell_type,y = Late1),colour="white",fill="#1ba7b3",side = "l")+
  geom_half_violin(
    data = meta %>% filter(group == "Adhesion"),
    aes(x = cell_type,y = Late1),colour="white",fill="#dfb424",side = "r")+
  theme_bw()+xlab("")+ylab("Late estrogen response")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,face = "bold",colour = "black",size =10 ),
        axis.text.y = element_text(face = "bold",colour = "black",size =10 ),
        axis.title.y =element_text(face = "bold",colour = "black",size =10 ))+
  stat_summary(data = meta, aes(x = cell_type,y = Late1, fill = group),
               fun.min = function(x){quantile(x)[2]},
               fun.max = function(x){quantile(x)[4]},
               geom = 'errorbar', color='black',
               width=0.01,size=0.5,
               position = position_dodge(width = 0.2))+
  stat_compare_means(data = meta, aes(x = cell_type,y = Late1, fill = group),
                     symnum.args=list(cutpoints = c(0, 0.001, 0.01, 0.05, 1),
                                      symbols = c("***", "**", "*", "-")),
                     label = "p.signif",
                     label.y = 0.17,
                     hide.ns = F)

ggsave("/data/DATAprocess/chentw/zigongneimo/temp/20240725/EPI_adhesion_Late_2.pdf",
       width = 3,height = 3)













#基质细胞进行富集分析----

STR=EnSc

DefaultAssay(STR)<-"RNA"
STR@active.ident=as.factor(STR$cell_type0718)
STR_markers <- FindAllMarkers(object=STR, only.pos = TRUE,min.pct = 0.25, 
                              logfc.threshold = 0.25)
STR_markers <-subset(STR_markers,STR_markers$p_val<0.05)

GO_STR=function(X){
  N=as.character(X)
  N=subset(STR_markers,STR_markers$cluster%in%N)
  N=N$gene
  G <- bitr(N, fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
  G=G$ENTREZID
  ego <- enrichGO(OrgDb="org.Hs.eg.db", gene = G, ont = "ALL",readable= TRUE)
  file1=paste("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/0718富集/",as.character(X),"_ego.csv",sep = "")
  print(file1)
  write.csv(ego,file1)
}

for (i in names(table(STR_markers$cluster))) {GO_STR(i)
}

#对机器学习进行检验操作----
library(mlbench)
library(Seurat)
library(caret)
library(glmnet)


data(Sonar)

inTraining <- createDataPartition(Sonar$Class, p = .75, list = FALSE)


training <- Sonar[inTraining,]
testing  <- Sonar[-inTraining,]


fitControl <- trainControl(
  method = "repeatedcv", 
  number = 10,
  repeats = 10)

set.seed(825)
gbmFit1 <- train(Class ~ ., 
                 data = training, 
                 method = "gbm", 
                 trControl = fitControl,
                 verbose = FALSE)
gbmFit1



# 网格搜索，首先设定超参数范围
gbmGrid <-  expand.grid(interaction.depth = c(1, 5, 9), 
                        n.trees = (1:30)*50, 
                        shrinkage = 0.1,
                        n.minobsinnode = 20)

nrow(gbmGrid)
## [1] 90
head(gbmGrid)

# 设置种子数，进行建模
set.seed(825)
gbmFit2 <- train(Class ~ ., 
                 data = training, 
                 method = "gbm", 
                 trControl = fitControl, 
                 verbose = FALSE, 
                 ## Now specify the exact models 
                 ## to evaluate:
                 tuneGrid = gbmGrid # 设定网格范围
)
gbmFit2

trellis.par.set(caretTheme())
plot(gbmFit2)  

trellis.par.set(caretTheme())
plot(gbmFit2, metric = "Kappa")

trellis.par.set(caretTheme())
plot(gbmFit2, metric = "Kappa", plotType = "level",
     scales = list(x = list(rot = 90)))


ggplot(gbmFit2) 


fitControl <- trainControl(method = "repeatedcv",
                           number = 10,
                           repeats = 10,
                           classProbs = TRUE, 
                           summaryFunction = twoClassSummary)

set.seed(825)
gbmFit3 <- train(Class ~ ., 
                 data = training, 
                 method = "gbm", 
                 trControl = fitControl, 
                 verbose = FALSE, 
                 tuneGrid = gbmGrid,
                 metric = "ROC" )
gbmFit3


predict(gbmFit3, newdata = head(testing), type = "prob")
predict(gbmFit3, newdata = head(testing))


gbmFit3$finalModel # 最终模型
gbmFit3$bestTune # 选择的超参数
gbmFit3$results # 包含各种指标的详细结果



# 多建立几个模型
set.seed(825)
svmFit <- train(Class ~ ., 
                data = training, 
                method = "svmRadial", 
                trControl = fitControl, 
                preProc = c("center", "scale"),
                tuneLength = 8,
                metric = "ROC")

set.seed(825)
rdaFit <- train(Class ~ ., 
                data = training, 
                method = "rda", 
                trControl = fitControl, 
                tuneLength = 4,
                metric = "ROC")

#将多个模型放在一起比较
resamps <- resamples(list(GBM = gbmFit3,
                          SVM = svmFit,
                          RDA = rdaFit))
resamps



##模型比较展示绘图
# 设主题
theme1 <- trellis.par.get()
theme1$plot.symbol$col = rgb(.2, .2, .2, .4)
theme1$plot.symbol$pch = 16
theme1$plot.line$col = rgb(1, 0, 0, .7)
theme1$plot.line$lwd <- 2

# 画图，箱线图
trellis.par.set(theme1)
bwplot(resamps, layout = c(3, 1))

# 密度图
trellis.par.set(theme1)
densityplot(resamps)


# 换个指标，点线图
trellis.par.set(caretTheme())
dotplot(resamps, metric = "ROC")


# 散点图
trellis.par.set(theme1)
xyplot(resamps, what = "BlandAltman")

# 散点图矩阵
splom(resamps)



##模型之间的显著性检测
difValues <- diff(resamps)
difValues

summary(difValues)

trellis.par.set(theme1)
bwplot(difValues, layout = c(3, 1))

trellis.par.set(caretTheme())
dotplot(difValues)
















#利用基质细胞进行机器学习挑选特征基因-----
EnSc=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0718.rds")

library(Seurat)
library(caret)
library(glmnet)



DimPlot(EnSc,group.by = "cell_type0718",label = T,label.size = 10)


data=as.data.frame(t(EnSc@assays$RNA@data))

data$type4=EnSc$type4







##对所有细胞类型中的四种主要细胞类型进行相关性分析绘图1----
umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")


umap_ctrl_IUA_allsingle$type4="N"

umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type2%in%"CTRL"]<-"Ctrl"
umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type2%in%"IUA1"]<-"Adjacy"
umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type2%in%"IUA2"]<-"Adhesion"




#3组间
umap_ctrl_IUA_allsingle$type4=factor(umap_ctrl_IUA_allsingle$type4,levels = c("Ctrl","Adjacy","Adhesion"))

umap_ctrl_IUA_allsingle$cell_type1110_1=paste(umap_ctrl_IUA_allsingle$type2,"_",umap_ctrl_IUA_allsingle$cell_type1110,sep = "")

umap_ctrl_IUA_allsingle$cell_type1110_1=factor(umap_ctrl_IUA_allsingle$cell_type1110_1,levels = c(paste("CTRL_",names(table(umap_ctrl_IUA_allsingle$cell_type1110)),sep = ""),
                                                                                                  paste("IUA1_",names(table(umap_ctrl_IUA_allsingle$cell_type1110)),sep = ""),
                                                                                                  paste("IUA2_",names(table(umap_ctrl_IUA_allsingle$cell_type1110)),sep = "")))

av=AverageExpression(umap_ctrl_IUA_allsingle,
                     group.by = "cell_type1110_1",
                     assays = "RNA")

av=av[[1]]


cg=names(tail(sort(apply(av,1,sd)),1000))


View(av[cg,])

N=cor(av[cg,],method = 'spearman')

N=as.data.frame(N)

N=N[1:4,]
N=N[,-c(1:4)]

N$CTRL=1
N$IUA1=1
N$IUA2=1

N$IUA1[1]=N[1,1]
N$IUA1[2]=N[2,2]
N$IUA1[3]=N[3,3]
N$IUA1[4]=N[4,4]


N$IUA2[1]=N[1,5]
N$IUA2[2]=N[2,6]
N$IUA2[3]=N[3,7]
N$IUA2[4]=N[4,8]


N=N[9:11]
pheatmap(N,cluster_cols = F, display_numbers = TRUE,cluster_rows = F)

colnames(N)[2]="Adjacy"
colnames(N)[3]="Adhesion"

rownames(N)=gsub("CTRL_","",rownames(N))
par(mar=c(3,9,2,3))

pdf(file ="/data/DATAprocess/chentw/zigongneimo/temp/P1.pdf",
    width =7,height = 6)
pheatmap(N,cluster_cols = F, display_numbers = TRUE,cluster_rows = F,
         fontsize = 20)
dev.off()












##对所有细胞类型中的四种主要细胞类型进行相关性分析绘图2----
umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")


umap_ctrl_IUA_allsingle$type5="N"

umap_ctrl_IUA_allsingle$type5[umap_ctrl_IUA_allsingle$type3%in%c("CTRL1","CTRL2")]<-"Ctrl"
umap_ctrl_IUA_allsingle$type5[umap_ctrl_IUA_allsingle$type3%in%c("IUA1","IUA2")]<-"miIUA-a"
umap_ctrl_IUA_allsingle$type5[umap_ctrl_IUA_allsingle$type3%in%c("IUA3","IUA4")]<-"miIUA-I"
umap_ctrl_IUA_allsingle$type5[umap_ctrl_IUA_allsingle$type3%in%c("IUA5","IUA6")]<-"moIUA-a"
umap_ctrl_IUA_allsingle$type5[umap_ctrl_IUA_allsingle$type3%in%c("IUA7")]<-"moIUA-I"





#3组间
umap_ctrl_IUA_allsingle$type5=factor(umap_ctrl_IUA_allsingle$type5,
                                     levels = c("Ctrl","miIUA-a","miIUA-I","moIUA-a","moIUA-I"))

umap_ctrl_IUA_allsingle$cell_type1110_1=paste(umap_ctrl_IUA_allsingle$type5,"_",umap_ctrl_IUA_allsingle$cell_type1110,sep = "")

umap_ctrl_IUA_allsingle$cell_type1110_1=factor(umap_ctrl_IUA_allsingle$cell_type1110_1,
                                               levels = c(paste("Ctrl_",names(table(umap_ctrl_IUA_allsingle$cell_type1110)),sep = ""),
                                                          paste("miIUA-a_",names(table(umap_ctrl_IUA_allsingle$cell_type1110)),sep = ""),
                                                          paste("miIUA-I_",names(table(umap_ctrl_IUA_allsingle$cell_type1110)),sep = ""),
                                                          paste("moIUA-a_",names(table(umap_ctrl_IUA_allsingle$cell_type1110)),sep = ""),
                                                          paste("moIUA-I_",names(table(umap_ctrl_IUA_allsingle$cell_type1110)),sep = "")))

av=AverageExpression(umap_ctrl_IUA_allsingle,
                     group.by = "cell_type1110_1",
                     assays = "RNA")

av=av[[1]]


cg=names(tail(sort(apply(av,1,sd)),1000))


View(av[cg,])

N=cor(av[cg,],method = 'spearman')

N=as.data.frame(N)

N=N[1:4,]
N=N[,-c(1:4)]

N$'Ctrl'=1
N$'miIUA-a'=1
N$'miIUA-I'=1
N$'moIUA-a'=1
N$'moIUA-I'=1

N$'miIUA-a'[1]=N[1,1]
N$'miIUA-a'[2]=N[2,2]
N$'miIUA-a'[3]=N[3,3]
N$'miIUA-a'[4]=N[4,4]

N$'miIUA-I'[1]=N[1,5]
N$'miIUA-I'[2]=N[2,6]
N$'miIUA-I'[3]=N[3,7]
N$'miIUA-I'[4]=N[4,8]

N$'moIUA-a'[1]=N[1,9]
N$'moIUA-a'[2]=N[2,10]
N$'moIUA-a'[3]=N[3,11]
N$'moIUA-a'[4]=N[4,12]

N$'moIUA-I'[1]=N[1,13]
N$'moIUA-I'[2]=N[2,14]
N$'moIUA-I'[3]=N[3,15]
N$'moIUA-I'[4]=N[4,16]




N=N[17:21]
pheatmap(N,cluster_cols = F, display_numbers = TRUE,cluster_rows = F)


rownames(N)=gsub("Ctrl_","",rownames(N))
par(mar=c(3,9,2,3))

pdf(file ="/data/DATAprocess/chentw/zigongneimo/temp/P1.pdf",
    width=8,height = 9)
pheatmap(N,cluster_cols = F, display_numbers = TRUE,cluster_rows = F,
         fontsize = 20)
dev.off()



#开始对最早的定义结果进行柱形图统计比例并绘制2
table_plot=as.data.frame(table(umap_ctrl_IUA_allsingle$type4,umap_ctrl_IUA_allsingle$cell_type1110))

pdf(file ="/data/DATAprocess/chentw/zigongneimo/temp/P2.pdf",
    width=12,height =4)
ggplot(data=table_plot,fill=Var2)+aes(x=Var1,y=stat(table_plot$Freq),fill=Var2)+geom_bar(stat="identity",width=0.75, position="fill")+
  theme_bw()+theme(legend.title=element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank())+ylab("Integrative Celltype(%)")+scale_fill_brewer(palette="Paired")+xlab("")+
  scale_y_continuous(expand=c(0,0),limits=c(0,1),breaks = seq(0,1,0.25),labels = c("0","25","50","75","100"))+
  theme(axis.text.x = element_text(size = 12,angle = 45,hjust = 1,vjust = 1,face = "bold",colour = "black"),
        axis.text.y = element_text(size = 12,face = "bold",colour = "black"),legend.text = element_text(size = 12,face = "bold",colour = "black"))+ylab("")+
  coord_flip()

dev.off()

M=as.data.frame(table(umap_ctrl_IUA_allsingle$type3,umap_ctrl_IUA_allsingle$cell_type1110))


M$ratio=10000

M$ratio[M$Var1%in%"CTRL1"]=M$Freq[M$Var1%in%"CTRL1"]/13852
M$ratio[M$Var1%in%"CTRL2"]=M$Freq[M$Var1%in%"CTRL2"]/9084
M$ratio[M$Var1%in%"IUA1"]=M$Freq[M$Var1%in%"IUA1"]/13454
M$ratio[M$Var1%in%"IUA2"]=M$Freq[M$Var1%in%"IUA2"]/18587
M$ratio[M$Var1%in%"IUA3"]=M$Freq[M$Var1%in%"IUA3"]/7087
M$ratio[M$Var1%in%"IUA4"]=M$Freq[M$Var1%in%"IUA4"]/5060
M$ratio[M$Var1%in%"IUA5"]=M$Freq[M$Var1%in%"IUA5"]/21681
M$ratio[M$Var1%in%"IUA6"]=M$Freq[M$Var1%in%"IUA6"]/5667
M$ratio[M$Var1%in%"IUA7"]=M$Freq[M$Var1%in%"IUA7"]/2037


write.csv(M,"/data/DATAprocess/chentw/zigongneimo/temp/ratio.csv")


umap_ctrl_IUA_allsingle$type4='Ctrl'

umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type3%in%c("IUA1","IUA2",
                                                                 "IUA5","IUA6")]<-"mi-IUA"

umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type3%in%c("IUA3",
                                                                 "IUA4","IUA7")]<-"mo-IUA"



M=as.data.frame(table(umap_ctrl_IUA_allsingle$type4,
                      umap_ctrl_IUA_allsingle$cell_type1110))


M$ratio=10000

M$ratio[M$Var1%in%"Ctrl"]=M$Freq[M$Var1%in%"Ctrl"]/22936
M$ratio[M$Var1%in%"mi-IUA"]=M$Freq[M$Var1%in%"mi-IUA"]/59389
M$ratio[M$Var1%in%"mo-IUA"]=M$Freq[M$Var1%in%"mo-IUA"]/14184

write.csv(M,"/data/DATAprocess/chentw/zigongneimo/temp/ratio2.csv")


##绘制几个基因的小提琴图
gene=c("CD14","CD3G","PTPRC","PECAM1","VWF","PDGFRA","COL1A1","EPCAM")
VlnPlot(umap_ctrl_IUA_allsingle,features = gene,group.by = "cell_type1110")


#2-1
P1=VlnPlot(umap_ctrl_IUA_allsingle,features = gene[1],pt.size = 0,ncol = 1,group.by = "cell_type1110")+
  scale_fill_manual(values = c("#DB0C15","#4CA636","#9E4894","#EE7A00","#387DB9","#482013","#F9E800","#5555A4"))+
  theme(axis.text = element_blank(),legend.position = "none")+xlab("")+ylab("")+
  theme(title = element_text(size =13),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.text = element_blank(),legend.position = "none",
        plot.subtitle = element_text(size = 3))+theme(text = element_text(size=0))





















Macr=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/Macr_allsingle.rds")


table(Macr$cell_type0610)

VlnPlot(Macr,features = "CD14",group.by = "cell_type0610",pt.size = 0)







#对之前的基质细胞差异基因进行upsetR图绘制----
EnSc=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0718.rds")

###上调
S0_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/S0/S0_deseq.csv")
Cluster0=subset(S0_deseq,S0_deseq$avg_log2FC>0)$X

S1_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/S1/S1_deseq.csv")
Cluster2=subset(S1_deseq,S1_deseq$avg_log2FC>0)$X

REV3L_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/REV3L/REV3L_deseq.csv")
Cluster1=subset(REV3L_deseq,REV3L_deseq$avg_log2FC>0)$X

BMP7_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/BMP7/BMP7_deseq.csv")
Cluster4=subset(BMP7_deseq,BMP7_deseq$avg_log2FC>0)$X


ISG15_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/ISG15/ISG15_deseq.csv")
Cluster6=subset(ISG15_deseq,ISG15_deseq$avg_log2FC>0)$X


P_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/P/P_deseq.csv")
Cluster5=subset(P_deseq,P_deseq$avg_log2FC>0)$X


Cycling_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/Cycling/Cycling_deseq.csv")
Cluster3=subset(Cycling_deseq,Cycling_deseq$avg_log2FC>0)$X


set_list <- list("Cluster0"=Cluster0,"Cluster1"=Cluster1,"Cluster2"=Cluster2,"Cluster3"=Cluster3,
                 "Cluster4"=Cluster4,"Cluster5"=Cluster5,"Cluster6"=Cluster6)


pdf("/data/DATAprocess/chentw/zigongneimo/temp/20240729/upset1.pdf",
    width = 10,height = 7)
upset(
  fromList(set_list),
  order.by = "freq",nsets = 7,shade.color="red",
  sets.bar.color=c("#EE7A00","#DB0C15","#4CA636",
                   "#387DB9","#9E4894","#5555A4",
                   "#482013"),
  sets.x.label="UP genes",mainbar.y.label="No. of up genes per cell type"
)

dev.off()

G=Reduce(intersect,list(Cluster0,Cluster1,Cluster2,Cluster3,Cluster4,Cluster5,Cluster6))





G <- bitr(G, fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
G=G$ENTREZID
ego <- enrichGO(OrgDb="org.Hs.eg.db", gene = G, ont = "ALL",readable= TRUE)

ego=ego@result



#对之前的monocle结果进行统计-----
library(monocle)
STR.cds=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/文件存放点/STR_EPI0513_1.rds")

STR.cds$cell_type0513=gsub("APOD+","Cluster7",STR.cds$cell_type0513)
STR.cds$cell_type0513=gsub("BMP7+/CD34+","Cluster4",STR.cds$cell_type0513)
STR.cds$cell_type0513=gsub("Cycling","Cluster3",STR.cds$cell_type0513)
STR.cds$cell_type0513=gsub("ISG15+","Cluster6",STR.cds$cell_type0513)
STR.cds$cell_type0513=gsub("Myofibroblast/Pericyte","Cluster5",STR.cds$cell_type0513)
STR.cds$cell_type0513=gsub("Normal","Cluster0",STR.cds$cell_type0513)
STR.cds$cell_type0513=gsub("OGN+","Cluster2",STR.cds$cell_type0513)
STR.cds$cell_type0513=gsub("PGR+","Cluster1",STR.cds$cell_type0513)

STR.cds$cell_type0513[STR.cds$cell_type0513%in%"BMP7+/CD34+"]="Cluster4"

STR_P1=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "cell_type0513",label=T)+ggtitle("")

STR_P2=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "Pseudotime")

STR_P1+STR_P2

plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "cell_type0513",label=T)+ggtitle("")+facet_wrap(~cell_type0513, nrow = 4)



#将SFPR4上皮细胞和其他所有基质细胞亚群进行monocle2分析
STR.cds=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/文件存放点/STR_EPI0513_2.rds")

STR.cds$cell_type0513=gsub("APOD+","Cluster7",STR.cds$cell_type0513)
STR.cds$cell_type0513=gsub("BMP7+/CD34+","Cluster4",STR.cds$cell_type0513)
STR.cds$cell_type0513=gsub("Cycling","Cluster3",STR.cds$cell_type0513)
STR.cds$cell_type0513=gsub("ISG15+","Cluster6",STR.cds$cell_type0513)
STR.cds$cell_type0513=gsub("Myofibroblast/Pericyte","Cluster5",STR.cds$cell_type0513)
STR.cds$cell_type0513=gsub("Normal","Cluster0",STR.cds$cell_type0513)
STR.cds$cell_type0513=gsub("OGN+","Cluster2",STR.cds$cell_type0513)
STR.cds$cell_type0513=gsub("PGR+","Cluster1",STR.cds$cell_type0513)

STR.cds$cell_type0513[STR.cds$cell_type0513%in%"BMP7+/CD34+"]="Cluster4"

STR_P1=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "cell_type0513",label=T)+ggtitle("")

STR_P2=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "Pseudotime")

STR_P1+STR_P2

plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "cell_type0513",
                     label=T)+ggtitle("")+facet_wrap(~cell_type0513)




#将SFPR4上皮细胞和其他所有基质细胞亚群进行monocle2分析
STR.cds=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/文件存放点/STR_0617.rds")

STR.cds$cell_type0103=gsub("APOD+","Cluster7",STR.cds$cell_type0103)
STR.cds$cell_type0103=gsub("BMP7+/CD34+","Cluster4",STR.cds$cell_type0103)
STR.cds$cell_type0103=gsub("Cycling","Cluster3",STR.cds$cell_type0103)
STR.cds$cell_type0103=gsub("ISG15+","Cluster6",STR.cds$cell_type0103)
STR.cds$cell_type0103=gsub("Myofibroblast/Pericyte","Cluster5",STR.cds$cell_type0103)
STR.cds$cell_type0103=gsub("Normal","Cluster0",STR.cds$cell_type0103)
STR.cds$cell_type0103=gsub("OGN+","Cluster2",STR.cds$cell_type0103)
STR.cds$cell_type0103=gsub("PGR+","Cluster1",STR.cds$cell_type0103)

STR.cds$cell_type0103[STR.cds$cell_type0103%in%"BMP7+/CD34+"]="Cluster4"

STR_P1=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "cell_type0103",label=T)+ggtitle("")

STR_P2=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "Pseudotime")

STR_P1+STR_P2

plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "cell_type0103",
                     label=T)+ggtitle("")+facet_wrap(~cell_type0103)



#将SFPR4上皮细胞和其他所有基质细胞亚群进行monocle2分析
STR.cds=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/文件存放点/STR_0621.rds")

STR.cds$cell_type0103=gsub("APOD+","Cluster7",STR.cds$cell_type0103)
STR.cds$cell_type0103=gsub("BMP7+/CD34+","Cluster4",STR.cds$cell_type0103)
STR.cds$cell_type0103=gsub("Cycling","Cluster3",STR.cds$cell_type0103)
STR.cds$cell_type0103=gsub("ISG15+","Cluster6",STR.cds$cell_type0103)
STR.cds$cell_type0103=gsub("Myofibroblast/Pericyte","Cluster5",STR.cds$cell_type0103)
STR.cds$cell_type0103=gsub("Normal","Cluster0",STR.cds$cell_type0103)
STR.cds$cell_type0103=gsub("OGN+","Cluster2",STR.cds$cell_type0103)
STR.cds$cell_type0103=gsub("PGR+","Cluster1",STR.cds$cell_type0103)

STR.cds$cell_type0103[STR.cds$cell_type0103%in%"BMP7+/CD34+"]="Cluster4"

STR_P1=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "cell_type0103",label=T)+ggtitle("")

STR_P2=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "Pseudotime")

STR_P1+STR_P2

plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "cell_type0103",
                     label=T)+ggtitle("")+facet_wrap(~cell_type0103)




#对STR进行velovyto.py-ALL----
###前期文件准备
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0718.rds")



setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/velocyto/20240730/")

STR$typenew=STR$cell_type0718


DimPlot(STR,group.by = "typenew",label = T,label.size = 7)+ggtitle("")


cellID_humanpost10x=Cells(STR)
cellID_humanpost10x=as.data.frame(cellID_humanpost10x)
colnames(cellID_humanpost10x)[1]<-"x"

cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"]=paste("H:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"]=paste("qian:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"]=paste("IUAshou:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"]=paste("IUAxia:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"]=paste("li:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"]=paste("IUAfeng:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"]=paste("IUA-GWY_additional:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"]=paste("IUA-BYQ:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"]=paste("IUA-WJM:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"],sep = "")
cellID_humanpost10x$x=gsub("-1_1","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_2","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_3","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_4","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_5","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_6","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_7","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_8","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_9","x",cellID_humanpost10x$x)

write.csv(cellID_humanpost10x, file = "cellID_humanpost10x.csv", row.names = FALSE)
#And finally we can extract our clusters with:
celltype=STR$typenew
celltype=as.data.frame(celltype)
rownames(celltype)=cellID_humanpost10x$x
write.csv(celltype, file = "celltype.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(STR, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")



rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"]=paste(rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"],"-0",sep = "")
rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"]=paste(rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"],"-1",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"],"-2",sep = "")
rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"]=paste(rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"],"-3",sep = "")
rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"]=paste(rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"],"-4",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"],"-5",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"],"-6",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"],"-7",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"],"-8",sep = "")
write.csv(celltype, file = "celltype2.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(STR, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")

DimPlot(STR,label = T,group.by = "typenew")+ggtitle("")


#对基质细胞的cluster0-cluster6进行monocle2分析----
EnSc=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0718.rds")

###先做非特异基因的monocle2----

DimPlot(EnSc,label = T)

DefaultAssay(EnSc)<-"RNA"

library(monocle)

EnSc_matrix<-as(as.matrix(GetAssayData(EnSc,slot = "counts")),'sparseMatrix')
EnSc_ann<-data.frame(gene_id=rownames(EnSc_matrix),gene_short_name=rownames(EnSc_matrix))
rownames(EnSc_ann)<-rownames(EnSc_matrix)

EnSc_fd<-new("AnnotatedDataFrame", data = EnSc_ann)

sample_ann<-EnSc@meta.data
rownames(sample_ann)<-colnames(EnSc_matrix)

EnSc_pd<-new("AnnotatedDataFrame", data =sample_ann)

EnSc.cds<-newCellDataSet(EnSc_matrix,phenoData =EnSc_pd,featureData =EnSc_fd,expressionFamily=negbinomial.size())


#预处理
EnSc.cds <- estimateSizeFactors(EnSc.cds)
EnSc.cds <- estimateDispersions(EnSc.cds)

disp_table <- dispersionTable(EnSc.cds)
unsup_clustering_genes <- subset(disp_table, mean_expression >= 0.1)
EnSc.cds <- setOrderingFilter(EnSc.cds, unsup_clustering_genes$gene_id)

EnSc.cds <- reduceDimension(
  EnSc.cds,
  max_components = 2,
  method = 'DDRTree')

EnSc.cds <- orderCells(EnSc.cds)
head(pData(EnSc.cds))


EnSc_P1=plot_cell_trajectory(EnSc.cds,cell_size = 1,color_by = "integrated_snn_res.0.35",label=T)+ggtitle("EnSc")

EnSc_P2=plot_cell_trajectory(EnSc.cds,cell_size = 1,color_by = "Pseudotime")

EnSc_P3=plot_cell_trajectory(EnSc.cds,cell_size = 1,color_by = "type3",label=T)+ggtitle("EnSc")



EnSc_P1+EnSc_P2

EnSc_P3











EnSc_P2=plot_cell_trajectory(EnSc.cds,cell_size = 1,color_by = "Pseudotime")


plot_cell_trajectory(EnSc.cds,cell_size = 3,color_by = "cell_type0103",label=T)+ggtitle("")+
  facet_wrap(~cell_type0103, nrow = 1)




saveRDS(EnSc.cds,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/文件存放点/STR_0731.rds")

EnSc.cds=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/文件存放点/STR_0731.rds")


###对特异基因进行monocle2分析----
library(monocle)

library(Seurat)
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0718.rds")

DefaultAssay(STR)<-"RNA"

STR_matrix<-as(as.matrix(STR@assays$RNA@counts),'sparseMatrix')

STR_ann<-data.frame(gene_id=rownames(STR_matrix),gene_short_name=rownames(STR_matrix))

rownames(STR_ann)<-rownames(STR_matrix)

STR_fd<-new("AnnotatedDataFrame", data = STR_ann)


sample_ann<-STR@meta.data
rownames(sample_ann)<-colnames(STR_matrix)

STR_pd<-new("AnnotatedDataFrame", data =sample_ann)

STR.cds<-newCellDataSet(STR_matrix,phenoData =STR_pd,
                        featureData =STR_fd,expressionFamily=negbinomial.size())


#预处理
STR.cds <- estimateSizeFactors(STR.cds)
STR.cds <- estimateDispersions(STR.cds)

STR.cds <- detectGenes(STR.cds, min_expr = 0.1)

disp_table <- dispersionTable(STR.cds)

unsup_clustering_genes <- subset(disp_table, mean_expression >= 0.1)

STR.cds <- setOrderingFilter(STR.cds, unsup_clustering_genes$gene_id)


STR.cds<- reduceDimension(STR.cds, max_components = 2, num_dim = 8, 
                          reduction_method = 'tSNE', verbose = T)

STR.cds<- clusterCells(STR.cds, num_clusters = 8)


expressed_genes <- row.names(subset(fData(STR.cds),
                                    num_cells_expressed >= 3))

clustering_DEG_genes <- differentialGeneTest(STR.cds[expressed_genes,],
                                             fullModelFormulaStr = "~ cell_type0718",cores = 8)

ordering_genes <- row.names(clustering_DEG_genes)[order(clustering_DEG_genes$qval)][1:800] #排序后取前800个基因

STR.cds<- setOrderingFilter(STR.cds, ordering_genes)

#开始绘图用DDrtree 方法进行降维、画图
STR.cds <- reduceDimension(
  STR.cds,
  max_components = 2,
  method = 'DDRTree')

STR.cds <- orderCells(STR.cds)
head(pData(STR.cds))


STR_P1=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "cell_type0718",label=T)+ggtitle("")

STR_P2=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "Pseudotime")

STR_P3=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "cell_type0718",label=T)+ggtitle("")+
  facet_wrap(~type0227, nrow = 1)

STR_P3
STR_P1+STR_P2

STR.cds$cell_type0305

STR_P1=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "cell_type0718",label=T)+ggtitle("")

STR_P3=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "cell_type0718",label=T)+ggtitle("")+
  facet_wrap(~cell_type0718, nrow = 2)
STR_P3



STR_P1+STR_P2

saveRDS(STR.cds,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/文件存放点/STR_0731_2.rds")


###将基质细胞的cluser0，3，4，取出来并进行monocle2分析----

library(monocle)

library(Seurat)
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0718.rds")


STR=subset(STR,cell_type0718%in%c("Cluster0","Cluster3","Cluster4"))


STR$cell_type0718=factor(STR$cell_type0718)

DefaultAssay(STR)<-"RNA"

STR_matrix<-as(as.matrix(STR@assays$RNA@counts),'sparseMatrix')

STR_ann<-data.frame(gene_id=rownames(STR_matrix),gene_short_name=rownames(STR_matrix))

rownames(STR_ann)<-rownames(STR_matrix)

STR_fd<-new("AnnotatedDataFrame", data = STR_ann)


sample_ann<-STR@meta.data
rownames(sample_ann)<-colnames(STR_matrix)

STR_pd<-new("AnnotatedDataFrame", data =sample_ann)

STR.cds<-newCellDataSet(STR_matrix,phenoData =STR_pd,
                        featureData =STR_fd,expressionFamily=negbinomial.size())


#预处理
STR.cds <- estimateSizeFactors(STR.cds)
STR.cds <- estimateDispersions(STR.cds)

STR.cds <- detectGenes(STR.cds, min_expr = 0.1)

disp_table <- dispersionTable(STR.cds)

unsup_clustering_genes <- subset(disp_table, mean_expression >= 0.1)

STR.cds <- setOrderingFilter(STR.cds, unsup_clustering_genes$gene_id)


STR.cds<- reduceDimension(STR.cds, max_components = 2, num_dim = 8, 
                          reduction_method = 'tSNE', verbose = T)

STR.cds<- clusterCells(STR.cds, num_clusters = 8)


expressed_genes <- row.names(subset(fData(STR.cds),
                                    num_cells_expressed >= 3))

clustering_DEG_genes <- differentialGeneTest(STR.cds[expressed_genes,],
                                             fullModelFormulaStr = "~ cell_type0718",cores = 8)

ordering_genes <- row.names(clustering_DEG_genes)[order(clustering_DEG_genes$qval)][1:800] #排序后取前800个基因

STR.cds<- setOrderingFilter(STR.cds, ordering_genes)

#开始绘图用DDrtree 方法进行降维、画图
STR.cds <- reduceDimension(
  STR.cds,
  max_components = 2,
  method = 'DDRTree')

STR.cds <- orderCells(STR.cds)
head(pData(STR.cds))


STR_P1=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "cell_type0718",label=T)+ggtitle("")

STR_P2=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "Pseudotime")

STR_P3=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "cell_type0718",label=T)+ggtitle("")+
  facet_wrap(~type0227, nrow = 1)

STR_P3
STR_P1+STR_P2

STR.cds$cell_type0305

STR_P1=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "cell_type0718",label=T)+ggtitle("")

STR_P3=plot_cell_trajectory(STR.cds,cell_size = 1,color_by = "cell_type0718",label=T)+ggtitle("")+
  facet_wrap(~cell_type0718, nrow = 2)
STR_P3



STR_P1+STR_P2

saveRDS(STR.cds,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/文件存放点/STR_0731_3.rds")







#对巨噬细胞进行转录因子分析----
library(dorothea)
library(Seurat)
library(dplyr)
library(tidyr)
library(tibble)
library(pheatmap)


Macr=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/Macr_allsingle.rds")


DimPlot(Macr,group.by = "cell_type0610",label = T,label.size = 12)

VlnPlot(Macr,group.by = "cell_type0610",features = "CD163",pt.size = 0)



dorothea_regulon_human<- get(data("dorothea_hs",package = "dorothea"))

regulon <- dorothea_regulon_human %>%    dplyr::filter(confidence %in% c("A","B","C"))
Macr <- run_viper(Macr,regulon,                  
                  options = list(method = "scale",minsize = 4,                                 
                                 eset.filter = FALSE,cores = 1,                                 
                                 verbose = FALSE))


DefaultAssay(object = Macr) <- "dorothea"
Macr <- ScaleData(Macr)
Macr <- RunPCA(Macr,features = rownames(Macr),verbose = FALSE)
Macr <- FindNeighbors(Macr,dims = 1:10,verbose = FALSE)
Macr <- FindClusters(Macr,resolution = 0.5,verbose = FALSE)


Macr@active.ident=as.factor(Macr$cell_type0610)

viper_scores_df <- GetAssayData(Macr,slot = "scale.data",                                
                                assay = "dorothea")%>%  data.frame(check.names = F)%>%  t()

CellsClusters <- data.frame(cell = names(Idents(Macr)),                           
                            cell_type = as.character(Idents(Macr)),                           
                            check.names = F)


#我们创建了一个数据框架，其中包含每个单元格的viper分数及其聚类
viper_scores_clusters <-viper_scores_df %>%  data.frame() %>%  
  rownames_to_column("cell") %>%  
  gather(tf,activity,-cell) %>%  inner_join(CellsClusters)


#我们按细胞群总结了viper得分
summarized_viper_scores <- viper_scores_clusters %>%  
  group_by(tf,cell_type) %>%  summarise(avg = mean(activity),
                                        std=sd(activity))


#选择20个变化最大的TF,(20*9，populations=180)
highly_variable_tfs <-summarized_viper_scores %>%  
  group_by(tf) %>%  mutate(var = var(avg)) %>%  ungroup() %>%  
  top_n(180,var) %>%  distinct(tf)


summarized_viper_scores_df <- summarized_viper_scores %>%  
  semi_join(highly_variable_tfs,by = "tf") %>%  
  dplyr::select(-std) %>%  spread(tf,avg) %>%  
  data.frame(row.names = 1,check.names = FALSE)


palette_length = 100
my_color = colorRampPalette(c("Darkblue","white","red"))(palette_length)

my_breaks <- c(seq(min(summarized_viper_scores_df),0,                   
                   length.out=ceiling(palette_length/2)+1),               
               seq(max(summarized_viper_scores_df)/palette_length,
                   max(summarized_viper_scores_df),
                   length.out=floor(palette_length/2)))


viper_hmap <- pheatmap(t(summarized_viper_scores_df),fontsize = 14,                       
                       fontsize_row = 10,                       
                       color=my_color,breaks = my_breaks,                       
                       main = "DoRothEA(ABC)",angle_col = 45,                       
                       treeheight_col = 0,border_color = NA)



#对巨噬细胞的M1和M2基因集进行评分----
Macr=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/Macr_allsingle.rds")

score=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/巨噬细胞评分.csv")


M1=score$M1_marker
M2=score$M2_marker

Macr=AddModuleScore(object=Macr,features=list(M1),name ="M1")

Macr=AddModuleScore(object=Macr,features=list(M2),name ="M2")

colnames(Macr@meta.data)

Macr$type4=factor(Macr$type4,levels = c("Ctrl","Adjacent","Adhesion"))

VlnPlot(Macr,features = "M11",group.by = "cell_type0610",pt.size = 0)+ggtitle("M1 Score")+xlab("")+
  scale_x_discrete(limits=c("CD24+","Proliferating","Uknow","CD301+","CD163+","Monocyte+"))

VlnPlot(Macr,features = "M11",group.by = "cell_type0610",pt.size = 0,split.by = "type4")+ggtitle("M1 Score")+xlab("")+
  scale_x_discrete(limits=c("CD24+","Proliferating","Uknow","CD301+","CD163+","Monocyte+"))

VlnPlot(Macr,features = "M21",group.by = "cell_type0610",
        split.by = "type4",pt.size = 0)+ggtitle("M2 Score")+xlab("")+
  scale_x_discrete(limits=c("Uknow","CD301+","CD24+","Proliferating","Monocyte+","CD163+"))

VlnPlot(Macr,features = "M21",group.by = "cell_type0610",pt.size = 0)+ggtitle("M2 Score")+xlab("")+
  scale_x_discrete(limits=c("Uknow","CD301+","CD24+","Proliferating","Monocyte+","CD163+"))


FeaturePlot(Macr,features = c("CD68","ACTA2"),
            min.cutoff = 0,order = T)


DimPlot(Macr,group.by = "cell_type0610",
        label = T,label.size = 9)

#对巨噬细胞表达CD68和ACTA2的细胞群进行标注
data=as.data.frame(t(as.data.frame(Macr@assays$RNA@data)))

Macr@meta.data$CD68=data$CD68
Macr@meta.data$ACTA2=data$ACTA2
Macr@meta.data$TAGLN=data$TAGLN
Macr@meta.data$AIFM2=data$AIFM2

Macr$type5="Other"
Macr$type5[Macr$CD68>0 & Macr$ACTA2>0]="T"

DimPlot(Macr,group.by = "type5",label = F,label.size = 8,order = T,
        pt.size = 0.6)+
  scale_color_manual(values=c('grey',"red"))+ggtitle("ACTA2/CD68")+
  theme(legend.position = "none")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/原图绘制/20240830/P1.pdf",
       width = 6,height = 5)

Macr$type5="Other"
Macr$type5[Macr$CD68>0 & Macr$AIFM2>0]="T"

DimPlot(Macr,group.by = "type5",label = F,label.size = 8,order = T,
        pt.size = 0.6)+
  scale_color_manual(values=c('grey',"red"))+ggtitle("AAIFM2/CD68")+
  theme(legend.position = "none")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/原图绘制/20240830/P2.pdf",
       width = 6,height = 5)




Macr$type5="Other"
Macr$type5[Macr$CD68>0 & Macr$TAGLN>0]="T"

DimPlot(Macr,group.by = "type5",label = F,label.size = 8,order = T,
        pt.size = 0.6)+
  scale_color_manual(values=c('grey',"red"))+ggtitle("TAGLN/CD68")+
  theme(legend.position = "none")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/原图绘制/20240830/P3.pdf",
       width = 6,height = 5)






#对肌成纤维细胞进行亚群整合----
EnSc=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0718.rds")

DimPlot(EnSc,group.by = "cell_type0305",label = T,label.size = 6)


Myfibroblast=subset(EnSc,cell_type0305%in%c("P"))

Myfibroblast <- CreateSeuratObject(counts = Myfibroblast@assays$RNA@counts, project = "Myfibroblast", min.cells = 3)
Myfibroblast[["percent.mt"]] <- PercentageFeatureSet(Myfibroblast, pattern = "^MT-")
VlnPlot(Myfibroblast, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

Myfibroblast <- subset(Myfibroblast, subset = percent.mt < 20 & nFeature_RNA > 500&nFeature_RNA < 7500)
Myfibroblast <- NormalizeData(Myfibroblast, normalization.method = "LogNormalize", 
                              scale.factor = 10000)
Myfibroblast <- FindVariableFeatures(Myfibroblast, selection.method = "vst", nfeatures = 2000)
VlnPlot(Myfibroblast, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

#Myfibroblast进行标准化然后PCA分析
DefaultAssay(Myfibroblast) <- "RNA"
all.genes_Myfibroblast <- rownames(Myfibroblast)
Myfibroblast <- ScaleData(object = Myfibroblast, features = all.genes_Myfibroblast)
Myfibroblast <- RunPCA(Myfibroblast, features = VariableFeatures(object = Myfibroblast))

#确定Myfibroblast的PC维数
Myfibroblast <- JackStraw(Myfibroblast, num.replicate = 100) 
Myfibroblast <- ScoreJackStraw(Myfibroblast, dims = 1:20) 
ElbowPlot(Myfibroblast,ndims = 30, reduction = "pca") 

#Myfibroblast进行细胞细胞聚类分析(UMAP)
Myfibroblast=RunUMAP(Myfibroblast, dims = 1:30)
DimPlot(Myfibroblast,reduction="umap",label=TRUE)
Myfibroblast <- FindNeighbors(Myfibroblast, dims = 1:30)
Myfibroblast <- FindClusters(Myfibroblast, resolution = 0.3)
Myfibroblast <- FindClusters(Myfibroblast, resolution = 0.6)
DimPlot(Myfibroblast,reduction="umap",label=TRUE)


#Myfibroblast进行细胞细胞聚类分析(tsne)
Myfibroblast=RunTSNE(Myfibroblast, dims = 1:30)
DimPlot(Myfibroblast,reduction="tsne",label=TRUE)
Myfibroblast <- FindNeighbors(Myfibroblast, dims = 1:30)
Myfibroblast <- FindClusters(Myfibroblast, resolution = 0.3)
Myfibroblast <- FindClusters(Myfibroblast, resolution = 0.6)
DimPlot(Myfibroblast,reduction="tsne",label=TRUE)



DimPlot(Myfibroblast,reduction="umap",label=TRUE)

FeaturePlot(Myfibroblast,features = c("ACTA2","COL1A1"))



#进行cellchat分析绘图-----
Macr=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/Macr_allsingle.rds")
EPI=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/rds文件存放点/EPI_harmony_1218_2.rds")
Immune=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/IMMUNE/rds存放点/Immune2.rds")
NK=subset(Immune,cell_type2%in%"NK")
EnDo=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/rds文件存放点/EnDo_harmony_1218_1.rds")
umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")

EnSc2=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

T=subset(Immune,cell_type2%in%c("Cycling T","T cells"))





all=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/cellchat/20240625/all.rds")
Ctrl=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/cellchat/20240625/Ctrl.rds")
Adjacent=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/cellchat/20240625/Adjacent.rds")
Adhesion=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/cellchat/20240625/Adhesion.rds")


object.list <- list(Ctrl = Ctrl, Adjacent = Adjacent , Adhesion = Adhesion)

cellchat <- mergeCellChat(object.list, add.names = names(object.list), cell.prefix = TRUE)


gg1 <- rankNet(cellchat, mode = "comparison", stacked = T, do.stat = TRUE,comparison = c(1,2,3))
gg2 <- rankNet(cellchat, mode = "comparison", stacked = F, do.stat = TRUE,comparison = c(1,2,3))
gg1 + gg2



netVisual_bubble(cellchat, 
                 comparison= c(1, 2), 
                 angle.x = 45)

p1<- netVisual_bubble(cellchat, 
                      comparison= c(1, 2), 
                      max.dataset = 2, 
                      title.name = "Increased signaling in adhesion", 
                      angle.x = 45, 
                      remove.isolate = T) #Increased为比较组通讯概率更强的配受体对信息

p2<- netVisual_bubble(cellchat, 
                      comparison= c(1, 2), 
                      max.dataset = 1, 
                      title.name = "Decreased signaling in adhesion", 
                      angle.x = 45, 
                      remove.isolate = T) #Decreased为对照组通讯概率更强的配受体对信息
p1+ p2



p7<- netVisual_bubble(cellchat, 
                      comparison= c(1, 3), 
                      max.dataset = 3, 
                      title.name = "Increased signaling in adhesion", 
                      angle.x = 45, 
                      remove.isolate = T) #Increased为比较组通讯概率更强的配受体对信息

p8<- netVisual_bubble(cellchat, 
                      comparison= c(1, 3), 
                      max.dataset = 1, 
                      title.name = "Decreased signaling in adhesion", 
                      angle.x = 45, 
                      remove.isolate = T) #Decreased为对照组通讯概率更强的配受体对信息
p7+ p8



#挑选一些受体配体对放表里面进行绘图
EPI$cell_type3="EnEc"
T$cell_type3="T"
NK$cell_type3="NK"
EnDo$cell_type3="EnDo"
Macr$cell_type3="Macr"
EnSc2$cell_type3="EnSc"

EnSc=EnSc2

T$type4="Adhesion"
T$type4[T$type2%in%"CTRL"]="Ctrl"
T$type4[T$type2%in%"IUA1"]="Adjacent"

NK$type4="Adhesion"
NK$type4[NK$type2%in%"CTRL"]="Ctrl"
NK$type4[NK$type2%in%"IUA1"]="Adjacent"

EnDo$type4="Adhesion"
EnDo$type4[EnDo$type2%in%"CTRL"]="Ctrl"
EnDo$type4[EnDo$type2%in%"IUA1"]="Adjacent"


all=merge(EPI,y = c(EnSc2,T,NK,EnDo,Macr))

all <- NormalizeData(all, normalization.method = "LogNormalize", 
                     scale.factor = 10000)

saveRDS(all,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/all_umap.rds")

RC=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/cellchat/20240625/RC.csv")

#先统计细胞平均表达量
Ctrl=subset(all,type4%in%"Ctrl")
Adjacent=subset(all,type4%in%"Adjacent")
Adhesion=subset(all,type4%in%"Adhesion")


EPI_Ctrl=subset(EPI,type4%in%"Ctrl")
T_Ctrl=subset(T,type4%in%"Ctrl")
NK_Ctrl=subset(NK,type4%in%"Ctrl")
EnDo_Ctrl=subset(EnDo,type4%in%"Ctrl")
Macr_Ctrl=subset(Macr,type4%in%"Ctrl")
EnSc_Ctrl=subset(EnSc,type4%in%"Ctrl")

EPI_Adjacent=subset(EPI,type4%in%"Adjacent")
T_Adjacent=subset(T,type4%in%"Adjacent")
NK_Adjacent=subset(NK,type4%in%"Adjacent")
EnDo_Adjacent=subset(EnDo,type4%in%"Adjacent")
Macr_Adjacent=subset(Macr,type4%in%"Adjacent")
EnSc_Adjacent=subset(EnSc,type4%in%"Adjacent")

EPI_Adhesion=subset(EPI,type4%in%"Adhesion")
T_Adhesion=subset(T,type4%in%"Adhesion")
NK_Adhesion=subset(NK,type4%in%"Adhesion")
EnDo_Adhesion=subset(EnDo,type4%in%"Adhesion")
Macr_Adhesion=subset(Macr,type4%in%"Adhesion")
EnSc_Adhesion=subset(EnSc,type4%in%"Adhesion")

#绘图【计算的代码不见了先根据之前的数据提取绘图,后面复现代码】
P_point2=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/cellchat/20240625/point4.csv")

P_point3=subset(P_point2,LR2%in%c("TNF-TNFRSF1B","TNF-TNFRSF1A","NAMPT-INSR",
                                  "NAMPT-ITGA5","MIF-CD74","EREG-EGFR","HBEGF-EGFR",
                                  "CCL5-CCR1","CCL3L1-CCR1","AREG-EGFR"))

P_point3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/cellchat/20240625/20241018绘图/P_point3.csv")

P_point3$ligand=gsub("Endo","EnDo",P_point3$ligand)
P_point3$receptor=gsub("Endo","EnDo",P_point3$receptor)


for (X in 1:162) {
  P_point3[X,]$LR=(AverageExpression(subset(all,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$ligand),features = P_point3[X,]$L1,group.by = "cell_type3")$RNA[1,]*
                     nrow(subset(all,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$ligand)@meta.data)+
                     AverageExpression(subset(all,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$receptor),features = P_point3[X,]$R1,group.by = "cell_type3")$RNA[1,]*
                     nrow(subset(all,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$receptor)@meta.data))/
    (nrow(subset(all,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$ligand)@meta.data)+nrow(subset(all,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$receptor)@meta.data))
}


#计算复合体的LR
P_point3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/cellchat/20240625/20241018绘图/P_point3.csv")

for (X in 163:235) {
  P_point3[X,]$LR=(AverageExpression(subset(all,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$ligand),features = P_point3[X,]$L1,group.by = "cell_type3")$RNA[1,]*
                     nrow(subset(all,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$ligand)@meta.data)+
                     AverageExpression(subset(all,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$receptor),features = P_point3[X,]$R1,group.by = "cell_type3")$RNA[1,]*
                     nrow(subset(all,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$receptor)@meta.data)+
                     AverageExpression(subset(all,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$receptor),features = P_point3[X,]$R2,group.by = "cell_type3")$RNA[1,]*
                     nrow(subset(all,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$receptor)@meta.data))/
    (nrow(subset(all,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$ligand)@meta.data)+nrow(subset(all,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$receptor)@meta.data)+
       nrow(subset(all,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$receptor)@meta.data))
}


#计算单独受体配体对的表达比例
gene=c(P_point3$L1,P_point3$R1,P_point3$R2)
gene=gene[!duplicated(gene)]

all$cell_type4=paste(all$type4,"_",all$cell_type3,sep = '')

P=DotPlot(all,features = gene,group.by = "cell_type4")

data2=P$data

data2$type4="M"

data2$type4[grep("Ctrl_",data2$id)]="Ctrl"
data2$type4[grep("Adjacent_",data2$id)]="Adjacent"
data2$type4[grep("Adhesion_",data2$id)]="Adhesion"


data2$cell_type3="M"


data2$cell_type3[grep("Ctrl_",data2$id)]=gsub("Ctrl_","",data2$id[grep("Ctrl_",data2$id)])
data2$cell_type3[grep("Adjacent_",data2$id)]=gsub("Adjacent_","",data2$id[grep("Adjacent_",data2$id)])
data2$cell_type3[grep("Adhesion_",data2$id)]=gsub("Adhesion_","",data2$id[grep("Adhesion_",data2$id)])


for (X in 1:162) {
  P_point3[X,]$proportion=(subset(data2,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$ligand&features.plot%in%P_point3[X,]$L1)$pct.exp*
                             nrow(subset(all,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$ligand)@meta.data)+
                             subset(data2,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$receptor&features.plot%in%P_point3[X,]$R1)$pct.exp*
                             nrow(subset(all,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$receptor)@meta.data))/
    (nrow(subset(all,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$ligand)@meta.data)+nrow(subset(all,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$receptor)@meta.data))
}


#计算复合体的比例
for (X in 163:234) {
  P_point3[X,]$proportion=(subset(data2,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$ligand&features.plot%in%P_point3[X,]$L1)$pct.exp*
                             nrow(subset(all,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$ligand)@meta.data)+
                             subset(data2,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$receptor&features.plot%in%P_point3[X,]$R1)$pct.exp*
                             nrow(subset(all,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$receptor)@meta.data)+
                             subset(data2,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$receptor&features.plot%in%P_point3[X,]$R2)$pct.exp*
                             nrow(subset(all,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$receptor)@meta.data))/
    (nrow(subset(all,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$ligand)@meta.data)+nrow(subset(all,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$ligand)@meta.data)+
       nrow(subset(all,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$receptor)@meta.data)+nrow(subset(all,type4%in%P_point3[X,]$type4&cell_type3%in%P_point3[X,]$receptor)@meta.data))
}



#开始气泡图绘制
P_point3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/cellchat/20240625/20241018绘图/P_point3.csv")

P_point3$type4=factor(P_point3$type4,levels = c("Ctrl","Adjacent","Adhesion"))


P_point3$LR=log2(P_point3$LR+1)

P_point3$type5=gsub("Endo","EnDo",P_point3$type5)

ggplot(P_point3,aes(type5,LR2))+geom_point(aes(size=P_point3$proportion,color=P_point3$LR))+
  scale_color_gradient2(low="#0E4171",mid = "white",high="#BE3840",midpoint = 1.5)+ theme_bw()+
  theme(axis.text.x = element_text(size=9,angle = 45,hjust = 1,vjust = 1,colour = "black",face="bold"),
        axis.text.y = element_text(size=9,colour = "black",face="bold"))+xlab("")+ylab("")+
  labs(size="cell proportion",color="log2(LR+1)")+scale_x_discrete(limits=c(paste("Ctrl","_",names(table(paste(P_point3$ligand,"_",P_point3$receptor,sep=""))),sep = ""),
                                                                            paste("Adjacent","_",names(table(paste(P_point3$ligand,"_",P_point3$receptor,sep=""))),sep = ""),
                                                                            paste("Adhesion","_",names(table(paste(P_point3$ligand,"_",P_point3$receptor,sep=""))),sep = "")))+
  scale_y_discrete(limits=c("CCL3-CCR1","CCL3L1-CCR1","CCL5-CCR1","AREG-EGFR","EREG-EGFR","HBEGF-EGFR",
                            "MIF-(CD74+CD44)","MIF-(CD74+CXCR4)","NAMPT-(ITGA5+ITGB1)","NAMPT-INSR","TNF-TNFRSF1A","TNF-TNFRSF1B"))

























#上面做的是ctrl和adhesion组之间的差异受体配体,继续选取三间22差异的受体配体对进行绘图[纤维化，趋化相关]

p1<- netVisual_bubble(cellchat, 
                      comparison= c(1, 2),sources.use = 4,
                      title.name = "Increased signaling in adjacent", 
                      angle.x = 45, 
                      remove.isolate = T) #Increased为比较组通讯概率更强的配受体对信息

p2<- netVisual_bubble(cellchat, 
                      comparison= c(1, 2), sources.use = 4,
                      title.name = "Decreased signaling in adjacent", 
                      angle.x = 45, 
                      remove.isolate = T) #Decreased为对照组通讯概率更强的配受体对信息
p1+ p2





p1<- netVisual_bubble(cellchat, 
                      comparison= c(1, 3),sources.use = 4,max.dataset = 3, 
                      title.name = "Increased signaling in adhesion", 
                      angle.x = 45, 
                      remove.isolate = T) #Increased为比较组通讯概率更强的配受体对信息

p2<- netVisual_bubble(cellchat, 
                      comparison= c(1, 3), sources.use = 4,
                      title.name = "Decreased signaling in adhesion", 
                      angle.x = 45, max.dataset = 1, 
                      remove.isolate = T) #Decreased为对照组通讯概率更强的配受体对信息
p1+ p2







#进行受体配体对差异热图绘制
LR3=c("TGFB1_TGFBR1","TGFB1_TGFBR2","PDGFB_PDGFRA" )



#对其他细胞和基质细胞互作的差异受体配体对（ECM,抗炎，促炎）----
p1<- netVisual_bubble(cellchat, 
                      comparison= c(1, 3),sources.use = 4,max.dataset = 3, 
                      title.name = "Increased signaling in adhesion", 
                      angle.x = 45, 
                      remove.isolate = T) #Increased为比较组通讯概率更强的配受体对信息

p2<- netVisual_bubble(cellchat, 
                      comparison= c(1, 3), sources.use = 4,
                      title.name = "Decreased signaling in adhesion", 
                      angle.x = 45, max.dataset = 1, 
                      remove.isolate = T) #Decreased为对照组通讯概率更强的配受体对信息
p1+ p2


netVisual_bubble(cellchat, targets.use = 3,
                 comparison= c(1, 3), 
                 angle.x = 45)



LR2=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/cellchat/20240625/20241018绘图/lr22.csv")




#开始进行cellchat，[STR和其他细胞类型]---------
all=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/all_umap.rds")

table(all$cell_type3)

all$cell_type4="Other"

all$cell_type4[all$cell_type3%in%"EnSc"]="EnSc"



Ctrl=subset(all,type4%in%"Ctrl")
Adjacent=subset(all,type4%in%"Adjacent")
Adhesion=subset(all,type4%in%"Adhesion")



###Ctrl
cellchat <- createCellChat(Ctrl@assays$RNA@data,meta = Ctrl@meta.data,group.by = "cell_type4")
cellchat=setIdent(cellchat,ident.use = "cell_type4")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Interaction weights/strength")

saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/20241024/Ctrl.rds")


###Adjacent
cellchat <- createCellChat(Adjacent@assays$RNA@data,meta = Adjacent@meta.data,group.by = "cell_type4")
cellchat=setIdent(cellchat,ident.use = "cell_type4")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Interaction weights/strength")

saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/20241024/Adjacent.rds")



###Adhesion
cellchat <- createCellChat(Adhesion@assays$RNA@data,meta = Adhesion@meta.data,group.by = "cell_type4")
cellchat=setIdent(cellchat,ident.use = "cell_type4")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Interaction weights/strength")

saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/20241024/Adhesion.rds")



###all
cellchat <- createCellChat(all@assays$RNA@data,meta = all@meta.data,group.by = "cell_type4")
cellchat=setIdent(cellchat,ident.use = "cell_type4")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Interaction weights/strength")

saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/20241024/all.rds")




#开始比较并进行绘图
Ctrl=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/20241024/Ctrl.rds")
Adjacent=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/20241024/Adjacent.rds")
Adhesion=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/20241024/Adhesion.rds")


object.list <- list(Ctrl = Ctrl, Adjacent = Adjacent , Adhesion = Adhesion)

cellchat <- mergeCellChat(object.list, add.names = names(object.list), cell.prefix = TRUE)


gg1 <- rankNet(cellchat, mode = "comparison", stacked = T, do.stat = TRUE,comparison = c(1,2,3))
gg2 <- rankNet(cellchat, mode = "comparison", stacked = F, do.stat = TRUE,comparison = c(1,2,3))
gg1 + gg2



netVisual_bubble(cellchat, 
                 comparison= c(1, 2), 
                 angle.x = 45)

p1<- netVisual_bubble(cellchat, 
                      comparison= c(1, 3), targets.use = 2,
                      title.name = "Increased signaling in adhesion", 
                      angle.x = 45, max.dataset = 1,
                      remove.isolate = T) #Increased为比较组通讯概率更强的配受体对信息

p2<- netVisual_bubble(cellchat, 
                      comparison= c(2, 3), targets.use = 2,
                      title.name = "Iecreased signaling in adhesion2", 
                      angle.x = 45, max.dataset = 3,
                      remove.isolate = T) #Decreased为对照组通讯概率更强的配受体对信息
p1+ p2



#绘图【计算的代码不见了先根据之前的数据提取绘图,后面复现代码】
P_point3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/20241024/LR3.csv")

all$cell_type5="K"


for (X in 70:108) {
  P_point3[X,]$LR=(AverageExpression(subset(all,type4%in%P_point3[X,]$type4),features = P_point3[X,]$L1,group.by = "cell_type5")$RNA[1,]*
                     nrow(subset(all,type4%in%P_point3[X,]$type4)@meta.data)+
                     AverageExpression(subset(all,cell_type4%in%"EnSc"&type4%in%P_point3[X,]$type4),features = P_point3[X,]$R1,group.by = "cell_type4")$RNA[1,]*
                     nrow(subset(all,cell_type4%in%"EnSc"&type4%in%P_point3[X,]$type4)@meta.data))/
    (nrow(subset(all,type4%in%P_point3[X,]$type4)@meta.data)+nrow(subset(all,cell_type4%in%"EnSc"&type4%in%P_point3[X,]$type4)@meta.data))
}



for (X in 1:69) {
  P_point3[X,]$LR=(AverageExpression(subset(all,type4%in%P_point3[X,]$type4),features = P_point3[X,]$L1,group.by = "cell_type5")$RNA[1,]*
                     nrow(subset(all,type4%in%P_point3[X,]$type4)@meta.data)+
                     AverageExpression(subset(all,cell_type4%in%"EnSc"&type4%in%P_point3[X,]$type4),features = P_point3[X,]$R1,group.by = "cell_type4")$RNA[1,]*
                     nrow(subset(all,cell_type4%in%"EnSc"&type4%in%P_point3[X,]$type4)@meta.data)+
                     AverageExpression(subset(all,cell_type4%in%"EnSc"&type4%in%P_point3[X,]$type4),features = P_point3[X,]$R2,group.by = "cell_type4")$RNA[1,]*
                     nrow(subset(all,cell_type4%in%"EnSc"&type4%in%P_point3[X,]$type4)@meta.data))/
    (nrow(subset(all,type4%in%P_point3[X,]$type4)@meta.data)+nrow(subset(all,cell_type4%in%"EnSc"&type4%in%P_point3[X,]$type4)@meta.data)+nrow(subset(all,cell_type4%in%"EnSc"&type4%in%P_point3[X,]$type4)@meta.data))
}


write.csv(P_point3,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/20241024/LR4.csv")



#开始热图绘制
LR4=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/20241024/LR7.csv",row.names = 1)


LR4=LR4[1:3]

xx=pheatmap(LR4,scale = "row",cluster_rows = F,cluster_cols = F,color = colorRampPalette(c("navy", "white", "firebrick3"))(50))


save_pheatmap_pdf <- function(x, filename, width=4, height=7) {
  stopifnot(!missing(x))
  stopifnot(!missing(filename))
  pdf(filename, width=width, height=height)
  grid::grid.newpage()
  grid::grid.draw(x$gtable)
  dev.off()
}



save_pheatmap_pdf(xx, "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/cellchat/20240625/20241018绘图/P2.pdf")






##寻找功能相关的差异转录因子并进行热图绘制----
EnSc=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0718.rds")

EnSc_deseq=FindMarkers(EnSc,ident.1 = "Adhesion",ident.2 = "Ctrl",
                       min.pct = 0.2,group.by = "type4")

EnSc_deseq=
  subset(EnSc_deseq,EnSc_deseq$p_val<0.01&EnSc_deseq$avg_log2FC>0)

write.csv(EnSc_deseq,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/富集分析/EnSc_deseq.csv")

scenic=read.csv("/data/DATAprocess/chentw/temp/2022/scenic.csv",row.names = 1)

gene1=rownames(EnSc_deseq)
gene2=scenic$x


gene3=intersect(gene1,gene2)






G <- bitr(gene3, fromType = "SYMBOL", toType = c("ENTREZID"), OrgDb = org.Hs.eg.db)
G=G$ENTREZID
ego <- enrichGO(OrgDb="org.Hs.eg.db", gene = G, ont = "ALL",readable= TRUE)

ego=ego@result

write.csv(ego,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/富集分析/ego.csv")



#对基质细胞进行SCPA分析-----
path1=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/SCPA分析/PATH12_2.csv",header = F,row.names = 1)

path1=t(path1)

path1=as.data.frame(path1)

EnSc=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0718.rds")



library(KEGGREST)

###选取一些通路内的基因
org <- keggList("organism")
head(org)
gs <- keggGet('hsa00061') 

gs[[1]]$GENE
genes <- unlist(lapply(gs[[1]]$GENE,function(x) strsplit(x,';')[[1]][1]))
gene_list<-genes[1:length(genes)%%2 ==0]
gene_list
write.csv(file="/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/SCPA分析/Fatty_acid_biosynthesis.csv",gene_list)





#开始统计SCPA中的几组基质细胞亚群的差异基因进行柱形图统计----
EnSc=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0718.rds")

DimPlot(EnSc)


path=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/SCPA分析/PATH12_2.csv",header = F,row.names = 1)

path=t(path)


path=as.data.frame(path)


S0_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/S0/S0_deseq.csv")
Cluster0=subset(S0_deseq,S0_deseq$p_val<0.05)$X

S1_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/S1/S1_deseq.csv")
Cluster2=subset(S1_deseq,S1_deseq$p_val<0.05)$X

REV3L_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/REV3L/REV3L_deseq.csv")
Cluster1=subset(REV3L_deseq,REV3L_deseq$p_val<0.05)$X

BMP7_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/BMP7/BMP7_deseq.csv")
Cluster4=subset(BMP7_deseq,BMP7_deseq$p_val<0.05)$X


ISG15_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/ISG15/ISG15_deseq.csv")
Cluster6=subset(ISG15_deseq,ISG15_deseq$p_val<0.05)$X


P_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/P/P_deseq.csv")
Cluster5=subset(P_deseq,P_deseq$p_val<0.05)$X


Cycling_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/Cycling/Cycling_deseq.csv")
Cluster3=subset(Cycling_deseq,Cycling_deseq$p_val<0.05)$X



#Cluster0在各个通路内的差异基因数目
rownames(as.data.frame(intersect(path$`AMPK signaling pathway`,Cluster0)))
rownames(as.data.frame(intersect(path$`AMPK signaling pathway`,Cluster1)))
rownames(as.data.frame(intersect(path$`AMPK signaling pathway`,Cluster2)))
rownames(as.data.frame(intersect(path$`AMPK signaling pathway`,Cluster3)))
rownames(as.data.frame(intersect(path$`AMPK signaling pathway`,Cluster4)))
rownames(as.data.frame(intersect(path$`AMPK signaling pathway`,Cluster5)))
rownames(as.data.frame(intersect(path$`AMPK signaling pathway`,Cluster6)))



rownames(as.data.frame(intersect(path$`HIF-1 signaling pathway`,Cluster0)))
rownames(as.data.frame(intersect(path$`HIF-1 signaling pathway`,Cluster1)))
rownames(as.data.frame(intersect(path$`HIF-1 signaling pathway`,Cluster2)))
rownames(as.data.frame(intersect(path$`HIF-1 signaling pathway`,Cluster3)))
rownames(as.data.frame(intersect(path$`HIF-1 signaling pathway`,Cluster4)))
rownames(as.data.frame(intersect(path$`HIF-1 signaling pathway`,Cluster5)))
rownames(as.data.frame(intersect(path$`HIF-1 signaling pathway`,Cluster6)))



rownames(as.data.frame(intersect(path$`Oxidative phosphorylation`,Cluster0)))
rownames(as.data.frame(intersect(path$`Oxidative phosphorylation`,Cluster1)))
rownames(as.data.frame(intersect(path$`Oxidative phosphorylation`,Cluster2)))
rownames(as.data.frame(intersect(path$`Oxidative phosphorylation`,Cluster3)))
rownames(as.data.frame(intersect(path$`Oxidative phosphorylation`,Cluster4)))
rownames(as.data.frame(intersect(path$`Oxidative phosphorylation`,Cluster5)))
rownames(as.data.frame(intersect(path$`Oxidative phosphorylation`,Cluster6)))



rownames(as.data.frame(intersect(path$`Glycolysis / Gluconeogenesis`,Cluster0)))
rownames(as.data.frame(intersect(path$`Glycolysis / Gluconeogenesis`,Cluster1)))
rownames(as.data.frame(intersect(path$`Glycolysis / Gluconeogenesis`,Cluster2)))
rownames(as.data.frame(intersect(path$`Glycolysis / Gluconeogenesis`,Cluster3)))
rownames(as.data.frame(intersect(path$`Glycolysis / Gluconeogenesis`,Cluster4)))
rownames(as.data.frame(intersect(path$`Glycolysis / Gluconeogenesis`,Cluster5)))
rownames(as.data.frame(intersect(path$`Glycolysis / Gluconeogenesis`,Cluster6)))


rownames(as.data.frame(intersect(path$`Pyruvate metabolism`,Cluster0)))
rownames(as.data.frame(intersect(path$`Pyruvate metabolism`,Cluster1)))
rownames(as.data.frame(intersect(path$`Pyruvate metabolism`,Cluster2)))
rownames(as.data.frame(intersect(path$`Pyruvate metabolism`,Cluster3)))
rownames(as.data.frame(intersect(path$`Pyruvate metabolism`,Cluster4)))
rownames(as.data.frame(intersect(path$`Pyruvate metabolism`,Cluster5)))
rownames(as.data.frame(intersect(path$`Pyruvate metabolism`,Cluster6)))

rownames(as.data.frame(intersect(path$`Fatty acid degradation`,Cluster0)))
rownames(as.data.frame(intersect(path$`Fatty acid degradation`,Cluster1)))
rownames(as.data.frame(intersect(path$`Fatty acid degradation`,Cluster2)))
rownames(as.data.frame(intersect(path$`Fatty acid degradation`,Cluster3)))
rownames(as.data.frame(intersect(path$`Fatty acid degradation`,Cluster4)))
rownames(as.data.frame(intersect(path$`Fatty acid degradation`,Cluster5)))
rownames(as.data.frame(intersect(path$`Fatty acid degradation`,Cluster6)))


rownames(as.data.frame(intersect(path$`Citrate cycle (TCA cycle)`,Cluster0)))
rownames(as.data.frame(intersect(path$`Citrate cycle (TCA cycle)`,Cluster1)))
rownames(as.data.frame(intersect(path$`Citrate cycle (TCA cycle)`,Cluster2)))
rownames(as.data.frame(intersect(path$`Citrate cycle (TCA cycle)`,Cluster3)))
rownames(as.data.frame(intersect(path$`Citrate cycle (TCA cycle)`,Cluster4)))
rownames(as.data.frame(intersect(path$`Citrate cycle (TCA cycle)`,Cluster5)))
rownames(as.data.frame(intersect(path$`Citrate cycle (TCA cycle)`,Cluster6)))



rownames(as.data.frame(intersect(path$`Pentose phosphate pathway`,Cluster0)))
rownames(as.data.frame(intersect(path$`Pentose phosphate pathway`,Cluster1)))
rownames(as.data.frame(intersect(path$`Pentose phosphate pathway`,Cluster2)))
rownames(as.data.frame(intersect(path$`Pentose phosphate pathway`,Cluster3)))
rownames(as.data.frame(intersect(path$`Pentose phosphate pathway`,Cluster4)))
rownames(as.data.frame(intersect(path$`Pentose phosphate pathway`,Cluster5)))
rownames(as.data.frame(intersect(path$`Pentose phosphate pathway`,Cluster6)))

rownames(as.data.frame(intersect(path$`Fatty acid degradation`,Cluster0)))
rownames(as.data.frame(intersect(path$`Fatty acid degradation`,Cluster1)))
rownames(as.data.frame(intersect(path$`Fatty acid degradation`,Cluster2)))
rownames(as.data.frame(intersect(path$`Fatty acid degradation`,Cluster3)))
rownames(as.data.frame(intersect(path$`Fatty acid degradation`,Cluster4)))
rownames(as.data.frame(intersect(path$`Fatty acid degradation`,Cluster5)))
rownames(as.data.frame(intersect(path$`Fatty acid degradation`,Cluster6)))

rownames(as.data.frame(intersect(path$`Fatty acid biosynthesis`,Cluster0)))
rownames(as.data.frame(intersect(path$`Fatty acid biosynthesis`,Cluster1)))
rownames(as.data.frame(intersect(path$`Fatty acid biosynthesis`,Cluster2)))
rownames(as.data.frame(intersect(path$`Fatty acid biosynthesis`,Cluster3)))
rownames(as.data.frame(intersect(path$`Fatty acid biosynthesis`,Cluster4)))
rownames(as.data.frame(intersect(path$`Fatty acid biosynthesis`,Cluster5)))
rownames(as.data.frame(intersect(path$`Fatty acid biosynthesis`,Cluster6)))


rownames(as.data.frame(intersect(path$`Glutathione metabolism`,Cluster0)))
rownames(as.data.frame(intersect(path$`Glutathione metabolism`,Cluster1)))
rownames(as.data.frame(intersect(path$`Glutathione metabolism`,Cluster2)))
rownames(as.data.frame(intersect(path$`Glutathione metabolism`,Cluster3)))
rownames(as.data.frame(intersect(path$`Glutathione metabolism`,Cluster4)))
rownames(as.data.frame(intersect(path$`Glutathione metabolism`,Cluster5)))
rownames(as.data.frame(intersect(path$`Glutathione metabolism`,Cluster6)))



rownames(as.data.frame(intersect(path$`Alanine, aspartate and glutamate metabolism`,Cluster0)))
rownames(as.data.frame(intersect(path$`Alanine, aspartate and glutamate metabolism`,Cluster1)))
rownames(as.data.frame(intersect(path$`Alanine, aspartate and glutamate metabolism`,Cluster2)))
rownames(as.data.frame(intersect(path$`Alanine, aspartate and glutamate metabolism`,Cluster3)))
rownames(as.data.frame(intersect(path$`Alanine, aspartate and glutamate metabolism`,Cluster4)))
rownames(as.data.frame(intersect(path$`Alanine, aspartate and glutamate metabolism`,Cluster5)))
rownames(as.data.frame(intersect(path$`Alanine, aspartate and glutamate metabolism`,Cluster6)))



rownames(as.data.frame(intersect(path$`Insulin signaling pathway`,Cluster0)))
rownames(as.data.frame(intersect(path$`Insulin signaling pathway`,Cluster1)))
rownames(as.data.frame(intersect(path$`Insulin signaling pathway`,Cluster2)))
rownames(as.data.frame(intersect(path$`Insulin signaling pathway`,Cluster3)))
rownames(as.data.frame(intersect(path$`Insulin signaling pathway`,Cluster4)))
rownames(as.data.frame(intersect(path$`Insulin signaling pathway`,Cluster5)))
rownames(as.data.frame(intersect(path$`Insulin signaling pathway`,Cluster6)))


rownames(as.data.frame(intersect(path$`Tyrosine metabolism`,Cluster0)))
rownames(as.data.frame(intersect(path$`Tyrosine metabolism`,Cluster1)))
rownames(as.data.frame(intersect(path$`Tyrosine metabolism`,Cluster2)))
rownames(as.data.frame(intersect(path$`Tyrosine metabolism`,Cluster3)))
rownames(as.data.frame(intersect(path$`Tyrosine metabolism`,Cluster4)))
rownames(as.data.frame(intersect(path$`Tyrosine metabolism`,Cluster5)))
rownames(as.data.frame(intersect(path$`Tyrosine metabolism`,Cluster6)))


rownames(as.data.frame(intersect(path$`Amino sugar and  nucleottde sugar metabolism`,Cluster0)))
rownames(as.data.frame(intersect(path$`Amino sugar and  nucleottde sugar metabolism`,Cluster1)))
rownames(as.data.frame(intersect(path$`Amino sugar and  nucleottde sugar metabolism`,Cluster2)))
rownames(as.data.frame(intersect(path$`Amino sugar and  nucleottde sugar metabolism`,Cluster3)))
rownames(as.data.frame(intersect(path$`Amino sugar and  nucleottde sugar metabolism`,Cluster4)))
rownames(as.data.frame(intersect(path$`Amino sugar and  nucleottde sugar metabolism`,Cluster5)))
rownames(as.data.frame(intersect(path$`Amino sugar and  nucleottde sugar metabolism`,Cluster6)))


rownames(as.data.frame(intersect(path$`Cysteine and methionine metabolism`,Cluster0)))
rownames(as.data.frame(intersect(path$`Cysteine and methionine metabolism`,Cluster1)))
rownames(as.data.frame(intersect(path$`Cysteine and methionine metabolism`,Cluster2)))
rownames(as.data.frame(intersect(path$`Cysteine and methionine metabolism`,Cluster3)))
rownames(as.data.frame(intersect(path$`Cysteine and methionine metabolism`,Cluster4)))
rownames(as.data.frame(intersect(path$`Cysteine and methionine metabolism`,Cluster5)))
rownames(as.data.frame(intersect(path$`Cysteine and methionine metabolism`,Cluster6)))



rownames(as.data.frame(intersect(path$`Glycerophospholipid metabolism`,Cluster0)))
rownames(as.data.frame(intersect(path$`Glycerophospholipid metabolism`,Cluster1)))
rownames(as.data.frame(intersect(path$`Glycerophospholipid metabolism`,Cluster2)))
rownames(as.data.frame(intersect(path$`Glycerophospholipid metabolism`,Cluster3)))
rownames(as.data.frame(intersect(path$`Glycerophospholipid metabolism`,Cluster4)))
rownames(as.data.frame(intersect(path$`Glycerophospholipid metabolism`,Cluster5)))
rownames(as.data.frame(intersect(path$`Glycerophospholipid metabolism`,Cluster6)))




rownames(as.data.frame(intersect(path$`Glycine serine and threoinine metabolism`,Cluster0)))
rownames(as.data.frame(intersect(path$`Glycine serine and threoinine metabolism`,Cluster1)))
rownames(as.data.frame(intersect(path$`Glycine serine and threoinine metabolism`,Cluster2)))
rownames(as.data.frame(intersect(path$`Glycine serine and threoinine metabolism`,Cluster3)))
rownames(as.data.frame(intersect(path$`Glycine serine and threoinine metabolism`,Cluster4)))
rownames(as.data.frame(intersect(path$`Glycine serine and threoinine metabolism`,Cluster5)))
rownames(as.data.frame(intersect(path$`Glycine serine and threoinine metabolism`,Cluster6)))



#开始绘图
SCPA_bar=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/SCPA分析/绘图/SCPA_bar.csv")


SCPA_bar2=gather(SCPA_bar,gene, value, colnames(SCPA_bar),-X)

colnames(SCPA_bar2)=c("Term","Group","Value")


SCPA_bar2$Group=gsub("all","Total N of genes/pathway",SCPA_bar2$Group)

SCPA_bar2$Group=factor(SCPA_bar2$Group,levels = c("Cluster6","Cluster5","Cluster4","Cluster3","Cluster2","Cluster1","Cluster0",
                                                  "Total N of genes/pathway"))


ggplot(SCPA_bar2)+aes(x=Value,y=Term,fill=Group)+geom_bar(stat="identity",width=0.55,color="black")+
  theme_classic()+scale_x_continuous(expand=c(0,0))+xlab("")+ylab("")+
  scale_y_discrete(limits=c("Fatty acid biosynthesis","Fatty acid elongation","Pentose phosphate pathway","Citrate cycle (TCA cycle)",
                            "Alanine, aspartate and glutamate metabolism","Glycine serine and threoinine metabolism","Cysteine and methionine metabolism",
                            "Fatty acid degradation","Pyruvate metabolism","Tyrosine metabolism","Amino sugar and  nucleottde sugar metabolism",
                            "Glutathione metabolism","Glycolysis / Gluconeogenesis","Glycerophospholipid metabolism","HIF-1 signaling pathway",
                            "AMPK signaling pathway","Insulin signaling pathway","Oxidative phosphorylation"))+
  theme(axis.text = element_text(face = "bold",color = "black",size = 12),
        axis.title  = element_text(face = "bold",color = "black",size = 12),
        legend.text = element_text(face = "bold",color = "black",size = 12),
        legend.title = element_text(face = "bold",color = "black",size = 0))+
  scale_fill_manual(values=c("#FB61D7","#AB92FF","#02B6EB","#0DC399","#53B400","#C8A010","#F8766E","#626262"))+
  xlab("N of genes")+ylab("Metabolic pathway")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/SCPA分析/绘图/EnSc_SCPA_A.pdf",
       width = 15,height = 7.2)


#对基质细胞中的SCPA通路统计算法P值进行绘图
SCPA_heatmap=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/SCPA分析/绘图/SCPA_heatmap.csv",row.names = 1)


xx=pheatmap(SCPA_heatmap)


save_pheatmap_pdf <- function(x, filename, width=7, height=5) {
  stopifnot(!missing(x))
  stopifnot(!missing(filename))
  pdf(filename, width=width, height=height)
  grid::grid.newpage()
  grid::grid.draw(x$gtable)
  dev.off()
}



save_pheatmap_pdf(xx, "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/SCPA分析/绘图/SCPA_heatmap.pdf")

#散点图
SCPA_point=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/SCPA分析/绘图/rest_data.csv",row.names = 1)



ggplot(SCPA_point)+aes(x=qval,y=path_rank)+geom_point(size=7,color="#9B395A")+
  theme_calc()+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,face = "bold",colour = "black",size =10 ),
        axis.text.y = element_text(face = "bold",colour = "black",size =10 ),
        axis.title.y =element_text(face = "bold",colour = "black",size =10 ),axis.title = element_text(face = "bold",colour = "black",size =10 ))+
  xlab("")+ylab("")+theme(legend.position = "none")+xlim(-6,8)+geom_text(data=SCPA_point,aes(x=qval,y=path_rank,label=SCPA_point$Pathway),label.padding=0.33)

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/SCPA分析/绘图/EnSc_SCPA_B.pdf",
       width = 13,height = 8.2)

#对巨噬细胞进行SCPA分析绘图----------
#首先对巨噬细胞各个亚群进行adhesion和ctrl组的差异基因分析
Macr=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/Macr_allsingle.rds")

Macr$type5=paste(Macr$type4,"_",Macr$cell_type0610,sep = "")

#CD163+
CD163_deseq=FindMarkers(Macr,ident.1 = "Adhesion_CD163+",ident.2 = "Ctrl_CD163+",
                        min.pct = 0.2,group.by = "type5")
CD163_deseq=subset(CD163_deseq,CD163_deseq$p_val<0.05)

CD163_deseq=rownames(CD163_deseq)

#CD24+
CD24_deseq=FindMarkers(Macr,ident.1 = "Adhesion_CD24+",ident.2 = "Ctrl_CD24+",
                       min.pct = 0.2,group.by = "type5")
CD24_deseq=subset(CD24_deseq,CD24_deseq$p_val<0.05)

CD24_deseq=rownames(CD24_deseq)

#CD301+
CD301_deseq=FindMarkers(Macr,ident.1 = "Adhesion_CD301+",ident.2 = "Ctrl_CD301+",
                        min.pct = 0.2,group.by = "type5")
CD301_deseq=subset(CD301_deseq,CD301_deseq$p_val<0.05)

CD301_deseq=rownames(CD301_deseq)



#Proliferating
Proliferating_deseq=FindMarkers(Macr,ident.1 = "Adhesion_Proliferating",ident.2 = "Ctrl_Proliferating",
                                min.pct = 0.2,group.by = "type5")
Proliferating_deseq=subset(Proliferating_deseq,Proliferating_deseq$p_val<0.05)

Proliferating_deseq=rownames(Proliferating_deseq)

#Monocyte+
Monocyte_deseq=FindMarkers(Macr,ident.1 = "Adhesion_Monocyte+",ident.2 = "Ctrl_Monocyte+",
                           min.pct = 0.2,group.by = "type5")
Monocyte_deseq=subset(Monocyte_deseq,Monocyte_deseq$p_val<0.05)

Monocyte_deseq=rownames(Monocyte_deseq)

#Uknow
Uknow_deseq=FindMarkers(Macr,ident.1 = "Adhesion_Uknow",ident.2 = "Ctrl_Uknow",
                        min.pct = 0.2,group.by = "type5")
Uknow_deseq=subset(Uknow_deseq,Uknow_deseq$p_val<0.05)
Uknow_deseq=rownames(Uknow_deseq)


path=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/SCPA分析/PATH12_2.csv",header = F,row.names = 1)

path=t(path)


path=as.data.frame(path)


#统计差异基因数目
rownames(as.data.frame(intersect(path$`AMPK signaling pathway`,CD163_deseq)))
rownames(as.data.frame(intersect(path$`AMPK signaling pathway`,CD24_deseq)))
rownames(as.data.frame(intersect(path$`AMPK signaling pathway`,CD301_deseq)))
rownames(as.data.frame(intersect(path$`AMPK signaling pathway`,Monocyte_deseq)))
rownames(as.data.frame(intersect(path$`AMPK signaling pathway`,Proliferating_deseq)))
rownames(as.data.frame(intersect(path$`AMPK signaling pathway`,Uknow_deseq)))

rownames(as.data.frame(intersect(path$`HIF-1 signaling pathway`,CD163_deseq)))
rownames(as.data.frame(intersect(path$`HIF-1 signaling pathway`,CD24_deseq)))
rownames(as.data.frame(intersect(path$`HIF-1 signaling pathway`,CD301_deseq)))
rownames(as.data.frame(intersect(path$`HIF-1 signaling pathway`,Monocyte_deseq)))
rownames(as.data.frame(intersect(path$`HIF-1 signaling pathway`,Proliferating_deseq)))
rownames(as.data.frame(intersect(path$`HIF-1 signaling pathway`,Uknow_deseq)))



rownames(as.data.frame(intersect(path$`Oxidative phosphorylation`,CD163_deseq)))
rownames(as.data.frame(intersect(path$`Oxidative phosphorylation`,CD24_deseq)))
rownames(as.data.frame(intersect(path$`Oxidative phosphorylation`,CD301_deseq)))
rownames(as.data.frame(intersect(path$`Oxidative phosphorylation`,Monocyte_deseq)))
rownames(as.data.frame(intersect(path$`Oxidative phosphorylation`,Proliferating_deseq)))
rownames(as.data.frame(intersect(path$`Oxidative phosphorylation`,Uknow_deseq)))



rownames(as.data.frame(intersect(path$`Glycolysis / Gluconeogenesis`,CD163_deseq)))
rownames(as.data.frame(intersect(path$`Glycolysis / Gluconeogenesis`,CD24_deseq)))
rownames(as.data.frame(intersect(path$`Glycolysis / Gluconeogenesis`,CD301_deseq)))
rownames(as.data.frame(intersect(path$`Glycolysis / Gluconeogenesis`,Monocyte_deseq)))
rownames(as.data.frame(intersect(path$`Glycolysis / Gluconeogenesis`,Proliferating_deseq)))
rownames(as.data.frame(intersect(path$`Glycolysis / Gluconeogenesis`,Uknow_deseq)))


rownames(as.data.frame(intersect(path$`Pyruvate metabolism`,CD163_deseq)))
rownames(as.data.frame(intersect(path$`Pyruvate metabolism`,CD24_deseq)))
rownames(as.data.frame(intersect(path$`Pyruvate metabolism`,CD301_deseq)))
rownames(as.data.frame(intersect(path$`Pyruvate metabolism`,Monocyte_deseq)))
rownames(as.data.frame(intersect(path$`Pyruvate metabolism`,Proliferating_deseq)))
rownames(as.data.frame(intersect(path$`Pyruvate metabolism`,Uknow_deseq)))

rownames(as.data.frame(intersect(path$`Fatty acid degradation`,CD163_deseq)))
rownames(as.data.frame(intersect(path$`Fatty acid degradation`,CD24_deseq)))
rownames(as.data.frame(intersect(path$`Fatty acid degradation`,CD301_deseq)))
rownames(as.data.frame(intersect(path$`Fatty acid degradation`,Monocyte_deseq)))
rownames(as.data.frame(intersect(path$`Fatty acid degradation`,Proliferating_deseq)))
rownames(as.data.frame(intersect(path$`Fatty acid degradation`,Uknow_deseq)))


rownames(as.data.frame(intersect(path$`Fatty acid elongation`,CD163_deseq)))
rownames(as.data.frame(intersect(path$`Fatty acid elongation`,CD24_deseq)))
rownames(as.data.frame(intersect(path$`Fatty acid elongation`,CD301_deseq)))
rownames(as.data.frame(intersect(path$`Fatty acid elongation`,Monocyte_deseq)))
rownames(as.data.frame(intersect(path$`Fatty acid elongation`,Proliferating_deseq)))
rownames(as.data.frame(intersect(path$`Fatty acid elongation`,Uknow_deseq)))



rownames(as.data.frame(intersect(path$`Citrate cycle (TCA cycle)`,CD163_deseq)))
rownames(as.data.frame(intersect(path$`Citrate cycle (TCA cycle)`,CD24_deseq)))
rownames(as.data.frame(intersect(path$`Citrate cycle (TCA cycle)`,CD301_deseq)))
rownames(as.data.frame(intersect(path$`Citrate cycle (TCA cycle)`,Monocyte_deseq)))
rownames(as.data.frame(intersect(path$`Citrate cycle (TCA cycle)`,Proliferating_deseq)))
rownames(as.data.frame(intersect(path$`Citrate cycle (TCA cycle)`,Uknow_deseq)))


rownames(as.data.frame(intersect(path$`Pentose phosphate pathway`,CD163_deseq)))
rownames(as.data.frame(intersect(path$`Pentose phosphate pathway`,CD24_deseq)))
rownames(as.data.frame(intersect(path$`Pentose phosphate pathway`,CD301_deseq)))
rownames(as.data.frame(intersect(path$`Pentose phosphate pathway`,Monocyte_deseq)))
rownames(as.data.frame(intersect(path$`Pentose phosphate pathway`,Proliferating_deseq)))
rownames(as.data.frame(intersect(path$`Pentose phosphate pathway`,Uknow_deseq)))


rownames(as.data.frame(intersect(path$`Fatty acid biosynthesis`,CD163_deseq)))
rownames(as.data.frame(intersect(path$`Fatty acid biosynthesis`,CD24_deseq)))
rownames(as.data.frame(intersect(path$`Fatty acid biosynthesis`,CD301_deseq)))
rownames(as.data.frame(intersect(path$`Fatty acid biosynthesis`,Monocyte_deseq)))
rownames(as.data.frame(intersect(path$`Fatty acid biosynthesis`,Proliferating_deseq)))
rownames(as.data.frame(intersect(path$`Fatty acid biosynthesis`,Uknow_deseq)))


rownames(as.data.frame(intersect(path$`Glutathione metabolism`,CD163_deseq)))
rownames(as.data.frame(intersect(path$`Glutathione metabolism`,CD24_deseq)))
rownames(as.data.frame(intersect(path$`Glutathione metabolism`,CD301_deseq)))
rownames(as.data.frame(intersect(path$`Glutathione metabolism`,Monocyte_deseq)))
rownames(as.data.frame(intersect(path$`Glutathione metabolism`,Proliferating_deseq)))
rownames(as.data.frame(intersect(path$`Glutathione metabolism`,Uknow_deseq)))



rownames(as.data.frame(intersect(path$`Alanine, aspartate and glutamate metabolism`,CD163_deseq)))
rownames(as.data.frame(intersect(path$`Alanine, aspartate and glutamate metabolism`,CD24_deseq)))
rownames(as.data.frame(intersect(path$`Alanine, aspartate and glutamate metabolism`,CD301_deseq)))
rownames(as.data.frame(intersect(path$`Alanine, aspartate and glutamate metabolism`,Monocyte_deseq)))
rownames(as.data.frame(intersect(path$`Alanine, aspartate and glutamate metabolism`,Proliferating_deseq)))
rownames(as.data.frame(intersect(path$`Alanine, aspartate and glutamate metabolism`,Uknow_deseq)))



rownames(as.data.frame(intersect(path$`Insulin signaling pathway`,CD163_deseq)))
rownames(as.data.frame(intersect(path$`Insulin signaling pathway`,CD24_deseq)))
rownames(as.data.frame(intersect(path$`Insulin signaling pathway`,CD301_deseq)))
rownames(as.data.frame(intersect(path$`Insulin signaling pathway`,Monocyte_deseq)))
rownames(as.data.frame(intersect(path$`Insulin signaling pathway`,Proliferating_deseq)))
rownames(as.data.frame(intersect(path$`Insulin signaling pathway`,Uknow_deseq)))


rownames(as.data.frame(intersect(path$`Tyrosine metabolism`,CD163_deseq)))
rownames(as.data.frame(intersect(path$`Tyrosine metabolism`,CD24_deseq)))
rownames(as.data.frame(intersect(path$`Tyrosine metabolism`,CD301_deseq)))
rownames(as.data.frame(intersect(path$`Tyrosine metabolism`,Monocyte_deseq)))
rownames(as.data.frame(intersect(path$`Tyrosine metabolism`,Proliferating_deseq)))
rownames(as.data.frame(intersect(path$`Tyrosine metabolism`,Uknow_deseq)))


rownames(as.data.frame(intersect(path$`Amino sugar and  nucleottde sugar metabolism`,CD163_deseq)))
rownames(as.data.frame(intersect(path$`Amino sugar and  nucleottde sugar metabolism`,CD24_deseq)))
rownames(as.data.frame(intersect(path$`Amino sugar and  nucleottde sugar metabolism`,CD301_deseq)))
rownames(as.data.frame(intersect(path$`Amino sugar and  nucleottde sugar metabolism`,Monocyte_deseq)))
rownames(as.data.frame(intersect(path$`Amino sugar and  nucleottde sugar metabolism`,Proliferating_deseq)))
rownames(as.data.frame(intersect(path$`Amino sugar and  nucleottde sugar metabolism`,Uknow_deseq)))


rownames(as.data.frame(intersect(path$`Cysteine and methionine metabolism`,CD163_deseq)))
rownames(as.data.frame(intersect(path$`Cysteine and methionine metabolism`,CD24_deseq)))
rownames(as.data.frame(intersect(path$`Cysteine and methionine metabolism`,CD301_deseq)))
rownames(as.data.frame(intersect(path$`Cysteine and methionine metabolism`,Monocyte_deseq)))
rownames(as.data.frame(intersect(path$`Cysteine and methionine metabolism`,Proliferating_deseq)))
rownames(as.data.frame(intersect(path$`Cysteine and methionine metabolism`,Uknow_deseq)))



rownames(as.data.frame(intersect(path$`Glycerophospholipid metabolism`,CD163_deseq)))
rownames(as.data.frame(intersect(path$`Glycerophospholipid metabolism`,CD24_deseq)))
rownames(as.data.frame(intersect(path$`Glycerophospholipid metabolism`,CD301_deseq)))
rownames(as.data.frame(intersect(path$`Glycerophospholipid metabolism`,Monocyte_deseq)))
rownames(as.data.frame(intersect(path$`Glycerophospholipid metabolism`,Proliferating_deseq)))
rownames(as.data.frame(intersect(path$`Glycerophospholipid metabolism`,Uknow_deseq)))



rownames(as.data.frame(intersect(path$`Glycine serine and threoinine metabolism`,CD163_deseq)))
rownames(as.data.frame(intersect(path$`Glycine serine and threoinine metabolism`,CD24_deseq)))
rownames(as.data.frame(intersect(path$`Glycine serine and threoinine metabolism`,CD301_deseq)))
rownames(as.data.frame(intersect(path$`Glycine serine and threoinine metabolism`,Monocyte_deseq)))
rownames(as.data.frame(intersect(path$`Glycine serine and threoinine metabolism`,Proliferating_deseq)))
rownames(as.data.frame(intersect(path$`Glycine serine and threoinine metabolism`,Uknow_deseq)))







#开始绘图
SCPA_bar=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/SCPA分析/绘图/Macr_SCPA_bar.csv")


SCPA_bar2=gather(SCPA_bar,gene, value, colnames(SCPA_bar),-pathway)

colnames(SCPA_bar2)=c("Term","Group","Value")


SCPA_bar2$Group=gsub("all","Total N of genes/pathway",SCPA_bar2$Group)

SCPA_bar2$Group=gsub("CD163.","CD163+",SCPA_bar2$Group)
SCPA_bar2$Group=gsub("CD24.","CD24+",SCPA_bar2$Group)
SCPA_bar2$Group=gsub("CD301.","CD301+",SCPA_bar2$Group)


SCPA_bar2$Group=factor(SCPA_bar2$Group,levels = c("Unknow","Proliferating","Monocyte","CD301+",
                                                  "CD24+","CD163+",
                                                  "Total N of genes/pathway"))


ggplot(SCPA_bar2)+aes(x=Value,y=Term,fill=Group)+geom_bar(stat="identity",width=0.55,color="black")+
  theme_classic()+scale_x_continuous(expand=c(0,0))+xlab("")+ylab("")+
  scale_y_discrete(limits=c("Fatty acid biosynthesis","Fatty acid elongation","Pentose phosphate pathway","Citrate cycle (TCA cycle)",
                            "Alanine, aspartate and glutamate metabolism","Glycine serine and threoinine metabolism","Cysteine and methionine metabolism",
                            "Fatty acid degradation","Pyruvate metabolism","Tyrosine metabolism","Amino sugar and  nucleottde sugar metabolism",
                            "Glutathione metabolism","Glycolysis / Gluconeogenesis","Glycerophospholipid metabolism","HIF-1 signaling pathway",
                            "AMPK signaling pathway","Insulin signaling pathway","Oxidative phosphorylation"))+
  theme(axis.text = element_text(face = "bold",color = "black",size = 12),
        axis.title  = element_text(face = "bold",color = "black",size = 12),
        legend.text = element_text(face = "bold",color = "black",size = 12),
        legend.title = element_text(face = "bold",color = "black",size = 0))+
  scale_fill_manual(values=c("#F564E3","#619CFF","#00BA38","#B79F00","#00BFC4","#F8766D","#626262"))+
  xlab("N of genes")+ylab("Metabolic pathway")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/SCPA分析/绘图/Macr_SCPA_A.pdf",
       width = 15,height = 7.2)



#散点图
SCPA_point=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/SCPA分析/绘图/Macr/Macr_rest_data.csv",row.names = 1)



ggplot(SCPA_point)+aes(x=qval,y=path_rank)+geom_point(size=7,color="#9B395A")+
  theme_calc()+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,face = "bold",colour = "black",size =10 ),
        axis.text.y = element_text(face = "bold",colour = "black",size =10 ),
        axis.title.y =element_text(face = "bold",colour = "black",size =10 ),axis.title = element_text(face = "bold",colour = "black",size =10 ))+
  xlab("")+ylab("")+theme(legend.position = "none")+xlim(-6,8)+geom_text(data=SCPA_point,aes(x=qval,y=path_rank,label=SCPA_point$Pathway),label.padding=0.33)


ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/SCPA分析/绘图/Macr_SCPA_B.pdf",
       width = 13,height = 8.2)




#热图
SCPA_heatmap=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/SCPA分析/绘图/Macr/Macr_heatmap.csv",row.names = 1)


xx=pheatmap(SCPA_heatmap)


save_pheatmap_pdf <- function(x, filename, width=7, height=5) {
  stopifnot(!missing(x))
  stopifnot(!missing(filename))
  pdf(filename, width=width, height=height)
  grid::grid.newpage()
  grid::grid.draw(x$gtable)
  dev.off()
}



save_pheatmap_pdf(xx, "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/SCPA分析/绘图/Macr_SCPA_C.pdf")


#decoupleR分析转录因子情况-----
library(decoupleR)
library(dplyr)
library(bslib)
library(tclust)
library(OmnipathR)
library(tidyr)
library(htmltools)
library(tibble)
###基质细胞所有转录因子差异----------
sce=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0718.rds")
Idents(sce)=sce$type4

net<-get_collectri(organism='human',split_complexes=FALSE)
net


mat<-as.matrix(sce@assays$RNA@data)



acts<-run_ulm(mat,net,minsize=5,
              .source='source',.target='target',.mor='mor')
acts


sce[['tfsulm']]<-acts%>%
  pivot_wider(id_cols='source',
              names_from='condition',
              values_from='score')%>%
  column_to_rownames('source')%>%
  Seurat::CreateAssayObject(.)
#更改默认的分析对象
DefaultAssay(object=sce)<-"tfsulm"
#缩放数据并重新保存
sce<-ScaleData(sce)
sce@assays$tfsulm@data<-sce@assays$tfsulm@scale.data

DefaultAssay(object=sce)<-"tfsulm"


n_tfs<-25
#提取活性数据并修改数据格式
df<-t(as.matrix(sce@assays$tfsulm@data))%>%
  as.data.frame()%>%
  mutate(cluster=Idents(sce))%>%
  pivot_longer(cols=-cluster,#指定除cluster列以外的都需要转换
               names_to="source",#列名放到source列中
               values_to="score")%>%#值放到score列中
  group_by(cluster,source)%>%#将数据按cluster和source两列进行分组
  summarise(mean=mean(score))
#获得不同簇中活性差异较大的TFs
tfs<-df%>%
  group_by(source)%>%
  summarise(std=sd(mean))%>%
  arrange(-abs(std))%>%
  head(n_tfs)%>%
  pull(source)

#修改数据格式
top_acts_mat<-df%>%
  filter(source%in%tfs)%>%
  pivot_wider(id_cols='cluster',
              names_from='source',
              values_from='mean')%>%
  column_to_rownames('cluster')%>%
  as.matrix()

#绘图
palette_length=100
my_color=colorRampPalette(c("Darkblue","white","red"))(palette_length)

my_breaks<-c(seq(-3,0,length.out=ceiling(palette_length/2)+1),
             seq(0.05,3,length.out=floor(palette_length/2)))


#显示
save_pheatmap_pdf <- function(x, filename, width=6, height=2.5) {
  stopifnot(!missing(x))
  stopifnot(!missing(filename))
  pdf(filename, width=width, height=height)
  grid::grid.newpage()
  grid::grid.draw(x$gtable)
  dev.off()
}







xx=pheatmap(top_acts_mat,
            border_color=NA,
            color=my_color,
            breaks=my_breaks,cluster_rows = F,cluster_cols = F)




save_pheatmap_pdf(xx, "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/SCPA分析/绘图/P1.pdf")



#




#
###基质细胞代谢相关的转录因子差异----------
sce=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0718.rds")
Idents(sce)=sce$type4

path=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/SCPA分析/PATH12_2.csv",header = F,row.names = 1)

path=t(path)


path=as.data.frame(path)

gene=c(path$`AMPK signaling pathway`,path$`HIF-1 signaling pathway`,path$`Oxidative phosphorylation`,path$`Glycolysis / Gluconeogenesis`,
       path$`Pyruvate metabolism`,path$`Fatty acid degradation`,path$`Citrate cycle (TCA cycle)`,path$`Pentose phosphate pathway`,
       path$`Fatty acid elongation`,path$`Fatty acid biosynthesis`,path$`Glutathione metabolism`,path$`Alanine, aspartate and glutamate metabolism`,
       path$`Insulin signaling pathway`,path$`Tyrosine metabolism`,path$`Amino sugar and  nucleottde sugar metabolism`,
       path$`Cysteine and methionine metabolism`,path$`Glycerophospholipid metabolism`,
       path$`Glycine serine and threoinine metabolism`)


gene=gene[-which(gene=="")]

table(duplicated(gene))

gene<-gene[!duplicated(gene)]



net<-get_collectri(organism='human',split_complexes=FALSE)
net


mat<-as.matrix(sce@assays$RNA@data)

mat=as.data.frame(mat)






mat=mat[gene,]

mat<-as.matrix(mat)


acts<-run_ulm(mat,net,minsize=5,
              .source='source',.target='target',.mor='mor')
acts


sce[['tfsulm']]<-acts%>%
  pivot_wider(id_cols='source',
              names_from='condition',
              values_from='score')%>%
  column_to_rownames('source')%>%
  Seurat::CreateAssayObject(.)
#更改默认的分析对象
DefaultAssay(object=sce)<-"tfsulm"
#缩放数据并重新保存
sce<-ScaleData(sce)
sce@assays$tfsulm@data<-sce@assays$tfsulm@scale.data

DefaultAssay(object=sce)<-"tfsulm"


n_tfs<-25
#提取活性数据并修改数据格式
df<-t(as.matrix(sce@assays$tfsulm@data))%>%
  as.data.frame()%>%
  mutate(cluster=Idents(sce))%>%
  pivot_longer(cols=-cluster,#指定除cluster列以外的都需要转换
               names_to="source",#列名放到source列中
               values_to="score")%>%#值放到score列中
  group_by(cluster,source)%>%#将数据按cluster和source两列进行分组
  summarise(mean=mean(score))


df=subset(df,df$source%in%gene)

#获得不同簇中活性差异较大的TFs
tfs<-df%>%
  group_by(source)%>%
  summarise(std=sd(mean))%>%
  arrange(-abs(std))%>%
  head(n_tfs)%>%
  pull(source)

#修改数据格式
top_acts_mat<-df%>%
  filter(source%in%tfs)%>%
  pivot_wider(id_cols='cluster',
              names_from='source',
              values_from='mean')%>%
  column_to_rownames('cluster')%>%
  as.matrix()

#绘图
palette_length=100
my_color=colorRampPalette(c("Darkblue","white","red"))(palette_length)

my_breaks<-c(seq(-3,0,length.out=ceiling(palette_length/2)+1),
             seq(0.05,3,length.out=floor(palette_length/2)))



#显示
save_pheatmap_pdf <- function(x, filename, width=6, height=2.5) {
  stopifnot(!missing(x))
  stopifnot(!missing(filename))
  pdf(filename, width=width, height=height)
  grid::grid.newpage()
  grid::grid.draw(x$gtable)
  dev.off()
}







xx=pheatmap(top_acts_mat,
            border_color=NA,
            color=my_color,
            breaks=my_breaks,cluster_rows = F,cluster_cols = F)




save_pheatmap_pdf(xx, "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/SCPA分析/绘图/P2.pdf")



#




#
###巨噬细胞所有转录因子差异----------
sce=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/Macr_allsingle.rds")
Idents(sce)=sce$type4

net<-get_collectri(organism='human',split_complexes=FALSE)
net


mat<-as.matrix(sce@assays$RNA@data)

mat=as.data.frame(mat)

mat<-as.matrix(mat)


acts<-run_ulm(mat,net,minsize=5,
              .source='source',.target='target',.mor='mor')
acts


sce[['tfsulm']]<-acts%>%
  pivot_wider(id_cols='source',
              names_from='condition',
              values_from='score')%>%
  column_to_rownames('source')%>%
  Seurat::CreateAssayObject(.)
#更改默认的分析对象
DefaultAssay(object=sce)<-"tfsulm"
#缩放数据并重新保存
sce<-ScaleData(sce)
sce@assays$tfsulm@data<-sce@assays$tfsulm@scale.data

DefaultAssay(object=sce)<-"tfsulm"


n_tfs<-25
#提取活性数据并修改数据格式
df<-t(as.matrix(sce@assays$tfsulm@data))%>%
  as.data.frame()%>%
  mutate(cluster=Idents(sce))%>%
  pivot_longer(cols=-cluster,#指定除cluster列以外的都需要转换
               names_to="source",#列名放到source列中
               values_to="score")%>%#值放到score列中
  group_by(cluster,source)%>%#将数据按cluster和source两列进行分组
  summarise(mean=mean(score))
#获得不同簇中活性差异较大的TFs
tfs<-df%>%
  group_by(source)%>%
  summarise(std=sd(mean))%>%
  arrange(-abs(std))%>%
  head(n_tfs)%>%
  pull(source)

#修改数据格式
top_acts_mat<-df%>%
  filter(source%in%tfs)%>%
  pivot_wider(id_cols='cluster',
              names_from='source',
              values_from='mean')%>%
  column_to_rownames('cluster')%>%
  as.matrix()

#绘图
palette_length=100
my_color=colorRampPalette(c("Darkblue","white","red"))(palette_length)

my_breaks<-c(seq(-3,0,length.out=ceiling(palette_length/2)+1),
             seq(0.05,3,length.out=floor(palette_length/2)))

#显示
pheatmap(top_acts_mat,
         border_color=NA,
         color=my_color,
         breaks=my_breaks,cluster_rows = F,cluster_cols = F)



#




#
###巨噬细胞代谢相关的转录因子差异----------
sce=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0718.rds")
Idents(sce)=sce$type4

path=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/SCPA分析/PATH12_2.csv",header = F,row.names = 1)

path=t(path)


path=as.data.frame(path)

gene=c(path$`AMPK signaling pathway`,path$`HIF-1 signaling pathway`,path$`Oxidative phosphorylation`,path$`Glycolysis / Gluconeogenesis`,
       path$`Pyruvate metabolism`,path$`Fatty acid degradation`,path$`Citrate cycle (TCA cycle)`,path$`Pentose phosphate pathway`,
       path$`Fatty acid elongation`,path$`Fatty acid biosynthesis`,path$`Glutathione metabolism`,path$`Alanine, aspartate and glutamate metabolism`,
       path$`Insulin signaling pathway`,path$`Tyrosine metabolism`,path$`Amino sugar and  nucleottde sugar metabolism`,
       path$`Cysteine and methionine metabolism`,path$`Glycerophospholipid metabolism`,
       path$`Glycine serine and threoinine metabolism`)


gene=gene[-which(gene=="")]

table(duplicated(gene))

gene<-gene[!duplicated(gene)]



net<-get_collectri(organism='human',split_complexes=FALSE)
net


mat<-as.matrix(sce@assays$RNA@data)

mat=as.data.frame(mat)






mat=mat[gene,]

mat<-as.matrix(mat)


acts<-run_ulm(mat,net,minsize=5,
              .source='source',.target='target',.mor='mor')
acts


sce[['tfsulm']]<-acts%>%
  pivot_wider(id_cols='source',
              names_from='condition',
              values_from='score')%>%
  column_to_rownames('source')%>%
  Seurat::CreateAssayObject(.)
#更改默认的分析对象
DefaultAssay(object=sce)<-"tfsulm"
#缩放数据并重新保存
sce<-ScaleData(sce)
sce@assays$tfsulm@data<-sce@assays$tfsulm@scale.data

DefaultAssay(object=sce)<-"tfsulm"


n_tfs<-25
#提取活性数据并修改数据格式
df<-t(as.matrix(sce@assays$tfsulm@data))%>%
  as.data.frame()%>%
  mutate(cluster=Idents(sce))%>%
  pivot_longer(cols=-cluster,#指定除cluster列以外的都需要转换
               names_to="source",#列名放到source列中
               values_to="score")%>%#值放到score列中
  group_by(cluster,source)%>%#将数据按cluster和source两列进行分组
  summarise(mean=mean(score))


df=subset(df,df$source%in%gene)

#获得不同簇中活性差异较大的TFs
tfs<-df%>%
  group_by(source)%>%
  summarise(std=sd(mean))%>%
  arrange(-abs(std))%>%
  head(n_tfs)%>%
  pull(source)

#修改数据格式
top_acts_mat<-df%>%
  filter(source%in%tfs)%>%
  pivot_wider(id_cols='cluster',
              names_from='source',
              values_from='mean')%>%
  column_to_rownames('cluster')%>%
  as.matrix()

#绘图
palette_length=100
my_color=colorRampPalette(c("Darkblue","white","red"))(palette_length)

my_breaks<-c(seq(-3,0,length.out=ceiling(palette_length/2)+1),
             seq(0.05,3,length.out=floor(palette_length/2)))

#显示
pheatmap(top_acts_mat,
         border_color=NA,
         color=my_color,
         breaks=my_breaks,cluster_rows = F,cluster_cols = F)



#




#



###对基质细胞进行monocle2伪时间分析并原图绘制-------
library(monocle)
EnSc=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0718.rds")

DefaultAssay(EnSc)<-"RNA"

EnSc_matrix<-as(as.matrix(EnSc@assays$RNA@counts),'sparseMatrix')

EnSc_ann<-data.frame(gene_id=rownames(EnSc_matrix),gene_short_name=rownames(EnSc_matrix))

rownames(EnSc_ann)<-rownames(EnSc_matrix)

EnSc_fd<-new("AnnotatedDataFrame", data = EnSc_ann)


sample_ann<-EnSc@meta.data
rownames(sample_ann)<-colnames(EnSc_matrix)

EnSc_pd<-new("AnnotatedDataFrame", data =sample_ann)

EnSc.cds<-newCellDataSet(EnSc_matrix,phenoData =EnSc_pd,
                         featureData =EnSc_fd,expressionFamily=negbinomial.size())


#预处理
EnSc.cds <- estimateSizeFactors(EnSc.cds)
EnSc.cds <- estimateDispersions(EnSc.cds)

EnSc.cds <- detectGenes(EnSc.cds, min_expr = 0.1)

disp_table <- dispersionTable(EnSc.cds)

unsup_clustering_genes <- subset(disp_table, mean_expression >= 0.1)

EnSc.cds <- setOrderingFilter(EnSc.cds, unsup_clustering_genes$gene_id)


EnSc.cds<- reduceDimension(EnSc.cds, max_components = 2, num_dim = 8, 
                           reduction_method = 'tSNE', verbose = T)

EnSc.cds<- clusterCells(EnSc.cds, num_clusters = 8)


expressed_genes <- row.names(subset(fData(EnSc.cds),
                                    num_cells_expressed >= 10))

clustering_DEG_genes <- differentialGeneTest(EnSc.cds[expressed_genes,],
                                             fullModelFormulaEnSc = "~ cell_type0718",cores = 8)

ordering_genes <- row.names(clustering_DEG_genes)[order(clustering_DEG_genes$qval)][1:1000] #排序后取前800个基因

EnSc.cds<- setOrderingFilter(EnSc.cds, ordering_genes)

#开始绘图用DDrtree 方法进行降维、画图
EnSc.cds <- reduceDimension(
  EnSc.cds,
  max_components = 2,
  method = 'DDRTree')

EnSc.cds <- orderCells(EnSc.cds)
head(pData(EnSc.cds))


EnSc_P1=plot_cell_trajectory(EnSc.cds,cell_size = 1,color_by = "cell_type0718",label=T)+ggtitle("")

EnSc_P2=plot_cell_trajectory(EnSc.cds,cell_size = 1,color_by = "Pseudotime")

EnSc_P3=plot_cell_trajectory(EnSc.cds,cell_size = 1,color_by = "cell_type0718",label=T)+ggtitle("")+
  facet_wrap(~cell_type0718, nrow = 1)

EnSc_P3
EnSc_P1+EnSc_P2


EnSc_P1=plot_cell_trajectory(EnSc.cds,cell_size = 1,color_by = "cell_type0305",label=T)+ggtitle("")

EnSc_P3=plot_cell_trajectory(EnSc.cds,cell_size = 1,color_by = "cell_type0305",label=T)+ggtitle("")+
  facet_wrap(~cell_type0305, nrow = 2)

saveRDS(EnSc.cds,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/20241024/EnSc.rds")










#进行6种细胞类型的衰老评分----
aging=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/衰老检测/衰老评分.csv")

##成纤维细胞衰老文章评分----
aging=aging$Gene.human.

all=AddModuleScore(object=all,features=list(aging))


all$type4=factor(all$type4,levels = c("Ctrl","Adjacent","Adhesion"))


VlnPlot(all,group.by = "cell_type3",features = "Cluster1",split.by = "type4",pt.size = 0)+
  ggtitle("aging score")



my_comparisons=list(c("Ctrl","Adhesion")) 

meta=all@meta.data

meta$group=meta$type4

ggplot()+
  geom_half_violin(
    data = meta %>% filter(group == "Ctrl"),
    aes(x = cell_type3,y = Cluster1),colour="white",fill="#1ba7b3",side = "l")+
  geom_half_violin(
    data = meta %>% filter(group == "Adhesion"),
    aes(x = cell_type3,y = Cluster1),colour="white",fill="#dfb424",side = "r")+
  theme_bw()+xlab("")+ylab("Early estrogen response")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,face = "bold",colour = "black",size =10 ),
        axis.text.y = element_text(face = "bold",colour = "black",size =10 ),
        axis.title.y =element_text(face = "bold",colour = "black",size =10 ))+
  stat_summary(data = meta, aes(x = cell_type3,y = Cluster1, fill = group),
               fun.min = function(x){quantile(x)[2]},
               fun.max = function(x){quantile(x)[4]},
               geom = 'errorbar', color='black',
               width=0.01,size=0.5,
               position = position_dodge(width = 0.2))+
  stat_compare_means(data = meta, aes(x = cell_type3,y = Cluster1, fill = group),
                     symnum.args=list(cutpoints = c(0, 0.001, 0.01, 0.05, 1),
                                      symbols = c("***", "**", "*", "-")),
                     label = "p.signif",
                     label.y = 0.4,
                     hide.ns = F)+ylim(0,0.4)






##子宫内膜衰老文章评分----
aging=c("CPEB1","DDB2","PTTG1","DHCR24","AR","CCND1","E2F1","RPS6KA6","ZMAT3",
        "EZH2","KCNJ12","VEGFA","ASPH","BMI1","BRCA1","TBX2")

all=AddModuleScore(object=all,features=list(aging))


all$type4=factor(all$type4,levels = c("Ctrl","Adjacent","Adhesion"))


VlnPlot(all,group.by = "cell_type3",features = "Cluster1",split.by = "type4",pt.size = 0)+
  ggtitle("aging score")



my_comparisons=list(c("Ctrl","Adhesion")) 

meta=all@meta.data

meta$group=meta$type4

ggplot()+
  geom_half_violin(
    data = meta %>% filter(group == "Ctrl"),
    aes(x = cell_type3,y = Cluster1),colour="white",fill="#1ba7b3",side = "l")+
  geom_half_violin(
    data = meta %>% filter(group == "Adhesion"),
    aes(x = cell_type3,y = Cluster1),colour="white",fill="#dfb424",side = "r")+
  theme_bw()+xlab("")+ylab("Aging Score")+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,face = "bold",colour = "black",size =10 ),
        axis.text.y = element_text(face = "bold",colour = "black",size =10 ),
        axis.title.y =element_text(face = "bold",colour = "black",size =10 ))+
  stat_summary(data = meta, aes(x = cell_type3,y = Cluster1, fill = group),
               fun.min = function(x){quantile(x)[2]},
               fun.max = function(x){quantile(x)[4]},
               geom = 'errorbar', color='black',
               width=0.01,size=0.5,
               position = position_dodge(width = 0.2))+
  stat_compare_means(data = meta, aes(x = cell_type3,y = Cluster1, fill = group),
                     symnum.args=list(cutpoints = c(0, 0.001, 0.01, 0.05, 1),
                                      symbols = c("***", "**", "*", "-")),
                     label = "p.signif",
                     label.y = 0.4,
                     hide.ns = F)+ylim(0,0.4)







#对富集通路进行汇总制图----
cluster0_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/S0_up_kegg.csv")
cluster0_go=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/S0_up_ego.csv")

cluster1_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/REV3L_up_kegg.csv")
cluster1_go=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/REV3L_up_ego.csv")

cluster2_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/S1_up_kegg.csv")
cluster2_go=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/S1_up_ego.csv")

cluster3_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/Cycling_up_kegg.csv")
cluster3_go=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/Cycling_up_ego.csv")

cluster4_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/BMP7_up_kegg.csv")
cluster4_go=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/BMP7_up_ego.csv")

cluster5_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/P_up_kegg.csv")
cluster5_go=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/P_up_ego.csv")

cluster6_kegg=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/ISG15_up_kegg.csv")
cluster6_go=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/KOBAS富集/ISG15_up_ego.csv")

term=c("Focal adhesion","Hippo signaling pathway","Estrogen signaling pathway",
       "Th17 cell differentiation","TGF-beta signaling pathway","extracellular matrix organization",
       "collagen-containing extracellular matrix","integrin binging","wound healing","collagen fibril organization")

cluster0_kegg=subset(cluster0_kegg,cluster0_kegg$X.Term%in%term)
cluster1_kegg=subset(cluster1_kegg,cluster1_kegg$X.Term%in%term)
cluster2_kegg=subset(cluster2_kegg,cluster2_kegg$X.Term%in%term)
cluster3_kegg=subset(cluster3_kegg,cluster3_kegg$X.Term%in%term)
cluster4_kegg=subset(cluster4_kegg,cluster4_kegg$X.Term%in%term)
cluster5_kegg=subset(cluster5_kegg,cluster5_kegg$X.Term%in%term)
cluster6_kegg=subset(cluster6_kegg,cluster6_kegg$X.Term%in%term)


cluster0_go=subset(cluster0_go,cluster0_go$X.Term%in%term)
cluster1_go=subset(cluster1_go,cluster1_go$X.Term%in%term)
cluster2_go=subset(cluster2_go,cluster2_go$X.Term%in%term)
cluster3_go=subset(cluster3_go,cluster3_go$X.Term%in%term)
cluster4_go=subset(cluster4_go,cluster4_go$X.Term%in%term)
cluster5_go=subset(cluster5_go,cluster5_go$X.Term%in%term)
cluster6_go=subset(cluster6_go,cluster6_go$X.Term%in%term)

cluster0=rbind(cluster0_kegg,cluster0_go)
cluster1=rbind(cluster1_kegg,cluster1_go)
cluster2=rbind(cluster2_kegg,cluster2_go)
cluster3=rbind(cluster3_kegg,cluster3_go)
cluster4=rbind(cluster4_kegg,cluster4_go)
cluster5=rbind(cluster5_kegg,cluster5_go)
cluster6=rbind(cluster6_kegg,cluster6_go)

cluster0$type="Cluster0"
cluster1$type="Cluster1"
cluster2$type="Cluster2"
cluster3$type="Cluster3"
cluster4$type="Cluster4"
cluster5$type="Cluster5"
cluster6$type="Cluster6"

plot2=rbind(cluster0,cluster1,cluster2,cluster3,cluster4,cluster5,cluster6)

ggplot(plot2)+aes(x=type,y=X.Term)+geom_point(aes(size=Input.number,color=-log10(P.Value)))+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,face = "bold",colour = "black",size =10 ),
        axis.text.y = element_text(face = "bold",colour = "black",size =10 ),
        axis.title.y =element_text(face = "bold",colour = "black",size =10 ),axis.title = element_text(face = "bold",colour = "black",size =10 ))+xlab("")+ylab("")+
  scale_color_gradient(low="white",high="red")+ggtitle("GO analysis/KEGG analysis of upregulated-DEGs")+labs(size="Count")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/2024/cellchat/20240625/20241018绘图/P1.pdf",width = 6.7,height = 4.5)




#将周医生测序大鼠数据里面造模中的CCL2和CCL5基因进行分析-----
#先提取出来FPKM这两个基因看看
gene3=c("ENSRNOG00000007159|Ccl2",
        "ENSRNOG00000010906|Ccl5")


A1=read.csv("/data/DATAprocess/chentw/ZZL_RNAseq/RAT/差异分析结果存放点/24h.ESC.ZMVSNormal.csv",row.names = 1)
A2=read.csv("/data/DATAprocess/chentw/ZZL_RNAseq/RAT/差异分析结果存放点/48h.ESC.ZMVSNormal.csv",row.names = 1)
A3=read.csv("/data/DATAprocess/chentw/ZZL_RNAseq/RAT/差异分析结果存放点/72h.ESC.ZMVSNormal.csv",row.names = 1)
A4=read.csv("/data/DATAprocess/chentw/ZZL_RNAseq/RAT/差异分析结果存放点/7d.ESC.ZMVSNormal.csv",row.names = 1)
A5=read.csv("/data/DATAprocess/chentw/ZZL_RNAseq/RAT/差异分析结果存放点/30d.MSC.ZMVSNormal.csv",row.names = 1)


A1=A1[gene3,]
A2=A2[gene3,]
A3=A3[gene3,]
A4=A4[gene3,]
A5=A5[gene3,]

A1$time="24h"
A2$time="48h"
A3$time="72h"
A4$time="7d"
A5$time="30d"


all3=rbind(A1,A2,A3,A4,A5)

all3$type2=rownames(all3)


all3$type2=gsub("21","2",all3$type2)
all3$type2=gsub("22","2",all3$type2)
all3$type2=gsub("23","2",all3$type2)
all3$type2=gsub("24","2",all3$type2)
all3$type2=gsub("51","5",all3$type2)
all3$type2=gsub("52","5",all3$type2)
all3$type2=gsub("53","5",all3$type2)
all3$type2=gsub("54","5",all3$type2)

ggplot(all3, aes(x=time, y=log2FoldChange, fill=type2))+
  geom_bar(position=position_dodge(), stat="identity")+
  facet_grid(.~all3$type2)+theme(panel.border = element_rect(color = "black", size = 1, fill = NA)) +
  theme_bw(base_size=20)+
  theme(axis.line=element_line(colour='black'),
        plot.title=element_text(hjust=0.5),
        legend.title=element_blank(),
        legend.position="bottom",
        legend.background=element_rect(fill='transparent'))+
  theme(axis.text.x=element_text(angle = 45,hjust = 1,vjust = 1,size="20",face = "bold",colour = "black"),
        axis.text.y=element_text(size="20",face = "bold",colour = "black"),
        axis.ticks.x = element_blank(),legend.text=element_text(size="20",face = "bold",colour = "black"),
        axis.title.y=element_text(size="20",face = "bold",colour = "black"),
        legend.position = "none")+xlab("")+
  ylab("% of total cells")+ scale_fill_manual(values=c("#006EBD","#E31E23","#007E30"))+
  scale_x_discrete(limits=c("24h","48h","72h","7d","30d"))+ylab("log2FC")+
  scale_y_continuous(expand=c(0,0),limits=c(0,5.5))

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/bar1.pdf",
       width = 8,height = 5)


#进行四类细胞类型的umap图比例绘制【补充1h的完善】----
#通过下面代码查看p值
library(ggpubr)
S1=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/1h/table_plot.csv")

my_comparisons=list(c("Ctrl","Adjacency"),c("Ctrl","Adhesion"))


ebtop<-function(x){
  return(mean(x)+sd(x)/sqrt(length(x)))
}
ebbottom<-function(x){
  return(mean(x)-sd(x)/sqrt(length(x)))
}



ggplot(data=S1,aes(x=group,y=ratio,fill=group))+
  stat_summary(geom = "bar",fun = "mean",
               position = position_dodge(0.9))+
  stat_summary(geom = "errorbar",
               fun.min = ebbottom,
               fun.max = ebtop,
               position = position_dodge(0.9),
               width=0.2)+facet_grid(.~Var1)+
  theme_classic()+
  theme(panel.border = element_rect(color = "black", size = 1, fill = NA)) +
  theme_bw(base_size=20)+
  theme(axis.line=element_line(colour='black'),
        plot.title=element_text(hjust=0.5),
        legend.title=element_blank(),
        legend.position="bottom",
        legend.background=element_rect(fill='transparent'))+
  theme(axis.text.x=element_blank(),
        axis.text.y=element_text(size="20",face = "bold",colour = "black"),
        axis.ticks.x = element_blank(),legend.text=element_text(size="20",face = "bold",colour = "black"),
        axis.title.y=element_text(size="20",face = "bold",colour = "black"))+xlab("")+
  ylab("% of total cells")+ scale_fill_manual(values=c("#006EBD","#E31E23","#007E30"))+
  stat_compare_means(comparisons = my_comparisons,
                     method = "t.test",label = "p.format")+ylim(0,120)

setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/")

umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")



umap_ctrl_IUA_allsingle$type2[umap_ctrl_IUA_allsingle$type2%in%"CTRL"]="Ctrl"
umap_ctrl_IUA_allsingle$type2[umap_ctrl_IUA_allsingle$type2%in%"IUA1"]="Adjacency"
umap_ctrl_IUA_allsingle$type2[umap_ctrl_IUA_allsingle$type2%in%"IUA2"]="Adhesion"


umap_ctrl_IUA_allsingle$type2=factor(umap_ctrl_IUA_allsingle$type2,
                                     levels = c("Ctrl","Adjacency","Adhesion"))
DimPlot(umap_ctrl_IUA_allsingle,group.by = "cell_type1110")

table_plot=as.data.frame(table(umap_ctrl_IUA_allsingle$type2,umap_ctrl_IUA_allsingle$cell_type1110))


ggplot(data=table_plot,fill=Var2)+aes(x=Var1,y=stat(table_plot$Freq),fill=Var2)+
  geom_bar(stat="identity",width=0.75, position="fill",color="black")+
  theme_bw()+theme(legend.title=element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank())+
  ylab("Integrative Celltype(%)")+
  scale_fill_brewer(palette="Paired")+xlab("")+
  scale_y_continuous(expand=c(0,0),limits=c(0,1),breaks = seq(0,1,0.25),labels = c("0","25","50","75","100"))+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,colour = "black",face = "bold",size = 22),
        axis.text.y = element_text(colour = "black",face = "bold",size = 22),
        axis.title.y = element_text(colour = "black",face = "bold",size = 22),
        legend.text = element_text(colour = "black",face = "bold",size = 22),
        panel.border = element_rect(color = "black", size = 1, fill = NA))+
  scale_fill_manual(values=c("#E37668","#88B532","#3BBEC0","#9F87AC"))+
  coord_flip()+ylab("")


ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/1h/h1.pdf",
       width = 15,height = 3)













#展示批次效应不高的结果根据review要求对图1e和S1c进行修改-----
umap_ctrl_IUA_allsingle$type4<-"type4"
umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type3%in%c("IUA1","IUA2","IUA5","IUA6")]<-"IUA-mild"
umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type3%in%c("IUA3","IUA4","IUA7")]<-"IUA-moderate"
umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type1%in%c("CTRL")]<-"CTRL"

umap_ctrl_IUA_allsingle$type5<-"type5"
umap_ctrl_IUA_allsingle$type5[umap_ctrl_IUA_allsingle$type1%in%"CTRL"]<- "Ctrl"
umap_ctrl_IUA_allsingle$type5[umap_ctrl_IUA_allsingle$type2%in%"Adjacency"&umap_ctrl_IUA_allsingle$type4%in%"IUA-mild"]<- "miIUA-a"
umap_ctrl_IUA_allsingle$type5[umap_ctrl_IUA_allsingle$type2%in%"Adhesion"&umap_ctrl_IUA_allsingle$type4%in%"IUA-mild"]<- "miIUA-I"
umap_ctrl_IUA_allsingle$type5[umap_ctrl_IUA_allsingle$type2%in%"Adjacency"&umap_ctrl_IUA_allsingle$type4%in%"IUA-moderate"]<- "moIUA-a"
umap_ctrl_IUA_allsingle$type5[umap_ctrl_IUA_allsingle$type2%in%"Adhesion"&umap_ctrl_IUA_allsingle$type4%in%"IUA-moderate"]<- "moIUA-I"


DimPlot(umap_ctrl_IUA_allsingle,split.by = "type3",label = F,label.size = 2,ncol = 5)+
  theme(title = element_text(size = 25),legend.key.size = unit(1,'cm'),legend.position = "none",
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),
        panel.border = element_blank(),axis.title = element_blank(),axis.text = element_blank(),
        axis.ticks = element_blank(),axis.line = element_blank())+ggtitle("")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/S1C/S1C.pdf",
       width =13,height = 6)




DimPlot(umap_ctrl_IUA_allsingle,split.by = "type5",label = F,label.size = 2,ncol = 5)+
  theme(title = element_text(size = 25),legend.key.size = unit(1,'cm'),legend.position = "none",
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),
        panel.border = element_blank(),axis.title = element_blank(),axis.text = element_blank(),
        axis.ticks = element_blank(),axis.line = element_blank())+ggtitle("")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/1E/1E.pdf",
       width =13,height = 3)






















#review1对Comment 4展示基质细胞亚群---------
STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")

STR$cell_type1227=paste("Cluster",STR$RNA_snn_res.0.3,sep = "")

DimPlot(STR,reduction="umap",label=T,label.size =7,pt.size = 0.9,group.by = "cell_type1227")+ggtitle("")+
  theme(title = element_text(size =0),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),axis.line = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 0))+theme(text = element_text(size=0))+
  scale_color_manual(values = c("#F87B73","#CD9600","#7CAE00","#00BE67","#00BFC4",
                                "#C981FF","#FF61CC","#00A9FF"))

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/补充1.pdf",width = 7,height = 7)



DimPlot(STR,reduction="umap",label=T,label.size =7,pt.size = 0.9,group.by = "cell_type1227",split.by = "type2")+ggtitle("")+
  theme(title = element_text(size =0),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),axis.line = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 0))+theme(text = element_text(size=0))+
  scale_color_manual(values = c("#F87B73","#CD9600","#7CAE00","#00BE67","#00BFC4",
                                "#C981FF","#FF61CC","#00A9FF"))

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/补充2.pdf",width = 21,height = 7)

DimPlot(STR,reduction="umap",label=T,label.size =7,pt.size = 0.9,group.by = "cell_type1227",split.by = "type3",
        ncol = 4)+ggtitle("")+
  theme(title = element_text(size =0),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),axis.line = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 0))+theme(text = element_text(size=0))+
  scale_color_manual(values = c("#F87B73","#CD9600","#7CAE00","#00BE67","#00BFC4",
                                "#C981FF","#FF61CC","#00A9FF"))

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/补充3.pdf",width = 28,height = 14)

#通过饼状图展现比例少
data=as.data.frame(table(STR$cell_type1227))

colnames(data)=c("category","value")
# 计算标签的坐标
data <- data %>%   arrange(desc(category)) %>%  mutate(prop = value / sum(data$value) *100) %>%  mutate(ypos = cumsum(prop)- 0.5*prop)

ggplot(data, aes(x = '', y = prop, fill = category)) +  
  geom_bar(stat = "identity", color='white', width = 1) +  
  scale_fill_brewer(palette = "RdYlBu") +   coord_polar("y", start = 0) +  theme_void() +  
  theme(legend.position = "right",        plot.title = element_text(hjust = 0.5)) +  
  geom_text_repel(aes(x=1.5, y=ypos,            label = paste0(round(prop), "%")),             
                  color = "black",             size = 4,            nudge_x = 0.15) +  
  labs(fill = "")+
  scale_fill_manual(values = c("#F87B73","#CD9600","#7CAE00","#00BE67","#00BFC4",
                               "#C981FF","#FF61CC","#00A9FF"))+
  theme(text = element_text(size=22,colour = "black",face = "bold"))

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/饼图1.pdf",width = 7,height = 7)



#对巨噬细胞的monocle结果进行绘图和调整补充-----
Macr=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/Macr_allsingle.rds")

macr=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/文件存放点/Macr2.rds")

plot_cell_trajectory(macr,cell_size = 1,color_by = "Pseudotime")

mygenes=c("IGF1","COL1A1","ITGA4","BTG1","MKI67","CENPF","GSN","NRP1")

plot_genes_in_pseudotime(macr[mygenes,], color_by = "cell_type0122",ncol = 2)

normalized_data <- as.data.frame(t(Macr@assays$RNA@data))


# 提取伪时间信息
pseudotime_data <- pData(macr)

mygenes=c("IGF1")
# 创建数据框，用于绘制图形
plot_data <- data.frame(
  pseudotime = pseudotime_data$Pseudotime,
  State=pseudotime_data$State,
  type3=pseudotime_data$type3,
  type2=pseudotime_data$type2,
  type1=pseudotime_data$type1,
  cell_type0122=pseudotime_data$cell_type0122,
  IGF1 = normalized_data$IGF1,
  COL1A1 = normalized_data$COL1A1,
  ITGA4 = normalized_data$ITGA4,
  BTG1 = normalized_data$BTG1,
  MKI67 = normalized_data$MKI67,
  CENPF = normalized_data$CENPF,
  GSN = normalized_data$GSN,
  NRP1 =normalized_data$NRP1)



# 使用 ggplot2 绘制基因表达随伪时间变化的图
library(ggplot2)


correlation <- cor(plot_data$pseudotime, plot_data$IGF1)

P1=ggplot(plot_data, aes(x = pseudotime, y = IGF1, color = cell_type0122)) + 
  geom_point(size=1) +  # 绘制散点
  geom_smooth(method = "loess", se = FALSE, color = "black") +theme_bw()+
  xlab("")+ylab("")+theme(text = element_text(size=12,face = "bold",colour = "black"))+
  scale_x_continuous(expand = c(0,0))+scale_y_continuous(expand = c(0,0))+
  scale_fill_manual(values = c("#F3786B","#AFA302","#08C23E","#0BB6BE","#62A2FC","#E578D0"))+
  theme(legend.position = "none")+
  stat_cor(method = "pearson", label.x = 1, label.y = 4,position = "identity")  

P2=ggplot(plot_data, aes(x = pseudotime, y = COL1A1, color = cell_type0122)) + 
  geom_point(size=1) +  # 绘制散点
  geom_smooth(method = "loess", se = FALSE, color = "black") +theme_bw()+
  xlab("")+ylab("")+theme(text = element_text(size=12,face = "bold",colour = "black"))+
  scale_x_continuous(expand = c(0,0))+scale_y_continuous(expand = c(0,0))+
  scale_fill_manual(values = c("#F3786B","#AFA302","#08C23E","#0BB6BE","#62A2FC","#E578D0"))+
  theme(legend.position = "none")

P3=ggplot(plot_data, aes(x = pseudotime, y = ITGA4, color = cell_type0122)) + 
  geom_point(size=1) +  # 绘制散点
  geom_smooth(method = "loess", se = FALSE, color = "black") +theme_bw()+
  xlab("")+ylab("")+theme(text = element_text(size=12,face = "bold",colour = "black"))+
  scale_x_continuous(expand = c(0,0))+scale_y_continuous(expand = c(0,0))+
  scale_fill_manual(values = c("#F3786B","#AFA302","#08C23E","#0BB6BE","#62A2FC","#E578D0"))+
  theme(legend.position = "none")

P4=ggplot(plot_data, aes(x = pseudotime, y = BTG1, color = cell_type0122)) + 
  geom_point(size=1) +  # 绘制散点
  geom_smooth(method = "loess", se = FALSE, color = "black") +theme_bw()+
  xlab("")+ylab("")+theme(text = element_text(size=12,face = "bold",colour = "black"))+
  scale_x_continuous(expand = c(0,0))+scale_y_continuous(expand = c(0,0))+
  scale_fill_manual(values = c("#F3786B","#AFA302","#08C23E","#0BB6BE","#62A2FC","#E578D0"))+
  theme(legend.position = "none")

P5=ggplot(plot_data, aes(x = pseudotime, y = MKI67, color = cell_type0122)) + 
  geom_point(size=1) +  # 绘制散点
  geom_smooth(method = "loess", se = FALSE, color = "black") +theme_bw()+
  xlab("")+ylab("")+theme(text = element_text(size=12,face = "bold",colour = "black"))+
  scale_x_continuous(expand = c(0,0))+scale_y_continuous(expand = c(0,0))+
  scale_fill_manual(values = c("#F3786B","#AFA302","#08C23E","#0BB6BE","#62A2FC","#E578D0"))+
  theme(legend.position = "none")

P6=ggplot(plot_data, aes(x = pseudotime, y = CENPF, color = cell_type0122)) + 
  geom_point(size=1) +  # 绘制散点
  geom_smooth(method = "loess", se = FALSE, color = "black") +theme_bw()+
  xlab("")+ylab("")+theme(text = element_text(size=12,face = "bold",colour = "black"))+
  scale_x_continuous(expand = c(0,0))+scale_y_continuous(expand = c(0,0))+
  scale_fill_manual(values = c("#F3786B","#AFA302","#08C23E","#0BB6BE","#62A2FC","#E578D0"))+
  theme(legend.position = "none")

P7=ggplot(plot_data, aes(x = pseudotime, y = GSN, color = cell_type0122)) + 
  geom_point(size=1) +  # 绘制散点
  geom_smooth(method = "loess", se = FALSE, color = "black") +theme_bw()+
  xlab("")+ylab("")+theme(text = element_text(size=12,face = "bold",colour = "black"))+
  scale_x_continuous(expand = c(0,0))+scale_y_continuous(expand = c(0,0))+
  scale_fill_manual(values = c("#F3786B","#AFA302","#08C23E","#0BB6BE","#62A2FC","#E578D0"))+
  theme(legend.position = "none")

P8=ggplot(plot_data, aes(x = pseudotime, y = NRP1, color = cell_type0122)) + 
  geom_point(size=1) +  # 绘制散点
  geom_smooth(method = "loess", se = FALSE, color = "black") +theme_bw()+
  xlab("")+ylab("")+theme(text = element_text(size=12,face = "bold",colour = "black"))+
  scale_x_continuous(expand = c(0,0))+scale_y_continuous(expand = c(0,0))+
  scale_fill_manual(values = c("#F3786B","#AFA302","#08C23E","#0BB6BE","#62A2FC","#E578D0"))+
  theme(legend.position = "none")


cowplot::plot_grid(P1,P2,P3,P4,P5,P6,P7,P8,ncol = 2)


ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/补充5.pdf",width = 6,height = 10)

cor(plot_data$pseudotime, plot_data$IGF1)
cor(plot_data$pseudotime, plot_data$COL1A1)
cor(plot_data$pseudotime, plot_data$ITGA4)
cor(plot_data$pseudotime, plot_data$BTG1)
cor(plot_data$pseudotime, plot_data$MKI67)
cor(plot_data$pseudotime, plot_data$CENPF)
cor(plot_data$pseudotime, plot_data$GSN)
cor(plot_data$pseudotime, plot_data$NRP1)



#查看CD14阳性巨噬细胞中S100A8的表达情况-----
Macr=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/Macr_allsingle.rds")

FeaturePlot(Macr,features = "CD14")


VlnPlot(Macr,features = "S100A8",group.by = "cell_type0610")

#提取CD14+的巨噬细胞
data2 <- as.data.frame(t(Macr@assays$RNA@data))

Macr$CD14_1=data2$CD14
Macr$S100A8_1=data2$S100A8


Macr2=subset(Macr@meta.data,Macr$CD14_1>1.5)

cell2=rownames(Macr2)

Macr2=Macr[,cell2]

Macr2$cell_type3="CD14+"

VlnPlot(Macr2,features = "S100A8",group.by = "cell_type3")




#修改图2C,2E颜色变成一致-----
###2C
EnSc=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0718.rds")
EnSc$RNA_snn_res.0.3=factor(EnSc$RNA_snn_res.0.3)

table_plot=as.data.frame(table(EnSc$type3,EnSc$RNA_snn_res.0.3))

ggplot(data=table_plot,fill=Var2)+aes(x=Var1,y=stat(table_plot$Freq),fill=Var2)+geom_bar(stat="identity",width=0.75, position="fill",color="black")+
  theme_bw()+theme(legend.title=element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank())+ylab("Integrative Celltype(%)")+scale_fill_brewer(palette="Paired")+xlab("")+
  scale_y_continuous(expand=c(0,0),limits=c(0,1),breaks = seq(0,1,0.25),labels = c("0","25","50","75","100"))+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,colour = "black",face = "bold",size = 18),
        axis.text.y = element_text(colour = "black",face = "bold",size = 18),
        axis.title.y = element_text(colour = "black",face = "bold",size = 18),
        legend.text = element_text(colour = "black",face = "bold",size = 18),
        panel.border = element_rect(color = "black", size = 1.5, fill = NA))+
  scale_fill_manual(values=c("#F87B73","#CD9600","#7CAE00","#00BE67","#00BFC4",
                             "#C981FF","#FF61CC","#00A9FF"))+
  scale_x_discrete(labels=c("P1","P2","P3","P4","P5","P6","P7","P8"))+
  ylab("Cluster proportion(%)")


ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/2C.pdf",width =5,height = 7)


table_plot=as.data.frame(table(EnSc$type2,EnSc$RNA_snn_res.0.3))

ggplot(data=table_plot,fill=Var2)+aes(x=Var1,y=stat(table_plot$Freq),fill=Var2)+geom_bar(stat="identity",width=0.75, position="fill",color="black")+
  theme_bw()+theme(legend.title=element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank())+ylab("Integrative Celltype(%)")+scale_fill_brewer(palette="Paired")+xlab("")+
  scale_y_continuous(expand=c(0,0),limits=c(0,1),breaks = seq(0,1,0.25),labels = c("0","25","50","75","100"))+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,colour = "black",face = "bold",size = 18),
        axis.text.y = element_text(colour = "black",face = "bold",size = 18),
        axis.title.y = element_text(colour = "black",face = "bold",size = 18),
        legend.text = element_text(colour = "black",face = "bold",size = 18),
        panel.border = element_rect(color = "black", size = 1.5, fill = NA))+
  scale_fill_manual(values=c("#F87B73","#CD9600","#7CAE00","#00BE67","#00BFC4",
                             "#C981FF","#FF61CC","#00A9FF"))+
  scale_x_discrete(labels=c("Ctrl","Adjacency","Adhesion"))+
  ylab("Cluster proportion(%)")


ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/2E.pdf",width =4,height = 7)



##2B绘图
EnSc=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0718.rds")


DimPlot(EnSc,reduction="umap",label=F,label.size =7,pt.size = 0.9,group.by = "type3")+ggtitle("")+
  theme(title = element_text(size =0),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),axis.line = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 0))+theme(text = element_text(size=0))+
  scale_color_manual(values=c("#DB0D13","#51A23E","#974E90","#EE7D00","#FDE015","#87181C",
                              "#387FBA","#3F2217"))

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/2B_1.pdf",width = 7,height = 7)


DimPlot(EnSc,reduction="umap",label=F,label.size =7,pt.size = 0.9,group.by = "type3")+ggtitle("")+
  theme(title = element_text(size =0),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 8),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),axis.line = element_blank(),
        plot.subtitle = element_text(size = 0))+theme(text = element_text(size=0))+
  scale_color_manual(values=c("#DB0D13","#51A23E","#974E90","#EE7D00","#FDE015","#87181C",
                              "#387FBA","#3F2217"))

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/2B_1.pdf",width = 7,height = 7)









#对基质细胞的cluster0-6进行差异基因查找------
EnSc=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0718.rds")

EnSc$type6=paste(EnSc$type4,"_",EnSc$cell_type0718,sep = "")

#Cluster0
Cluster0_deseq=FindMarkers(EnSc,ident.1 = "Adhesion_Cluster0",ident.2 = "Ctrl_Cluster0",
                           min.pct = 0.2,group.by = "type6")
Cluster0_deseq=
  subset(Cluster0_deseq,Cluster0_deseq$p_val<=0.05)
write.csv(Cluster0_deseq,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/基质细胞差异基因/Cluster0_deseq.csv")

#Cluster1
Cluster1_deseq=FindMarkers(EnSc,ident.1 = "Adhesion_Cluster1",ident.2 = "Ctrl_Cluster1",
                           min.pct = 0.2,group.by = "type6")
Cluster1_deseq=subset(Cluster1_deseq,Cluster1_deseq$p_val<=0.05)
write.csv(Cluster1_deseq,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/基质细胞差异基因/Cluster1_deseq.csv")

#Cluster2
Cluster2_deseq=FindMarkers(EnSc,ident.1 = "Adhesion_Cluster2",ident.2 = "Ctrl_Cluster2",
                           min.pct = 0.2,group.by = "type6")
Cluster2_deseq=subset(Cluster2_deseq,Cluster2_deseq$p_val<=0.05)
write.csv(Cluster2_deseq,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/基质细胞差异基因/Cluster2_deseq.csv")

#Cluster3
Cluster3_deseq=FindMarkers(EnSc,ident.1 = "Adhesion_Cluster3",ident.2 = "Ctrl_Cluster3",
                           min.pct = 0.2,group.by = "type6")
Cluster3_deseq=subset(Cluster3_deseq,Cluster3_deseq$p_val<=0.05)
write.csv(Cluster3_deseq,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/基质细胞差异基因/Cluster3_deseq.csv")

#Cluster4
Cluster4_deseq=FindMarkers(EnSc,ident.1 = "Adhesion_Cluster4",ident.2 = "Ctrl_Cluster4",
                           min.pct = 0.2,group.by = "type6")
Cluster4_deseq=subset(Cluster4_deseq,Cluster4_deseq$p_val<=0.05)
write.csv(Cluster4_deseq,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/基质细胞差异基因/Cluster4_deseq.csv")



#Cluster5
Cluster5_deseq=FindMarkers(EnSc,ident.1 = "Adhesion_Cluster5",ident.2 = "Ctrl_Cluster5",
                           min.pct = 0.2,group.by = "type6")
Cluster5_deseq=subset(Cluster5_deseq,Cluster5_deseq$p_val<=0.05)
write.csv(Cluster5_deseq,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/基质细胞差异基因/Cluster5_deseq.csv")



#Cluster6
Cluster6_deseq=FindMarkers(EnSc,ident.1 = "Adhesion_Cluster6",ident.2 = "Ctrl_Cluster6",
                           min.pct = 0.2,group.by = "type6")
Cluster6_deseq=subset(Cluster6_deseq,Cluster6_deseq$p_val<=0.05)
write.csv(Cluster6_deseq,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/基质细胞差异基因/Cluster6_deseq.csv")







#修改图2g颜色修改[重新更改基质细胞upset图]-----
#2G_1
###上调
Cluster0_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/基质细胞差异基因/Cluster0_deseq.csv")
Cluster0=subset(Cluster0_deseq,Cluster0_deseq$avg_log2FC>0)$X

Cluster1_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/基质细胞差异基因/Cluster1_deseq.csv")
Cluster1=subset(Cluster1_deseq,Cluster1_deseq$avg_log2FC>0)$X

Cluster2_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/基质细胞差异基因/Cluster2_deseq.csv")
Cluster2=subset(Cluster2_deseq,Cluster2_deseq$avg_log2FC>0)$X

Cluster3_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/基质细胞差异基因/Cluster3_deseq.csv")
Cluster3=subset(Cluster3_deseq,Cluster3_deseq$avg_log2FC>0)$X

Cluster4_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/基质细胞差异基因/Cluster4_deseq.csv")
Cluster4=subset(Cluster4_deseq,Cluster4_deseq$avg_log2FC>0)$X

Cluster5_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/基质细胞差异基因/Cluster5_deseq.csv")
Cluster5=subset(Cluster5_deseq,Cluster5_deseq$avg_log2FC>0)$X

Cluster6_deseq=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/基质细胞差异基因/Cluster6_deseq.csv")
Cluster6=subset(Cluster6_deseq,Cluster6_deseq$avg_log2FC>0)$X


set_list <- list("Cluster0"=Cluster0,"Cluster1"=Cluster1,"Cluster2"=Cluster2,"Cluster3"=Cluster3,
                 "Cluster4"=Cluster4,"Cluster5"=Cluster5,"Cluster6"=Cluster6)

pdf("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/2G_1.pdf",
    width = 10,height = 7)
upset(
  fromList(set_list),
  order.by = "freq",nsets = 8,shade.color="red",
  sets.bar.color=c("#F87B73","#7CAE00","#00BFC4","#CD9600","#FF61CC","#00BE67","#C981FF"),
  sets.x.label="UP genes",mainbar.y.label="No. of up genes per cell type"
)

dev.off()





#2G_2
G=Reduce(intersect,list(Cluster0,Cluster1,Cluster2,Cluster3,Cluster4,Cluster5,Cluster6))

EnSc=AddModuleScore(object=EnSc,features=list(G))

my_comparisons=list(c("CTRL","IUA1"),c("CTRL","IUA2"))


VlnPlot(EnSc,features = "Cluster1",group.by = "type2",pt.size = 0)+
  facet_grid(.~EnSc$cell_type0718)+
  theme_classic()+scale_y_continuous(limits=c(-0.34,2.2))+
  theme(panel.border = element_rect(color = "black", size = 1, fill = NA)) +
  theme_bw(base_size=20)+
  theme(axis.line=element_line(colour='black'),
        plot.title=element_text(hjust=0.5),
        legend.title=element_blank(),
        legend.position="bottom",
        legend.background=element_rect(fill='transparent'))+
  theme(axis.text.x=element_blank(),
        axis.text.y=element_text(size="20",face = "bold",colour = "black"),
        axis.ticks.x = element_blank(),
        legend.text=element_text(size="20",face = "bold",colour = "black"),
        axis.title.y=element_text(size="20",face = "bold",colour = "black"))+
  xlab("")+
  scale_fill_manual(values = c("#F57769","#06B63A","#619DFF"))+stat_compare_means(comparisons = my_comparisons,tip.length = 0.03,
                                                                                  bracket.size = 1.3,hide.ns = F)+ggtitle("")


ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/2G_2.pdf",
       width = 12,height = 7)









#对巨噬细胞进行图3颜色修改-----
###3C
Macr=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/Macr_allsingle.rds")


Macr$cell_type0610=factor(Macr$cell_type0610,levels = c("CD163+","CD24+","CD301+",
                                                        "Monocyte+","Proliferating",
                                                        "Uknow"))

Macr$cell_type0610=gsub("Monocyte+","Monocyte",Macr$cell_type0610)
Macr$cell_type0610=gsub("Uknow","Unknow",Macr$cell_type0610)


DimPlot(Macr,reduction="umap",label=T,label.size =7,pt.size = 0.9,group.by = "cell_type0610")+ggtitle("")+
  theme(title = element_text(size =0),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),axis.line = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 0))+theme(text = element_text(size=0))+
  scale_color_manual(values = c("#F3786B","#AFA302","#08C23E","#0BB6BE","#62A2FC","#E578D0"))

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/3C.pdf",width = 7,height = 7)


#3f
macr=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/monocle/文件存放点/Macr2.rds")



macr$type4=gsub("Adjacent","Adjacency",macr$type4)

macr$type4=factor(macr$type4,levels = c("Ctrl","Adjacency","Adhesion"))

plot_cell_trajectory(macr,cell_size = 1,color_by = "type4",
                     label=T)+ggtitle("")+facet_wrap(~type4)+
  theme(legend.position = "none"
        ,axis.text.x = element_text(colour = "black",face = "bold",size = 18),
        axis.text.y = element_text(colour = "black",face = "bold",size = 18),
        axis.title.y = element_text(colour = "black",face = "bold",size = 18),
        axis.title.x = element_text(colour = "black",face = "bold",size = 18),
        legend.text = element_text(colour = "black",face = "bold",size = 18),
        plot.subtitle = element_text(size = 0),panel.border = element_blank())

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/3F.pdf",width = 9,height = 6)


#3B
library(ggpubr)
S1=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/3B/table_plot.csv")

my_comparisons=list(c("Ctrl","Adjacency"),c("Ctrl","Adhesion"),c("Adjacency","Adhesion"))


ebtop<-function(x){
  return(mean(x)+sd(x)/sqrt(length(x)))
}
ebbottom<-function(x){
  return(mean(x)-sd(x)/sqrt(length(x)))
}


S1$group=factor(S1$group,levels = c("Ctrl","Adjacency","Adhesion"))



ggplot(data=S1,aes(x=group,y=ratio,fill=group),color="black")+
  stat_summary(geom = "bar",fun = "mean",color="black",
               position = position_dodge(0.9))+
  stat_summary(geom = "errorbar",
               fun.min = ebbottom,
               fun.max = ebtop,
               position = position_dodge(0.9),
               width=0.2)+facet_grid(.~Var2)+
  theme_classic()+
  theme(panel.border = element_rect(color = "black", size = 1, fill = NA)) +
  theme_bw(base_size=20)+
  theme(axis.line=element_line(colour='black'),
        plot.title=element_text(hjust=0.5),
        legend.title=element_blank(),
        legend.position="bottom",
        legend.background=element_rect(fill='transparent'))+
  theme(axis.text.x=element_blank(),
        axis.text.y=element_text(size="20",face = "bold",colour = "black"),
        axis.ticks.x = element_blank(),legend.text=element_text(size="20",face = "bold",colour = "black"),
        axis.title.y=element_text(size="20",face = "bold",colour = "black"))+xlab("")+
  ylab("% of total cells")+ scale_fill_manual(values=c("#F57769","#06B63A","#619DFF"))+
  stat_compare_means(comparisons = my_comparisons,label.y = c(70,75,85),
                     method = "t.test",label = "p.format")+scale_y_continuous(expand=c(0,0),limits=c(0,95))

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/3B.pdf",width = 10,height = 7)



#对S1D的比例饼图进行修改-----
umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")


#通过饼状图展现比例少

# 计算标签的坐标
umap_ctrl_IUA_allsingle$cell_type1110=gsub("cells","",umap_ctrl_IUA_allsingle$cell_type1110)

umap_ctrl_IUA_allsingle$cell_type1110_2=umap_ctrl_IUA_allsingle$cell_type1110

umap_ctrl_IUA_allsingle$cell_type1110=factor(umap_ctrl_IUA_allsingle$cell_type1110,
                                             levels = c("Stromal ","Immune ","Epithelium ","Endothelium "))


#Ctrl组
data1=as.data.frame(table(subset(umap_ctrl_IUA_allsingle,type1%in%"CTRL")$cell_type1110))
colnames(data1)=c("category","value")
data1 <- data1 %>%   arrange(desc(category)) %>%  mutate(prop = value / sum(data1$value) *100) %>%  mutate(ypos = cumsum(prop)- 0.5*prop)

ggplot(data1, aes(x = '', y = prop, fill = category)) +  
  geom_bar(stat = "identity", color='black', width = 1) +  
  scale_fill_brewer(palette = "RdYlBu") +   coord_polar("y", start = 0) +  theme_void() +  
  theme(legend.position = "right",        plot.title = element_text(hjust = 0.5)) +  
  labs(fill = "")+
  scale_fill_manual(values = c("#E77168","#7AA92C","#23B0B2","#A082B2"))+
  theme(text = element_text(size=22,colour = "black",face = "bold"))

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/S1D/Ctrl.pdf",width = 7,height = 7)


#Adjacency组
data2=as.data.frame(table(subset(umap_ctrl_IUA_allsingle,type2%in%"IUA1")$cell_type1110))
colnames(data2)=c("category","value")
data2 <- data2 %>%   arrange(desc(category)) %>%  mutate(prop = value / sum(data2$value) *100) %>%  mutate(ypos = cumsum(prop)- 0.5*prop)
ggplot(data2, aes(x = '', y = prop, fill = category)) +  
  geom_bar(stat = "identity", color='black', width = 1) +  
  scale_fill_brewer(palette = "RdYlBu") +   coord_polar("y", start = 0) +  theme_void() +  
  theme(legend.position = "right",        plot.title = element_text(hjust = 0.5)) +  
  labs(fill = "")+
  scale_fill_manual(values = c("#E77168","#7AA92C","#23B0B2","#A082B2"))+
  theme(text = element_text(size=22,colour = "black",face = "bold"))

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/S1D/Adjacency.pdf",width = 7,height = 7)



#Adhesion组
data2=as.data.frame(table(subset(umap_ctrl_IUA_allsingle,type2%in%"IUA2")$cell_type1110))
colnames(data2)=c("category","value")
data2 <- data2 %>%   arrange(desc(category)) %>%  mutate(prop = value / sum(data2$value) *100) %>%  mutate(ypos = cumsum(prop)- 0.5*prop)
ggplot(data2, aes(x = '', y = prop, fill = category)) +  
  geom_bar(stat = "identity", color='black', width = 1) +  
  scale_fill_brewer(palette = "RdYlBu") +   coord_polar("y", start = 0) +  theme_void() +  
  theme(legend.position = "right",        plot.title = element_text(hjust = 0.5)) +  
  labs(fill = "")+
  scale_fill_manual(values = c("#E77168","#7AA92C","#23B0B2","#A082B2"))+
  theme(text = element_text(size=22,colour = "black",face = "bold"))

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/S1D/Adhesion.pdf",width = 7,height = 7)

umap_ctrl_IUA_allsingle$type4='Ctrl'

umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type3%in%c("IUA1","IUA2",
                                                                 "IUA5","IUA6")]<-"mi-IUA"

umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type3%in%c("IUA3",
                                                                 "IUA4","IUA7")]<-"mo-IUA"


#mi-IUA组
data2=as.data.frame(table(subset(umap_ctrl_IUA_allsingle,type4%in%"mi-IUA")$cell_type1110))
colnames(data2)=c("category","value")
data2 <- data2 %>%   arrange(desc(category)) %>%  mutate(prop = value / sum(data2$value) *100) %>%  mutate(ypos = cumsum(prop)- 0.5*prop)
ggplot(data2, aes(x = '', y = prop, fill = category)) +  
  geom_bar(stat = "identity", color='black', width = 1) +  
  scale_fill_brewer(palette = "RdYlBu") +   coord_polar("y", start = 0) +  theme_void() +  
  theme(legend.position = "right",        plot.title = element_text(hjust = 0.5)) +  
  labs(fill = "")+
  scale_fill_manual(values = c("#E77168","#7AA92C","#23B0B2","#A082B2"))+
  theme(text = element_text(size=22,colour = "black",face = "bold"))

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/S1D/mi-IUA.pdf",width = 7,height = 7)




#mo-IUA组
data2=as.data.frame(table(subset(umap_ctrl_IUA_allsingle,type4%in%"mo-IUA")$cell_type1110))
colnames(data2)=c("category","value")
data2 <- data2 %>%   arrange(desc(category)) %>%  mutate(prop = value / sum(data2$value) *100) %>%  mutate(ypos = cumsum(prop)- 0.5*prop)
ggplot(data2, aes(x = '', y = prop, fill = category)) +  
  geom_bar(stat = "identity", color='black', width = 1) +  
  scale_fill_brewer(palette = "RdYlBu") +   coord_polar("y", start = 0) +  theme_void() +  
  theme(legend.position = "right",        plot.title = element_text(hjust = 0.5)) +  
  labs(fill = "")+
  scale_fill_manual(values = c("#E77168","#7AA92C","#23B0B2","#A082B2"))+
  theme(text = element_text(size=22,colour = "black",face = "bold"))

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/S1D/mo-IUA.pdf",width = 7,height = 7)




#mo-IUA组
data2=as.data.frame(table(subset(umap_ctrl_IUA_allsingle,type4%in%"mo-IUA")$cell_type1110))
colnames(data2)=c("category","value")
data2 <- data2 %>%   arrange(desc(category)) %>%  mutate(prop = value / sum(data2$value) *100) %>%  mutate(ypos = cumsum(prop)- 0.5*prop)
ggplot(data2, aes(x = '', y = prop, fill = category)) +  
  geom_bar(stat = "identity", color='black', width = 1) +  
  scale_fill_brewer(palette = "RdYlBu") +   coord_polar("y", start = 0) +  theme_void() +  
  theme(legend.position = "right",        plot.title = element_text(hjust = 0.5)) +  
  labs(fill = "")+
  scale_fill_manual(values = c("#E77168","#7AA92C","#23B0B2","#A082B2"))+
  theme(text = element_text(size=22,colour = "black",face = "bold"))

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/S1D/mo-IUA.pdf",width = 7,height = 7)

#CTRL1组
data2=as.data.frame(table(subset(umap_ctrl_IUA_allsingle,type3%in%"CTRL1")$cell_type1110))
colnames(data2)=c("category","value")
data2 <- data2 %>%   arrange(desc(category)) %>%  mutate(prop = value / sum(data2$value) *100) %>%  mutate(ypos = cumsum(prop)- 0.5*prop)
ggplot(data2, aes(x = '', y = prop, fill = category)) +  
  geom_bar(stat = "identity", color='black', width = 1) +  
  scale_fill_brewer(palette = "RdYlBu") +   coord_polar("y", start = 0) +  theme_void() +  
  theme(legend.position = "right",        plot.title = element_text(hjust = 0.5)) +  
  labs(fill = "")+
  scale_fill_manual(values = c("#E77168","#7AA92C","#23B0B2","#A082B2"))+
  theme(text = element_text(size=22,colour = "black",face = "bold"))

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/S1D/CTRL1.pdf",width = 7,height = 7)



#CTRL2组
data2=as.data.frame(table(subset(umap_ctrl_IUA_allsingle,type3%in%"CTRL2")$cell_type1110))
colnames(data2)=c("category","value")
data2 <- data2 %>%   arrange(desc(category)) %>%  mutate(prop = value / sum(data2$value) *100) %>%  mutate(ypos = cumsum(prop)- 0.5*prop)
ggplot(data2, aes(x = '', y = prop, fill = category)) +  
  geom_bar(stat = "identity", color='black', width = 1) +  
  scale_fill_brewer(palette = "RdYlBu") +   coord_polar("y", start = 0) +  theme_void() +  
  theme(legend.position = "right",        plot.title = element_text(hjust = 0.5)) +  
  labs(fill = "")+
  scale_fill_manual(values = c("#E77168","#7AA92C","#23B0B2","#A082B2"))+
  theme(text = element_text(size=22,colour = "black",face = "bold"))

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/S1D/CTRL2.pdf",width = 7,height = 7)


#IUA1组
data2=as.data.frame(table(subset(umap_ctrl_IUA_allsingle,type3%in%"IUA1")$cell_type1110))
colnames(data2)=c("category","value")
data2 <- data2 %>%   arrange(desc(category)) %>%  mutate(prop = value / sum(data2$value) *100) %>%  mutate(ypos = cumsum(prop)- 0.5*prop)
ggplot(data2, aes(x = '', y = prop, fill = category)) +  
  geom_bar(stat = "identity", color='black', width = 1) +  
  scale_fill_brewer(palette = "RdYlBu") +   coord_polar("y", start = 0) +  theme_void() +  
  theme(legend.position = "right",        plot.title = element_text(hjust = 0.5)) +  
  labs(fill = "")+
  scale_fill_manual(values = c("#E77168","#7AA92C","#23B0B2","#A082B2"))+
  theme(text = element_text(size=22,colour = "black",face = "bold"))

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/S1D/IUA1.pdf",width = 7,height = 7)

#IUA2组
data2=as.data.frame(table(subset(umap_ctrl_IUA_allsingle,type3%in%"IUA2")$cell_type1110))
colnames(data2)=c("category","value")
data2 <- data2 %>%   arrange(desc(category)) %>%  mutate(prop = value / sum(data2$value) *100) %>%  mutate(ypos = cumsum(prop)- 0.5*prop)
ggplot(data2, aes(x = '', y = prop, fill = category)) +  
  geom_bar(stat = "identity", color='black', width = 1) +  
  scale_fill_brewer(palette = "RdYlBu") +   coord_polar("y", start = 0) +  theme_void() +  
  theme(legend.position = "right",        plot.title = element_text(hjust = 0.5)) +  
  labs(fill = "")+
  scale_fill_manual(values = c("#E77168","#7AA92C","#23B0B2","#A082B2"))+
  theme(text = element_text(size=22,colour = "black",face = "bold"))

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/S1D/IUA2.pdf",width = 7,height = 7)


#IUA3组
data2=as.data.frame(table(subset(umap_ctrl_IUA_allsingle,type3%in%"IUA3")$cell_type1110))
colnames(data2)=c("category","value")
data2 <- data2 %>%   arrange(desc(category)) %>%  mutate(prop = value / sum(data2$value) *100) %>%  mutate(ypos = cumsum(prop)- 0.5*prop)
ggplot(data2, aes(x = '', y = prop, fill = category)) +  
  geom_bar(stat = "identity", color='black', width = 1) +  
  scale_fill_brewer(palette = "RdYlBu") +   coord_polar("y", start = 0) +  theme_void() +  
  theme(legend.position = "right",        plot.title = element_text(hjust = 0.5)) +  
  labs(fill = "")+
  scale_fill_manual(values = c("#E77168","#7AA92C","#23B0B2","#A082B2"))+
  theme(text = element_text(size=22,colour = "black",face = "bold"))

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/S1D/IUA3.pdf",width = 7,height = 7)

#IUA4组
data2=as.data.frame(table(subset(umap_ctrl_IUA_allsingle,type3%in%"IUA4")$cell_type1110))
colnames(data2)=c("category","value")
data2 <- data2 %>%   arrange(desc(category)) %>%  mutate(prop = value / sum(data2$value) *100) %>%  mutate(ypos = cumsum(prop)- 0.5*prop)
ggplot(data2, aes(x = '', y = prop, fill = category)) +  
  geom_bar(stat = "identity", color='black', width = 1) +  
  scale_fill_brewer(palette = "RdYlBu") +   coord_polar("y", start = 0) +  theme_void() +  
  theme(legend.position = "right",        plot.title = element_text(hjust = 0.5)) +  
  labs(fill = "")+
  scale_fill_manual(values = c("#E77168","#7AA92C","#23B0B2","#A082B2"))+
  theme(text = element_text(size=22,colour = "black",face = "bold"))

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/S1D/IUA4.pdf",width = 7,height = 7)


#IUA5组
data2=as.data.frame(table(subset(umap_ctrl_IUA_allsingle,type3%in%"IUA5")$cell_type1110))
colnames(data2)=c("category","value")
data2 <- data2 %>%   arrange(desc(category)) %>%  mutate(prop = value / sum(data2$value) *100) %>%  mutate(ypos = cumsum(prop)- 0.5*prop)
ggplot(data2, aes(x = '', y = prop, fill = category)) +  
  geom_bar(stat = "identity", color='black', width = 1) +  
  scale_fill_brewer(palette = "RdYlBu") +   coord_polar("y", start = 0) +  theme_void() +  
  theme(legend.position = "right",        plot.title = element_text(hjust = 0.5)) +  
  labs(fill = "")+
  scale_fill_manual(values = c("#E77168","#7AA92C","#23B0B2","#A082B2"))+
  theme(text = element_text(size=22,colour = "black",face = "bold"))

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/S1D/IUA5.pdf",width = 7,height = 7)




#IUA6组
data2=as.data.frame(table(subset(umap_ctrl_IUA_allsingle,type3%in%"IUA6")$cell_type1110))
colnames(data2)=c("category","value")
data2 <- data2 %>%   arrange(desc(category)) %>%  mutate(prop = value / sum(data2$value) *100) %>%  mutate(ypos = cumsum(prop)- 0.5*prop)
ggplot(data2, aes(x = '', y = prop, fill = category)) +  
  geom_bar(stat = "identity", color='black', width = 1) +  
  scale_fill_brewer(palette = "RdYlBu") +   coord_polar("y", start = 0) +  theme_void() +  
  theme(legend.position = "right",        plot.title = element_text(hjust = 0.5)) +  
  labs(fill = "")+
  scale_fill_manual(values = c("#E77168","#7AA92C","#23B0B2","#A082B2"))+
  theme(text = element_text(size=22,colour = "black",face = "bold"))

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/S1D/IUA6.pdf",width = 7,height = 7)





#IUA7组
data2=as.data.frame(table(subset(umap_ctrl_IUA_allsingle,type3%in%"IUA7")$cell_type1110))
colnames(data2)=c("category","value")
data2 <- data2 %>%   arrange(desc(category)) %>%  mutate(prop = value / sum(data2$value) *100) %>%  mutate(ypos = cumsum(prop)- 0.5*prop)
ggplot(data2, aes(x = '', y = prop, fill = category)) +  
  geom_bar(stat = "identity", color='black', width = 1) +  
  scale_fill_brewer(palette = "RdYlBu") +   coord_polar("y", start = 0) +  theme_void() +  
  theme(legend.position = "right",        plot.title = element_text(hjust = 0.5)) +  
  labs(fill = "")+
  scale_fill_manual(values = c("#E77168","#7AA92C","#23B0B2","#A082B2"))+
  theme(text = element_text(size=22,colour = "black",face = "bold"))

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/S1D/IUA7.pdf",width = 7,height = 7)





#对S1D的比例饼图进行修改[加上比例统计数字]-----
umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")


#通过饼状图展现比例少

# 计算标签的坐标
umap_ctrl_IUA_allsingle$cell_type1110=gsub("cells","",umap_ctrl_IUA_allsingle$cell_type1110)

umap_ctrl_IUA_allsingle$cell_type1110_2=umap_ctrl_IUA_allsingle$cell_type1110

umap_ctrl_IUA_allsingle$cell_type1110=factor(umap_ctrl_IUA_allsingle$cell_type1110,
                                             levels = c("Stromal ","Immune ","Epithelium ","Endothelium "))


#Ctrl组
data1=as.data.frame(table(subset(umap_ctrl_IUA_allsingle,type1%in%"CTRL")$cell_type1110))
colnames(data1)=c("category","value")
data1 <- data1 %>%   arrange(desc(category)) %>%  mutate(prop = value / sum(data1$value) *100) %>%  mutate(ypos = cumsum(prop)- 0.5*prop)


data1$prop2=round(data1$prop,2)

data1$prop2=paste(data1$prop2,"%",sep = "")

ggplot(data1, aes(x = '', y = prop, fill = category)) +  
  geom_bar(stat = "identity", color='black', width = 1) +  
  scale_fill_brewer(palette = "RdYlBu") +   coord_polar("y", start = 0) +  theme_void() +  
  theme(legend.position = "right",        plot.title = element_text(hjust = 0.5)) +  
  labs(fill = "")+
  scale_fill_manual(values = c("#E77168","#7AA92C","#23B0B2","#A082B2"))+
  theme(text = element_text(size=22,colour = "black",face = "bold"))+
  geom_label_repel(aes(label=prop2,y=prop),
                   nudge_x=0.6,
                   nudge_y=0.6,
                   size=5,
                   show.legend=F,
                   segment.color="grey50")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/S1D_2/Ctrl.pdf",width = 7,height = 7)


#Adjacency组
data2=as.data.frame(table(subset(umap_ctrl_IUA_allsingle,type2%in%"IUA1")$cell_type1110))
colnames(data2)=c("category","value")
data2 <- data2 %>%   arrange(desc(category)) %>%  mutate(prop = value / sum(data2$value) *100) %>%  mutate(ypos = cumsum(prop)- 0.5*prop)

data2$prop2=round(data2$prop,2)

data2$prop2=paste(data2$prop2,"%",sep = "")

ggplot(data2, aes(x = '', y = prop, fill = category)) +  
  geom_bar(stat = "identity", color='black', width = 1) +  
  scale_fill_brewer(palette = "RdYlBu") +   coord_polar("y", start = 0) +  theme_void() +  
  theme(legend.position = "right",        plot.title = element_text(hjust = 0.5)) +  
  labs(fill = "")+
  scale_fill_manual(values = c("#E77168","#7AA92C","#23B0B2","#A082B2"))+
  theme(text = element_text(size=22,colour = "black",face = "bold"))+
  geom_label_repel(aes(label=prop2,y=prop),
                   nudge_x=0.6,
                   nudge_y=0.6,
                   size=5,
                   show.legend=F,
                   segment.color="grey50")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/S1D_2/Adjacency.pdf",width = 7,height = 7)



#Adhesion组
data2=as.data.frame(table(subset(umap_ctrl_IUA_allsingle,type2%in%"IUA2")$cell_type1110))
colnames(data2)=c("category","value")
data2 <- data2 %>%   arrange(desc(category)) %>%  mutate(prop = value / sum(data2$value) *100) %>%  mutate(ypos = cumsum(prop)- 0.5*prop)

data2$prop2=round(data2$prop,2)

data2$prop2=paste(data2$prop2,"%",sep = "")

ggplot(data2, aes(x = '', y = prop, fill = category)) +  
  geom_bar(stat = "identity", color='black', width = 1) +  
  scale_fill_brewer(palette = "RdYlBu") +   coord_polar("y", start = 0) +  theme_void() +  
  theme(legend.position = "right",        plot.title = element_text(hjust = 0.5)) +  
  labs(fill = "")+
  scale_fill_manual(values = c("#E77168","#7AA92C","#23B0B2","#A082B2"))+
  theme(text = element_text(size=22,colour = "black",face = "bold"))+
  geom_label_repel(aes(label=prop2,y=prop),
                   nudge_x=0.6,
                   nudge_y=0.6,
                   size=5,
                   show.legend=F,
                   segment.color="grey50")


ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/S1D_2/Adhesion.pdf",width = 7,height = 7)

umap_ctrl_IUA_allsingle$type4='Ctrl'

umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type3%in%c("IUA1","IUA2",
                                                                 "IUA5","IUA6")]<-"mi-IUA"

umap_ctrl_IUA_allsingle$type4[umap_ctrl_IUA_allsingle$type3%in%c("IUA3",
                                                                 "IUA4","IUA7")]<-"mo-IUA"


#mi-IUA组
data2=as.data.frame(table(subset(umap_ctrl_IUA_allsingle,type4%in%"mi-IUA")$cell_type1110))
colnames(data2)=c("category","value")
data2 <- data2 %>%   arrange(desc(category)) %>%  mutate(prop = value / sum(data2$value) *100) %>%  mutate(ypos = cumsum(prop)- 0.5*prop)

data2$prop2=round(data2$prop,2)

data2$prop2=paste(data2$prop2,"%",sep = "")

ggplot(data2, aes(x = '', y = prop, fill = category)) +  
  geom_bar(stat = "identity", color='black', width = 1) +  
  scale_fill_brewer(palette = "RdYlBu") +   coord_polar("y", start = 0) +  theme_void() +  
  theme(legend.position = "right",        plot.title = element_text(hjust = 0.5)) +  
  labs(fill = "")+
  scale_fill_manual(values = c("#E77168","#7AA92C","#23B0B2","#A082B2"))+
  theme(text = element_text(size=22,colour = "black",face = "bold"))+
  geom_label_repel(aes(label=prop2,y=prop),
                   nudge_x=0.6,
                   nudge_y=0.6,
                   size=5,
                   show.legend=F,
                   segment.color="grey50")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/S1D_2/mi-IUA.pdf",width = 7,height = 7)




#mo-IUA组
data2=as.data.frame(table(subset(umap_ctrl_IUA_allsingle,type4%in%"mo-IUA")$cell_type1110))
colnames(data2)=c("category","value")
data2 <- data2 %>%   arrange(desc(category)) %>%  mutate(prop = value / sum(data2$value) *100) %>%  mutate(ypos = cumsum(prop)- 0.5*prop)
data2$prop2=round(data2$prop,2)

data2$prop2=paste(data2$prop2,"%",sep = "")

ggplot(data2, aes(x = '', y = prop, fill = category)) +  
  geom_bar(stat = "identity", color='black', width = 1) +  
  scale_fill_brewer(palette = "RdYlBu") +   coord_polar("y", start = 0) +  theme_void() +  
  theme(legend.position = "right",        plot.title = element_text(hjust = 0.5)) +  
  labs(fill = "")+
  scale_fill_manual(values = c("#E77168","#7AA92C","#23B0B2","#A082B2"))+
  theme(text = element_text(size=22,colour = "black",face = "bold"))+
  geom_label_repel(aes(label=prop2,y=prop),
                   nudge_x=0.6,
                   nudge_y=0.6,
                   size=5,
                   show.legend=F,
                   segment.color="grey50")
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/S1D_2/mo-IUA.pdf",width = 7,height = 7)



#CTRL1组
data2=as.data.frame(table(subset(umap_ctrl_IUA_allsingle,type3%in%"CTRL1")$cell_type1110))
colnames(data2)=c("category","value")
data2 <- data2 %>%   arrange(desc(category)) %>%  mutate(prop = value / sum(data2$value) *100) %>%  mutate(ypos = cumsum(prop)- 0.5*prop)
data2$prop2=round(data2$prop,2)

data2$prop2=paste(data2$prop2,"%",sep = "")

ggplot(data2, aes(x = '', y = prop, fill = category)) +  
  geom_bar(stat = "identity", color='black', width = 1) +  
  scale_fill_brewer(palette = "RdYlBu") +   coord_polar("y", start = 0) +  theme_void() +  
  theme(legend.position = "right",        plot.title = element_text(hjust = 0.5)) +  
  labs(fill = "")+
  scale_fill_manual(values = c("#E77168","#7AA92C","#23B0B2","#A082B2"))+
  theme(text = element_text(size=22,colour = "black",face = "bold"))+
  geom_label_repel(aes(label=prop2,y=prop),
                   nudge_x=0.6,
                   nudge_y=0.6,
                   size=5,
                   show.legend=F,
                   segment.color="grey50")
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/S1D_2/CTRL1.pdf",width = 7,height = 7)



#CTRL2组
data2=as.data.frame(table(subset(umap_ctrl_IUA_allsingle,type3%in%"CTRL2")$cell_type1110))
colnames(data2)=c("category","value")
data2 <- data2 %>%   arrange(desc(category)) %>%  mutate(prop = value / sum(data2$value) *100) %>%  mutate(ypos = cumsum(prop)- 0.5*prop)
data2$prop2=round(data2$prop,2)

data2$prop2=paste(data2$prop2,"%",sep = "")

ggplot(data2, aes(x = '', y = prop, fill = category)) +  
  geom_bar(stat = "identity", color='black', width = 1) +  
  scale_fill_brewer(palette = "RdYlBu") +   coord_polar("y", start = 0) +  theme_void() +  
  theme(legend.position = "right",        plot.title = element_text(hjust = 0.5)) +  
  labs(fill = "")+
  scale_fill_manual(values = c("#E77168","#7AA92C","#23B0B2","#A082B2"))+
  theme(text = element_text(size=22,colour = "black",face = "bold"))+
  geom_label_repel(aes(label=prop2,y=prop),
                   nudge_x=0.6,
                   nudge_y=0.6,
                   size=5,
                   show.legend=F,
                   segment.color="grey50")
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/S1D_2/CTRL2.pdf",width = 7,height = 7)


#IUA1组
data2=as.data.frame(table(subset(umap_ctrl_IUA_allsingle,type3%in%"IUA1")$cell_type1110))
colnames(data2)=c("category","value")
data2 <- data2 %>%   arrange(desc(category)) %>%  mutate(prop = value / sum(data2$value) *100) %>%  mutate(ypos = cumsum(prop)- 0.5*prop)
data2$prop2=round(data2$prop,2)

data2$prop2=paste(data2$prop2,"%",sep = "")

ggplot(data2, aes(x = '', y = prop, fill = category)) +  
  geom_bar(stat = "identity", color='black', width = 1) +  
  scale_fill_brewer(palette = "RdYlBu") +   coord_polar("y", start = 0) +  theme_void() +  
  theme(legend.position = "right",        plot.title = element_text(hjust = 0.5)) +  
  labs(fill = "")+
  scale_fill_manual(values = c("#E77168","#7AA92C","#23B0B2","#A082B2"))+
  theme(text = element_text(size=22,colour = "black",face = "bold"))+
  geom_label_repel(aes(label=prop2,y=prop),
                   nudge_x=0.6,
                   nudge_y=0.6,
                   size=5,
                   show.legend=F,
                   segment.color="grey50")
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/S1D_2/IUA1.pdf",width = 7,height = 7)

#IUA2组
data2=as.data.frame(table(subset(umap_ctrl_IUA_allsingle,type3%in%"IUA2")$cell_type1110))
colnames(data2)=c("category","value")
data2 <- data2 %>%   arrange(desc(category)) %>%  mutate(prop = value / sum(data2$value) *100) %>%  mutate(ypos = cumsum(prop)- 0.5*prop)
data2$prop2=round(data2$prop,2)

data2$prop2=paste(data2$prop2,"%",sep = "")

ggplot(data2, aes(x = '', y = prop, fill = category)) +  
  geom_bar(stat = "identity", color='black', width = 1) +  
  scale_fill_brewer(palette = "RdYlBu") +   coord_polar("y", start = 0) +  theme_void() +  
  theme(legend.position = "right",        plot.title = element_text(hjust = 0.5)) +  
  labs(fill = "")+
  scale_fill_manual(values = c("#E77168","#7AA92C","#23B0B2","#A082B2"))+
  theme(text = element_text(size=22,colour = "black",face = "bold"))+
  geom_label_repel(aes(label=prop2,y=prop),
                   nudge_x=0.6,
                   nudge_y=0.6,
                   size=5,
                   show.legend=F,
                   segment.color="grey50")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/S1D_2/IUA2.pdf",width = 7,height = 7)


#IUA3组
data2=as.data.frame(table(subset(umap_ctrl_IUA_allsingle,type3%in%"IUA3")$cell_type1110))
colnames(data2)=c("category","value")
data2 <- data2 %>%   arrange(desc(category)) %>%  mutate(prop = value / sum(data2$value) *100) %>%  mutate(ypos = cumsum(prop)- 0.5*prop)
data2$prop2=round(data2$prop,2)

data2$prop2=paste(data2$prop2,"%",sep = "")

ggplot(data2, aes(x = '', y = prop, fill = category)) +  
  geom_bar(stat = "identity", color='black', width = 1) +  
  scale_fill_brewer(palette = "RdYlBu") +   coord_polar("y", start = 0) +  theme_void() +  
  theme(legend.position = "right",        plot.title = element_text(hjust = 0.5)) +  
  labs(fill = "")+
  scale_fill_manual(values = c("#E77168","#7AA92C","#23B0B2","#A082B2"))+
  theme(text = element_text(size=22,colour = "black",face = "bold"))+
  geom_label_repel(aes(label=prop2,y=prop),
                   nudge_x=0.6,
                   nudge_y=0.6,
                   size=5,
                   show.legend=F,
                   segment.color="grey50")
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/S1D_2/IUA3.pdf",width = 7,height = 7)

#IUA4组
data2=as.data.frame(table(subset(umap_ctrl_IUA_allsingle,type3%in%"IUA4")$cell_type1110))
colnames(data2)=c("category","value")
data2 <- data2 %>%   arrange(desc(category)) %>%  mutate(prop = value / sum(data2$value) *100) %>%  mutate(ypos = cumsum(prop)- 0.5*prop)
data2$prop2=round(data2$prop,2)

data2$prop2=paste(data2$prop2,"%",sep = "")

ggplot(data2, aes(x = '', y = prop, fill = category)) +  
  geom_bar(stat = "identity", color='black', width = 1) +  
  scale_fill_brewer(palette = "RdYlBu") +   coord_polar("y", start = 0) +  theme_void() +  
  theme(legend.position = "right",        plot.title = element_text(hjust = 0.5)) +  
  labs(fill = "")+
  scale_fill_manual(values = c("#E77168","#7AA92C","#23B0B2","#A082B2"))+
  theme(text = element_text(size=22,colour = "black",face = "bold"))+
  geom_label_repel(aes(label=prop2,y=prop),
                   nudge_x=0.6,
                   nudge_y=0.6,
                   size=5,
                   show.legend=F,
                   segment.color="grey50")
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/S1D_2/IUA4.pdf",width = 7,height = 7)


#IUA5组
data2=as.data.frame(table(subset(umap_ctrl_IUA_allsingle,type3%in%"IUA5")$cell_type1110))
colnames(data2)=c("category","value")
data2 <- data2 %>%   arrange(desc(category)) %>%  mutate(prop = value / sum(data2$value) *100) %>%  mutate(ypos = cumsum(prop)- 0.5*prop)
data2$prop2=round(data2$prop,2)

data2$prop2=paste(data2$prop2,"%",sep = "")

ggplot(data2, aes(x = '', y = prop, fill = category)) +  
  geom_bar(stat = "identity", color='black', width = 1) +  
  scale_fill_brewer(palette = "RdYlBu") +   coord_polar("y", start = 0) +  theme_void() +  
  theme(legend.position = "right",        plot.title = element_text(hjust = 0.5)) +  
  labs(fill = "")+
  scale_fill_manual(values = c("#E77168","#7AA92C","#23B0B2","#A082B2"))+
  theme(text = element_text(size=22,colour = "black",face = "bold"))+
  geom_label_repel(aes(label=prop2,y=prop),
                   nudge_x=0.6,
                   nudge_y=0.6,
                   size=5,
                   show.legend=F,
                   segment.color="grey50")
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/S1D_2/IUA5.pdf",width = 7,height = 7)




#IUA6组
data2=as.data.frame(table(subset(umap_ctrl_IUA_allsingle,type3%in%"IUA6")$cell_type1110))
colnames(data2)=c("category","value")
data2 <- data2 %>%   arrange(desc(category)) %>%  mutate(prop = value / sum(data2$value) *100) %>%  mutate(ypos = cumsum(prop)- 0.5*prop)
data2$prop2=round(data2$prop,2)

data2$prop2=paste(data2$prop2,"%",sep = "")

ggplot(data2, aes(x = '', y = prop, fill = category)) +  
  geom_bar(stat = "identity", color='black', width = 1) +  
  scale_fill_brewer(palette = "RdYlBu") +   coord_polar("y", start = 0) +  theme_void() +  
  theme(legend.position = "right",        plot.title = element_text(hjust = 0.5)) +  
  labs(fill = "")+
  scale_fill_manual(values = c("#E77168","#7AA92C","#23B0B2","#A082B2"))+
  theme(text = element_text(size=22,colour = "black",face = "bold"))+
  geom_label_repel(aes(label=prop2,y=prop),
                   nudge_x=0.6,
                   nudge_y=0.6,
                   size=5,
                   show.legend=F,
                   segment.color="grey50")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/S1D_2/IUA6.pdf",width = 7,height = 7)





#IUA7组
data2=as.data.frame(table(subset(umap_ctrl_IUA_allsingle,type3%in%"IUA7")$cell_type1110))
colnames(data2)=c("category","value")
data2 <- data2 %>%   arrange(desc(category)) %>%  mutate(prop = value / sum(data2$value) *100) %>%  mutate(ypos = cumsum(prop)- 0.5*prop)
data2$prop2=round(data2$prop,2)

data2$prop2=paste(data2$prop2,"%",sep = "")

ggplot(data2, aes(x = '', y = prop, fill = category)) +  
  geom_bar(stat = "identity", color='black', width = 1) +  
  scale_fill_brewer(palette = "RdYlBu") +   coord_polar("y", start = 0) +  theme_void() +  
  theme(legend.position = "right",        plot.title = element_text(hjust = 0.5)) +  
  labs(fill = "")+
  scale_fill_manual(values = c("#E77168","#7AA92C","#23B0B2","#A082B2"))+
  theme(text = element_text(size=22,colour = "black",face = "bold"))+
  geom_label_repel(aes(label=prop2,y=prop),
                   nudge_x=0.6,
                   nudge_y=0.6,
                   size=5,
                   show.legend=F,
                   segment.color="grey50")
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/S1D_2/IUA7.pdf",width = 7,height = 7)





#将基质细胞亚群和巨噬细胞亚群进行通讯分析【通过添加减少细胞数目比例的参数在进行一次cellchat并查看改后效果】-----
EnSc=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0718.rds")
Macr=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/Macr_allsingle.rds")

table(EnSc$cell_type0718)
table(Macr$cell_type0610)

EnSc$cell_type1229=EnSc$cell_type0718
Macr$cell_type1229=Macr$cell_type0610

all=merge(EnSc,Macr)

all@meta.data$cell_type1229 <- gsub("Cluster","S",all@meta.data$cell_type1229)


all$type2=gsub("CTRL","Ctrl",all$type2)
all$type2=gsub("IUA1","Adjacency",all$type2)
all$type2=gsub("IUA2","Adhesion",all$type2)

Ctrl=subset(all,type2%in%"Ctrl")
Adjacency=subset(all,type2%in%"Adjacency")
Adhesion=subset(all,type2%in%"Adhesion")

library(Seurat)
library(CellChat)


#all
cellchat <- createCellChat(all@assays$RNA@data,meta = all@meta.data,group.by = "cell_type1229")
cellchat=setIdent(cellchat,ident.use = "cell_type1229")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Interaction weights/strength")

saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/fig5_cellchat/rds/all.rds")


#Ctrl组

cellchat <- createCellChat(Ctrl@assays$RNA@data,meta = Ctrl@meta.data,group.by = "cell_type1229")
cellchat=setIdent(cellchat,ident.use = "cell_type1229")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Interaction weights/strength")

saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/fig5_cellchat/rds/Ctrl.rds")



#Adjacency组
cellchat <- createCellChat(Adjacency@assays$RNA@data,meta = Adjacency@meta.data,group.by = "cell_type1229")
cellchat=setIdent(cellchat,ident.use = "cell_type1229")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Interaction weights/strength")

saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/fig5_cellchat/rds/Adjacency.rds")




#Adhesion组
cellchat <- createCellChat(Adhesion@assays$RNA@data,meta = Adhesion@meta.data,group.by = "cell_type1229")
cellchat=setIdent(cellchat,ident.use = "cell_type1229")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Interaction weights/strength")

saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/fig5_cellchat/rds/Adhesion.rds")



##整合新的结果看看和之前是否有什么不一样
all=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/fig5_cellchat/rds/all.rds")
Ctrl=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/fig5_cellchat/rds/Ctrl.rds")
Adjacency=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/fig5_cellchat/rds/Adjacency.rds")
Adhesion=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/fig5_cellchat/rds/Adhesion.rds")


object.list <- list(Ctrl = Ctrl, Adjacency = Adjacency , Adhesion = Adhesion)

cellchat <- mergeCellChat(object.list, add.names = names(object.list), cell.prefix = TRUE)


gg1 <- rankNet(cellchat, mode = "comparison", stacked = T, do.stat = TRUE,comparison = c(1,2,3))
gg2 <- rankNet(cellchat, mode = "comparison", stacked = F, do.stat = TRUE,comparison = c(1,2,3))
gg1 + gg2





gg1 <- netVisual_heatmap(cellchat, measure = "weight",comparison = c(1,2))
gg2 <- netVisual_heatmap(cellchat, measure = "weight",comparison = c(1,3))
gg1 + gg2

gg1 <-netVisual_diffInteraction(cellchat, weight.scale = T, measure = "weight",comparison = c(1,2))
gg2 <-netVisual_diffInteraction(cellchat, weight.scale = T, measure = "weight",comparison = c(1,3))
gg1 + gg2




pathways.show <- c("TGFb") 
weight.max <- getMaxWeight(object.list, slot.name = c("netP"), attribute = pathways.show) # control the edge weights across different datasets
par(mfrow = c(3,1), xpd=TRUE)
for (i in 1:length(object.list)) {
  netVisual_aggregate(object.list[[i]], signaling = pathways.show, layout = "circle", edge.weight.max = weight.max[1], edge.width.max = 10, signaling.name = paste(pathways.show, names(object.list)[i]))
}

pathways.show <- c("SPP1") 
weight.max <- getMaxWeight(object.list, slot.name = c("netP"), attribute = pathways.show) # control the edge weights across different datasets
par(mfrow = c(1,3), xpd=TRUE)
for (i in 1:length(object.list)) {
  netVisual_aggregate(object.list[[i]], signaling = pathways.show, layout = "circle", edge.weight.max = weight.max[1], edge.width.max = 10, signaling.name = paste(pathways.show, names(object.list)[i]))
}



#SPP1和complement信号通路在整体里面的通讯变化
pathways.show <- c("SPP1") 
par(mfrow = c(1,2), xpd=TRUE)
netVisual_aggregate(Adhesion, signaling = "SPP1", layout = "circle")

netVisual_aggregate(Adhesion, signaling = "TGFb", layout = "circle")



#将CD301+文章的巨噬细胞和IUA的巨噬细胞进行整合并查看我们定义的亚群如何----
WEN=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/CD301_WEN_DATA/rds存放点/all_harmony.rds")


DimPlot(WEN,label = T,label.size = 8)

#先对这些亚群进行定义【分大群定义即可】


STR_maker=c("LUM","COL1A1","DCN","MMP11")
FeaturePlot(WEN,features = STR_maker)

EPI_maker=c("EPCAM","KRT18","MMP7","WFDC2")
FeaturePlot(WEN,features = EPI_maker)

Endothelium_maker=c("VWF","CDH5")
FeaturePlot(WEN,features = Endothelium_maker)

smooth_maker=c("MCAM","RGS5","ACTA2")
FeaturePlot(WEN,features = smooth_maker)


Macr_marker=c("LST1","S100A9","S100A8","LYZ","IL1B","S100A12","CD68")
FeaturePlot(WEN,features = Macr_marker,min.cutoff = 0,order = T)












#对SCVI整合进行文件准备工作-----
library(anndata)
library(Seurat)
library(SingleCellExperiment)
library(SeuratDisk)

setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/scVI整合/rds/")


#CTRL1
CTRL1=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/H.rds")
SaveH5Seurat(CTRL1, filename = "CTRL1.h5Seurat")
Convert("CTRL1.h5Seurat", dest = "h5ad")

#CTRL2
CTRL2=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/qian.rds")
SaveH5Seurat(CTRL2, filename = "CTRL2.h5Seurat")
Convert("CTRL2.h5Seurat", dest = "h5ad")

IUA1=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/IUAshou.rds")
IUA2=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/IUAxia.rds")
IUA3=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/li.rds")
IUA4=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/IUAfeng.rds")
IUA5=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/IUA_GWY_additional.rds")
IUA6=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/IUA_BYQ.rds")
IUA7=readRDS("/mnt/chentw/zigongneimo/photo/11-samples/seurat-object/IUA_WJM.rds")
umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")


SaveH5Seurat(IUA1, filename = "IUA1.h5Seurat")
Convert("IUA1.h5Seurat", dest = "h5ad")

SaveH5Seurat(IUA2, filename = "IUA2.h5Seurat")
Convert("IUA2.h5Seurat", dest = "h5ad")

SaveH5Seurat(IUA3, filename = "IUA3.h5Seurat")
Convert("IUA3.h5Seurat", dest = "h5ad")

SaveH5Seurat(IUA4, filename = "IUA4.h5Seurat")
Convert("IUA4.h5Seurat", dest = "h5ad")

SaveH5Seurat(IUA5, filename = "IUA5.h5Seurat")
Convert("IUA5.h5Seurat", dest = "h5ad")

SaveH5Seurat(IUA6, filename = "IUA6.h5Seurat")
Convert("IUA6.h5Seurat", dest = "h5ad")

SaveH5Seurat(IUA7, filename = "IUA7.h5Seurat")
Convert("IUA7.h5Seurat", dest = "h5ad")

SaveH5Seurat(umap_ctrl_IUA_allsingle, filename = "umap_ctrl_IUA_allsingle.h5Seurat")
Convert("umap_ctrl_IUA_allsingle.h5Seurat", dest = "h5ad")

Z=as.data.frame(t(umap_ctrl_IUA_allsingle@assays$RNA@counts))

write.csv(Z,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/scVI整合/rds/path_to_raw_counts.csv")

seurat_meta <- umap_ctrl_IUA_allsingle@meta.data[, c("type1", "type2", "type3","cell_type","cell_type1110")]

write.csv(seurat_meta, "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/scVI整合/rds/seurat_meta.csv", row.names = FALSE)



#全部一起整合数据量过大提取出一部分来试着做一下------
umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")

umap_all_single_scRNA_sub <-subset(umap_ctrl_IUA_allsingle,downsample = 5000)

DimPlot(umap_all_single_scRNA_sub,group.by = "cell_type1110",label = T,label.size = 12,split.by = "type3")

table(umap_all_single_scRNA_sub$cell_type1110,umap_all_single_scRNA_sub$type3)

Z=as.data.frame(t(umap_all_single_scRNA_sub@assays$RNA@counts))

write.csv(Z,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/scVI整合/test/path_to_raw_counts.csv")


seurat_meta <- umap_all_single_scRNA_sub@meta.data[, c("type1", "type2", "type3","cell_type","cell_type1110")]

write.csv(seurat_meta, "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/scVI整合/test/seurat_meta.csv", row.names = FALSE)

setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/scVI整合/test/")

SaveH5Seurat(umap_all_single_scRNA_sub, filename = "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/scVI整合/test/umap_all_single_scRNA_sub.h5Seurat")
Convert("umap_all_single_scRNA_sub.h5Seurat", dest = "h5ad")



#测试2----
umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")


subcell <- sample(colnames(umap_ctrl_IUA_allsingle),4000)
umap_all_single_scRNAsub <- umap_ctrl_IUA_allsingle[,subcell]

SaveH5Seurat(umap_all_single_scRNAsub, filename = "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/scVI整合/test2/umap_all_single_scRNAsub.h5Seurat")
Convert("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/scVI整合/test2/umap_all_single_scRNAsub.h5Seurat", dest = "h5ad")



SaveH5Seurat(umap_all_single_scRNAsub[top_2000_genes,], filename = "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/scVI整合/test2/umap_all_single_scRNAsub2.h5Seurat")
Convert("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/scVI整合/test2/umap_all_single_scRNAsub2.h5Seurat", dest = "h5ad")



Z=as.data.frame(t(umap_all_single_scRNAsub@assays$RNA@counts))

write.csv(Z,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/scVI整合/test2/path_to_raw_counts.csv")


seurat_meta <- umap_all_single_scRNAsub@meta.data[, c("type1", "type2", "type3","cell_type","cell_type1110")]

write.csv(seurat_meta, "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/scVI整合/test2/seurat_meta.csv", row.names = FALSE)


umap_all_single_scRNAsub <- FindVariableFeatures(umap_all_single_scRNAsub, selection.method = "vst", nfeatures = 2000)
top_2000_genes <- VariableFeatures(umap_all_single_scRNAsub)


Z=as.data.frame(t(umap_all_single_scRNAsub@assays$RNA@counts[top_2000_genes,]))

write.csv(Z,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/scVI整合/test2/path_to_raw_counts2.csv")



#将SCVI整合的结果提取出来进行绘图-----
library(anndata)
umape=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/scVI整合/test2/umap_coordinates.csv")

seurat_meta=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/scVI整合/test2/seurat_meta.csv")


adata <- read_h5ad("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/scVI整合/test2/umap_all_single_scRNAsub2.h5ad")

meta= adata$obs

meta$umap1=umape$UMAP1
meta$umap2=umape$UMAP2

rownames(seurat_meta)=rownames(meta)

meta$cell_type1110=seurat_meta$cell_type1110

ggplot(meta)+aes(x=umap1,y=umap2)+geom_point(aes(color=cell_type1110))+
  facet_grid(.~meta$type3)

#将SCVI整合的结果提取出来进行绘图-所有细胞
library(anndata)
umape=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/scVI整合/rds/umap_coordinates.csv")

seurat_meta=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/scVI整合/rds/seurat_meta.csv")


adata <- read_h5ad("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/scVI整合/rds/umap_ctrl_IUA_allsingle.h5ad")

meta= adata$obs

meta$umap1=umape$UMAP1
meta$umap2=umape$UMAP2

rownames(seurat_meta)=rownames(meta)

meta$cell_type1110=seurat_meta$cell_type1110

meta$cell_type1110=factor(meta$cell_type1110,levels = c("Stromal cells","Immune cells",
                                                        "Epithelium cells","Endothelium cells"))

P1=ggplot(subset(meta,type3%in%"CTRL1"))+aes(x=umap1,y=umap2)+geom_point(aes(color=cell_type1110))+theme_bw(base_size=20)+
  theme(panel.border = element_rect(color = "black", size = 1, fill = NA))+
  theme(axis.line=element_line(colour='black'),
        plot.title=element_text(hjust=0.5),
        legend.title=element_blank(),
        legend.position="none",
        legend.background=element_rect(fill='transparent'),
        axis.text.x=element_text(size="20",face = "bold",colour = "black"),
        axis.text.y=element_text(size="20",face = "bold",colour = "black"),
        legend.text=element_text(size="20",face = "bold",colour = "black"),
        axis.title.y=element_text(size="20",face = "bold",colour = "black"),
        axis.title.x=element_text(size="20",face = "bold",colour = "black"))+
  scale_fill_manual(values=c("#E37668","#88B532","#3BBEC0","#9F87AC"))

P2=ggplot(subset(meta,type3%in%"CTRL2"))+aes(x=umap1,y=umap2)+geom_point(aes(color=cell_type1110))+theme_bw(base_size=20)+
  theme(panel.border = element_rect(color = "black", size = 1, fill = NA))+
  theme(axis.line=element_line(colour='black'),
        plot.title=element_text(hjust=0.5),
        legend.title=element_blank(),
        legend.position="none",
        legend.background=element_rect(fill='transparent'),
        axis.text.x=element_text(size="20",face = "bold",colour = "black"),
        axis.text.y=element_text(size="20",face = "bold",colour = "black"),
        legend.text=element_text(size="20",face = "bold",colour = "black"),
        axis.title.y=element_text(size="20",face = "bold",colour = "black"),
        axis.title.x=element_text(size="20",face = "bold",colour = "black"))+
  scale_fill_manual(values=c("#E37668","#88B532","#3BBEC0","#9F87AC"))

P3=ggplot(subset(meta,type3%in%"IUA1"))+aes(x=umap1,y=umap2)+geom_point(aes(color=cell_type1110))+theme_bw(base_size=20)+
  theme(panel.border = element_rect(color = "black", size = 1, fill = NA))+
  theme(axis.line=element_line(colour='black'),
        plot.title=element_text(hjust=0.5),
        legend.title=element_blank(),
        legend.position="none",
        legend.background=element_rect(fill='transparent'),
        axis.text.x=element_text(size="20",face = "bold",colour = "black"),
        axis.text.y=element_text(size="20",face = "bold",colour = "black"),
        legend.text=element_text(size="20",face = "bold",colour = "black"),
        axis.title.y=element_text(size="20",face = "bold",colour = "black"),
        axis.title.x=element_text(size="20",face = "bold",colour = "black"))+
  scale_fill_manual(values=c("#E37668","#88B532","#3BBEC0","#9F87AC"))

P4=ggplot(subset(meta,type3%in%"IUA2"))+aes(x=umap1,y=umap2)+geom_point(aes(color=cell_type1110))+theme_bw(base_size=20)+
  theme(panel.border = element_rect(color = "black", size = 1, fill = NA))+
  theme(axis.line=element_line(colour='black'),
        plot.title=element_text(hjust=0.5),
        legend.title=element_blank(),
        legend.position="none",
        legend.background=element_rect(fill='transparent'),
        axis.text.x=element_text(size="20",face = "bold",colour = "black"),
        axis.text.y=element_text(size="20",face = "bold",colour = "black"),
        legend.text=element_text(size="20",face = "bold",colour = "black"),
        axis.title.y=element_text(size="20",face = "bold",colour = "black"),
        axis.title.x=element_text(size="20",face = "bold",colour = "black"))+
  scale_fill_manual(values=c("#E37668","#88B532","#3BBEC0","#9F87AC"))


P5=ggplot(subset(meta,type3%in%"IUA3"))+aes(x=umap1,y=umap2)+geom_point(aes(color=cell_type1110))+theme_bw(base_size=20)+
  theme(panel.border = element_rect(color = "black", size = 1, fill = NA))+
  theme(axis.line=element_line(colour='black'),
        plot.title=element_text(hjust=0.5),
        legend.title=element_blank(),
        legend.position="none",
        legend.background=element_rect(fill='transparent'),
        axis.text.x=element_text(size="20",face = "bold",colour = "black"),
        axis.text.y=element_text(size="20",face = "bold",colour = "black"),
        legend.text=element_text(size="20",face = "bold",colour = "black"),
        axis.title.y=element_text(size="20",face = "bold",colour = "black"),
        axis.title.x=element_text(size="20",face = "bold",colour = "black"))+
  scale_fill_manual(values=c("#E37668","#88B532","#3BBEC0","#9F87AC"))


P6=ggplot(subset(meta,type3%in%"IUA4"))+aes(x=umap1,y=umap2)+geom_point(aes(color=cell_type1110))+theme_bw(base_size=20)+
  theme(panel.border = element_rect(color = "black", size = 1, fill = NA))+
  theme(axis.line=element_line(colour='black'),
        plot.title=element_text(hjust=0.5),
        legend.title=element_blank(),
        legend.position="none",
        legend.background=element_rect(fill='transparent'),
        axis.text.x=element_text(size="20",face = "bold",colour = "black"),
        axis.text.y=element_text(size="20",face = "bold",colour = "black"),
        legend.text=element_text(size="20",face = "bold",colour = "black"),
        axis.title.y=element_text(size="20",face = "bold",colour = "black"),
        axis.title.x=element_text(size="20",face = "bold",colour = "black"))+
  scale_fill_manual(values=c("#E37668","#88B532","#3BBEC0","#9F87AC"))


P7=ggplot(subset(meta,type3%in%"IUA5"))+aes(x=umap1,y=umap2)+geom_point(aes(color=cell_type1110))+theme_bw(base_size=20)+
  theme(panel.border = element_rect(color = "black", size = 1, fill = NA))+
  theme(axis.line=element_line(colour='black'),
        plot.title=element_text(hjust=0.5),
        legend.title=element_blank(),
        legend.position="none",
        legend.background=element_rect(fill='transparent'),
        axis.text.x=element_text(size="20",face = "bold",colour = "black"),
        axis.text.y=element_text(size="20",face = "bold",colour = "black"),
        legend.text=element_text(size="20",face = "bold",colour = "black"),
        axis.title.y=element_text(size="20",face = "bold",colour = "black"),
        axis.title.x=element_text(size="20",face = "bold",colour = "black"))+
  scale_fill_manual(values=c("#E37668","#88B532","#3BBEC0","#9F87AC"))



P8=ggplot(subset(meta,type3%in%"IUA6"))+aes(x=umap1,y=umap2)+geom_point(aes(color=cell_type1110))+theme_bw(base_size=20)+
  theme(panel.border = element_rect(color = "black", size = 1, fill = NA))+
  theme(axis.line=element_line(colour='black'),
        plot.title=element_text(hjust=0.5),
        legend.title=element_blank(),
        legend.position="none",
        legend.background=element_rect(fill='transparent'),
        axis.text.x=element_text(size="20",face = "bold",colour = "black"),
        axis.text.y=element_text(size="20",face = "bold",colour = "black"),
        legend.text=element_text(size="20",face = "bold",colour = "black"),
        axis.title.y=element_text(size="20",face = "bold",colour = "black"),
        axis.title.x=element_text(size="20",face = "bold",colour = "black"))+
  scale_fill_manual(values=c("#E37668","#88B532","#3BBEC0","#9F87AC"))




P9=ggplot(subset(meta,type3%in%"IUA7"))+aes(x=umap1,y=umap2)+geom_point(aes(color=cell_type1110))+theme_bw(base_size=20)+
  theme(panel.border = element_rect(color = "black", size = 1, fill = NA))+
  theme(axis.line=element_line(colour='black'),
        plot.title=element_text(hjust=0.5),
        legend.title=element_blank(),
        legend.position="none",
        legend.background=element_rect(fill='transparent'),
        axis.text.x=element_text(size="20",face = "bold",colour = "black"),
        axis.text.y=element_text(size="20",face = "bold",colour = "black"),
        legend.text=element_text(size="20",face = "bold",colour = "black"),
        axis.title.y=element_text(size="20",face = "bold",colour = "black"),
        axis.title.x=element_text(size="20",face = "bold",colour = "black"))+
  scale_fill_manual(values=c("#E37668","#88B532","#3BBEC0","#9F87AC"))




cowplot::plot_grid(P1,P2,P3,P4,P5,P6,P7,P8,P9,ncol = 5)




#展示SCVI整合结果和integrated结果都没有批次效应-----
umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")

meta2=umap_ctrl_IUA_allsingle@meta.data


library(anndata)
umape=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/scVI整合/rds/umap_coordinates.csv")

seurat_meta=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/scVI整合/rds/seurat_meta.csv")


adata <- read_h5ad("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/scVI整合/rds/umap_ctrl_IUA_allsingle.h5ad")

meta= adata$obs

meta$umap1=umape$UMAP1
meta$umap2=umape$UMAP2

meta3=meta[c("umap1","umap2")]

colnames(meta3)=c("UMAP_1","UMAP_2")



DimPlot(umap_ctrl_IUA_allsingle)



umap_ctrl_IUA_allsingle[["umap"]]@cell.embeddings=as.matrix(meta3)


DimPlot(umap_ctrl_IUA_allsingle,reduction="umap",label=F,label.size =7,pt.size = 0.9,split.by = "type3",
        ncol = 3)+
  theme(title = element_text(size =0),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 66),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),axis.line = element_blank(),
        plot.subtitle = element_text(size = 36))+theme(text = element_text(size=36))

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/SCVI整合1.pdf",width = 28,height = 21)

umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")

DimPlot(umap_ctrl_IUA_allsingle,reduction="umap",label=F,label.size =7,pt.size = 0.9,split.by = "type3",
        ncol = 3)+
  theme(title = element_text(size =0),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 66),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),axis.line = element_blank(),
        plot.subtitle = element_text(size = 36))+theme(text = element_text(size=36))


ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/integrated整合1.pdf",width = 28,height = 21)



umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")

set.seed(123)  # 设置随机种子以保证可重复性
random_order <- sample(ncol(umap_ctrl_IUA_allsingle))  # 随机打乱点的顺序


DimPlot(umap_ctrl_IUA_allsingle,reduction="umap",label=F,label.size =7,pt.size = 0.2,group.by  = "type3", cells = random_order,
        raster = F)+
  theme(title = element_text(size =0),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 22),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),axis.line = element_blank(),
        plot.subtitle = element_text(size = 12))+theme(text = element_text(size=12))+
  scale_fill_manual(values = c("#E9726D","#CA8D20","#8EA430","#30AD43","#2FB396","#27AFD0","#6894CB","#A878B1","#D766A1"))



ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/integrated整合3.pdf",width = 9,height = 7)


DimPlot(umap_ctrl_IUA_allsingle,reduction="umap",label=F,label.size =7,pt.size = 0.9,split.by = "type3",group.by="type3",
        ncol = 3)+
  theme(title = element_text(size =0),legend.key.size = unit(1,'cm'),legend.text = element_text(size = 66),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),axis.line = element_blank(),
        plot.subtitle = element_text(size = 36))+theme(text = element_text(size=36))+
  scale_fill_manual(values = c("#E9726D","#CA8D20","#8EA430","#30AD43","#2FB396","#27AFD0","#6894CB","#A878B1","#D766A1"))



ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/integrated整合4.pdf",width = 28,height = 21)






#将我们的基质和文章不同周期的基质进行monocle2分析----
###Ctrl-------
WEN_umap=readRDS("/mnt/chentw/zigongneimo/photo/文献数据操作数据/WEN.umap.rds")
EnSc=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0718.rds")

WEN_STR=subset(WEN_umap,cell_type%in%"Stromal fibroblasts")

EnSc_Ctrl=subset(EnSc,type2%in%"CTRL")


EnSc_Ctrl$donor="proliferative"

WEN_STR$type2="WEN_STR"

all2=merge(EnSc_Ctrl,WEN_STR)

all2 <- CreateSeuratObject(counts =all2@assays$RNA@counts, project = "all2", min.cells = 3,meta.data = all2@meta.data)

VlnPlot(all2, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

all2 <- NormalizeData(all2, normalization.method = "LogNormalize", 
                      scale.factor = 10000)
all2 <- FindVariableFeatures(all2, selection.method = "vst", nfeatures = 2000)
VlnPlot(all2, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

#all2进行标准化然后PCA分析
DefaultAssay(all2) <- "RNA"
all.genes_all2 <- rownames(all2)
all2 <- ScaleData(object = all2, features = all.genes_all2)
all2 <- RunPCA(all2, features = VariableFeatures(object = all2))


#all2进行细胞细胞聚类分析(UMAP)
all2=RunUMAP(all2, dims = 1:30)
DimPlot(all2,reduction="umap",label=TRUE)
all2 <- FindNeighbors(all2, dims = 1:30)
all2 <- FindClusters(all2, resolution = 0.3)
all2 <- FindClusters(all2, resolution = 0.6)
DimPlot(all2,reduction="umap",label=TRUE,group.by = "donor")



#收集上面all2的marker基因和高可变基因
#收集起来对count矩阵提取，这样就可以节省后续的运行时间
all2@active.ident=as.factor(all2$donor)
all2_markers <- FindAllMarkers(object=all2, only.pos = TRUE,min.pct = 0.25, 
                               logfc.threshold = 0.25)
all2_markers <-subset(all2_markers,all2_markers$p_val<0.05)

all2_markers=all2_markers$gene

hvg_genes <- VariableFeatures(all2)

gene3=c(all2_markers,hvg_genes)


##开始monocle2分析[ctrl-上皮]
library(monocle)

DefaultAssay(all2)<-"RNA"

all2_matrix=as.data.frame(all2@assays$RNA@counts)

all2_matrix=all2_matrix[gene3,]

all2_matrix<-as(as.matrix(all2_matrix),'sparseMatrix')



all2_ann<-data.frame(gene_id=rownames(all2_matrix),gene_short_name=rownames(all2_matrix))

rownames(all2_ann)<-rownames(all2_matrix)

all2_fd<-new("AnnotatedDataFrame", data = all2_ann)

phenoData <- all2@meta.data
phenoData$celltype <- all2$donor
all2_pd <- new('AnnotatedDataFrame', data = phenoData)



cds<-newCellDataSet(all2_matrix,phenoData =all2_pd,
                    featureData =all2_fd,expressionFamily=negbinomial.size())

cds <- estimateSizeFactors(cds)
cds <- estimateDispersions(cds)

cds<-detectGenes(cds,min_expr=0.1)
expressed_genes <- row.names(subset(fData(cds),
                                    num_cells_expressed>=10))

diff_test_res <- differentialGeneTest(cds[expressed_genes,],                                      
                                      fullModelFormulaStr = "~donor",cores=16)



sig_genes<-subset(diff_test_res,qval<0.01)
sig_genes <- sig_genes[order(sig_genes$pval),]


# 标记为用于细胞排序的基因
cds <- setOrderingFilter(cds, sig_genes$gene_short_name)


all2@active.ident=as.factor(all2$donor)

express_genes <- VariableFeatures(all2)

cds <- setOrderingFilter(cds[expressed_genes,],                          
                         express_genes)

disp_table <- dispersionTable(cds)
disp.genes <- subset(disp_table, mean_expression>=0.1&dispersion_empirical>=1*dispersion_fit)$gene_id

cds <- setOrderingFilter(cds[expressed_genes,],                          
                         disp.genes) 

cds <- reduceDimension(cds, max_components=2,                       
                       reduction_method = 'DDRTree',                        
                       verbose = T)

cds <- orderCells(cds)

saveRDS(cds,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/ctrl_STR.rds")

cds=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/ctrl_STR.rds")


P1=plot_cell_trajectory(cds,cell_size = 1,color_by = "donor",label=T)+ggtitle("")+
  facet_wrap(~donor)

P2=plot_cell_trajectory(cds,cell_size = 1,color_by = "Pseudotime")

P1+P2



###Adhesion-------
WEN_umap=readRDS("/mnt/chentw/zigongneimo/photo/文献数据操作数据/WEN.umap.rds")
EnSc=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0718.rds")

WEN_STR=subset(WEN_umap,cell_type%in%"Stromal fibroblasts")

EnSc_Adhesion=subset(EnSc,type2%in%"IUA2")


EnSc_Adhesion$donor="proliferative"

WEN_STR$type2="WEN_STR"

all2=merge(EnSc_Adhesion,WEN_STR)

all2 <- CreateSeuratObject(counts =all2@assays$RNA@counts, project = "all2", min.cells = 3,meta.data = all2@meta.data)

VlnPlot(all2, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

all2 <- NormalizeData(all2, normalization.method = "LogNormalize", 
                      scale.factor = 10000)
all2 <- FindVariableFeatures(all2, selection.method = "vst", nfeatures = 2000)
VlnPlot(all2, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

#all2进行标准化然后PCA分析
DefaultAssay(all2) <- "RNA"
all.genes_all2 <- rownames(all2)
all2 <- ScaleData(object = all2, features = all.genes_all2)
all2 <- RunPCA(all2, features = VariableFeatures(object = all2))


#all2进行细胞细胞聚类分析(UMAP)
all2=RunUMAP(all2, dims = 1:30)
DimPlot(all2,reduction="umap",label=TRUE)
all2 <- FindNeighbors(all2, dims = 1:30)
all2 <- FindClusters(all2, resolution = 0.3)
all2 <- FindClusters(all2, resolution = 0.6)
DimPlot(all2,reduction="umap",label=TRUE,group.by = "donor")



#收集上面all2的marker基因和高可变基因
#收集起来对count矩阵提取，这样就可以节省后续的运行时间
all2@active.ident=as.factor(all2$donor)
all2_markers <- FindAllMarkers(object=all2, only.pos = TRUE,min.pct = 0.25, 
                               logfc.threshold = 0.25)
all2_markers <-subset(all2_markers,all2_markers$p_val<0.05)

all2_markers=all2_markers$gene

hvg_genes <- VariableFeatures(all2)

gene3=c(all2_markers,hvg_genes)


##开始monocle2分析[ctrl-上皮]
library(monocle)

DefaultAssay(all2)<-"RNA"

all2_matrix=as.data.frame(all2@assays$RNA@counts)

all2_matrix=all2_matrix[gene3,]

all2_matrix<-as(as.matrix(all2_matrix),'sparseMatrix')



all2_ann<-data.frame(gene_id=rownames(all2_matrix),gene_short_name=rownames(all2_matrix))

rownames(all2_ann)<-rownames(all2_matrix)

all2_fd<-new("AnnotatedDataFrame", data = all2_ann)

phenoData <- all2@meta.data
phenoData$celltype <- all2$donor
all2_pd <- new('AnnotatedDataFrame', data = phenoData)



cds<-newCellDataSet(all2_matrix,phenoData =all2_pd,
                    featureData =all2_fd,expressionFamily=negbinomial.size())

cds <- estimateSizeFactors(cds)
cds <- estimateDispersions(cds)

cds<-detectGenes(cds,min_expr=0.1)
expressed_genes <- row.names(subset(fData(cds),
                                    num_cells_expressed>=10))

diff_test_res <- differentialGeneTest(cds[expressed_genes,],                                      
                                      fullModelFormulaStr = "~donor",cores=16)



sig_genes<-subset(diff_test_res,qval<0.01)
sig_genes <- sig_genes[order(sig_genes$pval),]


# 标记为用于细胞排序的基因
cds <- setOrderingFilter(cds, sig_genes$gene_short_name)


all2@active.ident=as.factor(all2$donor)

express_genes <- VariableFeatures(all2)

cds <- setOrderingFilter(cds[expressed_genes,],                          
                         express_genes)

disp_table <- dispersionTable(cds)
disp.genes <- subset(disp_table, mean_expression>=0.1&dispersion_empirical>=1*dispersion_fit)$gene_id

cds <- setOrderingFilter(cds[expressed_genes,],                          
                         disp.genes) 

cds <- reduceDimension(cds, max_components=2,                       
                       reduction_method = 'DDRTree',                        
                       verbose = T)

cds <- orderCells(cds)

saveRDS(cds,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/Adhesion_STR.rds")

cds=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/Adhesion_STR.rds")


P1=plot_cell_trajectory(cds,cell_size = 1,color_by = "donor",label=T)+ggtitle("")+
  facet_wrap(~donor)

P2=plot_cell_trajectory(cds,cell_size = 1,color_by = "Pseudotime")

P1+P2



#将我们的所有上皮和文章不同周期的上皮进行monocle2分析----
###Ctrl-------
WEN_umap=readRDS("/mnt/chentw/zigongneimo/photo/文献数据操作数据/WEN.umap.rds")
EnEc=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/EPI_allsingle.rds")

WEN_EPI=subset(WEN_umap,cell_type%in%c("Unciliated epithelia 1","Unciliated epithelia 2","Ciliated"))

EnEc_Ctrl=subset(EnEc,type2%in%"CTRL")

EnEc_Ctrl$donor="proliferative"

WEN_EPI$type2="WEN_EPI"

all2=merge(EnEc_Ctrl,WEN_EPI)

all2 <- CreateSeuratObject(counts =all2@assays$RNA@counts, project = "all2", min.cells = 3,meta.data = all2@meta.data)

VlnPlot(all2, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

all2 <- NormalizeData(all2, normalization.method = "LogNormalize", 
                      scale.factor = 10000)
all2 <- FindVariableFeatures(all2, selection.method = "vst", nfeatures = 2000)
VlnPlot(all2, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

#all2进行标准化然后PCA分析
DefaultAssay(all2) <- "RNA"
all.genes_all2 <- rownames(all2)
all2 <- ScaleData(object = all2, features = all.genes_all2)
all2 <- RunPCA(all2, features = VariableFeatures(object = all2))


#all2进行细胞细胞聚类分析(UMAP)
all2=RunUMAP(all2, dims = 1:30)
DimPlot(all2,reduction="umap",label=TRUE)
all2 <- FindNeighbors(all2, dims = 1:30)
all2 <- FindClusters(all2, resolution = 0.3)
all2 <- FindClusters(all2, resolution = 0.6)
DimPlot(all2,reduction="umap",label=TRUE,group.by = "donor")



#收集上面all2的marker基因和高可变基因
#收集起来对count矩阵提取，这样就可以节省后续的运行时间
all2@active.ident=as.factor(all2$donor)
all2_markers <- FindAllMarkers(object=all2, only.pos = TRUE,min.pct = 0.25, 
                               logfc.threshold = 0.25)
all2_markers <-subset(all2_markers,all2_markers$p_val<0.05)

all2_markers=all2_markers$gene

hvg_genes <- VariableFeatures(all2)

gene3=c(all2_markers,hvg_genes)


##开始monocle2分析[ctrl-上皮]
library(monocle)

DefaultAssay(all2)<-"RNA"

all2_matrix=as.data.frame(all2@assays$RNA@counts)

all2_matrix=all2_matrix[gene3,]

all2_matrix<-as(as.matrix(all2_matrix),'sparseMatrix')



all2_ann<-data.frame(gene_id=rownames(all2_matrix),gene_short_name=rownames(all2_matrix))

rownames(all2_ann)<-rownames(all2_matrix)

all2_fd<-new("AnnotatedDataFrame", data = all2_ann)

phenoData <- all2@meta.data
phenoData$celltype <- all2$donor
all2_pd <- new('AnnotatedDataFrame', data = phenoData)



cds<-newCellDataSet(all2_matrix,phenoData =all2_pd,
                    featureData =all2_fd,expressionFamily=negbinomial.size())

cds <- estimateSizeFactors(cds)
cds <- estimateDispersions(cds)

cds<-detectGenes(cds,min_expr=0.1)
expressed_genes <- row.names(subset(fData(cds),
                                    num_cells_expressed>=10))

diff_test_res <- differentialGeneTest(cds[expressed_genes,],                                      
                                      fullModelFormulaStr = "~donor",cores=16)



sig_genes<-subset(diff_test_res,qval<0.01)
sig_genes <- sig_genes[order(sig_genes$pval),]


# 标记为用于细胞排序的基因
cds <- setOrderingFilter(cds, sig_genes$gene_short_name)


all2@active.ident=as.factor(all2$donor)

express_genes <- VariableFeatures(all2)

cds <- setOrderingFilter(cds[expressed_genes,],                          
                         express_genes)

disp_table <- dispersionTable(cds)
disp.genes <- subset(disp_table, mean_expression>=0.1&dispersion_empirical>=1*dispersion_fit)$gene_id

cds <- setOrderingFilter(cds[expressed_genes,],                          
                         disp.genes) 

cds <- reduceDimension(cds, max_components=2,                       
                       reduction_method = 'DDRTree',                        
                       verbose = T)

cds <- orderCells(cds)

saveRDS(cds,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/ctrl_EPI.rds")

cds=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/ctrl_EPI.rds")


P1=plot_cell_trajectory(cds,cell_size = 1,color_by = "donor",label=T)+ggtitle("")+
  facet_wrap(~donor)

P2=plot_cell_trajectory(cds,cell_size = 1,color_by = "Pseudotime")

P1+P2

cds=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/ctrl_EPI.rds")



P1=plot_cell_trajectory(cds,cell_size = 1,color_by = "donor",label=T)+ggtitle("")+
  facet_wrap(~donor)

P2=plot_cell_trajectory(cds,cell_size = 1,color_by = "Pseudotime")

P1+P2

cds$donor=factor(cds$donor,levels = c("proliferative","secretory_early",
                                      "secretory_early-mid","secretory_mid","secretory_late"))

plot_cell_trajectory(cds,cell_size = 1,color_by = "donor",label=T)+ggtitle("")+
  facet_wrap(~donor)

plot_cell_trajectory(cds,cell_size = 1,color_by = "donor",label=T)+ggtitle("")+
  facet_wrap(~type2)

plot_cell_trajectory(cds,cell_size = 1,color_by = "cell_type",label=T)+ggtitle("")+
  facet_wrap(~cell_type)


###Adhesion-------
WEN_umap=readRDS("/mnt/chentw/zigongneimo/photo/文献数据操作数据/WEN.umap.rds")
EnEc=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/EPI_allsingle.rds")

WEN_EPI=subset(WEN_umap,cell_type%in%c("Unciliated epithelia 1","Unciliated epithelia 2","Ciliated"))

EnEc_Adhesion=subset(EnEc,type2%in%"IUA2")

EnEc_Adhesion$donor="proliferative"

WEN_EPI$type2="WEN_EPI"

all2=merge(EnEc_Adhesion,WEN_EPI)

all2 <- CreateSeuratObject(counts =all2@assays$RNA@counts, project = "all2", min.cells = 3,meta.data = all2@meta.data)

VlnPlot(all2, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

all2 <- NormalizeData(all2, normalization.method = "LogNormalize", 
                      scale.factor = 10000)
all2 <- FindVariableFeatures(all2, selection.method = "vst", nfeatures = 2000)
VlnPlot(all2, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

#all2进行标准化然后PCA分析
DefaultAssay(all2) <- "RNA"
all.genes_all2 <- rownames(all2)
all2 <- ScaleData(object = all2, features = all.genes_all2)
all2 <- RunPCA(all2, features = VariableFeatures(object = all2))


#all2进行细胞细胞聚类分析(UMAP)
all2=RunUMAP(all2, dims = 1:30)
DimPlot(all2,reduction="umap",label=TRUE)
all2 <- FindNeighbors(all2, dims = 1:30)
all2 <- FindClusters(all2, resolution = 0.3)
all2 <- FindClusters(all2, resolution = 0.6)
DimPlot(all2,reduction="umap",label=TRUE,group.by = "donor")



#收集上面all2的marker基因和高可变基因
#收集起来对count矩阵提取，这样就可以节省后续的运行时间
all2@active.ident=as.factor(all2$donor)
all2_markers <- FindAllMarkers(object=all2, only.pos = TRUE,min.pct = 0.25, 
                               logfc.threshold = 0.25)
all2_markers <-subset(all2_markers,all2_markers$p_val<0.05)

all2_markers=all2_markers$gene

hvg_genes <- VariableFeatures(all2)

gene3=c(all2_markers,hvg_genes)


##开始monocle2分析[Adhesion-上皮]
library(monocle)

DefaultAssay(all2)<-"RNA"

all2_matrix=as.data.frame(all2@assays$RNA@counts)

all2_matrix=all2_matrix[gene3,]

all2_matrix<-as(as.matrix(all2_matrix),'sparseMatrix')



all2_ann<-data.frame(gene_id=rownames(all2_matrix),gene_short_name=rownames(all2_matrix))

rownames(all2_ann)<-rownames(all2_matrix)

all2_fd<-new("AnnotatedDataFrame", data = all2_ann)

phenoData <- all2@meta.data
phenoData$celltype <- all2$donor
all2_pd <- new('AnnotatedDataFrame', data = phenoData)



cds<-newCellDataSet(all2_matrix,phenoData =all2_pd,
                    featureData =all2_fd,expressionFamily=negbinomial.size())

cds <- estimateSizeFactors(cds)
cds <- estimateDispersions(cds)

cds<-detectGenes(cds,min_expr=0.1)
expressed_genes <- row.names(subset(fData(cds),
                                    num_cells_expressed>=10))

diff_test_res <- differentialGeneTest(cds[expressed_genes,],                                      
                                      fullModelFormulaStr = "~donor",cores=16)



sig_genes<-subset(diff_test_res,qval<0.01)
sig_genes <- sig_genes[order(sig_genes$pval),]


# 标记为用于细胞排序的基因
cds <- setOrderingFilter(cds, sig_genes$gene_short_name)


all2@active.ident=as.factor(all2$donor)

express_genes <- VariableFeatures(all2)

cds <- setOrderingFilter(cds[expressed_genes,],                          
                         express_genes)

disp_table <- dispersionTable(cds)
disp.genes <- subset(disp_table, mean_expression>=0.1&dispersion_empirical>=1*dispersion_fit)$gene_id

cds <- setOrderingFilter(cds[expressed_genes,],                          
                         disp.genes) 

cds <- reduceDimension(cds, max_components=2,                       
                       reduction_method = 'DDRTree',                        
                       verbose = T)

cds <- orderCells(cds)

saveRDS(cds,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/Adhesion_EPI.rds")

cds=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/Adhesion_EPI.rds")



P1=plot_cell_trajectory(cds,cell_size = 1,color_by = "donor",label=T)+ggtitle("")+
  facet_wrap(~donor)

P2=plot_cell_trajectory(cds,cell_size = 1,color_by = "Pseudotime")

P1+P2

cds$donor=factor(cds$donor,levels = c("proliferative","secretory_early",
                                      "secretory_early-mid","secretory_mid","secretory_late"))

plot_cell_trajectory(cds,cell_size = 1,color_by = "donor",label=T)+ggtitle("")+
  facet_wrap(~donor)

plot_cell_trajectory(cds,cell_size = 1,color_by = "donor",label=T)+ggtitle("")+
  facet_wrap(~type2)

plot_cell_trajectory(cds,cell_size = 1,color_by = "cell_type",label=T)+ggtitle("")+
  facet_wrap(~cell_type)

#将我们的所有非纤毛上皮和文章不同周期的上皮进行monocle2分析----
library(monocle)
library(Seurat)
library(harmony)
library(dplyr)
library(SeuratDisk)

###Ctrl-------
WEN_umap=readRDS("/mnt/chentw/zigongneimo/photo/文献数据操作数据/WEN.umap.rds")
EnEc=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/EPI_allsingle.rds")

EnEc=subset(EnEc,cell_type%in%c("Cycling epithelial cells","Epithelials cells"))

WEN_EPI=subset(WEN_umap,cell_type%in%c("Unciliated epithelia 1","Unciliated epithelia 2"))

EnEc_Ctrl=subset(EnEc,type2%in%"CTRL")

EnEc_Ctrl$donor="proliferative"

WEN_EPI$type2="WEN_EPI"

all2=merge(EnEc_Ctrl,WEN_EPI)

all2 <- CreateSeuratObject(counts =all2@assays$RNA@counts, project = "all2", min.cells = 3,meta.data = all2@meta.data)

VlnPlot(all2, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

all2 <- NormalizeData(all2, normalization.method = "LogNormalize", 
                      scale.factor = 10000)
all2 <- FindVariableFeatures(all2, selection.method = "vst", nfeatures = 2000)
VlnPlot(all2, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

#all2进行标准化然后PCA分析
DefaultAssay(all2) <- "RNA"
all.genes_all2 <- rownames(all2)
all2 <- ScaleData(object = all2, features = all.genes_all2)
all2 <- RunPCA(all2, features = VariableFeatures(object = all2))


#all2进行细胞细胞聚类分析(UMAP)
all2=RunUMAP(all2, dims = 1:30)
DimPlot(all2,reduction="umap",label=TRUE)
all2 <- FindNeighbors(all2, dims = 1:30)
all2 <- FindClusters(all2, resolution = 0.3)
all2 <- FindClusters(all2, resolution = 0.6)
DimPlot(all2,reduction="umap",label=TRUE,group.by = "donor")



#收集上面all2的marker基因和高可变基因
#收集起来对count矩阵提取，这样就可以节省后续的运行时间
all2@active.ident=as.factor(all2$donor)
all2_markers <- FindAllMarkers(object=all2, only.pos = TRUE,min.pct = 0.25, 
                               logfc.threshold = 0.25)
all2_markers <-subset(all2_markers,all2_markers$p_val<0.05)

all2_markers=all2_markers$gene

hvg_genes <- VariableFeatures(all2)

gene3=c(all2_markers,hvg_genes)


##开始monocle2分析[ctrl-上皮]
library(monocle)

DefaultAssay(all2)<-"RNA"

all2_matrix=as.data.frame(all2@assays$RNA@counts)

all2_matrix=all2_matrix[gene3,]

all2_matrix<-as(as.matrix(all2_matrix),'sparseMatrix')



all2_ann<-data.frame(gene_id=rownames(all2_matrix),gene_short_name=rownames(all2_matrix))

rownames(all2_ann)<-rownames(all2_matrix)

all2_fd<-new("AnnotatedDataFrame", data = all2_ann)

phenoData <- all2@meta.data
phenoData$celltype <- all2$donor
all2_pd <- new('AnnotatedDataFrame', data = phenoData)



cds<-newCellDataSet(all2_matrix,phenoData =all2_pd,
                    featureData =all2_fd,expressionFamily=negbinomial.size())

cds <- estimateSizeFactors(cds)
cds <- estimateDispersions(cds)

cds<-detectGenes(cds,min_expr=0.1)
expressed_genes <- row.names(subset(fData(cds),
                                    num_cells_expressed>=10))

diff_test_res <- differentialGeneTest(cds[expressed_genes,],                                      
                                      fullModelFormulaStr = "~donor",cores=16)



sig_genes<-subset(diff_test_res,qval<0.01)
sig_genes <- sig_genes[order(sig_genes$pval),]


# 标记为用于细胞排序的基因
cds <- setOrderingFilter(cds, sig_genes$gene_short_name)


all2@active.ident=as.factor(all2$donor)

express_genes <- VariableFeatures(all2)

cds <- setOrderingFilter(cds[expressed_genes,],                          
                         express_genes)

disp_table <- dispersionTable(cds)
disp.genes <- subset(disp_table, mean_expression>=0.1&dispersion_empirical>=1*dispersion_fit)$gene_id

cds <- setOrderingFilter(cds[expressed_genes,],                          
                         disp.genes) 

cds <- reduceDimension(cds, max_components=2,                       
                       reduction_method = 'DDRTree',                        
                       verbose = T)

cds <- orderCells(cds)

saveRDS(cds,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/ctrl_Uncili_EPI.rds")

cds=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/ctrl_Uncili_EPI.rds")


P1=plot_cell_trajectory(cds,cell_size = 1,color_by = "donor",label=T)+ggtitle("")+
  facet_wrap(~donor)

P2=plot_cell_trajectory(cds,cell_size = 1,color_by = "Pseudotime")

P1+P2

cds=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/ctrl_Uncili_EPI.rds")



P1=plot_cell_trajectory(cds,cell_size = 1,color_by = "donor",label=T)+ggtitle("")+
  facet_wrap(~donor)

P2=plot_cell_trajectory(cds,cell_size = 1,color_by = "Pseudotime")

P1+P2

cds$donor=factor(cds$donor,levels = c("proliferative","secretory_early",
                                      "secretory_early-mid","secretory_mid","secretory_late"))

plot_cell_trajectory(cds,cell_size = 1,color_by = "donor",label=T)+ggtitle("")+
  facet_wrap(~donor)

plot_cell_trajectory(cds,cell_size = 1,color_by = "donor",label=T)+ggtitle("")+
  facet_wrap(~type2)

plot_cell_trajectory(cds,cell_size = 1,color_by = "cell_type",label=T)+ggtitle("")+
  facet_wrap(~cell_type)


###Adhesion-------
library(monocle)
library(Seurat)
library(harmony)
library(dplyr)
library(SeuratDisk)
WEN_umap=readRDS("/mnt/chentw/zigongneimo/photo/文献数据操作数据/WEN.umap.rds")
EnEc=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/EPI_allsingle.rds")

EnEc=subset(EnEc,cell_type%in%c("Cycling epithelial cells","Epithelials cells"))

WEN_EPI=subset(WEN_umap,cell_type%in%c("Unciliated epithelia 1","Unciliated epithelia 2"))

EnEc_Adhesion=subset(EnEc,type2%in%"IUA2")

EnEc_Adhesion$donor="proliferative"

WEN_EPI$type2="WEN_EPI"

all2=merge(EnEc_Adhesion,WEN_EPI)

all2 <- CreateSeuratObject(counts =all2@assays$RNA@counts, project = "all2", min.cells = 3,meta.data = all2@meta.data)

VlnPlot(all2, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

all2 <- NormalizeData(all2, normalization.method = "LogNormalize", 
                      scale.factor = 10000)
all2 <- FindVariableFeatures(all2, selection.method = "vst", nfeatures = 2000)
VlnPlot(all2, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

#all2进行标准化然后PCA分析
DefaultAssay(all2) <- "RNA"
all.genes_all2 <- rownames(all2)
all2 <- ScaleData(object = all2, features = all.genes_all2)
all2 <- RunPCA(all2, features = VariableFeatures(object = all2))


#all2进行细胞细胞聚类分析(UMAP)
all2=RunUMAP(all2, dims = 1:30)
DimPlot(all2,reduction="umap",label=TRUE)
all2 <- FindNeighbors(all2, dims = 1:30)
all2 <- FindClusters(all2, resolution = 0.3)
all2 <- FindClusters(all2, resolution = 0.6)
DimPlot(all2,reduction="umap",label=TRUE,group.by = "donor")



#收集上面all2的marker基因和高可变基因
#收集起来对count矩阵提取，这样就可以节省后续的运行时间
all2@active.ident=as.factor(all2$donor)
all2_markers <- FindAllMarkers(object=all2, only.pos = TRUE,min.pct = 0.25, 
                               logfc.threshold = 0.25)
all2_markers <-subset(all2_markers,all2_markers$p_val<0.05)

all2_markers=all2_markers$gene

hvg_genes <- VariableFeatures(all2)

gene3=c(all2_markers,hvg_genes)


##开始monocle2分析[Adhesion-上皮]
library(monocle)

DefaultAssay(all2)<-"RNA"

all2_matrix=as.data.frame(all2@assays$RNA@counts)

all2_matrix=all2_matrix[gene3,]

all2_matrix<-as(as.matrix(all2_matrix),'sparseMatrix')



all2_ann<-data.frame(gene_id=rownames(all2_matrix),gene_short_name=rownames(all2_matrix))

rownames(all2_ann)<-rownames(all2_matrix)

all2_fd<-new("AnnotatedDataFrame", data = all2_ann)

phenoData <- all2@meta.data
phenoData$celltype <- all2$donor
all2_pd <- new('AnnotatedDataFrame', data = phenoData)



cds<-newCellDataSet(all2_matrix,phenoData =all2_pd,
                    featureData =all2_fd,expressionFamily=negbinomial.size())

cds <- estimateSizeFactors(cds)
cds <- estimateDispersions(cds)

cds<-detectGenes(cds,min_expr=0.1)
expressed_genes <- row.names(subset(fData(cds),
                                    num_cells_expressed>=10))

diff_test_res <- differentialGeneTest(cds[expressed_genes,],                                      
                                      fullModelFormulaStr = "~donor",cores=16)



sig_genes<-subset(diff_test_res,qval<0.01)
sig_genes <- sig_genes[order(sig_genes$pval),]


# 标记为用于细胞排序的基因
cds <- setOrderingFilter(cds, sig_genes$gene_short_name)


all2@active.ident=as.factor(all2$donor)

express_genes <- VariableFeatures(all2)

cds <- setOrderingFilter(cds[expressed_genes,],                          
                         express_genes)

disp_table <- dispersionTable(cds)
disp.genes <- subset(disp_table, mean_expression>=0.1&dispersion_empirical>=1*dispersion_fit)$gene_id

cds <- setOrderingFilter(cds[expressed_genes,],                          
                         disp.genes) 

cds <- reduceDimension(cds, max_components=2,                       
                       reduction_method = 'DDRTree',                        
                       verbose = T)

cds <- orderCells(cds)

saveRDS(cds,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/Adhesion_Uncili_EPI.rds")

cds=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/Adhesion_Uncili_EPI.rds")



P1=plot_cell_trajectory(cds,cell_size = 1,color_by = "donor",label=T)+ggtitle("")+
  facet_wrap(~donor)

P2=plot_cell_trajectory(cds,cell_size = 1,color_by = "Pseudotime")

P1+P2

cds$donor=factor(cds$donor,levels = c("proliferative","secretory_early",
                                      "secretory_early-mid","secretory_mid","secretory_late"))

plot_cell_trajectory(cds,cell_size = 1,color_by = "Pseudotime")

plot_cell_trajectory(cds,cell_size = 1,color_by = "donor",label=T)+ggtitle("")+
  facet_wrap(~donor)

plot_cell_trajectory(cds,cell_size = 1,color_by = "donor",label=T)+ggtitle("")+
  facet_wrap(~type2)

plot_cell_trajectory(cds,cell_size = 1,color_by = "cell_type",label=T)+ggtitle("")+
  facet_wrap(~cell_type)

#all-----
all2=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/rds/all2.rds")

#收集上面all2的marker基因和高可变基因
#收集起来对count矩阵提取，这样就可以节省后续的运行时间
all2@active.ident=as.factor(all2$donor)
all2_markers <- FindAllMarkers(object=all2, only.pos = TRUE,min.pct = 0.25, 
                               logfc.threshold = 0.25)
all2_markers <-subset(all2_markers,all2_markers$p_val<0.05)

all2_markers=all2_markers$gene

hvg_genes <- VariableFeatures(all2)

gene3=c(all2_markers,hvg_genes)


##开始monocle2分析[Adhesion-上皮]
library(monocle)

DefaultAssay(all2)<-"RNA"

all2_matrix=as.data.frame(all2@assays$RNA@counts)

all2_matrix=all2_matrix[gene3,]

all2_matrix<-as(as.matrix(all2_matrix),'sparseMatrix')



all2_ann<-data.frame(gene_id=rownames(all2_matrix),gene_short_name=rownames(all2_matrix))

rownames(all2_ann)<-rownames(all2_matrix)

all2_fd<-new("AnnotatedDataFrame", data = all2_ann)

phenoData <- all2@meta.data
phenoData$celltype <- all2$donor
all2_pd <- new('AnnotatedDataFrame', data = phenoData)



cds<-newCellDataSet(all2_matrix,phenoData =all2_pd,
                    featureData =all2_fd,expressionFamily=negbinomial.size())

cds <- estimateSizeFactors(cds)
cds <- estimateDispersions(cds)

cds<-detectGenes(cds,min_expr=0.1)
expressed_genes <- row.names(subset(fData(cds),
                                    num_cells_expressed>=10))

diff_test_res <- differentialGeneTest(cds[expressed_genes,],                                      
                                      fullModelFormulaStr = "~donor")



sig_genes<-subset(diff_test_res,qval<0.01)
sig_genes <- sig_genes[order(sig_genes$pval),]

saveRDS(cds,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/all_Uncili_EPI.rds")



# 标记为用于细胞排序的基因
cds <- setOrderingFilter(cds, sig_genes$gene_short_name)


all2@active.ident=as.factor(all2$donor)

express_genes <- VariableFeatures(all2)

cds <- setOrderingFilter(cds[expressed_genes,],                          
                         express_genes)

disp_table <- dispersionTable(cds)
disp.genes <- subset(disp_table, mean_expression>=0.1&dispersion_empirical>=1*dispersion_fit)$gene_id

cds <- setOrderingFilter(cds[expressed_genes,],                          
                         disp.genes) 

cds <- reduceDimension(cds, max_components=2,                       
                       reduction_method = 'DDRTree',                        
                       verbose = T)

cds <- orderCells(cds)

saveRDS(cds,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/all_Uncili_EPI.rds")



cds=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/all_Uncili_EPI.rds")



P1=plot_cell_trajectory(cds,cell_size = 1,color_by = "donor",label=T)+ggtitle("")+
  facet_wrap(~donor)

P2=plot_cell_trajectory(cds,cell_size = 1,color_by = "Pseudotime")

P1+P2

cds$donor=factor(cds$donor,levels = c("proliferative","secretory_early",
                                      "secretory_early-mid","secretory_mid","secretory_late"))

plot_cell_trajectory(cds,cell_size = 1,color_by = "Pseudotime")

plot_cell_trajectory(cds,cell_size = 1,color_by = "donor",label=T)+ggtitle("")+
  facet_wrap(~donor)

plot_cell_trajectory(cds,cell_size = 1,color_by = "donor",label=T)+ggtitle("")+
  facet_wrap(~type2)

plot_cell_trajectory(cds,cell_size = 1,color_by = "cell_type",label=T)+ggtitle("")+
  facet_wrap(~cell_type)

#对非纤毛的monocle2结果进行WOI基因的两组查看和比较-----
Ctrl=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/ctrl_Uncili_EPI.rds")
Adhesion=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/Adhesion_Uncili_EPI.rds")
WOI_maker=read.csv("/data/DATAprocess/chentw/TJW_scRNA2/20230714/WOI.csv")


WOI=c(WOI_maker$UP,WOI_maker$DOWN[1:95])




WOI_UP=WOI_maker$UP
WOI_DOWN=WOI_maker$DOWN



gene1=WOI_UP[1:12]
gene2=WOI_UP[13:24]
gene3=WOI_UP[25:36]
gene4=WOI_UP[37:48]
gene5=WOI_UP[49:60]
gene6=WOI_UP[61:72]
gene7=WOI_UP[73:84]
gene8=WOI_UP[85:95]


#Ctrl组
Ctrl$donor[Ctrl$type2%in%"CTRL"]<-"Ctrl Pro"


plot_cell_trajectory(Ctrl,cell_size = 1,color_by = "Pseudotime")

plot_cell_trajectory(Ctrl,cell_size = 1,color_by = "donor")+
  facet_wrap(~donor)




plot_genes_in_pseudotime(Ctrl[gene1,], color_by = "donor",ncol = 2)






















#对上述两个分组进行WOI基因的评分----
WEN_umap=readRDS("/mnt/chentw/zigongneimo/photo/文献数据操作数据/WEN.umap.rds")
EnEc=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/EPI_allsingle.rds")

EnEc=subset(EnEc,cell_type%in%c("Cycling epithelial cells","Epithelials cells"))

WEN_EPI=subset(WEN_umap,cell_type%in%c("Unciliated epithelia 1","Unciliated epithelia 2"))

EnEc_Ctrl=subset(EnEc,type2%in%"CTRL")

EnEc_Ctrl$donor="proliferative"

WEN_EPI$type2="WEN_EPI"

all2=merge(EnEc_Ctrl,WEN_EPI)

all2 <- CreateSeuratObject(counts =all2@assays$RNA@counts, project = "all2", min.cells = 3,meta.data = all2@meta.data)

VlnPlot(all2, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

all2 <- NormalizeData(all2, normalization.method = "LogNormalize", 
                      scale.factor = 10000)
all2 <- FindVariableFeatures(all2, selection.method = "vst", nfeatures = 2000)
VlnPlot(all2, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

#all2进行标准化然后PCA分析
DefaultAssay(all2) <- "RNA"
all.genes_all2 <- rownames(all2)
all2 <- ScaleData(object = all2, features = all.genes_all2)
all2 <- RunPCA(all2, features = VariableFeatures(object = all2))


#all2进行细胞细胞聚类分析(UMAP)
all2=RunUMAP(all2, dims = 1:30)
DimPlot(all2,reduction="umap",label=TRUE)
all2 <- FindNeighbors(all2, dims = 1:30)
all2 <- FindClusters(all2, resolution = 0.3)
all2 <- FindClusters(all2, resolution = 0.6)
DimPlot(all2,reduction="umap",label=TRUE,group.by = "donor")
Ctrl=all2




WEN_umap=readRDS("/mnt/chentw/zigongneimo/photo/文献数据操作数据/WEN.umap.rds")
EnEc=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/EPI_allsingle.rds")

WEN_EPI=subset(WEN_umap,cell_type%in%c("Unciliated epithelia 1","Unciliated epithelia 2","Ciliated"))

EnEc_Adhesion=subset(EnEc,type2%in%"IUA2")

EnEc_Adhesion$donor="proliferative"

WEN_EPI$type2="WEN_EPI"

all2=merge(EnEc_Adhesion,WEN_EPI)

all2 <- CreateSeuratObject(counts =all2@assays$RNA@counts, project = "all2", min.cells = 3,meta.data = all2@meta.data)

VlnPlot(all2, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

all2 <- NormalizeData(all2, normalization.method = "LogNormalize", 
                      scale.factor = 10000)
all2 <- FindVariableFeatures(all2, selection.method = "vst", nfeatures = 2000)
VlnPlot(all2, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

#all2进行标准化然后PCA分析
DefaultAssay(all2) <- "RNA"
all.genes_all2 <- rownames(all2)
all2 <- ScaleData(object = all2, features = all.genes_all2)
all2 <- RunPCA(all2, features = VariableFeatures(object = all2))


#all2进行细胞细胞聚类分析(UMAP)
all2=RunUMAP(all2, dims = 1:30)
DimPlot(all2,reduction="umap",label=TRUE)
all2 <- FindNeighbors(all2, dims = 1:30)
all2 <- FindClusters(all2, resolution = 0.3)
all2 <- FindClusters(all2, resolution = 0.6)
DimPlot(all2,reduction="umap",label=TRUE,group.by = "donor")

Adhesion=all2





#对两组进行评分并进行
WOI_maker=read.csv("/data/DATAprocess/chentw/TJW_scRNA2/20230714/WOI.csv")


WOI=c(WOI_maker$UP,WOI_maker$DOWN[1:95])














#对三组基质细胞进行monocle2分析-----
WEN_umap=readRDS("/mnt/chentw/zigongneimo/photo/文献数据操作数据/WEN.umap.rds")
EnSc=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0718.rds")

WEN_STR=subset(WEN_umap,cell_type%in%"Stromal fibroblasts")

WEN_STR$type3=WEN_STR$orig.ident

EnSc$donor="proliferative"

WEN_STR$type2="WEN_STR"

all2=merge(EnSc,WEN_STR)

all2 <- CreateSeuratObject(counts =all2@assays$RNA@counts, project = "all2", min.cells = 3,meta.data = all2@meta.data)

VlnPlot(all2, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

all2 <- NormalizeData(all2, normalization.method = "LogNormalize", 
                      scale.factor = 10000)
all2 <- FindVariableFeatures(all2, selection.method = "vst", nfeatures = 2000)
VlnPlot(all2, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

#all2进行标准化然后PCA分析
DefaultAssay(all2) <- "RNA"
all.genes_all2 <- rownames(all2)
all2 <- ScaleData(object = all2, features = all.genes_all2)
all2 <- RunPCA(all2, features = VariableFeatures(object = all2))


#all2进行细胞细胞聚类分析(UMAP)
all2=RunUMAP(all2, dims = 1:30)
DimPlot(all2,reduction="umap",label=TRUE)
all2 <- FindNeighbors(all2, dims = 1:30)
all2 <- FindClusters(all2, resolution = 0.3)
all2 <- FindClusters(all2, resolution = 0.6)
DimPlot(all2,reduction="umap",label=TRUE,group.by = "donor")

DimPlot(all2,reduction="umap",label=TRUE,group.by = "type3")


saveRDS(all2,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/rds/EnSc_all2.rds")


#Ctrl组-----
#固定数目位1500一个时期
all2=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/rds/EnSc_all2.rds")

all2=subset(all2,type2%in%c("CTRL","WEN_STR"))

all2$donor[all2$type2%in%"CTRL"]<-"Pro"

all2@active.ident=as.factor(all2$donor)
all2 <- subset(all2,downsample = 1500)
DefaultAssay(all2)<-"RNA"

DimPlot(all2,reduction="umap",label=TRUE,group.by = "donor")


all2@active.ident=as.factor(all2$donor)
all2_markers <- FindAllMarkers(object=all2, only.pos = TRUE,min.pct = 0.25, 
                               logfc.threshold = 0.25)
all2_markers <-subset(all2_markers,all2_markers$p_val<0.05)

all2_markers=all2_markers$gene

hvg_genes <- VariableFeatures(all2)

gene3=c(all2_markers)


##开始monocle2分析[ctrl-上皮]
library(monocle)

DefaultAssay(all2)<-"RNA"

all2_matrix=as.data.frame(all2@assays$RNA@counts)

all2_matrix=all2_matrix[gene3,]

all2_matrix<-as(as.matrix(all2_matrix),'sparseMatrix')



all2_ann<-data.frame(gene_id=rownames(all2_matrix),gene_short_name=rownames(all2_matrix))

rownames(all2_ann)<-rownames(all2_matrix)

all2_fd<-new("AnnotatedDataFrame", data = all2_ann)

phenoData <- all2@meta.data
phenoData$celltype <- all2$donor
all2_pd <- new('AnnotatedDataFrame', data = phenoData)



cds<-newCellDataSet(all2_matrix,phenoData =all2_pd,
                    featureData =all2_fd,expressionFamily=negbinomial.size())

cds <- estimateSizeFactors(cds)
cds <- estimateDispersions(cds)

cds<-detectGenes(cds,min_expr=0.1)


diff_test_res <- differentialGeneTest(cds,                                      
                                      fullModelFormulaStr = "~donor",cores=16)

ordering_genes<-row.names(subset(diff_test_res,qval<0.01))

cds<-setOrderingFilter(cds,ordering_genes)

cds <- reduceDimension(cds, max_components=2,                       
                       reduction_method = 'DDRTree',                        
                       verbose = F)

cds <- orderCells(cds)

saveRDS(cds,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/ctrl_EnSc.rds")


cds=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/ctrl_EnSc.rds")


P1=plot_cell_trajectory(cds,cell_size = 1,color_by = "donor",label=T)+ggtitle("")+
  facet_wrap(~donor)

P2=plot_cell_trajectory(cds,cell_size = 1,color_by = "Pseudotime")

P1+P2

cds$donor=factor(cds$donor,levels = c("proliferative","secretory_early",
                                      "secretory_early-mid","secretory_mid","secretory_late"))

plot_cell_trajectory(cds,cell_size = 1,color_by = "Pseudotime")

plot_cell_trajectory(cds,cell_size = 1,color_by = "donor",label=T)+ggtitle("")+
  facet_wrap(~donor)

plot_cell_trajectory(cds,cell_size = 1,color_by = "donor",label=T)+ggtitle("")+
  facet_wrap(~type2)

plot_cell_trajectory(cds,cell_size = 1,color_by = "cell_type",label=T)+ggtitle("")+
  facet_wrap(~cell_type)





#Adhesion组-----
#固定数目位1500一个时期
all2=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/rds/EnSc_all2.rds")

all2=subset(all2,type2%in%c("IUA2","WEN_STR"))

all2$donor[all2$type2%in%"IUA2"]<-"Pro"

all2@active.ident=as.factor(all2$donor)
all2 <- subset(all2,downsample = 1500)
DefaultAssay(all2)<-"RNA"

DimPlot(all2,reduction="umap",label=TRUE,group.by = "donor")


all2@active.ident=as.factor(all2$donor)
all2_markers <- FindAllMarkers(object=all2, only.pos = TRUE,min.pct = 0.25, 
                               logfc.threshold = 0.25)
all2_markers <-subset(all2_markers,all2_markers$p_val<0.05)

all2_markers=all2_markers$gene

hvg_genes <- VariableFeatures(all2)

gene3=c(all2_markers)


##开始monocle2分析[ctrl-上皮]
library(monocle)

DefaultAssay(all2)<-"RNA"

all2_matrix=as.data.frame(all2@assays$RNA@counts)

all2_matrix=all2_matrix[gene3,]

all2_matrix<-as(as.matrix(all2_matrix),'sparseMatrix')



all2_ann<-data.frame(gene_id=rownames(all2_matrix),gene_short_name=rownames(all2_matrix))

rownames(all2_ann)<-rownames(all2_matrix)

all2_fd<-new("AnnotatedDataFrame", data = all2_ann)

phenoData <- all2@meta.data
phenoData$celltype <- all2$donor
all2_pd <- new('AnnotatedDataFrame', data = phenoData)



cds<-newCellDataSet(all2_matrix,phenoData =all2_pd,
                    featureData =all2_fd,expressionFamily=negbinomial.size())

cds <- estimateSizeFactors(cds)
cds <- estimateDispersions(cds)

cds<-detectGenes(cds,min_expr=0.1)


diff_test_res <- differentialGeneTest(cds,                                      
                                      fullModelFormulaStr = "~donor",cores=16)

ordering_genes<-row.names(subset(diff_test_res,qval<0.01))
ordering_genes
cds<-setOrderingFilter(cds,ordering_genes)

cds <- reduceDimension(cds, max_components=2,                       
                       reduction_method = 'DDRTree',                        
                       verbose = F)

cds <- orderCells(cds)

saveRDS(cds,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/Adhesion_EnSc.rds")



cds=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/Adhesion_EnSc.rds")


P1=plot_cell_trajectory(cds,cell_size = 1,color_by = "donor",label=T)+ggtitle("")+
  facet_wrap(~donor)

P2=plot_cell_trajectory(cds,cell_size = 1,color_by = "Pseudotime")

P1+P2

cds$donor=factor(cds$donor,levels = c("proliferative","secretory_early",
                                      "secretory_early-mid","secretory_mid","secretory_late"))

plot_cell_trajectory(cds,cell_size = 1,color_by = "Pseudotime")

plot_cell_trajectory(cds,cell_size = 1,color_by = "donor",label=T)+ggtitle("")+
  facet_wrap(~donor)

plot_cell_trajectory(cds,cell_size = 1,color_by = "donor",label=T)+ggtitle("")+
  facet_wrap(~type2)

plot_cell_trajectory(cds,cell_size = 1,color_by = "cell_type",label=T)+ggtitle("")+
  facet_wrap(~cell_type)



#all组----
all2=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/rds/EnSc_all2.rds")

all2$donor[all2$type2%in%"CTRL"]<-"Ctrl"
all2$donor[all2$type2%in%"IUA1"]<-"Adjacency"
all2$donor[all2$type2%in%"IUA2"]<-"Adhesion"


all2 <- subset(all2,downsample = 1500)
DefaultAssay(all2)<-"RNA"

DimPlot(all2,reduction="umap",label=TRUE,group.by = "donor")


all2@active.ident=as.factor(all2$donor)
all2_markers <- FindAllMarkers(object=all2, only.pos = TRUE,min.pct = 0.25, 
                               logfc.threshold = 0.25)
all2_markers <-subset(all2_markers,all2_markers$p_val<0.05)

all2_markers=all2_markers$gene

hvg_genes <- VariableFeatures(all2)

gene3=c(all2_markers)


##开始monocle2分析[ctrl-上皮]
library(monocle)

DefaultAssay(all2)<-"RNA"

all2_matrix=as.data.frame(all2@assays$RNA@counts)

all2_matrix=all2_matrix[gene3,]

all2_matrix<-as(as.matrix(all2_matrix),'sparseMatrix')



all2_ann<-data.frame(gene_id=rownames(all2_matrix),gene_short_name=rownames(all2_matrix))

rownames(all2_ann)<-rownames(all2_matrix)

all2_fd<-new("AnnotatedDataFrame", data = all2_ann)

phenoData <- all2@meta.data
phenoData$celltype <- all2$donor
all2_pd <- new('AnnotatedDataFrame', data = phenoData)



cds<-newCellDataSet(all2_matrix,phenoData =all2_pd,
                    featureData =all2_fd,expressionFamily=negbinomial.size())

cds <- estimateSizeFactors(cds)
cds <- estimateDispersions(cds)

cds<-detectGenes(cds,min_expr=0.1)


diff_test_res <- differentialGeneTest(cds,                                      
                                      fullModelFormulaStr = "~donor",cores=16)

ordering_genes<-row.names(subset(diff_test_res,qval<0.01))
ordering_genes
cds<-setOrderingFilter(cds,ordering_genes)

cds <- reduceDimension(cds, max_components=2,                       
                       reduction_method = 'DDRTree',                        
                       verbose = F)

cds <- orderCells(cds)

saveRDS(cds,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/all_EnSc.rds")



cds=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/Adhesion_EnSc.rds")


P1=plot_cell_trajectory(cds,cell_size = 1,color_by = "donor",label=T)+ggtitle("")+
  facet_wrap(~donor)

P2=plot_cell_trajectory(cds,cell_size = 1,color_by = "Pseudotime")

P1+P2

cds$donor=factor(cds$donor,levels = c("proliferative","secretory_early",
                                      "secretory_early-mid","secretory_mid","secretory_late"))

plot_cell_trajectory(cds,cell_size = 1,color_by = "Pseudotime")

plot_cell_trajectory(cds,cell_size = 1,color_by = "donor",label=T)+ggtitle("")+
  facet_wrap(~donor)

plot_cell_trajectory(cds,cell_size = 1,color_by = "type2",label=T)+ggtitle("")+
  facet_wrap(~type2)

plot_cell_trajectory(cds,cell_size = 1,color_by = "cell_type",label=T)+ggtitle("")+
  facet_wrap(~cell_type)

#对三组基质细胞进行monocle3分析-----
#Ctrl组-----
all2=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/rds/EnSc_all2.rds")


table(all2$donor)


DimPlot(all2,group.by = "donor",label = T)

library(monocle3)
table(all2$type2)

all2$donor[all2$type2%in%"CTRL"]<-"Ctrl"
all2$donor[all2$type2%in%"IUA1"]<-"Adjacency"
all2$donor[all2$type2%in%"IUA2"]<-"Adhesion"


all2@active.ident=as.factor(all2$donor)

all2 <- subset(all2,downsample = 1500)

table(all2$donor)



Ctrl=subset(all2,type2%in%c("CTRL","WEN_STR"))

expression_matrix=Ctrl@assays$RNA@data
cell_metadata=data.frame(Ctrl@meta.data)
gene_annotation=data.frame(expression_matrix[,1])
gene_annotation[,1]=row.names(gene_annotation)
colnames(gene_annotation)=c("gene_short_name")
##构建Monocle3cds对象
cds<-new_cell_data_set(expression_matrix,
                       cell_metadata=cell_metadata,
                       gene_metadata=gene_annotation)

#预处理，相当于Seurat流程normalize过程
#counts数据选择norm_method=c("log")
#data数据选择norm_method=c("none")
cds<-preprocess_cds(cds,num_dim=50,norm_method=c("none"))

#去除批次效应,多个样本时可以通过此方式去除批次
cds<-align_cds(cds,alignment_group="orig.ident")

##降维，默认是"Umap"方式
cds<-reduce_dimension(cds,cores=5)

##聚类分群，分辨率调小，是为了让细胞是一群可以更好展示
cds<-cluster_cells(cds,resolution=0.0000001)

##拟时序
cds<-learn_graph(cds)




myselect<-function(cds,select.classify,my_select){
  cell_ids<-which(colData(cds)[,select.classify]==my_select)
  closest_vertex<-
    cds@principal_graph_aux[["UMAP"]]$pr_graph_cell_proj_closest_vertex
  closest_vertex<-as.matrix(closest_vertex[colnames(cds),])
  root_pr_nodes<-
    igraph::V(principal_graph(cds)[["UMAP"]])$name[as.numeric(names
                                                              (which.max(table(closest_vertex[cell_ids,]))))]
  root_pr_nodes}

cds<-order_cells(cds,root_pr_nodes=myselect(cds,select.classify='donor',my_select="Ctrl"))

cds.embed<-cds@int_colData$reducedDims$UMAP
int.embed<-Embeddings(Ctrl,reduction="umap")
int.embed<-int.embed[rownames(cds.embed),]
cds@int_colData$reducedDims$UMAP<-int.embed

##不同细胞类型拟时序数值
##拟时序值越高表示细胞分化程度越高，这里仅为演示，并非真实分化情况
plot_cells(cds,color_cells_by="pseudotime",
           show_trajectory_graph=T)+plot_cells(cds,
                                               color_cells_by="donor",
                                               label_cell_groups=FALSE,
                                               label_leaves=FALSE,
                                               label_branch_points=FALSE,
                                               graph_label_size=1)+scale_color_manual(values=pal_jco("default",alpha=0.6)(9))


#对非纤毛上皮进行velocity分析------
###前期文件准备-----
WEN_umap=readRDS("/mnt/chentw/zigongneimo/photo/文献数据操作数据/WEN.umap.rds")
EnEc=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/EPI_allsingle.rds")

EnEc=subset(EnEc,cell_type%in%c("Cycling epithelial cells","Epithelials cells"))

WEN_EPI=subset(WEN_umap,cell_type%in%c("Unciliated epithelia 1","Unciliated epithelia 2"))


EnEc$donor="proliferative"

WEN_EPI$type2="WEN_EPI"

WEN_EPI$type3=paste("E",WEN_EPI$orig.ident,sep = "")

table(EnEc$donor)
table(WEN_EPI$donor)



all2=merge(EnEc,WEN_EPI)

all2 <- CreateSeuratObject(counts =all2@assays$RNA@counts, min.cells = 3,meta.data = all2@meta.data)

VlnPlot(all2, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

all2 <- NormalizeData(all2, normalization.method = "LogNormalize", 
                      scale.factor = 10000)
all2 <- FindVariableFeatures(all2, selection.method = "vst", nfeatures = 2000)
VlnPlot(all2, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

#all2进行标准化然后PCA分析
DefaultAssay(all2) <- "RNA"
all.genes_all2 <- rownames(all2)
all2 <- ScaleData(object = all2, features = all.genes_all2)
all2 <- RunPCA(all2, features = VariableFeatures(object = all2))


#all2进行细胞细胞聚类分析(UMAP)
all2=RunUMAP(all2, dims = 1:30)
DimPlot(all2,reduction="umap",label=TRUE)
all2 <- FindNeighbors(all2, dims = 1:30)
all2 <- FindClusters(all2, resolution = 0.3)
all2 <- FindClusters(all2, resolution = 0.6)
DimPlot(all2,reduction="umap",label=TRUE,group.by = "donor")

#进行harmony整合


all2=all_merge%>%RunHarmony("type3",plot_covergence=T)

DefaultAssay(all2) <- "RNA"


all2 <- RunPCA(all2, features = VariableFeatures(object = all2))


all2=RunUMAP(all2,reduction="harmony",dims=1:30)
all2=FindNeighbors(all2,dims=1:30,reduction = "harmony")
all2=FindClusters(all2, resolution = 0.15,reduction = "harmony")
all2=FindClusters(all2, resolution = 0.17,reduction = "harmony")
all2=FindClusters(all2, resolution = 0.18,reduction = "harmony")
all2=FindClusters(all2, resolution = 0.2,reduction = "harmony")


DimPlot(all2,group.by = "donor")



#保存对象
saveRDS(all2,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/rds/all2.rds")

all2=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/rds/all2.rds")

all2=subset(all2,type3!="IUA7")

#进行integraetd整合
all2.list <- SplitObject(all2, split.by="type3")
all2.list <- lapply(X = all2.list, FUN = function(x) {
  x <- NormalizeData(x)
  x <- FindVariableFeatures(x, selection.method = "vst", nfeatures = 2000)
})

k.fliter=min(200,min(sapply(all2.list,ncol)))
all2 <- FindIntegrationAnchors(object.list = all2.list, dims = 1:30,k.filter=k.fliter)
all2 <- IntegrateData(anchorset = all2, dims = 1:30,k.weight=k.fliter)

#进行标准化并进行PCA
DefaultAssay(all2) <- "RNA"
all.genes.all2 <- rownames(all2)
all2 <- ScaleData(object = all2, features = all.genes.all2)

DefaultAssay(all2) <- "integrated"
all.genes.all21 <- rownames(all2)
all2 <- ScaleData(object = all2, features = all.genes.all21)
all2 <- RunPCA(all2, features = VariableFeatures(object = all2))
DimPlot(all2,label = T)

#定义pca维数
all2 <- JackStraw(all2, num.replicate = 100) 
all2 <- ScoreJackStraw(all2, dims = 1:30) 
ElbowPlot(all2,ndims = 30, reduction = "pca") 

#进行all2细胞细胞聚类分析(先runumap降维确定形状再确定cluster)
#进行细胞细胞聚类分析(all2)
DefaultAssay(all2) <- "integrated"
all2=RunUMAP(all2, dims = 1:30)
DimPlot(all2,reduction="umap",label=TRUE)
all2 <- FindNeighbors(all2, dims = 1:30)
all2 <- FindClusters(all2, resolution = 0.3)
all2 <- FindClusters(all2, resolution = 0.15)

all2 <- FindClusters(all2, resolution = 0.6)
all2 <- FindClusters(all2, resolution = 0.45)
all2 <- FindClusters(all2, resolution = 0.05)
DefaultAssay(all2)<-"RNA"








#all2进行细胞细胞聚类分析(tsne)
all2=RunTSNE(WEN2_S,.TR, dims = 1:30)
DimPlot(all2,reduction="tsne",label=TRUE)
all2<- FindNeighbors(all2, dims = 1:30)
all2 <- FindClusters(all2, resolution = 0.3)
all2 <- FindClusters(all2, resolution = 0.15)
all2 <- FindClusters(all2, resolution = 0.05)
DimPlot(all2,reduction="tsne",label=TRUE)

saveRDS(all2,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/rds/all2_inter.rds")




DimPlot(all2,group.by = "type3",label=TRUE)


DimPlot(all2,group.by = "donor",label=TRUE)



#CTRL组velocity分析----
all2=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/rds/all2.rds")

setwd("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/velocyto/ALL-ALL/")

STR$typenew=STR$cell_type0305


DimPlot(STR,group.by = "typenew",label = T,label.size = 7)+ggtitle("")


cellID_humanpost10x=Cells(STR)
cellID_humanpost10x=as.data.frame(cellID_humanpost10x)
colnames(cellID_humanpost10x)[1]<-"x"

cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"]=paste("H:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"1"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"]=paste("qian:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"2"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"]=paste("IUAshou:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"3"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"]=paste("IUAxia:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"4"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"]=paste("li:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"5"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"]=paste("IUAfeng:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"6"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"]=paste("IUA-GWY_additional:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"7"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"]=paste("IUA-BYQ:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"8"],sep = "")
cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"]=paste("IUA-WJM:",cellID_humanpost10x$x[substr(cellID_humanpost10x$x,20,20)%in%"9"],sep = "")
cellID_humanpost10x$x=gsub("-1_1","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_2","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_3","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_4","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_5","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_6","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_7","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_8","x",cellID_humanpost10x$x)
cellID_humanpost10x$x=gsub("-1_9","x",cellID_humanpost10x$x)

write.csv(cellID_humanpost10x, file = "cellID_humanpost10x.csv", row.names = FALSE)
#And finally we can extract our clusters with:
celltype=STR$typenew
celltype=as.data.frame(celltype)
rownames(celltype)=cellID_humanpost10x$x
write.csv(celltype, file = "celltype.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(STR, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")



rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"]=paste(rownames(celltype)[substr(rownames(celltype),1,1)%in%"H"],"-0",sep = "")
rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"]=paste(rownames(celltype)[substr(rownames(celltype),1,4)%in%"qian"],"-1",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAshou"],"-2",sep = "")
rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"]=paste(rownames(celltype)[substr(rownames(celltype),1,6)%in%"IUAxia"],"-3",sep = "")
rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"]=paste(rownames(celltype)[substr(rownames(celltype),1,2)%in%"li"],"-4",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUAfeng"],"-5",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-GWY"],"-6",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-BYQ"],"-7",sep = "")
rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"]=paste(rownames(celltype)[substr(rownames(celltype),1,7)%in%"IUA-WJM"],"-8",sep = "")
write.csv(celltype, file = "celltype2.csv")


#To get UMAP or TSNE coordinates, we use the Embeddings function:
Embeddings=Embeddings(STR, reduction = "umap")
Embeddings=as.data.frame(Embeddings)
rownames(Embeddings)=rownames(celltype)
write.csv(Embeddings, file = "cell_embeddings.csv")

DimPlot(STR,label = T,group.by = "typenew")+ggtitle("")







#和基质细胞进行映射【异位症文章中】----
WEN_STR<-LoadH5Seurat("/data/DATAprocess/chentw/zigongneimo/WEN2/h5seurat文件/endo-2022_stromal.h5seurat",
                      assays="RNA")
STR_allsingle3=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")


#走一下标准化流程
WEN_STR <- SCTransform(WEN_STR)
WEN_STR <- RunPCA(WEN_STR)
WEN_STR <-RunUMAP(WEN_STR, return.model = TRUE,1:30)

DimPlot(WEN_STR,group.by = "subtypes",label = T)


STR_allsingle3 <- SCTransform(STR_allsingle3)

STR_allsingle3=STR_allsingle3%>%RunHarmony("type3",plot_covergence=T)

STR_allsingle3 <- RunPCA(STR_allsingle3)


STR_allsingle3=RunUMAP(STR_allsingle3,reduction="harmony",dims=1:30)
STR_allsingle3=FindNeighbors(STR_allsingle3,dims=1:30,reduction = "harmony")



DimPlot(STR_allsingle3,group.by = "cell_type",label = T)

DimPlot(STR_allsingle3,group.by = "orig.ident",label = T)

transfer.anchors <- FindTransferAnchors(reference = WEN_STR, query = STR_allsingle3, 
                                        dims = 1:30)

predictions <- TransferData(anchorset = transfer.anchors, refdata = WEN_STR$subtypes, 
                            dims = 1:30)

STR_allsingle3$predictions=predictions$predicted.id

STR_allsingle3=RunUMAP(STR_allsingle3,reduction="harmony",dims=1:30)

DimPlot(STR_allsingle3,group.by = "predictions",label = T,reduction = "umap")





#和上皮细胞进行映射【异位症文章中】----
WEN_EPI<-LoadH5Seurat("/data/DATAprocess/chentw/zigongneimo/WEN2/h5seurat文件/endo-2022_epithelial.h5seurat",
                      assays="RNA")
EPI_allsingle3=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/EPI_allsingle.rds")


#走一下标准化流程
WEN_EPI <- SCTransform(WEN_EPI)
WEN_EPI <- RunPCA(WEN_EPI)
WEN_EPI <-RunUMAP(WEN_EPI, return.model = TRUE,1:30)

DimPlot(WEN_EPI,group.by = "subtypes",label = T)


EPI_allsingle3 <- SCTransform(EPI_allsingle3)

EPI_allsingle3=EPI_allsingle3%>%RunHarmony("type3",plot_covergence=T)

EPI_allsingle3 <- RunPCA(EPI_allsingle3)


EPI_allsingle3=RunUMAP(EPI_allsingle3,reduction="harmony",dims=1:30)
EPI_allsingle3=FindNeighbors(EPI_allsingle3,dims=1:30,reduction = "harmony")



DimPlot(EPI_allsingle3,group.by = "cell_type",label = T)

DimPlot(EPI_allsingle3,group.by = "orig.ident",label = T)

transfer.anchors <- FindTransferAnchors(reference = WEN_EPI, query = EPI_allsingle3, 
                                        dims = 1:30)

predictions <- TransferData(anchorset = transfer.anchors, refdata = WEN_EPI$subtypes, 
                            dims = 1:30)

EPI_allsingle3$predictions=predictions$predicted.id

EPI_allsingle3=RunUMAP(EPI_allsingle3,reduction="harmony",dims=1:30)

DimPlot(EPI_allsingle3,group.by = "predictions",label = T,reduction = "umap")




#和基质细胞进行映射【空间转录组文章中】----
WEN_STR<-LoadH5Seurat("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/空间转录组scRNA文件/endometrium_all.h5seurat",
                      assays="RNA")

table(WEN_STR$Cell.type)

WEN_STR=subset(WEN_STR,Cell.type%in%c("eS","dS","Fibroblast C7","uSMC","PV MYH11","PV STEAP4"))

STR_allsingle3=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0304.rds")


#走一下标准化流程
WEN_STR <- SCTransform(WEN_STR)
WEN_STR <- RunPCA(WEN_STR)
WEN_STR <-RunUMAP(WEN_STR, return.model = TRUE,1:30)

DimPlot(WEN_STR,group.by = "Cell.type",label = T)


STR_allsingle3 <- SCTransform(STR_allsingle3)

STR_allsingle3=STR_allsingle3%>%RunHarmony("type3",plot_covergence=T)

STR_allsingle3 <- RunPCA(STR_allsingle3)


STR_allsingle3=RunUMAP(STR_allsingle3,reduction="harmony",dims=1:30)
STR_allsingle3=FindNeighbors(STR_allsingle3,dims=1:30,reduction = "harmony")



DimPlot(STR_allsingle3,group.by = "cell_type",label = T)

DimPlot(STR_allsingle3,group.by = "orig.ident",label = T)

transfer.anchors <- FindTransferAnchors(reference = WEN_STR, query = STR_allsingle3, 
                                        dims = 1:30)

predictions <- TransferData(anchorset = transfer.anchors, refdata = WEN_STR$Cell.type, 
                            dims = 1:30)

STR_allsingle3$predictions=predictions$predicted.id

STR_allsingle3=RunUMAP(STR_allsingle3,reduction="harmony",dims=1:30)

DimPlot(STR_allsingle3,group.by = "predictions",label = T,reduction = "umap")





#和上皮细胞进行映射【空间转录组文章中】----

WEN_STR<-LoadH5Seurat("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/空间转录组scRNA文件/endometrium_all.h5seurat",
                      assays="RNA")

table(WEN_STR$Cell.type)

WEN_EPI=subset(WEN_STR,Cell.type%in%c("Lumenal","SOX9","Glandular","Ciliated"))

EPI_allsingle3=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/EPI_allsingle.rds")


#走一下标准化流程
WEN_EPI <- SCTransform(WEN_EPI)
WEN_EPI <- RunPCA(WEN_EPI)
WEN_EPI <-RunUMAP(WEN_EPI, return.model = TRUE,1:30)

DimPlot(WEN_EPI,group.by = "subtypes",label = T)


EPI_allsingle3 <- SCTransform(EPI_allsingle3)

EPI_allsingle3=EPI_allsingle3%>%RunHarmony("type3",plot_covergence=T)

EPI_allsingle3 <- RunPCA(EPI_allsingle3)


EPI_allsingle3=RunUMAP(EPI_allsingle3,reduction="harmony",dims=1:30)
EPI_allsingle3=FindNeighbors(EPI_allsingle3,dims=1:30,reduction = "harmony")



DimPlot(EPI_allsingle3,group.by = "cell_type",label = T)

DimPlot(EPI_allsingle3,group.by = "orig.ident",label = T)

transfer.anchors <- FindTransferAnchors(reference = WEN_EPI, query = EPI_allsingle3, 
                                        dims = 1:30)

predictions <- TransferData(anchorset = transfer.anchors, refdata = WEN_EPI$Cell.type, 
                            dims = 1:30)

EPI_allsingle3$predictions=predictions$predicted.id

EPI_allsingle3=RunUMAP(EPI_allsingle3,reduction="harmony",dims=1:30)


DimPlot(WEN_EPI,group.by = "predictions",label = T,reduction = "umap")


P1=DimPlot(WEN_EPI,group.by = "Cell.type",label = T,repel = T,label.size = 7,reduction = "umap_scvi_sampl_cc")+
  xlab("umap1")+ylab("umap2")+
  theme(legend.text = element_text(size = 17))

P2=DimPlot(EPI_allsingle3,group.by = "predictions",label = T,reduction = "umap")+
  ggtitle("")+
  theme(legend.text = element_text(size = 17))

P1+P2
ggarrange(P1,P2,common.legend = T,legend = "right")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/NG_上皮映射1.pdf",
       width = 17,height = 7)

#对基质细胞每个亚群进行ahesion和ctrl的差异基因查找并进行绘图-----
#说明差异基因和细胞数目并没有直接关系或者不是完全正相关-----
#先统计上调基因

Cluster0=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/S0/S0_deseq.csv")
Cluster1=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/REV3L/REV3L_deseq.csv")
Cluster2=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/S1/S1_deseq.csv")
Cluster3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/Cycling/Cycling_deseq.csv")
Cluster4=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/BMP7/BMP7_deseq.csv")
Cluster5=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/P/P_deseq.csv")
Cluster6=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/不同细亚型AdheisonVSCtrl并进行富集/ISG15/ISG15_deseq.csv")

Cluster0=subset(Cluster0,p_val<0.05)
Cluster1=subset(Cluster1,p_val<0.05)
Cluster2=subset(Cluster2,p_val<0.05)
Cluster3=subset(Cluster3,p_val<0.05)
Cluster4=subset(Cluster4,p_val<0.05)
Cluster5=subset(Cluster5,p_val<0.05)
Cluster6=subset(Cluster6,p_val<0.05)


Cluster0$Change="up"
Cluster0$Change[Cluster0$avg_log2FC<0]<-"down"

Cluster1$Change="up"
Cluster1$Change[Cluster1$avg_log2FC<0]<-"down"

Cluster2$Change="up"
Cluster2$Change[Cluster2$avg_log2FC<0]<-"down"

Cluster3$Change="up"
Cluster3$Change[Cluster3$avg_log2FC<0]<-"down"

Cluster4$Change="up"
Cluster4$Change[Cluster4$avg_log2FC<0]<-"down"

Cluster5$Change="up"
Cluster5$Change[Cluster5$avg_log2FC<0]<-"down"

Cluster6$Change="up"
Cluster6$Change[Cluster6$avg_log2FC<0]<-"down"

table(Cluster0$Change)
table(Cluster1$Change)
table(Cluster2$Change)
table(Cluster3$Change)
table(Cluster4$Change)
table(Cluster5$Change)
table(Cluster6$Change)

EnSc_bar=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/ensc_bar.csv")


ggplot(EnSc_bar,aes(x=Number,y=CLuster,fill=Change))+
  geom_bar(stat="identity", width =0.8)+
  theme_classic(base_size =25) + 
  theme(panel.grid.major.y = element_line(), 
        legend.position ="none")+
  xlab("")+ylab("")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/1h/h1.pdf",
       width = 7,height = 7)












#对umap图的样本散点分布进行改良-----
library(Seurat)
umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")

table(umap_ctrl_IUA_allsingle$type3)

CTRL=subset(umap_ctrl_IUA_allsingle,type3%in%c("CTRL1","CTRL2"))

S2=subset(umap_ctrl_IUA_allsingle,type3%in%c("IUA1","IUA2","IUA5","IUA6"))

S3=subset(umap_ctrl_IUA_allsingle,type3%in%c("IUA3","IUA4","IUA7"))


P1=DimPlot(CTRL,reduction="umap",label=F,label.size =7,pt.size = 0.01,group.by = "type3",order = F)+
  theme(title = element_text(size =0),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),axis.line = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 36))+theme(text = element_text(size=36))+
  scale_color_manual(values=c("#DB0D13","#51A23E"))


P2=DimPlot(S2,reduction="umap",label=F,label.size =7,pt.size = 0.01,group.by = "type3",order = F)+
  theme(title = element_text(size =0),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),axis.line = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 36))+theme(text = element_text(size=36))+
  scale_color_manual(values=c("#974E90","#EE7D00","#387FBA","#3F2217"))


P3=DimPlot(S3,reduction="umap",label=F,label.size =7,pt.size = 0.01,group.by = "type3",order = F)+
  theme(title = element_text(size =0),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),axis.line = element_blank(),
        legend.position = "none",
        plot.subtitle = element_text(size = 36))+theme(text = element_text(size=36))+
  scale_color_manual(values=c("#FDE015","#87181C","#92CE6C"))



P1+P2+P3





ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/P22.pdf",
       width = 42,height = 14)










#计算巨噬细胞比例图并绘制柱形统计图----
Macr=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/Macr_allsingle.rds")


table(Macr$cell_type0610,Macr$type3)












#和NM文章增殖期数据进行聚类查看增殖期是否都聚在一起-----

#harmony整合
library(harmony)
library(Seurat)
library(dplyr)

WEN_umap=readRDS("/mnt/chentw/zigongneimo/photo/文献数据操作数据/WEN.umap.rds")

umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")

WEN_Pro=subset(WEN_umap,donor%in%"proliferative")

umap_ctrl_IUA_allsingle$donor="Pro"

all=merge(umap_ctrl_IUA_allsingle,WEN_Pro)


DefaultAssay(all)<-"RNA"

N=all@assays$RNA@counts

all=CreateSeuratObject(counts = N, meta.data = all@meta.data)

all=NormalizeData(all, normalization.method = "LogNormalize", 
                  scale.factor = 10000)
all <- FindVariableFeatures(all, selection.method = "vst", nfeatures = 2000)
VlnPlot(all, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)

mtGenes=grep('^MT-',rownames(all),value=TRUE)
RPGenes=c(grep('^RPL',rownames(all),value=TRUE),grep('^RPS',rownames(all),value=TRUE))
stress_gene2=grep('^HSP',rownames(all),value=TRUE)
HB_gene2=grep('^HB',rownames(all),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
stress_gene1=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
               "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/gene1.csv")
subgene3=subgene3$cellcyle_gene
subgene4=c("FOS","CXCL2","ZFP36","FOSB","DUSP1","ATF3","CXCL8","NR4A1","CXCL3","PPP1R15A",
           "JUNB","EGR1","HSPA1A","HSPA1B","SOCS3","KLF6","JUN","IER2","CXCL1","NFKBIA",
           "HSPA6","DNAJB1","IER3","CCNL1","MTRNR2L2","IER4","ID1","CEBPD","KRT6A","CYR61",
           "DEPP1","CLDN4","IRF1","DUSP2","BTG2","PLAUR","MAFF","KLF4","PHLDA2","TNFAIP3")
subgene5=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/512个应激反应相关基因.csv")
subgene5=subgene5$gene_symbol

subgene=c(mtGenes,RPGenes,stress_gene2,HB_gene2,HB.genes,s.genes,g2m.genes,stress_gene1,"MKI67",subgene3,subgene4,subgene5)

all@assays$RNA@var.features = setdiff(all@assays$RNA@var.features,subgene)  #去除subgene影响

all@assays$RNA@var.features

#all进行标准化然后PCA分析
DefaultAssay(all) <- "RNA"
all.genes_all <- rownames(all)
all <- ScaleData(object = all, features = all.genes_all)
all <- RunPCA(all, features = VariableFeatures(object = all))


#all进行细胞细胞聚类分析(UMAP)
all=RunUMAP(all, dims = 1:30)
DimPlot(all,reduction="umap",label=TRUE)
all <- FindNeighbors(all, dims = 1:30)
all <- FindClusters(all, resolution = 0.3)
all <- FindClusters(all, resolution = 0.6)
DimPlot(all,reduction="umap",label=TRUE)


#all进行细胞细胞聚类分析(tsne)
all=RunTSNE(all, dims = 1:30)
DimPlot(all,reduction="tsne",label=TRUE)
all <- FindNeighbors(all, dims = 1:30)
all <- FindClusters(all, resolution = 0.3)
DimPlot(all,reduction="tsne",label=TRUE)

DimPlot(all,reduction="umap",label=TRUE)



#开始尝试去除细胞样本批次并且进行新的一个保存

all=all%>%RunHarmony("orig.ident",plot_covergence=T)

DefaultAssay(all) <- "RNA"


all <- RunPCA(all, features = VariableFeatures(object = all))


all=RunUMAP(all,reduction="harmony",dims=1:30)

all=FindNeighbors(all,dims=1:30,reduction = "harmony")
all=FindClusters(all, resolution = 0.15,reduction = "harmony")
all=FindClusters(all, resolution = 0.3,reduction = "harmony")
all=FindClusters(all, resolution = 0.6,reduction = "harmony")
all=FindClusters(all, resolution = 0.9,reduction = "harmony")
all=FindClusters(all, resolution = 1.2,reduction = "harmony")



saveRDS(all,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/NM_ALL_harmony.rds")



#进行integrated整合

WEN_Pro=subset(WEN_umap,donor%in%"proliferative")

umap_ctrl_IUA_allsingle$donor="Pro"

all=merge(umap_ctrl_IUA_allsingle,WEN_Pro)


DefaultAssay(all)<-"RNA"

N=all@assays$RNA@counts

all=CreateSeuratObject(counts = N, meta.data = all@meta.data)

all=NormalizeData(all, normalization.method = "LogNormalize", 
                  scale.factor = 10000)
all <- FindVariableFeatures(all, selection.method = "vst", nfeatures = 2000)
VlnPlot(all, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)

mtGenes=grep('^MT-',rownames(all),value=TRUE)
RPGenes=c(grep('^RPL',rownames(all),value=TRUE),grep('^RPS',rownames(all),value=TRUE))
stress_gene2=grep('^HSP',rownames(all),value=TRUE)
HB_gene2=grep('^HB',rownames(all),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
stress_gene1=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
               "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/gene1.csv")
subgene3=subgene3$cellcyle_gene
subgene4=c("FOS","CXCL2","ZFP36","FOSB","DUSP1","ATF3","CXCL8","NR4A1","CXCL3","PPP1R15A",
           "JUNB","EGR1","HSPA1A","HSPA1B","SOCS3","KLF6","JUN","IER2","CXCL1","NFKBIA",
           "HSPA6","DNAJB1","IER3","CCNL1","MTRNR2L2","IER4","ID1","CEBPD","KRT6A","CYR61",
           "DEPP1","CLDN4","IRF1","DUSP2","BTG2","PLAUR","MAFF","KLF4","PHLDA2","TNFAIP3")
subgene5=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/512个应激反应相关基因.csv")
subgene5=subgene5$gene_symbol

subgene=c(mtGenes,RPGenes,stress_gene2,HB_gene2,HB.genes,s.genes,g2m.genes,stress_gene1,"MKI67",subgene3,subgene4,subgene5)

all@assays$RNA@var.features = setdiff(all@assays$RNA@var.features,subgene)  #去除subgene影响

all@assays$RNA@var.features

#all进行标准化然后PCA分析
DefaultAssay(all) <- "RNA"
all.genes_all <- rownames(all)
all <- ScaleData(object = all, features = all.genes_all)
all <- RunPCA(all, features = VariableFeatures(object = all))


#all进行细胞细胞聚类分析(UMAP)
all=RunUMAP(all, dims = 1:30)
DimPlot(all,reduction="umap",label=TRUE)
all <- FindNeighbors(all, dims = 1:30)
all <- FindClusters(all, resolution = 0.3)
all <- FindClusters(all, resolution = 0.6)
DimPlot(all,reduction="umap",label=TRUE)


#all进行细胞细胞聚类分析(tsne)
all=RunTSNE(all, dims = 1:30)
DimPlot(all,reduction="tsne",label=TRUE)
all <- FindNeighbors(all, dims = 1:30)
all <- FindClusters(all, resolution = 0.3)
DimPlot(all,reduction="tsne",label=TRUE)

DimPlot(all,reduction="umap",label=TRUE)


all.list <- SplitObject(all, split.by = "orig.ident")

all.list <- lapply(X = all.list, FUN = function(x) {
  x <- NormalizeData(x)
  x <- FindVariableFeatures(x, selection.method = "vst", nfeatures = 2000)
})


features <- SelectIntegrationFeatures(object.list = all.list)


mtGenes=grep('^MT-',rownames(all),value=TRUE)
RPGenes=grep('^RP',rownames(all),value=TRUE)
stress_gene2=grep('^HSP',rownames(all),value=TRUE)
HB_gene2=grep('^HB',rownames(all),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
stress_gene1=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
               "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/gene1.csv")
subgene3=subgene3$cellcyle_gene
subgene4=c("FOS","CXCL2","ZFP36","FOSB","DUSP1","ATF3","CXCL8","NR4A1","CXCL3","PPP1R15A",
           "JUNB","EGR1","HSPA1A","HSPA1B","SOCS3","KLF6","JUN","IER2","CXCL1","NFKBIA",
           "HSPA6","DNAJB1","IER3","CCNL1","MTRNR2L2","IER4","ID1","CEBPD","KRT6A","CYR61",
           "DEPP1","CLDN4","IRF1","DUSP2","BTG2","PLAUR","MAFF","KLF4","PHLDA2","TNFAIP3")



subgene=c(mtGenes,RPGenes,stress_gene2,HB_gene2,HB.genes,s.genes,g2m.genes,stress_gene1,"MKI67",subgene3,subgene4)


features=features[-which(features%in%subgene)]

all <- FindIntegrationAnchors(object.list = all.list, anchor.features = features)

all <- IntegrateData(anchorset = all)

#进行标准化并进行PCA
DefaultAssay(all) <- "RNA"
all.genes.all <- rownames(all)
all <- ScaleData(object = all, features = all.genes.all)

DefaultAssay(all) <- "integrated"
all.genes.all1 <- rownames(all)
all <- ScaleData(object = all, features = all.genes.all1)
all <- RunPCA(all, features = features)
DimPlot(all,label = T)

#进行all细胞细胞聚类分析(先runumap降维确定形状再确定cluster)
all=RunUMAP(all, dims = 1:30)
all <- FindNeighbors(all, dims = 1:30)
all <- FindClusters(all, resolution = 0.15)

all=RunTSNE(all, dims = 1:30)
DimPlot(all,reduction="tsne",label=TRUE)
all <- FindNeighbors(all, dims = 1:30)
all <- FindClusters(all, resolution = 0.3)

saveRDS(all,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/NM_ALL_integrated.rds")


####查看harmony聚类情况----
all_harmony=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/NM_ALL_harmony.rds")

DimPlot(all_harmony,raster=FALSE,group.by = "orig.ident",split.by = "orig.ident")





###和CD301巨噬细胞进行整合-----
library(Seurat)
library(harmony)
library(dplyr)

WEN2_harmony=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/CD301_WEN_DATA/rds存放点/all_harmony.rds")
umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")

all=merge(umap_ctrl_IUA_allsingle,WEN2_harmony)


DefaultAssay(all)<-"RNA"

N=all@assays$RNA@counts

all=CreateSeuratObject(counts = N, meta.data = all@meta.data)

all=NormalizeData(all, normalization.method = "LogNormalize", 
                  scale.factor = 10000)
all <- FindVariableFeatures(all, selection.method = "vst", nfeatures = 2000)
VlnPlot(all, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)

mtGenes=grep('^MT-',rownames(all),value=TRUE)
RPGenes=c(grep('^RPL',rownames(all),value=TRUE),grep('^RPS',rownames(all),value=TRUE))
stress_gene2=grep('^HSP',rownames(all),value=TRUE)
HB_gene2=grep('^HB',rownames(all),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
stress_gene1=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
               "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/gene1.csv")
subgene3=subgene3$cellcyle_gene
subgene4=c("FOS","CXCL2","ZFP36","FOSB","DUSP1","ATF3","CXCL8","NR4A1","CXCL3","PPP1R15A",
           "JUNB","EGR1","HSPA1A","HSPA1B","SOCS3","KLF6","JUN","IER2","CXCL1","NFKBIA",
           "HSPA6","DNAJB1","IER3","CCNL1","MTRNR2L2","IER4","ID1","CEBPD","KRT6A","CYR61",
           "DEPP1","CLDN4","IRF1","DUSP2","BTG2","PLAUR","MAFF","KLF4","PHLDA2","TNFAIP3")
subgene5=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/512个应激反应相关基因.csv")
subgene5=subgene5$gene_symbol

subgene=c(mtGenes,RPGenes,stress_gene2,HB_gene2,HB.genes,s.genes,g2m.genes,stress_gene1,"MKI67",subgene3,subgene4,subgene5)

all@assays$RNA@var.features = setdiff(all@assays$RNA@var.features,subgene)  #去除subgene影响

all@assays$RNA@var.features

#all进行标准化然后PCA分析
DefaultAssay(all) <- "RNA"
all.genes_all <- rownames(all)
all <- ScaleData(object = all, features = all.genes_all)
all <- RunPCA(all, features = VariableFeatures(object = all))


#all进行细胞细胞聚类分析(UMAP)
all=RunUMAP(all, dims = 1:30)
DimPlot(all,reduction="umap",label=TRUE)
all <- FindNeighbors(all, dims = 1:30)
all <- FindClusters(all, resolution = 0.3)
all <- FindClusters(all, resolution = 0.6)
DimPlot(all,reduction="umap",label=TRUE)


#all进行细胞细胞聚类分析(tsne)
all=RunTSNE(all, dims = 1:30)
DimPlot(all,reduction="tsne",label=TRUE)
all <- FindNeighbors(all, dims = 1:30)
all <- FindClusters(all, resolution = 0.3)
DimPlot(all,reduction="tsne",label=TRUE)

DimPlot(all,reduction="umap",label=TRUE)



#开始尝试去除细胞样本批次并且进行新的一个保存

all=all%>%RunHarmony("orig.ident",plot_covergence=T)

DefaultAssay(all) <- "RNA"


all <- RunPCA(all, features = VariableFeatures(object = all))


all=RunUMAP(all,reduction="harmony",dims=1:30)

all=FindNeighbors(all,dims=1:30,reduction = "harmony")
all=FindClusters(all, resolution = 0.15,reduction = "harmony")
all=FindClusters(all, resolution = 0.3,reduction = "harmony")
all=FindClusters(all, resolution = 0.6,reduction = "harmony")
all=FindClusters(all, resolution = 0.9,reduction = "harmony")
all=FindClusters(all, resolution = 1.2,reduction = "harmony")



saveRDS(all,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/CD301_ALL_harmony.rds")

DimPlot(all,group.by = "orig.ident",raster = F)






#和IUA子宫内膜分泌期进行整合-----
library(anndata)
library(Seurat)
library(SingleCellExperiment)
library(SeuratDisk)
library(harmony)
library(dplyr)


WEN2<-LoadH5Seurat("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/IUA_SEC/GSE215968_sc_AS_vs_WOI_Control.h5seurat",
                   assays="RNA")

DimPlot(WEN2)

N=WEN2@assays$RNA@counts

WEN2=CreateSeuratObject(counts = N, meta.data = WEN2@meta.data)

WEN2=NormalizeData(WEN2, normalization.method = "LogNormalize", 
                   scale.factor = 10000)
WEN2 <- FindVariableFeatures(WEN2, selection.method = "vst", nfeatures = 2000)

#WEN2进行标准化然后PCA分析
DefaultAssay(WEN2) <- "RNA"
WEN2.genes_WEN2 <- rownames(WEN2)
WEN2 <- ScaleData(object = WEN2, features = WEN2.genes_WEN2)
WEN2 <- RunPCA(WEN2, features = VariableFeatures(object = WEN2))


#WEN2进行细胞细胞聚类分析(UMAP)
WEN2=RunUMAP(WEN2, dims = 1:30)
DimPlot(WEN2,reduction="umap",label=TRUE)
WEN2 <- FindNeighbors(WEN2, dims = 1:30)
WEN2 <- FindClusters(WEN2, resolution = 0.3)
WEN2 <- FindClusters(WEN2, resolution = 0.6)
DimPlot(WEN2,reduction="umap",label=TRUE,raster = F)
saveRDS(WEN2,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/IUA_sec.rds")



WEN2<-readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/IUA_sec.rds")

umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")



all=merge(umap_ctrl_IUA_allsingle,WEN2)


DefaultAssay(all)<-"RNA"

N=all@assays$RNA@counts

all=CreateSeuratObject(counts = N, meta.data = all@meta.data)

all=NormalizeData(all, normalization.method = "LogNormalize", 
                  scale.factor = 10000)
all <- FindVariableFeatures(all, selection.method = "vst", nfeatures = 2000)
VlnPlot(all, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)

mtGenes=grep('^MT-',rownames(all),value=TRUE)
RPGenes=c(grep('^RPL',rownames(all),value=TRUE),grep('^RPS',rownames(all),value=TRUE))
stress_gene2=grep('^HSP',rownames(all),value=TRUE)
HB_gene2=grep('^HB',rownames(all),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
stress_gene1=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
               "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/gene1.csv")
subgene3=subgene3$cellcyle_gene
subgene4=c("FOS","CXCL2","ZFP36","FOSB","DUSP1","ATF3","CXCL8","NR4A1","CXCL3","PPP1R15A",
           "JUNB","EGR1","HSPA1A","HSPA1B","SOCS3","KLF6","JUN","IER2","CXCL1","NFKBIA",
           "HSPA6","DNAJB1","IER3","CCNL1","MTRNR2L2","IER4","ID1","CEBPD","KRT6A","CYR61",
           "DEPP1","CLDN4","IRF1","DUSP2","BTG2","PLAUR","MAFF","KLF4","PHLDA2","TNFAIP3")
subgene5=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/512个应激反应相关基因.csv")
subgene5=subgene5$gene_symbol

subgene=c(mtGenes,RPGenes,stress_gene2,HB_gene2,HB.genes,s.genes,g2m.genes,stress_gene1,"MKI67",subgene3,subgene4,subgene5)

all@assays$RNA@var.features = setdiff(all@assays$RNA@var.features,subgene)  #去除subgene影响

all@assays$RNA@var.features

#all进行标准化然后PCA分析
DefaultAssay(all) <- "RNA"
all.genes_all <- rownames(all)
all <- ScaleData(object = all, features = all.genes_all)
all <- RunPCA(all, features = VariableFeatures(object = all))


#all进行细胞细胞聚类分析(UMAP)
all=RunUMAP(all, dims = 1:30)
DimPlot(all,reduction="umap",label=TRUE)
all <- FindNeighbors(all, dims = 1:30)
all <- FindClusters(all, resolution = 0.3)
all <- FindClusters(all, resolution = 0.6)
DimPlot(all,reduction="umap",label=TRUE)


#all进行细胞细胞聚类分析(tsne)
all=RunTSNE(all, dims = 1:30)
DimPlot(all,reduction="tsne",label=TRUE)
all <- FindNeighbors(all, dims = 1:30)
all <- FindClusters(all, resolution = 0.3)
DimPlot(all,reduction="tsne",label=TRUE)

DimPlot(all,reduction="umap",label=TRUE)



#开始尝试去除细胞样本批次并且进行新的一个保存

all=all%>%RunHarmony("orig.ident",plot_covergence=T)

DefaultAssay(all) <- "RNA"


all <- RunPCA(all, features = VariableFeatures(object = all))


all=RunUMAP(all,reduction="harmony",dims=1:30)

all=FindNeighbors(all,dims=1:30,reduction = "harmony")
all=FindClusters(all, resolution = 0.15,reduction = "harmony")
all=FindClusters(all, resolution = 0.3,reduction = "harmony")
all=FindClusters(all, resolution = 0.6,reduction = "harmony")
all=FindClusters(all, resolution = 0.9,reduction = "harmony")
all=FindClusters(all, resolution = 1.2,reduction = "harmony")



saveRDS(all,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/IUAsec_ALL_harmony.rds")


#查看增殖期文章和我们数据增殖期是否聚在一起-----

all=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/NM_ALL_harmony.rds")


table(all$type2)

all$type3[all$orig.ident%in%c("19")]<-"NM_CTRL1"
all$type3[all$orig.ident%in%c("63")]<-"NM_CTRL2"

all$type2[all$orig.ident%in%c("19","63")]<-"NM_CTRL"

DimPlot(all,split.by = "type2",raster = F,label = 12,ncol = 2)

DimPlot(all,split.by = "type2",raster = F,label = 12,ncol = 2,group.by = "cell_type")+
  theme(legend.position = "none")




all$batch="1"
all$batch[all$type2%in%"NM_CTRL"]="2"



# 2. 按样本聚合 Harmony 坐标
library(Seurat)
library(harmony)
library(dendextend)
library(dplyr)

metadata <- all@meta.data
sample_names <- unique(metadata$type3)
harmony_embeddings <- Embeddings(all, "harmony")[, 1:30]

sample_matrix <- t(sapply(sample_names, function(x) {
  cells <- rownames(metadata)[metadata$type3 == x]
  colMeans(harmony_embeddings[cells, ])
}))

# 3. 层次聚类
dist_matrix <- dist(sample_matrix, method = "euclidean")
hclust_res <- hclust(dist_matrix, method = "ward.D2")

# 4. 绘制树状图
dend <- as.dendrogram(hclust_res) %>%
  set("branches_k_color", k = 3) %>%
  set("labels_cex", 0.8)
plot(dend, main = "Sample Clustering After Harmony")



#热图绘制
harmony_embeddings <- Embeddings(all, reduction = "harmony")

# 获取样本信息（假设meta.data中的列名为'type3'）
type3_info <- all$type3

# 计算每个样本在Harmony嵌入中的平均坐标
type3_means <- aggregate(harmony_embeddings, by = list(type3 = type3_info), FUN = mean)
rownames(type3_means) <- type3_means$type3
type3_means <- type3_means[, -1]  # 移除样本名列

# 计算相关系数矩阵
cor_matrix <- cor(t(type3_means))

# 准备样本注释（例如批次信息，假设列名为'batch'）
annotation_df <- data.frame(
  Batch = all@meta.data$batch[match(rownames(type3_means), all@meta.data$type3)]
)
rownames(annotation_df) <- rownames(type3_means)

# 绘制热图并添加批次注释
pheatmap(cor_matrix,
         color = colorRampPalette(c("blue", "white", "red"))(100),
         cluster_rows = TRUE,
         cluster_cols = TRUE,
         annotation_row = annotation_df,
         annotation_col = annotation_df,
         display_numbers = TRUE,
         number_color = "black",
         number_format = "%.2f",
         main = "样本相关性热图（Harmony去批次后）")



AverageExpression(all,group.by = "type3",assays = "RNA")









#查看增殖期IUA和数据的关系-----
all=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/CD301_ALL_harmony.rds")

table(all$orig.ident)

all$type3[all$orig.ident%in%c("CTRL1")]<-"EMM_CTRL1"
all$type3[all$orig.ident%in%c("CTRL2")]<-"EMM_CTRL2"
all$type3[all$orig.ident%in%c("CTRL3")]<-"EMM_CTRL3"

all$type3[all$orig.ident%in%c("IUA1")]<-"EMM_IUA1"
all$type3[all$orig.ident%in%c("IUA2")]<-"EMM_IUA2"
all$type3[all$orig.ident%in%c("IUA3")]<-"EMM_IUA3"




all$type2[all$orig.ident%in%c("CTRL1","CTRL2","CTRL3")]<-"EMM_CTRL"
all$type2[all$orig.ident%in%c("IUA1","IUA2","IUA3")]<-"EMM_IUAL"


DimPlot(all,split.by = "type2",raster = F,label = 12,ncol = 2)

DimPlot(all,split.by = "type2",raster = F,label = 12,ncol = 2,group.by = "cell_type")+
  theme(legend.position = "none")





#查看分泌期IUA和数据的关系-----
all=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/IUAsec_ALL_harmony.rds")

table(all$orig.ident)

all$type4="NC_IUA"
all$type4[all$orig.ident%in%c("19","20","29","39","57","58")]<-"NC_CTRL"

all$type4[all$type2%in%c("CTRL")]<-"Ctrl"


all$type4[all$type2%in%c("IUA1")]<-"Adjacency"

all$type4[all$type2%in%c("IUA2")]<-"Adhesion"

table(all$type4)


DimPlot(all,split.by = "type4",raster = F,label = 12,ncol = 2)

DimPlot(all,split.by = "type4",raster = F,label = 12,ncol = 2,group.by = "cell_type")+
  theme(legend.position = "none")


all$cell_type[all$type4%in%c("NC_CTRL","NC_IUA")]=all$principal_cell_types[all$type4%in%c("NC_CTRL","NC_IUA")]

DimPlot(all,split.by = "type4",raster = F,label = 12,ncol = 2,group.by = "cell_type",repel = T)+
  theme(legend.position = "none")




#所有细胞进行一个整合------
WEN2<-readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/IUA_sec.rds")
all2=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/CD301_ALL_harmony.rds")
WEN_umap=readRDS("/mnt/chentw/zigongneimo/photo/文献数据操作数据/WEN.umap.rds")

WEN2=subset(WEN2,Group%in%"AS")

WEN2$cell_type=WEN2$principal_cell_types

WEN2$donor="NC_Sec_IUA"



#开始定义CD301文章的细胞类型[映射的方法辅助判断]

all2$type3[all2$orig.ident%in%c("CTRL1")]<-"EMM_CTRL1"
all2$type3[all2$orig.ident%in%c("CTRL2")]<-"EMM_CTRL2"
all2$type3[all2$orig.ident%in%c("CTRL3")]<-"EMM_CTRL3"

all2$type3[all2$orig.ident%in%c("IUA1")]<-"EMM_IUA1"
all2$type3[all2$orig.ident%in%c("IUA2")]<-"EMM_IUA2"
all2$type3[all2$orig.ident%in%c("IUA3")]<-"EMM_IUA3"

DefaultAssay(all2) <- "RNA"


all2 <- RunPCA(all2, features = VariableFeatures(object = all2))


all2=RunUMAP(all2,reduction="harmony",dims=1:30)

all2=FindNeighbors(all2,dims=1:30,reduction = "harmony")
all2=FindClusters(all2, resolution = 0.15,reduction = "harmony")


all2$cell_type[all2$type3%in%c("EMM_CTRL1","EMM_CTRL2","EMM_CTRL3","EMM_IUA1","EMM_IUA3","EMM_IUA3")]<-
  all2@active.ident[all2$type3%in%c("EMM_CTRL1","EMM_CTRL2","EMM_CTRL3","EMM_IUA1","EMM_IUA3","EMM_IUA3")]

all2$type2[all2$type3%in%c("EMM_CTRL1","EMM_CTRL2","EMM_CTRL3")]<-
  "EMM_CTRL"

all2$type2[all2$type3%in%c("EMM_IUA1","EMM_IUA2","EMM_IUA3")]<-"EMM_IUA"



DimPlot(all2,split.by = "type2",raster = F,label =T,ncol = 2,group.by = "cell_type",repel = T)+theme(legend.position = "none")


DimPlot(all2,split.by = "type2",raster = F,label =T,ncol = 2,repel = T)+theme(legend.position = "none")


#取出并定义细胞类型
CD301_ALL=subset(all2,type2%in%c("EMM_CTRL","EMM_IUA"))
DimPlot(CD301_ALL,label = T,label.size = 12)

CD301_ALL$cell_type="Stromal cells"

CD301_ALL$cell_type[CD301_ALL@active.ident%in%"5"]="Macrophage"
CD301_ALL$cell_type[CD301_ALL@active.ident%in%"10"]="B cells"
CD301_ALL$cell_type[CD301_ALL@active.ident%in%"8"]="Endothelial cells"
CD301_ALL$cell_type[CD301_ALL@active.ident%in%"1"]="T cells"
CD301_ALL$cell_type[CD301_ALL@active.ident%in%"4"]="NK-T cells"
CD301_ALL$cell_type[CD301_ALL@active.ident%in%"9"]="Ciliated epithelial cells"
CD301_ALL$cell_type[CD301_ALL@active.ident%in%"11"]="Mast cells"
CD301_ALL$cell_type[CD301_ALL@active.ident%in%"3"]="Epithelial cells"

DimPlot(CD301_ALL,group.by = "cell_type",raster = F,label =T,ncol = 2,repel = T)

CD301_ALL$donor="EMM_Pro_late_Ctrl"
CD301_ALL$donor[CD301_ALL$type2%in%"EMM_IUA"]="EMM_Pro_late_IUA"

all4=subset(all2,type2%in%c("CTRL","IUA1","IUA2"))

all4$donor=paste("Pro_",all4$type2,sep = "")


#开始整合
library(Seurat)
library(harmony)
library(dplyr)

all=merge(all4,WEN_umap)
all=merge(all,CD301_ALL)
all=merge(all,WEN2)


DefaultAssay(all)<-"RNA"

N=all@assays$RNA@counts

all=CreateSeuratObject(counts = N, meta.data = all@meta.data)

all=NormalizeData(all, normalization.method = "LogNormalize", 
                  scale.factor = 10000)
all <- FindVariableFeatures(all, selection.method = "vst", nfeatures = 2000)
VlnPlot(all, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)

mtGenes=grep('^MT-',rownames(all),value=TRUE)
RPGenes=c(grep('^RPL',rownames(all),value=TRUE),grep('^RPS',rownames(all),value=TRUE))
stress_gene2=grep('^HSP',rownames(all),value=TRUE)
HB_gene2=grep('^HB',rownames(all),value=TRUE)
HB.genes=c("HBA1","HBA2","HBB","HBD","HBE1","HBG1","HBG2","HBM",
           "HBQ1","HBZ")
s.genes <- Seurat::cc.genes$s.genes
g2m.genes <- Seurat::cc.genes$g2m.genes
stress_gene1=c("FOSB","FOS","JUN","JUNB","JUND","ATF3","EGR1","HSPA1A","HSPA1B","HSP90AB1","HSPA8",
               "HSPB1","IER3","IER2","BTG1","BTG2","DUSP1")

subgene3=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/gene1.csv")
subgene3=subgene3$cellcyle_gene
subgene4=c("FOS","CXCL2","ZFP36","FOSB","DUSP1","ATF3","CXCL8","NR4A1","CXCL3","PPP1R15A",
           "JUNB","EGR1","HSPA1A","HSPA1B","SOCS3","KLF6","JUN","IER2","CXCL1","NFKBIA",
           "HSPA6","DNAJB1","IER3","CCNL1","MTRNR2L2","IER4","ID1","CEBPD","KRT6A","CYR61",
           "DEPP1","CLDN4","IRF1","DUSP2","BTG2","PLAUR","MAFF","KLF4","PHLDA2","TNFAIP3")
subgene5=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/512个应激反应相关基因.csv")
subgene5=subgene5$gene_symbol

subgene=c(mtGenes,RPGenes,stress_gene2,HB_gene2,HB.genes,s.genes,g2m.genes,stress_gene1,"MKI67",subgene3,subgene4,subgene5)

all@assays$RNA@var.features = setdiff(all@assays$RNA@var.features,subgene)  #去除subgene影响

all@assays$RNA@var.features

#all进行标准化然后PCA分析
DefaultAssay(all) <- "RNA"
all.genes_all <- rownames(all)
all <- ScaleData(object = all, features = all.genes_all)
all <- RunPCA(all, features = VariableFeatures(object = all))


#all进行细胞细胞聚类分析(UMAP)
all=RunUMAP(all, dims = 1:30)
DimPlot(all,reduction="umap",label=TRUE)
all <- FindNeighbors(all, dims = 1:30)
all <- FindClusters(all, resolution = 0.3)
all <- FindClusters(all, resolution = 0.6)
DimPlot(all,reduction="umap",label=TRUE)


#all进行细胞细胞聚类分析(tsne)
all=RunTSNE(all, dims = 1:30)
DimPlot(all,reduction="tsne",label=TRUE)
all <- FindNeighbors(all, dims = 1:30)
all <- FindClusters(all, resolution = 0.3)
DimPlot(all,reduction="tsne",label=TRUE)

DimPlot(all,reduction="umap",label=TRUE)



#开始尝试去除细胞样本批次并且进行新的一个保存

all=all%>%RunHarmony("orig.ident",plot_covergence=T)

DefaultAssay(all) <- "RNA"


all <- RunPCA(all, features = VariableFeatures(object = all))


all=RunUMAP(all,reduction="harmony",dims=1:30)

all=FindNeighbors(all,dims=1:30,reduction = "harmony")
all=FindClusters(all, resolution = 0.15,reduction = "harmony")
all=FindClusters(all, resolution = 0.3,reduction = "harmony")
all=FindClusters(all, resolution = 0.6,reduction = "harmony")
all=FindClusters(all, resolution = 0.9,reduction = "harmony")
all=FindClusters(all, resolution = 1.2,reduction = "harmony")



saveRDS(all,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/ALL_harmony.rds")






#对上述进行绘图----
all=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/ALL_harmony.rds")

table(all$donor2)

all$donor3="Pro"
all$donor3[all$donor2%in%c("Ctrl_Sec","IUA_Sec")]="Sec"

#对细胞类型名称进行更改
all$cell_type2="N"
all$cell_type2[all$cell_type%in%
                 c("Cycling stromal cells","Erythrocyte","Stromal","Stromal cells","Stromal fibroblasts",
                   "Perivascular cells",
                   "Transient state cells","Perivascular","Smooth muscle cells","AS-uAS-uSMC")]="Stromal cells"

all$cell_type2[all$cell_type%in%
                 c("Endothelium","Endothelial cells","Endothelia")]="Endothelial cells"

all$cell_type2[all$cell_type%in%
                 c("Macrophages","Macrophage","Erythrocytes","DC")]="Macrophage"

all$cell_type2[all$cell_type%in%
                 c("B cells")]="B cells"

all$cell_type2[all$cell_type%in%
                 c("Ciliated epithelial cells","Ciliated Epithelium",
                   "Cliated epithelial cells","Ciliated")]="Ciliated Epithelium"

all$cell_type2[all$cell_type%in%
                 c("Cycling epithelial cells","Epithelial cells","Epithelials cells",
                   "Epithelium","Unciliated epithelia 1","Unciliated epithelia 2",
                   "AS-Epithelium")]="Epithelium cells"

all$cell_type2[all$cell_type%in%
                 c("CD8+ T cells","DN T cells","T cells","MAIT cells")]="T cells"

all$cell_type2[all$cell_type%in%
                 c("Mast cells")]="Mast cells"

all$cell_type2[all$cell_type%in%
                 c("NK-T cells","NK","Cycling NK-T cells","Cytotoxic NK")]="NK cells"

all$cell_type2[all@active.ident%in%"22"&all$cell_type2%in%"N"]="B cells"
all$cell_type2[all@active.ident%in%"22"&all$cell_type2%in%"Macrophage"]="B cells"

all$cell_type2[all@active.ident%in%c("6","25","28")&all$cell_type2%in%"N"]="NK cells"
all$cell_type2[all@active.ident%in%c("5","10","18")&all$cell_type2%in%"N"]="T cells"
all$cell_type2[all@active.ident%in%"27"&all$cell_type2%in%"N"]="Mast cells"


all$cell_type2[all$cell_type2%in%"N"]="T cells"

DimPlot(all,split.by = "donor2",raster = F,label =T,ncol = 3,group.by = "cell_type2",repel = T)


DimPlot(all,split.by = "donor3",raster = F,label =T,ncol = 3,repel = T,
        group.by = "cell_type2",label.size = 7)+ggtitle("")
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/整合图1.pdf",
       width = 17,height = 7)

DimPlot(all,raster = F,label =T,repel = T,
        group.by = "cell_type2",label.size = 7)+ggtitle("")
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/整合图1.pdf",
       width = 9,height = 7)

DimPlot(all,raster = F,label =T,repel = T,order = F,
        shuffle = TRUE, pt.size = 0.005, 
        group.by = "donor3",label.size = 7)+ggtitle("")+
  scale_color_manual(values=c("#3384A1","#A53030"))
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/整合图2_donor.pdf",
       width = 9,height = 7)


table(all$donor)

all$type5="Wang et al."
all$type5[all$donor%in%"Pro_IUA2"]<-"Adhesion"

all$type5[all$donor%in%"Pro_IUA1"]<-"Adjacency"
all$type5[all$donor%in%"Pro_CTRL"]<-"Ctrl"

all$type5[all$donor%in%c("EMM_Pro_late_IUA","EMM_Pro_late_Ctrl")]<-"Lv et al."
all$type5[all$donor%in%c("NC_Sec_IUA")]<-"Santamaria et al."

table(all$type5)

all$type5=factor(all$type5,levels = c("Ctrl","Adjacency","Adhesion",
                                      "Lv et al.","Santamaria et al.","Wang et al."))

DimPlot(subset(all,donor3%in%"Pro"),raster = F,label =F,repel = T,order = F,
        shuffle = TRUE, pt.size = 0.005, 
        group.by = "type5",label.size = 7)+ggtitle("")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/整合图2_Pro.pdf",
       width =9,height = 7)

DimPlot(subset(all,donor3%in%"Sec"),raster = F,label =F,repel = T,order = F,
        shuffle = TRUE, pt.size = 0.005, 
        group.by = "type5",label.size = 7)+ggtitle("")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/整合图2_Sec.pdf",
       width =9,height = 7)


#对多个分组进行split后一个个进行细胞类型绘制
table(all$type5)

DimPlot(subset(all,type5%in%"Ctrl"),raster = F,label =T,
        repel = T,
        group.by = "cell_type2",label.size = 7)+ggtitle("")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/整合图3_Ctrl.pdf",
       width =9,height = 7)

DimPlot(subset(all,type5%in%"Adjacency"),raster = F,label =T,
        repel = T,
        group.by = "cell_type2",label.size = 7)+ggtitle("")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/整合图3_Adjacency.pdf",
       width =9,height = 7)

DimPlot(subset(all,type5%in%"Adhesion"),raster = F,label =T,
        repel = T,
        group.by = "cell_type2",label.size = 7)+ggtitle("")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/整合图3_Adhesion.pdf",
       width =9,height = 7)


DimPlot(subset(all,type5%in%"Lv et al."),raster = F,label =T,
        repel = T,
        group.by = "cell_type2",label.size = 7)+ggtitle("")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/整合图3_LV.pdf",
       width =9,height = 7)


DimPlot(subset(all,type5%in%"Santamaria et al."),raster = F,label =T,
        repel = T,
        group.by = "cell_type2",label.size = 7)+ggtitle("")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/整合图3_Santamaria.pdf",
       width =9,height = 7)

DimPlot(subset(all,type5%in%"Wang et al."),raster = F,label =T,
        repel = T,
        group.by = "cell_type2",label.size = 7)+ggtitle("")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/整合图3_wang.pdf",
       width =9,height = 7)



DimPlot(all,split.by = "type5",raster = F,label =T,ncol = 6,repel = T,
        group.by = "cell_type2",label.size = 7)+ggtitle("")+
  theme(title = element_text(size =0),legend.text = element_text(size = 0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),axis.line = element_blank(),
        plot.subtitle = element_text(size = 36))+theme(text = element_text(size=36))

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/整合图3.pdf",
       width = 33,height = 7)


DimPlot(all,split.by = "type5",raster = F,label =F,ncol = 6,repel = T,
        group.by = "type5",label.size = 7)+ggtitle("")+
  theme(title = element_text(size =0),
        panel.grid.major = element_blank(),panel.grid.minor = element_blank(),panel.border = element_blank(),
        axis.title = element_blank(),axis.text = element_blank(),axis.ticks = element_blank(),axis.line = element_blank(),
        plot.subtitle = element_text(size = 36))+theme(text = element_text(size=36))

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/整合图4.pdf",
       width = 33,height = 7)

DimPlot(subset(all,type5%in%"Ctrl"),raster = F,label =T,
        repel = T,
        group.by = "type5",label.size = 7)+ggtitle("")+
  scale_color_manual(values = "#F8766D")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/整合图3_Ctrl.pdf",
       width =9,height = 7)

DimPlot(subset(all,type5%in%"Adjacency"),raster = F,label =T,
        repel = T,
        group.by = "type5",label.size = 7)+ggtitle("")+
  scale_color_manual(values = "#B79F00")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/整合图3_Adjacency.pdf",
       width =9,height = 7)

DimPlot(subset(all,type5%in%"Adhesion"),raster = F,label =T,
        repel = T,
        group.by = "type5",label.size = 7)+ggtitle("")+
  scale_color_manual(values = "#00BA38")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/整合图3_Adhesion.pdf",
       width =9,height = 7)


DimPlot(subset(all,type5%in%"Lv et al."),raster = F,label =T,
        repel = T,
        group.by = "type5",label.size = 7)+ggtitle("")+
  scale_color_manual(values = "#00BFC4")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/整合图3_LV.pdf",
       width =9,height = 7)


DimPlot(subset(all,type5%in%"Santamaria et al."),raster = F,label =T,
        repel = T,
        group.by = "type5",label.size = 7)+ggtitle("")+
  scale_color_manual(values = "#619CFF")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/整合图3_Santamaria.pdf",
       width =9,height = 7)

DimPlot(subset(all,type5%in%"Wang et al."),raster = F,label =T,
        repel = T,
        group.by = "type5",label.size = 7)+ggtitle("")+
  scale_color_manual(values = "#F564E3")


ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/整合图3_wang.pdf",
       width =9,height = 7)



#












####将非纤毛上皮去除进行monocle分析------
all=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/ALL_harmony.rds")

DimPlot(all,group.by = "cell_type",split.by = "donor",label = T,raster = F,ncol =4,repel = T)+
  theme(legend.position = "none")

all$type4="Ctrl"
all$type4[all$donor%in%c("EMM_Pro_late_IUA","NC_Sec_IUA","Pro_IUA2")]="Adhesion"
all$type4[all$donor%in%c("Pro_IUA1")]="Adjacency"

DimPlot(all,group.by = "cell_type",split.by = "type4",label = T,raster = F,repel = T)+
  theme(legend.position = "none")

all$donor2="Ctrl_Sec"
all$donor2[all$donor%in%c("proliferative","Pro_CTRL","EMM_Pro_late_Ctrl")]="Ctrl_Pro"
all$donor2[all$donor%in%c("EMM_Pro_late_IUA","Pro_IUA2")]="IUA_Pro"
all$donor2[all$donor%in%c("Pro_IUA1")]="Adjacency_Pro"
all$donor2[all$donor%in%c("NC_Sec_IUA")]="IUA_Sec"


DimPlot(all,group.by = "cell_type",split.by = "donor2",label = T,raster = F,repel = T)+
  theme(legend.position = "none")



#取出非纤毛上皮进行monocle2分析[sec和pro差异前800基因]-----
NM_EPI=subset(all,cell_type%in%c("Cycling epithelial cells","Epithelial cells","Epithelials cells","Epithelium",
                                 "Unciliated epithelia 1","Unciliated epithelia 2","AS-Epithelium"))


saveRDS(NM_EPI,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/EPI_ALL.rds")

table(all$cell_type)
NM_STR=subset(all,cell_type%in%c("Stromal","Stromal cells","Stromal fibroblasts",
                                 "Cycling stromal cells","AS-uAS-uSMC","Perivascular"))


saveRDS(NM_STR,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/STR_ALL.rds")



#ALL----
library(monocle)
library(Seurat)
library(dplyr)
NM_EPI=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/EPI_ALL.rds")
DefaultAssay(NM_EPI)<-"RNA"

table(NM_EPI$donor2)

NM_EPI$donor2[NM_EPI$donor2%in%c("Adjacency_Pro","Ctrl_Pro","IUA_Pro")]="Pro"
NM_EPI$donor2[NM_EPI$donor2%in%c("Ctrl_Sec","IUA_Sec")]="Sec"

table(NM_EPI$donor2)

NM_EPI@active.ident=as.factor(NM_EPI$donor2)
NM_EPI_markers <- FindAllMarkers(object=NM_EPI, only.pos = TRUE,min.pct = 0.25, 
                                 logfc.threshold = 0.25)
NM_EPI_markers <-subset(NM_EPI_markers,NM_EPI_markers$p_val<0.05)



NM_EPI_markers_top400 <- NM_EPI_markers %>% group_by(cluster) %>% top_n(n = 400, wt = avg_log2FC)


NM_EPI_markers=NM_EPI_markers_top400$gene




gene3=c(NM_EPI_markers)


##开始monocle2分析[ctrl-上皮]
library(monocle)

DefaultAssay(NM_EPI)<-"RNA"

NM_EPI_matrix=as.data.frame(NM_EPI@assays$RNA@counts)

NM_EPI_matrix=NM_EPI_matrix[gene3,]

NM_EPI_matrix<-as(as.matrix(NM_EPI_matrix),'sparseMatrix')



NM_EPI_ann<-data.frame(gene_id=rownames(NM_EPI_matrix),gene_short_name=rownames(NM_EPI_matrix))

rownames(NM_EPI_ann)<-rownames(NM_EPI_matrix)

NM_EPI_fd<-new("AnnotatedDataFrame", data = NM_EPI_ann)

phenoData <- NM_EPI@meta.data
phenoData$celltype <- NM_EPI$donor
NM_EPI_pd <- new('AnnotatedDataFrame', data = phenoData)



cds<-newCellDataSet(NM_EPI_matrix,phenoData =NM_EPI_pd,
                    featureData =NM_EPI_fd,expressionFamily=negbinomial.size())

cds <- estimateSizeFactors(cds)
cds <- estimateDispersions(cds)

cds<-detectGenes(cds,min_expr=0.1)
expressed_genes <- row.names(subset(fData(cds),
                                    num_cells_expressed>=10))

diff_test_res <- differentialGeneTest(cds[expressed_genes,],                                      
                                      fullModelFormulaStr = "~donor2",cores=16)



sig_genes<-subset(diff_test_res,qval<0.01)
sig_genes <- sig_genes[order(sig_genes$pval),]


# 标记为用于细胞排序的基因
cds <- setOrderingFilter(cds, sig_genes$gene_short_name)


NM_EPI@active.ident=as.factor(NM_EPI$donor)

express_genes <- VariableFeatures(NM_EPI)

cds <- setOrderingFilter(cds[expressed_genes,],                          
                         express_genes)

disp_table <- dispersionTable(cds)
disp.genes <- subset(disp_table, mean_expression>=0.1&dispersion_empirical>=1*dispersion_fit)$gene_id

cds <- setOrderingFilter(cds[expressed_genes,],                          
                         disp.genes) 

cds <- reduceDimension(cds, max_components=2,                       
                       reduction_method = 'DDRTree',                        
                       verbose = F)

cds <- orderCells(cds)

saveRDS(cds,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/EPI_all.rds")


plot_cell_trajectory(cds,cell_size = 1,color_by = "donor2",label=T)+ggtitle("")+
  facet_wrap(~type4, nrow = 1)

P2=plot_cell_trajectory(cds,cell_size = 1,color_by = "Pseudotime")

P1+P2









#Ctrl------
library(monocle)
library(Seurat)
library(dplyr)

NM_EPI=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/EPI_ALL.rds")


table(NM_EPI$donor2)

NM_EPI_CTRL=subset(NM_EPI,donor2%in%c("Ctrl_Pro","Ctrl_Sec"))




DefaultAssay(NM_EPI_CTRL)<-"RNA"

NM_EPI_CTRL@active.ident=as.factor(NM_EPI_CTRL$donor2)
NM_EPI_CTRL_markers <- FindAllMarkers(object=NM_EPI_CTRL, only.pos = TRUE,min.pct = 0.25, 
                                      logfc.threshold = 0.25)
NM_EPI_CTRL_markers <-subset(NM_EPI_CTRL_markers,NM_EPI_CTRL_markers$p_val<0.05)



NM_EPI_CTRL_markers_top400 <- NM_EPI_CTRL_markers %>% group_by(cluster) %>% top_n(n = 400, wt = avg_log2FC)


NM_EPI_CTRL_markers=NM_EPI_CTRL_markers_top400$gene




gene3=c(NM_EPI_CTRL_markers)


##开始monocle2分析[ctrl-上皮]
library(monocle)

DefaultAssay(NM_EPI_CTRL)<-"RNA"

NM_EPI_CTRL_matrix=as.data.frame(NM_EPI_CTRL@assays$RNA@counts)

NM_EPI_CTRL_matrix=NM_EPI_CTRL_matrix[gene3,]

NM_EPI_CTRL_matrix<-as(as.matrix(NM_EPI_CTRL_matrix),'sparseMatrix')



NM_EPI_CTRL_ann<-data.frame(gene_id=rownames(NM_EPI_CTRL_matrix),gene_short_name=rownames(NM_EPI_CTRL_matrix))

rownames(NM_EPI_CTRL_ann)<-rownames(NM_EPI_CTRL_matrix)

NM_EPI_CTRL_fd<-new("AnnotatedDataFrame", data = NM_EPI_CTRL_ann)

phenoData <- NM_EPI_CTRL@meta.data
phenoData$celltype <- NM_EPI_CTRL$donor
NM_EPI_CTRL_pd <- new('AnnotatedDataFrame', data = phenoData)



cds<-newCellDataSet(NM_EPI_CTRL_matrix,phenoData =NM_EPI_CTRL_pd,
                    featureData =NM_EPI_CTRL_fd,expressionFamily=negbinomial.size())

cds <- estimateSizeFactors(cds)
cds <- estimateDispersions(cds)

cds<-detectGenes(cds,min_expr=0.1)
expressed_genes <- row.names(subset(fData(cds),
                                    num_cells_expressed>=10))

diff_test_res <- differentialGeneTest(cds[expressed_genes,],                                      
                                      fullModelFormulaStr = "~donor2",cores=16)



sig_genes<-subset(diff_test_res,qval<0.01)
sig_genes <- sig_genes[order(sig_genes$pval),]


# 标记为用于细胞排序的基因
cds <- setOrderingFilter(cds, sig_genes$gene_short_name)


NM_EPI_CTRL@active.ident=as.factor(NM_EPI_CTRL$donor)

express_genes <- VariableFeatures(NM_EPI_CTRL)

cds <- setOrderingFilter(cds[expressed_genes,],                          
                         express_genes)

disp_table <- dispersionTable(cds)
disp.genes <- subset(disp_table, mean_expression>=0.1&dispersion_empirical>=1*dispersion_fit)$gene_id

cds <- setOrderingFilter(cds[expressed_genes,],                          
                         disp.genes) 

cds <- reduceDimension(cds, max_components=2,                       
                       reduction_method = 'DDRTree',                        
                       verbose = F)

cds <- orderCells(cds)

saveRDS(cds,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/EPI_all_CTRL.rds")

P1=plot_cell_trajectory(cds,cell_size = 1,color_by = "donor2",label=T)+ggtitle("")

P2=plot_cell_trajectory(cds,cell_size = 1,color_by = "Pseudotime")

P3=plot_cell_trajectory(cds,cell_size = 1,color_by = "cell_type0718",label=T)+ggtitle("")+
  facet_wrap(~cell_type0718, nrow = 1)

P3
P1+P2













#IUA------
library(monocle)
library(Seurat)
library(dplyr)

NM_EPI=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/EPI_ALL.rds")


table(NM_EPI$donor2)

NM_EPI_CTRL=subset(NM_EPI,donor2%in%c("Ctrl_Pro","Ctrl_Sec"))




DefaultAssay(NM_EPI_CTRL)<-"RNA"


#选择ctrl组的差异基因在IUA再进行一次monocle2分析绘图
NM_EPI_CTRL@active.ident=as.factor(NM_EPI_CTRL$donor2)
NM_EPI_CTRL_markers <- FindAllMarkers(object=NM_EPI_CTRL, only.pos = TRUE,min.pct = 0.25, 
                                      logfc.threshold = 0.25)
NM_EPI_CTRL_markers <-subset(NM_EPI_CTRL_markers,NM_EPI_CTRL_markers$p_val<0.05)


NM_EPI_CTRL_markers_top400 <- NM_EPI_CTRL_markers %>% group_by(cluster) %>% top_n(n = 400, wt = avg_log2FC)


NM_EPI_CTRL_markers=NM_EPI_CTRL_markers_top400$gene

gene3=c(NM_EPI_CTRL_markers)


table(NM_EPI$donor2)

NM_EPI_IUA=subset(NM_EPI,donor2%in%c("IUA_Pro","IUA_Sec"))


##开始monocle2分析[IUA-上皮]
library(monocle)

DefaultAssay(NM_EPI_IUA)<-"RNA"

NM_EPI_IUA_matrix=as.data.frame(NM_EPI_IUA@assays$RNA@counts)

NM_EPI_IUA_matrix=NM_EPI_IUA_matrix[gene3,]

NM_EPI_IUA_matrix<-as(as.matrix(NM_EPI_IUA_matrix),'sparseMatrix')



NM_EPI_IUA_ann<-data.frame(gene_id=rownames(NM_EPI_IUA_matrix),
                           gene_short_name=rownames(NM_EPI_IUA_matrix))

rownames(NM_EPI_IUA_ann)<-rownames(NM_EPI_IUA_matrix)

NM_EPI_IUA_fd<-new("AnnotatedDataFrame", data = NM_EPI_IUA_ann)

phenoData <- NM_EPI_IUA@meta.data
phenoData$celltype <- NM_EPI_IUA$donor
NM_EPI_IUA_pd <- new('AnnotatedDataFrame', data = phenoData)



cds<-newCellDataSet(NM_EPI_IUA_matrix,phenoData =NM_EPI_IUA_pd,
                    featureData =NM_EPI_IUA_fd,expressionFamily=negbinomial.size())

cds <- estimateSizeFactors(cds)
cds <- estimateDispersions(cds)

cds<-detectGenes(cds,min_expr=0.1)
expressed_genes <- row.names(subset(fData(cds),
                                    num_cells_expressed>=10))

diff_test_res <- differentialGeneTest(cds[expressed_genes,],                                      
                                      fullModelFormulaStr = "~donor2",cores=16)



sig_genes<-subset(diff_test_res,qval<0.01)
sig_genes <- sig_genes[order(sig_genes$pval),]


# 标记为用于细胞排序的基因
cds <- setOrderingFilter(cds, sig_genes$gene_short_name)


NM_EPI_IUA@active.ident=as.factor(NM_EPI_IUA$donor)

express_genes <- VariableFeatures(NM_EPI_IUA)

cds <- setOrderingFilter(cds[expressed_genes,],                          
                         express_genes)

disp_table <- dispersionTable(cds)
disp.genes <- subset(disp_table, mean_expression>=0.1&dispersion_empirical>=1*dispersion_fit)$gene_id

cds <- setOrderingFilter(cds[expressed_genes,],                          
                         disp.genes) 

cds <- reduceDimension(cds, max_components=2,                       
                       reduction_method = 'DDRTree',                        
                       verbose = F)

cds <- orderCells(cds)

saveRDS(cds,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/EPI_all_IUA.rds")



P1=plot_cell_trajectory(cds,cell_size = 1,color_by = "donor2",label=T)+ggtitle("")

P2=plot_cell_trajectory(cds,cell_size = 1,color_by = "Pseudotime")

P3=plot_cell_trajectory(cds,cell_size = 1,color_by = "cell_type0718",label=T)+ggtitle("")+
  facet_wrap(~cell_type0718, nrow = 1)

P3
P1+P2


#取出基质细胞进行monocle2分析----
#ALL----
library(monocle)
library(Seurat)
library(dplyr)
NM_STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/STR_ALL.rds")
DefaultAssay(NM_STR)<-"RNA"

table(NM_STR$donor2)

NM_STR$donor2[NM_STR$donor2%in%c("Adjacency_Pro","Ctrl_Pro","IUA_Pro")]="Pro"
NM_STR$donor2[NM_STR$donor2%in%c("Ctrl_Sec","IUA_Sec")]="Sec"

table(NM_STR$donor2)

NM_STR@active.ident=as.factor(NM_STR$donor2)
NM_STR_markers <- FindAllMarkers(object=NM_STR, only.pos = TRUE,min.pct = 0.25, 
                                 logfc.threshold = 0.25)
NM_STR_markers <-subset(NM_STR_markers,NM_STR_markers$p_val<0.05)



NM_STR_markers_top400 <- NM_STR_markers %>% group_by(cluster) %>% top_n(n = 400, wt = avg_log2FC)


NM_STR_markers=NM_STR_markers_top400$gene




gene3=c(NM_STR_markers)


##开始monocle2分析[ctrl-上皮]
library(monocle)

DefaultAssay(NM_STR)<-"RNA"

NM_STR_matrix=as.data.frame(NM_STR@assays$RNA@counts)

NM_STR_matrix=NM_STR_matrix[gene3,]

NM_STR_matrix<-as(as.matrix(NM_STR_matrix),'sparseMatrix')



NM_STR_ann<-data.frame(gene_id=rownames(NM_STR_matrix),gene_short_name=rownames(NM_STR_matrix))

rownames(NM_STR_ann)<-rownames(NM_STR_matrix)

NM_STR_fd<-new("AnnotatedDataFrame", data = NM_STR_ann)

phenoData <- NM_STR@meta.data
phenoData$celltype <- NM_STR$donor
NM_STR_pd <- new('AnnotatedDataFrame', data = phenoData)



cds<-newCellDataSet(NM_STR_matrix,phenoData =NM_STR_pd,
                    featureData =NM_STR_fd,expressionFamily=negbinomial.size())

cds <- estimateSizeFactors(cds)
cds <- estimateDispersions(cds)

cds<-detectGenes(cds,min_expr=0.1)
expressed_genes <- row.names(subset(fData(cds),
                                    num_cells_expressed>=10))

diff_test_res <- differentialGeneTest(cds[expressed_genes,],                                      
                                      fullModelFormulaStr = "~donor2",cores=16)



sig_genes<-subset(diff_test_res,qval<0.01)
sig_genes <- sig_genes[order(sig_genes$pval),]


# 标记为用于细胞排序的基因
cds <- setOrderingFilter(cds, sig_genes$gene_short_name)


NM_STR@active.ident=as.factor(NM_STR$donor)

express_genes <- VariableFeatures(NM_STR)

cds <- setOrderingFilter(cds[expressed_genes,],                          
                         express_genes)

disp_table <- dispersionTable(cds)
disp.genes <- subset(disp_table, mean_expression>=0.1&dispersion_empirical>=1*dispersion_fit)$gene_id

cds <- setOrderingFilter(cds[expressed_genes,],                          
                         disp.genes) 

cds <- reduceDimension(cds, max_components=2,                       
                       reduction_method = 'DDRTree',                        
                       verbose = F)

cds <- orderCells(cds)

saveRDS(cds,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/EPI_all.rds")

plot_cell_trajectory(cds,cell_size = 1,color_by = "donor2",label=T)+ggtitle("")+
  facet_wrap(~type4, nrow = 1)

P2=plot_cell_trajectory(cds,cell_size = 1,color_by = "Pseudotime")

P1+P2









#Ctrl------
library(monocle)
library(Seurat)
library(dplyr)

NM_STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/STR_ALL.rds")


table(NM_STR$donor2)

NM_STR_CTRL=subset(NM_STR,donor2%in%c("Ctrl_Pro","Ctrl_Sec"))




DefaultAssay(NM_STR_CTRL)<-"RNA"

NM_STR_CTRL@active.ident=as.factor(NM_STR_CTRL$donor2)
NM_STR_CTRL_markers <- FindAllMarkers(object=NM_STR_CTRL, only.pos = TRUE,min.pct = 0.25, 
                                      logfc.threshold = 0.25)
NM_STR_CTRL_markers <-subset(NM_STR_CTRL_markers,NM_STR_CTRL_markers$p_val<0.05)



NM_STR_CTRL_markers_top400 <- NM_STR_CTRL_markers %>% group_by(cluster) %>% top_n(n = 400, wt = avg_log2FC)


NM_STR_CTRL_markers=NM_STR_CTRL_markers_top400$gene

NM_STR_CTRL <- subset(NM_STR_CTRL,downsample = 10000)


gene3=c(NM_STR_CTRL_markers)


##开始monocle2分析[ctrl-上皮]
library(monocle)

DefaultAssay(NM_STR_CTRL)<-"RNA"

NM_STR_CTRL_matrix=as.data.frame(NM_STR_CTRL@assays$RNA@counts)

NM_STR_CTRL_matrix=NM_STR_CTRL_matrix[gene3,]

NM_STR_CTRL_matrix<-as(as.matrix(NM_STR_CTRL_matrix),'sparseMatrix')



NM_STR_CTRL_ann<-data.frame(gene_id=rownames(NM_STR_CTRL_matrix),gene_short_name=rownames(NM_STR_CTRL_matrix))

rownames(NM_STR_CTRL_ann)<-rownames(NM_STR_CTRL_matrix)

NM_STR_CTRL_fd<-new("AnnotatedDataFrame", data = NM_STR_CTRL_ann)

phenoData <- NM_STR_CTRL@meta.data
phenoData$celltype <- NM_STR_CTRL$donor
NM_STR_CTRL_pd <- new('AnnotatedDataFrame', data = phenoData)



cds<-newCellDataSet(NM_STR_CTRL_matrix,phenoData =NM_STR_CTRL_pd,
                    featureData =NM_STR_CTRL_fd,expressionFamily=negbinomial.size())

cds <- estimateSizeFactors(cds)
cds <- estimateDispersions(cds)

cds<-detectGenes(cds,min_expr=0.1)
expressed_genes <- row.names(subset(fData(cds),
                                    num_cells_expressed>=10))

diff_test_res <- differentialGeneTest(cds[expressed_genes,],                                      
                                      fullModelFormulaStr = "~donor2",cores=16)



sig_genes<-subset(diff_test_res,qval<0.01)
sig_genes <- sig_genes[order(sig_genes$pval),]


# 标记为用于细胞排序的基因
cds <- setOrderingFilter(cds, sig_genes$gene_short_name)


NM_STR_CTRL@active.ident=as.factor(NM_STR_CTRL$donor)

express_genes <- VariableFeatures(NM_STR_CTRL)

cds <- setOrderingFilter(cds[expressed_genes,],                          
                         express_genes)

disp_table <- dispersionTable(cds)
disp.genes <- subset(disp_table, mean_expression>=0.1&dispersion_empirical>=1*dispersion_fit)$gene_id

cds <- setOrderingFilter(cds[expressed_genes,],                          
                         disp.genes) 

cds <- reduceDimension(cds, max_components=2,                       
                       reduction_method = 'DDRTree',                        
                       verbose = F)

cds <- orderCells(cds)

saveRDS(cds,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/STR_ALL_CTRL.rds")

P1=plot_cell_trajectory(cds,cell_size = 1,color_by = "donor2",label=T)+ggtitle("")

P2=plot_cell_trajectory(cds,cell_size = 1,color_by = "Pseudotime")

P3=plot_cell_trajectory(cds,cell_size = 1,color_by = "cell_type0718",label=T)+ggtitle("")+
  facet_wrap(~cell_type0718, nrow = 1)

P3
P1+P2













#IUA------
library(monocle)
library(Seurat)
library(dplyr)

NM_STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/STR_ALL.rds")


table(NM_STR$donor2)

NM_STR_CTRL=subset(NM_STR,donor2%in%c("Ctrl_Pro","Ctrl_Sec"))




DefaultAssay(NM_STR_CTRL)<-"RNA"


#选择ctrl组的差异基因在IUA再进行一次monocle2分析绘图
NM_STR_CTRL@active.ident=as.factor(NM_STR_CTRL$donor2)
NM_STR_CTRL_markers <- FindAllMarkers(object=NM_STR_CTRL, only.pos = TRUE,min.pct = 0.25, 
                                      logfc.threshold = 0.25)
NM_STR_CTRL_markers <-subset(NM_STR_CTRL_markers,NM_STR_CTRL_markers$p_val<0.05)


NM_STR_CTRL_markers_top400 <- NM_STR_CTRL_markers %>% group_by(cluster) %>% top_n(n = 400, wt = avg_log2FC)


NM_STR_CTRL_markers=NM_STR_CTRL_markers_top400$gene

gene3=c(NM_STR_CTRL_markers)


table(NM_STR$donor2)

NM_STR_IUA=subset(NM_STR,donor2%in%c("IUA_Pro","IUA_Sec"))

NM_STR_IUA@active.ident=as.factor(NM_STR_IUA$donor2)
NM_STR_IUA <- subset(NM_STR_IUA,downsample = 10000)

##开始monocle2分析[IUA-上皮]
library(monocle)

DefaultAssay(NM_STR_IUA)<-"RNA"

NM_STR_IUA_matrix=as.data.frame(NM_STR_IUA@assays$RNA@counts)

NM_STR_IUA_matrix=NM_STR_IUA_matrix[gene3,]

NM_STR_IUA_matrix<-as(as.matrix(NM_STR_IUA_matrix),'sparseMatrix')



NM_STR_IUA_ann<-data.frame(gene_id=rownames(NM_STR_IUA_matrix),
                           gene_short_name=rownames(NM_STR_IUA_matrix))

rownames(NM_STR_IUA_ann)<-rownames(NM_STR_IUA_matrix)

NM_STR_IUA_fd<-new("AnnotatedDataFrame", data = NM_STR_IUA_ann)

phenoData <- NM_STR_IUA@meta.data
phenoData$celltype <- NM_STR_IUA$donor
NM_STR_IUA_pd <- new('AnnotatedDataFrame', data = phenoData)



cds<-newCellDataSet(NM_STR_IUA_matrix,phenoData =NM_STR_IUA_pd,
                    featureData =NM_STR_IUA_fd,expressionFamily=negbinomial.size())

cds <- estimateSizeFactors(cds)
cds <- estimateDispersions(cds)

cds<-detectGenes(cds,min_expr=0.1)
expressed_genes <- row.names(subset(fData(cds),
                                    num_cells_expressed>=10))

diff_test_res <- differentialGeneTest(cds[expressed_genes,],                                      
                                      fullModelFormulaStr = "~donor2",cores=16)



sig_genes<-subset(diff_test_res,qval<0.01)
sig_genes <- sig_genes[order(sig_genes$pval),]


# 标记为用于细胞排序的基因
cds <- setOrderingFilter(cds, sig_genes$gene_short_name)


NM_STR_IUA@active.ident=as.factor(NM_STR_IUA$donor)

express_genes <- VariableFeatures(NM_STR_IUA)

cds <- setOrderingFilter(cds[expressed_genes,],                          
                         express_genes)

disp_table <- dispersionTable(cds)
disp.genes <- subset(disp_table, mean_expression>=0.1&dispersion_empirical>=1*dispersion_fit)$gene_id

cds <- setOrderingFilter(cds[expressed_genes,],                          
                         disp.genes) 

cds <- reduceDimension(cds, max_components=2,                       
                       reduction_method = 'DDRTree',                        
                       verbose = F)

cds <- orderCells(cds)



saveRDS(cds,
        "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/STR_ALL_IUA.rds")



P1=plot_cell_trajectory(cds,cell_size = 1,color_by = "donor2",label=T)+ggtitle("")

P2=plot_cell_trajectory(cds,cell_size = 1,color_by = "Pseudotime")

P3=plot_cell_trajectory(cds,cell_size = 1,color_by = "cell_type0718",label=T)+ggtitle("")+
  facet_wrap(~cell_type0718, nrow = 1)

P3
P1+P2


#对上皮monocle结果进行展示----
#CTRL组----
cds=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/EPI_all_CTRL.rds")

plot_cell_trajectory(cds,cell_size = 1,color_by = "State",label=T)+ggtitle("")

plot_cell_trajectory(cds,cell_size = 1,color_by = "donor2",label=T)+ggtitle("")
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/Ctrl_monocle1.pdf",
       width = 7,height = 7)

plot_cell_trajectory(cds,cell_size = 1,color_by = "Pseudotime")
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/Ctrl_monocle2.pdf",
       width = 7,height = 7)



cds$type5="Wang et al."
cds$type5[cds$donor%in%"Pro_IUA2"]<-"Adhesion"

cds$type5[cds$donor%in%"Pro_IUA1"]<-"Adjacency"
cds$type5[cds$donor%in%"Pro_CTRL"]<-"Ctrl"

cds$type5[cds$donor%in%c("EMM_Pro_late_IUA","EMM_Pro_late_Ctrl")]<-"Lv et al."
cds$type5[cds$donor%in%c("NC_Sec_IUA")]<-"Santamaria et al."


plot_cell_trajectory(cds,cell_size = 1,color_by = "donor2",label=T)+ggtitle("")+
  facet_wrap(~type5, nrow = 1)


ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/Ctrl_monocle3.pdf",
       width = 14,height = 7)


mygenes=c("IGFBP1","PAEP","DKK1","S100P","MAOA","SPP1","FOXO1","PRL",
          "MKI67","CCND1","PCNA","ESR1","MYC")

plot_genes_in_pseudotime(cds["IGFBP1",], color_by = "donor2",ncol = 2)



NM_EPI=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/EPI_ALL.rds")
cds=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/EPI_all_CTRL.rds")

table(NM_EPI$donor2)

NM_EPI_CTRL=subset(NM_EPI,donor2%in%c("Ctrl_Pro","Ctrl_Sec"))

normalized_data <- as.data.frame(t(NM_EPI_CTRL@assays$RNA@data))


# 提取伪时间信息
pseudotime_data=pData(cds)

# 创建数据框，用于绘制图形
plot_data <- data.frame(
  pseudotime = pseudotime_data$Pseudotime,
  State=pseudotime_data$State,
  type3=pseudotime_data$type3,
  type2=pseudotime_data$type2,
  donor2=pseudotime_data$donor2,
  donor=pseudotime_data$donor,
  ESR1=normalized_data$ESR1,
  FOXO1=normalized_data$FOXO1,
  PAEP=normalized_data$PAEP,
  SPP1=normalized_data$SPP1)



# 使用 ggplot2 绘制基因表达随伪时间变化的图
library(ggplot2)
library(ggpubr)

correlation <- cor(plot_data$pseudotime, plot_data$ESR1)

cor(plot_data$pseudotime, plot_data$ESR1)
cor(plot_data$pseudotime, plot_data$FOXO1)
cor(plot_data$pseudotime, plot_data$PAEP)
cor(plot_data$pseudotime, plot_data$SPP1)

P1=ggplot(plot_data, aes(x = pseudotime, y = ESR1, color = donor2)) + 
  geom_point(size=1) +  # 绘制散点
  geom_smooth(method = "loess", se = FALSE, color = "black") +theme_bw()+
  xlab("")+ylab("")+theme(text = element_text(size=12,face = "bold",colour = "black"))+
  scale_x_continuous(expand = c(0,0))+scale_y_continuous(expand = c(0,0))+
  scale_fill_manual(values = c("#F3786B","#AFA302"))+
  theme(legend.position = "none")+ggtitle("ESR1")

P2=ggplot(plot_data, aes(x = pseudotime, y = FOXO1, color = donor2)) + 
  geom_point(size=1) +  # 绘制散点
  geom_smooth(method = "loess", se = FALSE, color = "black") +theme_bw()+
  xlab("")+ylab("")+theme(text = element_text(size=12,face = "bold",colour = "black"))+
  scale_x_continuous(expand = c(0,0))+scale_y_continuous(expand = c(0,0))+
  scale_fill_manual(values = c("#F3786B","#AFA302"))+
  theme(legend.position = "none") +ggtitle("FOXO1")

P3=ggplot(plot_data, aes(x = pseudotime, y = PAEP, color = donor2)) + 
  geom_point(size=1) +  # 绘制散点
  geom_smooth(method = "loess", se = FALSE, color = "black") +theme_bw()+
  xlab("")+ylab("")+theme(text = element_text(size=12,face = "bold",colour = "black"))+
  scale_x_continuous(expand = c(0,0))+scale_y_continuous(expand = c(0,0))+
  scale_fill_manual(values = c("#F3786B","#AFA302"))+
  theme(legend.position = "none")+ggtitle("PAEP")


P4=ggplot(plot_data, aes(x = pseudotime, y = SPP1, color = donor2)) + 
  geom_point(size=1) +  # 绘制散点
  geom_smooth(method = "loess", se = FALSE, color = "black") +theme_bw()+
  xlab("")+ylab("")+theme(text = element_text(size=12,face = "bold",colour = "black"))+
  scale_x_continuous(expand = c(0,0))+scale_y_continuous(expand = c(0,0))+
  scale_fill_manual(values = c("#F3786B","#AFA302"))+
  theme(legend.position = "none")+ggtitle("SPP1")

ggarrange(P1,P2,P3,P4,ncol=1)

#IUA组-----
cds=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/EPI_all_IUA.rds")



cds$type5="Wang et al."
cds$type5[cds$donor%in%"Pro_IUA2"]<-"Adhesion"

cds$type5[cds$donor%in%"Pro_IUA1"]<-"Adjacency"
cds$type5[cds$donor%in%"Pro_CTRL"]<-"Ctrl"

cds$type5[cds$donor%in%c("EMM_Pro_late_IUA","EMM_Pro_late_Ctrl")]<-"Lv et al."
cds$type5[cds$donor%in%c("NC_Sec_IUA")]<-"Santamaria et al."


plot_cell_trajectory(cds,cell_size = 1,color_by = "donor2",label=T)+ggtitle("")
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/IUA_monocle1.pdf",
       width = 7,height = 7)

plot_cell_trajectory(cds,cell_size = 1,color_by = "Pseudotime")
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/IUA_monocle2.pdf",
       width = 7,height = 7)



plot_cell_trajectory(cds,cell_size = 1,color_by = "donor2",label=T)+ggtitle("")+
  facet_wrap(~type5, nrow = 1)


ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/IUA_monocle3.pdf",
       width = 14,height = 7)

N=cds@featureData@data$gene_short_name

N2=intersect(N,mygenes)

plot_genes_in_pseudotime(cds[N2,],ncol = 2,color_by = "donor2")


cds2=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/EPI_all_IUA.rds")

table(NM_EPI$donor2)

NM_EPI_IUA=subset(NM_EPI,donor2%in%c("IUA_Pro","IUA_Sec"))

normalized_data2 <- as.data.frame(t(NM_EPI_IUA@assays$RNA@data))


# 提取伪时间信息
pseudotime_data2=pData(cds2)

# 创建数据框，用于绘制图形
plot_data2 <- data.frame(
  pseudotime = pseudotime_data2$Pseudotime,
  State=pseudotime_data2$State,
  type3=pseudotime_data2$type3,
  type2=pseudotime_data2$type2,
  donor2=pseudotime_data2$donor2,
  donor=pseudotime_data2$donor,
  ESR1=normalized_data2$ESR1,
  FOXO1=normalized_data2$FOXO1,
  PAEP=normalized_data2$PAEP,
  SPP1=normalized_data2$SPP1)



# 使用 ggplot2 绘制基因表达随伪时间变化的图
library(ggplot2)
library(ggpubr)


cor(plot_data2$pseudotime, plot_data2$ESR1)
cor(plot_data2$pseudotime, plot_data2$FOXO1)
cor(plot_data2$pseudotime, plot_data2$PAEP)
cor(plot_data2$pseudotime, plot_data2$SPP1)


P5=ggplot(plot_data2, aes(x = pseudotime, y = ESR1, color = donor2)) + 
  geom_point(size=1) +  # 绘制散点
  geom_smooth(method = "loess", se = FALSE, color = "black") +theme_bw()+
  xlab("")+ylab("")+theme(text = element_text(size=12,face = "bold",colour = "black"))+
  scale_x_continuous(expand = c(0,0))+scale_y_continuous(expand = c(0,0))+
  scale_fill_manual(values = c("#F3786B","#AFA302"))+
  theme(legend.position = "none")+ggtitle("ESR1")

P6=ggplot(plot_data2, aes(x = pseudotime, y = FOXO1, color = donor2)) + 
  geom_point(size=1) +  # 绘制散点
  geom_smooth(method = "loess", se = FALSE, color = "black") +theme_bw()+
  xlab("")+ylab("")+theme(text = element_text(size=12,face = "bold",colour = "black"))+
  scale_x_continuous(expand = c(0,0))+scale_y_continuous(expand = c(0,0))+
  scale_fill_manual(values = c("#F3786B","#AFA302"))+
  theme(legend.position = "none") +ggtitle("FOXO1")

P7=ggplot(plot_data2, aes(x = pseudotime, y = PAEP, color = donor2)) + 
  geom_point(size=1) +  # 绘制散点
  geom_smooth(method = "loess", se = FALSE, color = "black") +theme_bw()+
  xlab("")+ylab("")+theme(text = element_text(size=12,face = "bold",colour = "black"))+
  scale_x_continuous(expand = c(0,0))+scale_y_continuous(expand = c(0,0))+
  scale_fill_manual(values = c("#F3786B","#AFA302"))+
  theme(legend.position = "none")+ggtitle("PAEP")


P8=ggplot(plot_data2, aes(x = pseudotime, y = SPP1, color = donor2)) + 
  geom_point(size=1) +  # 绘制散点
  geom_smooth(method = "loess", se = FALSE, color = "black") +theme_bw()+
  xlab("")+ylab("")+theme(text = element_text(size=12,face = "bold",colour = "black"))+
  scale_x_continuous(expand = c(0,0))+scale_y_continuous(expand = c(0,0))+
  scale_fill_manual(values = c("#F3786B","#AFA302"))+
  theme(legend.position = "none")+ggtitle("SPP1")

cowplot::plot_grid(P1,P5,P2,P6,P3,P7,P4,P8,ncol=2)

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/corP2.pdf",
       width = 6,height = 12)





#绘制一张整合的图-----

#对上皮细胞进行分支热图绘制-----
cds=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/EPI_all_CTRL.rds")
library(monocle)

cds$type5="Wang et al."
cds$type5[cds$donor%in%"Pro_IUA2"]<-"Adhesion"

cds$type5[cds$donor%in%"Pro_IUA1"]<-"Adjacency"
cds$type5[cds$donor%in%"Pro_CTRL"]<-"Ctrl"

cds$type5[cds$donor%in%c("EMM_Pro_late_IUA","EMM_Pro_late_Ctrl")]<-"Lv et al."
cds$type5[cds$donor%in%c("NC_Sec_IUA")]<-"Santamaria et al."


plot_cell_trajectory(cds,cell_size = 1,color_by = "donor2",label=T)+ggtitle("")


plot_cell_trajectory(cds,cell_size = 1,color_by = "State",label=T)+ggtitle("")

BEAM_branch3 <- BEAM(cds, branch_point = 3, cores = 20)

BEAM_branch3 <- BEAM_branch3[order(BEAM_branch3$qval),]
BEAM_branch3 <- BEAM_branch3[,c("gene_short_name", "pval", "qval")]
head(BEAM_branch3)

BEAM_res = BEAM_branch3

pdf("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/EPI_Ctrl_monocle_3.pdf",
    width = 5,height = 9)
my_branched_heatmap <- plot_genes_branched_heatmap(cds[row.names(subset(BEAM_res, qval < 1e-4)),], 
                                                   branch_point = 3,num_clusters = 4, use_gene_short_name = TRUE,
                                                   show_rownames = F,return_heatmap = TRUE)
dev.off()

saveRDS(my_branched_heatmap,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/EPI_CTRL_my_branched_heatmap.rds")

#取出各个cluster基因进行富集
annotation_df <- data.frame(
  Cluster=my_branched_heatmap[["annotation_row"]][["Cluster"]],
  gene=my_branched_heatmap[["ph"]][["tree_row"]][["labels"]]
)

table(annotation_df$Cluster) 
write.csv(annotation_df,
          "/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/EPI_CTRL_my_branched_heatmap.csv")

gene3

plot_genes_in_pseudotime(cds1[N2,],ncol = 2,color_by = "donor2")









#对富集进行柱形图绘制-----
#CTRL----
term=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/分支热图富集/CTRL_branch_term.csv")

term1=subset(term,Cluster%in%"1")
term1=term1[order(-term1$P.Value),]

ggplot(term1,aes(x=-log10(P.Value),y=X.Term))+
  geom_bar(stat="identity", width =0.8,color="black",fill="#C87DFF")+
  theme_classic(base_size =25) + theme(panel.border = element_blank())+
  theme(legend.position ="none",axis.text = element_text(face="bold",color="black"))+
  xlab("")+ylab("")+scale_y_discrete(limits=term1$X.Term)
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/分支热图富集/CTRL_branch_term1.pdf",
       height = 6,width = 10)


term2=subset(term,Cluster%in%"2")
term2=term2[order(-term2$P.Value),]

ggplot(term2,aes(x=-log10(P.Value),y=X.Term))+
  geom_bar(stat="identity", width =0.8,color="black",fill="#F8776D")+
  theme_classic(base_size =25) + theme(panel.border = element_blank())+
  theme(legend.position ="none",axis.text = element_text(face="bold",color="black"))+
  xlab("")+ylab("")+scale_y_discrete(limits=term2$X.Term)
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/分支热图富集/CTRL_branch_term2.pdf",
       height = 6,width = 10)


term3=subset(term,Cluster%in%"3")
term3=term3[order(-term3$P.Value),]

ggplot(term3,aes(x=-log10(P.Value),y=X.Term))+
  geom_bar(stat="identity", width =0.8,color="black",fill="#FE81F0")+
  theme_classic(base_size =25) + theme(panel.border = element_blank())+
  theme(legend.position ="none",axis.text = element_text(face="bold",color="black"))+
  xlab("")+ylab("")+scale_y_discrete(limits=term3$X.Term)
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/分支热图富集/CTRL_branch_term3.pdf",
       height = 6,width = 10)


term4=subset(term,Cluster%in%"4")
term4=term4[order(-term4$P.Value),]

ggplot(term4,aes(x=-log10(P.Value),y=X.Term))+
  geom_bar(stat="identity", width =0.8,color="black",fill="#7DAF00")+
  theme_classic(base_size =25) + theme(panel.border = element_blank())+
  theme(legend.position ="none",axis.text = element_text(face="bold",color="black"))+
  xlab("")+ylab("")+scale_y_discrete(limits=term4$X.Term)
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/分支热图富集/CTRL_branch_term4.pdf",
       height = 6,width = 10)


term1=subset(term,Cluster%in%"1")
term1=term1[order(-term1$P.Value),]

ggplot(term1,aes(x=-log10(P.Value),y=X.Term))+
  geom_bar(stat="identity", width =0.8,color="black",fill="#C87DFF")+
  theme_classic(base_size =25) + theme(panel.border = element_blank())+
  theme(legend.position ="none",axis.text = element_blank())+
  xlab("")+ylab("")+scale_y_discrete(limits=term1$X.Term)
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/分支热图富集/CTRL_branch_term1_1.pdf",
       height = 6,width = 6)


term2=subset(term,Cluster%in%"2")
term2=term2[order(-term2$P.Value),]

ggplot(term2,aes(x=-log10(P.Value),y=X.Term))+
  geom_bar(stat="identity", width =0.8,color="black",fill="#F8776D")+
  theme_classic(base_size =25) + theme(panel.border = element_blank())+
  theme(legend.position ="none",axis.text = element_blank())+
  xlab("")+ylab("")+scale_y_discrete(limits=term2$X.Term)
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/分支热图富集/CTRL_branch_term2_1.pdf",
       height = 6,width = 6)


term3=subset(term,Cluster%in%"3")
term3=term3[order(-term3$P.Value),]

ggplot(term3,aes(x=-log10(P.Value),y=X.Term))+
  geom_bar(stat="identity", width =0.8,color="black",fill="#FE81F0")+
  theme_classic(base_size =25) + theme(panel.border = element_blank())+
  theme(legend.position ="none",axis.text = element_blank())+
  xlab("")+ylab("")+scale_y_discrete(limits=term3$X.Term)
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/分支热图富集/CTRL_branch_term3_1.pdf",
       height = 6,width = 6)


term4=subset(term,Cluster%in%"4")
term4=term4[order(-term4$P.Value),]

ggplot(term4,aes(x=-log10(P.Value),y=X.Term))+
  geom_bar(stat="identity", width =0.8,color="black",fill="#7DAF00")+
  theme_classic(base_size =25) + theme(panel.border = element_blank())+
  theme(legend.position ="none",axis.text = element_blank())+
  xlab("")+ylab("")+scale_y_discrete(limits=term4$X.Term)
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/分支热图富集/CTRL_branch_term4_1.pdf",
       height = 6,width = 6)

#IUA----
term=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/分支热图富集/IUA_branch_term.csv")

term1=subset(term,Cluster%in%"1")
term1=term1[order(-term1$P.Value),]

ggplot(term1,aes(x=-log10(P.Value),y=X.Term))+
  geom_bar(stat="identity", width =0.8,color="black",fill="#00C4FF")+
  theme_classic(base_size =25) + theme(panel.border = element_blank())+
  theme(legend.position ="none",axis.text = element_text(face="bold",color="black"))+
  xlab("")+ylab("")+scale_y_discrete(limits=term1$X.Term)
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/分支热图富集/IUA_branch_term1.pdf",
       height = 6,width = 10)


term2=subset(term,Cluster%in%"2")
term2=term2[order(-term2$P.Value),]

ggplot(term2,aes(x=-log10(P.Value),y=X.Term))+
  geom_bar(stat="identity", width =0.8,color="black",fill="#FF9289")+
  theme_classic(base_size =25) + theme(panel.border = element_blank())+
  theme(legend.position ="none",axis.text = element_text(face="bold",color="black"))+
  xlab("")+ylab("")+scale_y_discrete(limits=term2$X.Term)
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/分支热图富集/IUA_branch_term2.pdf",
       height = 6,width = 10)


term3=subset(term,Cluster%in%"3")
term3=term3[order(-term3$P.Value),]

ggplot(term3,aes(x=-log10(P.Value),y=X.Term))+
  geom_bar(stat="identity", width =0.8,color="black",fill="#E199FF")+
  theme_classic(base_size =25) + theme(panel.border = element_blank())+
  theme(legend.position ="none",axis.text = element_text(face="bold",color="black"))+
  xlab("")+ylab("")+scale_y_discrete(limits=term3$X.Term)
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/分支热图富集/IUA_branch_term3.pdf",
       height = 6,width = 10)


term4=subset(term,Cluster%in%"4")
term4=term4[order(-term4$P.Value),]

ggplot(term4,aes(x=-log10(P.Value),y=X.Term))+
  geom_bar(stat="identity", width =0.8,color="black",fill="#96CA00")+
  theme_classic(base_size =25) + theme(panel.border = element_blank())+
  theme(legend.position ="none",axis.text = element_text(face="bold",color="black"))+
  xlab("")+ylab("")+scale_y_discrete(limits=term4$X.Term)
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/分支热图富集/IUA_branch_term4.pdf",
       height = 6,width = 10)


term5=subset(term,Cluster%in%"5")
term5=term5[order(-term5$P.Value),]

ggplot(term5,aes(x=-log10(P.Value),y=X.Term))+
  geom_bar(stat="identity", width =0.8,color="black",fill="#FF80E7")+
  theme_classic(base_size =25) + theme(panel.border = element_blank())+
  theme(legend.position ="none",axis.text = element_text(face="bold",color="black"))+
  xlab("")+ylab("")+scale_y_discrete(limits=term5$X.Term)
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/分支热图富集/IUA_branch_term5.pdf",
       height = 6,width = 10)



term1=subset(term,Cluster%in%"1")
term1=term1[order(-term1$P.Value),]

ggplot(term1,aes(x=-log10(P.Value),y=X.Term))+
  geom_bar(stat="identity", width =0.8,color="black",fill="#00C4FF")+
  theme_classic(base_size =25) + theme(panel.border = element_blank())+
  theme(legend.position ="none",axis.text = element_blank())+
  xlab("")+ylab("")+scale_y_discrete(limits=term1$X.Term)
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/分支热图富集/IUA_branch_term1_1.pdf",
       height = 6,width = 6)


term2=subset(term,Cluster%in%"2")
term2=term2[order(-term2$P.Value),]

ggplot(term2,aes(x=-log10(P.Value),y=X.Term))+
  geom_bar(stat="identity", width =0.8,color="black",fill="#FF9289")+
  theme_classic(base_size =25) + theme(panel.border = element_blank())+
  theme(legend.position ="none",axis.text = element_blank())+
  xlab("")+ylab("")+scale_y_discrete(limits=term2$X.Term)
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/分支热图富集/IUA_branch_term2_1.pdf",
       height = 6,width = 6)


term3=subset(term,Cluster%in%"3")
term3=term3[order(-term3$P.Value),]

ggplot(term3,aes(x=-log10(P.Value),y=X.Term))+
  geom_bar(stat="identity", width =0.8,color="black",fill="#E199FF")+
  theme_classic(base_size =25) + theme(panel.border = element_blank())+
  theme(legend.position ="none",axis.text = element_blank())+
  xlab("")+ylab("")+scale_y_discrete(limits=term3$X.Term)
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/分支热图富集/IUA_branch_term3_1.pdf",
       height = 6,width = 6)


term4=subset(term,Cluster%in%"4")
term4=term4[order(-term4$P.Value),]

ggplot(term4,aes(x=-log10(P.Value),y=X.Term))+
  geom_bar(stat="identity", width =0.8,color="black",fill="#96CA00")+
  theme_classic(base_size =25) + theme(panel.border = element_blank())+
  theme(legend.position ="none",axis.text = element_blank())+
  xlab("")+ylab("")+scale_y_discrete(limits=term4$X.Term)
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/分支热图富集/IUA_branch_term4_1.pdf",
       height = 6,width = 6)



term5=subset(term,Cluster%in%"5")
term5=term5[order(-term5$P.Value),]

ggplot(term5,aes(x=-log10(P.Value),y=X.Term))+
  geom_bar(stat="identity", width =0.8,color="black",fill="#FF80E7")+
  theme_classic(base_size =25) + theme(panel.border = element_blank())+
  theme(legend.position ="none",axis.text = element_blank())+
  xlab("")+ylab("")+scale_y_discrete(limits=term5$X.Term)
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/分支热图富集/IUA_branch_term5_1.pdf",
       height = 6,width = 6)

#展示上述基因的富集结果-----
cds1=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/EPI_all_CTRL.rds")
cds2=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/EPI_all_IUA.rds")



gene3=c("FOXO1","ESR1","PAEP","SPP1")

cds1$donor2=gsub("Ctrl_","",cds1$donor2)
cds2$donor2=gsub("IUA_","",cds2$donor2)

P1=plot_genes_in_pseudotime(cds1[gene3,],ncol = 1,color_by = "donor2")
P2=plot_genes_in_pseudotime(cds2[gene3,],ncol = 1,color_by = "donor2")


pdf("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/p1.pdf",
    width = 6,height = 10)
ggarrange(P1,P2,common.legend = T)
dev.off()



#对基质monocle结果进行展示----
cds=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/STR_ALL_CTRL.rds")

plot_cell_trajectory(cds,cell_size = 1,color_by = "State",label=T)+ggtitle("")

plot_cell_trajectory(cds,cell_size = 1,color_by = "donor2",label=T)+ggtitle("")
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/STR_Ctrl_monocle1.pdf",
       width = 7,height = 7)

plot_cell_trajectory(cds,cell_size = 1,color_by = "Pseudotime")
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/STR_Ctrl_monocle2.pdf",
       width = 7,height = 7)



cds$type5="Wang et al."
cds$type5[cds$donor%in%"Pro_IUA2"]<-"Adhesion"

cds$type5[cds$donor%in%"Pro_IUA1"]<-"Adjacency"
cds$type5[cds$donor%in%"Pro_CTRL"]<-"Ctrl"

cds$type5[cds$donor%in%c("EMM_Pro_late_IUA","EMM_Pro_late_Ctrl")]<-"Lv et al."
cds$type5[cds$donor%in%c("NC_Sec_IUA")]<-"Santamaria et al."


plot_cell_trajectory(cds,cell_size = 1,color_by = "donor2",label=T)+ggtitle("")+
  facet_wrap(~type5, nrow = 1)


ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/STR_Ctrl_monocle3.pdf",
       width = 14,height = 7)


cds=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/STR_ALL_IUA.rds")



cds$type5="Wang et al."
cds$type5[cds$donor%in%"Pro_IUA2"]<-"Adhesion"

cds$type5[cds$donor%in%"Pro_IUA1"]<-"Adjacency"
cds$type5[cds$donor%in%"Pro_CTRL"]<-"Ctrl"

cds$type5[cds$donor%in%c("EMM_Pro_late_IUA","EMM_Pro_late_Ctrl")]<-"Lv et al."
cds$type5[cds$donor%in%c("NC_Sec_IUA")]<-"Santamaria et al."


plot_cell_trajectory(cds,cell_size = 1,color_by = "donor2",label=T)+ggtitle("")
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/STR_IUA_monocle1.pdf",
       width = 7,height = 7)

plot_cell_trajectory(cds,cell_size = 1,color_by = "Pseudotime")
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/STR_IUA_monocle2.pdf",
       width = 7,height = 7)



plot_cell_trajectory(cds,cell_size = 1,color_by = "donor2",label=T)+ggtitle("")+
  facet_wrap(~type5, nrow = 1)


ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/monocle/STR_IUA_monocle3.pdf",
       width = 14,height = 7)









#对monocle分支热图的cluster基因进行富集柱形图绘制----
#CTRL组----
#Cluster1





#对数据进行映射NCB基质细胞、上皮细胞亚群和我们的亚群进行映射并进行图谱展示-----
#上皮
library(SeuratDisk)
library(Seurat)
library(ggpubr)

WEN_EPI<-LoadH5Seurat("/data/DATAprocess/chentw/zigongneimo/WEN2/h5seurat文件/endo-2022_epithelial.h5seurat",
                      assays="RNA")


EPI_allsingle3=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/EPI_allsingle.rds")



WEN_EPI=NormalizeData(WEN_EPI, normalization.method = "LogNormalize", 
                      scale.factor = 10000)

common_features=intersect(rownames(EPI_allsingle3),rownames(WEN_EPI))


WEN_EPI=RunUMAP(WEN_EPI,return.model = T,dims = 1:30)
DimPlot(WEN_EPI,group.by = "subtypes")



anchors=FindTransferAnchors(reference = WEN_EPI,
                            query = EPI_allsingle3,
                            normalization.method = "LogNormalize",
                            reference.reduction = "pca",features = common_features,
                            dims = 1:30)

EPI3=MapQuery(anchorset = anchors,
              query = EPI_allsingle3,
              reference = WEN_EPI,
              refdata = list(celltype.l1="subtypes"),
              reference.reduction = "pca",
              reduction.model = "umap")

EPI3$predicted.celltype.l1=factor(EPI3$predicted.celltype.l1,levels = 
                                    names(table(WEN_EPI$subtypes)))


P1=DimPlot(WEN_EPI,group.by = "subtypes",label = T,repel = T,label.size = 7)+ggtitle("")+
  theme(legend.text = element_text(size = 17))+
  ggtitle("Tan et.at.")


P2=DimPlot(EPI3,group.by = "predicted.celltype.l1",label = T,repel = T,label.size = 7,reduction="ref.umap")+
  ggtitle("")+
  scale_color_manual(values=c("#FF6D64","#A4A511","#00C283","#00C2C7",
                              "#00B3F7","#9592FE","#FB63EF","#FF55B6"))+
  theme(legend.text = element_text(size = 17))


ggarrange(P1,P2,common.legend = T,legend = "right")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/上皮映射1.pdf",
       width = 17,height = 7)



#基质
library(SeuratDisk)
library(Seurat)


WEN_STR<-LoadH5Seurat("/data/DATAprocess/chentw/zigongneimo/WEN2/h5seurat文件/endo-2022_stromal.h5seurat",
                      assays="RNA")


EnSc=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0718.rds")


WEN_STR <- NormalizeData(WEN_STR, normalization.method = "LogNormalize", 
                         scale.factor = 10000)
WEN_STR <- FindVariableFeatures(WEN_STR, selection.method = "vst", nfeatures = 2000)
VlnPlot(WEN_STR, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

#WEN_STR进行标准化然后PCA分析
DefaultAssay(WEN_STR) <- "RNA"
all.genes_WEN_STR <- rownames(WEN_STR)
WEN_STR <- ScaleData(object = WEN_STR, features = all.genes_WEN_STR)
WEN_STR <- RunPCA(WEN_STR, features = VariableFeatures(object = WEN_STR))
WEN_STR=RunUMAP(WEN_STR,return.model = T,dims = 1:30)
DimPlot(WEN_STR,group.by = "Cell.type",reduction = "umap")

common_features=intersect(rownames(EnSc),rownames(WEN_STR))


WEN_STR=RunUMAP(WEN_STR,return.model = T,dims = 1:30)

DimPlot(WEN_STR,group.by = "subtypes",label = T,label.size = 7,repel = T)



anchors=FindTransferAnchors(reference = WEN_STR,
                            query = EnSc,
                            normalization.method = "LogNormalize",
                            reference.reduction = "pca",
                            features = common_features,
                            dims = 1:30)

EnSc3=MapQuery(anchorset = anchors,
               query = EnSc,
               reference = WEN_STR,
               refdata = list(celltype.l1="subtypes"),
               reference.reduction = "pca",
               reduction.model = "umap")





P1=DimPlot(WEN_STR,group.by = "subtypes",label = T,repel = T)


P1=DimPlot(WEN_STR,group.by = "subtypes",label = T,repel = T,label.size = 7)+ggtitle("")+
  theme(legend.text = element_text(size = 17))


P2=DimPlot(EnSc3,group.by = "predicted.celltype.l1",label = T,repel = T,label.size = 7,reduction="ref.umap")+
  ggtitle("")+
  theme(legend.text = element_text(size = 17))


ggarrange(P1,P2,common.legend = T,legend = "right")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/NCB_基质映射1.pdf",
       width = 17,height = 7)















#和空间转录组NG，数据上皮基质细胞进行映射-----
library(Seurat)
library(dplyr)
library(SeuratDisk)



WEN<-LoadH5Seurat("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/空间转录组scRNA文件/endometrium_all.h5seurat",
                  assays="RNA")

table(WEN$Cell.type)

WEN_EPI=subset(WEN,Cell.type%in%c("Lumenal","SOX9","Glandular","Ciliated"))

EPI_allsingle3=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/EPI_allsingle.rds")


common_features=intersect(rownames(EPI_allsingle3),rownames(WEN_EPI))


WEN_EPI <- NormalizeData(WEN_EPI, normalization.method = "LogNormalize", 
                         scale.factor = 10000)
WEN_EPI <- FindVariableFeatures(WEN_EPI, selection.method = "vst", nfeatures = 2000)
VlnPlot(WEN_EPI, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

#WEN_EPI进行标准化然后PCA分析
DefaultAssay(WEN_EPI) <- "RNA"
all.genes_WEN_EPI <- rownames(WEN_EPI)
WEN_EPI <- ScaleData(object = WEN_EPI, features = all.genes_WEN_EPI)
WEN_EPI <- RunPCA(WEN_EPI, features = VariableFeatures(object = WEN_EPI))
WEN_EPI=RunUMAP(WEN_EPI,return.model = T,dims = 1:30)
DimPlot(WEN_EPI,group.by = "Cell.type",reduction = "umap")

anchors=FindTransferAnchors(reference = WEN_EPI,
                            query = EPI_allsingle3,
                            normalization.method = "LogNormalize",
                            reference.reduction = "pca",features = common_features,
                            dims = 1:30)

EPI3=MapQuery(anchorset = anchors,
              query = EPI_allsingle3,
              reference = WEN_EPI,
              refdata = list(celltype.l1="Cell.type"),
              reference.reduction = "pca",
              reduction.model = "umap")


P1=DimPlot(WEN_EPI,group.by = "Cell.type",label = T,repel = T,
           reduction = "umap")

P1=DimPlot(WEN_EPI,group.by = "Cell.type",label = T,repel = T,label.size = 7,reduction = "umap_scvi_sampl_cc")+
  ggtitle("")+
  theme(legend.text = element_text(size = 17))+xlab("umap1")+ylab("umap2")


P2=DimPlot(EPI3,group.by = "predicted.celltype.l1",label = T,repel = T,label.size = 7,
           reduction="umap")+
  ggtitle("")+
  theme(legend.text = element_text(size = 17))


ggarrange(P1,P2,common.legend = T,legend = "right")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/NG_上皮映射1.pdf",
       width = 17,height = 7)

#基质细胞映射
WENa_STR<-LoadH5Seurat("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/空间转录组scRNA文件/endometrium_all.h5seurat",
                       assays="RNA")

table(WEN_STR$Cell.type)

WEN_STR=subset(WEN_STR,Cell.type%in%c("eS","dS","Fibroblast C7","uSMC","PV MYH11","PV STEAP4"))


table(WEN$Cell.type)

EnSc=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0718.rds")


common_features=intersect(rownames(EnSc),rownames(WEN_STR))


WEN_STR <- NormalizeData(WEN_STR, normalization.method = "LogNormalize", 
                         scale.factor = 10000)
WEN_STR <- FindVariableFeatures(WEN_STR, selection.method = "vst", nfeatures = 2000)
VlnPlot(WEN_STR, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

#WEN_STR进行标准化然后PCA分析
DefaultAssay(WEN_STR) <- "RNA"
all.genes_WEN_STR <- rownames(WEN_STR)
WEN_STR <- ScaleData(object = WEN_STR, features = all.genes_WEN_STR)
WEN_STR <- RunPCA(WEN_STR, features = VariableFeatures(object = WEN_STR))
WEN_STR=RunUMAP(WEN_STR,return.model = T,dims = 1:30)
DimPlot(WEN_STR,group.by = "Cell.type",reduction = "umap")

anchors=FindTransferAnchors(reference = WEN_STR,
                            query = EnSc,
                            normalization.method = "LogNormalize",
                            reference.reduction = "pca",features = common_features,
                            dims = 1:30)

EnSc3=MapQuery(anchorset = anchors,
               query = EnSc,
               reference = WEN_STR,
               refdata = list(celltype.l1="Cell.type"),
               reference.reduction = "pca",
               reduction.model = "umap")


P1=DimPlot(WEN_STR,group.by = "Cell.type",label = T,repel = T,
           reduction = "umap")

P1=DimPlot(WEN_STR,group.by = "Cell.type",label = T,repel = T,label.size = 7,reduction = "umap")+
  ggtitle("")+
  theme(legend.text = element_text(size = 17))+xlab("umap1")+ylab("umap2")


P2=DimPlot(EnSc3,group.by = "predicted.celltype.l1",label = T,repel = T,label.size = 7,
           reduction="ref.umap")+
  ggtitle("")+
  theme(legend.text = element_text(size = 17))


ggarrange(P1,P2,common.legend = T,legend = "right")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/映射结果汇总/NG_基质映射1.pdf",
       width = 17,height = 7)


#和PNAS数据进行整合映射-----




#和IUA_sec整合映射-----
WEN2<-readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/IUA_sec.rds")

table(WEN2$cell_subtypes)


WEN_EPI=subset(WEN2,cell_subtypes%in%c("AS-Epithelium","Epi-Ciliated","Epi-Glandular",
                                       "Epi-Glandular secretoryd","Epi-Luminal","Epi-PDGFRA","Epi-SOX9"))

DimPlot(WEN_EPI,group.by = "cell_subtypes",label = T,repel = T,
        reduction = "umap")



EPI_allsingle3=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/EPI_allsingle.rds")


common_features=intersect(rownames(EPI_allsingle3),rownames(WEN_EPI))

WEN_EPI <- NormalizeData(WEN_EPI, normalization.method = "LogNormalize", 
                         scale.factor = 10000)
WEN_EPI <- FindVariableFeatures(WEN_EPI, selection.method = "vst", nfeatures = 2000)
VlnPlot(WEN_EPI, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

#WEN_EPI进行标准化然后PCA分析
DefaultAssay(WEN_EPI) <- "RNA"
all.genes_WEN_EPI <- rownames(WEN_EPI)
WEN_EPI <- ScaleData(object = WEN_EPI, features = all.genes_WEN_EPI)
WEN_EPI <- RunPCA(WEN_EPI, features = VariableFeatures(object = WEN_EPI))
WEN_EPI=RunUMAP(WEN_EPI,return.model = T,dims = 1:30)
DimPlot(WEN_EPI,group.by = "cell_subtypes",reduction = "umap")


anchors=FindTransferAnchors(reference = WEN_EPI,
                            query = EPI_allsingle3,
                            normalization.method = "LogNormalize",
                            reference.reduction = "pca",features = common_features,
                            dims = 1:30)

EPI3=MapQuery(anchorset = anchors,
              query = EPI_allsingle3,
              reference = WEN_EPI,
              refdata = list(celltype.l1="cell_subtypes"),
              reference.reduction = "pca",
              reduction.model = "umap")

P1=DimPlot(WEN_EPI,group.by = "cell_subtypes",label = T,repel = T,label.size = 7,
           reduction="umap")+
  ggtitle("")+
  theme(legend.text = element_text(size = 17))



P2=DimPlot(EPI3,group.by = "predicted.celltype.l1",label = T,repel = T,label.size = 7,
           reduction="ref.umap")+
  ggtitle("")+
  theme(legend.text = element_text(size = 17))


ggarrange(P1,P2,common.legend = T,legend = "right")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/IUASEC_上皮映射1.pdf",
       width = 17,height = 7)


WEN_STR=subset(WEN2,cell_subtypes%in%c("AS-uSMC","Stroma","Cycling Stroma","Cycling PV","PV STEAP4","PV MYH11","Myofibroblasts"))

DimPlot(WEN_STR,group.by = "cell_subtypes",label = T,repel = T,
        reduction = "umap")


EnSc=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0718.rds")


common_features=intersect(rownames(EnSc),rownames(WEN_STR))

WEN_STR <- NormalizeData(WEN_STR, normalization.method = "LogNormalize", 
                         scale.factor = 10000)
WEN_STR <- FindVariableFeatures(WEN_STR, selection.method = "vst", nfeatures = 2000)
VlnPlot(WEN_STR, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

#WEN_STR进行标准化然后PCA分析
DefaultAssay(WEN_STR) <- "RNA"
all.genes_WEN_STR <- rownames(WEN_STR)
WEN_STR <- ScaleData(object = WEN_STR, features = all.genes_WEN_STR)
WEN_STR <- RunPCA(WEN_STR, features = VariableFeatures(object = WEN_STR))
WEN_STR=RunUMAP(WEN_STR,return.model = T,dims = 1:30)
DimPlot(WEN_STR,group.by = "cell_subtypes",reduction = "umap")


anchors=FindTransferAnchors(reference = WEN_STR,
                            query = EnSc,
                            normalization.method = "LogNormalize",
                            reference.reduction = "pca",features = common_features,
                            dims = 1:30)

STR3=MapQuery(anchorset = anchors,
              query = EnSc,
              reference = WEN_STR,
              refdata = list(celltype.l1="cell_subtypes"),
              reference.reduction = "pca",
              reduction.model = "umap")

P1=DimPlot(WEN_STR,group.by = "cell_subtypes",label = T,repel = T,label.size = 7,
           reduction="umap")+
  ggtitle("")+
  theme(legend.text = element_text(size = 17))



P2=DimPlot(STR3,group.by = "predicted.celltype.l1",label = T,repel = T,label.size = 7,
           reduction="ref.umap")+
  ggtitle("")+
  theme(legend.text = element_text(size = 17))


ggarrange(P1,P2,common.legend = T,legend = "right")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/IUASEC_上皮映射1.pdf",
       width = 17,height = 7)


#和子宫内膜整合图谱（NG）进行映射分析-----
library(anndata)
library(Seurat)
library(SingleCellExperiment)
library(SeuratDisk)
library(zellkonverter)
library(hdf5r)

library(Seurat)
library("reticulate")

anndata <- import("anndata")


adata <- anndata$read_h5ad("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/子宫内膜整合数据/endometriumAtlasV2_cells_with_counts.h5ad")



# 提取基因表达矩阵（稀疏矩阵格式）
# 转换为 R 的矩阵格式（如果需要）
X_dense <- as.data.frame(adata$X)




# 提取细胞元数据
obs <- adata$obs

# 提取基因元数据
var <- adata$var



rownames(X_dense)=rownames(obs)
colnames(X_dense)=rownames(var)


X_dense=as.matrix(X_dense)

# 创建 Seurat 对象
seurat_obj <- CreateSeuratObject(
  counts = t(X_dense),  # 转置矩阵以符合 Seurat 的格式
  meta.data = obs
)


saveRDS(seurat_obj,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/子宫内膜整合数据/all_WEN3.rds")

seurat_obj=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/子宫内膜整合数据/all_WEN3.rds")

seurat_obj <- NormalizeData(seurat_obj, normalization.method = "LogNormalize", 
                            scale.factor = 10000)
seurat_obj <- FindVariableFeatures(seurat_obj, selection.method = "vst", nfeatures = 2000)

#seurat_obj进行标准化然后PCA分析
DefaultAssay(seurat_obj) <- "RNA"
all.genes_seurat_obj <- rownames(seurat_obj)
seurat_obj <- ScaleData(object = seurat_obj, features = all.genes_seurat_obj)
seurat_obj <- RunPCA(seurat_obj, features = VariableFeatures(object = seurat_obj))

#确定seurat_obj的PC维数
seurat_obj <- JackStraw(seurat_obj, num.replicate = 100) 
seurat_obj <- ScoreJackStraw(seurat_obj, dims = 1:20) 
ElbowPlot(seurat_obj,ndims = 30, reduction = "pca") 

#seurat_obj进行细胞细胞聚类分析(UMAP)
seurat_obj=RunUMAP(seurat_obj, dims = 1:30)
DimPlot(seurat_obj,reduction="umap",label=TRUE)
seurat_obj <- FindNeighbors(seurat_obj, dims = 1:30)
seurat_obj <- FindClusters(seurat_obj, resolution = 0.3)
seurat_obj <- FindClusters(seurat_obj, resolution = 0.6)
DimPlot(seurat_obj,reduction="umap",label=TRUE)


# 提取 UMAP 坐标
umap_coords <- adata$obsm$get("X_umap")

# 将 UMAP 坐标转换为 data.frame
umap_df <- as.data.frame(umap_coords)
colnames(umap_df) <- c("UMAP_1", "UMAP_2")

DimPlot(seurat_obj,reduction="umap",label=TRUE,raster = F,group.by = "celltype")

saveRDS(seurat_obj,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/子宫内膜整合数据/all_WEN3.rds")



#细致的细胞类型的映射-上皮
WEN_EPI=subset(seurat_obj,lineage%in%c("Epithelial"))

DimPlot(WEN_EPI,group.by = "celltype",label = T,repel = T,
        reduction = "umap")



EPI_allsingle3=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/EPI_allsingle.rds")


common_features=intersect(rownames(EPI_allsingle3),rownames(WEN_EPI))

WEN_EPI <- NormalizeData(WEN_EPI, normalization.method = "LogNormalize", 
                         scale.factor = 10000)
WEN_EPI <- FindVariableFeatures(WEN_EPI, selection.method = "vst", nfeatures = 2000)
VlnPlot(WEN_EPI, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

#WEN_EPI进行标准化然后PCA分析
DefaultAssay(WEN_EPI) <- "RNA"
all.genes_WEN_EPI <- rownames(WEN_EPI)
WEN_EPI <- ScaleData(object = WEN_EPI, features = all.genes_WEN_EPI)
WEN_EPI <- RunPCA(WEN_EPI, features = VariableFeatures(object = WEN_EPI))
WEN_EPI=RunUMAP(WEN_EPI,return.model = T,dims = 1:30)
DimPlot(WEN_EPI,group.by = "celltype",reduction = "umap")


anchors=FindTransferAnchors(reference = WEN_EPI,
                            query = EPI_allsingle3,
                            normalization.method = "LogNormalize",
                            reference.reduction = "pca",features = common_features,
                            dims = 1:30)

EPI3=MapQuery(anchorset = anchors,
              query = EPI_allsingle3,
              reference = WEN_EPI,
              refdata = list(celltype.l1="celltype"),
              reference.reduction = "pca",
              reduction.model = "umap")

P1=DimPlot(WEN_EPI,group.by = "celltype",label = T,repel = T,label.size = 7,
           reduction="umap")+
  ggtitle("")+
  theme(legend.text = element_text(size = 17))



P2=DimPlot(EPI3,group.by = "predicted.celltype.l1",label = T,repel = T,label.size = 7,
           reduction="ref.umap")+
  ggtitle("")+
  theme(legend.text = element_text(size = 17))


ggarrange(P1,P2,common.legend = T,legend = "right")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/NG_integrated_EPI2.pdf",
       width = 17,height = 7)







#广泛的细胞类型的映射
WEN_EPI=subset(seurat_obj,lineage%in%c("Epithelial"))


WEN_EPI$label_long=factor(WEN_EPI$label_long)

table(WEN_EPI$label_long)

WEN_EPI$celltype2="N"

WEN_EPI$celltype2[WEN_EPI$label_long%in%c("1 | SOX9 basalis","2 | SOX9 functionalis I",
                                          "3 | SOX9 functionalis II","4 | SOX9 luminal")]<-"SOX9+"

WEN_EPI$celltype2[WEN_EPI$label_long%in%c("5 | Cycling")]<-"Cycling"

WEN_EPI$celltype2[WEN_EPI$label_long%in%c("6 | preGlandular","7 | Glandular",
                                          "8 | Glandular secretory","9 | Glandular secretory FGF7")]<-"Glandular"

WEN_EPI$celltype2[WEN_EPI$label_long%in%c("10 | preLuminal","11 | Luminal")]<-"Luminal"


WEN_EPI$celltype2[WEN_EPI$label_long%in%c("12 | preCiliated","13 | Ciliated")]<-"Ciliated"


WEN_EPI$celltype2[WEN_EPI$label_long%in%c("14 | MUC5B","15 | KRT5")]<-"Endocervical"

WEN_EPI$celltype2[WEN_EPI$label_long%in%c("16 | eHormones")]<-"Hormones"


DimPlot(WEN_EPI,group.by = "celltype2",label = T,repel = T,
        reduction = "umap")



EPI_allsingle3=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/EPI_allsingle.rds")


common_features=intersect(rownames(EPI_allsingle3),rownames(WEN_EPI))

WEN_EPI <- NormalizeData(WEN_EPI, normalization.method = "LogNormalize", 
                         scale.factor = 10000)
WEN_EPI <- FindVariableFeatures(WEN_EPI, selection.method = "vst", nfeatures = 2000)
VlnPlot(WEN_EPI, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

#WEN_EPI进行标准化然后PCA分析
DefaultAssay(WEN_EPI) <- "RNA"
all.genes_WEN_EPI <- rownames(WEN_EPI)
WEN_EPI <- ScaleData(object = WEN_EPI, features = all.genes_WEN_EPI)
WEN_EPI <- RunPCA(WEN_EPI, features = VariableFeatures(object = WEN_EPI))
WEN_EPI=RunUMAP(WEN_EPI,return.model = T,dims = 1:30)
DimPlot(WEN_EPI,group.by = "celltype2",reduction = "umap")


anchors=FindTransferAnchors(reference = WEN_EPI,
                            query = EPI_allsingle3,
                            normalization.method = "LogNormalize",
                            reference.reduction = "pca",features = common_features,
                            dims = 1:30)

EPI3=MapQuery(anchorset = anchors,
              query = EPI_allsingle3,
              reference = WEN_EPI,
              refdata = list(celltype.l1="celltype2"),
              reference.reduction = "pca",
              reduction.model = "umap")

P1=DimPlot(WEN_EPI,group.by = "celltype2",label = T,repel = T,label.size = 7,
           reduction="umap")+
  ggtitle("")+
  theme(legend.text = element_text(size = 17))



P2=DimPlot(EPI3,group.by = "predicted.celltype.l1",label = T,repel = T,label.size = 7,
           reduction="ref.umap")+
  ggtitle("")+
  theme(legend.text = element_text(size = 17))


ggarrange(P1,P2,common.legend = T,legend = "right")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/NG_integrated_EPI2.pdf",
       width = 17,height = 7)







#基质细胞
seurat_obj=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/子宫内膜整合数据/all_WEN3.rds")



WEN_STR=subset(seurat_obj,lineage%in%c("Mesenchymal"))

DimPlot(WEN_STR,group.by = "celltype",label = T,repel = T,
        reduction = "umap",raster = F)

EnSc=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0718.rds")

common_features=intersect(rownames(EnSc),rownames(WEN_STR))

WEN_STR <- NormalizeData(WEN_STR, normalization.method = "LogNormalize", 
                         scale.factor = 10000)
WEN_STR <- FindVariableFeatures(WEN_STR, selection.method = "vst", nfeatures = 2000)
VlnPlot(WEN_STR, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

#WEN_STR进行标准化然后PCA分析
DefaultAssay(WEN_STR) <- "RNA"
all.genes_WEN_STR <- rownames(WEN_STR)
WEN_STR <- ScaleData(object = WEN_STR, features = all.genes_WEN_STR)
WEN_STR <- RunPCA(WEN_STR, features = VariableFeatures(object = WEN_STR))
WEN_STR=RunUMAP(WEN_STR,return.model = T,dims = 1:30)
DimPlot(WEN_STR,group.by = "celltype",reduction = "umap")


anchors=FindTransferAnchors(reference = WEN_STR,
                            query = EnSc,
                            normalization.method = "LogNormalize",
                            reference.reduction = "pca",features = common_features,
                            dims = 1:30)

STR3=MapQuery(anchorset = anchors,
              query = EnSc,
              reference = WEN_STR,
              refdata = list(celltype.l1="celltype"),
              reference.reduction = "pca",
              reduction.model = "umap")

P1=DimPlot(WEN_STR,group.by = "celltype",label = T,repel = T,label.size = 7,
           reduction="umap",raster = F)+
  ggtitle("")+
  theme(legend.text = element_text(size = 17))



P2=DimPlot(STR3,group.by = "predicted.celltype.l1",label = T,repel = T,label.size = 7,
           reduction="ref.umap")+
  ggtitle("")+
  theme(legend.text = element_text(size = 17))


ggarrange(P1,P2,common.legend = T,legend = "right")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/NG_integrated_STR1.pdf",
       width = 17,height = 7)





#广泛的细胞类型的映射
WEN_STR=subset(seurat_obj,lineage%in%c("Mesenchymal"))

WEN_STR$label_long=factor(WEN_STR$label_long)

table(WEN_STR$label_long)

WEN_STR$celltype2="N"

WEN_STR$celltype2[WEN_STR$label_long%in%c("17 | uSMCs",
                                          "18 | mPV")]<-"Myometrial"

WEN_STR$celltype2[WEN_STR$label_long%in%c("19 | ePV 1a","20 | ePV 1b","21 | ePV 2")]<-"Endometrial PV"

WEN_STR$celltype2[WEN_STR$label_long%in%c("22 | eStromal MMPs",
                                          "23 | eStromal","24 | eStromal cycling")]<-"Pro Stromal"

WEN_STR$celltype2[WEN_STR$label_long%in%c("25 | dStromal early","26 | dStromal mid",
                                          "27 | dStromal late")]<-"Sec Stromal"


WEN_STR$celltype2[WEN_STR$label_long%in%c("28 | Fibroblast basalis")]<-"Fibroblast"


WEN_STR$celltype2[WEN_STR$label_long%in%c("29 | HOXA13")]<-"Endocervical"

WEN_STR$celltype2[WEN_STR$label_long%in%c("30 | sHormones","31 | dHormones")]<-"Hormones"


DimPlot(WEN_STR,group.by = "celltype2",label = T,repel = T,
        reduction = "umap",raster = F)



EnSc=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/STR_harmony2/rds文件存放点/STR_0718.rds")


common_features=intersect(rownames(EnSc),rownames(WEN_STR))

WEN_STR <- NormalizeData(WEN_STR, normalization.method = "LogNormalize", 
                         scale.factor = 10000)
WEN_STR <- FindVariableFeatures(WEN_STR, selection.method = "vst", nfeatures = 2000)
VlnPlot(WEN_STR, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

#WEN_STR进行标准化然后PCA分析
DefaultAssay(WEN_STR) <- "RNA"
all.genes_WEN_STR <- rownames(WEN_STR)
WEN_STR <- ScaleData(object = WEN_STR, features = all.genes_WEN_STR)
WEN_STR <- RunPCA(WEN_STR, features = VariableFeatures(object = WEN_STR))
WEN_STR=RunUMAP(WEN_STR,return.model = T,dims = 1:30)
DimPlot(WEN_STR,group.by = "celltype2",reduction = "umap")


anchors=FindTransferAnchors(reference = WEN_STR,
                            query = EnSc,
                            normalization.method = "LogNormalize",
                            reference.reduction = "pca",features = common_features,
                            dims = 1:30)

STR3=MapQuery(anchorset = anchors,
              query = EnSc,
              reference = WEN_STR,
              refdata = list(celltype.l1="celltype2"),
              reference.reduction = "pca",
              reduction.model = "umap")

P1=DimPlot(WEN_STR,group.by = "celltype2",label = T,repel = T,label.size = 7,
           reduction="umap",raster = F)+
  ggtitle("")+
  theme(legend.text = element_text(size = 17))



P2=DimPlot(STR3,group.by = "predicted.celltype.l1",label = T,repel = T,label.size = 7,
           reduction="ref.umap")+
  ggtitle("")+
  theme(legend.text = element_text(size = 17))


ggarrange(P1,P2,common.legend = T,legend = "right")

ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/NG_integrated_STR2.pdf",
       width = 17,height = 7)




#查看基质细胞的比例和之前的细胞类型
EnSc$cell_type0719=STR3$predicted.celltype.l1

DimPlot(EnSc,group.by = "cell_type0718",label = T,label.size = 12)+
  DimPlot(EnSc,group.by = "cell_type0719",label = T,label.size = 12)







#和增殖期子宫内膜宫腔粘连文章进行映射分析-----
WEN=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/CD301_WEN_DATA/rds存放点/all_harmony.rds")



#对上皮基质细胞进行差异基因分析-----
NM_EPI=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/EPI_ALL.rds")

NM_EPI1=subset(NM_EPI,donor2%in%c("Ctrl_Pro","IUA_Pro"))
NM_EPI_deseq=FindMarkers(NM_EPI1,ident.1 = "Adhesion",ident.2 = "Ctrl",
                         min.pct = 0.2,group.by = "type4")

NM_EPI_deseq=
  subset(NM_EPI_deseq,NM_EPI_deseq$p_val<0.01)

write.csv(NM_EPI_deseq,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_EPI_Pro_deseq.csv")



NM_EPI2=subset(NM_EPI,donor2%in%c("Ctrl_Sec","IUA_Sec"))
NM_EPI_deseq=FindMarkers(NM_EPI2,ident.1 = "Adhesion",ident.2 = "Ctrl",
                         min.pct = 0.2,group.by = "type4")

NM_EPI_deseq=
  subset(NM_EPI_deseq,NM_EPI_deseq$p_val<0.01)

write.csv(NM_EPI_deseq,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_EPI_Sec_deseq.csv")


NM_STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/STR_ALL.rds")

NM_STR1=subset(NM_STR,donor2%in%c("Ctrl_Pro","IUA_Pro"))
NM_STR_deseq=FindMarkers(NM_STR1,ident.1 = "Adhesion",ident.2 = "Ctrl",
                         min.pct = 0.2,group.by = "type4")

NM_STR_deseq=
  subset(NM_STR_deseq,NM_STR_deseq$p_val<0.01)

write.csv(NM_STR_deseq,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_STR_Pro_deseq.csv")



NM_STR2=subset(NM_STR,donor2%in%c("Ctrl_Sec","IUA_Sec"))
NM_STR_deseq=FindMarkers(NM_STR2,ident.1 = "Adhesion",ident.2 = "Ctrl",
                         min.pct = 0.2,group.by = "type4")

NM_STR_deseq=
  subset(NM_STR_deseq,NM_STR_deseq$p_val<0.01)

write.csv(NM_STR_deseq,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_STR_Sec_deseq.csv")




#对上皮细胞差异基因进行veen图绘制-----

Injury_30D=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_EPI_Pro_deseq.csv")
Treated_30D=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_EPI_Sec_deseq.csv")

Injury_30D_up=subset(Injury_30D,avg_log2FC>0)
Treated_30D_up=subset(Treated_30D,avg_log2FC>0)
Injury_30D_down=subset(Injury_30D,avg_log2FC<0)
Treated_30D_down=subset(Treated_30D,avg_log2FC<0)


Injury_30D_up=Injury_30D_up$X
Treated_30D_up=Treated_30D_up$X

Injury_30D_down=Injury_30D_down$X
Treated_30D_down=Treated_30D_down$X




list('Pro Up'=Injury_30D_up,'Sec Up'=Treated_30D_up) %>% 
  ggvenn(show_percentage = T,show_elements = F,label_sep = ",",
         digits = 1,stroke_color = "white",
         fill_color = c("#1E90FF", "#FF8C00"),
         set_name_color = c("#1E90FF","#FF8C00"))

list('Pro down'=Injury_30D_down,'Sec down'=Treated_30D_down) %>% 
  ggvenn(show_percentage = T,show_elements = F,label_sep = ",",
         digits = 1,stroke_color = "white",
         fill_color = c("#1E90FF", "#FF8C00"),
         set_name_color = c("#1E90FF","#FF8C00"))
#对基质细胞差异基因进行veen图绘制-----
library(ggvenn)
library(ggVennDiagram)

Injury_30D=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_STR_Pro_deseq.csv")
Treated_30D=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_STR_Sec_deseq.csv")

Injury_30D_up=subset(Injury_30D,avg_log2FC>0)
Treated_30D_up=subset(Treated_30D,avg_log2FC>0)
Injury_30D_down=subset(Injury_30D,avg_log2FC<0)
Treated_30D_down=subset(Treated_30D,avg_log2FC<0)


Injury_30D_up=Injury_30D_up$X
Treated_30D_up=Treated_30D_up$X

Injury_30D_down=Injury_30D_down$X
Treated_30D_down=Treated_30D_down$X




list('Pro Up'=Injury_30D_up,'Sec Up'=Treated_30D_up) %>% 
  ggvenn(show_percentage = T,show_elements = F,label_sep = ",",
         digits = 1,stroke_color = "white",
         fill_color = c("#1E90FF", "#FF8C00"),
         set_name_color = c("#1E90FF","#FF8C00"))

list('Pro down'=Injury_30D_down,'Sec down'=Treated_30D_down) %>% 
  ggvenn(show_percentage = T,show_elements = F,label_sep = ",",
         digits = 1,stroke_color = "white",
         fill_color = c("#1E90FF", "#FF8C00"),
         set_name_color = c("#1E90FF","#FF8C00"))


#对上皮富集通路取出通路进行气泡图绘制----
NM_EPI_PRO_up=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_EPI_PRO_up.csv")
NM_EPI_SEC_up=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_EPI_SEC_up.csv")

NM_EPI_PRO_down=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_EPI_PRO_down.csv")
NM_EPI_SEC_down=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_EPI_SEC_down.csv")

NM_EPI_PRO_up=subset(NM_EPI_PRO_up,NM_EPI_PRO_up$P.Value<0.01)
NM_EPI_SEC_up=subset(NM_EPI_SEC_up,NM_EPI_SEC_up$P.Value<0.01)
NM_EPI_PRO_down=subset(NM_EPI_PRO_down,NM_EPI_PRO_down$P.Value<0.01)
NM_EPI_SEC_down=subset(NM_EPI_SEC_down,NM_EPI_SEC_down$P.Value<0.01)

NM_EPI_PRO_up=NM_EPI_PRO_up$X.Term
NM_EPI_SEC_up=NM_EPI_SEC_up$X.Term
NM_EPI_PRO_down=NM_EPI_PRO_down$X.Term
NM_EPI_SEC_down=NM_EPI_SEC_down$X.Term



#找到up中的通路和down中不同的通路
A=intersect(NM_EPI_PRO_up,NM_EPI_SEC_up)
B=c(NM_EPI_PRO_down,NM_EPI_SEC_down)

A1=A[-which(A%in%B)]

A1=as.data.frame(A1)

write.csv(A1,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/A1_up.csv")


#查看pro_IUA中特异存在的信号通路
A=NM_EPI_PRO_up
B=c(NM_EPI_PRO_down,NM_EPI_SEC_down,NM_EPI_SEC_up)

A1=A[-which(A%in%B)]

A1=as.data.frame(A1)

write.csv(A1,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/PRO-EPI_up_unique.csv")






#查看sec_IUA中特异存在的信号通路
A=NM_EPI_SEC_up
B=c(NM_EPI_PRO_down,NM_EPI_SEC_down,NM_EPI_PRO_up)

A1=A[-which(A%in%B)]

A1=as.data.frame(A1)

write.csv(A1,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/SEC-EPI_up_unique.csv")





#找到down中的通路和up中不同的通路
A=intersect(NM_EPI_PRO_down,NM_EPI_SEC_down)
B=c(NM_EPI_PRO_up,NM_EPI_SEC_up)

A1=A[-which(A%in%B)]

A1=as.data.frame(A1)

write.csv(A1,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/A1_down.csv")




#查看pro_IUA中特异存在的信号通路
A=NM_EPI_PRO_down
B=c(NM_EPI_PRO_up,NM_EPI_SEC_down,NM_EPI_SEC_up)

A1=A[-which(A%in%B)]

A1=as.data.frame(A1)

write.csv(A1,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/PRO-EPI_down_unique.csv")






#查看sec_IUA中特异存在的信号通路
A=NM_EPI_SEC_down
B=c(NM_EPI_PRO_down,NM_EPI_SEC_up,NM_EPI_PRO_up)

A1=A[-which(A%in%B)]

A1=as.data.frame(A1)

write.csv(A1,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/SEC-EPI_down_unique.csv")















#选出通路进行绘制
NM_EPI_PRO_up=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_EPI_PRO_up.csv")
NM_EPI_SEC_up=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_EPI_SEC_up.csv")
NM_EPI_PRO_down=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_EPI_PRO_down.csv")
NM_EPI_SEC_down=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_EPI_SEC_down.csv")

NM_EPI_PRO_up=subset(NM_EPI_PRO_up,NM_EPI_PRO_up$P.Value<0.01)
NM_EPI_SEC_up=subset(NM_EPI_SEC_up,NM_EPI_SEC_up$P.Value<0.01)
NM_EPI_PRO_down=subset(NM_EPI_PRO_down,NM_EPI_PRO_down$P.Value<0.01)
NM_EPI_SEC_down=subset(NM_EPI_SEC_down,NM_EPI_SEC_down$P.Value<0.01)


TERM=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/EPI_term.csv")
TERM=TERM$TERM


NM_EPI_PRO_up=subset(NM_EPI_PRO_up,NM_EPI_PRO_up$X.Term%in%TERM)
NM_EPI_SEC_up=subset(NM_EPI_SEC_up,NM_EPI_SEC_up$X.Term%in%TERM)
NM_EPI_PRO_down=subset(NM_EPI_PRO_down,NM_EPI_PRO_down$X.Term%in%TERM)
NM_EPI_SEC_down=subset(NM_EPI_SEC_down,NM_EPI_SEC_down$X.Term%in%TERM)

NM_EPI_PRO_up$type="PRO_up"
NM_EPI_SEC_up$type="SEC_up"
NM_EPI_PRO_down$type="PRO_down"
NM_EPI_SEC_down$type="SEC_down"

plot2=rbind(NM_EPI_PRO_down,NM_EPI_SEC_down,NM_EPI_PRO_up,NM_EPI_SEC_up)


New<-plot2[!duplicated(plot2$X.Term),]



ggplot(plot2)+aes(x=type,y=X.Term)+geom_point(aes(size=Input.number,color=-log10(P.Value)))+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,face = "bold",colour = "black",size =10 ),
        axis.text.y = element_text(face = "bold",colour = "black",size =10 ),
        axis.title.y =element_text(face = "bold",colour = "black",size =10 ),
        axis.title = element_text(face = "bold",colour = "black",size =10 ))+
  xlab("")+ylab("")+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])))+
  ggtitle("GO analysis/KEGG analysis of DEGs")+labs(size="Count")+
  scale_x_discrete(limits=c("PRO_down","SEC_down","PRO_up","SEC_up"))+
  scale_y_discrete(limits=New$X.Term)+scale_size_continuous(range = c(4,8))


ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/EPI_term.pdf",
       width = 7,height = 6.5)




ggplot(plot2)+aes(x=type,y=X.Term)+geom_point(aes(size=Input.number,color=-log10(P.Value)))+
  theme_bw()+
  theme(axis.title.y =element_text(face = "bold",colour = "black",size =10 ),
        axis.title = element_text(face = "bold",colour = "black",size =10 ),axis.text = element_blank())+
  xlab("")+ylab("")+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])))+
  ggtitle("GO analysis/KEGG analysis of DEGs")+labs(size="Count")+
  scale_x_discrete(limits=c("PRO_down","SEC_down","PRO_up","SEC_up"))+
  scale_y_discrete(limits=New$X.Term)+scale_size_continuous(range = c(4,8))



ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/EPI_term2.pdf",
       width = 3,height = 6.5)





#对基质富集通路取出通路进行气泡图绘制----
NM_STR_PRO_up=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_STR_PRO_up.csv")
NM_STR_SEC_up=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_STR_SEC_up.csv")

NM_STR_PRO_down=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_STR_PRO_down.csv")
NM_STR_SEC_down=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_STR_SEC_down.csv")

NM_STR_PRO_up=subset(NM_STR_PRO_up,NM_STR_PRO_up$P.Value<0.01)
NM_STR_SEC_up=subset(NM_STR_SEC_up,NM_STR_SEC_up$P.Value<0.01)
NM_STR_PRO_down=subset(NM_STR_PRO_down,NM_STR_PRO_down$P.Value<0.01)
NM_STR_SEC_down=subset(NM_STR_SEC_down,NM_STR_SEC_down$P.Value<0.01)

NM_STR_PRO_up=NM_STR_PRO_up$X.Term
NM_STR_SEC_up=NM_STR_SEC_up$X.Term
NM_STR_PRO_down=NM_STR_PRO_down$X.Term
NM_STR_SEC_down=NM_STR_SEC_down$X.Term



#找到up中的通路和down中不同的通路
A=intersect(NM_STR_PRO_up,NM_STR_SEC_up)
B=c(NM_STR_PRO_down,NM_STR_SEC_down)

A1=A[-which(A%in%B)]

A1=as.data.frame(A1)

write.csv(A1,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/STR_A1_up.csv")


#查看pro_IUA中特异存在的信号通路
A=NM_STR_PRO_up
B=c(NM_STR_PRO_down,NM_STR_SEC_down,NM_STR_SEC_up)

A1=A[-which(A%in%B)]

A1=as.data.frame(A1)

write.csv(A1,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/PRO-STR_up_unique.csv")






#查看sec_IUA中特异存在的信号通路
A=NM_STR_SEC_up
B=c(NM_STR_PRO_down,NM_STR_SEC_down,NM_STR_PRO_up)

A1=A[-which(A%in%B)]

A1=as.data.frame(A1)

write.csv(A1,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/SEC-STR_up_unique.csv")





#找到down中的通路和up中不同的通路
A=intersect(NM_STR_PRO_down,NM_STR_SEC_down)
B=c(NM_STR_PRO_up,NM_STR_SEC_up)

A1=A[-which(A%in%B)]

A1=as.data.frame(A1)

write.csv(A1,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/STR_A1_down.csv")




#查看pro_IUA中特异存在的信号通路
A=NM_STR_PRO_down
B=c(NM_STR_PRO_up,NM_STR_SEC_down,NM_STR_SEC_up)

A1=A[-which(A%in%B)]

A1=as.data.frame(A1)

write.csv(A1,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/PRO-STR_down_unique.csv")






#查看sec_IUA中特异存在的信号通路
A=NM_STR_SEC_down
B=c(NM_STR_PRO_down,NM_STR_SEC_up,NM_STR_PRO_up)

A1=A[-which(A%in%B)]

A1=as.data.frame(A1)

write.csv(A1,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/SEC-STR_down_unique.csv")





#选出通路进行绘制
NM_STR_PRO_up=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_STR_PRO_up.csv")
NM_STR_SEC_up=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_STR_SEC_up.csv")
NM_STR_PRO_down=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_STR_PRO_down.csv")
NM_STR_SEC_down=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_STR_SEC_down.csv")

NM_STR_PRO_up=subset(NM_STR_PRO_up,NM_STR_PRO_up$P.Value<0.01)
NM_STR_SEC_up=subset(NM_STR_SEC_up,NM_STR_SEC_up$P.Value<0.01)
NM_STR_PRO_down=subset(NM_STR_PRO_down,NM_STR_PRO_down$P.Value<0.01)
NM_STR_SEC_down=subset(NM_STR_SEC_down,NM_STR_SEC_down$P.Value<0.01)


TERM=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/STR_term.csv")
TERM=TERM$TERM


NM_STR_PRO_up=subset(NM_STR_PRO_up,NM_STR_PRO_up$X.Term%in%TERM)
NM_STR_SEC_up=subset(NM_STR_SEC_up,NM_STR_SEC_up$X.Term%in%TERM)
NM_STR_PRO_down=subset(NM_STR_PRO_down,NM_STR_PRO_down$X.Term%in%TERM)
NM_STR_SEC_down=subset(NM_STR_SEC_down,NM_STR_SEC_down$X.Term%in%TERM)

NM_STR_PRO_up$type="PRO_up"
NM_STR_SEC_up$type="SEC_up"
NM_STR_PRO_down$type="PRO_down"
NM_STR_SEC_down$type="SEC_down"

plot2=rbind(NM_STR_PRO_down,NM_STR_SEC_down,NM_STR_PRO_up,NM_STR_SEC_up)


New<-plot2[!duplicated(plot2$X.Term),]



ggplot(plot2)+aes(x=type,y=X.Term)+geom_point(aes(size=Input.number,color=-log10(P.Value)))+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,face = "bold",colour = "black",size =10 ),
        axis.text.y = element_text(face = "bold",colour = "black",size =10 ),
        axis.title.y =element_text(face = "bold",colour = "black",size =10 ),
        axis.title = element_text(face = "bold",colour = "black",size =10 ))+
  xlab("")+ylab("")+
  scale_color_gradient(low="white",high="red")+ggtitle("GO analysis/KEGG analysis of DEGs")+labs(size="Count")+
  scale_x_discrete(limits=c("PRO_down","SEC_down","PRO_up","SEC_up"))+
  scale_y_discrete(limits=New$X.Term)+scale_size_continuous(range = c(4,8))


ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/STR_term1.pdf",
       width = 7.5,height = 7.5)


ggplot(plot2)+aes(x=type,y=X.Term)+geom_point(aes(size=Input.number,color=-log10(P.Value)))+
  theme_bw()+
  theme(axis.text.x = element_text(angle = 45,hjust = 1,vjust = 1,face = "bold",colour = "black",size =10 ),
        axis.text.y = element_text(face = "bold",colour = "black",size =10 ),
        axis.title.y =element_text(face = "bold",colour = "black",size =10 ),
        axis.title = element_text(face = "bold",colour = "black",size =10 ))+
  xlab("")+ylab("")+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])))+
  ggtitle("GO analysis/KEGG analysis of DEGs")+labs(size="Count")+
  scale_x_discrete(limits=c("PRO_down","SEC_down","PRO_up","SEC_up"))+
  scale_y_discrete(limits=New$X.Term)+scale_size_continuous(range = c(4,8))



ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/STR_term2.pdf",
       width = 7,height = 6)


ggplot(plot2)+aes(x=type,y=X.Term)+geom_point(aes(size=Input.number,color=-log10(P.Value)))+
  theme_bw()+
  theme(axis.title.y =element_text(face = "bold",colour = "black",size =10 ),
        axis.title = element_text(face = "bold",colour = "black",size =10 ),axis.text = element_blank())+
  xlab("")+ylab("")+
  scale_color_gradientn(colours = rev(c("#FFD92F","#FEE391",brewer.pal(11, "Spectral")[7:11])))+
  ggtitle("GO analysis/KEGG analysis of DEGs")+labs(size="Count")+
  scale_x_discrete(limits=c("PRO_down","SEC_down","PRO_up","SEC_up"))+
  scale_y_discrete(limits=New$X.Term)+scale_size_continuous(range = c(4,8))



ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/STR_term2.pdf",
       width = 3,height = 6.5)



#对上皮差异基因进行热图绘制----
NM_EPI=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/EPI_ALL.rds")

NM_EPI_deseq1=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_EPI_Pro_deseq.csv")
NM_EPI_deseq2=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_EPI_Sec_deseq.csv")



NM_EPI_deseq1_up=subset(NM_EPI_deseq1,avg_log2FC>0)
NM_EPI_deseq2_up=subset(NM_EPI_deseq2,avg_log2FC>0)
NM_EPI_deseq1_down=subset(NM_EPI_deseq1,avg_log2FC<0)
NM_EPI_deseq2_down=subset(NM_EPI_deseq2,avg_log2FC<0)


up3=intersect(NM_EPI_deseq1_up$X,NM_EPI_deseq2_up$X)
down3=intersect(NM_EPI_deseq1_down$X,NM_EPI_deseq2_down$X)


NM_EPI=subset(NM_EPI,donor2%in%c("Ctrl_Pro","Ctrl_Sec","IUA_Pro","IUA_Sec"))
DefaultAssay(NM_EPI) <- "RNA"
all.genes_NM_EPI <- rownames(NM_EPI)
NM_EPI <- ScaleData(object = NM_EPI, features = all.genes_NM_EPI)


#doheatmap热图
DoHeatmap(NM_EPI,group.by="donor2",features = c(up3,down3),label=F)+
  theme(axis.text.y = element_blank())
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/P52.pdf",width = 6,height = 6)




#ComplexHeatmap热图绘制
library(ComplexHeatmap)
library(Seurat)


NM_EPI=subset(NM_EPI,donor2%in%c("Ctrl_Pro","Ctrl_Sec","IUA_Pro","IUA_Sec"))
DefaultAssay(NM_EPI) <- "RNA"
all.genes_NM_EPI <- rownames(NM_EPI)
NM_EPI <- ScaleData(object = NM_EPI, features = all.genes_NM_EPI)

data<-GetAssayData(NM_EPI,slot="scale.data")

g=c(up3,down3)

celltype_info<-sort(NM_EPI$donor2)
table(celltype_info)

celltype_info<-factor(celltype_info,levels=c("Ctrl_Pro","Ctrl_Sec","IUA_Pro","IUA_Sec"))

mat<-as.matrix(data[g,names(celltype_info)])


col<-c("#C87DFF","#7DAF00","#00C0C5","#F8776D")
names(col)<-levels(celltype_info)

top_anno<-HeatmapAnnotation(
  cluster=anno_block(gp=gpar(fill=col),#设置填充色
                     labels=levels(celltype_info),
                     labels_gp=gpar(cex=0.5,col="white")))#设置字体

Heatmap(mat,name="Expression",
        row_order=1:nrow(mat),
        cluster_rows=FALSE,
        cluster_columns=FALSE,
        show_column_names=F,
        border=T,show_row_names = F,
        row_names_gp=gpar(fontface='italic',fontsize=8),
        column_split=celltype_info,
        top_annotation=top_anno,#在热图上边增加注释
        column_title=NULL#不需要列标题
)

#对上皮差异基因[选取top的]进行热图绘制----
NM_EPI=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/EPI_ALL.rds")

NM_EPI_deseq1=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_EPI_Pro_deseq.csv")
NM_EPI_deseq2=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_EPI_Sec_deseq.csv")



NM_EPI_deseq1_up=subset(NM_EPI_deseq1,avg_log2FC>0)
NM_EPI_deseq2_up=subset(NM_EPI_deseq2,avg_log2FC>0)
NM_EPI_deseq1_down=subset(NM_EPI_deseq1,avg_log2FC<0)
NM_EPI_deseq2_down=subset(NM_EPI_deseq2,avg_log2FC<0)


NM_EPI_deseq1_up=NM_EPI_deseq1_up[1:100,]
NM_EPI_deseq2_up=NM_EPI_deseq2_up[1:100,]

NM_EPI_deseq1_down=NM_EPI_deseq1_down[1:100,]
NM_EPI_deseq2_down=NM_EPI_deseq2_down[1:100,]








up3=intersect(NM_EPI_deseq1_up$X,NM_EPI_deseq2_up$X)
down3=intersect(NM_EPI_deseq1_down$X,NM_EPI_deseq2_down$X)


NM_EPI=subset(NM_EPI,donor2%in%c("Ctrl_Pro","Ctrl_Sec","IUA_Pro","IUA_Sec"))
DefaultAssay(NM_EPI) <- "RNA"
all.genes_NM_EPI <- rownames(NM_EPI)
NM_EPI <- ScaleData(object = NM_EPI, features = all.genes_NM_EPI)


#doheatmap热图
DoHeatmap(NM_EPI,group.by="donor2",features = c(up3,down3),label=F)+
  theme(axis.text.y = element_blank())
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/P52.pdf",width = 6,height = 6)




#ComplexHeatmap热图绘制
library(ComplexHeatmap)
library(Seurat)
library(circlize)

NM_EPI=subset(NM_EPI,donor2%in%c("Ctrl_Pro","Ctrl_Sec","IUA_Pro","IUA_Sec"))
DefaultAssay(NM_EPI) <- "RNA"
all.genes_NM_EPI <- rownames(NM_EPI)
NM_EPI <- ScaleData(object = NM_EPI, features = all.genes_NM_EPI)

data<-GetAssayData(NM_EPI,slot="scale.data",assay = "RNA")

g=c(up3,down3)

celltype_info<-sort(NM_EPI$donor2)
table(celltype_info)

celltype_info<-factor(celltype_info,levels=c("Ctrl_Pro","Ctrl_Sec","IUA_Pro","IUA_Sec"))

mat<-as.matrix(data[g,names(celltype_info)])


col<-c("#C87DFF","#7DAF00","#00C0C5","#F8776D")
names(col)<-levels(celltype_info)

top_anno<-HeatmapAnnotation(
  cluster=anno_block(gp=gpar(fill=col),#设置填充色
                     labels=levels(celltype_info),
                     labels_gp=gpar(cex=0.5,col="white")))#设置字体

pdf("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/EPI_heatmap.pdf",width = 8,height = 6)
Heatmap(mat,name="Expression",
        row_order=1:nrow(mat),
        cluster_rows=FALSE,
        cluster_columns=FALSE,
        show_column_names=F,
        border=T,
        row_names_gp=gpar(fontface='italic',fontsize=8),
        column_split=celltype_info,
        top_annotation=top_anno,#在热图上边增加注释
        column_title=NULL#不需要列标题
)
dev.off()


#对基质差异基因进行热图绘制----
NM_STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/STR_ALL.rds")

NM_STR_deseq1=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_STR_Pro_deseq.csv")
NM_STR_deseq2=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_STR_Sec_deseq.csv")



NM_STR_deseq1_up=subset(NM_STR_deseq1,avg_log2FC>0)
NM_STR_deseq2_up=subset(NM_STR_deseq2,avg_log2FC>0)
NM_STR_deseq1_down=subset(NM_STR_deseq1,avg_log2FC<0)
NM_STR_deseq2_down=subset(NM_STR_deseq2,avg_log2FC<0)


NM_STR_deseq1_up=NM_STR_deseq1_up
NM_STR_deseq2_up=NM_STR_deseq2_up

NM_STR_deseq1_down=NM_STR_deseq1_down
NM_STR_deseq2_down=NM_STR_deseq2_down








up3=intersect(NM_STR_deseq1_up$X,NM_STR_deseq2_up$X)
down3=intersect(NM_STR_deseq1_down$X,NM_STR_deseq2_down$X)


NM_STR=subset(NM_STR,donor2%in%c("Ctrl_Pro","Ctrl_Sec","IUA_Pro","IUA_Sec"))
DefaultAssay(NM_STR) <- "RNA"
all.genes_NM_STR <- rownames(NM_STR)
NM_STR <- ScaleData(object = NM_STR, features = all.genes_NM_STR)





#ComplexHeatmap热图绘制
library(ComplexHeatmap)
library(Seurat)
library(circlize)

NM_STR=subset(NM_STR,donor2%in%c("Ctrl_Pro","Ctrl_Sec","IUA_Pro","IUA_Sec"))
DefaultAssay(NM_STR) <- "RNA"
all.genes_NM_STR <- rownames(NM_STR)
NM_STR <- ScaleData(object = NM_STR, features = all.genes_NM_STR)

data<-GetAssayData(NM_STR,slot="scale.data",assay = "RNA")

g=c(up3,down3)

celltype_info<-sort(NM_STR$donor2)
table(celltype_info)

celltype_info<-factor(celltype_info,levels=c("Ctrl_Pro","Ctrl_Sec","IUA_Pro","IUA_Sec"))

mat<-as.matrix(data[g,names(celltype_info)])


col<-c("#C87DFF","#7DAF00","#00C0C5","#F8776D")
names(col)<-levels(celltype_info)

top_anno<-HeatmapAnnotation(
  cluster=anno_block(gp=gpar(fill=col),#设置填充色
                     labels=levels(celltype_info),
                     labels_gp=gpar(cex=0.5,col="white")))#设置字体

pdf("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/STR_heatmap2.pdf",width = 16,height = 14)
Heatmap(mat,name="Expression",
        row_order=1:nrow(mat),
        cluster_rows=FALSE,
        cluster_columns=FALSE,
        show_column_names=F,
        border=T,
        row_names_gp=gpar(fontface='italic',fontsize=8),
        column_split=celltype_info,
        top_annotation=top_anno,#在热图上边增加注释
        column_title=NULL#不需要列标题
)
dev.off()














#对基质差异基因[选取top的]进行热图绘制----
NM_STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/STR_ALL.rds")

NM_STR_deseq1=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_STR_Pro_deseq.csv")
NM_STR_deseq2=read.csv("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_STR_Sec_deseq.csv")



NM_STR_deseq1_up=subset(NM_STR_deseq1,avg_log2FC>0)
NM_STR_deseq2_up=subset(NM_STR_deseq2,avg_log2FC>0)
NM_STR_deseq1_down=subset(NM_STR_deseq1,avg_log2FC<0)
NM_STR_deseq2_down=subset(NM_STR_deseq2,avg_log2FC<0)


NM_STR_deseq1_up=NM_STR_deseq1_up[1:100,]
NM_STR_deseq2_up=NM_STR_deseq2_up[1:100,]

NM_STR_deseq1_down=NM_STR_deseq1_down[1:100,]
NM_STR_deseq2_down=NM_STR_deseq2_down[1:100,]








up3=intersect(NM_STR_deseq1_up$X,NM_STR_deseq2_up$X)
down3=intersect(NM_STR_deseq1_down$X,NM_STR_deseq2_down$X)


NM_STR=subset(NM_STR,donor2%in%c("Ctrl_Pro","Ctrl_Sec","IUA_Pro","IUA_Sec"))
DefaultAssay(NM_STR) <- "RNA"
all.genes_NM_STR <- rownames(NM_STR)
NM_STR <- ScaleData(object = NM_STR, features = all.genes_NM_STR)





#ComplexHeatmap热图绘制
library(ComplexHeatmap)
library(Seurat)
library(circlize)

NM_STR=subset(NM_STR,donor2%in%c("Ctrl_Pro","Ctrl_Sec","IUA_Pro","IUA_Sec"))
DefaultAssay(NM_STR) <- "RNA"
all.genes_NM_STR <- rownames(NM_STR)
NM_STR <- ScaleData(object = NM_STR, features = all.genes_NM_STR)

data<-GetAssayData(NM_STR,slot="scale.data",assay = "RNA")

g=c(up3,down3)

celltype_info<-sort(NM_STR$donor2)
table(celltype_info)

celltype_info<-factor(celltype_info,levels=c("Ctrl_Pro","Ctrl_Sec","IUA_Pro","IUA_Sec"))

mat<-as.matrix(data[g,names(celltype_info)])


col<-c("#C87DFF","#7DAF00","#00C0C5","#F8776D")
names(col)<-levels(celltype_info)

top_anno<-HeatmapAnnotation(
  cluster=anno_block(gp=gpar(fill=col),#设置填充色
                     labels=levels(celltype_info),
                     labels_gp=gpar(cex=0.5,col="white")))#设置字体

pdf("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/STR_heatmap.pdf",width = 8,height = 6)
Heatmap(mat,name="Expression",
        row_order=1:nrow(mat),
        cluster_rows=FALSE,
        cluster_columns=FALSE,
        show_column_names=F,
        border=T,
        row_names_gp=gpar(fontface='italic',fontsize=8),
        column_split=celltype_info,
        top_annotation=top_anno,#在热图上边增加注释
        column_title=NULL#不需要列标题
)
dev.off()














#对基质细胞进行差异基因分析-----
NM_STR=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/STR_ALL.rds")

NM_STR1=subset(NM_STR,donor2%in%c("Ctrl_Pro","IUA_Pro"))
NM_STR_deseq=FindMarkers(NM_STR1,ident.1 = "Adhesion",ident.2 = "Ctrl",
                         min.pct = 0.2,group.by = "type4")

NM_STR_deseq=
  subset(NM_STR_deseq,NM_STR_deseq$p_val<0.01)

write.csv(NM_STR_deseq,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_STR_Pro_deseq.csv")



NM_STR2=subset(NM_STR,donor2%in%c("Ctrl_Sec","IUA_Sec"))
NM_STR_deseq=FindMarkers(NM_STR2,ident.1 = "Adhesion",ident.2 = "Ctrl",
                         min.pct = 0.2,group.by = "type4")

NM_STR_deseq=
  subset(NM_STR_deseq,NM_STR_deseq$p_val<0.01)

write.csv(NM_STR_deseq,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/差异基因与富集/NM_STR_Sec_deseq.csv")
#对scrna数据进行构建-----
#4861STDY7387181------
S1_count=Read10X("/data/DATAprocess/chentw/NG/scRNA/data/4861STDY7771115/",
                 cell.column = 2)
S1 <- CreateSeuratObject(counts = S1_count, project = "S1", min.cells = 3)
S1[["percent.mt"]] <- PercentageFeatureSet(S1, pattern = "^MT-")
VlnPlot(S1, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)
S1 <- subset(S1, subset = nFeature_RNA < 5500 & percent.mt < 20 & 
               nFeature_RNA > 200)
S1 <- NormalizeData(S1, normalization.method = "LogNormalize", 
                    scale.factor = 10000)
S1 <- FindVariableFeatures(S1, selection.method = "vst", nfeatures = 2000)
VlnPlot(S1, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3)

#S1进行标准化然后PCA分析
DefaultAssay(S1) <- "RNA"
all.genes_S1 <- rownames(S1)
S1 <- ScaleData(object = S1, features = all.genes_S1)
S1 <- RunPCA(S1, features = VariableFeatures(object = S1))

#确定S1的PC维数
S1 <- JackStraw(S1, num.replicate = 100) 
S1 <- ScoreJackStraw(S1, dims = 1:20) 
ElbowPlot(S1,ndims = 30, reduction = "pca") 

#S1进行细胞细胞聚类分析(UMAP)
S1=RunUMAP(S1, dims = 1:30)
DimPlot(S1,reduction="umap",label=TRUE)
S1 <- FindNeighbors(S1, dims = 1:30)
S1 <- FindClusters(S1, resolution = 0.3)
S1 <- FindClusters(S1, resolution = 0.6)
DimPlot(S1,reduction="umap",label=TRUE)


#S1进行细胞细胞聚类分析(tsne)
S1=RunTSNE(S1, dims = 1:30)
DimPlot(S1,reduction="tsne",label=TRUE)
S1 <- FindNeighbors(S1, dims = 1:30)
S1 <- FindClusters(S1, resolution = 0.3)
DimPlot(S1,reduction="tsne",label=TRUE)

#保存对象
saveRDS(S1,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/空转数据/rds/4861STDY7387181.rds")










#读取参考数据----
library(Seurat)
library(ggplot2)
library(patchwork)
library(dplyr)

#创建对象------
#152810----
setwd("/data/DATAprocess/chentw/NG/S1/")
S1 <- CreateSeuratObject(counts = Read10X("/data/DATAprocess/chentw/NG/S1/filtered_feature_bc_matrix/"), 
                         assay = "Spatial")


S1_img <- Read10X_Image(image.dir = file.path("./spatial"), filter.matrix = TRUE)

S1_img <- S1_img[Cells(x = S1)]
DefaultAssay(S1 = S1_img) <- "Spatial"
S1[["slice1"]] <- S1_img


S1 <- NormalizeData(S1, normalization.method = "LogNormalize", 
                    scale.factor = 10000)
S1 <- FindVariableFeatures(S1, selection.method = "vst", nfeatures = 2000)


#S1进行标准化然后PCA分析
DefaultAssay(S1) <- "RNA"
all.genes_S1 <- rownames(S1)
S1 <- ScaleData(object = S1, features = all.genes_S1)
S1 <- RunPCA(S1, features = VariableFeatures(object = S1))


#确定S1的PC维数
S1 <- JackStraw(S1, num.replicate = 100) 
S1 <- ScoreJackStraw(S1, dims = 1:20) 
ElbowPlot(S1,ndims = 30, reduction = "pca") 

#S1进行细胞细胞聚类分析(UMAP)
S1=RunUMAP(S1, dims = 1:30)
DimPlot(S1,reduction="umap",label=TRUE)
S1 <- FindNeighbors(S1, dims = 1:30)
S1 <- FindClusters(S1, resolution = 0.3)
S1 <- FindClusters(S1, resolution = 0.6)
DimPlot(S1,reduction="umap",label=TRUE)

saveRDS(S1,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/空转数据/rds/152810.rds")


#152811----
setwd("/data/DATAprocess/chentw/NG/152811/")
S1 <- CreateSeuratObject(counts = Read10X("/data/DATAprocess/chentw/NG/152811/filtered_feature_bc_matrix/"), 
                         assay = "Spatial")


S1_img <- Read10X_Image(image.dir = file.path("./spatial"), filter.matrix = TRUE)

S1_img <- S1_img[Cells(x = S1)]
DefaultAssay(S1 = S1_img) <- "Spatial"
S1[["slice1"]] <- S1_img


S1 <- NormalizeData(S1, normalization.method = "LogNormalize", 
                    scale.factor = 10000)
S1 <- FindVariableFeatures(S1, selection.method = "vst", nfeatures = 2000)


#S1进行标准化然后PCA分析
DefaultAssay(S1) <- "RNA"
all.genes_S1 <- rownames(S1)
S1 <- ScaleData(object = S1, features = all.genes_S1)
S1 <- RunPCA(S1, features = VariableFeatures(object = S1))


#确定S1的PC维数
S1 <- JackStraw(S1, num.replicate = 100) 
S1 <- ScoreJackStraw(S1, dims = 1:20) 
ElbowPlot(S1,ndims = 30, reduction = "pca") 

#S1进行细胞细胞聚类分析(UMAP)
S1=RunUMAP(S1, dims = 1:30)
DimPlot(S1,reduction="umap",label=TRUE)
S1 <- FindNeighbors(S1, dims = 1:30)
S1 <- FindClusters(S1, resolution = 0.3)
S1 <- FindClusters(S1, resolution = 0.6)
DimPlot(S1,reduction="umap",label=TRUE)
saveRDS(S1,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/空转数据/rds/152811.rds")



#152807----
setwd("/data/DATAprocess/chentw/NG/152807/")
S1 <- CreateSeuratObject(counts = Read10X("/data/DATAprocess/chentw/NG/152807/filtered_feature_bc_matrix/"), 
                         assay = "Spatial")


S1_img <- Read10X_Image(image.dir = file.path("./spatial"), filter.matrix = TRUE)

S1_img <- S1_img[Cells(x = S1)]
DefaultAssay(S1 = S1_img) <- "Spatial"
S1[["slice1"]] <- S1_img


S1 <- NormalizeData(S1, normalization.method = "LogNormalize", 
                    scale.factor = 10000)
S1 <- FindVariableFeatures(S1, selection.method = "vst", nfeatures = 2000)


#S1进行标准化然后PCA分析
DefaultAssay(S1) <- "RNA"
all.genes_S1 <- rownames(S1)
S1 <- ScaleData(object = S1, features = all.genes_S1)
S1 <- RunPCA(S1, features = VariableFeatures(object = S1))


#确定S1的PC维数
S1 <- JackStraw(S1, num.replicate = 100) 
S1 <- ScoreJackStraw(S1, dims = 1:20) 
ElbowPlot(S1,ndims = 30, reduction = "pca") 

#S1进行细胞细胞聚类分析(UMAP)
S1=RunUMAP(S1, dims = 1:30)
DimPlot(S1,reduction="umap",label=TRUE)
S1 <- FindNeighbors(S1, dims = 1:30)
S1 <- FindClusters(S1, resolution = 0.3)
S1 <- FindClusters(S1, resolution = 0.6)
DimPlot(S1,reduction="umap",label=TRUE)

saveRDS(S1,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/空转数据/rds/152807.rds")

#152806----
setwd("/data/DATAprocess/chentw/NG/152806/")
S1 <- CreateSeuratObject(counts = Read10X("/data/DATAprocess/chentw/NG/152806/filtered_feature_bc_matrix/"), 
                         assay = "Spatial")


S1_img <- Read10X_Image(image.dir = file.path("./spatial"), filter.matrix = TRUE)

S1_img <- S1_img[Cells(x = S1)]
DefaultAssay(S1 = S1_img) <- "Spatial"
S1[["slice1"]] <- S1_img


S1 <- NormalizeData(S1, normalization.method = "LogNormalize", 
                    scale.factor = 10000)
S1 <- FindVariableFeatures(S1, selection.method = "vst", nfeatures = 2000)


#S1进行标准化然后PCA分析
DefaultAssay(S1) <- "RNA"
all.genes_S1 <- rownames(S1)
S1 <- ScaleData(object = S1, features = all.genes_S1)
S1 <- RunPCA(S1, features = VariableFeatures(object = S1))


#确定S1的PC维数
S1 <- JackStraw(S1, num.replicate = 100) 
S1 <- ScoreJackStraw(S1, dims = 1:20) 
ElbowPlot(S1,ndims = 30, reduction = "pca") 

#S1进行细胞细胞聚类分析(UMAP)
S1=RunUMAP(S1, dims = 1:30)
DimPlot(S1,reduction="umap",label=TRUE)
S1 <- FindNeighbors(S1, dims = 1:30)
S1 <- FindClusters(S1, resolution = 0.3)
S1 <- FindClusters(S1, resolution = 0.6)
DimPlot(S1,reduction="umap",label=TRUE)

saveRDS(S1,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/空转数据/rds/152806.rds")





#查看空间基因的表达--------
#152810----
P1=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/空转数据/rds/152806.rds")

S1=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/空转数据/rds/152807.rds")
P2=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/空转数据/rds/152810.rds")

S2=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/空转数据/rds/152811.rds")



Macr_gene=c("CD68","CD163","CLEC10A",'CD24',
            "MKI67","S100A8")

SpatialFeaturePlot(P1, features = Macr_gene)

SpatialFeaturePlot(P2, features = Macr_gene)

SpatialFeaturePlot(S1, features = Macr_gene)

SpatialFeaturePlot(S2, features = Macr_gene)


STR_gene=c("PDGFRB","LUM","DCN","VIM","TGFBI","PGRMC1","SERPINF1",
           "OGN","BMP7","MKI67","REV3L","APOD","ISG15","RGS5","NOTCH3")


SpatialFeaturePlot(P1, features = STR_gene[1:8],ncol = 4)
SpatialFeaturePlot(P1, features = STR_gene[9:15],ncol = 4)

SpatialFeaturePlot(P2, features = STR_gene[1:8],ncol = 4)
SpatialFeaturePlot(P2, features = STR_gene[9:15],ncol = 4)

SpatialFeaturePlot(S1, features = STR_gene[1:8],ncol = 4)
SpatialFeaturePlot(S1, features = STR_gene[9:15],ncol = 4)


SpatialFeaturePlot(S2, features = STR_gene[1:8],ncol = 4)
SpatialFeaturePlot(S2, features = STR_gene[9:15],ncol = 4)

#对空转数据进行细胞分群并进行基因表达比较-----
P1=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/空转数据/rds/152806.rds")

SpatialDimPlot(P1, group.by = "seurat_clusters",label = T)

VlnPlot(P1, group.by = "seurat_clusters",features = STR_gene)

gene=c("PDGFRB","LUM","DCN","EPCAM","LGR5","WNT7A","FOXJ1","RFX2","PTPRC","CLDN4","CD34","SOX18")
VlnPlot(P1, group.by = "seurat_clusters",features = gene,pt.size=0)


S1=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/空转数据/rds/152807.rds")
SpatialDimPlot(S1, group.by = "seurat_clusters",label = T)
gene=c("PDGFRB","LUM","DCN","EPCAM","LGR5","WNT7A","FOXJ1","RFX2","PTPRC","CLDN4","CD34","SOX18")
VlnPlot(S1, group.by = "seurat_clusters",features = gene,pt.size=0)


P2=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/空转数据/rds/152810.rds")
SpatialDimPlot(P2, group.by = "seurat_clusters",label = T)
gene=c("PDGFRB","LUM","DCN","EPCAM","LGR5","WNT7A","FOXJ1","RFX2","PTPRC","CLDN4","CD34","SOX18")
VlnPlot(P2, group.by = "seurat_clusters",features = gene,pt.size=0)



S2=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/空转数据/rds/152811.rds")
SpatialDimPlot(S2, group.by = "seurat_clusters",label = T)
gene=c("PDGFRB","LUM","DCN","EPCAM","LGR5","WNT7A","FOXJ1","RFX2","PTPRC","CLDN4","CD34","SOX18")
VlnPlot(S2, group.by = "seurat_clusters",features = gene,pt.size=0)


#将空转数据进行整合
P1=subset(P1,seurat_clusters%in%c("3","6","4"))
P2=subset(P2,seurat_clusters%in%c("2","3","4","5"))
S1=subset(S1,seurat_clusters%in%c("2","4","5","6"))
S2=subset(S1,seurat_clusters%in%c("2","3","4"))

P1$type3="P1"
P2$type3="P2"
S1$type3="S1"
S2$type3="S2"

all=merge(P1,y = c(P2,S1,S2))


N=all@assays$Spatial@counts


all_single=CreateSeuratObject(counts = N, meta.data = all@meta.data)

all_single=NormalizeData(all_single, normalization.method = "LogNormalize", 
                         scale.factor = 10000)
all_single <- FindVariableFeatures(all_single, selection.method = "vst", nfeatures = 2000)


#all_single进行标准化然后PCA分析
DefaultAssay(all_single) <- "RNA"
all.genes_all_single <- rownames(all_single)
all_single <- ScaleData(object = all_single, features = all.genes_all_single)
all_single <- RunPCA(all_single, features = VariableFeatures(object = all_single))


#all_single进行细胞细胞聚类分析(UMAP)
all_single=RunUMAP(all_single, dims = 1:30)
DimPlot(all_single,reduction="umap",label=TRUE)
all_single <- FindNeighbors(all_single, dims = 1:30)
all_single <- FindClusters(all_single, resolution = 0.3)
all_single <- FindClusters(all_single, resolution = 0.6)
DimPlot(all_single,reduction="umap",label=TRUE)
DimPlot(all_single,reduction="umap",label=TRUE)

#all_single进行细胞细胞聚类分析(tsne)
all_single=RunTSNE(all_single, dims = 1:30)
DimPlot(all_single,reduction="tsne",label=TRUE)
all_single <- FindNeighbors(all_single, dims = 1:30)
all_single <- FindClusters(all_single, resolution = 0.3)
DimPlot(all_single,reduction="tsne",label=TRUE)

DimPlot(all_single,reduction="umap",label=TRUE,group.by = "type3")



#开始尝试去除细胞样本批次并且进行新的一个保存

all_single.list <- SplitObject(all_single, split.by="type3")
all_single.list <- lapply(X = all_single.list, FUN = function(x) {
  x <- NormalizeData(x)
  x <- FindVariableFeatures(x, selection.method = "vst", nfeatures = 2000)
})

k.fliter=min(200,min(sapply(all_single.list,ncol)))
all_single <- FindIntegrationAnchors(object.list = all_single.list, dims = 1:30,k.filter=k.fliter)
all_single <- IntegrateData(anchorset = all_single, dims = 1:30,k.weight=k.fliter)

#进行标准化并进行PCA
DefaultAssay(all_single) <- "RNA"
all.genes.all_single <- rownames(all_single)
all_single <- ScaleData(object = all_single, features = all.genes.all_single)

DefaultAssay(all_single) <- "integrated"
all.genes.all_single1 <- rownames(all_single)
all_single <- ScaleData(object = all_single, features = all.genes.all_single1)
all_single <- RunPCA(all_single, features = VariableFeatures(object = all_single))
DimPlot(all_single,label = T)

DefaultAssay(all_single) <- "integrated"
all_single=RunUMAP(all_single, dims = 1:30)
DimPlot(all_single,reduction="umap",label=TRUE)
all_single <- FindNeighbors(all_single, dims = 1:30)
all_single <- FindClusters(all_single, resolution = 0.3)
all_single <- FindClusters(all_single, resolution = 0.15)

all_single <- FindClusters(all_single, resolution = 0.6)
all_single <- FindClusters(all_single, resolution = 0.45)
DefaultAssay(all_single)<-"RNA"

DimPlot(all_single,label = T,group.by = "type3")

DimPlot(all_single,label = T)

saveRDS(all_single,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/空转数据/rds/all_single.rds")

gene=c("PDGFRB","LUM","DCN","EPCAM","LGR5","WNT7A","FOXJ1","RFX2","PTPRC","CLDN4","CD34","SOX18")
VlnPlot(all_single, features = gene,pt.size=0)


all_single$cell_type2="Immune cells"
all_single$cell_type2[all_single@active.ident%in%c("0","1","5","6","8")]="Stromal cells"
all_single$cell_type2[all_single@active.ident%in%c("3","7")]="Epithelium cells"
all_single$cell_type2[all_single@active.ident%in%c("4")]="Endothelium cells"


#对上述细胞类型进行通讯

#all
cellchat <- createCellChat(all_single@assays$RNA@data,meta = all_single@meta.data,
                           group.by = "cell_type2")
cellchat=setIdent(cellchat,ident.use = "cell_type2")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Interaction weights/strength")

saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/空转数据/cellchat/all.rds")





#对我们的数据三组各取4000细胞和空转数据结合并进行后续绘图
all_single=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/空转数据/rds/all_single.rds")

all_single$cell_type1110=all_single$cell_type2

umap_ctrl_IUA_allsingle=readRDS("/data/DATAprocess/chentw/zigongneimo/rds文件存放点/umap_ctrl_IUA_allsingle.rds")

umap_ctrl_IUA_allsingle@active.ident=as.factor(umap_ctrl_IUA_allsingle$type2)
umap2 <- subset(umap_ctrl_IUA_allsingle,downsample = 4615)
DefaultAssay(umap2)<-"RNA"
table(umap2$type2)


#开始整合
all_umap=merge(all_single,umap2)



all_umap <- CreateSeuratObject(counts =all_umap@assays$RNA@counts, project = "all_umap", min.cells = 3,meta.data = all_umap@meta.data)

all_umap[["percent.mt"]] <- PercentageFeatureSet(all_umap, pattern = "^MT-")
VlnPlot(all_umap, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

all_umap <- NormalizeData(all_umap, normalization.method = "LogNormalize", 
                          scale.factor = 10000)
all_umap <- FindVariableFeatures(all_umap, selection.method = "vst", nfeatures = 2000)
VlnPlot(all_umap, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3,pt.size = 0)

#all_umap进行标准化然后PCA分析
DefaultAssay(all_umap) <- "RNA"
all.genes_all_umap <- rownames(all_umap)
all_umap <- ScaleData(object = all_umap, features = all.genes_all_umap)
all_umap <- RunPCA(all_umap, features = VariableFeatures(object = all_umap))


#all_umap进行细胞细胞聚类分析(UMAP)
all_umap=RunUMAP(all_umap, dims = 1:30)
DimPlot(all_umap,reduction="umap",label=TRUE)
all_umap <- FindNeighbors(all_umap, dims = 1:30)
all_umap <- FindClusters(all_umap, resolution = 0.3)
all_umap <- FindClusters(all_umap, resolution = 0.6)
DimPlot(all_umap,reduction="umap",label=TRUE,group.by = "type3")


#all_umap进行细胞细胞聚类分析(tsne)
all_umap=RunTSNE(all_umap, dims = 1:30)
DimPlot(all_umap,reduction="tsne",label=TRUE)
all_umap <- FindNeighbors(all_umap, dims = 1:30)
all_umap <- FindClusters(all_umap, resolution = 0.3)
all_umap <- FindClusters(all_umap, resolution = 0.6)
DimPlot(all_umap,reduction="tsne",label=TRUE)



all_harmony=all_umap%>%RunHarmony("orig.ident",plot_covergence=T)

DefaultAssay(all_harmony) <- "RNA"


all_harmony <- RunPCA(all_harmony, features = VariableFeatures(object = all_harmony))


all_harmony=RunUMAP(all_harmony,reduction="harmony",dims=1:30)
all_harmony=FindNeighbors(all_harmony,dims=1:30,reduction = "harmony")
all_harmony=FindClusters(all_harmony, resolution = 0.15,reduction = "harmony")
all_harmony=FindClusters(all_harmony, resolution = 0.17,reduction = "harmony")
all_harmony=FindClusters(all_harmony, resolution = 0.18,reduction = "harmony")
all_harmony=FindClusters(all_harmony, resolution = 0.2,reduction = "harmony")

DimPlot(all_harmony,label=TRUE,group.by = "type3")







#integrated整合

all_harmony.list <- SplitObject(all_harmony, split.by="type3")
all_harmony.list <- lapply(X = all_harmony.list, FUN = function(x) {
  x <- NormalizeData(x)
  x <- FindVariableFeatures(x, selection.method = "vst", nfeatures = 2000)
})

k.fliter=min(200,min(sapply(all_harmony.list,ncol)))
all_harmony <- FindIntegrationAnchors(object.list = all_harmony.list, dims = 1:30,k.filter=k.fliter)
all_harmony <- IntegrateData(anchorset = all_harmony, dims = 1:30,k.weight=k.fliter)

#进行标准化并进行PCA
DefaultAssay(all_harmony) <- "RNA"
all.genes.all_harmony <- rownames(all_harmony)
all_harmony <- ScaleData(object = all_harmony, features = all.genes.all_harmony)

DefaultAssay(all_harmony) <- "integrated"
all.genes.all_harmony1 <- rownames(all_harmony)
all_harmony <- ScaleData(object = all_harmony, features = all.genes.all_harmony1)
all_harmony <- RunPCA(all_harmony, features = VariableFeatures(object = all_harmony))
DimPlot(all_harmony,label = T)

DefaultAssay(all_harmony) <- "integrated"
all_harmony=RunUMAP(all_harmony, dims = 1:30)
DimPlot(all_harmony,reduction="umap",label=TRUE)
all_harmony <- FindNeighbors(all_harmony, dims = 1:30)
all_harmony <- FindClusters(all_harmony, resolution = 0.3)
all_harmony <- FindClusters(all_harmony, resolution = 0.15)

all_harmony <- FindClusters(all_harmony, resolution = 0.6)
all_harmony <- FindClusters(all_harmony, resolution = 0.45)
DefaultAssay(all_harmony)<-"RNA"

DimPlot(all_harmony,label = T,group.by = "type3")


DimPlot(all_harmony,label = T,group.by = "cell_type1110")

all_harmony$type2[all_harmony$type3%in%c("P1","P2","S1","S2")]<-"NG"

DimPlot(all_harmony,label = T,group.by = "cell_type1110",split.by = "type2")

#保存对象
saveRDS(all_harmony,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/空转数据/rds/all_integrated1.rds")






#对空转数据进行细胞类型更改
all_harmony=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/空转数据/rds/all_integrated1.rds")

DimPlot(all_harmony,label = T,split.by = "type2")

all_harmony$cell_type1110[all_harmony$type2%in%"NG"]<-"Stromal cells"
all_harmony$cell_type1110[all_harmony$type2%in%"NG"&all_harmony@active.ident%in%"10"]="Endothelium cells"
all_harmony$cell_type1110[all_harmony$type2%in%"NG"&all_harmony@active.ident%in%c("3","12")]="Epithelium cells"

all_harmony$cell_type1110[all_harmony$type2%in%"NG"&all_harmony@active.ident%in%c("13","6")]="Immune cells"

DimPlot(all_harmony,label = T,group.by = "cell_type1110",split.by = "type2")


#提取空转组进行cellchat----
NG=subset(all_harmony,type2%in%"NG")

cellchat <- createCellChat(NG@assays$RNA@data,meta = NG@meta.data,group.by = "cell_type1110")
cellchat=setIdent(cellchat,ident.use = "cell_type1110")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Interaction weights/strength")

saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/CELLCHAT2/NG.rds")


CTRL=subset(all_harmony,type2%in%"CTRL")

cellchat <- createCellChat(CTRL@assays$RNA@data,meta = CTRL@meta.data,group.by = "cell_type1110")
cellchat=setIdent(cellchat,ident.use = "cell_type1110")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Interaction weights/strength")

saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/CELLCHAT2/CTRL.rds")


Adhesion=subset(all_harmony,type2%in%"IUA2")
cellchat <- createCellChat(Adhesion@assays$RNA@data,meta = Adhesion@meta.data,group.by = "cell_type1110")
cellchat=setIdent(cellchat,ident.use = "cell_type1110")
groupSize=as.numeric(table(cellchat@idents))
CellChatDB=CellChatDB.human
str(CellChatDB)
showDatabaseCategory(CellChatDB)
CellChatDB.use=subsetDB(CellChatDB,search = "Secreted Signaling")
cellchat@DB=CellChatDB.use
cellchat=subsetData(cellchat)
future::plan("multiprocess",workers=4)
cellchat=identifyOverExpressedGenes(cellchat)
cellchat=identifyOverExpressedInteractions(cellchat)
cellchat=projectData(cellchat,PPI.human)
cellchat=computeCommunProb(cellchat,raw.use = F,population.size = T)
cellchat=filterCommunication(cellchat,min.cells = 10)
df.net=subsetCommunication(cellchat)
cellchat=computeCommunProbPathway(cellchat)
df.netp=subsetCommunication(cellchat,slot.name='netP')
cellchat=aggregateNet(cellchat)
groupSize=as.numeric(table(cellchat@idents))
mat=cellchat@net$count
par(mfrow=c(1,2),xpd=T)
netVisual_circle(cellchat@net$count,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Number of interactions")
netVisual_circle(cellchat@net$weight,vertex.weight = groupSize,weight.scale = T,
                 label.edge = T,title.name = "Interaction weights/strength")

saveRDS(cellchat,"/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/CELLCHAT2/Adhesion.rds")

NG=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/CELLCHAT2/NG.rds")
Ctrl=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/CELLCHAT2/CTRL.rds")
Adhesion=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/修改图补充图/CELLCHAT2/Adhesion.rds")

NG=netAnalysis_computeCentrality(NG)
Ctrl=netAnalysis_computeCentrality(Ctrl)
Adhesion=netAnalysis_computeCentrality(Adhesion)


object.list <- list(Ctrl = Ctrl, NG = NG , Adhesion = Adhesion)

cellchat <- mergeCellChat(object.list, add.names = names(object.list), cell.prefix = TRUE)


gg1 <- rankNet(cellchat, mode = "comparison", stacked = T, do.stat = TRUE,comparison = c(1,2,3))
gg2 <- rankNet(cellchat, mode = "comparison", stacked = F, do.stat = TRUE,comparison = c(1,2,3))
gg1 + gg2


gg1 <- netVisual_diffInteraction(cellchat, weight.scale = T, measure = "weight",comparison = c(1,2))
gg2 <- netVisual_diffInteraction(cellchat, weight.scale = T, measure = "weight",comparison = c(3,2))


gg1 <- netVisual_heatmap(cellchat,measure = "weight",comparison = c(1,2))
gg2 <- netVisual_heatmap(cellchat, measure = "weight",comparison = c(3,2))
gg1 + gg2

gg1 <- netVisual_bubble(cellchat, comparison = c(1, 2), max.dataset = 2, title.name = "Increased signaling in LS", angle.x = 45, remove.isolate = T)
gg2 <- netVisual_bubble(cellchat,comparison = c(3, 2), max.dataset = 3, title.name = "Decreased signaling in LS", angle.x = 45, remove.isolate = T)
gg1 + gg2




#查看空转数据结果----
BiocManager::install("magick")
library(magick)

data_dir <- "/data/DATAprocess/chentw/NG/S1/" 
spatial_data <- Load10X_Spatial(data.dir = data_dir)

SpatialFeaturePlot(spatial_data, 
                   features = NULL,  # 仅展示H&E图像
                   pt.size.factor = 0,  # 点的大小
                   alpha = c(0, 1),  # 透明度
                   image.alpha = 1)+theme(legend.position = "none")
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/空转数据/pdf/S1_152807.pdf",
       width = 6.7,height = 5)





data_dir <- "/data/DATAprocess/chentw/NG/" 
spatial_data <- Load10X_Spatial(data.dir = data_dir)

SpatialFeaturePlot(spatial_data, 
                   features = NULL,  # 仅展示H&E图像
                   pt.size.factor = 0,  # 点的大小
                   alpha = c(0, 1),  # 透明度
                   image.alpha = 1)+theme(legend.position = "none")
ggsave("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/空转数据/pdf/S1_152807.pdf",
       width = 6.7,height = 5)




#重新绘制基因featureplot进行一个美化----
library(Seurat)
library(MASS)
library(ggplot2)
library(spatstat)





S1=readRDS("/data/DATAprocess/chentw/zigongneimo/zigongneimo2/空转数据/rds/152807.rds")

DimPlot(S1,group.by = "Spatial_snn_res.0.3")
# 提取目标细胞（以某一细胞类型为例）
target_cells <- WhichCells(S1, idents = "1")
spatial_coords <- GetTissueCoordinates(S1)[target_cells, ]

# 计算细胞密度
spatial_ppp <- ppp(spatial_coords$x, spatial_coords$y, 
                   xrange = range(spatial_coords$x), 
                   yrange = range(spatial_coords$y))
cell_density <- density(spatial_ppp, sigma = 0.1)

# 绘制密度图
density_df <- as.data.frame(cell_density)
ggplot(density_df, aes(x = x, y = y, fill = value)) +
  geom_tile() +
  scale_fill_viridis_c(name = "Density") +
  theme_minimal() +
  labs(title = "Cell Density of T Cells", x = "X Coordinate", y = "Y Coordinate")


SpatialFeaturePlot(P1, features = Macr_gene)


